{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import re"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "file_train_text = \"../../data/training_text/training_text\"\n",
    "file_train_variant = \"../../data/training_variants/training_variants\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ragou/.local/lib/python3.8/site-packages/pandas/util/_decorators.py:311: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n",
      "  return func(*args, **kwargs)\n"
     ]
    }
   ],
   "source": [
    "train_text = pd.read_csv(file_train_text, sep = '\\|\\|')\n",
    "train_text.index.name = \"ID\"\n",
    "train_text.columns = [\"Text\"]\n",
    "\n",
    "train_variant = pd.read_csv(file_train_variant)\n",
    "train_variant.set_index(\"ID\",inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Y371H</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CBL</td>\n",
       "      <td>K382E</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>CBL</td>\n",
       "      <td>C381A</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>CBL</td>\n",
       "      <td>C396R</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Variation  Class\n",
       "ID                      \n",
       "8   CBL     Y371H      4\n",
       "11  CBL     K382E      4\n",
       "13  CBL     C381A      4\n",
       "21  CBL     C396R      4"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_variant[train_variant[\"Variation\"].isin([\"C396R\", \"H398Q\", \"Y371H\", \"K382E\", \"C381A\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Variation  Class\n",
       "ID                      \n",
       "2   CBL     Q249E      2"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_variant[train_variant[\"Variation\"] == \"Q249E\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 Text\n",
       "ID                                                   \n",
       "6   Oncogenic mutations in the monomeric Casitas B..."
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_text.loc[[6],]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_count = train_variant[\"Gene\"].value_counts().loc[lambda x : x > 100].to_dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_class = train_variant[train_variant[\"Gene\"].isin(gene_count.keys())]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA4EAAAFNCAYAAAC+H2oqAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/MnkTPAAAACXBIWXMAAAsTAAALEwEAmpwYAAAcvElEQVR4nO3dfbBlVXkn4N8rDVFQAaFDoCFCRtQQJyr2KEqijqgRv3AcZYxRiWOKpEqNikkkJhUdJ6ZMYuJn4hQBFSZGRfADE2J0xK9kIrFBogJmbAkKLR8tIirGCPrOH2e3XtvGvug9Z/e9+3mqTt29115nn/fs7qru311rr13dHQAAAKbhNmMXAAAAwOIIgQAAABMiBAIAAEyIEAgAADAhQiAAAMCECIEAAAATIgQCsGpU1Yur6i9H/PwPVtWvDNu/VFXvXcFzX1xVDx62V/R7VtULq+rUlTofAKubEAjALqWqnlxVm6rqa1V1VVX9bVX93Nh1ba+739TdD99Zv6p6Y1X9/jLO9zPd/cEfta6qenBVXbnduf+gu3/lRz03AGuDEAjALqOqTkryyiR/kOSAJD+Z5M+THDdiWXNVVevGrgGAaRECAdglVNXeSV6S5Jnd/fbuvrG7b+rud3f3b97Ce95WVVdX1Q1V9eGq+pklxx5ZVZdU1VeraktV/cbQvn9V/XVVfbmqvlRVH6mqHf57WFUPq6pPD+d/bZJacuyXq+rvh+2qqldU1bVV9ZWq+mRV3aOqTkzyS0l+axjZfPfQ//KqekFVfSLJjVW1bmh76JKPv21VvXWo/8KquueSz+6qusuS/TdW1e9X1V5J/jbJQcPnfa2qDtp+emlVPXaYfvrlYYrrTy85dnlV/UZVfWL43m+tqtsu448QgFVCCARgV3H/JLdN8o5b8Z6/TXJ4kh9PcmGSNy05dlqSX+3uOyS5R5LzhvbnJ7kyyfrMRhtfmKS3P3FV7Z/k7Ul+N8n+ST6b5OhbqOPhSR6Y5K5J9k5yfJLruvuUoaY/6u7bd/djlrznF5M8Ksk+3X3zDs55XJK3JblTkr9K8s6q2v0Wr0SS7r4xybFJvjB83u27+wvbfa+7JnlzkucO1+DcJO+uqj2WdDs+ySOSHJbkZ5P88g/6XABWFyEQgF3Ffkm+eAuBaIe6+/Xd/dXu/vckL05yz2FEMUluSnJEVd2xu6/v7guXtB+Y5M7DSONHuvv7QmCSRya5uLvP6u6bMpumevUtlHJTkjskuXuS6u5Lu/uqnZT/6u6+orv/7RaOX7Dks/80s4B81E7OuRz/LcnfdPf7hnO/PMntkjxgu9q+0N1fSvLuJPdagc8FYBchBAKwq7guyf7LvUeuqnarqpdV1Wer6itJLh8O7T/8/K+ZBbnPVdWHqur+Q/sfJ9mc5L1VdVlVnXwLH3FQkiu27QxB8Yoddezu85K8NsmfJbm2qk6pqjvu5Cvs8Fw7Ot7d385s9PKgnbxnOQ5K8rntzn1Fkg1L+iwNu19PcvsV+FwAdhFCIAC7in9M8u9JHrfM/k/ObMrkQzObgnno0F5J0t0f6+7jMpsq+s4kZw7tX+3u53f3TyV5bJKTquqYHZz/qiSHbNupqlq6v73ufnV33yfJEZlNC912H+OORhl/UPs2Sz/7NkkOTrJtaufXk+y5pO9P3IrzfiHJnZece9v32rKT9wGwRgiBAOwSuvuGJL+X5M+q6nFVtWdV7V5Vx1bVH+3gLXfILDRel1kg+oNtB6pqj+E5fnsPUx6/kuTbw7FHV9VdhvBzQ5JvbTu2nb9J8jNV9fhhdPLX871h6zuq6j9V1f2Ge/ZuTPKNJee8JslP3crLkST3WfLZzx2+60eHYxclefIwGvqIJA9a8r5rkuy3ZFrs9s5M8qiqOmao9/nDuf/vD1EjAKuQEAjALqO7/yTJSZktxrI1s2mKz8psJG97Z2Q2rXFLkkvy3YC0zVOTXD5MFf21zFbpTGYLyfyfJF/LbPTxz7v7Azuo5YtJnpjkZZkFzcOT/MMtlH7HJH+R5Pqhpusym3aazBaoOWJYiXNH3+OWvCuz+/euH77L44dAmyTPSfKYJF8evtd3ztvdn85s4ZfLhs/8nimk3f0vSZ6S5DVJvjic5zHd/c1bURsAq1jt+F54AAAA1iIjgQAAABMytxBYVa8fHpr7qSVtd6qq91XVZ4af+w7tVVWvrqrNw8Npj1zynhOG/p+pqhPmVS8AAMAUzHMk8I2ZPWh2qZOTvL+7D0/y/mE/mT3Y9vDhdWKS1yWz0JjkRUnul+S+SV60LTgCAABw680tBHb3h5N8abvm45KcPmyfnu8uA35ckjN65qNJ9qmqA5P8QpL3dfeXuvv6JO/L9wdLAAAAlmnR9wQe0N1XDdtXJzlg2N6Q731o7pVD2y21AwAA8ENYN9YHd3dX1YotTVpVJ2Y2lTR77bXXfe5+97uv1KkBAABWlQsuuOCL3b1+R8cWHQKvqaoDu/uqYbrntUP7liSHLOl38NC2JcmDt2v/4I5O3N2nJDklSTZu3NibNm1a2coBAABWiar63C0dW/R00HOSbFvh84TMHoS7rf1pwyqhRyW5YZg2+ndJHl5V+w4Lwjx8aAMAAOCHMLeRwKp6c2ajePtX1ZWZrfL5siRnVtUzknwuyfFD93OTPDLJ5iRfT/L0JOnuL1XV/0zysaHfS7p7+8VmAAAAWKbqXrHb8nYZpoMCAABTVlUXdPfGHR1b9HRQAAAARiQEAgAATIgQCAAAMCFCIAAAwIQIgQAAABMiBAIAAEyIEAgAADAhQiAAAMCErBu7AAAAdu7Sl543dgmrwk//zkPGLgF2eUYCAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJmSUEFhVz6uqi6vqU1X15qq6bVUdVlXnV9XmqnprVe0x9P2xYX/zcPzQMWoGAABYCxYeAqtqQ5JfT7Kxu++RZLckT0ryh0le0d13SXJ9kmcMb3lGkuuH9lcM/QAAAPghjDUddF2S21XVuiR7JrkqyUOSnDUcPz3J44bt44b9DMePqapaXKkAAABrx8JDYHdvSfLyJJ/PLPzdkOSCJF/u7puHblcm2TBsb0hyxfDem4f++y2yZgAAgLVijOmg+2Y2undYkoOS7JXkEStw3hOralNVbdq6deuPejoAAIA1aYzpoA9N8q/dvbW7b0ry9iRHJ9lnmB6aJAcn2TJsb0lySJIMx/dOct32J+3uU7p7Y3dvXL9+/by/AwAAwKo0Rgj8fJKjqmrP4d6+Y5JckuQDSZ4w9DkhybuG7XOG/QzHz+vuXmC9AAAAa8YY9wSen9kCLxcm+eRQwylJXpDkpKranNk9f6cNbzktyX5D+0lJTl50zQAAAGvFup13WXnd/aIkL9qu+bIk991B328keeIi6gIAAFjrxnpEBAAAACMQAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAlZN3YBAGvNhx74oLFLWBUe9OEPjV0CAEySkUAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmJBRQmBV7VNVZ1XVp6vq0qq6f1XdqareV1WfGX7uO/Stqnp1VW2uqk9U1ZFj1AwAALAWjDUS+Kok7+nuuye5Z5JLk5yc5P3dfXiS9w/7SXJsksOH14lJXrf4cgEAANaGhYfAqto7yQOTnJYk3f3N7v5ykuOSnD50Oz3J44bt45Kc0TMfTbJPVR240KIBAADWiDFGAg9LsjXJG6rq41V1alXtleSA7r5q6HN1kgOG7Q1Jrljy/iuHNgAAAG6lMULguiRHJnldd987yY357tTPJEl3d5K+NSetqhOralNVbdq6deuKFQsAALCWjBECr0xyZXefP+yflVkovGbbNM/h57XD8S1JDlny/oOHtu/R3ad098bu3rh+/fq5FQ8AALCaLTwEdvfVSa6oqrsNTcckuSTJOUlOGNpOSPKuYfucJE8bVgk9KskNS6aNAgAAcCusG+lzn53kTVW1R5LLkjw9s0B6ZlU9I8nnkhw/9D03ySOTbE7y9aEvAAAAP4RRQmB3X5Rk4w4OHbODvp3kmfOuCQAAYArGek4gAAAAIxACAQAAJkQIBAAAmBAhEAAAYELGWh10l3Cf3zxj7BJWhQv++GljlwAAAKwQI4EAAAATIgQCAABMiBAIAAAwIUIgAADAhAiBAAAAE7KsEFhV719OGwAAALu2H/iIiKq6bZI9k+xfVfsmqeHQHZNsmHNtAAAArLCdPSfwV5M8N8lBSS7Id0PgV5K8dn5lAQAAMA8/MAR296uSvKqqnt3dr1lQTQAAAMzJzkYCkyTd/ZqqekCSQ5e+p7vPmFNdAAAAzMGyQmBV/e8k/yHJRUm+NTR3EiEQAABgFVlWCEyyMckR3d3zLAYAAID5Wu5zAj+V5CfmWQgAAADzt9yRwP2TXFJV/5Tk37c1dvdj51IVAAAAc7HcEPjieRYBAADAYix3ddAPzbsQAAAA5m+5q4N+NbPVQJNkjyS7J7mxu+84r8IAAABYecsdCbzDtu2qqiTHJTlqXkUBAAAwH8tdHfQ7euadSX5h5csBAABgnpY7HfTxS3Zvk9lzA78xl4oAAACYm+WuDvqYJds3J7k8symhAAAArCLLvSfw6fMuBAAAgPlb1j2BVXVwVb2jqq4dXmdX1cHzLg4AAICVtdyFYd6Q5JwkBw2vdw9tAAAArCLLDYHru/sN3X3z8HpjkvVzrAsAAIA5WG4IvK6qnlJVuw2vpyS5bp6FAQAAsPKWGwL/e5Ljk1yd5KokT0jyy3OqCQAAgDlZ7iMiXpLkhO6+Pkmq6k5JXp5ZOAQAAGCVWO5I4M9uC4BJ0t1fSnLv+ZQEAADAvCw3BN6mqvbdtjOMBC53FBEAAIBdxHKD3J8k+ceqetuw/8QkL51PSaxln3/Jfxy7hFXhJ3/vk2OXAADAGrWsENjdZ1TVpiQPGZoe392XzK8sAAAA5mHZUzqH0Cf4AQAArGLLvScQAACANUAIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBCRguBVbVbVX28qv562D+sqs6vqs1V9daq2mNo/7Fhf/Nw/NCxagYAAFjtxhwJfE6SS5fs/2GSV3T3XZJcn+QZQ/szklw/tL9i6AcAAMAPYZQQWFUHJ3lUklOH/UrykCRnDV1OT/K4Yfu4YT/D8WOG/gAAANxKY40EvjLJbyX59rC/X5Ivd/fNw/6VSTYM2xuSXJEkw/Ebhv4AAADcSgsPgVX16CTXdvcFK3zeE6tqU1Vt2rp160qeGgAAYM0YYyTw6CSPrarLk7wls2mgr0qyT1WtG/ocnGTLsL0lySFJMhzfO8l125+0u0/p7o3dvXH9+vXz/QYAAACr1MJDYHf/dncf3N2HJnlSkvO6+5eSfCDJE4ZuJyR517B9zrCf4fh53d0LLBkAAGDN2JWeE/iCJCdV1ebM7vk7bWg/Lcl+Q/tJSU4eqT4AAIBVb93Ou8xPd38wyQeH7cuS3HcHfb6R5IkLLQwAAGCN2pVGAgEAAJgzIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYkHVjFwAArD4vfcoTxi5hVfidvzxr7BIAvo+RQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQjwsHoBV77XPf/fYJawKz/qTx4xdAgC7ACOBAAAAEyIEAgAATIgQCAAAMCFCIAAAwIQIgQAAABMiBAIAAEyIEAgAADAhCw+BVXVIVX2gqi6pqour6jlD+52q6n1V9Znh575De1XVq6tqc1V9oqqOXHTNAAAAa8UYI4E3J3l+dx+R5Kgkz6yqI5KcnOT93X14kvcP+0lybJLDh9eJSV63+JIBAADWhoWHwO6+qrsvHLa/muTSJBuSHJfk9KHb6UkeN2wfl+SMnvlokn2q6sDFVg0AALA2jHpPYFUdmuTeSc5PckB3XzUcujrJAcP2hiRXLHnblUPb9uc6sao2VdWmrVu3zq9oAACAVWy0EFhVt09ydpLndvdXlh7r7k7St+Z83X1Kd2/s7o3r169fwUoBAADWjlFCYFXtnlkAfFN3v31ovmbbNM/h57VD+5Ykhyx5+8FDGwAAALfSGKuDVpLTklza3X+65NA5SU4Ytk9I8q4l7U8bVgk9KskNS6aNAgAAcCusG+Ezj07y1CSfrKqLhrYXJnlZkjOr6hlJPpfk+OHYuUkemWRzkq8nefpCqwUAAFhDFh4Cu/vvk9QtHD5mB/07yTPnWhQAAMBEjLo6KAAAAIslBAIAAEyIEAgAADAhQiAAAMCECIEAAAATIgQCAABMiBAIAAAwIUIgAADAhAiBAAAAEyIEAgAATIgQCAAAMCFCIAAAwIQIgQAAABMiBAIAAEyIEAgAADAhQiAAAMCECIEAAAATIgQCAABMiBAIAAAwIUIgAADAhAiBAAAAEyIEAgAATIgQCAAAMCFCIAAAwIQIgQAAABOybuwCgPk6+jVHj13CqvAPz/6HsUsAAFgII4EAAAATIgQCAABMiBAIAAAwIUIgAADAhAiBAAAAEyIEAgAATIgQCAAAMCFCIAAAwIQIgQAAABMiBAIAAEyIEAgAADAhQiAAAMCECIEAAAATsm7sAgAAAJLkzLfdd+wSVoXjn/hPP9L7jQQCAABMiBAIAAAwIUIgAADAhAiBAAAAE2JhGAAA2IEXv/jFY5ewKrhOq8+qGQmsqkdU1b9U1eaqOnnsegAAAFajVRECq2q3JH+W5NgkRyT5xao6YtyqAAAAVp9VEQKT3DfJ5u6+rLu/meQtSY4buSYAAIBVZ7WEwA1Jrliyf+XQBgAAwK1Q3T12DTtVVU9I8oju/pVh/6lJ7tfdz1rS58QkJw67d0vyLwsvdGXsn+SLYxcxMa754rnmi+eaL55rvniu+eK55ovnmi/ear3md+7u9Ts6sFpWB92S5JAl+wcPbd/R3ackOWWRRc1DVW3q7o1j1zElrvniueaL55ovnmu+eK754rnmi+eaL95avOarZTrox5IcXlWHVdUeSZ6U5JyRawIAAFh1VsVIYHffXFXPSvJ3SXZL8vruvnjksgAAAFadVRECk6S7z01y7th1LMCqn9K6Crnmi+eaL55rvniu+eK55ovnmi+ea754a+6ar4qFYQAAAFgZq+WeQAAAAFaAELiLqKrXV9W1VfWpsWuZgqo6pKo+UFWXVNXFVfWcsWta66rqtlX1T1X1z8M1/x9j1zQVVbVbVX28qv567FqmoKour6pPVtVFVbVp7HqmoKr2qaqzqurTVXVpVd1/7JrWsqq62/D3e9vrK1X13LHrWuuq6nnDv5+fqqo3V9Vtx65prauq5wzX++K19nfcdNBdRFU9MMnXkpzR3fcYu561rqoOTHJgd19YVXdIckGSx3X3JSOXtmZVVSXZq7u/VlW7J/n7JM/p7o+OXNqaV1UnJdmY5I7d/eix61nrquryJBu7ezU+U2pVqqrTk3yku08dVhHfs7u/PHJZk1BVu2X22K77dffnxq5nraqqDZn9u3lEd/9bVZ2Z5NzufuO4la1dVXWPJG9Jct8k30zyniS/1t2bRy1shRgJ3EV094eTfGnsOqaiu6/q7guH7a8muTTJhnGrWtt65mvD7u7Dy2+h5qyqDk7yqCSnjl0LzENV7Z3kgUlOS5Lu/qYAuFDHJPmsALgQ65LcrqrWJdkzyRdGrmet++kk53f317v75iQfSvL4kWtaMUIgk1dVhya5d5LzRy5lzRumJV6U5Nok7+tu13z+Xpnkt5J8e+Q6pqSTvLeqLqiqE8cuZgIOS7I1yRuGac+nVtVeYxc1IU9K8uaxi1jruntLkpcn+XySq5Lc0N3vHbeqNe9TSX6+qvarqj2TPDLJISPXtGKEQCatqm6f5Owkz+3ur4xdz1rX3d/q7nslOTjJfYepFsxJVT06ybXdfcHYtUzMz3X3kUmOTfLMYbo/87MuyZFJXtfd905yY5KTxy1pGoapt49N8raxa1nrqmrfJMdl9kuPg5LsVVVPGbeqta27L03yh0nem9lU0IuSfGvMmlaSEMhkDfelnZ3kTd399rHrmZJhqtYHkjxi5FLWuqOTPHa4R+0tSR5SVX85bklr3/Ab+3T3tUnekdn9JMzPlUmuXDKz4KzMQiHzd2ySC7v7mrELmYCHJvnX7t7a3TcleXuSB4xc05rX3ad19326+4FJrk/y/8auaaUIgUzSsEjJaUku7e4/HbueKaiq9VW1z7B9uyQPS/LpUYta47r7t7v74O4+NLMpW+d1t98cz1FV7TUsNpVhSuLDM5tSxJx099VJrqiquw1NxySxyNdi/GJMBV2Uzyc5qqr2HP4Pc0xm6xkwR1X148PPn8zsfsC/GreilbNu7AKYqao3J3lwkv2r6sokL+ru08atak07OslTk3xyuEctSV7Y3eeOV9Kad2CS04eV5G6T5Mzu9sgC1poDkrxj9n+0rEvyV939nnFLmoRnJ3nTMD3xsiRPH7meNW/4JcfDkvzq2LVMQXefX1VnJbkwyc1JPp7klHGrmoSzq2q/JDcleeZaWnTKIyIAAAAmxHRQAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgFgGarqJ6rqLVX12aq6oKrOraq7VpXnAAKwqnhOIADsxPBw5nckOb27nzS03TOz5wICwKpiJBAAdu4/J7mpu//Xtobu/uckV2zbr6pDq+ojVXXh8HrA0H5gVX24qi6qqk9V1c9X1W5V9cZh/5NV9bzFfyUApspIIADs3D2SXLCTPtcmeVh3f6OqDk/y5iQbkzw5yd9190urarckeya5V5IN3X2PJKmqfeZVOABsTwgEgJWxe5LXVtW9knwryV2H9o8leX1V7Z7knd19UVVdluSnquo1Sf4myXvHKBiAaTIdFAB27uIk99lJn+cluSbJPTMbAdwjSbr7w0kemGRLkjdW1dO6+/qh3weT/FqSU+dTNgB8PyEQAHbuvCQ/VlUnbmuoqp9NcsiSPnsnuaq7v53kqUl2G/rdOck13f0XmYW9I6tq/yS36e6zk/xukiMX8zUAwHRQANip7u6q+i9JXllVL0jyjSSXJ3nukm5/nuTsqnpakvckuXFof3CS36yqm5J8LcnTkmxI8oaq2vbL2N+e93cAgG2qu8euAQAAgAUxHRQAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmJD/DwTZnzDiG3ukAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 1080x360 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure(figsize=(15,5))\n",
    "sns.countplot(x = train_variant[\"Class\"])\n",
    "plt.title(\"Class distribution\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA3sAAAHwCAYAAAAfJXbRAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/MnkTPAAAACXBIWXMAAAsTAAALEwEAmpwYAAAjSElEQVR4nO3de7ztdV3n8fcnwPsNhxMikMeULKnEPJqXMs2Z8TIVWkr6cJRxVGpCs8lJaXSUUiceo6mTlQ6NJjyGvOUlbRjvd8fUAyEK6EgqAnI5miJegsDP/LF+G5fbffbZ53DW3vt8eT4fj/3Ya/1u67sWi8V68bvs6u4AAAAwlh/a6AEAAACw94k9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9ANZdVW2tqq6q/af7/6eqjt1L2/75qvrs3P0vVtW/3BvbnrZ3TlU9YG9tDwAWRewBDKyqHl1VH6uqb1XV5dPt36qq2uixzevuh3b3KbtabgrEO+9iWx/q7rvsjXFV1aur6vnLtn9kd79/b2x/b6iqp1TV9qq6qqpevcL8B1XVZ6rq21X1vqq6w9y8G1fVq6rqG1V1aVX97roOfheq6t9V1Yc3ehwA+yqxBzCoqnp6kv+e5IVJbpfk4CS/meR+SW60gUNbmKU9hfuiqjp4D1f9cpLnJ3nVCts8KMmbkvyXJLdNsj3J6+YWOTHJEUnukOSBSZ5RVQ/Zw3EAsMmIPYABVdWtk/xhkt/q7r/u7it75u+7+7HdfdW03I2r6kVV9aWquqyqXlFVN53mPaCqLqqqp097BS+pqifMPcZO111hPPtNy36lqj6f5N8sm//+qnrSdPvOVfWBqrpiWv510/QPTot/sqq+WVW/PjfGZ1bVpUn+cmnasiHcs6rOraqvVdVfVtVNpm3+wJ6jpb2HVXVcksdmFkDfrKq3TfOvOyx0eg1eWlVfnn5eWlU3Xsvrt4ITpzH+XlXdbpXlvk93v6m735LkqyvM/tUk53T3G7r7nzKLu7tV1Y9P849N8rzu/lp3n5fkL5L8u509VlU9uarOq6orp7H+zDT9J6Z/hl+fDnP9lbl1rvtnO93/vtd8er1/s6o+N63/ZzXzE0lekeQ+0+v/9bW+JgDMiD2AMd0nyY2T/M0uljspyY8lOSrJnZMcmuQ5c/Nvl+TW0/QnJvmzqjpwjevOe3KSX0py9yTbkjxylTE9L8k7kxyY5LAkL0uS7r7/NP9u3X2L7l7aQ3W7zPZa3SHJcTvZ5mOTPDjJnaYxP3uVx8/0eCcnOS3Jf5se75dXWOxZSe6d2WtwtyT3Wrbt1V6/5Y5P8ttJfjrJZ6vqrVX1iKo6YFdjXcWRST4595y+leQfkhw5jeOQ+fnT7SNX2lBVPSqzWHx8klsl+ZUkX53G97bM/pn9cJKnJjmtqnbnUNpfSnLPzJ77MUkePMXnbyb56PT632Y3tgdAxB7AqA5K8pXuvmZpQlX932nPyXeq6v5VVZnF0X/s7n/s7iuT/Nckj57bzj8n+cPu/ufuPj3JN5PcZY3rzjsmyUu7+8Lu/sckf7TK2P85s3C7fXf/U3fv6pyt7yZ5bndf1d3f2ckyfzr32C9I8phdbHOtHpvZ63N5d+9I8gdJHjc3f8XXb6UNdfd3u/vd3f24zCL3LUn+Y5KLq+p5ezi+WyS5Ytm0K5LccpqXZfOX5q3kSZmF7yemvcTnd/cFmcXuLZKc1N1Xd/d7k/xtdu81Pqm7v97dX0ryvsziGYDrSewBjOmrSQ6aP4etu+877R35amaf/1uS3CzJGVMEfj3J26fp121nPhiTfDuzL/ZrWXfe7ZNcOHf/glXG/owkleTj0yGB/34Xz3XHdIjiapY/9u13sfxa3T7f/1yWb3tnr9+qpng+O8lZSQ7ITgJxDb6Z2V64ebdKcuU0L8vmL81byeGZ7RVc7vZJLuzu785NuyCzvZlrdenc7TW9RgDsmtgDGNNHk1yV5OhVlvlKku8kObK7bzP93Lq71/JFe3fXvSSzWFjyIzvbcHdf2t1P7u7bJ/mNJH9eq1+Bs9cw3uWP/eXp9rcyi9YkyQrnyu1q21/ObC/kStvebVV1WFWdUFXnJnltkh2ZHbZ6zB5u8pzMDi9d2v7NMzuU9Zzu/lpm/1zuNrf83aZ1VnLhtO5yX05yeFXNf6f4kSQXT7e/7zXO7NDWtVrLP1sAdkLsAQyou7+e2SGFf15Vj6yqW1bVD1XVUUluPi3z3cwuyPGSqvrhJKmqQ6vqwWvY/u6u+/okvz3FzIFJTtjZtqvqUVV12HT3a5l94V/aa3RZkh/d1fhWcPz02LfN7Dy7pfP9PpnZ+WtHTRdtOXHZert6vNckeXZVbanZlS+fk+R/7cH4UlUnZhZad8nsXLUjuvt506GNq623/zT2/ZLsV1U3mduj++YkP1lVvzYt85wkZ3f3Z6b5p07jP3C6aMuTk7x6Jw/1P5P8p6q6x3QBlTvX7M84fCyzvXHPqKoDavY3CH85s1hNZnsnf7WqbjZF+xN342W5LMlhVTXk1WMBFk3sAQyqu/9bkt/N7LDIy6af/5HkmUn+77TYM5Ocn+TvquobSd6dtR8yuDvr/kWSd2QWV2dm9ucAduaeST5WVd9M8tYkT+vuz0/zTkxyynTo6O7s7fqrzC4g8vnMDkV8fpJ09//L7Kql707yuSTLzw98ZZK7To/3lhW2+/zM/pzB2Uk+NT2356+w3Fq8JbPzFJ/Q3R/s7rXu1Xp2ZntZT0jyb6fbz06S6TzCX8vsPMWvJfnZfP95lc/N7PW4IMkHkrywu9++0oN09xum7fxVZod6viXJbbv76szi7qGZ7fH98ySPnwvKlyS5OrP33ymZXfRmrd6bWQBfWlVf2Y31AEhSa/9vCQAAAPsKe/YAAAAGJPYAAAAGJPYAAAAGJPYAAAAGJPYAAAAGtP+uF9m8DjrooN66detGDwMAAGBDnHHGGV/p7i0rzdunY2/r1q3Zvn37Rg8DAABgQ1TVBTub5zBOAACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAe2/0QPYaPf4vVM3eghsQme88PEbPQQAALhe7NkDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAY0MJir6oOr6r3VdW5VXVOVT1tmn5iVV1cVWdNPw+bW+f3q+r8qvpsVT14UWMDAAAY3f4L3PY1SZ7e3WdW1S2TnFFV75rmvaS7XzS/cFXdNcmjkxyZ5PZJ3l1VP9bd1y5wjAAAAENa2J697r6ku8+cbl+Z5Lwkh66yytFJXtvdV3X3F5Kcn+ReixofAADAyNblnL2q2prk7kk+Nk16SlWdXVWvqqoDp2mHJrlwbrWLsnocAgAAsBMLj72qukWSNyb5ne7+RpKXJ7lTkqOSXJLkj3dze8dV1faq2r5jx469PVwAAIAhLDT2quqAzELvtO5+U5J092XdfW13fzfJX+R7h2penOTwudUPm6Z9n+4+ubu3dfe2LVu2LHL4AAAA+6xFXo2zkrwyyXnd/eK56YfMLfaIJJ+ebr81yaOr6sZVdcckRyT5+KLGBwAAMLJFXo3zfkkel+RTVXXWNO0/J3lMVR2VpJN8MclvJEl3n1NVr09ybmZX8jzelTgBAAD2zMJir7s/nKRWmHX6Kuu8IMkLFjUmAACAG4p1uRonAAAA60vsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADGhhsVdVh1fV+6rq3Ko6p6qeNk2/bVW9q6o+N/0+cJpeVfUnVXV+VZ1dVT+zqLEBAACMbpF79q5J8vTuvmuSeyc5vqrumuSEJO/p7iOSvGe6nyQPTXLE9HNckpcvcGwAAABDW1jsdfcl3X3mdPvKJOclOTTJ0UlOmRY7JcnDp9tHJzm1Z/4uyW2q6pBFjQ8AAGBk63LOXlVtTXL3JB9LcnB3XzLNujTJwdPtQ5NcOLfaRdM0AAAAdtPCY6+qbpHkjUl+p7u/MT+vuztJ7+b2jquq7VW1fceOHXtxpAAAAONYaOxV1QGZhd5p3f2mafJlS4dnTr8vn6ZfnOTwudUPm6Z9n+4+ubu3dfe2LVu2LG7wAAAA+7BFXo2zkrwyyXnd/eK5WW9Ncux0+9gkfzM3/fHTVTnvneSKucM9AQAA2A37L3Db90vyuCSfqqqzpmn/OclJSV5fVU9MckGSY6Z5pyd5WJLzk3w7yRMWODYAAIChLSz2uvvDSWonsx+0wvKd5PhFjQcAAOCGZF2uxgkAAMD6EnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAAD2n+jBwCs7Et/+FMbPQQ2oR95zqc2eggAwD7Cnj0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABLSz2qupVVXV5VX16btqJVXVxVZ01/Txsbt7vV9X5VfXZqnrwosYFAABwQ7DIPXuvTvKQFaa/pLuPmn5OT5KqumuSRyc5clrnz6tqvwWODQAAYGgLi73u/mCSf1zj4kcneW13X9XdX0hyfpJ7LWpsAAAAo9uIc/aeUlVnT4d5HjhNOzTJhXPLXDRN+wFVdVxVba+q7Tt27Fj0WAEAAPZJ6x17L09ypyRHJbkkyR/v7ga6++Tu3tbd27Zs2bKXhwcAADCGdY297r6su6/t7u8m+Yt871DNi5McPrfoYdM0AAAA9sBux15VHVhVP70nD1ZVh8zdfUSSpSt1vjXJo6vqxlV1xyRHJPn4njwGAAAAyf5rWaiq3p/kV6blz0hyeVV9pLt/d5V1XpPkAUkOqqqLkjw3yQOq6qgkneSLSX4jSbr7nKp6fZJzk1yT5PjuvnbPnhIAAABrir0kt+7ub1TVk5Kc2t3PraqzV1uhux+zwuRXrrL8C5K8YI3jAQAAYBVrPYxz/+kQzGOS/O0CxwMAAMBesNbY+4Mk70hyfnd/oqp+NMnnFjcsAAAAro+1HsZ5SXdfd1GW7v58Vb14QWMCAADgelrrnr2XrXEaAAAAm8Cqe/aq6j5J7ptkS1XNX3nzVkn2W+TAAAAA2HO7OozzRkluMS13y7np30jyyEUNCgAAgOtn1djr7g8k+UBVvbq7L1inMQEAAHA9rfUCLTeuqpOTbJ1fp7t/cRGDAgAA4PpZa+y9IckrkvzPJNcubjgAAADsDWuNvWu6++ULHQkA+4z7vex+Gz0ENqGPPPUjGz0EAOas9U8vvK2qfquqDqmq2y79LHRkAAAA7LG17tk7dvr9e3PTOsmP7t3hAAAAsDesKfa6+46LHggAAAB7z5pir6oev9L07j517w4HAACAvWGth3Hec+72TZI8KMmZScQeAADAJrTWwzifOn+/qm6T5LWLGBAAAADX31qvxrnct5I4jw8AAGCTWus5e2/L7OqbSbJfkp9I8vpFDQoAAIDrZ63n7L1o7vY1SS7o7osWMB4AAAD2gjUdxtndH0jymSS3THJgkqsXOSgAAACunzXFXlUdk+TjSR6V5JgkH6uqRy5yYAAAAOy5tR7G+awk9+zuy5OkqrYkeXeSv17UwAAAANhza70a5w8thd7kq7uxLgAAAOtsrXv23l5V70jymun+ryc5fTFDAgAA4PpaNfaq6s5JDu7u36uqX03yc9OsjyY5bdGDAwAAYM/sas/eS5P8fpJ095uSvClJquqnpnm/vMCxAQAAsId2dd7dwd39qeUTp2lbFzIiAAAArrddxd5tVpl30704DgAAAPaiXcXe9qp68vKJVfWkJGcsZkgAAABcX7s6Z+93kry5qh6b78XdtiQ3SvKIBY4LAACA62HV2Ovuy5Lct6oemOQnp8n/u7vfu/CRAQAAsMfW9Hf2uvt9Sd634LEAAACwl6z1j6oDAGx6H7j/L2z0ENiEfuGDH9joIcCG2NUFWgAAANgH2bMHAADr4E+f/raNHgKbzFP++JcXun179gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAa0sNirqldV1eVV9em5abetqndV1eem3wdO06uq/qSqzq+qs6vqZxY1LgAAgBuCRe7Ze3WShyybdkKS93T3EUneM91PkocmOWL6OS7Jyxc4LgAAgOEtLPa6+4NJ/nHZ5KOTnDLdPiXJw+emn9ozf5fkNlV1yKLGBgAAMLr1Pmfv4O6+ZLp9aZKDp9uHJrlwbrmLpmkAAADsgQ27QEt3d5Le3fWq6riq2l5V23fs2LGAkQEAAOz71jv2Lls6PHP6ffk0/eIkh88td9g07Qd098ndva27t23ZsmWhgwUAANhXrXfsvTXJsdPtY5P8zdz0x09X5bx3kivmDvcEAABgN+2/qA1X1WuSPCDJQVV1UZLnJjkpyeur6olJLkhyzLT46UkeluT8JN9O8oRFjQsAAOCGYGGx192P2cmsB62wbCc5flFjAQAAuKHZsAu0AAAAsDhiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYED7b8SDVtUXk1yZ5Nok13T3tqq6bZLXJdma5ItJjunur23E+AAAAPZ1G7ln74HdfVR3b5vun5DkPd19RJL3TPcBAADYA5vpMM6jk5wy3T4lycM3bigAAAD7to2KvU7yzqo6o6qOm6Yd3N2XTLcvTXLwSitW1XFVtb2qtu/YsWM9xgoAALDP2ZBz9pL8XHdfXFU/nORdVfWZ+Znd3VXVK63Y3ScnOTlJtm3btuIyAAAAN3Qbsmevuy+efl+e5M1J7pXksqo6JEmm35dvxNgAAABGsO6xV1U3r6pbLt1O8q+TfDrJW5McOy12bJK/We+xAQAAjGIjDuM8OMmbq2rp8f+qu99eVZ9I8vqqemKSC5IcswFjAwAAGMK6x153fz7J3VaY/tUkD1rv8QAAAIxoM/3pBQAAAPYSsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADCgTRd7VfWQqvpsVZ1fVSds9HgAAAD2RZsq9qpqvyR/luShSe6a5DFVddeNHRUAAMC+Z1PFXpJ7JTm/uz/f3VcneW2Sozd4TAAAAPuczRZ7hya5cO7+RdM0AAAAdkN190aP4TpV9cgkD+nuJ033H5fkZ7v7KXPLHJfkuOnuXZJ8dt0HOq6DknxlowcBK/DeZDPz/mSz8t5kM/P+3Hvu0N1bVpqx/3qPZBcuTnL43P3DpmnX6e6Tk5y8noO6oaiq7d29baPHAct5b7KZeX+yWXlvspl5f66PzXYY5yeSHFFVd6yqGyV5dJK3bvCYAAAA9jmbas9ed19TVU9J8o4k+yV5VXefs8HDAgAA2OdsqthLku4+PcnpGz2OGyiHx7JZeW+ymXl/sll5b7KZeX+ug011gRYAAAD2js12zh4AAAB7gdgbSFVdW1VnVdUnq+rMqrrvNH1rVX1nmnduVZ1aVQdM8w6oqpOq6nPTOh+tqofObfOoquqqesiyx3pVVV1eVZ9e32fJCKrqX0zvx7Oq6tKqunjufk+/P11Vb6iqm03rfLGqPjXN2z63redV1dnT9HdW1e037pmxr5v7HF36OWGavn9V/dfps3Jp3rNWWW9rVT2gqq6Y7n+mql60cc+MEcy9z5Y+Hw9d5bP0Rqu8n9+/7HN0W1W9f8OeGPuk9freWVWHV9X7pm2dU1VPW/9nu+/adOfscb18p7uPSpKqenCSP0ryC9O8f+juo6pqvyTvSnJMktOSPC/JIUl+sruvqqqD59ZJksck+fD0++1z01+d5E+TnLqwZ8OwuvurSY5Kkqo6Mck3u/tF0/1vzr2PT0vym0lePK36wO5e/jd5Xtjd/2Va/reTPGdaB/bEdZ+jyzw/ye2S/FR3/1NV3TLJ01dbr6q2JvlQd/9SVd00yd9X1Zu7+yOLGTo3AN9Z9vn463P3T8zcZ+k0bWfv5yT54ap6aHf/n4WOmJGt1/fOa5I8vbvPnD57z6iqd3X3uYt9emMQe+O6VZKvLZ/Y3ddW1ceTHDrtMXlykjt291XT/MuSvD5JqqqSPCrJv0ryoaq6SXf/07TcB6cvMrBIH0ry06st0N3fmLt78yRORGavmvus3Dr3GXhlkhPXuo3u/k5VnZXk0EWMkRukXX4+7sILkzwridhjb1jY987uviTJJdPyV1bVeZl9loq9NRB7Y7np9GXiJpn9X5NfXL5AVd0kyc8meVqSOyf50rIvy/Pum+QL3f0P0+Ed/ybJGxcwbvgBVbV/kofme/9nr5O8s6o6yf/o7pPnln1BkscnuSLJA9d7rAxl6XN0yR8lOS+zz8or17jeF7r7EfMzq+rAJEck+eBeHCs3UCt8Pu7MD7yfu/t10+2PJnlEVT0wyWrvbdiZdf/eOe1ouHuSj+2dpzA+5+yN5TvdfVR3/3iShyQ5dfq/JElyp+lfyMuSXNLdZ69he49J8trp9mun+7BoS//x2J7kS0leOU3/ue7+mcy+4BxfVfdfWqG7n9Xdh2d2iMhT1nm8jGXpc3Tp53XLF6iqJ0znolxYVYevsN586P18VX0yycVJ3tHdl67Hk2BYO/t83JldvZ+fn+TZCxgnNwzr+r2zqm6RWfz9zirByDJib1Dd/dEkByXZMk36h+m46jsluUdV/UqS85P8SFXdavn60zHWv5bkOVX1xSQvS/KQ6VhpWKT5LydP7e6rk6S7L55+X57kzUnutcK6p2X2voW9aemz8pZJ0t1/OX2eXpFkv12s+6HuvluSI5M8saqOWuRAGd6Kn497qrvfm+SmSe69d4bHDdWiv3dOF3h5Y5LTuvtNC346QxF7g6qqH8/sS8hX56dPF7c4Icnvd/e3M/u/gv+9qm40rbelqh6V5EFJzu7uw7t7a3ffIbN/yb7v0CRYD1V187kP/Jsn+ddJPj3dP2Ju0aOTfGb9R8jI5j4r/3Q6JGnpi8mNdmMbX0hyUpJnLmSQsOeen+QZGz0I9m2L/N457S18ZZLzuvvFYbeIvbHcdDq06Kwkr0tybHdfu8Jyb0lys6r6+cwO39iR5Nya/RmFv03yjcx2nb952XpvnKanql6T2fH+d6mqi6rqiQt4PrDk4CQfng6H+3iS/93dS+eqnFSzy5CfnVkEuiQz18d1n6PTz0nT9GdldoGAT1fV32d2cYxTknx5N7b9iiT3d3Er1tHO3s/X6e7TM/seALtrvb533i/J45L84tx7+WGLeEIjqm4XrgMAABiNPXsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAMKeqDq6qv6qqz1fVGVX10aryN0YB2OeIPQCYTH+89y1JPtjdP9rd90jy6CSHbejAAGAPiD0A+J5fTHJ1d79iaUJ3X9DdL6uq/arqhVX1iao6u6p+I0mq6gFV9f6q+uuq+kxVnTZFY6rqHlX1gWkP4Tuq6pANel4A3ADtv9EDAIBN5MgkZ+5k3hOTXNHd96yqGyf5SFW9c5p392ndLyf5SJL7VdXHkrwsydHdvaOqfj3JC5L8+4U+AwCYiD0A2Imq+rMkP5fk6iQXJPnpqnrkNPvWSY6Y5n28uy+a1jkrydYkX0/yk0neNe3o2y/JJes3egBu6MQeAHzPOUl+belOdx9fVQcl2Z7kS0me2t3vmF+hqh6Q5Kq5Sddm9t/XSnJOd99nwWMGgBU5Zw8Avue9SW5SVf9hbtrNpt/vSPIfquqAJKmqH6uqm6+yrc8m2VJV95mWP6CqjlzEoAFgJfbsAcCku7uqHp7kJVX1jCQ7knwryTOTvCGzwzPPnC7AsiPJw1fZ1tXTIZ9/UlW3zuy/uS/NbO8hACxcdfdGjwEAAIC9zGGcAAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAA/r/+sCQoIjN2bkAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 1080x576 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure(figsize=(15,8))\n",
    "\n",
    "sns.barplot( x = list(gene_count.keys()), y = list(gene_count.values()))\n",
    "plt.title(\"Gene distribution > 100 count\")\n",
    "plt.ylabel(\"Counts\")\n",
    "plt.xlabel(\"Gene\")\n",
    "\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA3sAAAHwCAYAAAAfJXbRAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/MnkTPAAAACXBIWXMAAAsTAAALEwEAmpwYAAAwcUlEQVR4nO3de5xdZX0v/s+XJIgiApWLkJkk4I1LLBFQalVEPFJsvSGIIu3hppxetHj0HC8/j1WP9mhrFfByjoceRagW0FaPqIgiaEEFOeESi1wsRZJMRIhUULkIhOf3x+zomCaZmczs2ZOV9/v12q/Ze61nPeu796zMzCfPs9aq1loAAADolq0GXQAAAADTT9gDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhD4BZp6oWVVWrqrm911+pquOmqe9nV9VNY17fWlX/YTr67vX3/ao6ZLr664eqOr6qvrUJ232yqt7Tj5oAmH7CHsAWrKpeWVXfrap7quqO3vM/raoadG1jtdZe0Fo7a7x2vYD4hHH6uqy19uTpqGt94ae1tm9r7ZvT0T8ATIWwB7CFqqo3Jjk9yfuTPC7Jrkn+OMkzk2w9wNL6Zu1IIQBsCYQ9gC1QVW2f5L8n+dPW2j+01n7eRl3TWju2tfbLXrtHVNXfVNWKqrq9qj5WVY/srTukqkaq6o29UcHbquqEMfvY4LbrqWdOr+1PquqWJH+wzvpvVtWre8+fUFX/VFV399qf11t+aa/5sqr6RVW9YkyNb66qHyc5c+2ydUp4WlVdX1U/raozq2qbXp//brrj2tHDqjo5ybFJ3tTb3xd76381LbT3GZxWVT/qPU6rqkdM5PNbz2d0QlXdUFU/r6pbquo/jVk33vfisVV1flX9rKquTPL4De2n1/5ZVfWdqrqrqlZW1fHrabNjVX2pqlb3PrcvVdXQmPXH9+r8eVX9sKqO3dj3D4DpJ+wBbJmekeQRSb4wTrv3JXlSkiVJnpBkfpK/GLP+cUm27y0/KclHq2rHCW471muSvDDJU5McmOSojdT07iRfS7JjkqEkH06S1trBvfX7tdYe3VpbGyIel+S3kixMcvIG+jw2ye9lNAQ9Kcl/28j+09vfGUk+neSve/t70XqavS3J72T0M9gvydPX6Xtjn9+67sjoZ/SYJCckObWq9p9gXx9Ncn+S3ZKc2HusV1UtTPKVjH6uO/dqv3Y9TbdKcmZGP9cFSe5L8pFeH9sm+VCSF7TWtkvyu2P6WO/3D4DpJ+wBbJl2SvKT1tpDaxeMGcm5r6oO7p23d3KS/9xa+7fW2s+T/I8krxzTz4NJ/ntr7cHW2gVJfpHkyRPcdqyjk5zWWlvZWvu3JO/dSO0PZjRg7N5au7+1Nt6FRh5O8o7W2i9ba/dtoM1Hxuz7L5McM06fE3VsRj+fO1prq5O8K8kfjVm/3s9vfR211r7cWvvX3gjsP2U0MD17vL6qak6SI5P8RWvtntbadUk2dv7jq5J8vbV2Tq+vO1tr166nnjtba//YWru39/39yyTPGdPk4SSLq+qRrbXbWmvfH1PnZL5/AGwiYQ9gy3Rnkp3GnsPWWvvd1toOvXVbZXRU51FJruqFwLuSXNhb/qt+xgbGJPcmefQEtx1r9yQrx7xevpHa35SkklxZo1e+3OAoVc/q1tr947RZd9+7j9N+onbPb76Xdfve0Of371TVC6rqiqr6t97n+fsZDe3j9bVzkrmZ+Oc7nORfN7J+bT2Pqqr/XVXLq+pnSS5NskNVzWmt3ZPkFRk9B/S2qvpyVe3V23Sy3z8ANpGwB7BlujzJL5O8ZCNtfpLRqXn7ttZ26D22b62tN4xMcdvbMhoy1lqwoY5baz9urb2mtbZ7kv+U5H/Wxq/A2SZQ77r7/lHv+T0ZDa1Jkqp63CT7/lFGR7HW1/eE9c7z+8ckf5Nk114ovyCjoWk8q5M8lAl+vhkNhRs9p6/njRkdhTyotfaYJGun0VaStNa+2lp7fkanjt6Y5G97yyf7/QNgEwl7AFug1tpdGZ1S+D+r6qiq2q6qtqqqJUm27bV5OKN/oJ9aVbskSVXNr6rfm0D/k932M0n+vKqGeueZvWVDfVfVy8dcCOSnGQ1cD/de355kz/HqW48/6+37tzJ6nt3a8/2WJdm3qpb0LtryznW2G29/5yT5b1W1c1XtlNFzFj+1CfVtndFzLFcneaiqXpDksIls2Fpbk+RzSd7ZG43bJ8nG7ln46ST/oaqOrqq5vYu7LFlPu+0yGujv6n1u71i7oqp2raqX9M7d+2VGp5Q+3Fu3se8fANNI2APYQrXW/jrJGzI6re723uN/J3lzku/0mr05yc1JruhN1ft6NnBO2XpMZtu/TfLVjIarqzMaTjbkaUm+W1W/SHJ+klNaa7f01r0zyVm9qaNHT7DOJPn7jJ4Dd0tGpzC+J0laaz/I6FVLv57kX5Kse37Zx5Ps09vf/11Pv+9JsjTJ95L8c++9Tfqm5L1z4v48o6H4pxk9r+78SXTx2oxO6fxxkk9m9MIqG9rXioxOEX1jkn/L6IVV9ltP09OSPDKjo7hXZHSa7lpbZfTY+lGvj+ck+ZPeuo19/wCYRtXaRGa3AAAAsDkxsgcAANBBwh4AAEAHCXsAAAAdJOwBAAB0UN/CXlV9oqruqKrrxix7f1XdWFXfq6rPV9UOY9a9tapurqqbJnJZbwAAADasb1fjrKqDM3pfnbNba4t7yw5Lcklr7aGq+qskaa29uXfPn3OSPD3J7hm9xPWTevcG2qCddtqpLVq0qC/1AwAAzHZXXXXVT1prO69v3dx+7bS1dmlVLVpn2dfGvLwiyVG95y9Jcm5r7ZdJflhVN2c0+F2+sX0sWrQoS5cunb6iAQAANiNVtXxD6wZ5zt6JSb7Sez4/ycox60Z6ywAAANgEAwl7VfW2JA8l+fQmbHtyVS2tqqWrV6+e/uIAAAA6YMbDXlUdn+SFSY5tvz5hcFWS4THNhnrL/p3W2hmttQNbawfuvPN6p6YCAABs8fp2zt76VNXhSd6U5DmttXvHrDo/yd9X1QczeoGWJya5ciZrAwAAtkwPPvhgRkZGcv/99w+6lA3aZpttMjQ0lHnz5k14m76Fvao6J8khSXaqqpEk70jy1iSPSHJRVSXJFa21P26tfb+qPpPk+oxO7/yz8a7ECQAAMB1GRkay3XbbZdGiRenllFmltZY777wzIyMj2WOPPSa8XT+vxnnMehZ/fCPt/zLJX/arHgAAgPW5//77Z23QS5KqymMf+9hM9polg7waJwAAwKwwW4PeWptSn7AHAAAwjh//+Md55Stfmcc//vE54IAD8vu///v5wQ9+kMWLFw+6tA2a0Qu0AAAAbG5aazniiCNy3HHH5dxzz02SLFu2LLfffvuAK9s4I3sAAAAb8Y1vfCPz5s3LH//xH/9q2X777Zfh4V/fPe7WW2/Ns5/97Oy///7Zf//9853vfCdJctttt+Xggw/OkiVLsnjx4lx22WVZs2ZNjj/++CxevDhPecpTcuqpp/albiN7AAAAG3HdddflgAMO2GibXXbZJRdddFG22Wab/Mu//EuOOeaYLF26NH//93+f3/u938vb3va2rFmzJvfee2+uvfbarFq1Ktddd12S5K677upL3cIeAADAFD344IN57Wtfm2uvvTZz5szJD37wgyTJ0572tJx44ol58MEH89KXvjRLlizJnnvumVtuuSWve93r8gd/8Ac57LDD+lKTaZwAAAAbse++++aqq67aaJtTTz01u+66a5YtW5alS5fmgQceSJIcfPDBufTSSzN//vwcf/zxOfvss7Pjjjtm2bJlOeSQQ/Kxj30sr371q/tSt7AHAACwEYceemh++ctf5owzzvjVsu9973tZuXLlr17ffffd2W233bLVVlvl7/7u77JmzZokyfLly7PrrrvmNa95TV796lfn6quvzk9+8pM8/PDDOfLII/Oe97wnV199dV/qNo0TAABgI6oqn//85/P6178+f/VXf5VtttkmixYtymmnnfarNn/6p3+aI488MmeffXYOP/zwbLvttkmSb37zm3n/+9+fefPm5dGPfnTOPvvsrFq1KieccEIefvjhJMl73/ve/tTdWutLxzPhwAMPbEuXLh10GQAAwGbshhtuyN577z3oMsa1vjqr6qrW2oHra28aJwAAQAcJewAAAB0k7AEAAHSQsAfAFmfh8IJU1YQfC4cXDLpkAJg0V+MEYIuzYmRllp35xQm33++EF/WxGgDoDyN7AAAAHSTsAQAADNiJJ56YXXbZJYsXL562PoU9AACAMRYtXDipc7vHeyxauHDcfR5//PG58MILp/V9OGcPAABgjOUrVuTum2+ctv62f8Je47Y5+OCDc+utt07bPhMjewAAAJ0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAADdswxx+QZz3hGbrrppgwNDeXjH//4lPt0NU4AAIAxFi5YMKEraE6mv/Gcc84507a/tYQ9AACAMW5dvnzQJUwL0zgBAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAAGLCVK1fmuc99bvbZZ5/su+++Of3006fcp7AHAAAwxqKFC1NV0/ZYtHDhuPucO3duPvCBD+T666/PFVdckY9+9KO5/vrrp/Q+3GcPAABgjOUrVuSOy787bf3t8oyDxm2z2267ZbfddkuSbLfddtl7772zatWq7LPPPpu8XyN7AAAAs8itt96aa665JgcdNH5I3BhhDwAAYJb4xS9+kSOPPDKnnXZaHvOYx0ypL2EPAABgFnjwwQdz5JFH5thjj83LXvayKfcn7AEAAAxYay0nnXRS9t5777zhDW+Ylj6FPQAAgAH79re/nb/7u7/LJZdckiVLlmTJkiW54IILptSnq3ECAACMsXDBggldQXMy/Y3nWc96Vlpr07bPRNgDAAD4DbcuXz7oEqaFaZwAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAADNj999+fpz/96dlvv/2y77775h3veMeU+xT2AAAAxli4YEGqatoeE7nP3iMe8YhccsklWbZsWa699tpceOGFueKKK6b0PtxnDwAAYIwVK1fmlvMvnrb+9nzx88ZtU1V59KMfnSR58MEH8+CDD6aqprRfI3sAAACzwJo1a7JkyZLssssuef7zn5+DDjpoSv0JewAAALPAnDlzcu2112ZkZCRXXnllrrvuuin1J+wBAADMIjvssEOe+9zn5sILL5xSP8IeAADAgK1evTp33XVXkuS+++7LRRddlL322mtKfbpACwAAwIDddtttOe6447JmzZo8/PDDOfroo/PCF75wSn0KewAAAGMsGB6e0BU0J9PfeH77t38711xzzbTtMxH2AAAAfsPyFSsGXcK0cM4eAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAALPAmjVr8tSnPnXK99dbS9gDAAAYY+HwglTVtD0WDi+Y0H5PP/307L333tP2Pvp2n72q+kSSFya5o7W2uLfst5Kcl2RRkluTHN1a+2lVVZLTk/x+knuTHN9au7pftQEAAGzIipGVWXbmF6etv/1OeNG4bUZGRvLlL385b3vb2/LBD35wWvbbz5G9TyY5fJ1lb0lycWvtiUku7r1OkhckeWLvcXKS/9XHugAAAGaV17/+9fnrv/7rbLXV9EW0voW91tqlSf5tncUvSXJW7/lZSV46ZvnZbdQVSXaoqt36VRsAAMBs8aUvfSm77LJLDjjggGntd6bP2du1tXZb7/mPk+zaez4/ycox7UZ6ywAAADrt29/+ds4///wsWrQor3zlK3PJJZfkD//wD6fc78Au0NJaa0naZLerqpOramlVLV29enUfKgMAAJg5733vezMyMpJbb7015557bg499NB86lOfmnK/Mx32bl87PbP39Y7e8lVJhse0G+ot+3daa2e01g5srR24884797VYAACAzVXfrsa5AecnOS7J+3pfvzBm+Wur6twkByW5e8x0TwAAgBmzYGh4QlfQnEx/E3XIIYfkkEMOmZb99vPWC+ckOSTJTlU1kuQdGQ15n6mqk5IsT3J0r/kFGb3tws0ZvfXCCf2qCwAAYGOWr1wx6BKmRd/CXmvtmA2set562rYkf9avWgAAALY0A7tACwAAAP0j7AEAAHSQsAcAANBBwh4AAEAHzfStFwAAAFiPRYsWZbvttsucOXMyd+7cLF26dEr9CXsAAABjLBhekJUjK6etv+Gh4ayY4O0cvvGNb2SnnXaalv0KewAAAGOsHFmZSz7w2Wnr79A3vnza+poM5+wBAADMAlWVww47LAcccEDOOOOMKfdnZA8AAGAW+Na3vpX58+fnjjvuyPOf//zstddeOfjggze5PyN7AAAAs8D8+fOTJLvsskuOOOKIXHnllVPqT9gDAAAYsHvuuSc///nPf/X8a1/7WhYvXjylPk3jBAAAGLDbb789RxxxRJLkoYceyqte9aocfvjhU+pT2AMAABhjeGh4Wq+gOTw0PG6bPffcM8uWLZu2fSbCHgAAwG+Y6D3xZjvn7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAB0kLAHAAAwC9x111056qijstdee2XvvffO5ZdfPqX+hD0AAIAxFgwNp6qm7bFgAvfZS5JTTjklhx9+eG688cYsW7Yse++995Teh/vsAQAAjLFy1UjOe/tZ09bfK9593Lht7r777lx66aX55Cc/mSTZeuuts/XWW09pv0b2AAAABuyHP/xhdt5555xwwgl56lOfmle/+tW55557ptSnsAcAADBgDz30UK6++ur8yZ/8Sa655ppsu+22ed/73jelPoU9AACAARsaGsrQ0FAOOuigJMlRRx2Vq6++ekp9CnsAAAAD9rjHPS7Dw8O56aabkiQXX3xx9tlnnyn16QItAAAAs8CHP/zhHHvssXnggQey55575swzz5xSf8IeAADAGMPzhyZ0Bc3J9DcRS5YsydKlS6dtv8IeAADAGCtGVg66hGnhnD0AAIAOEvYAAAA6SNgDANgMLBhekKqa1GPB8IJBlw0MkHP2AAA2AytHVuaSD3x2Utsc+saX96kaYHNgZA8AAKCDhD0AAIABu+mmm7JkyZJfPR7zmMfktNNOm1KfpnECAACMMTw0nJFVI9PW39D8oawc53YOT37yk3PttdcmSdasWZP58+fniCOOmNJ+hT0AAIAxRlaN5PRTPjJt/Z1y+msn1f7iiy/O4x//+CxcuHBK+zWNEwAAYBY599xzc8wxx0y5H2EPAABglnjggQdy/vnn5+Uvn/rVdIU9AACAWeIrX/lK9t9//+y6665T7kvYAwAAmCXOOeecaZnCmQh7AAAAs8I999yTiy66KC972cumpT9X4wQAABhjaP7QpK+gOV5/E7HtttvmzjvvnLb9CnsAAABjjHdPvM2FaZwAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAADNipp56afffdN4sXL84xxxyT+++/f8p9CnsAAABjDM0fSlVN22O8++ytWrUqH/rQh7J06dJcd911WbNmTc4999wpvw/32QMAABhj1Y9W5a0nvnva+nvvJ94+bpuHHnoo9913X+bNm5d77703u++++5T3a2SPgRseGp7U/4wMDw0PumQAACZowST/1luwBf6tN3/+/PyX//JfsmDBguy2227Zfvvtc9hhh025XyN7DNzIqpGcfspHJtz+lNNf28dqAACYTitXjeS8t5814favePdxfaxmdvrpT3+aL3zhC/nhD3+YHXbYIS9/+cvzqU99Kn/4h384pX6N7AEAAAzQ17/+9eyxxx7ZeeedM2/evLzsZS/Ld77znSn3K+wBAAAM0IIFC3LFFVfk3nvvTWstF198cfbee+8p9yvsQUctWrhw0leKWrRw4aDLBgDY4hx00EE56qijsv/+++cpT3lKHn744Zx88slT7tc5e9BRy1esyN033zipbbZ/wl59qgYAYPMxf/f5E7qC5mT6G8+73vWuvOtd75q2fSbCHgAAwG8YWTUy6BKmhWmcAAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcNJOxV1X+uqu9X1XVVdU5VbVNVe1TVd6vq5qo6r6q2HkRtAAAAXTDjYa+q5if58yQHttYWJ5mT5JVJ/irJqa21JyT5aZKTZro2AACAQTj99NOzePHi7LvvvjnttNOmpc9BTeOcm+SRVTU3yaOS3Jbk0CT/0Ft/VpKXDqY0AABgSzZ//lCqatoe8+cPbXR/1113Xf72b/82V155ZZYtW5YvfelLufnmm6f8Pmb8PnuttVVV9TdJViS5L8nXklyV5K7W2kO9ZiNJxr/zIAAAwDT70Y9W5cSj/3za+vvEZz600fU33HBDDjrooDzqUY9KkjznOc/J5z73ubzpTW+a0n4HMY1zxyQvSbJHkt2TbJvk8Elsf3JVLa2qpatXr+5TlQAAADNj8eLFueyyy3LnnXfm3nvvzQUXXJCVK1dOud8ZH9lL8h+S/LC1tjpJqupzSZ6ZZIeqmtsb3RtKsmp9G7fWzkhyRpIceOCBbWZKBgAA6I+99947b37zm3PYYYdl2223zZIlSzJnzpwp9zuIc/ZWJPmdqnpUVVWS5yW5Psk3khzVa3Ncki8MoDYAAIAZd9JJJ+Wqq67KpZdemh133DFPetKTptznjIe91tp3M3ohlquT/HOvhjOSvDnJG6rq5iSPTfLxma4NAABgEO64444kyYoVK/K5z30ur3rVq6bc5yCmcaa19o4k71hn8S1Jnj6AcgAAAAbqyCOPzJ133pl58+blox/9aHbYYYcp9zmQsAcAADBb7b77/HGvoDnZ/sZz2WWXTdv+1hL2AAAAxli1amTQJUyLQd1UHQAAgD4S9gAAADpI2AMAALZ4rc3uW3hvSn3CHgAAsEXbZpttcuedd87awNday5133pltttlmUtu5QAsAALBFGxoaysjISFavXj3oUjZom222ydDQ0KS2EfYAAIAt2rx587LHHnsMuoxpZxonAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBM2rh8IJU1aQeC4cXDLpsAIDNztxBFwBsWVaMrMyyM784qW32O+FFfaoGAKC7jOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB00obBXVc+cyDIAAABmh4mO7H14gssAAACYBeZubGVVPSPJ7ybZuareMGbVY5LM6WdhAAAAbLqNhr0kWyd5dK/ddmOW/yzJUf0qCgAAgKnZaNhrrf1Tkn+qqk+21pbPUE0AAABM0Xgje2s9oqrOSLJo7DattUP7URQAAABTM9Gw99kkH0vyf5Ks6V85AAAATIeJhr2HWmv/q6+VAAAAMG0meuuFL1bVn1bVblX1W2sffa0MAACATTbRkb3jel//65hlLcme01sOAAAA02FCYa+1tke/CwEAAGD6TCjsVdV/XN/y1trZm7LTqtohoxd7WZzREcITk9yU5LyMXvHz1iRHt9Z+uin9AwAAbOkmes7e08Y8np3knUlePIX9np7kwtbaXkn2S3JDkrckubi19sQkF/deAwAAsAkmOo3zdWNf90bmzt2UHVbV9kkOTnJ8r+8HkjxQVS9Jckiv2VlJvpnkzZuyDwAAgC3dREf21nVPkk09j2+PJKuTnFlV11TV/6mqbZPs2lq7rdfmx0l23cT+AQAAtngTCntV9cWqOr/3+HJGz6/7/Cbuc26S/ZP8r9baUzMaHH9jymZrrWX0XL711XJyVS2tqqWrV6/exBIAgC3FooULU1WTeixauHDQZQNM2URvvfA3Y54/lGR5a21kE/c5kmSktfbd3ut/yGjYu72qdmut3VZVuyW5Y30bt9bOSHJGkhx44IHrDYQAAGstX7Eid99846S22f4Je/WpGoCZM6GRvdbaPyW5Mcl2SXZM8sCm7rC19uMkK6vqyb1Fz0tyfZLz8+v7+R2X5Aubug8AAIAt3URvvXB0kvdn9KIpleTDVfVfW2v/sIn7fV2ST1fV1kluSXJCRoPnZ6rqpCTLkxy9iX0DAABs8SY6jfNtSZ7WWrsjSapq5yRfz+gUzElrrV2b5MD1rHrepvQHAADAb5ro1Ti3Whv0eu6cxLYAAADMsImO7F1YVV9Nck7v9SuSXNCfkgAAAJiqjYa9qnpCRu9/91+r6mVJntVbdXmST/e7OAAAADbNeCN7pyV5a5K01j6X5HNJUlVP6a17UR9rAwAAYBONd97drq21f153YW/Zor5UBAAAwJSNF/Z22Mi6R05jHQAAAEyj8cLe0qp6zboLq+rVSa7qT0kAAABM1Xjn7L0+yeer6tj8OtwdmGTrJEf0sS4AAACmYKNhr7V2e5LfrarnJlncW/zl1tolfa8MAACATTah++y11r6R5Bt9rgUAAIBpMt45ewAAAGyGhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAOqyqJvxYMDQ86HKBaTR30AUAANA/5739rAm3fcW7j+tjJcBMM7IHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAzAwuEFqaoJP+ZuNWfQJQObmbmDLgAAYEu0YmRllp35xQm33++EF/WxGqCLjOwBAAB0kLAHAADQQcLeLLFo4cJJzdtftHDhoEsGAABmMefszRLLV6zIHZd/d8Ltd3nGQX2sBgAA2NwZ2QMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpoYGGvquZU1TVV9aXe6z2q6rtVdXNVnVdVWw+qNgAAgM3dIEf2Tklyw5jXf5Xk1NbaE5L8NMlJA6kKAACgAwYS9qpqKMkfJPk/vdeV5NAk/9BrclaSlw6iNgAAgC4Y1MjeaUnelOTh3uvHJrmrtfZQ7/VIkvkDqAsAAKATZjzsVdULk9zRWrtqE7c/uaqWVtXS1atXT3N1AAAA3TCIkb1nJnlxVd2a5NyMTt88PckOVTW312Yoyar1bdxaO6O1dmBr7cCdd955JuoFAADY7Mx42GutvbW1NtRaW5TklUkuaa0dm+QbSY7qNTsuyRdmujYAAICumE332XtzkjdU1c0ZPYfv4wOuBwAAYLM1d/wm/dNa+2aSb/ae35Lk6YOsBwAAoCtm08geAAAA00TYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoINmPOxV1XBVfaOqrq+q71fVKb3lv1VVF1XVv/S+7jjTtQEAAHTFIEb2HkryxtbaPkl+J8mfVdU+Sd6S5OLW2hOTXNx7DQAAwCaY8bDXWruttXZ17/nPk9yQZH6SlyQ5q9fsrCQvnenaAAAAumKg5+xV1aIkT03y3SS7ttZu6636cZJdB1UXAADA5m5gYa+qHp3kH5O8vrX2s7HrWmstSdvAdidX1dKqWrp69eoZqBQAAEiSBcMLUlWTeszZas6gy95izR3ETqtqXkaD3qdba5/rLb69qnZrrd1WVbsluWN927bWzkhyRpIceOCB6w2EAADA9Fs5sjKXfOCzk9rm0De+vE/VMJ5BXI2zknw8yQ2ttQ+OWXV+kuN6z49L8oWZrg0AAKArBjGy98wkf5Tkn6vq2t6y/y/J+5J8pqpOSrI8ydEDqA0AAKATZjzstda+laQ2sPp5M1kLAABAVw30apwAAAD0h7AHAADQQcIeAAD02aKFCyd1u4JFCxcOumQ6YCC3XgAAgC3J8hUrcvfNN064/fZP2KuP1bClMLIHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AwBQtXLAgVTWpx9w5cwZdNtBxcwddAADA5m7FypW55fyLJ7XNni9+Xp+qARhlZA8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9rYgC4cXpKom9Vg4vGDQZcOkLRganvSxvmBoeNBls4kWLpj8z7a5c+YMumwA6Lu5gy6AmbNiZGWWnfnFSW2z3wkv6lM10D8rV43kvLefNaltXvHu4/pUDf22YuXK3HL+xZPaZs8XP69P1QDA7GFkDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD3GNanL17tVw2ZvMt/vRQsXDrpcZrlFCxdO+rYIjiuAmTPZW3PN3cqtazYnbr3AuC75wGcn3PbQN768j5UwE+64/LsTbrvLMw7qYyV0wfIVK3L3zTdOapvtn7BXn6oBYF2TvTWX23JtXozsAQAAdJCwBwAA0EGmcQJTUlWTaj93zqbN9Z/MfuY4n4A+mOyxPjw0nBUrV/SpGgAYn7AHTMkt5188qfZ7vvh5m7Qf544yaJM5BhPHIQCDZxonAABAB826sFdVh1fVTVV1c1W9ZdD1bIpNudT4vLmTH2Sd7D42dfpcv+uaU5M/DCe7j6H5Q314pzA58+cPTeq4ne+43eIMDw1P6hgZHhoedMkAzGKzahpnVc1J8tEkz08ykuT/VdX5rbXrB1vZ5MzUpcZnavrcZJ339rMm1f4V7z5u0vt464nvnlT7937i7ZPeB0y3H/1oVU48+s8n3P4Tn/lQH6thNhpZNZLTT/nIhNufcvpr+1gNAJu72Tay9/QkN7fWbmmtPZDk3CQvGXBNAAAAm53ZFvbmJ1k55vVIbxkAAACTUK21QdfwK1V1VJLDW2uv7r3+oyQHtdZeO6bNyUlO7r18cpKbZrxQkmSnJD8ZdBFs8RyHzAaOQ2YDxyGD5hgcnIWttZ3Xt2JWnbOXZFWSsWebD/WW/Upr7YwkZ8xkUfx7VbW0tXbgoOtgy+Y4ZDZwHDIbOA4ZNMfg7DTbpnH+vyRPrKo9qmrrJK9Mcv6AawIAANjszKqRvdbaQ1X12iRfTTInySdaa98fcFkAAACbnVkV9pKktXZBkgsGXQfjMpWW2cBxyGzgOGQ2cBwyaI7BWWhWXaAFAACA6THbztkDAABgGgh7JEmqak1VXTvm8Zbe8rlV9T+q6l/GrHvbRrZbVFWHVNXdvdc3VtXfDO6dsbmoqseOOY5+XFWrxrxuva/XVdVnq+pRvW1urap/7q1bOqavd1fV93rLv1ZVuw/unbG5GfNzbe3xNn8jx+bWG/n5+c11jssDq+qbA3tjzGpjjqNlVXV1Vf1ub/miqrqvt+76qjq7qub11s2rqvf1fkdfXVWXV9ULxvS5pPfz8/B19vWJqrqjqq6b2XfJbDdTx2FVDVfVN3p9fb+qTpn5d7tlmHXn7DEw97XWlqxn+XuSPC7JU1pr91fVdkneuLHtqmpRkstaay+sqkcmuaaqPt9a+3Z/SqcLWmt3JlmSJFX1ziS/aK39Te/1L9YeZ1X16SR/nOSDvU2f21pb974+72+tvb3X/s+T/EVvG5iI+9Y53l4x5vU7M+bY7C3b0M/PJNmlql7QWvtKXyumC8Yed7+X5L1JntNb96+ttSVVNSfJRUmOTvLpJO9OsluSxa21X1bVrmO2SZJjknyr9/XCMcs/meQjSc7u27thczVTx+FDSd7YWru697flVVV1UWvt+v6+vS2PsMcG9UZPXpNkUWvt/iRprf08yTsn2kdr7b6qujbJ/H7UyBbpsiS/vbEGrbWfjXm5bRInJ7Opxj3exvH+JG9LIuwxGY9J8tN1F7bW1lTVlUnmj/kdvUdr7Ze99bcn+UySVFUleXmS5ye5rKq2GfO7/NLef8zCxvTtOGyt3Zbktl77n1fVDRn9W1HYm2bCHms9shfK1npvkhuSrOgFvIls98PW2hFjV1bVjkmemOTSaayVLVRVzU3ygvz6fwZbkq9VVUvyv1trZ4xp+5dJ/mOSu5M8d6ZrZfO3nuNtQ/7dz8/W2nm955cnOaKqnptkYz9LYe1xtE1GR0kOXbdBVW2T5KAkpyR5QkZ/R/9s3XY9v5vR38v/2ps+/AdJ/rEPddMtM34c9v7j4alJvjs9b4GxnLPHWve11paMeZy3boOqOqE3V3tlVQ2vZ7uxQe/ZVbUsyaokX22t/Xgm3gSdtfaXz9IkK5J8vLf8Wa21/TP6B/mfVdXBazdorb2ttTac0Skmr53hetm8beh425Dxfn6+J8l/60OddMva42ivJIcnObs3KpIkj+8dk7cnua219r0J9HdMknN7z8/tvYbxzOhxWFWPzmj4e/1GAiNTIOyxMTcnWdCbS53W2pm9edx3Z/Sm9xtzWWttvyT7Jjmpqpb0s1A6b+wf069rrT2QJK21Vb2vdyT5fJKnr2fbTyc5cuZKpQPWe7xtqtbaJUkemeR3pqc8uq61dnmSnZLs3Fv0r73fv49PckBVvTi//h39mHW3751TdWSSv6iqW5N8OMnha3+fw0T0+zjsXeDlH5N8urX2uT6/nS2WsMcGtdbuzej/aH+kN2S/9h/u1pPo44dJ3pfkzX0pki1WVW075hfGtkkOS3Jd7/UTxzR9SZIbZ75C+A3vSfKmQRfB5qGq9srof6reOXZ572JUb0ny1jG/o0+vqq172+1cVS9P8rwk32utDbfWFrXWFmb0j+rfONUCNqafx2FvtPDjSW5orX0w9I2wx1qPrN+8dPj7esvfltETaK+rqmsyerGCs5L8aBJ9fyzJwU4GZ5rtmuRbvenCVyb5cmtt7blV76vRy+Z/L6Mh0CWd6acN/fz8ldbaBUlWD6A2Nh+/Oo6SnJfkuNbamvW0+79JHlVVz87o9ODVSa6v0dsofCnJzzI6Ve7z62z3j73lqapzMno+6ZOraqSqTurD+2HzNFPH4TOT/FGSQ8f87Pz9fryhLV215iJ1AAAAXWNkDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AGKOqdq2qv6+qW6rqqqq6vKrcnwyAzY6wBwA9vRv9/t8kl7bW9mytHZDklUmGBloYAGwCYQ8Afu3QJA+01j62dkFrbXlr7cNVNaeq3l9V/6+qvldV/ylJquqQqvpmVf1DVd1YVZ/uhcZU1QFV9U+9EcKvVtVuA3pfAGyB5g66AACYRfZNcvUG1p2U5O7W2tOq6hFJvl1VX+ute2pv2x8l+XaSZ1bVd5N8OMlLWmurq+oVSf4yyYl9fQcA0CPsAcAGVNVHkzwryQNJlif57ao6qrd6+yRP7K27srU20tvm2iSLktyVZHGSi3oDfXOS3DZz1QOwpRP2AODXvp/kyLUvWmt/VlU7JVmaZEWS17XWvjp2g6o6JMkvxyxak9Hfr5Xk+621Z/S5ZgBYL+fsAcCvXZJkm6r6kzHLHtX7+tUkf1JV85Kkqp5UVdtupK+bkuxcVc/otZ9XVfv2o2gAWB8jewDQ01prVfXSJKdW1ZuSrE5yT5I3J/lsRqdnXt27AMvqJC/dSF8P9KZ8fqiqts/o79zTMjp6CAB9V621QdcAAADANDONEwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADooP8fbaQ8hx3YeEAAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 1080x576 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure(figsize=(15,8))\n",
    "sns.histplot(data=train_variant[train_variant[\"Gene\"].isin(gene_count.keys())],\n",
    "             x=\"Gene\", hue=\"Class\", multiple=\"dodge\", shrink=.7)\n",
    "plt.title(\"Gene distribution and class\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene             Variation  Class\n",
       "ID                                     \n",
       "0   FAM58A  Truncating Mutations      1\n",
       "1      CBL                 W802*      2\n",
       "2      CBL                 Q249E      2\n",
       "3      CBL                 N454D      3\n",
       "4      CBL                 L399V      4"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_variant.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Gene, Variation, Class]\n",
       "Index: []"
      ]
     },
     "execution_count": 200,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_variant[train_variant[\"Gene\"] == \"RAD51\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 Text\n",
       "ID                                                   \n",
       "0   Cyclin-dependent kinases (CDKs) regulate a var...\n",
       "1    Abstract Background  Non-small cell lung canc...\n",
       "2    Abstract Background  Non-small cell lung canc...\n",
       "3   Recent evidence has demonstrated that acquired...\n",
       "4   Oncogenic mutations in the monomeric Casitas B..."
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_text.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "amino_acid_dict = {'C' : 'CYS', 'D' : 'ASP', 'S' : 'SER', 'Q' : 'GLN', 'K' : 'LYS',\n",
    "    'I' : 'ILE', 'P' : 'PRO', 'T' : 'THR', 'F' : 'PHE', 'N' : 'ASN', \n",
    "    'G' : 'GLY', 'H' : 'HIS', 'L' : 'LEU', 'R' : 'ARG', 'W' : 'TRP', \n",
    "    'A' : 'ALA', 'V' : 'VAL', 'E' : 'GLU', 'Y' : 'TYR', 'M' : 'MET'}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Fonction pour la selection\n",
    "\n",
    "## clean_text\n",
    "Nettoie l'article, lower puis enlève les points qui pourrait nuire à la detection des phrases, (comme les \"...\", ou \"fig.A\")\n",
    "\n",
    "\n",
    "## find_match\n",
    "Va détecter la phrase qui contient le mot, puis prend 1 phrase avant et après (au total 3 phrase) (voir diapo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def amino_three(amino):\n",
    "    return amino_acid_dict[amino]\n",
    "    \n",
    "def decompose_variation(variation):\n",
    "    decompose_aa_pos_aa = re.compile(\"([a-z]{1,})(\\d+)([a-z]{1,})\")\n",
    "    list_variation = decompose_aa_pos_aa.search(variation)\n",
    "    if list_variation:\n",
    "        aa1 = list_variation.group(1)\n",
    "        aa2 = list_variation.group(3)\n",
    "\n",
    "        if len(aa1) + len(aa2) == 2:\n",
    "            amino1 = amino_acid_dict[aa1.upper()].lower()\n",
    "            position = list_variation.group(2)\n",
    "            amino2 = amino_acid_dict[aa2.upper()].lower()\n",
    "            return [amino1,position,amino2]\n",
    "    return False\n",
    "\n",
    "def decompose_fusion(variation):\n",
    "    decompose_g1_g2_fusion = re.compile(\"(\\w+)\\s?(\\?|-)\\s?(\\w+)\\? fusion\")\n",
    "    list_variation = decompose_g1_g2_fusion.search(variation)\n",
    "    if list_variation:\n",
    "        gene1 = list_variation.group(1)\n",
    "        gene2 = list_variation.group(3)\n",
    "        print(gene1,gene2,variation, list_variation.group(0))\n",
    "        return \"(\" + gene1 + \"|\" + gene2 + \")\"\n",
    "    return False\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'concatenate_data' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-9-5c66f2ab830b>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mconcatenate_data\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"Variation\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mdecompose_variation\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlower\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'concatenate_data' is not defined"
     ]
    }
   ],
   "source": [
    "concatenate_data[\"Variation\"].apply(lambda x: print(x,decompose_variation(x.lower())))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def clean_text(article):\n",
    "    dot3 = re.compile(\"[.]{2,}\")\n",
    "    fig = re.compile(\"fig[s]?\\.\")\n",
    "    decimal = re.compile(\"\\d+\\.\\d+\")    \n",
    "    etal = re.compile(\"et al\\.\")\n",
    "    ie = re.compile(\"i.e.\")\n",
    "    inc = re.compile(\"inc.\")\n",
    "    mutation_point = re.compile(\"[p|c]\\.\")\n",
    "    \n",
    "    clean_article = article.lower()\n",
    "    clean_article = dot3.sub(\".\", clean_article)\n",
    "    clean_article = fig.sub(\"\", clean_article)\n",
    "    clean_article = decimal.sub(\"\", clean_article)\n",
    "    clean_article = etal.sub(\"\", clean_article)\n",
    "    clean_article = ie.sub(\"\", clean_article)    \n",
    "    clean_article = inc.sub(\"\", clean_article)    \n",
    "    clean_article = mutation_point.sub(\"\", clean_article)    \n",
    "    \n",
    "    \n",
    "    return clean_article\n",
    "\n",
    "def join_tuple_string(strings_tuple):\n",
    "    return ' '.join(strings_tuple)\n",
    "\n",
    "def find_match(text, word):\n",
    "    clean = clean_text(text)\n",
    "    word = word.lower()\n",
    "    target_sentence = \"([^.]*{}[^.]*\\.)\".format(word)\n",
    "    before_after_target = \"([^.]*\\.){0,1}\"    \n",
    "    match_exp = re.compile(before_after_target + target_sentence + before_after_target)\n",
    "    match_text = match_exp.findall(clean)\n",
    "    final_match = \"\".join(list(map(join_tuple_string, match_text)))\n",
    "    return final_match\n",
    "\n",
    "def extract_match(line):\n",
    "    # Mettre celui qui consomme le moins en 1er --> ceux qui sont facile à faire sans trop de calcul !\n",
    "    # 1er : best match\n",
    "    # 2e ?\n",
    "    \n",
    "    \n",
    "    # Cleaning text     \n",
    "    text = clean_text(line[\"Text\"])\n",
    "    variation = line[\"Variation\"].lower()\n",
    "    gene = line[\"Gene\"].lower()\n",
    "    print(line.name)\n",
    "    # Try first match with inital variation value\n",
    "    # Quality score = 1 \n",
    "    initial_match = find_match(text, variation)    \n",
    "    if len(initial_match) != 0:\n",
    "        print(\"First match ! \", variation)\n",
    "        return initial_match, 1\n",
    "    \n",
    "\n",
    "    # Fusion of two genes\n",
    "    # Quality score = 2\n",
    "    if \"fusion\" in variation:\n",
    "        fusion_gene = decompose_fusion(variation)\n",
    "        if fusion_gene:\n",
    "            match_fusion_gene = find_match(text, fusion_gene)\n",
    "            if len(match_fusion_gene) != 0:\n",
    "                print(\"Fusion gene1 | gene2\", variation)\n",
    "                return match_fusion_gene, 2\n",
    "\n",
    "        # Try to match the word fusion at least..\n",
    "        # Quality score 4 (bad)\n",
    "        match_fusion = find_match(text,\"fusion\")\n",
    "        if len(match_fusion) != 0:\n",
    "            print(\"FUSION\", variation)\n",
    "            return match_fusion, 4\n",
    "        \n",
    "    aa_pos_aa = decompose_variation(variation)\n",
    "    if aa_pos_aa :\n",
    "        if len(aa_pos_aa) == 3:\n",
    "            # If we success to split variation in 3 group --> aa1 pos aa2\n",
    "            # Second try without the last amino acid --> y371\n",
    "            match_variation_aa_pos = find_match(text, variation[:-1])\n",
    "            if len(match_variation_aa_pos) != 0:\n",
    "                print(\"y371 aa_pos\", variation)\n",
    "                \n",
    "                return match_variation_aa_pos, 2\n",
    "\n",
    "            # Third try with 3 letter code of amino acid --> tyr371ser\n",
    "            match_aa_pos_aa = find_match(text, \"\".join(aa_pos_aa))\n",
    "            if len(match_aa_pos_aa) != 0:\n",
    "                print(\"aa_pos_aa\", variation)\n",
    "                return match_aa_pos_aa, 1\n",
    "            \n",
    "            # Try with 3 letter code without the last aa --> tyr371\n",
    "            match_aa_pos = find_match(text, aa_pos_aa[0] + aa_pos_aa[1])\n",
    "            if len(match_aa_pos) != 0:\n",
    "                print(\"aa_pos\", variation)\n",
    "                \n",
    "                return match_aa_pos,2\n",
    "            \n",
    "            # Match position only --> 371\n",
    "            match_pos = find_match(text, aa_pos_aa[1])\n",
    "            if len(match_pos) != 0:\n",
    "                print(\"pos\", variation)\n",
    "                \n",
    "                return match_pos,4\n",
    "            \n",
    "            # Match position around the real position --> 370 - 379\n",
    "            match_pos_weak = find_match(text, aa_pos_aa[1][:-1] + \"[0-9]\")\n",
    "            if len(match_pos_weak) != 0:\n",
    "                print(\"pos weak\", variation)\n",
    "                \n",
    "                return match_pos_weak,5\n",
    "            \n",
    "            \n",
    "            \n",
    "    return [],0\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 388,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'16[0-9]'"
      ]
     },
     "execution_count": 388,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = \"165\"\n",
    "a[:-1] + \"[0-9]\"\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\" transformed nih3t3 cells showed pi3 kinase-dependent activation of akt and the transformed phenotype was reverted by treatment with the pi3 kinase inhibitor ly294002 (supplementary  9).  when introduced into lin-\\u2009sca1+\\u2009c-kit+ (lsk) hspcs, c-cbl mutants (c-cbl(gln367pro) and c-cbl(tyr371ser)), as well as a mouse lymphoma-der oncogenic mutant (c-cbl(70z)), significantly promoted the replating capacity of these progenitors ( 2a).  because c-cbl negaty modulates tyroskinase signalling, and all c-cbl mutations, uding those previously reported13, 14, 15, 16, affected the critical domains for its enzymatic activity involved in this modulation, c-cbl was postulated to have a tumour suppressor function; loss-of-function could be a mechanism for the oncogenicity of these c-cbl mutants3, 5. if you requassistance to access this image, or to obtain a text description, please contact npg@nature. com a, prolonged replating capacity of lsk cells transduced with mutant c-cbl (c-cbl(gln367pro) and c-cbl(tyr371ser)), compared to mock- or wild-type c-cbl-transduced cells.  replating capacity in methylcellulose culture is shown as mean colony number (and s.  high resolution image and legend (161k) download powerpoint sl(544k) sl may be downloaded for educational use, according to the terms descr in nature publishing group's lsing policy.    because tyroskinase signalling is central to cytokresponses in haematopoietic cells and its deregulation is a common feature of myeloprolatdisorders23, we next exam the effects of c-cbl mutations (c-cbl(gln367pro) and c-cbl(tyr371ser)) and the loss of wild-type c-cbl alleles on the responses of lsk hspcs to various cytok.  in serum-free conditions, c-cbl-/- lsk cells showed a modestly enhanced prolatresponse to a variety of cytok, uding scf, il3 and tpo, compared to c-cbl+/+ cells (mock columns in  4a).\""
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "extract_match(concatenate_data.loc[19,:])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exemple à la main juste pour tester"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "word_to_search = \"FAM58A\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "article = train_text.loc[0].values[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "34"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article.count(word_to_search)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 498,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>klkmklm</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>kk</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ar</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         0  1\n",
       "0  klkmklm  1\n",
       "1       kk  2\n",
       "2       ar  5"
      ]
     },
     "execution_count": 498,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame([[\"klkmklm\",1],['kk',2],[\"ar\",5]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "clean_match = find_match(article, word_to_search)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Before selection : total lenght 39672 and 34 occurences of the word FAM58A\n",
      "After selection :  total lenght 9214 and 34 occurences of the word FAM58A\n"
     ]
    }
   ],
   "source": [
    "print(\"Before selection : total lenght {} and {} occurences of the word {}\".format(\n",
    "    len(article),article.count(word_to_search) , word_to_search))\n",
    "\n",
    "print(\"After selection :  total lenght {} and {} occurences of the word {}\".format(\n",
    "    len(clean_match),clean_match.count(word_to_search.lower()), word_to_search))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>D171N</td>\n",
       "      <td>4</td>\n",
       "      <td>Introduction  Myelodysplastic syndromes (MDS) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>A122*</td>\n",
       "      <td>1</td>\n",
       "      <td>Introduction  Myelodysplastic syndromes (MDS) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>Fusions</td>\n",
       "      <td>1</td>\n",
       "      <td>The Runt-related transcription factor 1 gene (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>R80C</td>\n",
       "      <td>4</td>\n",
       "      <td>The RUNX1/AML1 gene is the most frequent targe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>K83E</td>\n",
       "      <td>4</td>\n",
       "      <td>The most frequent mutations associated with le...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3316 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene             Variation  Class  \\\n",
       "ID                                          \n",
       "0     FAM58A  Truncating Mutations      1   \n",
       "1        CBL                 W802*      2   \n",
       "2        CBL                 Q249E      2   \n",
       "3        CBL                 N454D      3   \n",
       "4        CBL                 L399V      4   \n",
       "...      ...                   ...    ...   \n",
       "3316   RUNX1                 D171N      4   \n",
       "3317   RUNX1                 A122*      1   \n",
       "3318   RUNX1               Fusions      1   \n",
       "3319   RUNX1                  R80C      4   \n",
       "3320   RUNX1                  K83E      4   \n",
       "\n",
       "                                                   Text  \n",
       "ID                                                       \n",
       "0     Cyclin-dependent kinases (CDKs) regulate a var...  \n",
       "1      Abstract Background  Non-small cell lung canc...  \n",
       "2      Abstract Background  Non-small cell lung canc...  \n",
       "3     Recent evidence has demonstrated that acquired...  \n",
       "4     Oncogenic mutations in the monomeric Casitas B...  \n",
       "...                                                 ...  \n",
       "3316  Introduction  Myelodysplastic syndromes (MDS) ...  \n",
       "3317  Introduction  Myelodysplastic syndromes (MDS) ...  \n",
       "3318  The Runt-related transcription factor 1 gene (...  \n",
       "3319  The RUNX1/AML1 gene is the most frequent targe...  \n",
       "3320  The most frequent mutations associated with le...  \n",
       "\n",
       "[3316 rows x 4 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concatenate_data = pd.merge(train_variant, train_text, on=\"ID\").dropna()\n",
    "concatenate_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 504,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "1\n",
      "First match !  w802*\n",
      "2\n",
      "First match !  q249e\n",
      "3\n",
      "First match !  n454d\n",
      "4\n",
      "First match !  l399v\n",
      "5\n",
      "First match !  v391i\n",
      "6\n",
      "First match !  v430m\n",
      "7\n",
      "First match !  deletion\n",
      "8\n",
      "First match !  y371h\n",
      "9\n",
      "First match !  c384r\n",
      "10\n",
      "First match !  p395a\n",
      "11\n",
      "First match !  k382e\n",
      "12\n",
      "First match !  r420q\n",
      "13\n",
      "First match !  c381a\n",
      "14\n",
      "First match !  p428l\n",
      "15\n",
      "First match !  d390y\n",
      "16\n",
      "17\n",
      "First match !  q367p\n",
      "18\n",
      "First match !  m374v\n",
      "19\n",
      "aa_pos_aa y371s\n",
      "20\n",
      "First match !  h94y\n",
      "21\n",
      "First match !  c396r\n",
      "22\n",
      "First match !  g375p\n",
      "23\n",
      "First match !  s376f\n",
      "24\n",
      "First match !  p417a\n",
      "25\n",
      "First match !  h398y\n",
      "26\n",
      "First match !  s2g\n",
      "27\n",
      "First match !  y846c\n",
      "28\n",
      "First match !  c228t\n",
      "29\n",
      "First match !  h412y\n",
      "30\n",
      "First match !  h876q\n",
      "31\n",
      "First match !  promoter mutations\n",
      "32\n",
      "First match !  p704s\n",
      "33\n",
      "First match !  amplification\n",
      "34\n",
      "First match !  c250t\n",
      "35\n",
      "First match !  g1809r\n",
      "36\n",
      "First match !  g1809k\n",
      "37\n",
      "First match !  d1709e\n",
      "38\n",
      "First match !  d1709a\n",
      "39\n",
      "First match !  e1705a\n",
      "40\n",
      "First match !  d1810a\n",
      "41\n",
      "First match !  truncating mutations\n",
      "42\n",
      "First match !  e1705k\n",
      "43\n",
      "First match !  deletion\n",
      "44\n",
      "pos weak t1365m\n",
      "45\n",
      "First match !  v648g\n",
      "46\n",
      "47\n",
      "First match !  a707t\n",
      "48\n",
      "First match !  promoter hypermethylation\n",
      "49\n",
      "pos weak r1343l\n",
      "50\n",
      "First match !  a209t\n",
      "51\n",
      "First match !  y280h\n",
      "52\n",
      "First match !  d927g\n",
      "53\n",
      "First match !  n510k\n",
      "54\n",
      "First match !  f248s\n",
      "55\n",
      "First match !  l708p\n",
      "56\n",
      "First match !  v995m\n",
      "57\n",
      "First match !  y412f\n",
      "58\n",
      "First match !  f74s\n",
      "59\n",
      "First match !  r1040l\n",
      "60\n",
      "First match !  r453c\n",
      "61\n",
      "62\n",
      "First match !  a1022e\n",
      "63\n",
      "pos weak q984k\n",
      "64\n",
      "First match !  t605m\n",
      "65\n",
      "First match !  k218t\n",
      "66\n",
      "pos weak n1125i\n",
      "67\n",
      "y371 aa_pos y35c\n",
      "68\n",
      "First match !  y35n\n",
      "69\n",
      "y371 aa_pos y35h\n",
      "70\n",
      "First match !  deletion\n",
      "71\n",
      "72\n",
      "73\n",
      "74\n",
      "First match !  deletion\n",
      "75\n",
      "76\n",
      "First match !  p284l\n",
      "77\n",
      "First match !  q276*\n",
      "78\n",
      "First match !  t286a\n",
      "79\n",
      "First match !  t283a\n",
      "80\n",
      "81\n",
      "First match !  i290r\n",
      "82\n",
      "First match !  a77s\n",
      "83\n",
      "First match !  a77p\n",
      "84\n",
      "First match !  a77t\n",
      "85\n",
      "First match !  m90i\n",
      "86\n",
      "First match !  amplification\n",
      "87\n",
      "First match !  overexpression\n",
      "88\n",
      "89\n",
      "First match !  n45s\n",
      "90\n",
      "First match !  r487q\n",
      "91\n",
      "aa_pos_aa c41y\n",
      "92\n",
      "First match !  g52r\n",
      "93\n",
      "aa_pos_aa p83l\n",
      "94\n",
      "First match !  s241l\n",
      "95\n",
      "First match !  s387y\n",
      "96\n",
      "97\n",
      "First match !  r537p\n",
      "98\n",
      "First match !  d404g\n",
      "99\n",
      "First match !  p525l\n",
      "100\n",
      "101\n",
      "First match !  deletion\n",
      "102\n",
      "103\n",
      "First match !  r976h\n",
      "104\n",
      "105\n",
      "First match !  t1219i\n",
      "106\n",
      "pos g39e\n",
      "107\n",
      "First match !  deletion\n",
      "108\n",
      "109\n",
      "First match !  r1095h\n",
      "110\n",
      "111\n",
      "112\n",
      "113\n",
      "114\n",
      "115\n",
      "First match !  deletion\n",
      "116\n",
      "First match !  truncating mutations\n",
      "117\n",
      "118\n",
      "First match !  truncating mutations\n",
      "119\n",
      "First match !  deletion\n",
      "120\n",
      "121\n",
      "First match !  k700r\n",
      "122\n",
      "First match !  r625c\n",
      "123\n",
      "First match !  e622d\n",
      "124\n",
      "First match !  h662q\n",
      "125\n",
      "y371 aa_pos r625d\n",
      "126\n",
      "First match !  k666m\n",
      "127\n",
      "y371 aa_pos h662r\n",
      "128\n",
      "First match !  r625h\n",
      "129\n",
      "First match !  k666t\n",
      "130\n",
      "First match !  r625l\n",
      "131\n",
      "y371 aa_pos e622q\n",
      "132\n",
      "First match !  r625g\n",
      "133\n",
      "First match !  k666n\n",
      "134\n",
      "y371 aa_pos k666r\n",
      "135\n",
      "First match !  k700e\n",
      "136\n",
      "pos weak r957q\n",
      "137\n",
      "First match !  deletion\n",
      "138\n",
      "139\n",
      "140\n",
      "141\n",
      "142\n",
      "First match !  g719a\n",
      "143\n",
      "First match !  g735s\n",
      "144\n",
      "145\n",
      "146\n",
      "147\n",
      "148\n",
      "First match !  g724s\n",
      "149\n",
      "150\n",
      "First match !  egfrviii\n",
      "151\n",
      "pos weak v765a\n",
      "152\n",
      "First match !  egfrvii\n",
      "153\n",
      "First match !  l858m\n",
      "154\n",
      "First match !  h870r\n",
      "155\n",
      "156\n",
      "First match !  h773l\n",
      "157\n",
      "First match !  t710a\n",
      "158\n",
      "First match !  l747p\n",
      "159\n",
      "y371 aa_pos e709v\n",
      "160\n",
      "First match !  d761y\n",
      "161\n",
      "162\n",
      "y371 aa_pos h773y\n",
      "163\n",
      "First match !  v851a\n",
      "164\n",
      "FUSION egfr-rad51 fusion\n",
      "165\n",
      "First match !  d770_n771insnpg\n",
      "166\n",
      "First match !  e746_a750del\n",
      "167\n",
      "First match !  a864t\n",
      "168\n",
      "pos weak e746q\n",
      "169\n",
      "170\n",
      "First match !  egfrvv\n",
      "171\n",
      "172\n",
      "y371 aa_pos e709q\n",
      "173\n",
      "First match !  e804g\n",
      "174\n",
      "175\n",
      "176\n",
      "First match !  g796s\n",
      "177\n",
      "First match !  r836c\n",
      "178\n",
      "179\n",
      "First match !  g857e\n",
      "180\n",
      "First match !  g719s\n",
      "181\n",
      "First match !  l747v\n",
      "182\n",
      "pos weak t783a\n",
      "183\n",
      "First match !  l692f\n",
      "184\n",
      "185\n",
      "pos v726m\n",
      "186\n",
      "First match !  e746v\n",
      "187\n",
      "188\n",
      "First match !  exon 19 deletion\n",
      "189\n",
      "First match !  s768i\n",
      "190\n",
      "First match !  c620y\n",
      "191\n",
      "First match !  s921r\n",
      "192\n",
      "y371 aa_pos n826y\n",
      "193\n",
      "194\n",
      "First match !  h773_v774insh\n",
      "195\n",
      "196\n",
      "First match !  c628y\n",
      "197\n",
      "First match !  exon 20 insertion\n",
      "198\n",
      "First match !  exon 19 insertion\n",
      "199\n",
      "First match !  l704n\n",
      "200\n",
      "First match !  l861q\n",
      "201\n",
      "First match !  l747f\n",
      "202\n",
      "First match !  n826s\n",
      "203\n",
      "First match !  l747_e749del\n",
      "204\n",
      "First match !  amplification\n",
      "205\n",
      "First match !  l858q\n",
      "206\n",
      "207\n",
      "First match !  r108k\n",
      "208\n",
      "pos weak v774a\n",
      "209\n",
      "First match !  s645c\n",
      "210\n",
      "First match !  l858r\n",
      "211\n",
      "First match !  egfr-kdd\n",
      "212\n",
      "213\n",
      "pos weak v774m\n",
      "214\n",
      "215\n",
      "pos weak e746g\n",
      "216\n",
      "First match !  l861p\n",
      "217\n",
      "First match !  k745m\n",
      "218\n",
      "219\n",
      "First match !  l747s\n",
      "220\n",
      "First match !  s492r\n",
      "221\n",
      "First match !  s768_d770dup\n",
      "222\n",
      "223\n",
      "First match !  l861r\n",
      "224\n",
      "First match !  g719d\n",
      "225\n",
      "226\n",
      "First match !  a763_y764insfqea\n",
      "227\n",
      "First match !  s784f\n",
      "228\n",
      "First match !  y801h\n",
      "229\n",
      "230\n",
      "First match !  v742a\n",
      "231\n",
      "232\n",
      "233\n",
      "First match !  t725m\n",
      "234\n",
      "First match !  l703p\n",
      "235\n",
      "First match !  t790m\n",
      "236\n",
      "First match !  a767_v769dup\n",
      "237\n",
      "First match !  e866k\n",
      "238\n",
      "First match !  egfrviv\n",
      "239\n",
      "First match !  s768_v769insvas\n",
      "240\n",
      "241\n",
      "First match !  e734q\n",
      "242\n",
      "First match !  r776h\n",
      "243\n",
      "First match !  t263p\n",
      "244\n",
      "First match !  t785a\n",
      "245\n",
      "First match !  a750p\n",
      "246\n",
      "First match !  l838v\n",
      "247\n",
      "First match !  p753s\n",
      "248\n",
      "249\n",
      "250\n",
      "First match !  i853t\n",
      "251\n",
      "First match !  l833v\n",
      "252\n",
      "First match !  r324l\n",
      "253\n",
      "First match !  v769m\n",
      "254\n",
      "First match !  w731l\n",
      "255\n",
      "First match !  e884k\n",
      "256\n",
      "257\n",
      "First match !  g719c\n",
      "258\n",
      "First match !  e330k\n",
      "259\n",
      "260\n",
      "First match !  r841k\n",
      "261\n",
      "262\n",
      "263\n",
      "First match !  r776c\n",
      "264\n",
      "First match !  e868g\n",
      "265\n",
      "First match !  p848l\n",
      "266\n",
      "y371 aa_pos l861f\n",
      "267\n",
      "268\n",
      "FUSION egfr-purb fusion\n",
      "269\n",
      "First match !  e709g\n",
      "270\n",
      "First match !  e709k\n",
      "271\n",
      "First match !  p596l\n",
      "272\n",
      "First match !  e709a\n",
      "273\n",
      "First match !  d587h\n",
      "274\n",
      "y371 aa_pos g863d\n",
      "275\n",
      "pos weak l838p\n",
      "276\n",
      "277\n",
      "First match !  v843i\n",
      "278\n",
      "First match !  g465e\n",
      "279\n",
      "FUSION igh-nkx2 fusion\n",
      "280\n",
      "FUSION trb-nkx2-1 fusion\n",
      "281\n",
      "First match !  amplification\n",
      "282\n",
      "First match !  fusions\n",
      "283\n",
      "FUSION tra-nkx2-1 fusion\n",
      "284\n",
      "First match !  a339v\n",
      "285\n",
      "286\n",
      "First match !  deletion\n",
      "287\n",
      "288\n",
      "First match !  amplification\n",
      "289\n",
      "First match !  brd4-nut fusion\n",
      "290\n",
      "aa_pos_aa s102f\n",
      "291\n",
      "First match !  deletion\n",
      "292\n",
      "First match !  k373e\n",
      "293\n",
      "First match !  i157t\n",
      "294\n",
      "First match !  e161del\n",
      "295\n",
      "First match !  p85l\n",
      "296\n",
      "First match !  s428f\n",
      "297\n",
      "First match !  r117g\n",
      "298\n",
      "First match !  e321k\n",
      "299\n",
      "First match !  amplification\n",
      "300\n",
      "First match !  tmprss2-etv1 fusion\n",
      "301\n",
      "FUSION tmprss2-etv5 fusion\n",
      "302\n",
      "FUSION tmprss2-etv4 fusion\n",
      "303\n",
      "y371 aa_pos t75m\n",
      "304\n",
      "First match !  fusions\n",
      "305\n",
      "aa_pos_aa m160v\n",
      "306\n",
      "pos g35r\n",
      "307\n",
      "pos k28m\n",
      "308\n",
      "pos g35v\n",
      "309\n",
      "First match !  i279p\n",
      "310\n",
      "First match !  r331p\n",
      "311\n",
      "First match !  s330a\n",
      "312\n",
      "First match !  s308a\n",
      "313\n",
      "314\n",
      "First match !  k320e\n",
      "315\n",
      "First match !  s1986f\n",
      "316\n",
      "First match !  amplification\n",
      "317\n",
      "First match !  d2033n\n",
      "318\n",
      "First match !  s1986y\n",
      "319\n",
      "FUSION ezr-ros1 fusion\n",
      "320\n",
      "First match !  l1947r\n",
      "321\n",
      "FUSION ros1-cd74 fusion\n",
      "322\n",
      "FUSION lima1-ros1 fusion\n",
      "323\n",
      "First match !  v2098i\n",
      "324\n",
      "First match !  g2032r\n",
      "325\n",
      "First match !  g1971e\n",
      "326\n",
      "FUSION kdelr2-ros1 fusion\n",
      "327\n",
      "First match !  slc34a2-ros1 fusion\n",
      "328\n",
      "First match !  l1951r\n",
      "329\n",
      "FUSION msn-ros1 fusion\n",
      "330\n",
      "First match !  fusions\n",
      "331\n",
      "FUSION lrig3-ros1 fusion\n",
      "332\n",
      "FUSION sdc4-ros1 fusion\n",
      "333\n",
      "First match !  tfg-ros1 fusion\n",
      "334\n",
      "First match !  g2101a\n",
      "335\n",
      "FUSION cep85l-ros1 fusion\n",
      "336\n",
      "First match !  cd74-ros1 fusion\n",
      "337\n",
      "First match !  c2060g\n",
      "338\n",
      "FUSION ccdc6-ros1 fusion\n",
      "339\n",
      "First match !  e1935g\n",
      "340\n",
      "FUSION fig-ros1 fusion\n",
      "341\n",
      "342\n",
      "First match !  r732q\n",
      "343\n",
      "First match !  deletion\n",
      "344\n",
      "345\n",
      "First match !  d254n\n",
      "346\n",
      "347\n",
      "First match !  a298t\n",
      "348\n",
      "349\n",
      "First match !  d257n\n",
      "350\n",
      "351\n",
      "352\n",
      "First match !  fusions\n",
      "353\n",
      "aa_pos_aa y1414c\n",
      "354\n",
      "355\n",
      "356\n",
      "FUSION ep300-moz fusion\n",
      "357\n",
      "First match !  d1399y\n",
      "358\n",
      "FUSION ep300-mll fusion\n",
      "359\n",
      "First match !  deletion\n",
      "360\n",
      "aa_pos_aa p1502l\n",
      "361\n",
      "362\n",
      "aa_pos_aa p531l\n",
      "363\n",
      "First match !  a530v\n",
      "364\n",
      "aa_pos_aa p531s\n",
      "365\n",
      "First match !  a530t\n",
      "366\n",
      "First match !  y532c\n",
      "367\n",
      "First match !  p531a\n",
      "368\n",
      "369\n",
      "First match !  s215g\n",
      "370\n",
      "y371 aa_pos r273l\n",
      "371\n",
      "pos r202s\n",
      "372\n",
      "First match !  h179q\n",
      "373\n",
      "First match !  k120e\n",
      "374\n",
      "First match !  c277q\n",
      "375\n",
      "First match !  r290h\n",
      "376\n",
      "First match !  r337p\n",
      "377\n",
      "First match !  p278s\n",
      "378\n",
      "First match !  y234h\n",
      "379\n",
      "First match !  c277w\n",
      "380\n",
      "pos s241t\n",
      "381\n",
      "First match !  v173e\n",
      "382\n",
      "First match !  q331r\n",
      "383\n",
      "First match !  h115n\n",
      "384\n",
      "pos p219s\n",
      "385\n",
      "First match !  r249w\n",
      "386\n",
      "pos v274f\n",
      "387\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "First match !  r175c\n",
      "388\n",
      "First match !  r248w\n",
      "389\n",
      "First match !  s121a\n",
      "390\n",
      "First match !  y220s\n",
      "391\n",
      "First match !  p153h\n",
      "392\n",
      "First match !  r181c\n",
      "393\n",
      "First match !  r337c\n",
      "394\n",
      "First match !  k120n\n",
      "395\n",
      "First match !  g245d\n",
      "396\n",
      "First match !  r175h\n",
      "397\n",
      "First match !  r181l\n",
      "398\n",
      "First match !  g244s\n",
      "399\n",
      "pos s240r\n",
      "400\n",
      "First match !  r273c\n",
      "401\n",
      "First match !  e221q\n",
      "402\n",
      "First match !  q331h\n",
      "403\n",
      "First match !  y220c\n",
      "404\n",
      "First match !  r342p\n",
      "405\n",
      "First match !  c242f\n",
      "406\n",
      "First match !  c242s\n",
      "407\n",
      "y371 aa_pos c135s\n",
      "408\n",
      "First match !  m237k\n",
      "409\n",
      "aa_pos v197l\n",
      "410\n",
      "First match !  s121c\n",
      "411\n",
      "First match !  s241f\n",
      "412\n",
      "413\n",
      "First match !  r248k\n",
      "414\n",
      "First match !  h214r\n",
      "415\n",
      "First match !  l330r\n",
      "416\n",
      "pos weak p151s\n",
      "417\n",
      "First match !  truncating mutations\n",
      "418\n",
      "First match !  r110l\n",
      "419\n",
      "First match !  c176f\n",
      "420\n",
      "First match !  h179y\n",
      "421\n",
      "First match !  c277r\n",
      "422\n",
      "First match !  l344r\n",
      "423\n",
      "First match !  r267q\n",
      "424\n",
      "First match !  r156h\n",
      "425\n",
      "First match !  d186a\n",
      "426\n",
      "First match !  c275s\n",
      "427\n",
      "First match !  e285k\n",
      "428\n",
      "429\n",
      "First match !  v157d\n",
      "430\n",
      "pos y236s\n",
      "431\n",
      "First match !  r337h\n",
      "432\n",
      "First match !  r213q\n",
      "433\n",
      "First match !  r280k\n",
      "434\n",
      "First match !  r110p\n",
      "435\n",
      "First match !  r249s\n",
      "436\n",
      "First match !  h168n\n",
      "437\n",
      "First match !  r290g\n",
      "438\n",
      "First match !  h193p\n",
      "439\n",
      "First match !  n239s\n",
      "440\n",
      "First match !  l348s\n",
      "441\n",
      "pos weak m133t\n",
      "442\n",
      "First match !  h179l\n",
      "443\n",
      "First match !  g244r\n",
      "444\n",
      "First match !  p316l\n",
      "445\n",
      "First match !  h297n\n",
      "446\n",
      "First match !  v272l\n",
      "447\n",
      "First match !  g244d\n",
      "448\n",
      "First match !  g245s\n",
      "449\n",
      "First match !  l348f\n",
      "450\n",
      "First match !  y163c\n",
      "451\n",
      "First match !  h179n\n",
      "452\n",
      "First match !  n235d\n",
      "453\n",
      "First match !  h168r\n",
      "454\n",
      "First match !  e285v\n",
      "455\n",
      "First match !  i255f\n",
      "456\n",
      "First match !  r158h\n",
      "457\n",
      "First match !  a161t\n",
      "458\n",
      "First match !  p278a\n",
      "459\n",
      "First match !  r249m\n",
      "460\n",
      "First match !  f341c\n",
      "461\n",
      "First match !  k292t\n",
      "462\n",
      "First match !  g266r\n",
      "463\n",
      "First match !  h179r\n",
      "464\n",
      "pos c141y\n",
      "465\n",
      "First match !  v143a\n",
      "466\n",
      "pos weak e326l\n",
      "467\n",
      "First match !  c229r\n",
      "468\n",
      "First match !  k291q\n",
      "469\n",
      "First match !  q144r\n",
      "470\n",
      "First match !  r248q\n",
      "471\n",
      "First match !  p250l\n",
      "472\n",
      "First match !  a347t\n",
      "473\n",
      "First match !  v157f\n",
      "474\n",
      "pos i251s\n",
      "475\n",
      "First match !  f134y\n",
      "476\n",
      "First match !  p152l\n",
      "477\n",
      "pos g199r\n",
      "478\n",
      "First match !  i162m\n",
      "479\n",
      "First match !  c135r\n",
      "480\n",
      "481\n",
      "First match !  a161s\n",
      "482\n",
      "First match !  g334r\n",
      "483\n",
      "First match !  v173l\n",
      "484\n",
      "First match !  r158l\n",
      "485\n",
      "First match !  h193n\n",
      "486\n",
      "First match !  r175l\n",
      "487\n",
      "pos p47s\n",
      "488\n",
      "First match !  r280a\n",
      "489\n",
      "First match !  r280t\n",
      "490\n",
      "First match !  k292i\n",
      "491\n",
      "First match !  y236c\n",
      "492\n",
      "y371 aa_pos r248l\n",
      "493\n",
      "First match !  r156c\n",
      "494\n",
      "pos y236d\n",
      "495\n",
      "pos t150i\n",
      "496\n",
      "First match !  l344p\n",
      "497\n",
      "First match !  h233n\n",
      "498\n",
      "First match !  p278l\n",
      "499\n",
      "First match !  k291e\n",
      "500\n",
      "First match !  e258v\n",
      "501\n",
      "pos weak s241y\n",
      "502\n",
      "First match !  c238f\n",
      "503\n",
      "First match !  f212y\n",
      "504\n",
      "First match !  r282w\n",
      "505\n",
      "First match !  r267p\n",
      "506\n",
      "First match !  h214q\n",
      "507\n",
      "First match !  t123a\n",
      "508\n",
      "First match !  s215c\n",
      "509\n",
      "First match !  r282q\n",
      "510\n",
      "First match !  r273h\n",
      "511\n",
      "First match !  g245a\n",
      "512\n",
      "First match !  c135y\n",
      "513\n",
      "First match !  r283q\n",
      "514\n",
      "First match !  r181h\n",
      "515\n",
      "First match !  p151h\n",
      "516\n",
      "First match !  f328v\n",
      "517\n",
      "First match !  y234c\n",
      "518\n",
      "First match !  p151a\n",
      "519\n",
      "First match !  r273g\n",
      "520\n",
      "First match !  v197e\n",
      "521\n",
      "pos e286k\n",
      "522\n",
      "First match !  k120m\n",
      "523\n",
      "First match !  i195t\n",
      "524\n",
      "First match !  g356a\n",
      "525\n",
      "First match !  r158c\n",
      "526\n",
      "First match !  h214n\n",
      "527\n",
      "First match !  c238s\n",
      "528\n",
      "pos weak l145r\n",
      "529\n",
      "First match !  p278r\n",
      "530\n",
      "First match !  r337l\n",
      "531\n",
      "First match !  g266e\n",
      "532\n",
      "533\n",
      "First match !  deletion\n",
      "534\n",
      "First match !  r321q\n",
      "535\n",
      "First match !  p305l\n",
      "536\n",
      "First match !  f346v\n",
      "537\n",
      "First match !  v319d\n",
      "538\n",
      "First match !  w368h\n",
      "539\n",
      "First match !  deletion\n",
      "540\n",
      "First match !  d450e\n",
      "541\n",
      "First match !  y406h\n",
      "542\n",
      "First match !  k420a\n",
      "543\n",
      "First match !  t413n\n",
      "544\n",
      "First match !  s276l\n",
      "545\n",
      "First match !  d300h\n",
      "546\n",
      "First match !  k375a\n",
      "547\n",
      "First match !  n387p\n",
      "548\n",
      "First match !  y426a\n",
      "549\n",
      "First match !  r428a\n",
      "550\n",
      "First match !  d300n\n",
      "551\n",
      "First match !  r310h\n",
      "552\n",
      "First match !  d450h\n",
      "553\n",
      "First match !  y298a\n",
      "554\n",
      "First match !  k341a\n",
      "555\n",
      "First match !  v356r\n",
      "556\n",
      "First match !  d258n\n",
      "557\n",
      "First match !  r373h\n",
      "558\n",
      "First match !  r287a\n",
      "559\n",
      "First match !  deletion\n",
      "560\n",
      "First match !  y297a\n",
      "561\n",
      "First match !  y238a\n",
      "562\n",
      "First match !  r243a\n",
      "563\n",
      "First match !  r373q\n",
      "564\n",
      "First match !  r292a\n",
      "565\n",
      "First match !  e239a\n",
      "566\n",
      "First match !  d408y\n",
      "567\n",
      "First match !  y384a\n",
      "568\n",
      "First match !  w406a\n",
      "569\n",
      "First match !  d408h\n",
      "570\n",
      "First match !  r268a\n",
      "571\n",
      "First match !  d408e\n",
      "572\n",
      "First match !  v277d\n",
      "573\n",
      "First match !  s425c\n",
      "574\n",
      "First match !  y237a\n",
      "575\n",
      "First match !  p336s\n",
      "576\n",
      "First match !  d537e\n",
      "577\n",
      "First match !  w509r\n",
      "578\n",
      "First match !  k507q\n",
      "579\n",
      "First match !  d537y\n",
      "580\n",
      "First match !  g508s\n",
      "581\n",
      "582\n",
      "First match !  r441p\n",
      "583\n",
      "First match !  r361c\n",
      "584\n",
      "First match !  l43v\n",
      "585\n",
      "First match !  truncating mutations\n",
      "586\n",
      "First match !  r497h\n",
      "587\n",
      "First match !  r515g\n",
      "588\n",
      "First match !  w398v\n",
      "589\n",
      "First match !  g419v\n",
      "590\n",
      "First match !  r420h\n",
      "591\n",
      "First match !  y513a\n",
      "592\n",
      "First match !  v465m\n",
      "593\n",
      "594\n",
      "First match !  v370d\n",
      "595\n",
      "First match !  r100t\n",
      "596\n",
      "First match !  p130s\n",
      "597\n",
      "First match !  deletion\n",
      "598\n",
      "First match !  k507a\n",
      "599\n",
      "First match !  e330g\n",
      "600\n",
      "First match !  k428a\n",
      "601\n",
      "First match !  r496h\n",
      "602\n",
      "First match !  e330k\n",
      "603\n",
      "First match !  c324y\n",
      "604\n",
      "605\n",
      "First match !  d351h\n",
      "606\n",
      "pos weak a532h\n",
      "607\n",
      "pos weak d357y\n",
      "608\n",
      "First match !  h132y\n",
      "609\n",
      "First match !  amplification\n",
      "610\n",
      "First match !  r24c\n",
      "611\n",
      "First match !  k22a\n",
      "612\n",
      "First match !  overexpression\n",
      "613\n",
      "First match !  a197t\n",
      "614\n",
      "First match !  amplification\n",
      "615\n",
      "First match !  r465h\n",
      "616\n",
      "617\n",
      "First match !  r479l\n",
      "618\n",
      "First match !  s462y\n",
      "619\n",
      "First match !  s227a\n",
      "620\n",
      "pos d520n\n",
      "621\n",
      "First match !  r479q\n",
      "622\n",
      "aa_pos r479h\n",
      "623\n",
      "First match !  r658q\n",
      "624\n",
      "First match !  r505c\n",
      "625\n",
      "First match !  r689q\n",
      "626\n",
      "First match !  k11r\n",
      "627\n",
      "First match !  t205a\n",
      "628\n",
      "pos r505l\n",
      "629\n",
      "First match !  deletion\n",
      "630\n",
      "First match !  i347m\n",
      "631\n",
      "First match !  copy number loss\n",
      "632\n",
      "pos weak g423r\n",
      "633\n",
      "First match !  r482q\n",
      "634\n",
      "First match !  amplification\n",
      "635\n",
      "First match !  deletion\n",
      "636\n",
      "637\n",
      "First match !  s31r\n",
      "638\n",
      "First match !  deletion\n",
      "639\n",
      "First match !  p133t\n",
      "640\n",
      "First match !  v109g\n",
      "641\n",
      "642\n",
      "First match !  q50p\n",
      "643\n",
      "First match !  p114l\n",
      "644\n",
      "First match !  e69g\n",
      "645\n",
      "y371 aa_pos g67r\n",
      "646\n",
      "First match !  l97r\n",
      "647\n",
      "First match !  d84g\n",
      "648\n",
      "649\n",
      "First match !  v126d\n",
      "650\n",
      "651\n",
      "First match !  d84h\n",
      "652\n",
      "First match !  m53i\n",
      "653\n",
      "y371 aa_pos t77p\n",
      "654\n",
      "First match !  n71i\n",
      "655\n",
      "First match !  g67s\n",
      "656\n",
      "First match !  n71s\n",
      "657\n",
      "y371 aa_pos r80p\n",
      "658\n",
      "First match !  d84v\n",
      "659\n",
      "First match !  d74n\n",
      "660\n",
      "y371 aa_pos d74y\n",
      "661\n",
      "First match !  p81l\n",
      "662\n",
      "First match !  l32p\n",
      "663\n",
      "y371 aa_pos d108h\n",
      "664\n",
      "First match !  deletion\n",
      "665\n",
      "y371 aa_pos g35r\n",
      "666\n",
      "First match !  d84y\n",
      "667\n",
      "First match !  n71k\n",
      "668\n",
      "First match !  r24p\n",
      "669\n",
      "First match !  r99p\n",
      "670\n",
      "First match !  a148t\n",
      "671\n",
      "First match !  g23d\n",
      "672\n",
      "First match !  r80l\n",
      "673\n",
      "First match !  g35a\n",
      "674\n",
      "First match !  h83y\n",
      "675\n",
      "First match !  a36p\n",
      "676\n",
      "First match !  p48l\n",
      "677\n",
      "aa_pos_aa d108n\n",
      "678\n",
      "First match !  p81t\n",
      "679\n",
      "First match !  g101w\n",
      "680\n",
      "First match !  a57v\n",
      "681\n",
      "First match !  r87l\n",
      "682\n",
      "First match !  h83n\n",
      "683\n",
      "First match !  d84n\n",
      "684\n",
      "First match !  p114s\n",
      "685\n",
      "First match !  i49s\n",
      "686\n",
      "First match !  r112g\n",
      "687\n",
      "First match !  s56i\n",
      "688\n",
      "689\n",
      "First match !  a60v\n",
      "690\n",
      "First match !  r87p\n",
      "691\n",
      "First match !  e88k\n",
      "692\n",
      "693\n",
      "694\n",
      "695\n",
      "First match !  deletion\n",
      "696\n",
      "First match !  hypermethylation\n",
      "697\n",
      "First match !  e35*\n",
      "698\n",
      "First match !  p40t\n",
      "699\n",
      "First match !  d86n\n",
      "700\n",
      "First match !  a23e\n",
      "701\n",
      "First match !  deletion\n",
      "702\n",
      "First match !  deletion\n",
      "703\n",
      "First match !  e812k\n",
      "704\n",
      "pos l49h\n",
      "705\n",
      "First match !  r678q\n",
      "706\n",
      "First match !  t798m\n",
      "707\n",
      "First match !  d808n\n",
      "708\n",
      "y371 aa_pos v659e\n",
      "709\n",
      "First match !  s760a\n",
      "710\n",
      "711\n",
      "First match !  l866m\n",
      "712\n",
      "First match !  c630y\n",
      "713\n",
      "First match !  i767m\n",
      "714\n",
      "First match !  s1002r\n",
      "715\n",
      "First match !  e719g\n",
      "716\n",
      "First match !  k753e\n",
      "717\n",
      "First match !  e719k\n",
      "718\n",
      "719\n",
      "First match !  g776s\n",
      "720\n",
      "First match !  s310f\n",
      "721\n",
      "First match !  s335c\n",
      "722\n",
      "First match !  l785f\n",
      "723\n",
      "pos weak l915m\n",
      "724\n",
      "First match !  c334s\n",
      "725\n",
      "726\n",
      "First match !  g660d\n",
      "727\n",
      "First match !  v777a\n",
      "728\n",
      "First match !  k753m\n",
      "729\n",
      "First match !  v842i\n",
      "730\n",
      "pos weak v750e\n",
      "731\n",
      "pos weak n319d\n",
      "732\n",
      "733\n",
      "First match !  v773a\n",
      "734\n",
      "First match !  e717k\n",
      "735\n",
      "First match !  k753a\n",
      "736\n",
      "First match !  t733i\n",
      "737\n",
      "First match !  y835f\n",
      "738\n",
      "First match !  s653c\n",
      "739\n",
      "740\n",
      "First match !  d769h\n",
      "741\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "First match !  l726i\n",
      "742\n",
      "First match !  h878y\n",
      "743\n",
      "First match !  exon 20 insertions\n",
      "744\n",
      "First match !  amplification\n",
      "745\n",
      "First match !  l755s\n",
      "746\n",
      "First match !  d821n\n",
      "747\n",
      "First match !  g309a\n",
      "748\n",
      "aa_pos_aa e265k\n",
      "749\n",
      "First match !  d769y\n",
      "750\n",
      "First match !  t798i\n",
      "751\n",
      "First match !  l755p\n",
      "752\n",
      "y371 aa_pos y1248f\n",
      "753\n",
      "First match !  p780l\n",
      "754\n",
      "First match !  v777l\n",
      "755\n",
      "First match !  s783p\n",
      "756\n",
      "First match !  d769a\n",
      "757\n",
      "First match !  c311r\n",
      "758\n",
      "759\n",
      "First match !  d277h\n",
      "760\n",
      "First match !  r143q\n",
      "761\n",
      "First match !  g309e\n",
      "762\n",
      "First match !  e321g\n",
      "763\n",
      "First match !  l726f\n",
      "764\n",
      "First match !  v777m\n",
      "765\n",
      "First match !  y803n\n",
      "766\n",
      "First match !  n857s\n",
      "767\n",
      "First match !  d845a\n",
      "768\n",
      "769\n",
      "First match !  s310y\n",
      "770\n",
      "First match !  v794m\n",
      "771\n",
      "772\n",
      "First match !  s846i\n",
      "773\n",
      "First match !  a1131t\n",
      "774\n",
      "First match !  g284r\n",
      "775\n",
      "First match !  p262h\n",
      "776\n",
      "First match !  v714m\n",
      "777\n",
      "First match !  t389k\n",
      "778\n",
      "First match !  amplification\n",
      "779\n",
      "First match !  q809r\n",
      "780\n",
      "First match !  r669c\n",
      "781\n",
      "First match !  a232v\n",
      "782\n",
      "First match !  v104m\n",
      "783\n",
      "First match !  e452k\n",
      "784\n",
      "First match !  ezr-erbb4 fusion\n",
      "785\n",
      "First match !  r306s\n",
      "786\n",
      "First match !  e563k\n",
      "787\n",
      "First match !  v348l\n",
      "788\n",
      "First match !  t244r\n",
      "789\n",
      "First match !  y285c\n",
      "790\n",
      "First match !  r544w\n",
      "791\n",
      "First match !  m887i\n",
      "792\n",
      "First match !  d595v\n",
      "793\n",
      "First match !  fusions\n",
      "794\n",
      "First match !  n181s\n",
      "795\n",
      "First match !  r838q\n",
      "796\n",
      "First match !  g785s\n",
      "797\n",
      "First match !  g936r\n",
      "798\n",
      "First match !  e317k\n",
      "799\n",
      "First match !  e542k\n",
      "800\n",
      "First match !  e872k\n",
      "801\n",
      "First match !  k935i\n",
      "802\n",
      "First match !  e836k\n",
      "803\n",
      "804\n",
      "pos weak e606g\n",
      "805\n",
      "806\n",
      "807\n",
      "808\n",
      "809\n",
      "aa_pos_aa r631c\n",
      "810\n",
      "y371 aa_pos d609e\n",
      "811\n",
      "First match !  d513y\n",
      "812\n",
      "aa_pos_aa v536m\n",
      "813\n",
      "aa_pos_aa r487w\n",
      "814\n",
      "815\n",
      "aa_pos_aa l461v\n",
      "816\n",
      "pos r592h\n",
      "817\n",
      "pos y24c\n",
      "818\n",
      "aa_pos_aa d423n\n",
      "819\n",
      "820\n",
      "First match !  v242f\n",
      "821\n",
      "aa_pos_aa r601q\n",
      "822\n",
      "First match !  a349p\n",
      "823\n",
      "First match !  p47a\n",
      "824\n",
      "First match !  m299i\n",
      "825\n",
      "First match !  r173c\n",
      "826\n",
      "First match !  truncating mutations\n",
      "827\n",
      "First match !  deletion\n",
      "828\n",
      "First match !  k52r\n",
      "829\n",
      "First match !  r109*\n",
      "830\n",
      "831\n",
      "aa_pos_aa r689s\n",
      "832\n",
      "First match !  deletion\n",
      "833\n",
      "834\n",
      "aa_pos_aa l230p\n",
      "835\n",
      "836\n",
      "aa_pos_aa s786f\n",
      "837\n",
      "FUSION bcr-abl1 fusion\n",
      "838\n",
      "First match !  l248v\n",
      "839\n",
      "First match !  g250e\n",
      "840\n",
      "First match !  m244v\n",
      "841\n",
      "First match !  fusions\n",
      "842\n",
      "First match !  e355a\n",
      "843\n",
      "First match !  l387m\n",
      "844\n",
      "First match !  e281k\n",
      "845\n",
      "First match !  e459k\n",
      "846\n",
      "First match !  f311l\n",
      "847\n",
      "First match !  y253f\n",
      "848\n",
      "First match !  t315i\n",
      "849\n",
      "First match !  e279k\n",
      "850\n",
      "First match !  e255v\n",
      "851\n",
      "First match !  e255k\n",
      "852\n",
      "First match !  h396p\n",
      "853\n",
      "First match !  y253h\n",
      "854\n",
      "First match !  v299l\n",
      "855\n",
      "First match !  l384m\n",
      "856\n",
      "First match !  v379i\n",
      "857\n",
      "First match !  m351t\n",
      "858\n",
      "First match !  v289f\n",
      "859\n",
      "First match !  q252h\n",
      "860\n",
      "First match !  f317l\n",
      "861\n",
      "First match !  h396r\n",
      "862\n",
      "First match !  f359c\n",
      "863\n",
      "864\n",
      "First match !  tmprss2-erg fusion\n",
      "865\n",
      "FUSION ewsr1-erg fusion\n",
      "866\n",
      "First match !  fus-erg fusion\n",
      "867\n",
      "First match !  fusions\n",
      "868\n",
      "869\n",
      "870\n",
      "871\n",
      "872\n",
      "First match !  a633t\n",
      "873\n",
      "First match !  r481g\n",
      "874\n",
      "First match !  h845y\n",
      "875\n",
      "876\n",
      "FUSION kdr-pdgfra fusion\n",
      "877\n",
      "First match !  n870s\n",
      "878\n",
      "First match !  q579r\n",
      "879\n",
      "First match !  r748g\n",
      "880\n",
      "First match !  r841k\n",
      "881\n",
      "882\n",
      "First match !  n659k\n",
      "883\n",
      "First match !  h570r\n",
      "884\n",
      "First match !  y555c\n",
      "885\n",
      "886\n",
      "First match !  e996k\n",
      "887\n",
      "First match !  v536e\n",
      "888\n",
      "First match !  s478p\n",
      "889\n",
      "890\n",
      "First match !  h650q\n",
      "891\n",
      "892\n",
      "y371 aa_pos a633v\n",
      "893\n",
      "894\n",
      "First match !  i562m\n",
      "895\n",
      "896\n",
      "First match !  v561a\n",
      "897\n",
      "898\n",
      "pos weak k830r\n",
      "899\n",
      "900\n",
      "901\n",
      "First match !  d842i\n",
      "902\n",
      "y371 aa_pos n659r\n",
      "903\n",
      "First match !  kif5b-pdgfra fusion\n",
      "904\n",
      "First match !  y849s\n",
      "905\n",
      "pos weak i834v\n",
      "906\n",
      "907\n",
      "First match !  g853d\n",
      "908\n",
      "First match !  d842v\n",
      "909\n",
      "First match !  t674i\n",
      "910\n",
      "First match !  p577s\n",
      "911\n",
      "912\n",
      "First match !  amplification\n",
      "913\n",
      "First match !  w349c\n",
      "914\n",
      "First match !  d846y\n",
      "915\n",
      "First match !  n848k\n",
      "916\n",
      "917\n",
      "First match !  l507p\n",
      "918\n",
      "First match !  n659s\n",
      "919\n",
      "First match !  v561d\n",
      "920\n",
      "921\n",
      "922\n",
      "First match !  strn-pdgfra fusion\n",
      "923\n",
      "First match !  d842y\n",
      "924\n",
      "First match !  y849c\n",
      "925\n",
      "First match !  bcr-pdgfra fusion\n",
      "926\n",
      "927\n",
      "First match !  etv6-pdgfra fusion\n",
      "928\n",
      "First match !  v658a\n",
      "929\n",
      "930\n",
      "First match !  f808l\n",
      "931\n",
      "FUSION kank1-pdgfrb fusion\n",
      "932\n",
      "First match !  r987w\n",
      "933\n",
      "FUSION nin-pdgfrb fusion\n",
      "934\n",
      "First match !  gpiap1-pdgfrb fusion\n",
      "935\n",
      "First match !  golga4-pdgfrb fusion\n",
      "936\n",
      "FUSION wdr48-pdgfrb fusion\n",
      "937\n",
      "First match !  tp53bp1-pdgfrb fusion\n",
      "938\n",
      "First match !  fusions\n",
      "939\n",
      "First match !  prkg2-pdgfrb fusion\n",
      "940\n",
      "First match !  ebf1-pdgfrb fusion\n",
      "941\n",
      "FUSION atf7ip-pdgfrb fusion\n",
      "942\n",
      "First match !  amplification\n",
      "943\n",
      "y371 aa_pos e1071w\n",
      "944\n",
      "First match !  pde4dip-pdgfrb fusion\n",
      "945\n",
      "First match !  bin2-pdgfrb fusion\n",
      "946\n",
      "aa_pos_aa r561c\n",
      "947\n",
      "FUSION cep85l-pdgfrb fusion\n",
      "948\n",
      "First match !  git2-pdgfrb fusion\n",
      "949\n",
      "FUSION sptbn1-pdgfrb fusion\n",
      "950\n",
      "FUSION hip1-pdgfrb fusion\n",
      "951\n",
      "FUSION kiaa1509-pdgfrb fusion\n",
      "952\n",
      "FUSION etv6-pdgfrb fusion\n",
      "953\n",
      "aa_pos_aa v665a\n",
      "954\n",
      "First match !  truncating mutations\n",
      "955\n",
      "First match !  deletion\n",
      "956\n",
      "First match !  v354e\n",
      "957\n",
      "First match !  d402y\n",
      "958\n",
      "First match !  p480l\n",
      "959\n",
      "960\n",
      "First match !  overexpression\n",
      "961\n",
      "First match !  m1_e165del\n",
      "962\n",
      "First match !  fusions\n",
      "963\n",
      "First match !  v534e\n",
      "964\n",
      "First match !  l536p\n",
      "965\n",
      "First match !  l469v\n",
      "966\n",
      "aa_pos_aa s463p\n",
      "967\n",
      "First match !  e380q\n",
      "968\n",
      "969\n",
      "FUSION esr1-yap1 fusion\n",
      "970\n",
      "aa_pos_aa l536q\n",
      "971\n",
      "First match !  l536h\n",
      "972\n",
      "First match !  v422del\n",
      "973\n",
      "First match !  esr1-ccdc170 fusion\n",
      "974\n",
      "pos weak l536r\n",
      "975\n",
      "First match !  s432l\n",
      "976\n",
      "First match !  y537c\n",
      "977\n",
      "First match !  y537s\n",
      "978\n",
      "First match !  tmprss2-etv1 fusion\n",
      "979\n",
      "First match !  fusions\n",
      "980\n",
      "FUSION ewsr1-etv1 fusion\n",
      "981\n",
      "First match !  r418g\n",
      "982\n",
      "First match !  r369q\n",
      "983\n",
      "First match !  fusions\n",
      "984\n",
      "First match !  p214l\n",
      "985\n",
      "First match !  r369w\n",
      "986\n",
      "First match !  385_418del\n",
      "987\n",
      "FUSION etv6-flt3 fusion\n",
      "988\n",
      "First match !  etv6-ntrk3 fusion\n",
      "989\n",
      "First match !  m224r\n",
      "990\n",
      "aa_pos_aa f158c\n",
      "991\n",
      "First match !  k82t\n",
      "992\n",
      "y371 aa_pos s35q\n",
      "993\n",
      "First match !  l117p\n",
      "994\n",
      "First match !  h68r\n",
      "995\n",
      "First match !  r246k\n",
      "996\n",
      "First match !  t417i\n",
      "997\n",
      "First match !  g305w\n",
      "998\n",
      "999\n",
      "1000\n",
      "aa_pos_aa h105r\n",
      "1001\n",
      "1002\n",
      "First match !  g305r\n",
      "1003\n",
      "1004\n",
      "aa_pos_aa h206d\n",
      "1005\n",
      "First match !  g1035s\n",
      "1006\n",
      "First match !  l191h\n",
      "1007\n",
      "First match !  l180p\n",
      "1008\n",
      "First match !  r1097h\n",
      "1009\n",
      "First match !  deletion\n",
      "1010\n",
      "First match !  s1036p\n",
      "1011\n",
      "First match !  g1567d\n",
      "1012\n",
      "First match !  l410r\n",
      "1013\n",
      "First match !  v299g\n",
      "1014\n",
      "First match !  h1620r\n",
      "1015\n",
      "1016\n",
      "First match !  l1584r\n",
      "1017\n",
      "First match !  q1554h\n",
      "1018\n",
      "First match !  t1623i\n",
      "1019\n",
      "First match !  h597y\n",
      "1020\n",
      "First match !  e1552del\n",
      "1021\n",
      "First match !  t1203k\n",
      "1022\n",
      "First match !  v1673f\n",
      "1023\n",
      "First match !  v705m\n",
      "1024\n",
      "First match !  p1675l\n",
      "1025\n",
      "First match !  g1596v\n",
      "1026\n",
      "First match !  r905w\n",
      "1027\n",
      "First match !  r905q\n",
      "1028\n",
      "First match !  v1673d\n",
      "1029\n",
      "First match !  r611w\n",
      "1030\n",
      "First match !  l830r\n",
      "1031\n",
      "First match !  s1653p\n",
      "1032\n",
      "First match !  r905g\n",
      "1033\n",
      "First match !  s1653f\n",
      "1034\n",
      "First match !  v769e\n",
      "1035\n",
      "First match !  q1503p\n",
      "1036\n",
      "First match !  deletion\n",
      "1037\n",
      "First match !  a889p\n",
      "1038\n",
      "First match !  w1610g\n",
      "1039\n",
      "First match !  r462c\n",
      "1040\n",
      "First match !  r1200w\n",
      "1041\n",
      "First match !  l792r\n",
      "1042\n",
      "First match !  l493v\n",
      "1043\n",
      "First match !  l493p\n",
      "1044\n",
      "First match !  c696y\n",
      "1045\n",
      "First match !  l826p\n",
      "1046\n",
      "First match !  a614d\n",
      "1047\n",
      "First match !  l146r\n",
      "1048\n",
      "First match !  r611q\n",
      "1049\n",
      "First match !  s1498n\n",
      "1050\n",
      "First match !  f615s\n",
      "1051\n",
      "First match !  y598c\n",
      "1052\n",
      "First match !  v705e\n",
      "1053\n",
      "First match !  l844r\n",
      "1054\n",
      "First match !  l448p\n",
      "1055\n",
      "First match !  p1709l\n",
      "1056\n",
      "First match !  e75g\n",
      "1057\n",
      "1058\n",
      "FUSION ewsr1-yy1 fusion\n",
      "1059\n",
      "First match !  ewsr1-ddit3 fusion\n",
      "1060\n",
      "1061\n",
      "First match !  fusions\n",
      "1062\n",
      "FUSION ewsr1-fli1 fusion\n",
      "1063\n",
      "FUSION ewsr1-nr4a3 fusion\n",
      "1064\n",
      "FUSION ewsr1-creb1 fusion\n",
      "1065\n",
      "FUSION ewsr1-etv4 fusion\n",
      "1066\n",
      "FUSION ewsr1-atf1 fusion\n",
      "1067\n",
      "FUSION ewsr1-wt1 fusion\n",
      "1068\n",
      "First match !  r206h\n",
      "1069\n",
      "First match !  g328e\n",
      "1070\n",
      "First match !  g328v\n",
      "1071\n",
      "pos weak g87r\n",
      "1072\n",
      "1073\n",
      "First match !  deletion\n",
      "1074\n",
      "First match !  s453fs*\n",
      "1075\n",
      "pos weak l455m\n",
      "1076\n",
      "First match !  f400i\n",
      "1077\n",
      "First match !  amplification\n",
      "1078\n",
      "First match !  m253k\n",
      "1079\n",
      "First match !  amplification\n",
      "1080\n",
      "1081\n",
      "pos weak y646n\n",
      "1082\n",
      "First match !  y646f\n",
      "1083\n",
      "pos weak y646h\n",
      "1084\n",
      "pos weak y646c\n",
      "1085\n",
      "pos weak y646s\n",
      "1086\n",
      "First match !  a677g\n",
      "1087\n",
      "First match !  overexpression\n",
      "1088\n",
      "First match !  amplification\n",
      "1089\n",
      "First match !  amplification\n",
      "1090\n",
      "First match !  k409q\n",
      "1091\n",
      "1092\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1093\n",
      "First match !  t1481fs\n",
      "1094\n",
      "1095\n",
      "First match !  e1286v\n",
      "1096\n",
      "First match !  t779fs\n",
      "1097\n",
      "First match !  truncating mutations\n",
      "1098\n",
      "First match !  deletion\n",
      "1099\n",
      "First match !  a95d\n",
      "1100\n",
      "First match !  e685v\n",
      "1101\n",
      "First match !  c91s\n",
      "1102\n",
      "First match !  truncating mutations\n",
      "1103\n",
      "First match !  f170i\n",
      "1104\n",
      "First match !  i47f\n",
      "1105\n",
      "First match !  c91a\n",
      "1106\n",
      "First match !  f81v\n",
      "1107\n",
      "First match !  deletion\n",
      "1108\n",
      "First match !  s858r\n",
      "1110\n",
      "First match !  truncating mutations\n",
      "1111\n",
      "First match !  h492r\n",
      "1112\n",
      "First match !  y510c\n",
      "1113\n",
      "First match !  deletion\n",
      "1114\n",
      "First match !  l274p\n",
      "1115\n",
      "First match !  w183a\n",
      "1116\n",
      "First match !  l210r\n",
      "1117\n",
      "1118\n",
      "1119\n",
      "First match !  r970c\n",
      "1120\n",
      "y371 aa_pos y1003c\n",
      "1121\n",
      "1122\n",
      "First match !  f1200i\n",
      "1123\n",
      "First match !  n375s\n",
      "1124\n",
      "aa_pos v1092i\n",
      "1125\n",
      "First match !  y1230c\n",
      "1126\n",
      "First match !  y1003f\n",
      "1127\n",
      "First match !  amplification\n",
      "1128\n",
      "1129\n",
      "1130\n",
      "First match !  d1010h\n",
      "1131\n",
      "1132\n",
      "1133\n",
      "First match !  overexpression\n",
      "1134\n",
      "First match !  y1003*\n",
      "1135\n",
      "First match !  y1235d\n",
      "1136\n",
      "1137\n",
      "First match !  d1010y\n",
      "1138\n",
      "1139\n",
      "1140\n",
      "1141\n",
      "1142\n",
      "pos weak m1250t\n",
      "1143\n",
      "First match !  ptprz1-met fusion\n",
      "1144\n",
      "1145\n",
      "First match !  y1230h\n",
      "1146\n",
      "First match !  e168d\n",
      "1147\n",
      "1148\n",
      "1149\n",
      "1150\n",
      "First match !  t992i\n",
      "1151\n",
      "First match !  s34f\n",
      "1152\n",
      "First match !  q157p\n",
      "1153\n",
      "First match !  s34y\n",
      "1154\n",
      "First match !  s3660l\n",
      "1155\n",
      "First match !  a1685s\n",
      "1156\n",
      "1157\n",
      "First match !  deletion\n",
      "1158\n",
      "First match !  deletion\n",
      "1159\n",
      "First match !  a4419s\n",
      "1160\n",
      "aa_pos_aa p4309a\n",
      "1161\n",
      "1162\n",
      "First match !  deletion\n",
      "1163\n",
      "1164\n",
      "First match !  r38h\n",
      "1165\n",
      "First match !  i31m\n",
      "1166\n",
      "First match !  g106d\n",
      "1167\n",
      "First match !  k111e\n",
      "1168\n",
      "First match !  r88q\n",
      "1169\n",
      "First match !  r108h\n",
      "1170\n",
      "First match !  m1043i\n",
      "1171\n",
      "First match !  e542g\n",
      "1172\n",
      "First match !  g106v\n",
      "1173\n",
      "First match !  d350g\n",
      "1174\n",
      "First match !  e545q\n",
      "1175\n",
      "First match !  v344a\n",
      "1176\n",
      "pos e78k\n",
      "1177\n",
      "1178\n",
      "First match !  r93q\n",
      "1179\n",
      "First match !  n1044k\n",
      "1180\n",
      "First match !  e116k\n",
      "1181\n",
      "First match !  v344g\n",
      "1182\n",
      "First match !  t1025s\n",
      "1183\n",
      "First match !  c420r\n",
      "1184\n",
      "First match !  k111n\n",
      "1185\n",
      "First match !  q60k\n",
      "1186\n",
      "First match !  n345i\n",
      "1187\n",
      "First match !  e545k\n",
      "1188\n",
      "First match !  e365k\n",
      "1189\n",
      "First match !  e81k\n",
      "1190\n",
      "First match !  q546k\n",
      "1191\n",
      "First match !  q546p\n",
      "1192\n",
      "First match !  e542k\n",
      "1193\n",
      "First match !  h701p\n",
      "1194\n",
      "First match !  r93w\n",
      "1195\n",
      "First match !  h1047r\n",
      "1196\n",
      "First match !  q546e\n",
      "1197\n",
      "First match !  a1066v\n",
      "1198\n",
      "First match !  amplification\n",
      "1199\n",
      "1200\n",
      "First match !  h1047l\n",
      "1201\n",
      "First match !  p449t\n",
      "1202\n",
      "First match !  p539r\n",
      "1203\n",
      "First match !  a1020v\n",
      "1204\n",
      "aa_pos_aa g914r\n",
      "1205\n",
      "y371 aa_pos e542q\n",
      "1206\n",
      "y371 aa_pos q546r\n",
      "1207\n",
      "First match !  e579k\n",
      "1208\n",
      "First match !  d1029y\n",
      "1209\n",
      "First match !  r115l\n",
      "1210\n",
      "1211\n",
      "First match !  g118d\n",
      "1212\n",
      "First match !  e545a\n",
      "1213\n",
      "First match !  h1047y\n",
      "1214\n",
      "First match !  e453a\n",
      "1215\n",
      "First match !  c378r\n",
      "1216\n",
      "First match !  e545g\n",
      "1217\n",
      "First match !  i391m\n",
      "1218\n",
      "First match !  e542v\n",
      "1219\n",
      "First match !  m1043v\n",
      "1220\n",
      "pos weak n553s\n",
      "1221\n",
      "First match !  e552k\n",
      "1222\n",
      "FUSION acpp-pik3cb fusion\n",
      "1223\n",
      "pos weak r48w\n",
      "1224\n",
      "First match !  fusions\n",
      "1225\n",
      "First match !  d1067v\n",
      "1226\n",
      "First match !  d1067y\n",
      "1227\n",
      "First match !  e633k\n",
      "1228\n",
      "1229\n",
      "First match !  d1067a\n",
      "1230\n",
      "First match !  l193f\n",
      "1231\n",
      "First match !  n82k\n",
      "1232\n",
      "First match !  h68y\n",
      "1233\n",
      "First match !  e124q\n",
      "1234\n",
      "First match !  e142d\n",
      "1235\n",
      "First match !  e135k\n",
      "1236\n",
      "First match !  y53h\n",
      "1237\n",
      "First match !  p123m\n",
      "1238\n",
      "First match !  e525k\n",
      "1239\n",
      "First match !  n334k\n",
      "1240\n",
      "First match !  e1021k\n",
      "1241\n",
      "First match !  fusions\n",
      "1242\n",
      "First match !  amplification\n",
      "1243\n",
      "First match !  yap1-mamld1 fusion\n",
      "1244\n",
      "First match !  yap1-fam118b fusion\n",
      "1245\n",
      "First match !  t576del\n",
      "1246\n",
      "First match !  g376r\n",
      "1247\n",
      "1248\n",
      "First match !  l370fs\n",
      "1249\n",
      "First match !  e218*\n",
      "1250\n",
      "First match !  deletion\n",
      "1251\n",
      "1252\n",
      "First match !  n564d\n",
      "1253\n",
      "First match !  w237_y242del\n",
      "1254\n",
      "First match !  e137k\n",
      "1255\n",
      "First match !  e160*\n",
      "1256\n",
      "1257\n",
      "1258\n",
      "First match !  d560y\n",
      "1259\n",
      "First match !  k379e\n",
      "1260\n",
      "First match !  r574fs\n",
      "1261\n",
      "1262\n",
      "1263\n",
      "First match !  k288q\n",
      "1264\n",
      "First match !  r162*\n",
      "1265\n",
      "First match !  r348*\n",
      "1266\n",
      "First match !  r262t\n",
      "1267\n",
      "First match !  n564k\n",
      "1268\n",
      "First match !  e439del\n",
      "1269\n",
      "aa_pos_aa g373r\n",
      "1270\n",
      "1271\n",
      "First match !  amplification\n",
      "1272\n",
      "First match !  n561d\n",
      "1273\n",
      "First match !  d557h\n",
      "1274\n",
      "First match !  a727v\n",
      "1275\n",
      "First match !  a171v\n",
      "1276\n",
      "1278\n",
      "First match !  g12v\n",
      "1279\n",
      "First match !  g13d\n",
      "1280\n",
      "First match !  amplification\n",
      "1281\n",
      "First match !  a146t\n",
      "1282\n",
      "First match !  k117r\n",
      "1283\n",
      "First match !  q61h\n",
      "1284\n",
      "y371 aa_pos g13r\n",
      "1285\n",
      "First match !  g13v\n",
      "1286\n",
      "First match !  a146v\n",
      "1287\n",
      "First match !  q61l\n",
      "1288\n",
      "First match !  q61k\n",
      "1289\n",
      "First match !  k117n\n",
      "1290\n",
      "First match !  t58i\n",
      "1291\n",
      "First match !  f28l\n",
      "1292\n",
      "First match !  g13c\n",
      "1293\n",
      "First match !  g12a\n",
      "1294\n",
      "First match !  q22k\n",
      "1295\n",
      "First match !  g12s\n",
      "1296\n",
      "First match !  y64a\n",
      "1297\n",
      "pos p34r\n",
      "1298\n",
      "First match !  g12c\n",
      "1299\n",
      "pos a59t\n",
      "1300\n",
      "1301\n",
      "First match !  r265s\n",
      "1302\n",
      "First match !  r100*\n",
      "1303\n",
      "aa_pos_aa m587v\n",
      "1304\n",
      "First match !  a111p\n",
      "1305\n",
      "1306\n",
      "First match !  r265c\n",
      "1307\n",
      "1308\n",
      "First match !  g101s\n",
      "1309\n",
      "1310\n",
      "pos weak e172k\n",
      "1311\n",
      "pos v506a\n",
      "1312\n",
      "1313\n",
      "1314\n",
      "pos weak r755s\n",
      "1315\n",
      "First match !  t117m\n",
      "1316\n",
      "pos g67r\n",
      "1317\n",
      "1318\n",
      "First match !  r217c\n",
      "1319\n",
      "First match !  i219v\n",
      "1320\n",
      "pos weak r725c\n",
      "1321\n",
      "pos weak p648s\n",
      "1322\n",
      "pos weak p648l\n",
      "1323\n",
      "1324\n",
      "pos weak a120s\n",
      "1325\n",
      "pos v384d\n",
      "1326\n",
      "First match !  r487*\n",
      "1327\n",
      "First match !  g67w\n",
      "1328\n",
      "pos weak l622h\n",
      "1329\n",
      "pos weak l582f\n",
      "1330\n",
      "First match !  k618t\n",
      "1331\n",
      "1332\n",
      "1333\n",
      "pos weak l749p\n",
      "1334\n",
      "First match !  t82a\n",
      "1335\n",
      "First match !  amplification\n",
      "1336\n",
      "First match !  amplification\n",
      "1337\n",
      "1338\n",
      "1339\n",
      "First match !  fusions\n",
      "1340\n",
      "First match !  e267g\n",
      "1341\n",
      "First match !  l321a\n",
      "1342\n",
      "First match !  q59e\n",
      "1343\n",
      "First match !  e40k\n",
      "1344\n",
      "First match !  k179m\n",
      "1345\n",
      "First match !  l362r\n",
      "1346\n",
      "First match !  v270a\n",
      "1347\n",
      "y371 aa_pos r370c\n",
      "1348\n",
      "First match !  l52r\n",
      "1349\n",
      "First match !  e49k\n",
      "1350\n",
      "First match !  e17k\n",
      "1351\n",
      "First match !  q79e\n",
      "1352\n",
      "First match !  g311d\n",
      "1353\n",
      "First match !  l78t\n",
      "1354\n",
      "First match !  q79k\n",
      "1355\n",
      "First match !  v271a\n",
      "1356\n",
      "First match !  p42t\n",
      "1357\n",
      "First match !  k39n\n",
      "1358\n",
      "First match !  w80a\n",
      "1359\n",
      "First match !  d32y\n",
      "1360\n",
      "First match !  t195i\n",
      "1361\n",
      "First match !  d325a\n",
      "1362\n",
      "First match !  k189n\n",
      "1363\n",
      "First match !  n53a\n",
      "1364\n",
      "First match !  d323h\n",
      "1365\n",
      "First match !  r23a\n",
      "1366\n",
      "First match !  l202f\n",
      "1367\n",
      "First match !  c77f\n",
      "1368\n",
      "First match !  d32h\n",
      "1369\n",
      "First match !  k181m\n",
      "1370\n",
      "First match !  i289m\n",
      "1371\n",
      "First match !  r170w\n",
      "1372\n",
      "First match !  r368c\n",
      "1373\n",
      "First match !  g161v\n",
      "1374\n",
      "First match !  fusions\n",
      "1375\n",
      "y371 aa_pos h355m\n",
      "1376\n",
      "First match !  amplification\n",
      "1377\n",
      "First match !  e17k\n",
      "1378\n",
      "First match !  d399n\n",
      "1379\n",
      "FUSION fgfr1op1-fgfr1 fusion\n",
      "1380\n",
      "First match !  a263v\n",
      "1381\n",
      "First match !  cep110-fgfr1 fusion\n",
      "1382\n",
      "First match !  znf198-fgfr1 fusion\n",
      "1383\n",
      "First match !  e664k\n",
      "1384\n",
      "First match !  amplification\n",
      "1385\n",
      "FUSION fgfr1-tacc1 fusion\n",
      "1386\n",
      "First match !  k656e\n",
      "1387\n",
      "First match !  fusions\n",
      "1388\n",
      "First match !  bcr-fgfr1 fusion\n",
      "1389\n",
      "FUSION erlin2?fgfr1 fusion\n",
      "1390\n",
      "First match !  n546k\n",
      "1391\n",
      "First match !  cux1-fgfr1 fusion\n",
      "1392\n",
      "First match !  g382d\n",
      "1393\n",
      "pos a500t\n",
      "1394\n",
      "First match !  k650e\n",
      "1395\n",
      "First match !  d617g\n",
      "1396\n",
      "First match !  k650t\n",
      "1397\n",
      "First match !  d641n\n",
      "1398\n",
      "FUSION igh-fgfr3 fusion\n",
      "1399\n",
      "First match !  n540s\n",
      "1400\n",
      "First match !  d641g\n",
      "1401\n",
      "pos weak y647c\n",
      "1402\n",
      "First match !  k650q\n",
      "1403\n",
      "First match !  s371c\n",
      "1404\n",
      "pos weak e627d\n",
      "1405\n",
      "First match !  k650r\n",
      "1406\n",
      "First match !  r248c\n",
      "1408\n",
      "pos weak v677i\n",
      "1409\n",
      "First match !  g637w\n",
      "1410\n",
      "FUSION fgfr3 - baiap2l1 fusion\n",
      "1411\n",
      "pos weak d646y\n",
      "1412\n",
      "1413\n",
      "pos weak h643d\n",
      "1414\n",
      "First match !  r669g\n",
      "1415\n",
      "First match !  i538v\n",
      "1416\n",
      "First match !  e466k\n",
      "1417\n",
      "First match !  f384l\n",
      "1418\n",
      "First match !  amplification\n",
      "1419\n",
      "First match !  g375c\n",
      "1420\n",
      "First match !  g697c\n",
      "1421\n",
      "First match !  v555m\n",
      "1422\n",
      "First match !  y373c\n",
      "1423\n",
      "First match !  s249c\n",
      "1424\n",
      "First match !  a391e\n",
      "1425\n",
      "First match !  k650m\n",
      "1426\n",
      "First match !  k650n\n",
      "1427\n",
      "First match !  g370c\n",
      "1428\n",
      "pos n653h\n",
      "1429\n",
      "First match !  fusions\n",
      "1430\n",
      "First match !  g380r\n",
      "1431\n",
      "First match !  f133v\n",
      "1432\n",
      "First match !  m117v\n",
      "1433\n",
      "First match !  r121q\n",
      "1434\n",
      "First match !  d140g\n",
      "1435\n",
      "First match !  y87c\n",
      "1436\n",
      "First match !  p94a\n",
      "1437\n",
      "First match !  w131a\n",
      "1438\n",
      "First match !  w131g\n",
      "1439\n",
      "First match !  s119n\n",
      "1440\n",
      "First match !  s80r\n",
      "1441\n",
      "First match !  g75r\n",
      "1442\n",
      "First match !  k129e\n",
      "1443\n",
      "First match !  e50k\n",
      "1444\n",
      "First match !  f133l\n",
      "1445\n",
      "First match !  y87n\n",
      "1446\n",
      "First match !  d130a\n",
      "1447\n",
      "First match !  n296i\n",
      "1448\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "First match !  f102c\n",
      "1449\n",
      "First match !  m117i\n",
      "1450\n",
      "1451\n",
      "First match !  r251q\n",
      "1452\n",
      "First match !  k517r\n",
      "1453\n",
      "First match !  e731k\n",
      "1454\n",
      "First match !  y375c\n",
      "1455\n",
      "First match !  k659n\n",
      "1456\n",
      "First match !  fusions\n",
      "1457\n",
      "First match !  t730s\n",
      "1458\n",
      "First match !  fgfr2-mgea5 fusion\n",
      "1459\n",
      "First match !  v755i\n",
      "1460\n",
      "First match !  g271e\n",
      "1461\n",
      "First match !  l770v\n",
      "1462\n",
      "First match !  r203c\n",
      "1463\n",
      "pos e565g\n",
      "1464\n",
      "First match !  fgfr2-ahcyl1 fusion\n",
      "1465\n",
      "First match !  a648t\n",
      "1466\n",
      "pos weak n549s\n",
      "1467\n",
      "First match !  p253r\n",
      "1468\n",
      "First match !  g701s\n",
      "1469\n",
      "First match !  t341p\n",
      "1470\n",
      "FUSION fgfr2?pphln1 fusion\n",
      "1471\n",
      "First match !  n549k\n",
      "1472\n",
      "First match !  s252w\n",
      "1473\n",
      "First match !  fgfr2-bicc1 fusion\n",
      "1474\n",
      "First match !  v248d\n",
      "1475\n",
      "First match !  k526e\n",
      "1476\n",
      "First match !  l617f\n",
      "1477\n",
      "First match !  i642v\n",
      "1478\n",
      "pos weak m535i\n",
      "1479\n",
      "pos weak k525e\n",
      "1480\n",
      "First match !  e636k\n",
      "1481\n",
      "FUSION fgfr2-tacc3 fusion\n",
      "1482\n",
      "FUSION fgfr2-kiaa1967 fusion\n",
      "1483\n",
      "1484\n",
      "First match !  c278f\n",
      "1485\n",
      "First match !  a389t\n",
      "1486\n",
      "pos weak l617m\n",
      "1487\n",
      "1488\n",
      "First match !  c382r\n",
      "1489\n",
      "pos weak k641n\n",
      "1490\n",
      "First match !  fgfr2-fam76a fusion\n",
      "1491\n",
      "First match !  fgfr2-ccdc6 fusion\n",
      "1492\n",
      "First match !  e219k\n",
      "1493\n",
      "First match !  k641r\n",
      "1494\n",
      "First match !  k310r\n",
      "1495\n",
      "First match !  d101y\n",
      "1496\n",
      "First match !  n549t\n",
      "1497\n",
      "First match !  m391r\n",
      "1498\n",
      "pos weak v564i\n",
      "1499\n",
      "pos weak k659e\n",
      "1500\n",
      "First match !  e475k\n",
      "1501\n",
      "pos n535k\n",
      "1502\n",
      "First match !  v550e\n",
      "1503\n",
      "First match !  amplification\n",
      "1504\n",
      "First match !  deletion\n",
      "1505\n",
      "First match !  truncating mutations\n",
      "1506\n",
      "First match !  g1128a\n",
      "1507\n",
      "First match !  g1128s\n",
      "1508\n",
      "First match !  l1122v\n",
      "1509\n",
      "First match !  t1151m\n",
      "1510\n",
      "First match !  npm-alk fusion\n",
      "1511\n",
      "First match !  g1123d\n",
      "1512\n",
      "First match !  d1203n\n",
      "1513\n",
      "First match !  ranbp2-alk fusion\n",
      "1514\n",
      "First match !  t1087i\n",
      "1515\n",
      "1516\n",
      "First match !  e1384k\n",
      "1517\n",
      "1518\n",
      "First match !  i1171n\n",
      "1519\n",
      "First match !  y1278s\n",
      "1520\n",
      "First match !  g1202r\n",
      "1521\n",
      "First match !  f1174i\n",
      "1522\n",
      "First match !  g881d\n",
      "1523\n",
      "First match !  s1206c\n",
      "1524\n",
      "First match !  i1171t\n",
      "1525\n",
      "First match !  cad-alk fusion\n",
      "1526\n",
      "First match !  l1152r\n",
      "1527\n",
      "First match !  c1156f\n",
      "1528\n",
      "First match !  g1123s\n",
      "1529\n",
      "First match !  s1206r\n",
      "1530\n",
      "First match !  f1174l\n",
      "1531\n",
      "First match !  s1206y\n",
      "1532\n",
      "First match !  strn-alk fusion\n",
      "1533\n",
      "FUSION ranbp1-alk fusion\n",
      "1534\n",
      "First match !  f1245c\n",
      "1535\n",
      "1536\n",
      "1537\n",
      "First match !  i1250t\n",
      "1538\n",
      "First match !  amplification\n",
      "1539\n",
      "First match !  t1343i\n",
      "1540\n",
      "1541\n",
      "First match !  l1198f\n",
      "1542\n",
      "First match !  f1245v\n",
      "1543\n",
      "First match !  g1201e\n",
      "1544\n",
      "First match !  v1180l\n",
      "1545\n",
      "First match !  d1091n\n",
      "1546\n",
      "First match !  kif5b-alk fusion\n",
      "1547\n",
      "First match !  l1196m\n",
      "1548\n",
      "First match !  p1139s\n",
      "1549\n",
      "1550\n",
      "First match !  g1269s\n",
      "1551\n",
      "First match !  g1269a\n",
      "1552\n",
      "First match !  l1196q\n",
      "1553\n",
      "First match !  a1099t\n",
      "1554\n",
      "y371 aa_pos g1286r\n",
      "1555\n",
      "pos weak r1060h\n",
      "1556\n",
      "First match !  d1270g\n",
      "1557\n",
      "y371 aa_pos e1210k\n",
      "1558\n",
      "First match !  n1178h\n",
      "1559\n",
      "First match !  k1062m\n",
      "1560\n",
      "First match !  fusions\n",
      "1561\n",
      "First match !  r1192p\n",
      "1562\n",
      "pos weak r1231q\n",
      "1563\n",
      "First match !  v597a\n",
      "1564\n",
      "First match !  d1349h\n",
      "1565\n",
      "First match !  r1464*\n",
      "1566\n",
      "First match !  i1183t\n",
      "1567\n",
      "First match !  l1198p\n",
      "1568\n",
      "First match !  c1156y\n",
      "1569\n",
      "First match !  l1152p\n",
      "1570\n",
      "First match !  h694r\n",
      "1571\n",
      "First match !  i1170n\n",
      "1572\n",
      "aa_pos r1275l\n",
      "1573\n",
      "First match !  i1170s\n",
      "1574\n",
      "First match !  a1234t\n",
      "1575\n",
      "First match !  s100f\n",
      "1576\n",
      "First match !  g325r\n",
      "1577\n",
      "1578\n",
      "First match !  amplification\n",
      "1579\n",
      "aa_pos_aa s505n\n",
      "1580\n",
      "First match !  w515k\n",
      "1581\n",
      "First match !  w515l\n",
      "1582\n",
      "First match !  p106l\n",
      "1583\n",
      "1584\n",
      "y371 aa_pos s217c\n",
      "1585\n",
      "y371 aa_pos s217t\n",
      "1586\n",
      "First match !  s217a\n",
      "1587\n",
      "First match !  t24a\n",
      "1588\n",
      "First match !  p86h\n",
      "1589\n",
      "First match !  d121g\n",
      "1590\n",
      "First match !  r64p\n",
      "1591\n",
      "y371 aa_pos y98h\n",
      "1592\n",
      "First match !  s68w\n",
      "1593\n",
      "pos v84l\n",
      "1594\n",
      "First match !  l128f\n",
      "1595\n",
      "First match !  truncating mutations\n",
      "1596\n",
      "First match !  v155a\n",
      "1597\n",
      "First match !  g114r\n",
      "1598\n",
      "y371 aa_pos h115r\n",
      "1599\n",
      "First match !  l118p\n",
      "1600\n",
      "First match !  r200w\n",
      "1601\n",
      "First match !  y98n\n",
      "1602\n",
      "First match !  q164r\n",
      "1603\n",
      "First match !  a149p\n",
      "1604\n",
      "First match !  l188q\n",
      "1605\n",
      "First match !  r69c\n",
      "1606\n",
      "First match !  y112h\n",
      "1607\n",
      "First match !  i151s\n",
      "1608\n",
      "First match !  l158p\n",
      "1609\n",
      "First match !  n78s\n",
      "1610\n",
      "First match !  f119s\n",
      "1611\n",
      "First match !  s65w\n",
      "1612\n",
      "First match !  q145h\n",
      "1613\n",
      "1614\n",
      "First match !  h191d\n",
      "1615\n",
      "pos weak l188v\n",
      "1616\n",
      "First match !  l118r\n",
      "1617\n",
      "First match !  w117c\n",
      "1618\n",
      "First match !  l63p\n",
      "1619\n",
      "First match !  e70k\n",
      "1620\n",
      "First match !  p25l\n",
      "1621\n",
      "First match !  r167w\n",
      "1622\n",
      "First match !  r82p\n",
      "1623\n",
      "First match !  s65l\n",
      "1624\n",
      "y371 aa_pos l158v\n",
      "1625\n",
      "First match !  s65a\n",
      "1626\n",
      "First match !  q96p\n",
      "1627\n",
      "First match !  deletion\n",
      "1628\n",
      "First match !  r167q\n",
      "1629\n",
      "First match !  v321m\n",
      "1630\n",
      "pos weak r154w\n",
      "1631\n",
      "pos weak r228k\n",
      "1632\n",
      "First match !  p326l\n",
      "1633\n",
      "1634\n",
      "y371 aa_pos n234i\n",
      "1635\n",
      "First match !  r134q\n",
      "1636\n",
      "pos g85r\n",
      "1637\n",
      "First match !  r304*\n",
      "1638\n",
      "aa_pos q72l\n",
      "1640\n",
      "1641\n",
      "First match !  d835n\n",
      "1642\n",
      "First match !  n676k\n",
      "1643\n",
      "1644\n",
      "First match !  d835a\n",
      "1645\n",
      "First match !  f590g\n",
      "1646\n",
      "First match !  t167a\n",
      "1647\n",
      "First match !  d324n\n",
      "1648\n",
      "First match !  y591d\n",
      "1649\n",
      "First match !  f594l\n",
      "1650\n",
      "First match !  v557i\n",
      "1651\n",
      "First match !  i867s\n",
      "1652\n",
      "First match !  d835del\n",
      "1653\n",
      "1654\n",
      "1655\n",
      "1656\n",
      "1657\n",
      "First match !  n676d\n",
      "1658\n",
      "1659\n",
      "pos a627t\n",
      "1660\n",
      "First match !  g831e\n",
      "1661\n",
      "First match !  r834q\n",
      "1662\n",
      "First match !  n676s\n",
      "1663\n",
      "First match !  m664i\n",
      "1664\n",
      "First match !  f691l\n",
      "1665\n",
      "First match !  m737i\n",
      "1666\n",
      "First match !  v194m\n",
      "1667\n",
      "First match !  v592a\n",
      "1668\n",
      "First match !  s451f\n",
      "1669\n",
      "y371 aa_pos s451e\n",
      "1670\n",
      "First match !  d835h\n",
      "1671\n",
      "First match !  y842c\n",
      "1672\n",
      "First match !  d835e\n",
      "1673\n",
      "First match !  y364h\n",
      "1674\n",
      "1675\n",
      "1676\n",
      "1677\n",
      "1678\n",
      "First match !  v579a\n",
      "1679\n",
      "1680\n",
      "First match !  n841i\n",
      "1681\n",
      "First match !  y572c\n",
      "1682\n",
      "1683\n",
      "First match !  d835y\n",
      "1684\n",
      "1685\n",
      "First match !  d839g\n",
      "1686\n",
      "First match !  amplification\n",
      "1687\n",
      "1688\n",
      "First match !  y599f\n",
      "1689\n",
      "First match !  g857a\n",
      "1690\n",
      "y371 aa_pos e5k\n",
      "1691\n",
      "First match !  e41a\n",
      "1692\n",
      "pos r20q\n",
      "1693\n",
      "First match !  g207e\n",
      "1694\n",
      "1695\n",
      "First match !  r802*\n",
      "1696\n",
      "1697\n",
      "First match !  i668v\n",
      "1698\n",
      "pos i18v\n",
      "1699\n",
      "First match !  s46i\n",
      "1700\n",
      "1701\n",
      "First match !  s46n\n",
      "1702\n",
      "First match !  t485k\n",
      "1703\n",
      "First match !  e541k\n",
      "1704\n",
      "1705\n",
      "First match !  p95h\n",
      "1706\n",
      "First match !  e1099k\n",
      "1707\n",
      "1708\n",
      "1709\n",
      "First match !  s297f\n",
      "1710\n",
      "First match !  p286r\n",
      "1711\n",
      "First match !  s459f\n",
      "1712\n",
      "pos weak s279y\n",
      "1713\n",
      "First match !  f367s\n",
      "1714\n",
      "First match !  v411l\n",
      "1715\n",
      "y371 aa_pos p286h\n",
      "1716\n",
      "pos weak l424i\n",
      "1717\n",
      "First match !  l424v\n",
      "1718\n",
      "aa_pos_aa s24f\n",
      "1719\n",
      "aa_pos_aa r11k\n",
      "1720\n",
      "y371 aa_pos a40e\n",
      "1721\n",
      "aa_pos_aa p26s\n",
      "1722\n",
      "1723\n",
      "First match !  i1307k\n",
      "1724\n",
      "First match !  n1026s\n",
      "1725\n",
      "First match !  deletion\n",
      "1726\n",
      "First match !  r640g\n",
      "1727\n",
      "First match !  a290t\n",
      "1728\n",
      "First match !  amplification\n",
      "1729\n",
      "1730\n",
      "1731\n",
      "First match !  deletion\n",
      "1732\n",
      "1733\n",
      "pos weak l390f\n",
      "1734\n",
      "First match !  d603n\n",
      "1735\n",
      "1736\n",
      "1737\n",
      "First match !  a636p\n",
      "1738\n",
      "First match !  a272v\n",
      "1739\n",
      "pos weak d603g\n",
      "1740\n",
      "aa_pos_aa g751r\n",
      "1741\n",
      "1742\n",
      "1743\n",
      "1744\n",
      "First match !  s860l\n",
      "1745\n",
      "First match !  truncating mutations\n",
      "1746\n",
      "First match !  s723f\n",
      "1747\n",
      "1748\n",
      "1749\n",
      "First match !  r389*\n",
      "1750\n",
      "First match !  y139d\n",
      "1751\n",
      "First match !  r132c\n",
      "1752\n",
      "First match !  r100a\n",
      "1753\n",
      "First match !  g123r\n",
      "1754\n",
      "First match !  r132h\n",
      "1755\n",
      "First match !  r132g\n",
      "1756\n",
      "First match !  r132q\n",
      "1757\n",
      "First match !  p33s\n",
      "1758\n",
      "First match !  a134d\n",
      "1759\n",
      "pos weak v294m\n",
      "1760\n",
      "pos weak h133q\n",
      "1761\n",
      "pos weak i130m\n",
      "1762\n",
      "pos i99m\n",
      "1763\n",
      "First match !  g70d\n",
      "1764\n",
      "First match !  r49c\n",
      "1765\n",
      "First match !  e190k\n",
      "1766\n",
      "First match !  r140q\n",
      "1767\n",
      "First match !  r172s\n",
      "1768\n",
      "First match !  r172g\n",
      "1769\n",
      "First match !  r172k\n",
      "1770\n",
      "First match !  r172m\n",
      "1771\n",
      "First match !  c528s\n",
      "1772\n",
      "First match !  e571k\n",
      "1773\n",
      "First match !  deletion\n",
      "1774\n",
      "First match !  truncating mutations\n",
      "1775\n",
      "1776\n",
      "First match !  deletion\n",
      "1777\n",
      "First match !  r1446h\n",
      "1778\n",
      "y371 aa_pos r1446c\n",
      "1779\n",
      "First match !  truncating mutations\n",
      "1780\n",
      "pos weak w1502a\n",
      "1781\n",
      "First match !  q1500p\n",
      "1782\n",
      "pos weak l702h\n",
      "1783\n",
      "1784\n",
      "First match !  p380r\n",
      "1785\n",
      "First match !  q58l\n",
      "1786\n",
      "First match !  arv567es\n",
      "1787\n",
      "First match !  a159t\n",
      "1788\n",
      "First match !  t878a\n",
      "1789\n",
      "pos weak h875y\n",
      "1790\n",
      "First match !  g216r\n",
      "1791\n",
      "pos weak w742l\n",
      "1792\n",
      "pos weak w742c\n",
      "1793\n",
      "pos weak t878s\n",
      "1794\n",
      "First match !  amplification\n",
      "1795\n",
      "1796\n",
      "First match !  f876l\n",
      "1797\n",
      "First match !  l272f\n",
      "1798\n",
      "First match !  q120e\n",
      "1799\n",
      "pos f877l\n",
      "1800\n",
      "First match !  g248v\n",
      "1801\n",
      "First match !  ar-v7\n",
      "1802\n",
      "First match !  n217i\n",
      "1803\n",
      "First match !  s214a\n",
      "1804\n",
      "y371 aa_pos s214t\n",
      "1805\n",
      "First match !  q347_a348del\n",
      "1806\n",
      "First match !  f351l\n",
      "1807\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "First match !  s214c\n",
      "1808\n",
      "First match !  g322s\n",
      "1809\n",
      "First match !  t37a\n",
      "1810\n",
      "First match !  g17v\n",
      "1811\n",
      "aa_pos_aa g17e\n",
      "1812\n",
      "aa_pos_aa g14v\n",
      "1813\n",
      "First match !  e40q\n",
      "1814\n",
      "aa_pos_aa l57v\n",
      "1815\n",
      "First match !  r5q\n",
      "1816\n",
      "1817\n",
      "First match !  t19n\n",
      "1818\n",
      "First match !  y42c\n",
      "1819\n",
      "First match !  e40t\n",
      "1820\n",
      "First match !  overexpression\n",
      "1821\n",
      "First match !  e40l\n",
      "1822\n",
      "First match !  g17a\n",
      "1823\n",
      "First match !  y34c\n",
      "1824\n",
      "First match !  e40w\n",
      "1825\n",
      "First match !  e40n\n",
      "1826\n",
      "1827\n",
      "First match !  w257g\n",
      "1828\n",
      "y371 aa_pos s256w\n",
      "1829\n",
      "pos p179r\n",
      "1830\n",
      "aa_pos r183p\n",
      "1831\n",
      "First match !  r182w\n",
      "1832\n",
      "pos r258h\n",
      "1833\n",
      "First match !  s256f\n",
      "1834\n",
      "pos r258c\n",
      "1835\n",
      "pos p179l\n",
      "1836\n",
      "aa_pos r183g\n",
      "1837\n",
      "y371 aa_pos w257c\n",
      "1838\n",
      "aa_pos r183q\n",
      "1839\n",
      "First match !  deletion\n",
      "1840\n",
      "First match !  truncating mutations\n",
      "1841\n",
      "First match !  r1625c\n",
      "1842\n",
      "First match !  m1255i\n",
      "1843\n",
      "First match !  a1374v\n",
      "1844\n",
      "First match !  g1125a\n",
      "1845\n",
      "First match !  h114y\n",
      "1846\n",
      "First match !  p186s\n",
      "1847\n",
      "First match !  s270l\n",
      "1848\n",
      "First match !  r264c\n",
      "1849\n",
      "First match !  h284n\n",
      "1850\n",
      "y371 aa_pos h284p\n",
      "1851\n",
      "1852\n",
      "First match !  r339w\n",
      "1853\n",
      "First match !  deletion\n",
      "1854\n",
      "First match !  amplification\n",
      "1855\n",
      "First match !  p2273s\n",
      "1856\n",
      "First match !  a1519t\n",
      "1857\n",
      "First match !  d2512g\n",
      "1858\n",
      "First match !  l1433s\n",
      "1859\n",
      "First match !  l2427r\n",
      "1860\n",
      "First match !  v2006l\n",
      "1861\n",
      "First match !  i2500f\n",
      "1862\n",
      "pos weak r2430m\n",
      "1863\n",
      "First match !  l2431p\n",
      "1864\n",
      "First match !  a8s\n",
      "1865\n",
      "First match !  r2505q\n",
      "1866\n",
      "First match !  i2500m\n",
      "1867\n",
      "First match !  e2419k\n",
      "1868\n",
      "First match !  t1977k\n",
      "1869\n",
      "First match !  f1888i\n",
      "1870\n",
      "First match !  c1483f\n",
      "1871\n",
      "First match !  w1456r\n",
      "1872\n",
      "First match !  v2006i\n",
      "1873\n",
      "First match !  a41p\n",
      "1874\n",
      "First match !  f1888v\n",
      "1875\n",
      "First match !  t1977r\n",
      "1876\n",
      "First match !  s2215t\n",
      "1877\n",
      "First match !  l2230v\n",
      "1878\n",
      "First match !  t1977i\n",
      "1879\n",
      "First match !  a1459p\n",
      "1880\n",
      "y371 aa_pos c1483w\n",
      "1881\n",
      "First match !  a2034v\n",
      "1882\n",
      "First match !  f1888l\n",
      "1883\n",
      "First match !  q2223k\n",
      "1884\n",
      "First match !  m2327i\n",
      "1885\n",
      "First match !  s2215f\n",
      "1886\n",
      "First match !  t1977s\n",
      "1887\n",
      "First match !  f2108l\n",
      "1888\n",
      "First match !  y1463s\n",
      "1889\n",
      "First match !  p2476l\n",
      "1890\n",
      "First match !  k1452n\n",
      "1891\n",
      "First match !  s2215y\n",
      "1892\n",
      "First match !  e2014k\n",
      "1893\n",
      "y371 aa_pos d2512y\n",
      "1894\n",
      "First match !  e1799k\n",
      "1895\n",
      "First match !  c1483y\n",
      "1896\n",
      "First match !  l1460p\n",
      "1897\n",
      "First match !  r2505*\n",
      "1898\n",
      "y371 aa_pos a41t\n",
      "1899\n",
      "First match !  c1483r\n",
      "1900\n",
      "First match !  deletion\n",
      "1901\n",
      "First match !  mll-tet1 fusion\n",
      "1902\n",
      "1903\n",
      "1904\n",
      "1905\n",
      "First match !  t910m\n",
      "1906\n",
      "1907\n",
      "First match !  g1232d\n",
      "1908\n",
      "First match !  truncating mutations\n",
      "1909\n",
      "First match !  deletion\n",
      "1910\n",
      "First match !  truncating mutations\n",
      "1911\n",
      "1912\n",
      "First match !  r514c\n",
      "1913\n",
      "First match !  e518k\n",
      "1914\n",
      "First match !  d384n\n",
      "1915\n",
      "First match !  v321m\n",
      "1916\n",
      "First match !  q477e\n",
      "1917\n",
      "y371 aa_pos d473g\n",
      "1918\n",
      "First match !  d473h\n",
      "1919\n",
      "First match !  h231r\n",
      "1920\n",
      "First match !  q635e\n",
      "1921\n",
      "First match !  n219d\n",
      "1922\n",
      "First match !  t241m\n",
      "1923\n",
      "pos weak l221r\n",
      "1924\n",
      "First match !  s387n\n",
      "1925\n",
      "First match !  i408v\n",
      "1926\n",
      "First match !  l412f\n",
      "1927\n",
      "First match !  w535l\n",
      "1928\n",
      "First match !  amplification\n",
      "1929\n",
      "First match !  f460l\n",
      "1930\n",
      "First match !  v414a\n",
      "1931\n",
      "First match !  a459v\n",
      "1932\n",
      "pos k575m\n",
      "1933\n",
      "First match !  e518a\n",
      "1934\n",
      "First match !  c49y\n",
      "1935\n",
      "First match !  f123i\n",
      "1936\n",
      "pos weak g116s\n",
      "1937\n",
      "First match !  g123s\n",
      "1938\n",
      "First match !  l232li\n",
      "1939\n",
      "First match !  l251p\n",
      "1940\n",
      "First match !  d387v\n",
      "1941\n",
      "First match !  l225li\n",
      "1942\n",
      "pos weak e127g\n",
      "1943\n",
      "pos s250p\n",
      "1944\n",
      "First match !  d401n\n",
      "1945\n",
      "First match !  e626k\n",
      "1946\n",
      "First match !  n81y\n",
      "1947\n",
      "First match !  d83v\n",
      "1948\n",
      "First match !  y69h\n",
      "1949\n",
      "First match !  e77k\n",
      "1950\n",
      "First match !  k4e\n",
      "1951\n",
      "1952\n",
      "First match !  t17a\n",
      "1953\n",
      "First match !  e1978*\n",
      "1954\n",
      "First match !  truncating mutations\n",
      "1955\n",
      "First match !  deletion\n",
      "1956\n",
      "First match !  n2875k\n",
      "1957\n",
      "First match !  d2870a\n",
      "1958\n",
      "First match !  d1853n\n",
      "1959\n",
      "aa_pos_aa g591v\n",
      "1960\n",
      "1961\n",
      "1962\n",
      "First match !  e322k\n",
      "1963\n",
      "First match !  amplification\n",
      "1964\n",
      "First match !  d32a\n",
      "1965\n",
      "First match !  s33f\n",
      "1966\n",
      "First match !  s37c\n",
      "1967\n",
      "First match !  g34e\n",
      "1968\n",
      "First match !  d32n\n",
      "1969\n",
      "First match !  k335i\n",
      "1970\n",
      "First match !  s33y\n",
      "1971\n",
      "First match !  s45a\n",
      "1972\n",
      "First match !  s45y\n",
      "1973\n",
      "First match !  s33c\n",
      "1974\n",
      "First match !  s37f\n",
      "1975\n",
      "First match !  s37y\n",
      "1976\n",
      "First match !  h36p\n",
      "1977\n",
      "First match !  s37a\n",
      "1978\n",
      "First match !  g34v\n",
      "1979\n",
      "1980\n",
      "First match !  t41a\n",
      "1981\n",
      "First match !  s33a\n",
      "1982\n",
      "First match !  s45f\n",
      "1983\n",
      "1984\n",
      "First match !  n387k\n",
      "1985\n",
      "First match !  i35s\n",
      "1986\n",
      "First match !  p44a\n",
      "1987\n",
      "First match !  s45p\n",
      "1988\n",
      "pos s23r\n",
      "1989\n",
      "pos t28i\n",
      "1990\n",
      "y371 aa_pos l115r\n",
      "1991\n",
      "First match !  c121s\n",
      "1992\n",
      "First match !  k57t\n",
      "1993\n",
      "First match !  i204t\n",
      "1994\n",
      "pos weak i111a\n",
      "1995\n",
      "First match !  r49l\n",
      "1996\n",
      "First match !  p387s\n",
      "1997\n",
      "First match !  f129l\n",
      "1998\n",
      "First match !  q56_v60del\n",
      "1999\n",
      "pos weak s123t\n",
      "2000\n",
      "First match !  q56p\n",
      "2001\n",
      "First match !  f53c\n",
      "2002\n",
      "First match !  k97m\n",
      "2003\n",
      "First match !  e102_i103del\n",
      "2004\n",
      "y371 aa_pos v60m\n",
      "2005\n",
      "First match !  k59del\n",
      "2006\n",
      "pos d65n\n",
      "2007\n",
      "First match !  amplification\n",
      "2008\n",
      "First match !  q58_e62del\n",
      "2009\n",
      "First match !  y130c\n",
      "2010\n",
      "First match !  f53l\n",
      "2011\n",
      "pos weak i111p\n",
      "2012\n",
      "First match !  p124l\n",
      "2013\n",
      "First match !  i103n\n",
      "2014\n",
      "First match !  i111n\n",
      "2015\n",
      "First match !  p124q\n",
      "2016\n",
      "First match !  v60e\n",
      "2017\n",
      "First match !  v211d\n",
      "2018\n",
      "pos weak e120q\n",
      "2019\n",
      "First match !  f53s\n",
      "2020\n",
      "First match !  g128v\n",
      "2021\n",
      "First match !  p306h\n",
      "2022\n",
      "pos weak i111r\n",
      "2023\n",
      "First match !  e203k\n",
      "2024\n",
      "2025\n",
      "First match !  k57e\n",
      "2026\n",
      "First match !  n382h\n",
      "2027\n",
      "First match !  p124s\n",
      "2028\n",
      "pos l37p\n",
      "2029\n",
      "First match !  r47q\n",
      "2030\n",
      "First match !  s222d\n",
      "2031\n",
      "First match !  d67n\n",
      "2032\n",
      "First match !  s226d\n",
      "2033\n",
      "First match !  f57v\n",
      "2034\n",
      "First match !  f57l\n",
      "2035\n",
      "First match !  k101m\n",
      "2036\n",
      "First match !  l46f\n",
      "2037\n",
      "First match !  n126d\n",
      "2038\n",
      "First match !  c125s\n",
      "2039\n",
      "First match !  f57c\n",
      "2040\n",
      "First match !  s222d\n",
      "2041\n",
      "First match !  e207k\n",
      "2042\n",
      "First match !  v35m\n",
      "2043\n",
      "First match !  truncating mutations\n",
      "2044\n",
      "2045\n",
      "First match !  m269r\n",
      "2046\n",
      "First match !  r552s\n",
      "2047\n",
      "First match !  r552g\n",
      "2048\n",
      "First match !  g434r\n",
      "2049\n",
      "First match !  e846k\n",
      "2050\n",
      "First match !  n233y\n",
      "2051\n",
      "pos t73i\n",
      "2052\n",
      "FUSION pvt1-myc fusion\n",
      "2053\n",
      "FUSION igl-myc fusion\n",
      "2054\n",
      "First match !  myc-nick\n",
      "2055\n",
      "First match !  fusions\n",
      "2056\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-16-0534602b3241>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mclean_match_data\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mconcatenate_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mextract_match\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mclean_match\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mclean_match_data\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m\"Text\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"Score\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36mapply\u001b[0;34m(self, func, axis, raw, result_type, args, **kwargs)\u001b[0m\n\u001b[1;32m   8739\u001b[0m             \u001b[0mkwargs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   8740\u001b[0m         )\n\u001b[0;32m-> 8741\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   8742\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   8743\u001b[0m     def applymap(\n",
      "\u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/apply.py\u001b[0m in \u001b[0;36mapply\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    686\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_raw\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    687\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 688\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_standard\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    689\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    690\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0magg\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/apply.py\u001b[0m in \u001b[0;36mapply_standard\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    810\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    811\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mapply_standard\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 812\u001b[0;31m         \u001b[0mresults\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mres_index\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_series_generator\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    813\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    814\u001b[0m         \u001b[0;31m# wrap results\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/apply.py\u001b[0m in \u001b[0;36mapply_series_generator\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    826\u001b[0m             \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mv\u001b[0m \u001b[0;32min\u001b[0m \u001b[0menumerate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseries_gen\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    827\u001b[0m                 \u001b[0;31m# ignore SettingWithCopy here in case the user mutates\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 828\u001b[0;31m                 \u001b[0mresults\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mv\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    829\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresults\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mABCSeries\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    830\u001b[0m                     \u001b[0;31m# If we have a view on v, we need to make a copy because\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-16-0534602b3241>\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(x)\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mclean_match_data\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mconcatenate_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mextract_match\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mclean_match\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mclean_match_data\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0;34m\"Text\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"Score\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-15-d6e1ba8e965f>\u001b[0m in \u001b[0;36mextract_match\u001b[0;34m(line)\u001b[0m\n\u001b[1;32m     64\u001b[0m         \u001b[0;31m# Try to match the word fusion at least..\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     65\u001b[0m         \u001b[0;31m# Quality score 4 (bad)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 66\u001b[0;31m         \u001b[0mmatch_fusion\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfind_match\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtext\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\"fusion\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     67\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmatch_fusion\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     68\u001b[0m             \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"FUSION\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvariation\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-15-d6e1ba8e965f>\u001b[0m in \u001b[0;36mfind_match\u001b[0;34m(text, word)\u001b[0m\n\u001b[1;32m     29\u001b[0m     \u001b[0mbefore_after_target\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"([^.]*\\.){0,1}\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     30\u001b[0m     \u001b[0mmatch_exp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mre\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcompile\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbefore_after_target\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mtarget_sentence\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mbefore_after_target\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 31\u001b[0;31m     \u001b[0mmatch_text\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmatch_exp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfindall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mclean\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     32\u001b[0m     \u001b[0mfinal_match\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmap\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mjoin_tuple_string\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmatch_text\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     33\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mfinal_match\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "clean_match_data = concatenate_data.apply(lambda x: extract_match(x), axis = 1)\n",
    "clean_match = pd.DataFrame(list(clean_match_data), columns = [\"Text\",\"Score\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 507,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Text</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>here, we deorphancdk10 by tifying cyclin m, th...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>the c-cbl loh also posity correlated with egf...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>the c-cbl loh also posity correlated with egf...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>most of the changes were novel, although 4 ca...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>all mutations from the second group were pred...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3311</th>\n",
       "      <td>21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutatio...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3312</th>\n",
       "      <td>21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutatio...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3313</th>\n",
       "      <td>lpxn is preferentially expressed in hematopoi...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3314</th>\n",
       "      <td>) conversely, mutations mapping to the first d...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3315</th>\n",
       "      <td>mutation analysis in fpd/aml pedigrees. elect...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3316 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Text  Score\n",
       "0     here, we deorphancdk10 by tifying cyclin m, th...      3\n",
       "1      the c-cbl loh also posity correlated with egf...      1\n",
       "2      the c-cbl loh also posity correlated with egf...      1\n",
       "3      most of the changes were novel, although 4 ca...      1\n",
       "4      all mutations from the second group were pred...      1\n",
       "...                                                 ...    ...\n",
       "3311  21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutatio...      1\n",
       "3312  21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutatio...      3\n",
       "3313   lpxn is preferentially expressed in hematopoi...      1\n",
       "3314  ) conversely, mutations mapping to the first d...      1\n",
       "3315   mutation analysis in fpd/aml pedigrees. elect...      1\n",
       "\n",
       "[3316 rows x 2 columns]"
      ]
     },
     "execution_count": 507,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clean_match"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Explications du problème\n",
    "Fonction poubelle juste pour voir ceux qui sont pas détectable --> len(match) == 0\n",
    "\n",
    "Si ils sont pas détectable, c'est que le nom donné pour leurs variation sont pas dans l'article, donc je vérfie à la main sur le site https://regex101.com/r/lvQZIf/1 qui sert à visualiser les expressions régulières.\n",
    "En copiant collant tout l'article, je regarde comment je peux détecter la variation avec un autre nom ducoup, comme le nom de base ne marche pas.\n",
    "\n",
    "Par exemple Id = 19, y371s n'est pas dans l'article mais il y est sous la forme tyr371ser.\n",
    "\n",
    "Cet regexpr ne marche pas (le texte est en minuscule) -> `([^.]*\\.){0,1}([^.]*y371s[^.]*\\.)([^.]*\\.){0,2}`\n",
    "\n",
    "Mais celle la marchera -> `([^.]*\\.){0,1}([^.]*tyr371ser[^.]*\\.)([^.]*\\.){0,2}` (ça donne 3 match sur le site pour l'article id = 19)\n",
    "\n",
    "(pour les regex, il faut juste changer le nom au milieu après `*` et avant `[^.]`)\n",
    "\n",
    "Autre exemple, ID = 102, mutation r922* qui n'est mentionné sous aucunes formes, ni même la position 922. Mais, il y a des mots qui peuvent faire penser à cette mutation du genre \"nonsense mutations\" car l'étoile (\\*) veut dire que r922 est devenu un codon stop, donc c'est une mutation non sens. Faudrait voir si je peux convertir ces mutations en mot pour faciliter la détection, ou utiliser un champ lexical.\n",
    "\n",
    "\n",
    "Donc là, j'essaie de faire plus ou moins au cas par cas (pas tous, y en a 880...) pour déterminer des groupes de variations indétectable pour revoir la détection.\n",
    "Par exemple, pour les variations de type `y371s`, je peux prendre `y371`, ou tout simplement `371` qui peuvent détecter la variation dans certain cas.\n",
    "\n",
    "\n",
    "Par contre, pour ceux où y a vraiment pas d'infos dans l'article avec la variation, du genre `t1365m` ID = 44; je sais pas quoi faire d'eux :)\n",
    "Car la position 1365 n'est même pas mentionné dans l'article, donc je vois pas comment détecter les infos liées à cette mutation...\n",
    "\n",
    "Soit je garde l'article entier, soit je prend le nom du gène pour la détection, mais le pb avec ça, c'est que ducoup tout ceux qui sont indétectables de ce gène là, auront les mêmes lignes de détecté car je détecterai uniquement en utilisant le nom du gène, donc y aura pas de différences entre eux même si la variation diffère.\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "## Ne lance pas la cellule juste en bas, ça prend un certain temps et de toute façon, tu as déjà la sortie avec tout ceux qui ne marchent pas.\n",
    "\n",
    "La sortie est de la forme : \n",
    "`variation | id | len(article_original) | len(match)`\n",
    "\n",
    "Certaines variations sont des mots complets du genre `truncating mutations`, un peu relou pour la détection, ce qu'on pourrait faire, c'est utiliser des mots qui ressemble à truncating grace à des méthodes de traitements de texte qui va prendre que le préfix par exemple et automatiser ça pour les mots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def check_line(x):\n",
    "    extract = extract_match(x)\n",
    "    score = extract[1]\n",
    "    if score == 0:\n",
    "        print(x[\"Variation\"].lower(), x.name, len(x[\"Text\"]))\n",
    "\n",
    "        \n",
    "concatenate_data.apply(lambda x : check_line(x) , axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Variable ID pour faciliter le print des deux cellules suivantes pour choper le text, la variation et le gene d'un ID et les copier coller sur le site regex\n",
    "\n",
    "Pour la détection, je teste juste avec le nom de la variation, puis vraiment si y a rien, je regarde du coté du nom du gène"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 553,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "ID = 3311"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 555,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"the aml1/evi-1 chimeric gene is generated by the t(3;21)(q26;q22) translocation and plays a pivotal role in progression of hematopoietic stem cell malignancies such as chronic myelocytic leukemia and myelodysplastic syndrome. in aml1/evi-1, an n-terminal half of aml1 including a runt homology domain is fused to the entire zinc-finger evi-1 protein. aml1 is essential for hematopoietic cell development in fetal liver and its lineage-specific differentiation in adult. in contrast, evi-1 is barely expressed in normal hematopoietic cells, but it is overexpressed in chronic myelocytic leukemia in blastic crisis and myelodysplastic syndrome-derived leukemia. there are at least four mechanisms identified in aml1/evi-1 fusion protein that possibly lead into malignant transformation of hematopoietic stem cells. firstly, aml1/evi-1 exerts dominant-negative effects over aml1-induced transcriptional activation. although target genes repressed by aml1/evi-1 are still not known, binding competition to a specific dna sequence and histone deacetylase recruitment through a co-repressor ctbp in evi-1 part are conceivable underlying mechanisms for the dominant-negative effects. secondly, aml1/evi-1 interferes with tgfbeta signaling and antagonizes the growth-inhibitory effects of tgfbeta. the first zinc-finger domain of evi-1 associates with smad3, a tgfbeta signal transducer, and represses its transcriptional activity by recruiting histone deacetylase through ctbp that interacts with evi-1. thirdly, aml1/evi-1 blocks jnk activity and prevents stress-induced apoptosis. aml1/evi-1 associates with jnk through the first zinc-finger domain of evi-1 and disturbs the association between jnk and its substrates. lastly, aml1/evi-1 enhances ap-1 activity by activating the c-fos promoter depending on the second zinc-finger domain of evi-1, and promotes cell proliferation. all these functions cooperatively contribute to the malignant transformation of the hematopoietic stem cells by aml1/evi-1.  keywords: aml1/evi-1, apoptosis, dominant-negative effects, jnk, tgfbeta, transcription factor top of page introduction in more than the past 10 years, most of the recurrent chromosomal abnormalities associated with specific subtypes of human leukemia have been molecularly characterized. these studies advanced our understanding in molecular mechanisms of leukemogenesis and provided deep insights for hematopoietic cell development. in particular, reciprocal translocations involving transcription factor-encoding genes play an important role in leukemogenesis through either inappropriate expression of a transcription factor or pathogenic generation of a chimeric transcription molecule. hirai and his colleagues at the university of tokyo have cloned several chimeric genes, including the aml1/evi-1 gene generated by the t(3;21)(q26;q22) translocation (mitani et al., 1994), and established a number of unique molecular models for leukemogenesis induced by this molecule. in this short review, i will focus on the molecular mechanisms of leukemogenesis by aml1/evi-1 chimeric protein, which have been clarified in his laboratory. i dedicate this review to late dr hisamaru hirai who had been my reliable supervisor as well as helpful mentor, but regrettably passed away on august 23, 2003.  top of page cloning of the aml1/evi-1 gene we first established skh1 leukemia cell line from a patient with chronic myelocytic leukemia (cml) in megakaryoblastic crisis acquiring the t(3;21)(q26;q22) translocation in addition to the t(9;22)(q34;q11) translocation. the aml1/evi-1 fusion gene generated by the t(3;21)(q26;q22) translocation was cloned from this cell line 10 years ago. the t(3;21)(q26;q22) translocation fuses the aml1 gene on 21q22 and the evi-1 gene on 3q26. the t(3;21) (q26;q22) translocation is occasionally observed in cml in blastic crisis or myelodysplastic (mds)-derived leukemia. since it is very rare that de novo acute leukemia carries the t(3;21)(q26;q22) translocation, the appearance of this chromosomal abnormality may trigger the transformation of chronic hematopoietic stem cell disorder into acute leukemic phase. actually, treatment with antisense oligonucleotide complementary to the coding sequence of the aml1/evi-1 junction markedly inhibits the growth of skh1 cells that carry both the t(3;21) (q26;q22) and the t(9;22) (q34;q11) translocations, but does not inhibit the growth of k562 cells that carry only the t(9;22) (q34;q11) translocation (mitani et al., 1995). thus, the aml1/evi-1 fusion protein could maintain proliferation capacity of the t(3;21)-carrying clones arising after blastic transformation of cml.  top of page structure of aml1/evi-1 the aml1 (now referred to runx1) gene was originally isolated as a gene that is located at the translocation breakpoint of chromosome 21 in the t(8;21)(q22;q22) translocation specifically found in acute myeloblastic leukemia (aml) (m2 according to the french–american–british classification) (miyoshi et al., 1991). aml1 has a runt homology domain (rhd) at the n-terminus that is highly homologous to the product of drosophila segmentation gene runt and alpha subunit of polyomavirus enhancer-binding protein 2 (pebp2 or pae2) (daga et al., 1992; bae et al., 1993; ogawa et al., 1993b) (figure 1). the aml1 gene is thought to be ubiquitously expressed in multiple hematopoietic lineages because human leukemia cell lines derived from myeloid, b and t lymphoid cells show normal aml1 transcripts (miyoshi et al., 1991). aml1 heterodimerizes with non-dna-binding subunit pebp2beta to form an active dna-binding complex and binds to a specific dna consensus sequence named the pebp2 site that contains r/taccrac through rhd (ogawa et al., 1993a, 1993b). pebp2beta enhances the dna-binding activity of aml1 and protects aml1 from ubiquitin-proteasome-mediated degradation (huang et al., 2001). the proline–serine–threonine (pst) domain at the c-terminus is required for transcriptional activation. aml1-mediated transcription depends on direct binding of transcriptional co-activators p300, cbp and p/caf that have intrinsic histone acetyltransferase (hat) activity (kitabayashi et al., 1998). aml1 is indispensable for expression of a number of hematopoietic lineage-specific genes, including the myeloperoxidase (mpo), macrophage colony-stimulating factor (m-csf) receptor, granulocyte-macrophage colony-stimulating factor (gm-csf), interleukin-3 (il-3), t-cell receptor (tcr) and np-3 genes. gene knockout experiments show that null mutation in either aml1 (okuda et al., 1996) or pebp2beta (sasaki et al., 1996; wang et al., 1996a, 1996b) is embryonic lethal at e12.5 due to complete lack of fetal liver hematopoiesis and lethal central nervous hemorrhages, although primitive yolk sac erythropoiesis appears normal. using conditional gene inactivation strategy, aml1 is shown to play a role in t-cell development (taniuchi et al., 2002). thus, aml1 is a key transcription factor for the hematopoietic cell development, proliferation and differentiation.  figure 1. figure 1 - unfortunately we are unable to provide accessible alternative text for this. if you require assistance to access this image, please contact help@nature.com or the author schematic structure of wild-type aml1, evi-1 and aml1/evi-1. wild-type aml1 possesses rhd at the n-terminus and pst region at the c-terminus. in aml1/evi-1, n-terminal aml1 sequences are abruptly interrupted at the end of the rhd and followed by almost the entire coding region of evi-1  full figure and legend (74k)  the evi-1 (ecotropic viral integration site) gene was initially identified as a common locus of retrovirus integration in myeloid tumors in akxd mouse (morishita et al., 1988). evi-1 is a transcriptional regulator that possesses two cys2his2-type zinc-finger domains and acidic domain (morishita et al., 1990). expression of evi-1 is barely detectable in healthy murine or human hematopoietic cells. however, evi-1 is highly expressed in mds or aml patients showing 3q26 abnormalities such as inv(3)(q21q26) or t(3;3)(q21;q26) (morishita et al., 1992). even in the absence of evident 3q36 abnormalities, elevated expression of evi-1 is reported in cml in blastic crisis and mds-derived leukemia (ogawa et al., 1996a). these findings suggest a critical role for evi-1 in malignant transformation of hematopoietic cells as a dominant oncogene.  in aml1/evi-1 chimeric cdna, an open reading frame of 4185 nucleotides encodes a 1395 amino-acid protein (mitani et al., 1994). the n-terminal portion of aml1 is abruptly interrupted at the end of rhd, followed by a 5'non-coding sequence of evi-1 cdna that is translated through the entire coding region. therefore, aml1/evi-1 fusion protein is a chimeric transcription molecule that consists of rhd of aml1 and two zinc-finger domains of evi-1. in skh1 cells, two major transcripts in sizes of 8.2 and 7.0 kb are transcribed from the aml1/evi-1 chimeric gene, and western analysis shows that aml1/evi-1 is a 180 kda chimeric molecule. aml1/evi-1 promotes pre-existing stem cell disorder into acute leukemia through multiple mechanisms. i will describe the unique functions of each dna-binding domain of aml1/evi-1 fusion molecule in detail below.  top of page dominant-negative effects over aml1 aml1 transactivates tww-tk-luc reporter that contains pebp2 sites derived from the tcrbeta promoter (tanaka et al., 1995). however, aml1/evi-1 does not have such transactivation ability, presumably because it does not contain the pst domain that is necessary for the reporter transactivation. notably, aml1/evi-1 reduces the reporter transactivation by aml1 in a dose-dependent manner. the gel shift assay shows that aml1/evi-1 binds to the pebp2 site more tightly than aml1 does. therefore, aml1/evi-1 could dominantly interfere with the aml1-induced transactivation by competing for the specific dna binding. deletion mutation analysis of aml1/evi-1 shows that rhd is responsible for the dominant-negative effect. to exert transcription activation, aml1 needs to heterodimerize with pebp2beta through its rhd. aml1 mainly locates in the nucleus, while pebp2beta locates in the cytoplasm (tanaka et al., 1998). immunofluorescence analysis shows that aml1/evi-1 locates in the nucleus with or without rhd. aml1/evi-1 seems to drag pebp2beta into the nucleus because co-expression of the full-length aml1/evi-1 relocates pebp2beta from the cytoplasm to the nucleus. this effect is rhd-dependent because aml1/evi-1 lacking rhd does not have such effect, which is consistent with the fact that aml1 associates with pebp2beta through rhd. since pebp2beta more effectively associates with aml1/evi-1 than aml1, pebp2beta more efficiently translocates into the nucleus by aml1/evi-1 than by aml1. this differential effect between aml1/evi-1 and aml1 presumably causes the dominant-negative effect of aml1/evi-1 over aml1 and accounts for one of the mechanisms through which this chimeric protein contributes to leukemogenesis.  recently, we found that aml1/evi-1 interacts with c-terminal binding protein (ctbp), which is essential for repressing the aml1-induced transactivation (izutsu et al., 2002) (figure 2). ctbp was originally identified as a protein that interacts with a c-terminal portion of adenovirus e1a protein. to date, two highly related proteins, termed ctbp1 and ctbp2, have been identified in both mice and humans. ctbp belongs to a member of co-repressor proteins that mediates repression by associating with several transcription factors, including krüppel-like factor (bklf), friend of gata (fog) and t-cell factor (tcf). although it is not clear how ctbp mediates transcriptional repression, it is supposed that histone deacetylase (hdac)-1 that interacts with ctbp is involved in the process. we previously determined that evi-1 interacts with ctbp exclusively through one of the two potential ctbp-binding amino-acid sequences (n-terminal pfdlt and c-terminal pldls) that locate next to the second zinc-finger domain (izutsu et al., 2001). these two sequences are conserved in aml1/evi-1 and one of them mediates interaction with ctbp. aml1/evi-1 interferes with the aml1-induced transactivation, while aml1/evi-1 with mutations in pldls does not. therefore, aml1/evi-1 requires ctbp as a co-repressor to dominantly inhibit the aml1-induced transactivation. an hdac inhibitor trichostatin a relieves this dominant-negative effect of aml1/evi-1. therefore, it is conceivable that aml1/evi-1 positively prevents the aml1-induced transcription by recruiting hdac through ctbp.  figure 2. figure 2 - unfortunately we are unable to provide accessible alternative text for this. if you require assistance to access this image, please contact help@nature.com or the author aml1/evi-1 exerts dominant-negative effects over wild-type- aml1. aml1 becomes an active transcription factor by heterodimerizing with pebp2beta through rhd. pebp2beta increases the dna-binding ability of aml1 and protects aml1 from the ubiquitin system-mediated degradation. as aml1/evi-1 binds to pebp2beta more tightly than aml1, aml1/evi-1 competes aml1 out from the pebp2 sites. moreover, aml1/evi-1 associates with a co-repressor ctbp and thereby recruits hdac. aml1/evi-1 actively represses the transcription of potential aml1 target genes. ctbp, c-terminal binding protein; hdac, histone deacetylase; pebp2, polyomavirus enhancer binding protein 2  full figure and legend (124k)  aml1/evi-1 blocks granulocytic differentiation of a murine myeloid cell line 32dcl3 induced by g-csf (tanaka et al., 1995). parental 32dcl3 cells acquire mature granulocytic phenotype characterized by cytoplasmic granules and segmented/circular nucleus when cultured with g-csf for several days. a robust induction of mpo mrna is also observed by northern analysis. 32dcl3 cells overexpressing aml1/evi-1 maintain immature morphological phenotypes characterized by a large unsegmented nucleus even in the presence of g-csf (izutsu et al., 2002). in parallel, there is poor induction of mpo mrna in these cells. thus, overexpression of aml1/evi-1 blocks g-csf-induced differentiation of 32dcl3 cells into mature granulocytic phenotype. on the contrary, in 32dcl3 cells overexpressing mutant aml1/evi-1 that is unable to interact with ctbp, the morphological changes and induction of mpo mrna are indistinguishable from those of parental 32dcl3 cells. these results suggest that aml1/evi-1 blocks g-csf-induced granulocytic maturation by interacting with ctbp. taken together, ctbp-mediated repression of gene transcription could be one of the mechanisms for aml1/evi-1-mediated block in granulocyte differentiation.  top of page repression of transforming growth factor bold italic beta (tgfbold italic beta)-mediated growth inhibition cell growth and differentiation are tightly regulated by delicate balance of growth factors and growth-inhibitory factors. tgfbeta inhibits proliferation of a wide range of cell types including epithelial, endothelial and hematopoietic cells. binding of tgfbeta to the heteromeric serine/threonine kinase receptor complex leads to direct phosphorylation of intracellular signal transducers smad2 and smad3, followed by the formation of heteromeric complexes with a common signal transducer smad4, and subsequent their translocation into the nucleus. once in the nucleus, smad complexes are thought to act as a transcriptional activator to inhibit cellular proliferation. we demonstrated that evi-1 perturbs tgfbeta signaling and inhibits proliferation of a wide variety of cells. evi-1 represses transcriptional activation of tgfbeta-responsive reporters such as the plasminogen-activator inhibitor-1 (pai-1) or p15 promoter-luciferase construct (kurokawa et al., 1998a). there is no effect of evi-1 on the basal activity of these promoters, indicating that this inhibitory effect works specifically toward tgfbeta-induced transcriptional activation. in growth inhibition experiment, evi-1 antagonizes the anti-proliferative effects of tgfbeta against tgfbeta-sensitive mv1lu cells. biochemically, most of retinoblastoma (rb) proteins in the evi-1-expressing cells remain in the hyperphosphorylated forms even in the presence of tgfbeta, whereas the control cells treated with tgfbeta accumulate hypophosphorylated rb and show a decreased level of rb protein. the first zinc-finger domain of evi-1 mediates interaction with smad3, and the small region next to the second zinc-finger domain (repression domain) is required to repress tgfbeta signaling. the association between evi-1 and ctbp is essential for the inhibition of pai-1 reporter construct. trichostatin a treatment alleviates evi-1-mediated repression of tgfbeta-responsive reporters, suggesting that hdac is involved in this repression. consistent with this, mv1lu cells overexpressing evi-1 with a mutated pldls show normal growth inhibition as parental mv1lu cells when treated with tgfbeta.  similar to evi-1, aml1/evi-1 inhibits transactivation of tgfbeta-responsive promoters (kurokawa et al., 1998b). the constitutive expression of aml1/evi-1 or evi-1 in 32dcl3 cells overcomes tgfbeta-mediated inhibition of cell growth. from these data, aml1/evi-1 and evi-1 can potentially block growth inhibition of hematopoietic cells mediated by tgfbeta. molm-1 is a human megakaryoblastoid cell line carrying the inv(3)(q21q26) and endogenously expressing a truncated form of evi-1 proteins in which the c-terminal amino acids of wild-type evi-1 are replaced by five amino acids (ogawa et al., 1996b). evi-1 in molm-1 cells retain both the first zinc-finger and the repression domains that are required for repression of tgfbeta signaling. thus, molm-1 cells do not respond to tgfbeta-mediated growth inhibition. however, it becomes tgfbeta sensitive when treated with antisense oligonucleotides complementary to the sequence encoding the n-terminus of the first zinc-finger domain of evi-1. just like wild-type-evi-1, aml1/evi-1 physically interacts with smad3 and inhibits the smad3 activity (figure 3). aml1/evi-1 also associates with ctbp through the ctbp-binding consensus sequence pldls. as endogenous ctbp proteins are detected in skh1 cells, it could be possible that tgfbeta signaling is repressed in this leukemia cell line. while the interaction of ctbp and hdac1 is clearly demonstrated, association between aml1/evi-1 and hadc1 is not yet determined. however, it is speculated that aml1/evi-1 recruits hdacs through ctbp to repress tgfbeta-responsive transcription. thus, in leukemia with the t(3;21) translocation, aml1/evi-1 may contribute to leukemogenesis by blocking the growth-inhibitory effect of tgfbeta.  figure 3. figure 3 - unfortunately we are unable to provide accessible alternative text for this. if you require assistance to access this image, please contact help@nature.com or the author aml1/evi-1 represses tgfbeta-mediated growth inhibitory signal. (a) upon tgfbeta binding to the heteromeric serine/threonine kinase receptor complex, type ii receptor activates type i receptor and thereby phosphorylates smad3. phosphorylated smad3 forms a complex with smad4 and translocates into the nucleus, where they work as a transcription factor to inhibit cellular growth. (b) aml1/evi-1 binds to smad3 through the first zinc-finger domain in evi-1 and actively interferes with smad3 transcriptional activity. recruitment of hdac by evi-1 via ctbp is an underlying mechanism. tgfbeta transforming growth factor beta; r i, type i receptor; r ii, type ii receptor; ctbp, c-terminal binding protein; hdac, histone deacetylase  full figure and legend (114k)  top of page inhibition of c-jun n-terminal kinase mitogen-activated protein (map) kinase cascades are important signaling pathways that are involved in a wide variety of biological response mechanisms. in vertebrates, at least three pathways have been identified: erk, c-jun n-terminal kinase (jnk) and p38. generally, erk signaling is involved in controlling cell proliferation and differentiation, while jnk or p38 signaling plays an important role in triggering apoptosis in response to cellular stresses such as uv light, italic gamma-radiation, osmotic shock, protein synthesis inhibitors, tumor necrosis factor-alpha (tnfalpha) and interleukin-1. evi-1 acts as an inhibitor of jnk, either when overexpressed jnk is unstimulated or stimulated by uv light, anisomycin, sorbitol and tnfalpha (kurokawa et al., 2000). however, evi-1 does not affect the kinase activities of overexpressed erk and p38. furthermore, rat1 cells overexpressing evi-1 show downregulated endogenous jnk activity. in molm-1 cells, the endogenous jnk activity is reduced. however, when treated with evi-1 antisense oligonucleotide, the jnk activity is restored. the same results are obtained in the experiment using human endometrial carcinoma cell line hec1b that expresses evi-1 at a high level. consistently, the presence of evi-1 actually reduces phosphorylated substrates of c-jun. jnk activation requires phosphorylation on two specific amino-acid residues thr183 and tyr185 by mmk4 or mmk7. although evi-1 physically interacts with jnk, it does not affect its phosphorylation status. however, the binding between jnk and c-jun is significantly reduced by concomitant expression of evi-1. this suggests that evi-1 inhibits c-jun phosphorylation by inhibiting the association between jnk and c-jun (figure 4). evi-1 associates jnk through the first zinc-finger domain. this interaction is essential for inhibition of jnk activity. evi-1 constitutively associates with jnk in hec1b cells, and this interaction is increased by the treatment with uv light.  figure 4. figure 4 - unfortunately we are unable to provide accessible alternative text for this. if you require assistance to access this image, please contact help@nature.com or the author evi-1 inhibits jnk activity. (a) various kinds of stresses such as uv light, italic gamma-radiation, osmotic shock, heat shock and protein synthesis inhibitor activate jnk by phosphorylation. activated jnk translocates into the nucleus, phosphorylates transcription factors such as jun and triggers apoptosis induction. (b) evi-1 blocks association between jnk and its substrates and thereby represses jnk activity to induce apoptosis. jnk, c-jun n-terminal kinase  full figure and legend (131k)  we also demonstrated that evi-1 inhibits apoptotic cell death by interfering with jnk signaling pathway. 293 cells undergo apoptosis via jnk signaling induced by uv light. overexpression of evi-1 in 293 cells significantly represses apoptosis induction by uv light. on the other hand, hec1b cells are resistant to apoptosis, presumably because they express endogenous evi-1. however, hec1b cells treated with evi-1-specific antisense oligonucleotide easily undergo apoptosis. stress activation of jnk promotes upregulation of fasl expression in t lymphocytes, which is one of the mechanisms potentially causing apoptosis. fasl expression induced by uv stimulation in jurkat cells is prevented by evi-1 depending on the first zinc-finger domain. evi-1 actually blocks apoptosis induction by uv light, and represses activation of fasl reporter stimulated by uv light and anisomycin in jurkat cells. furthermore, evi-1 inhibits apoptosis induction by tnfalpha in u937 cells by repressing jnk activity. in conclusion, evi-1 blocks both the molecular and biological activities of jnk.  it has not been determined whether aml1/evi-1 possesses the similar anti-apoptotic effect as evi-1 does. however, because this function is dependent on the first zinc-finger domain in evi-1, it is reasonable to speculate that aml1/evi-1 also prevents apoptosis induction by inhibiting jnk.  top of page stimulation of proliferation ap-1 (fos/jun heterodimer or jun/jun homodimer) is activated by growth stimuli, including growth factors, phorbol esters such as 12-o-tetradecanoylphorbol 13 -acetate (tpa) and various oncogene products, and stimulates transactivation through tpa responsive site (tre). ap-1 functions as a positive or negative regulator in a variety of cellular differentiation and proliferation processes. evi-1 raises ap-1 activity in nih3t3 cells and embryonal carcinoma (ec) cell line p19 (tanaka et al., 1994). evi-1-transfected p19 cells show differentiated phenotypes, characterized by flattened and enlarged morphology. these changes are indistinguishable from the morphological changes seen in p19 cells treated with retinoic acid or transfected with c-jun. the stage-specific embryonic antigen ssea-1, known as a stem cell marker in ec cells, is detected in parental p19 cells, but not in evi-1-transfected p19 cells. in contrast to ssea-1, the heat shock protein hsp47, known as a differentiation marker, is detected in evi-1-transfected p19 cells, but not in parental cells. in addition, expression of c-jun and c-fos is increased in evi-1-tansfected cells. evi-1 enhances the activity of the c-fos promoter in nih3t3 and p19 cells, depending on the second zinc-finger domain (figure 5). it is shown that the second zinc-finger domain is essential for both the activation of ap-1 and transactivation of the c-fos promoter.  figure 5. figure 5 - unfortunately we are unable to provide accessible alternative text for this. if you require assistance to access this image, please contact help@nature.com or the author evi-1 stimulates ap-1 activity. evi-1 activates c-fos transcription. fos associates with jun to form a heterodimer and stimulates tre-mediated transcription. increased ap-1 activity by evi-1 promotes cellular proliferation. erk, extracellular-regulated kinase; tcf, ternary complex factor; srf, serum response factor; tre, tpa responsive element; ap-1, activation protein 1  full figure and legend (84k)  in order to evaluate the transforming activity of aml1/evi-1, the fusion cdna is introduced into rat1 fibroblasts (kurokawa et al., 1995). rat1 cells expressing aml1/evi-1 form macroscopic colonies in soft agar, while the mock-transfected cells produce tiny, barely macroscopic ones. this indicates that aml1/evi-1 is a transforming gene. introduction of aml1/evi-1 into the rat1 clones harboring bcr/abl confers enhanced capacity for anchorage-independent growth. aml1/evi-1 also stimulates ap-1 activity. as the second zinc-finger domain is required for both transformation of rat1 cells and an increase in ap-1 activity, it is conceivable that aml1/evi-1 transforms rat1 cells by ap-1 activation. notably, aml1/evi-1-transformed colonies express c-jun mrna. since aml1/evi-1 exaggerates the transforming activity of bcr/abl, aml1/evi-1 could perform a critical role in leukemic progression of cml.  top of page conclusion similar to other aml1-related chimeras, aml1/evi-1 exerts dominant-negative effects over aml1-induced transcription. however, target genes that may have a key role in leukemogenesis are still not elucidated. on the other hand, almost the entire coding region of evi-1 is expressed under the aml1 promoter in leukemic cells carrying aml1/evi-1 chimeric gene. since ectopic expression of evi-1 is thought to be involved in leukemic transformation of cml and mds, evi-1 portion in aml1/evi-1 fusion molecule may also play a critical role in leukemogenesis. notably, aml1/evi-1 and evi-1 share similar functions; the first zinc-finger and the repression domains exhibit anti-growth repression effect by blocking tgfbeta signaling, the first zinc-finger domain exhibits anti-apoptotic effect by repressing jnk signaling, and the second zinc-finger domain exhibits proliferation stimulation effect by increasing ap-1 activity. all these effects are related to leukemic cell proliferation, directly or indirectly. the molecular mechanisms of leukemogenesis by aml1/evi-1 are summarized in figure 6. the multi-functions of aml1/evi-1 fusion molecule are mostly demonstrated by in vitro experiments. our next step is to generate aml1/evi-1 leukemia model mice and demonstrate these functions in vivo. it is plausible that aml1/evi-1 knock-in mice are embryonic lethal in the mid-gestation because of the dominant-negative effects of aml1/evi-1 over normal aml1, as is the case with aml1/eto knock-in mice. thus, we are planning to make conditional aml1/evi-1 knock-in mice to enable temporal and special expression control of the chimeric molecule.   abstract  runx1-evi1 is a chimeric gene generated by t(3;21)(q26;q22) observed in patients with aggressive transformation of myelodysplastic syndrome or chronic myelogenous leukemia. runx1-evi1 has oncogenic potentials through dominant-negative effect over wild-type runx1, inhibition of jun kinase (jnk) pathway, stimulation of cell growth via ap-1, suppression of tgf-β-mediated growth inhibition and repression of c/ebpα. runx1-evi1 heterozygous knock-in mice die in uteri due to central nervous system (cns) hemorrhage and severe defects in definitive hematopoiesis as runx1–/– mice do, indicating that runx1-evi1 dominantly suppresses functions of wild-type runx1 in vivo. acute myelogenous leukemia is induced in mice transplanted with bone marrow cells expressing runx1-evi1, and a runx1-evi1 knock-in chimera mouse developed acute megakaryoblastic leukemia. these results suggest that runx1-evi1 plays indispensable roles in leukemogenesis of t(3;21)-positive leukemia. major leukemogenic effect of runx1-evi1 is mainly through histone deacetyltransferase recruitment via c-terminal binding protein. histone deacetyltransferase could be a target in molecular therapy of runx1-evi1-expressing leukemia. (cancer sci 2008; 99: 1878–1883) the t(3;21)(q26;q22) chromosomal translocation occurs in patients with aggressive transformation of myelodysplastic syndrome (mds) or chronic myelogenous leukemia (cml). the presence of this chromosomal translocation indicates poor prognosis.(1–3) in the chromosomal joining region of t(3;21)(q26;q22), the runx1 gene on 21q22 is fused with the evi1 (ecotropic viral integration site-1) gene on 3q26.5. previously, we cloned the runx1-evi1 fusion gene from a case with blastic crisis of cml developing additional to the t(3;21) translocation.(4) this chimeric transcription factor is believed to be a molecular culprit for the leukemic progression of stem cell malignancies caused by t(3;21)(q26;q22). molecular and biological function of wild-type runx1  the runx1 protein mainly consists of two functional domains; the runt homology domain (which is known as a dna-binding domain), and the proline-, serine- and threonine-rich (pst) region (which is known as a putative transcriptional activation domain) (fig. 1). runx1 forms a heterodimeric active transcriptional complex with the non-dna-binding β subunit (cbfβ/pebp2β) and binds to a specific dna consensus sequence (accrca) named pebp2.(5–9) runx1- or cbfβ-deficient mice are embryonic lethal at day 12.5 of gestation (e12.5), showing massive hemorrhage in the central nervous system (cns) and lack of hematopoiesis in the fetal liver.(10–14) a recent study demonstrated that inactivation of runx1 in adult mice results in megakaryocyte maturation arrest, block in t- and b-lymphocyte development and increase in hematopoietic precursor cells.(15) a subsequent study by the same group showed that the number of quiescent hematopoietic stem cells (hsc) is also negatively regulated by runx1.(16) clinically, mutations in runx1 have been identified in 15% to 40% of mds-refractory anemia with excess of blasts (raeb) and mds/acute myelogenous leukemia (aml).(17,18) patients with mds/aml with the runx1 mutations have a significantly worse prognosis than those without them.  figure 1.  figure 1. open in figure viewerdownload powerpoint slide schematic structure of wild-type runx1, evi1 and runx1-evi1 molecules. wild-type runx1 has runt homology domain at the n-terminus and proline-, serine- and threonine-rich (pst) region at the c-terminus. in runx1-evi1, runx1 protein is truncated at the end of the runt homology domain and followed by almost the entire coding region of evi1 molecule including the prdi-bf1-riz1 (pr) domain. molecular and biological function of wild-type evi1  the evi1 gene was initially identified as a frequent retrovirus integration site in myeloid tumors in akxd mice.(19) evi1 expression is low in normal hematopoietic cells, but it is highly expressed in some patients of mds or aml.(20,21) evi1 has four major functional domains; the two zinc finger (zf) domains, the c-terminal binding protein (ctbp)-binding site, and the acidic domain (ad) (fig. 1). the ctbp-binding site is located between the two zf domains, and the ad is located in the most c-terminus. evi1 is reported to interfere with transforming growth factor-β (tgf-β) signaling, and antagonize its growth inhibitory effect through targeting an intracellular signal transducer smad3.(22) evi1 is also known to enhance ap-1 activity(23) and to block c-jun n-terminal kinase (jnk) activity,(24) as described below. evi1–/– mice die at e10.5, and hsc in evi1–/– embryos are markedly decreased in numbers, with defective self-renewing proliferation and repopulating capacity.(25) the study also shows that evi1 directly regulates the transcription of gata-2, which controls both the maintenance and proliferation of hsc. it was also recently reported that the decreased colony forming capacity of evi1–/– para-aortic splanchnopleural (p-sp) cells could be rescued by retroviral gata-2 expression, and by blocking of tgfβ signaling using an in vitro cultivation system of murine p-sp regions.(26)  the mds1 gene was originally identified as a distinct gene from the evi1 gene, and the putative intergenic splicing form between mds1 and evi1 found in normal and leukemic cells was designated as mds1/evi1.(27) because the breakpoint in the chromosome 3 of t(3;21)(q26;q22) is located in the 5′ region of the mds1 gene, and the resultant runx1-evi1 fusion gene includes the first and second exons of the mds1 gene between runx1 and evi1 sequences, runx1-evi1 is also called as runx1-mds1/evi1. however, at present, mds1 is recognized as one of the alternative spliced form of the evi1 gene.(28) in mds1/evi1, a 188 amino-acid region called the pr domain, which consists of the first and second exons of mds1 and the second and third exons of evi1, is fused to the n-terminal of the evi1 molecule. the pr domain was originally identified in human retinoblastoma protein-binding protein riz1(29) and the human transcriptional repressor prdi-bf1,(30) and has been found in at least 17 kinds of diverse proteins to date. a large body of evidence suggests that pr domain family (prdm) proteins are involved in regulation of cellular growth as well as tumorigenesis.(28) it is interesting that the pr domains are generally located in the n-terminal region of proteins, and that alternative splicing creates two forms of the pr domain proteins; a long form that has the pr domain (pr+) and a short form that lacks the pr domain (pr–). decreased expression of pr+ molecules and/or overexpression of pr– molecules are observed in a variety of cancer cells, suggesting a functional antagonism, in which the pr+ forms contribute to tumor suppression while the pr– forms are oncogenic.  for the evi1 gene, evi1 (pr-) is highly expressed in cases with human aml or mds as a consequence of chromosomal rearrangements involving 3q26. increased expression of evi1 (pr-) is also observed without 3q26 abnormalities. recent studies have shown that increased expression of evi1 (pr-) in aml, which occurs in approximately 10% of the cases, is associated with unfavorable outcomes.(20,21) molecular structure and function of runx1-evi1  molecular structure of runx1-evi1.\\u2002 the runx1-evi1 fusion gene is translated in frame to generate a chimeric transcription factor in which runx1-evi1 chimeric complementary dna (cdna), an open reading frame of 4185 nucleotides, encodes a 1395 amino-acid protein. the n-terminal of runx1 molecule including its runt dna-binding domain is fused to almost the entire portion of evi1 molecule (fig. 1). therefore, the runx1-evi1 fusion protein is a chimeric transcription molecule that consists of the runt domain of runx1 and two zf domains, ctbp-binding site and ad of evi1. dominant-negative function over wild-type runx1.\\u2002 runx1-evi1 dominantly suppresses the transactivation capacity of runx1 through the pebp2 sites (fig. 2).(5–9) competition for the pebp2 site-binding is proposed to be a mechanism of such dominant negative effects, since runx1-evi1 binds to the pebp2 sites with higher affinity than runx1 does. on binding to the pebp2 site, runx1-evi1 is believed to recruit corepressor complex via ctbp, since it is reported that runx1-evi1 requires interaction with ctbp to repress runx1-induced transactivation.(31) in addition, the association with ctbp is also required for runx1-evi1 to block myeloid differentiation of 32dcl3 cells induced by granulocyte colony-stimulating factor (g-csf), indicating that the association with ctbp is critical for runx1-evi1 to exert its biological functions in vivo. taken together, it is suggested that one of the mechanisms for runx1-evi1-meidated leukemogenesis is the dominant-negative effects over wild-type runx1 through an aberrant recruitment of a transcriptional corepressor complex.  figure 2.  figure 2. open in figure viewerdownload powerpoint slide (a) runx1-evi1 exerts dominant-negative effects over wild-type runx1. runx1 forms an active transcription factor by heterodimerizing with cbfβ through runt homology domain. cbfβ increases the dna-binding ability of runx1 and protects runx1 from degradation. (b) because runx1-evi1 binds cbfβ more strongly than runx1, runx1-evi1 competes runx1 out from the pebp2 sites. runx1-evi1 associates with ctbp and recruits hdac. runx1-evi1 actively represses the transcription of the runx1 target genes. stimulation of cell growth through ap-1.\\u2002 the transcription factor ap-1 (fos/jun heterodimer or jun/jun homodimer) represents a prototype of regulatory protein that converts extracellular signals into gene expressions. ap-1 is activated by growth stimuli, including growth factors, phorbol esters such as 12-0-tetradec-anoylphorbol 13-acetate (tpa), and various transforming oncogene products. in addition, ap-1 functions as a positive or negative regulator in a variety of cellular differentiation processes, such as changes in the differentiation processes of embryonal carcinoma cell lines, preadipocytes and myoblasts. previously, we showed that rat1 cells expressing runx1-evi1 chimeric protein formed macroscopic colonies in soft agar,(32) indicating that runx1-evi1 has oncogenic potential. removal of the second zf domain within the evi1 sequence totally abrogated the ability of runx1-evi1 to transform rat1 cells. we showed that the transforming effect is correlated with the ap-1 activation induced by runx1-evi1.(33) inhibition of jnk pathway.\\u2002 jnk is activated preferentially by extracellular stress stimuli including uv light, γ-radiation, osmotic shock, protein synthesis inhibitors, tumor necrosis factor-α and interleukin-1.(34) the jnk pathway is thought to play an important role in triggering apoptosis in response to cellular stresses. the activated jnks translocate into the nucleus where they phosphorylate transcription factors such as c-jun, and strongly augment their transcriptional activity. we showed that evi1 physically interacts with and thereby inhibits the function of jnk.(24) for instance, evi1 inhibits stress-induced cell death by inhibiting jnk. this inhibition of cell death by evi1 could contribute to oncogenic transformation of cells. it has not been determined whether runx1-evi1 possesses the similar antiapoptotic effect as evi1 does. however, given that this antiapoptotic function is dependent on the first zf domain in evi1, it is reasonable to speculate that runx1-evi1 also inhibits cell apoptosis by inhibiting jnk. suppression of tgf-β -mediated growth inhibition.\\u2002 tgf-β is one of the best characterized members of growth inhibitory factors. tgf-β inhibits proliferation of a wide range of cell types including epithelial, endothelial and hematopoietic cells.(35,36) intracellular mechanisms that transmit tgf-β signaling have been clarified in detail. when tgf-β binds to its receptors, the down-stream signaling molecules smad2 and smad3 are phosphorylated by the activated tgf-β receptors, followed by oligomerization with smad4. the smad2/4 or smad3/4 complexes accumulate in the nucleus, interact with dna, and activate the transcription of tgf-β-responsive genes.(37,38) this process is apparently simple, but many proteins, including inhibitory smads, participate in regulating the process and modify cellular responses to the stimuli. previously we reported that evi1 antagonizes the growth-inhibitory effects of tgf-β in the epithelial cells.(22) consistent with this finding, runx1-evi1 also represses tgf-β-mediated growth inhibition of a murine hematopoietic precursor cell line, 32dcl3.(39) the ability of runx1-evi1 to repress tgf-β signaling depends on the two separate regions of the evi1 portion, one of which is the first zf domain. runx1-evi1 interacts with smad3 through this domain, and represses the smad3 activity as evi1 does.  recently we have demonstrated that evi1 represses smad-induced transcription by recruiting ctbp as a corepressor,(40) indicating that runx1-evi1 represses smad3 activity through recruitment of ctbp.(31) evi1 and runx1-evi1 associate with ctbp through one of the consensus binding motifs, and this association is required for efficient inhibition of tgf-β signaling. a specific inhibitor for histone deacetylase (hdac) alleviates evi1-mediated repression of tgf-β signaling, suggesting that hdac is involved in transcriptional repression by evi1. the association with ctbp is required for runx1-evi1 to block myeloid differentiation of 32dcl3 cells induced by g-csf.  several chimeric proteins, such as runx1-eto,(41,42) pml-rarα(43) and bcr-abl(44) have self-interaction domains that are critical for their oncogenic capacity. runx1-evi1 was reported to homo-oligomerize through at least three oligomerization regions,(45) i.e. the runt domain, the first and the second zf domains. deletion of the second zf domain significantly reduces the differentiation block of primary murine bone marrow progenitors by runx1-evi1. a point mutation that inhibits ctbp binding also completely abrogates the effects of runx1-evi1 on differentiation. these results imply the importance of homo-oligomerization for runx1-evi1 chimeric proteins to exert its leukemogenic potentials. repression of c/ebpα.\\u2002 c/ebpα (ccaat/enhancer binding protein α) is a leucine zipper transcription factor that regulates the expression of target genes containing c/ebp sites in their promoter regions.(46,47) of the genes related to hematopoiesis, these genes include cebpa itself,(48) cebpe(48,49) and granulocyte colony-stimulating factor (g-csf) receptor.(48,50) inducible expression of c/ebpα is sufficient to trigger terminal granulocytic differentiation(51–54) and blocks monocytic differentiation program.(51,53) conversely, inactivation of c/ebpα in cebpa knock-out mice showed profound defects in the granulocytic differentiation, while all other hematopoietic cells are present in normal number, indicating its critical role in the granulopoiesis.(55) several lines of evidence suggest that mutations in cebpa is one of the major molecular events that could cause myeloid malignancies. ten per cent of patients with aml m1 or m2, according to the french-american-british (fab) classification, without frequent cytogenetic abnormalities such as t(8;21)(q22;q22), carry heterozygous cebpa gene mutations that result in truncated proteins with a dominant negative effects.(49,56,57) the runx1-eto fusion protein generated by the t(8;21) translocation represses the transcription of cebpa by suppressing its auto-regulatory loop in gene transcription,(48) while the pml-rarα fusion protein generated by t(15;17)(q21;q22) inhibits the function of c/ebpα.(58) we confirmed that runx1-evi1 suppressed the differentiation of lg-3 cells that differentiate along the myeloid lineage upon exposure to g-csf.(59) coexpression of c/ebpα restored the differentiation ability of the runx1-evi1-expressing lg-3 cells. we also found that runx1-evi1 associates with c/ebpα. runx1-evi1 suppresses c/ebpα-mediated transcription of the cebpa promoter in a ctbp-binding site-dependent fashion. in a gel-shift assay, runx1-evi1 down-regulated the dna-binding activity of c/ebpα. therefore, the recruitment of hdac by runx1-evi1 and interference with the dna binding of c/ebpα could be the mechanisms for the repression of c/ebpα by runx1-evi1. these results indicate that inhibition of c/ebpα is related to the leukemogenic potential of runx1-evi1. helbling et al.(60) have reported that runx1-evi1 represses c/ebpα in a different way. they reported that runx1-evi1 reduces the level of c/ebpα protein but not the level of its mrna in u937 cells and in leukemic blasts of patients carrying the runx1-evi1 translocation, and that runx1-evi1 up-regulates the expression of calreticulin, a putative inhibitor of cebpα translation. calreticulin has calcium storage and chaperone function, and is postulated to be involved in the development of leukemia.(61) the small interference rna (sirna) against calreticulin showed that runx1-evi1 inhibited c/ebpα expression in a post-transcriptional mechanism through calreticulin. biological function of runx1-evi1  runx1-evi1 transforms the most immature hematopoietic cells.\\u2002 as described above, senyuk et al. showed that runx1-evi1 transforms primary murine bone marrow progenitors, depending on both of the ctbp-binding site and the second zf domain.(45,62) recently, takeshita et al.(63) introduced runx1-evi1 and its mutants in murine bone marrow cells and evaluated their transforming activities by colony replating assays. the transforming activity of runx1-evi1 was lost when any of the known functional domains of evi1, the first and the second zf domains, ad at the c-terminus or ctbp-binding site, was deleted from the chimeric protein. although runx1-evi1 is known to repress function of wild-type runx1 dominantly, forced expression of evi1 did not transform the runx1–/– bone marrow cells, indicating that the existence of runx1-evi1 means more than a simple combination of the presence of evi1 and the absence of runx1. interestingly, unlike the mll-enl or runx1-eto leukemia-related chimeric proteins, which transform hematopoietic progenitors, runx1-evi1 transforms only the hematopoietic stem cell fraction (c-kit + sca-1 + lin–). moreover, runx1-evi1-transformed cells show a cell-marker profile distinct from that of the cells transformed by runx1-eto, which also suppresses runx1 function. the nature of runx1-evi1-leukemia as hematopoietic stem cell tumors might be a consequence of these oncogenic preference of runx1-evi1. a bone marrow transplantation model.\\u2002 cuenco et al.(64) analyzed the effect of the human runx1-evi1 fusion gene in mouse bone marrow cells using retroviral transduction system. mice transplanted with runx1-evi1-expressed bone marrow cells developed acute leukemia 5–13 months after the transplantation. the disease can be transferred into secondary recipient mice with a much shorter latency period. morphological analysis of peripheral blood and bone marrow smears demonstrated the presence of myeloid blast cells and immature cells differentiated into both myelocytic and monocytic lineages. cytochemical and flow cytometric analysis confirmed that these mice had a disease recapitulating human aml. these observations indicate that the expression of runx1-evi1 can induce aml in mice, with extended latency period suggesting a requirement for additional perturbations. a study by the same group has demonstrated a cooperation of bcr-abl and runx1-evi1 in blocking myeloid differentiation and rapid induction of aml in mouse model.(65) the study showed that runx1-evi1 alone does not block myeloid differentiation in the mouse bone marrow during the 4 months of preleukemia stage, while coexpression of bcr-abl and runx1-evi1 can block myeloid differentiation and induce aml rapidly. they also showed that both runx1 and evi1 portions are required for runx1-evi1 to cooperate with bcr-abl in the induction of aml in mice.(66)  recently, it has been shown in a virus transduced experiment that mice transplanted with bone marrow cells expressing runx1 mutants developed mds/aml within 4–13 months.(67) interestingly, the analysis of the viral integration sites showed that evi1 seemed to be a collaborating gene for the runx1(d171n) mutant for the induction of mds/aml. the disease has common phenotype characterized by marked hepatosplenomegaly, myeloid dysplasia, leukocytosis and biphenotypic surface markers. the collaboration between runx1(d171n) and evi1 was confirmed by a bone marrow transplant (bmt) model, where coexpression of runx1(d171n) and evi1 induced acute leukemia of the same phenotype with much shorter latency. these results suggest that a combination of dominant-negative effect over runx1 and the oncogenic property of evi1, both of which are components of runx1-evi1, could cause mds or mds/aml without additional hits. it seems important that runx1 should not be inactivated completely in order to cooperate with evi1, because expression of evi1 did not transform the runx1-deleted murine bone marrow cells.(63) runx1-evi1 knock-in mice.\\u2002 we knocked-in the runx1-evi1 chimeric gene into the mouse runx1 genomic locus to explore the effect of runx1-evi1 in developmental hematopoiesis in vivo.(68) our knock-in expression of runx1-evi1 fusion gene results in embryonic lethality between e12.5 and e14.5, with cns hemorrhage and a lack of fetal liver hematopoiesis. post-enucleated erythrocytes were absent in the peripheral blood from e12.5 runx1-evi1/+ embryos, whereas nucleated erythroblasts were abundantly observed. these findings indicate that runx1-evi1/+heterozygous mice fail to establish definitive hematopoiesis in the fetal liver as runx1–/– mice. therefore, runx1-evi1 was first demonstrated to have in vivo dominant inhibitory effects over runx1. electron microscopic examination of the e13.5 fetal liver showed that a fewer number of erythroid and myeloid progenitors, and dysplastic megakaryocytes that were defective for demarcation membrane exist in the runx1-evi1/+ fetal liver.  on in vitro hematopoietic colony forming assays, the fetal liver from e13.5 runx1-evi1/+embryos contained multilineage hematopoietic progenitors, while that from e12.5 runx1–/– embryos was reported to contain no definitive hematopoietic progenitors.(10,11) no erythroid colonies were seen in the livers at both e12.5 and e13.5. the cfu-gemm-derived colonies from e13.5 runx1-evi1/+embryos included few erythroblasts, numerous agranular granulocytes with a delayed differentiation, and dysplastic megakaryocytes. moreover, serial in vitro replating assays showed higher self-renewal capacity of hematopoietic progenitors in e13.5 runx1-evi1/+ fetal livers than that in wild-type fetal livers.  on semiquantitative reverse transcriptase-polymerase chain reaction (rt-pcr) method with fetal livers, the expression of pu.1 mrna in runx1-evi1/+fetal liver cells was comparable with that in the wild-type cells, whereas its expression level was markedly decreased in runx1–/– fetal liver cells, as has been reported previously.(69) the maintained expression of pu.1 in runx1-evi1/+fetal liver cells may support the survival of their multilineage hematopoietic progenitors up to e13.5, and enhance their monocyte/macrophage lineage differentiation. these differences in gene expression pattern provide logical explanation for the distinct hematopoietic capacity between runx1–/– and runx1-evi1/+ fetal liver cells. these data suggest that the sufficient expression of the pu.1 gene is the prerequisite for definitive hematopoiesis in the fetal liver. runx1-evi1 knock-in chimeric mice.\\u2002 of the six chimeric mice created, one mouse which died at 5 months of age showed marked hepatosplenomegaly. wright-giemsa staining of stump preparation from the enlarged spleen demonstrated massive infiltration of large dysplastic cells, some of which contained multilobulated nuclei with various size of cytoplasm reminiscent of megakaryoblastic leukemia.(70) histology section showed disrupted gross architecture of the spleen, with white and red pulp intermingling, and the electron microscopic analysis of the infiltrating cells in the spleen showed 20% of the cells positive for platelet-peroxidase, indicating that this chimeric mouse developed megakaryoblastic leukemia. the important aspect of our observation is that runx1-evi1 protein could be leukemogenic per se, in contract to runx1-eto, which requires additional hits to induce leukemia.(70–72) this clear difference in the pathophysiological outcome likely arises from the evi1 portion of runx1-evi1 protein. oncogenic functions of evi1 described above may cause stronger oncogenic capacity of runx1-evi1 than runx1-eto. another important aspect is that the affected lineage in human leukemia is recapitulated in the experimental animal. our observation indicates strong causal relationship between the expression of runx1-evi1 protein and megakaryoblastic leukemia. indeed, the runx1-evi1 chimeric gene was isolated from a patient developing megakaryoblastic crisis in chronic myelocytic leukemia with additional chromosome t(3;21). clinical aspects  as mentioned above, the major leukemogenic effect of runx1-evi1 is through hdac-recruitment via ctbp (fig. 3). thus repression of hdac activity is thought to suppress oncogenic effect of runx1-evi1. hdac inhibitors (hdaci) are the member of a new class of chemical agents that epigenetically modulate gene transcription by enhancing the acetylation of core nucleosomal histones and are expected to be promising anticancer agents against various types of tumors. suberoylanilide hydroxamic acid (vorinostat), one of the newly synthesized hdaci, is the most clinically successful hdaci that controls cutaneous t-cell lymphoma (ctcl) effectively.(73) conventional hdaci, such as valproic acid and butylate, are also reported to be effective in treating some types of hematological malignancies in clinical trials. recently, we have reported that hdaci (trichostatin a and vpa) triggers apoptosis in human leukemic cell lines expressing runx1-related chimeric proteins such as runx1-eto, tel-runx1 or runx1-evi1.(74) a cell line without runx1-related chimeras is less affected by the hdaci treatment. these data suggest that hdaci seems to be an attractive choice in the molecular targeting therapy of runx1-evi1-expressing leukemia.  figure 3.  figure 3. open in figure viewerdownload powerpoint slide runx1-evi1 is a multifunctional oncoprotein. runx1-evi1 exerts oncogenic function through three pathways: dominant-negative effect over runx1, repression of tgfβ signaling and inhibition of c/ebpα activity. another potential therapeutic agent targeted to runx1-evi1 is arsenic trioxide (ato). shackelford et al.(75) found that ato degrades runx1-evi1. the ato treatment induces the differentiation and apoptosis of runx1-evi1-expressing leukemic cells in vitro and elongates the survival of mice transplanted with these cells in vivo. they also demonstrated that ato targets runx1-evi1 via two moieties, mds1 and evi1 moieties. the evi1 induces a ubiquitin-proteasome pathway and mds1 induces a proteasome-independent pathway. with abundant experiences of clinical use in treating acute promyelocytic leukemia, ato could be used clinically as a targeted therapy for runx1-evi1-positive human leukemia. summary  the fusion protein runx1-evi1 is a multifunctional protein demonstrated by its diverse role in regulating differentiation, proliferation, apoptosis and self-renewal capacity. it functions as a dominant-negative suppressor of runx1 and has oncogenic properties of evi1. there is no doubt that runx1-evi1 plays a major role in t(3;21)-related mds and mds/aml. it remains to be determined whether runx1-evi1 induces leukemia by itself, or needs additional genetic events. a conditional knock-in experiment of the runx1-evi1 gene would help clarify this question.\""
      ]
     },
     "execution_count": 555,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concatenate_data.loc[ID,\"Text\"].lower()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 554,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "runx1-evi1 fusion runx1\n"
     ]
    }
   ],
   "source": [
    "print(concatenate_data.loc[ID,\"Variation\"].lower(),\n",
    "concatenate_data.loc[ID,\"Gene\"].lower())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "CBL = train_variant[train_variant[\"Gene\"] == \"RUNX1\"]\n",
    "article_cbl = train_text.loc[CBL.index,]\n",
    "\n",
    "CBL.to_csv(\"cbl.txt\")\n",
    "dtf = pd.merge(pd.DataFrame(clean_article_CBL.index), clean_article_CBL, on =\"ID\")\n",
    "np.savetxt(\"cbl_clean_article.txt\",dtf, fmt = \"%d||%s\", header= \",\".join(dtf.columns), comments='')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 534,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_data = pd.merge(train_variant,clean_match, on = \"ID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 552,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "      <td>the c-cbl loh also posity correlated with egf...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "      <td>the c-cbl loh also posity correlated with egf...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "      <td>most of the changes were novel, although 4 ca...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "      <td>all mutations from the second group were pred...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CBL</td>\n",
       "      <td>V391I</td>\n",
       "      <td>4</td>\n",
       "      <td>all mutations from the second group were pred...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3310</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>7</td>\n",
       "      <td>this truncated protein conta the rhd but not ...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3311</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>RUNX1-EVI1 Fusion</td>\n",
       "      <td>4</td>\n",
       "      <td>21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutatio...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3313</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>H78Q</td>\n",
       "      <td>4</td>\n",
       "      <td>lpxn is preferentially expressed in hematopoi...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3314</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>G42R</td>\n",
       "      <td>6</td>\n",
       "      <td>) conversely, mutations mapping to the first d...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3315</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>RUNX1-RUNX1T1 Fusion</td>\n",
       "      <td>4</td>\n",
       "      <td>mutation analysis in fpd/aml pedigrees. elect...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2460 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene             Variation  Class  \\\n",
       "ID                                         \n",
       "1       CBL                 W802*      2   \n",
       "2       CBL                 Q249E      2   \n",
       "3       CBL                 N454D      3   \n",
       "4       CBL                 L399V      4   \n",
       "5       CBL                 V391I      4   \n",
       "...     ...                   ...    ...   \n",
       "3310  RUNX1         Amplification      7   \n",
       "3311  RUNX1     RUNX1-EVI1 Fusion      4   \n",
       "3313  RUNX1                  H78Q      4   \n",
       "3314  RUNX1                  G42R      6   \n",
       "3315  RUNX1  RUNX1-RUNX1T1 Fusion      4   \n",
       "\n",
       "                                                   Text  Score  \n",
       "ID                                                              \n",
       "1      the c-cbl loh also posity correlated with egf...      1  \n",
       "2      the c-cbl loh also posity correlated with egf...      1  \n",
       "3      most of the changes were novel, although 4 ca...      1  \n",
       "4      all mutations from the second group were pred...      1  \n",
       "5      all mutations from the second group were pred...      1  \n",
       "...                                                 ...    ...  \n",
       "3310   this truncated protein conta the rhd but not ...      1  \n",
       "3311  21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutatio...      1  \n",
       "3313   lpxn is preferentially expressed in hematopoi...      1  \n",
       "3314  ) conversely, mutations mapping to the first d...      1  \n",
       "3315   mutation analysis in fpd/aml pedigrees. elect...      1  \n",
       "\n",
       "[2460 rows x 5 columns]"
      ]
     },
     "execution_count": 552,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data[new_data[\"Score\"] == 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 588,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "truncating mutations 1110 0\n",
      "deletion 1113 0\n",
      "l274p 1114 0\n",
      "w183a 1115 0\n",
      "x1007_splice 1118 0\n",
      "v1188l 1121 0\n",
      "f1200i 1122 0\n",
      "v1092i 1124 0\n",
      "y1003f 1126 0\n",
      "x963_splice 1129 0\n",
      "x1009_splice 1132 0\n",
      "overexpression 1133 0\n",
      "y1003* 1134 0\n",
      "h1094l 1136 0\n",
      "m1250t 1142 0\n",
      "h1106d 1144 0\n",
      "y1230h 1145 0\n",
      "n1100y 1149 0\n",
      "s3660l 1154 0\n",
      "truncating mutations 1156 0\n",
      "truncating mutations 1163 0\n",
      "i31m 1165 0\n",
      "m1043i 1170 0\n",
      "g106v 1172 0\n",
      "d350g 1173 0\n",
      "e545q 1174 0\n",
      "n1068fs*4 1177 0\n",
      "n1044k 1179 0\n",
      "t1025s 1182 0\n",
      "q60k 1185 0\n",
      "e545k 1187 0\n",
      "e365k 1188 0\n",
      "e81k 1189 0\n",
      "h701p 1193 0\n",
      "p449_l455del 1199 0\n",
      "p449t 1201 0\n",
      "a1020v 1203 0\n",
      "q546r 1206 0\n",
      "e579k 1207 0\n",
      "d1029y 1208 0\n",
      "g106_r108del 1210 0\n",
      "e453a 1214 0\n",
      "c378r 1215 0\n",
      "i391m 1217 0\n",
      "fusions 1224 0\n",
      "d1067y 1226 0\n",
      "e1051k 1228 0\n",
      "d1067a 1229 0\n",
      "y53h 1236 0\n",
      "p123m 1237 0\n",
      "fusions 1241 0\n",
      "amplification 1242 0\n",
      "yap1-fam118b fusion 1244 0\n",
      "e218* 1249 0\n",
      "i559_d560insdkrmns 1251 0\n",
      "n564d 1252 0\n",
      "w237_y242del 1253 0\n",
      "x582_splice 1257 0\n",
      "k459_s460delinsn 1262 0\n",
      "r348* 1265 0\n",
      "r262t 1266 0\n",
      "e439del 1268 0\n",
      "k376n 1270 0\n",
      "amplification 1271 0\n",
      "n561d 1272 0\n",
      "d557h 1273 0\n",
      "s273c 1276 0\n",
      "truncating mutations 1277 0\n",
      "amplification 1280 0\n",
      "q61k 1288 0\n",
      "k117n 1289 0\n",
      "r100* 1302 0\n",
      "q689r 1313 0\n",
      "w714* 1317 0\n",
      "l622h 1328 0\n",
      "l749p 1333 0\n",
      "truncating mutations 1338 0\n",
      "fusions 1339 0\n",
      "e267g 1340 0\n",
      "k179m 1344 0\n",
      "r370c 1347 0\n",
      "e49k 1349 0\n",
      "g311d 1352 0\n",
      "v271a 1355 0\n",
      "t195i 1360 0\n",
      "l202f 1366 0\n",
      "k181m 1369 0\n",
      "r368c 1372 0\n",
      "fusions 1374 0\n",
      "amplification 1376 0\n",
      "d399n 1378 0\n",
      "a263v 1380 0\n",
      "cep110-fgfr1 fusion 1381 0\n",
      "amplification 1384 0\n",
      "k656e 1386 0\n",
      "n546k 1390 0\n",
      "a500t 1393 0\n",
      "d617g 1395 0\n",
      "igh-fgfr3 fusion 1398 0\n",
      "d641g 1400 0\n",
      "y647c 1401 0\n",
      "s371c 1403 0\n",
      "e627d 1404 0\n",
      "r248c 1406 0\n",
      "c582f 1412 0\n",
      "h643d 1413 0\n",
      "r669g 1414 0\n",
      "e466k 1416 0\n",
      "f384l 1417 0\n",
      "amplification 1418 0\n",
      "g697c 1420 0\n",
      "v555m 1421 0\n",
      "y373c 1422 0\n",
      "s249c 1423 0\n",
      "a391e 1424 0\n",
      "n653h 1428 0\n",
      "fusions 1429 0\n",
      "g380r 1430 0\n",
      "f133v 1431 0\n",
      "d140g 1434 0\n",
      "w131a 1437 0\n",
      "w131g 1438 0\n",
      "g75r 1441 0\n",
      "d130a 1446 0\n",
      "m117i 1449 0\n",
      "truncating mutations 1450 0\n",
      "k517r 1452 0\n",
      "y375c 1454 0\n",
      "k659n 1455 0\n",
      "t730s 1457 0\n",
      "fgfr2-mgea5 fusion 1458 0\n",
      "v755i 1459 0\n",
      "l770v 1461 0\n",
      "r203c 1462 0\n",
      "fgfr2-ahcyl1 fusion 1464 0\n",
      "a648t 1465 0\n",
      "n549s 1466 0\n",
      "g701s 1468 0\n",
      "t341p 1469 0\n",
      "n549k 1471 0\n",
      "s252w 1472 0\n",
      "fgfr2-bicc1 fusion 1473 0\n",
      "i642v 1477 0\n",
      "fgfr2-kiaa1967 fusion 1482 0\n",
      "w290_i291delinsc 1487 0\n",
      "c382r 1488 0\n",
      "k641n 1489 0\n",
      "fgfr2-ccdc6 fusion 1491 0\n",
      "e219k 1492 0\n",
      "k310r 1494 0\n",
      "m391r 1497 0\n",
      "k659e 1499 0\n",
      "amplification 1503 0\n",
      "deletion 1504 0\n",
      "truncating mutations 1505 0\n",
      "g1128s 1507 0\n",
      "ranbp2-alk fusion 1513 0\n",
      "l1240v 1517 0\n",
      "y1278s 1519 0\n",
      "g881d 1522 0\n",
      "cad-alk fusion 1525 0\n",
      "l1152r 1526 0\n",
      "g1123s 1528 0\n",
      "s1206r 1529 0\n",
      "s1206y 1531 0\n",
      "m1328i 1535 0\n",
      "t1343i 1539 0\n",
      "t1151dup 1540 0\n",
      "g1201e 1543 0\n",
      "v1180l 1544 0\n",
      "p1139s 1548 0\n",
      "l1204f 1549 0\n",
      "a1099t 1553 0\n",
      "r1060h 1555 0\n",
      "e1210k 1557 0\n",
      "fusions 1560 0\n",
      "v597a 1563 0\n",
      "d1349h 1564 0\n",
      "r1464* 1565 0\n",
      "i1183t 1566 0\n",
      "l1198p 1567 0\n",
      "c1156y 1568 0\n",
      "h694r 1570 0\n",
      "r1275l 1572 0\n",
      "i1170s 1573 0\n",
      "r133* 1577 0\n",
      "q233* 1583 0\n",
      "d121g 1589 0\n",
      "l128f 1594 0\n",
      "r200w 1600 0\n",
      "q164r 1602 0\n",
      "a149p 1603 0\n",
      "l158p 1608 0\n",
      "l188v 1615 0\n",
      "l118r 1616 0\n",
      "w117c 1617 0\n",
      "e70k 1619 0\n",
      "s65l 1623 0\n",
      "r304* 1637 0\n",
      "flt3 internal tandem duplications 1640 0\n",
      "n676k 1642 0\n",
      "y599_d600insglyvdfreyey 1643 0\n",
      "d324n 1647 0\n",
      "y591d 1648 0\n",
      "i836del 1653 0\n",
      "w603_e604insdreyeydlkw 1656 0\n",
      "n676d 1657 0\n",
      "s840_n841insgs 1658 0\n",
      "a627t 1659 0\n",
      "n676s 1662 0\n",
      "m664i 1663 0\n",
      "f691l 1664 0\n",
      "v194m 1666 0\n",
      "s451f 1668 0\n",
      "l601_k602insreyeydl 1674 0\n",
      "l611_e612inscssdneyfyvdfreyeydlkwefprenl 1676 0\n",
      "e612_f613insgyvdfreyeydlkwefrprenlef 1677 0\n",
      "y599_d600inspapqimststlisenmnia 1679 0\n",
      "y572c 1681 0\n",
      "e598_y599insglvqvtgssdneyfyvdfreye 1682 0\n",
      "y599_d600inseyeyeyey 1684 0\n",
      "d839g 1685 0\n",
      "d600_l601insfreyeyd 1687 0\n",
      "g857a 1689 0\n",
      "i668v 1697 0\n",
      "t485k 1702 0\n",
      "e1099k 1706 0\n",
      "422_605trunc 1707 0\n",
      "v411l 1714 0\n",
      "p286h 1715 0\n",
      "l424v 1717 0\n",
      "n1026s 1724 0\n",
      "r680* 1742 0\n",
      "s723f 1746 0\n",
      "q429* 1747 0\n",
      "q816* 1748 0\n",
      "r389* 1749 0\n",
      "r100a 1752 0\n",
      "g123r 1753 0\n",
      "v294m 1759 0\n",
      "r140q 1766 0\n",
      "r172s 1767 0\n",
      "deletion 1773 0\n",
      "w1502a 1780 0\n",
      "q1500p 1781 0\n",
      "q58_q59insl 1783 0\n",
      "p380r 1784 0\n",
      "arv567es 1786 0\n",
      "w742c 1792 0\n",
      "t878s 1793 0\n",
      "amplification 1794 0\n",
      "f876l 1796 0\n",
      "f877l 1799 0\n",
      "ar-v7 1801 0\n",
      "n217i 1802 0\n",
      "q347_a348del 1805 0\n",
      "f351l 1806 0\n",
      "s214c 1807 0\n",
      "g322s 1808 0\n",
      "amplification 1816 0\n",
      "overexpression 1820 0\n",
      "p179l 1835 0\n",
      "r183q 1838 0\n",
      "truncating mutations 1840 0\n",
      "r1625c 1841 0\n",
      "m1255i 1842 0\n",
      "a1374v 1843 0\n",
      "g1125a 1844 0\n",
      "h114y 1845 0\n",
      "r264c 1848 0\n",
      "h284n 1849 0\n",
      "h284p 1850 0\n",
      "truncating mutations 1851 0\n",
      "r339w 1852 0\n",
      "deletion 1853 0\n",
      "amplification 1854 0\n",
      "p2273s 1855 0\n",
      "a1519t 1856 0\n",
      "l2427r 1859 0\n",
      "r2430m 1862 0\n",
      "l2431p 1863 0\n",
      "e2419k 1867 0\n",
      "w1456r 1871 0\n",
      "f1888v 1874 0\n",
      "t1977r 1875 0\n",
      "l2230v 1877 0\n",
      "a1459p 1879 0\n",
      "c1483w 1880 0\n",
      "a2034v 1881 0\n",
      "q2223k 1883 0\n",
      "m2327i 1884 0\n",
      "t1977s 1886 0\n",
      "f2108l 1887 0\n",
      "y1463s 1888 0\n",
      "e2014k 1892 0\n",
      "r2505* 1897 0\n",
      "mll-tet1 fusion 1901 0\n",
      "r978* 1903 0\n",
      "deletion 1909 0\n",
      "truncating mutations 1910 0\n",
      "truncating mutations 1911 0\n",
      "r514c 1912 0\n",
      "d384n 1914 0\n",
      "q477e 1916 0\n",
      "d473g 1917 0\n",
      "d473h 1918 0\n",
      "q635e 1920 0\n",
      "n219d 1921 0\n",
      "l221r 1923 0\n",
      "s387n 1924 0\n",
      "i408v 1925 0\n",
      "v414a 1930 0\n",
      "e518a 1933 0\n",
      "g116s 1936 0\n",
      "d387v 1940 0\n",
      "l225li 1941 0\n",
      "e127g 1942 0\n",
      "s250p 1943 0\n",
      "d401n 1944 0\n",
      "e626k 1945 0\n",
      "e1978* 1953 0\n",
      "d1853n 1958 0\n",
      "g591v 1959 0\n",
      "exon 13 deletion 1960 0\n",
      "k442nfs* 1961 0\n",
      "e322k 1962 0\n",
      "amplification 1963 0\n",
      "k335i 1969 0\n",
      "s45a 1971 0\n",
      "c121s 1991 0\n",
      "i111a 1994 0\n",
      "p387s 1996 0\n",
      "f129l 1997 0\n",
      "q56_v60del 1998 0\n",
      "e102_i103del 2003 0\n",
      "k59del 2005 0\n",
      "q58_e62del 2008 0\n",
      "f53l 2010 0\n",
      "i111p 2011 0\n",
      "v211d 2017 0\n",
      "p306h 2021 0\n",
      "i111r 2022 0\n",
      "n382h 2026 0\n",
      "l37p 2028 0\n",
      "s222d 2030 0\n",
      "d67n 2031 0\n",
      "s226d 2032 0\n",
      "truncating mutations 2043 0\n",
      "t244_i245inscpt 2044 0\n",
      "t73i 2051 0\n",
      "myc-nick 2054 0\n",
      "igh-myc fusion 2056 0\n",
      "igk-myc fusion 2057 0\n",
      "amplification 2059 0\n",
      "truncating mutations 2062 0\n",
      "l142p 2064 0\n",
      "p70r 2067 0\n",
      "deletion 2069 0\n",
      "s1303n 2076 0\n",
      "h1382y 2078 0\n",
      "m1293a 2079 0\n",
      "n1387a 2080 0\n",
      "amplification 2088 0\n",
      "truncating mutations 2091 0\n",
      "truncating mutations 2093 0\n",
      "l996f 2100 0\n",
      "g909r 2101 0\n",
      "truncating mutations 2111 0\n",
      "d289del 2118 0\n",
      "amplification 2122 0\n",
      "256_286trunc 2123 0\n",
      "l283_d294del 2128 0\n",
      "p287s 2129 0\n",
      "p287t 2130 0\n",
      "igh-bcl2 fusion 2131 0\n",
      "g430c 2136 0\n",
      "r415g 2139 0\n",
      "g423v 2141 0\n",
      "g186r 2143 0\n",
      "g480w 2144 0\n",
      "g333s 2145 0\n",
      "r571w 2147 0\n",
      "e157g 2153 0\n",
      "a121p 2155 0\n",
      "y68h 2164 0\n",
      "c71y 2171 0\n",
      "r173c 2174 0\n",
      "i32del 2176 0\n",
      "f347l 2177 0\n",
      "n276s 2180 0\n",
      "s362l 2182 0\n",
      "p169h 2188 0\n",
      "deletion 2191 0\n",
      "r130k 2192 0\n",
      "l325f 2196 0\n",
      "c105f 2200 0\n",
      "t131s 2206 0\n",
      "f341v 2207 0\n",
      "h93q 2210 0\n",
      "a126d 2211 0\n",
      "g251c 2212 0\n",
      "f241s 2213 0\n",
      "d92e 2216 0\n",
      "d162g 2220 0\n",
      "k289e 2223 0\n",
      "y68d 2225 0\n",
      "r173p 2240 0\n",
      "d331g 2241 0\n",
      "k62r 2243 0\n",
      "r161g 2245 0\n",
      "t319del 2250 0\n",
      "l345q 2251 0\n",
      "k128t 2255 0\n",
      "d326n 2263 0\n",
      "d92v 2266 0\n",
      "r173h 2268 0\n",
      "h123y 2270 0\n",
      "truncating mutations 2279 0\n",
      "r1515h 2280 0\n",
      "r201w 2282 0\n",
      "fusions 2283 0\n",
      "cic-dux4 fusion 2284 0\n",
      "truncating mutations 2285 0\n",
      "h410r 2286 0\n",
      "d661y 2287 0\n",
      "f384l 2291 0\n",
      "f384y 2292 0\n",
      "f384v 2293 0\n",
      "c569y 2297 0\n",
      "p48t 2299 0\n",
      "f958s 2301 0\n",
      "l783f 2306 0\n",
      "r879c 2309 0\n",
      "truncating mutations 2311 0\n",
      "g1079d 2312 0\n",
      "l910p 2313 0\n",
      "y1035c 2314 0\n",
      "f958v 2315 0\n",
      "s729c 2316 0\n",
      "k648n 2317 0\n",
      "l611v 2318 0\n",
      "d544h 2320 0\n",
      "w1038c 2324 0\n",
      "pcm1-jak2 fusion 2326 0\n",
      "r683t 2328 0\n",
      "f537_k539delinsl 2329 0\n",
      "l983f 2330 0\n",
      "pax5-jak2 fusion 2332 0\n",
      "i1018f 2334 0\n",
      "atf7ip-jak2 fusion 2335 0\n",
      "s1039f 2336 0\n",
      "v1075f 2338 0\n",
      "r683s 2340 0\n",
      "y1045w 2341 0\n",
      "amplification 2343 0\n",
      "r683g 2346 0\n",
      "i1018w 2347 0\n",
      "tel-jak2 fusion 2348 0\n",
      "r683k 2349 0\n",
      "fusions 2350 0\n",
      "amplification 2351 0\n",
      "m136r 2355 0\n",
      "d194y 2359 0\n",
      "r304w 2360 0\n",
      "truncating mutations 2362 0\n",
      "h174r 2363 0\n",
      "a205t 2364 0\n",
      "f354l 2366 0\n",
      "amplification 2367 0\n",
      "deletion 2368 0\n",
      "r498l 2376 0\n",
      "d106a 2378 0\n",
      "q510e 2379 0\n",
      "m504v 2380 0\n",
      "k590r 2385 0\n",
      "g464a 2388 0\n",
      "s502t 2390 0\n",
      "y489c 2395 0\n",
      "k1436q 2397 0\n",
      "v1398d 2400 0\n",
      "truncating mutations 2403 0\n",
      "deletion 2408 0\n",
      "r1204g 2409 0\n",
      "amplification 2411 0\n",
      "r1088c 2412 0\n",
      "v1565i 2413 0\n",
      "truncating mutations 2414 0\n",
      "p1311t 2415 0\n",
      "r1835p 2420 0\n",
      "truncating mutations 2421 0\n",
      "g1803a 2422 0\n",
      "a1830t 2423 0\n",
      "g1743r 2424 0\n",
      "y105c 2429 0\n",
      "l1407p 2431 0\n",
      "d96n 2437 0\n",
      "p1856s 2440 0\n",
      "f1761i 2441 0\n",
      "a1823t 2442 0\n",
      "i1807s 2443 0\n",
      "l246v 2446 0\n",
      "v191i 2447 0\n",
      "h1746n 2455 0\n",
      "p1749r 2456 0\n",
      "h1746q 2457 0\n",
      "s1101n 2464 0\n",
      "t1691i 2465 0\n",
      "h1421y 2467 0\n",
      "e1794d 2471 0\n",
      "s308a 2473 0\n",
      "r1699l 2475 0\n",
      "v1804a 2476 0\n",
      "e23fs 2477 0\n",
      "v1534m 2487 0\n",
      "s1512i 2493 0\n",
      "m1783i 2494 0\n",
      "e1682v 2495 0\n",
      "m1783l 2496 0\n",
      "k1702e 2501 0\n",
      "r866c 2503 0\n",
      "d1344h 2508 0\n",
      "v271l 2509 0\n",
      "f1761s 2511 0\n",
      "e1586g 2512 0\n",
      "s1841a 2513 0\n",
      "p798l 2516 0\n",
      "e1836k 2524 0\n",
      "p1812s 2526 0\n",
      "q1756fs 2529 0\n",
      "g1738r 2530 0\n",
      "n1647k 2532 0\n",
      "q1785h 2533 0\n",
      "r1589p 2535 0\n",
      "l165p 2536 0\n",
      "c1767s 2539 0\n",
      "s1486c 2546 0\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "p1859r 2547 0\n",
      "d1778n 2553 0\n",
      "w1837r 2556 0\n",
      "v1833e 2557 0\n",
      "p1806a 2558 0\n",
      "d1733g 2568 0\n",
      "s1297del 2571 0\n",
      "l1764p 2574 0\n",
      "g1770v 2576 0\n",
      "a1752v 2577 0\n",
      "g1763v 2579 0\n",
      "d1739y 2582 0\n",
      "e1660g 2583 0\n",
      "s1715r 2584 0\n",
      "m1628v 2586 0\n",
      "v1838e 2592 0\n",
      "g1738e 2595 0\n",
      "v1809f 2598 0\n",
      "m1008i 2602 0\n",
      "e1282v 2603 0\n",
      "e1214k 2604 0\n",
      "v1665m 2605 0\n",
      "r1589h 2608 0\n",
      "n1819s 2610 0\n",
      "v1808a 2611 0\n",
      "p1771l 2616 0\n",
      "r1758g 2624 0\n",
      "l1267s 2628 0\n",
      "e143k 2629 0\n",
      "t1720a 2630 0\n",
      "q1826h 2643 0\n",
      "s1164i 2645 0\n",
      "q1756c 2650 0\n",
      "r841q 2651 0\n",
      "m1663k 2656 0\n",
      "s1715n 2657 0\n",
      "e1250k 2660 0\n",
      "s1613c 2661 0\n",
      "l668f 2662 0\n",
      "p1614s 2663 0\n",
      "v1378i 2667 0\n",
      "t1685i 2669 0\n",
      "s1473p 2673 0\n",
      "e1346k 2675 0\n",
      "r133h 2676 0\n",
      "a1843t 2677 0\n",
      "d1778g 2678 0\n",
      "t599_v600insv 2685 0\n",
      "l485f 2686 0\n",
      "l485_p490del 2687 0\n",
      "l485_p490delinsf 2688 0\n",
      "a598v 2689 0\n",
      "i326v 2690 0\n",
      "q201h 2691 0\n",
      "a246p 2693 0\n",
      "g596v 2694 0\n",
      "v600r 2695 0\n",
      "v471f 2696 0\n",
      "t529m 2697 0\n",
      "g478c 2699 0\n",
      "t529i 2702 0\n",
      "k483e 2706 0\n",
      "r506_k507insvlr 2707 0\n",
      "r671q 2711 0\n",
      "v487_p492delinsa 2716 0\n",
      "p61braf 2718 0\n",
      "agk-braf fusion 2720 0\n",
      "t599dup 2721 0\n",
      "s151a 2722 0\n",
      "papss1-braf fusion 2725 0\n",
      "g469v 2730 0\n",
      "mkrn1-braf fusion 2731 0\n",
      "i463s 2732 0\n",
      "g466a 2737 0\n",
      "l584f 2740 0\n",
      "l485_q494del 2742 0\n",
      "t599_v600inseat 2745 0\n",
      "amplification 2748 0\n",
      "g469del 2750 0\n",
      "akap9-braf fusion 2752 0\n",
      "e586k 2756 0\n",
      "v600d_k601insfglat 2760 0\n",
      "d287h 2761 0\n",
      "e275k 2764 0\n",
      "d594e 2767 0\n",
      "n581y 2769 0\n",
      "t599r 2771 0\n",
      "g464e 2773 0\n",
      "p168t 2774 0\n",
      "r2336p 2775 0\n",
      "t2250a 2776 0\n",
      "a2351g 2777 0\n",
      "v894i 2780 0\n",
      "g2748d 2781 0\n",
      "r2502c 2782 0\n",
      "r2418g 2784 0\n",
      "k1690n 2786 0\n",
      "y3092c 2787 0\n",
      "i2675v 2788 0\n",
      "a2643g 2789 0\n",
      "l2106p 2790 0\n",
      "n1878k 2791 0\n",
      "t582p 2794 0\n",
      "t1354m 2795 0\n",
      "r2304c 2797 0\n",
      "s384f 2798 0\n",
      "s142i 2801 0\n",
      "n2436i 2802 0\n",
      "n319t 2803 0\n",
      "e3002k 2805 0\n",
      "g1194d 2806 0\n",
      "q2858r 2807 0\n",
      "f1524v 2812 0\n",
      "k1434i 2814 0\n",
      "k2472t 2815 0\n",
      "s1670a 2816 0\n",
      "y3098h 2817 0\n",
      "r1190w 2820 0\n",
      "c1365y 2821 0\n",
      "a2717s 2822 0\n",
      "n372h 2823 0\n",
      "r2520q 2824 0\n",
      "s1172l 2825 0\n",
      "q2384k 2826 0\n",
      "p655r 2829 0\n",
      "l1019v 2830 0\n",
      "r2888c 2831 0\n",
      "n517s 2833 0\n",
      "s2483n 2837 0\n",
      "d2723h 2839 0\n",
      "deletion 2840 0\n",
      "l929s 2841 0\n",
      "a1170v 2845 0\n",
      "r2336h 2846 0\n",
      "d806h 2847 0\n",
      "v1643a 2848 0\n",
      "h1966y 2849 0\n",
      "t630i 2850 0\n",
      "e2856a 2851 0\n",
      "s1733f 2852 0\n",
      "e462g 2853 0\n",
      "i1349t 2854 0\n",
      "r2973c 2856 0\n",
      "d935n 2857 0\n",
      "d2665g 2858 0\n",
      "d3170g 2859 0\n",
      "r2602t 2860 0\n",
      "r2842c 2861 0\n",
      "c554w 2862 0\n",
      "n1228d 2863 0\n",
      "h1918y 2865 0\n",
      "g602r 2867 0\n",
      "r2659k 2871 0\n",
      "l2865v 2872 0\n",
      "n987i 2873 0\n",
      "d1420y 2874 0\n",
      "t2681r 2875 0\n",
      "k513r 2877 0\n",
      "n56t 2878 0\n",
      "w2626c 2879 0\n",
      "s206c 2880 0\n",
      "d2723g 2882 0\n",
      "t2722r 2883 0\n",
      "h2074n 2884 0\n",
      "g2274v 2885 0\n",
      "r2318q 2886 0\n",
      "r174c 2888 0\n",
      "d1280v 2891 0\n",
      "t3211k 2892 0\n",
      "r3052q 2893 0\n",
      "s2670l 2894 0\n",
      "q1396r 2896 0\n",
      "k607t 2897 0\n",
      "e106g 2900 0\n",
      "q538p 2906 0\n",
      "d29h 2924 0\n",
      "r544s 2935 0\n",
      "c481s 2936 0\n",
      "y551f 2937 0\n",
      "e14* 2938 0\n",
      "q209l 2942 0\n",
      "amplification 2943 0\n",
      "d717v 2947 0\n",
      "a1065t 2948 0\n",
      "q472h 2949 0\n",
      "deletion 2950 0\n",
      "truncating mutations 2951 0\n",
      "r844h 2953 0\n",
      "q227r 2955 0\n",
      "deletion 2958 0\n",
      "truncating mutations 2959 0\n",
      "d816f 2963 0\n",
      "e839k 2968 0\n",
      "k550_k558del 2970 0\n",
      "y553_k558del 2974 0\n",
      "p551_e554del 2976 0\n",
      "t417_d419delinsrg 2977 0\n",
      "t417_d419delinsi 2978 0\n",
      "n655k 2980 0\n",
      "d816y 2981 0\n",
      "c443y 2983 0\n",
      "p577_w582delinspyd 2989 0\n",
      "d820g 2990 0\n",
      "d419del 2994 0\n",
      "t574_r588delinsl 2996 0\n",
      "l576p 3001 0\n",
      "v559del 3003 0\n",
      "q556_k558del 3006 0\n",
      "p551_w557delinsl 3007 0\n",
      "d816h 3011 0\n",
      "i653t 3013 0\n",
      "n564_y578del 3015 0\n",
      "d737n 3018 0\n",
      "k558_v559del 3024 0\n",
      "v569_l576del 3027 0\n",
      "y553n 3029 0\n",
      "w557_v559delinsc 3036 0\n",
      "w557_v560del 3042 0\n",
      "y418_d419del 3044 0\n",
      "r796g 3047 0\n",
      "v555_v559del 3048 0\n",
      "e490k 3052 0\n",
      "m552_w557del 3055 0\n",
      "d816a 3056 0\n",
      "v559a 3057 0\n",
      "t574instqlpyd 3058 0\n",
      "i1616t 3082 0\n",
      "g2430a 3089 0\n",
      "mir143-notch1 fusion 3093 0\n",
      "v1671i 3094 0\n",
      "a2425t 3095 0\n",
      "1_2009trunc 3096 0\n",
      "2010_2471trunc 3097 0\n",
      "c157y 3098 0\n",
      "r882c 3105 0\n",
      "amplification 3108 0\n",
      "truncating mutations 3111 0\n",
      "truncating mutations 3112 0\n",
      "g264s 3113 0\n",
      "hmga2-rad51b fusion 3114 0\n",
      "truncating mutations 3115 0\n",
      "a11_g12insga 3119 0\n",
      "y32s 3124 0\n",
      "n116h 3127 0\n",
      "r164q 3131 0\n",
      "wildtype 3135 0\n",
      "a146v 3136 0\n",
      "k147e 3138 0\n",
      "amplification 3151 0\n",
      "y40a 3152 0\n",
      "truncating mutations 3160 0\n",
      "esrp1-raf1 fusion 3164 0\n",
      "atg7-raf1 fusion 3166 0\n",
      "amplification 3168 0\n",
      "s427g 3169 0\n",
      "srgap3-raf1 fusion 3170 0\n",
      "p261l 3171 0\n",
      "f161l 3174 0\n",
      "t196a 3176 0\n",
      "truncating mutations 3177 0\n",
      "p291qfs*51 3178 0\n",
      "n551k 3183 0\n",
      "truncating mutations 3184 0\n",
      "amplification 3186 0\n",
      "q61h 3194 0\n",
      "l481f 3198 0\n",
      "truncating mutations 3202 0\n",
      "r698w 3205 0\n",
      "n480del 3209 0\n",
      "deletion 3213 0\n",
      "amplification 3214 0\n",
      "r342q 3222 0\n",
      "chtop-ntrk1 fusion 3226 0\n",
      "tfg-ntrk1 fusion 3228 0\n",
      "r715g 3233 0\n",
      "etv6-ntrk3 fusion 3234 0\n",
      "btbd1-ntrk3 fusion 3235 0\n",
      "fusions 3236 0\n",
      "g623r 3238 0\n",
      "deletion 3239 0\n",
      "g505s 3244 0\n",
      "g774v 3245 0\n",
      "i638f 3246 0\n",
      "t654i 3248 0\n",
      "deletion 3253 0\n",
      "casp8l 3254 0\n",
      "c248t 3255 0\n",
      "ret-ccdc6 fusion 3257 0\n",
      "s891a 3262 0\n",
      "e632_l633del 3263 0\n",
      "c618r 3264 0\n",
      "k603q 3266 0\n",
      "d631g 3267 0\n",
      "e921k 3268 0\n",
      "fusions 3269 0\n",
      "s904f 3270 0\n",
      "l790f 3271 0\n",
      "r873q 3272 0\n",
      "s765p 3274 0\n",
      "r897q 3276 0\n",
      "c634s 3279 0\n",
      "c634w 3281 0\n",
      "m980t 3282 0\n",
      "kif5b-ret fusion 3284 0\n",
      "r833c 3285 0\n",
      "v804m 3286 0\n",
      "i852m 3290 0\n",
      "c620r 3291 0\n",
      "f893l 3295 0\n",
      "y791f 3296 0\n",
      "r177* 3297 0\n",
      "r139g 3299 0\n",
      "r177q 3301 0\n",
      "s70fsx93 3305 0\n",
      "w279* 3306 0\n",
      "h78q 3313 0\n",
      "g42r 3314 0\n",
      "runx1-runx1t1 fusion 3315 0\n"
     ]
    }
   ],
   "source": [
    "count = new_data[new_data[\"Score\"] == 1].apply(lambda line: check_line(line), axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 578,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Int64Index([ 290,  677,  697,  748,  946,  953,  966,  970, 1074, 1109,\n",
       "            ...\n",
       "            3303, 3304, 3305, 3306, 3308, 3310, 3311, 3313, 3314, 3315],\n",
       "           dtype='int64', name='ID', length=1445)"
      ]
     },
     "execution_count": 578,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index_hacker = count[count == 0].index\n",
    "index_hacker"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 584,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"follicular lymphoma, the most common indolent non-hodgkin lymphoma, remains a significant clinical burden, as the majority of affected individuals undergo multiple relapses and eventually develop resistance to standard therapies. furthermore, in a subset of individuals, follicular lymphoma transforms into the more aggressive diffuse large b cell lymphoma (dlbcl), with this transformation associated with poor clinical outcomes2, 3, 4. recent genetic profiling and case studies of donor-derived follicular lymphoma following stem cell transplantation suggest the putative existence of a 'long-lived' tumor-initiating progenitor cell compartment from which successive disease events occur5, 6, 7, 8. an in-depth characterization and chronicling of the underlying genetic events from follicular lymphoma to transformed follicular lymphoma will serve as a guide in the development of effective targeted therapies. to address this need, we conducted whole-genome or whole-exome sequencing of sequential follicular lymphoma–transformed follicular lymphoma (fl-tfl) pairs and matched germline samples from 10 cases (fig. 1a and online methods), with deep targeted sequencing of 28 genes in an extension cohort (supplementary fig. 1 and supplementary table 1).  figure 1: clinical timelines and somatic mutation profiles for the discovery cases. clinical timelines and somatic mutation profiles for the discovery cases. (a) biopsy information with disease event timelines for the ten whole-genome or whole-exome sequenced follicular lymphoma (fl) cases. (b) distribution of base substitution patterns for all somatic snvs identified in the six paired follicular lymphoma–transformed follicular lymphoma (fl-tfl) genomes. somatic snvs for each case were sorted into three categories: shared (variants present in both follicular lymphoma and transformed follicular lymphoma samples), follicular lymphoma specific and transformed follicular lymphoma specific. aberrant somatic hypermutation (ashm) did not contribute to the majority of variants detected, with the exception of previously described gene targets (supplementary table 13).  full size image (137 kb) figures index next for the ten index cases (six whole-genome sequenced and four whole-exome sequenced), the sequencing yielded mean genomic and exonic coverage of 37× and 110×, respectively, with 96% of the targeted bases covered by >10-fold (supplementary table 2). in the 6 whole-genome sequenced cases, we detected approximately 10,000 somatic variants per tumor (supplementary table 3). g>a/c>t transitions were the most common nucleotide substitutions, consistent with data from other cancer types (supplementary fig. 2a). we observed a higher frequency of transversions in the transformation-specific variants (p = 0.006; fig. 1b and supplementary fig. 2b), reflected in the lower transition/transversion (ti/tv) ratios for transformed samples (p = 0.008; supplementary fig. 2c). focusing on protein-altering changes, we identified between 21 and 143 nonsynonymous somatic variants per sample, with an increased frequency of mutations present after transformation in the majority of cases (supplementary fig. 3). in total, 1,560 protein-altering variants affecting 908 genes were detected across the 10 cases, comprising missense changes (84.8%), short indels (8.9%) and nonsense mutations (6.3%) (supplementary table 4). we verified 235 single-nucleotide variants (snvs) and 48 indels out of 293 variants, corresponding to 118 genes, thereby achieving concordance rates of 98% and 88%, respectively (supplementary table 5). interrogation of the six whole-genome sequenced cases identified copy number alterations (cnas), previously unreported structural variants, fusion transcripts and recurrent mutations in noncoding regions (5′ utrs, 3′ utrs and promoter regions) (supplementary fig. 4 and supplementary table 6).  to determine the evolutionary relationship between the 10 paired follicular lymphoma–transformed follicular lymphoma samples, we constructed phylogenetic trees for each case using the somatic variants detected (only nonsynonymous variants and all variants). each tree showed branching rather than linear evolution, with the 'trunk' representing genetic events shared by the follicular lymphoma tumor(s) and their paired transformed follicular lymphoma samples, thus supporting the presence of an ancestral cpc. two distinct patterns of evolution from the cpc emerged. in the first pattern, all cases with the exception of s5 and s9 showed high clonal semblance between the paired follicular lymphoma and transformed follicular lymphoma tumors, and we refer to this pattern as evolution through a 'rich' ancestral cpc (fig. 2a,b and supplementary figs. 5a–f and 6). the mutations harbored in each cpc demonstrated an enrichment for genes involved in chromatin regulation, with concurrent mutations occurring in kmt2d (also known as mll2), which encodes a histone methyltransferase responsible for histone h3 lysine 4 (h3k4) trimethylation, and other histone-modifying genes (crebbp, ep300, ezh2 and mef2b)9, 10, 11, 12. apart from frequent mutations in histone modifiers, other mutational targets in the eight rich cpcs included genes involved in immune modulation (b2m, cd58 and tnfrsf14), jak-stat signaling (socs1 and stat6) and b cell receptor (bcr)–nf-κb signaling (bcl10, card11 and cd79b), with many of these genes previously reported in dlbcl10, 11, 12, 13, 14, 15, 16. in contrast, a second, unique pattern of evolution was seen in cases s5 and s9, in which only four nonsynonymous mutations were shared by the follicular lymphoma and transformed follicular lymphoma samples (referred to as evolution through a 'sparse' cpc) (fig. 2c,d). intriguingly, in both cases, different kmt2d (s5 and s9), tnfrsf14 (s5) and crebbp (s9) mutations occurred in the diagnostic follicular lymphoma and transformed follicular lymphoma biopsies, appearing to have arisen from independent clones and reflecting a convergent pattern of evolution. these observations strongly support tumoral dependency on crebbp, kmt2d and tnfrsf14 alterations during lymphomagenesis and progression.  figure 2: patterns of evolution of follicular lymphoma to transformed follicular lymphoma for four cases. patterns of evolution of follicular lymphoma to transformed follicular lymphoma for four cases. (a–d) in each case, a phylogenetic tree was constructed using nonsynonymous mutations and is shown against the corresponding event timeline. all trees have shared mutations (trunk), which represent a common ancestral origin from a cpc, and unique branches, depicting the mutations present in that biopsy alone. the length of individual branches denotes the number of mutations separating the two disease events. s2 (a) and s7 (b), both with three follicular lymphomas (blue) and a single transformed follicular lymphoma (red) biopsies, show divergent clonal evolution from a rich cpc pool (highlighted by the dashed lines) from which subsequent events arise. s5 (c) and s9 (d) show a sparse cpc (four shared mutations between follicular lymphoma and transformed follicular lymphoma samples in both cases) and convergent evolution, in which similar genetic alterations occur independently on separate branches of the tree. all paired tumors were confirmed to be clonally related through bcl2-igh rearrangement analyses. the pattern of clonal evolution was identical with all types of somatic variants considered (supplementary fig. 6). functional annotation of the genes was based on previously reported functions10, 11, 12, 13, 14, 15, 16. gl, germ line.  full size image (198 kb) previous figures index next to evaluate the mutation prevalence for the genes identified in our discovery cohort, we performed deep targeted resequencing of 28 candidate genes (mainly prioritizing genes involved in functional processes not previously implicated in follicular lymphoma (supplementary tables 7 and 8)) in an extension cohort of 100 independent follicular lymphoma biopsies and 32 paired follicular lymphoma–transformed follicular lymphoma cases (including the 10 index cases), achieving a mean depth of coverage of 840×. a prominent feature of the genetic landscape was the presence of mutations targeting genes involved in chromatin regulation through histones and histone modifications (fig. 3 and supplementary fig. 7), confirming previous studies9, 10, 11, 12. notably, over 70% of cases had concurrent mutations in at least two of the histone-modifying enzymes screened (crebbp, ezh2, mef2b and kmt2d). kmt2d was mutated in 82% of our cohort, with approximately half of the cases harboring multiple mutations. mutation frequencies for the histone acetyltransferase crebbp and the histone methyltransferase ezh2 were higher than in earlier studies at 64% and 20%, respectively9, 10, 11.  figure 3: distribution of mutations in 28 genes in follicular lymphoma. distribution of mutations in 28 genes in follicular lymphoma. the heat map shows the pattern of case-specific, concurrent and mutually exclusive mutations in 100 follicular lymphoma cases. each column represents an individual affected case, and each row denotes a specific gene assigned to one of five functional categories (labeled on the left). the bar graph on the right shows the frequency of mutations found for each gene across all samples. a heat map corresponding to the paired follicular lymphoma–transformed follicular lymphoma series is included in supplementary figure 7b.  full size image (299 kb) previous figures index next  notably, linker histone genes were recurrently mutated in follicular lymphoma. these genes encode proteins that facilitate the folding of higher order chromatin structures and regulate the access of histone-modifying enzymes and chromatin-remodeling complexes to their target genes17, 18, 19. overall, we observed mutations affecting at least one histone h1 gene in 28% of our series, with hist1h1c and hist1h1e being the most frequently mutated (fig. 3 and supplementary table 9). most of the histone h1 alterations were missense and were clustered in the highly conserved globular domain (supplementary fig. 8), targeting residues directly involved in dna binding20, 21, 22, 23. to assess the functional relevance of these mutations, we generated histone h1 (hist1h1c, hist1h1d and hist1h1e) triple-knockout mouse embryonic stem cells (escs) overexpressing human hist1h1c encoding wild-type or ser102phe histone h1 protein18 (supplementary fig. 9). the p.ser102phe alteration in histone h1 resulted in greatly impaired association with chromatin relative to wild-type protein (fig. 4a,b), suggesting that the identified mutations most likely lead to a loss-of-function phenotype by reducing the binding affinities and residence times of the histone h1 subtypes in chromatin-compromising chromatin compaction and the regulation of specific genes.  figure 4: functional analyses of mutations in linker histone hist1h1c and ebf1. functional analyses of mutations in linker histone hist1h1c and ebf1. (a) reversed-phase (rp)-hplc profiles of histones extracted from chromatin isolated from histone h1 triple-knockout (tko) mouse escs expressing wild-type or ser102phe human histone h1. the ser102phe mutant demonstrated higher hydrophobicity than the wild-type protein. mh1a, mouse histone h1a; mh1b, mouse histone h1b; mh2b, mouse histone h2b; hh1c, human histone h1c. (b) ratio of individual histone h1 variants (and total histone h1) to the nucleosome of the indicated escs. the ratio is calculated38, 39 from the hplc analysis in a and demonstrates that the total histone h1 levels in histone h1 triple-knockout escs expressing human histone h1c ser102phe were reduced compared to cells expressing wild-type human histone h1c, as a result of the weaker association of the mutant histone with chromatin (only 35% of wild-type association). (c) schematics of the domains and locations of alterations in ebf1 detected by targeted resequencing (ncbi protein reference sequence np_076870.1). dbd, dna-binding domain; ipt, immunoglobulin domain; hlh, helix-loop-helix domain. coding mutations are colored green (missense), yellow (nonsense) or red (frameshift). alterations at the same amino acid in different cases are depicted. highlighted variants were selected for functional analyses. (d) quantitative rt-pcr (qrt-pcr)-based gene expression in ebf1-deficient primary mouse pre-pro-b cells transduced with virus expressing wild-type or mutant ebf1. all three ebf1 mutants had similar expression levels. both dbd mutants (gly171asp and ser238thr) exhibit abrogation of target gene activation. igll1 and cd79b transcript levels were normalized to the levels observed with control (pmys) vector, and ebf1 transcript levels of mutant ebf1 constructs were normalized to the levels of wild-type ebf1 (wt). error bars, s.d. of duplicate experiments.  full size image (250 kb) previous figures index next we observed frequent mutations in components of the jak-stat signaling pathway, including in stat6 (12%) and socs1 (8%) (supplementary fig. 10). similar socs1 and stat6 mutations have been described in other germinal center lymphomas24, 25 and have been shown to contribute to constitutive stat6 activation and the promotion of tumor cell survival in lymphoma cell lines26, 27, 28.  constitutive activation of the antiapoptotic nf-κb signaling pathway, caused by mutations affecting positive and negative regulators of signaling, is an established feature of the activated b cell–like subtype of dlbcl (abc dlbcl). in contrast, these mutations occur infrequently in germinal center b cell–like dlbcl (gcb dlbcl)13, 14, 29, 30. interestingly, we detected mutually exclusive mutations in this pathway in one-third of follicular lymphomas, with a predilection for card11 (11%) and tnfaip3 (11%) (fig. 3). the type and location of the detected aberrations were reminiscent of the mutations found in abc dlbcl, with the exception of variants in cd79b (supplementary fig. 10), which included frameshift mutations affecting the immunoreceptor tyrosine-based activation motif (itam) domain in follicular lymphoma in contrast to the predominant missense mutations affecting the tyr196 residue reported in abc dlbcl14. a number of emerging therapeutic agents target the nf-κb signaling cascade, and such treatments may be beneficial to a preselected subset of patients with follicular lymphoma harboring mutations in this pathway.  we identified mutations in genes important in b cell development in 17% of our cohort (fig. 3 and supplementary fig. 10). we focused on early b cell factor 1 (ebf1), a transcription factor frequently deleted in high-risk acute lymphoblastic leukemia31. the ebf1 protein functions as a homodimer and binds the consensus sequence 5′-tcccnnggga-′3 (refs. 32,33), and more than half of the mutations in our series affected conserved residues of the dna-binding domain (dbd) (fig. 4c). binding of ebf1 to target genes can elicit different responses, including transcriptional activation or repression or h3k4 dimethylation34. in particular, transcriptional activation of igll1, encoding the λ5 surrogate light chain of the pre–b cell receptor, has been used as an assay to assess the binding and function of ebf1 (ref. 33). therefore, we tested the effects of three ebf1 mutations, encoding p.gly171asp and p.ser238thr (located in the dbd) and p.arg381ser (located in the helix-loop-helix (hlh) domain), on the ability to activate expression of the ebf1 target genes igll1 and cd79b in ebf1-deficient primary mouse pre-pro-b cells (fig. 4d). gly171asp and ser238thr mutant lines demonstrated markedly impaired upregulation of both igll1 and cd79b expression, most likely reflecting reduced binding to regulatory elements in their respective promoters. the hlh mutant, arg381ser, resulted in a more modest reduction in target gene expression. taken together, the data suggest that the ebf1 mutations detected in follicular lymphoma lead to loss of function and a reduction in ebf1 target gene expression.  to discriminate early from late genetic events, we integrated variant allele frequencies (vafs) generated from our high-depth sequencing data with snp array copy number profiles for 29 cases (26 follicular lymphoma–transformed follicular lymphoma pairs and 3 diagnostic follicular lymphomas), using the absolute algorithm35. this scheme allowed us to correct for variations in tumor content and ploidy as well as locus-specific cnas, thereby establishing the accurate clonal or subclonal status of mutations in our 28 selected genes. mutations present in nearly all tumor cells (clonal) would suggest early events and therefore represent initiating 'driver' genes. our results showed that mutations in histone-modifying genes (kmt2d, crebbp and ezh2) as well as mutations in stat6 and tnfrsf14 were predominantly clonal events (fig. 5a). these results expand on the findings of a recent study implicating crebbp mutation as an early event in follicular lymphoma36. interestingly, many of these driver genes (crebbp, kmt2d and tnfrsf14) harbored compound heterozygous mutations that were frequently accompanied by either deletions or acquired uniparental disomy (aupd), serving as pathogenic mechanisms to render tumor cells homozygous for a pre-existing abnormality (supplementary fig. 11). moreover, our longitudinal analyses of paired follicular lymphoma–transformed follicular lymphoma samples demonstrated that mutations in these genes were stable and remained clonally dominant, irrespective of therapy, in the majority of cases as the disease progressed (fig. 5b and supplementary fig. 12), reaffirming that these mutations represent key events in the founding cpc that repopulates the tumor at transformation. targeted resequencing in additional follicular lymphoma–transformed follicular lymphoma pairs confirmed the findings in our discovery cohort, demonstrating evolution predominantly occurring via rich cpcs (supplementary figs. 7 and 12).  figure 5: early and late mutations in follicular lymphoma and transformed follicular lymphoma. early and late mutations in follicular lymphoma and transformed follicular lymphoma. (a) clonality profiles of 28 recurrent gene mutations. in cases with genes harboring multiple mutations, the mutation with the highest vaf was considered. (b) longitudinal analyses of paired follicular lymphoma–transformed follicular lymphoma samples show early initiating mutations with corrected vafs that were unchanged with disease progression in two representative cases. (c) acquisition of clonal ebf1 mutations (boxed) at transformation shown for two examples. (d) acquisition of nf-κb gene mutations (boxed) at transformation, particularly in myd88, which had no mutations in the extension cohort of 100 follicular lymphomas, suggesting that these mutations are predominantly late genetic events during the course of disease.  full size image (183 kb) previous figures index we next evaluated the follicular lymphoma–transformed follicular lymphoma pairs for previously unreported transformation-associated events. using copy number profiling, we found an increase in the number of regions affected by cnas and, notably, an enrichment of particular cnas at transformation, such as amplifications of ezh2, mdm2, myc and rel (supplementary fig. 13 and supplementary table 10). ebf1 mutations were acquired and were restricted to the transformation biopsies in 5 of 32 cases, with vafs indicating clonal events (fig. 5c). remarkably, we also found the acquisition of alterations affecting regulators of nf-κb signaling at transformation, which specifically included mutations in myd88 and tnfaip3 (fig. 5d and supplementary fig. 12). four of the five myd88 mutations identified were restricted to the transformation biopsy, with none harboring the p.leu265pro alteration common to other b cell lymphomas30, 37 (supplementary fig. 10). in all cases, these 'progressor' variants did not exist in the antecedent follicular lymphoma biopsies, suggesting that they were gained during the most recent clonal expansion leading to the transformation event. the acquisition of these late genetic events underscores the need for temporal mutational profiling as we move toward an era of precision medicine. it was particularly striking that the time interval from the biopsy of the antecedent follicular lymphoma to transformation in these cases was brief (range of 0.27–2.41 years), raising the possibility that the acquisition of particular genetic events is selectively advantageous through promoting higher fitness for the clone that drives transformation.  in summary, we report the most comprehensive sequencing effort so far in follicular lymphoma. the mutational landscape of follicular lymphoma predominantly indicates epigenetic 'addiction' but highlights co-occurring aberrations in genes involved in b cell development as well as jak-stat and nf-κb signaling. by longitudinal profiling, we provide unequivocal evidence for a reservoir ancestral population, enriched in early mutations in driver genes, that propagates successive disease events. we did not identify a single compelling genetic event responsible for transformation but instead demonstrate that the acquisition of distinct genetic alterations may prompt the onset of aggressive disease. the high frequency of relapse suggests that cpcs are resistant to standard therapies and that adopting a stratified treatment approach targeting the specific early genetic lesions identified in the putative cpc may ultimately offer the best chance of eradicating these cells and curing follicular lymphoma.  methods main• methods• accession codes• references• acknowledgments• author information• supplementary information case samples. ten individuals with follicular lymphoma were selected on the basis of the availability of sequential tumor lymph node biopsies (a sample with follicular lymphoma and a subsequent sample with histological transformation to dlbcl (transformed follicular lymphoma)) and matched germline samples consisting of remission bone marrows or peripheral blood specimens. clonality for each of the paired follicular lymphoma and transformed follicular lymphoma samples was confirmed by bcl2-igh breakpoint analysis, as previously described40. the clinical characteristics of these cases are presented in supplementary table 1. overall, 34 samples (10 paired follicular lymphoma–transformed follicular lymphoma–germline samples and 2 additional relapsed follicular lymphoma samples for cases s2 and s7) were evaluated. an extension cohort of 22 additional paired follicular lymphoma and transformed follicular lymphoma samples and 100 independent follicular lymphoma cases were included in our validation experiments (supplementary table 8). all biopsies were previously histologically reviewed to confirm diagnosis and tumor content before dna extraction from fresh-frozen tumor samples. tumor content was further estimated using the absolute35 (version 1.0.4) and ascat41 (version 2.1) algorithms. written consent was obtained for the collection and use of specimens for research purposes, with ethical approval obtained from the london research ethics committee (lrec) of the east london and the city health authority (10/h0704/65 and 06/q0605/69).  whole-genome sequencing. genomic dna (5 μg) from matched tumor and germline samples was fragmented to generate libraries with an average insert size of 300 bp and sequenced using the hiseq 2000 platform with paired-end reads of 100 bp, according to the manufacturer's protocols (illumina). detailed coverage data for all cases are shown in supplementary table 2.  whole-genome sequencing read mapping and somatic mutation detection. raw sequencing data were aligned to the reference human genome (hg19) using the illumina elandv2 aligner. alignments were sorted and converted to bam format. pcr duplicates were removed, and clusters of poorly aligned and anomalous reads were de novo assembled into contigs using the casava (v1.8) package (see urls). we implemented the strelka algorithm42 to identify somatic snvs and indels for matched tumor-germline samples, selecting this algorithm for its sensitivity of mutation detection with varying tumor purities. after candidate indel detection and realignment, we called somatic variants and performed filtering. only somatic calls originating from homozygous reference alleles in normal samples were considered. a joint somatic quality score (q) of ≥15 for snvs and ≥30 for indels was applied for all calls retained after filtering. the complete list of variants is detailed in supplementary table 4.  whole-exome sequencing and analysis. capture libraries were constructed from 3 μg of tumor and germline dna using the agilent sureselect xt human all exon v4 kit. enriched exome libraries were multiplexed and sequenced on the illumina hiseq 2500 platform to generate 100-bp paired-end reads (supplementary table 2). sequencing reads were aligned to the reference genome using burrows-wheeler aligner (bwa)43. sam to bam conversion and marking of pcr duplicates were performed using picard tools (version 1.86), followed by local realignment around indels and base quality score recalibration using the genome analysis toolkit (gatk)44 (v2.3.9). somatic snvs and indels were identified using the strelka pipeline.  annotation and recurrently mutated genes. all somatic snvs and short indels retained after filtering were annotated using the snpnexus tool45. the ensembl database (v63) was used for gene and transcript identification and the detection of amino acid changes. any variants overlapping with common and germline polymorphisms present in the dbsnp132 and hapmap databases were masked. the allele frequencies for individual variants in both germline and tumor samples were determined by read count. somatic variants identified with both whole-genome and whole-exome sequencing are included in supplementary table 4.  copy number and structural variant analyses from whole-genome sequencing data. cnas were identified using varscan2 (ref. 46) (v2.3.5), with each tumor sample compared to the matching germline sample. samples were segmented using the circular binary segmentation (cbs) algorithm47 implemented in the r bioconductor package dnacopy (v2.12), which allowed splits to be reversed if the copy number ratio for tumor and normal samples was within 3 s.d. structural variants were called using pindel48 (v0.2.4t), with the exception of translocations, which were called using default parameters. only calls that had no supporting reads in the normal sample but had ≥8 supporting reads in the tumor sample were retained. translocations (fusion transcripts) were identified using the delly49 and breakdancer50 algorithms on paired-end reads, only retaining structural variants called by both methods with ≥5 supporting reads in the tumor sample.  identification of recurrent mutations in promoter regions. using ensembl gene and transcript annotations, the somatic variants in the region from −2,000 to +250 bp (within the 5′ utr) relative to the transcription start site (tss) for each transcript of protein-coding genes were identified across whole-genome sequencing data for all cases. promoter regions recurrently mutated in multiple cases were identified, and mutation hotspots in these regions were also noted (supplementary table 6).  mutation and indel validation. snvs and indels were confirmed by bidirectional sanger sequencing, fluorescence-based fragment size analysis or targeted deep sequencing (fluidigm-miseq) as detailed below (supplementary table 5). concordance data for selected variants called by whole-genome sequencing, whole-exome sequencing and validation targeted deep sequencing with estimated vafs are shown in supplementary figure 14.  phylogenetic analysis. evolutionary trees were constructed for each of the ten cases analyzed from whole-genome or whole-exome sequencing data on the basis of the distance matrix between germline, follicular lymphoma and transformed follicular lymphoma samples derived from the numbers of somatic variants (all variants as well as nonsynonymous snvs and indels) from each biopsy, using the neighbor-joining algorithm51 implemented in the phylip package (see urls). once the consensus phylogenetic tree was determined, it was redrawn starting from germline and leading to the putative cpc, then progressing to the follicular lymphoma and transformed follicular lymphoma samples, with the branch length proportional to the number of somatic changes, i.e., genetic distance, between the samples. trees were plotted against the clinical timelines.  high-density snp arrays. copy number analysis and identification of regions of copy-neutral loss of heterozygosity (cnloh, also known as aupd) were performed on a cohort of 26 paired follicular lymphoma–transformed follicular lymphoma and 3 diagnostic follicular lymphoma dna samples (including 7 of our discovery cases) using affymetrix snp 6.0 microarrays. further details are given in the supplementary note.  targeted resequencing. twenty-eight candidate genes (supplementary table 8) were sequenced in the extension cohort cases using the access array platform (fluidigm). all samples were screened in duplicate, with a number of matched germline and normal tonsil dna samples included as controls in each run. access array amplification was performed in a multiplex format with genomic dna (50 ng) according to the manufacturer's recommendation. deep sequencing was performed on multiplexed library pools using the illumina miseq platform. after demultiplexing and fastq file generation for the raw data, reads were aligned to the reference genome using the bowtie2 algorithm52. samtools53 was used to generate sorted bam files for each sample, and the varscan2 tool was used to examine the pileup file to call variants, including snps and short indels, implementing a minimum threshold vaf of 5% and fisher's exact test p value of <0.05. only bases meeting the minimum base quality score of 20 from reads meeting the minimum mapping quality score of 20 were considered. a minimum read depth of 10 at a position was required to make calls, and variants with >90% of reads supporting one strand were excluded. only concordant variants present in both duplicates or confirmed by independent sanger sequencing were retained. identified variants were annotated using snpnexus to exclude variants reported in dbsnp132 and to identify variants that had nonsynonymous consequences or affected splice sites.  estimating the clonality of mutations. tumor purity, ploidy and absolute dna copy numbers were determined from affymetrix snp 6.0 data (n = 26 follicular lymphoma–transformed follicular lymphoma pairs and 3 diagnostic follicular lymphomas) by implementing the absolute algorithm (supplementary table 11). the ascat algorithm was also used to determine tumor purity and ploidy to ensure high concordance across both algorithms (supplementary table 11). these estimates were used to compute and classify individual somatic mutations obtained from our deep sequencing analyses as clonal or subclonal on the basis of the posterior probability that the cancer cell fraction (ccf) exceeded 0.95, as previously reported54 (supplementary table 12).  identification of recurrent targets of aberrant somatic hypermutation. shm hotspots typically occur within the 2-kb window downstream of the tss, and we focused our analyses on these regions55. the total number of snvs and the average mutation density across the 14 whole-genome sequenced tumor samples were investigated. for each gene, the ratio of transition to transversion mutations, the ratio of mutations at c:g to a:t sites and the percentage of snvs within the hotspot motif wrcy, their associated p values and the shm indicator were calculated in accordance with previous studies56. the probability of observing the actual number of snvs or more for each target gene was calculated on the basis of the poisson distribution given the expected number of snvs per shm target region if those snvs were uniformly randomly distributed in all target genes. a list of shm gene targets is included in supplementary table 13.  ebf1 mutagenesis, retroviral expression and target gene analysis. three follicular lymphoma–derived mutants (p.gly171asp and p.ser238thr alterations were located in the dbd, and p.arg381ser was located in the hlh domain) were generated to assess their ability to activate ebf1 target gene expression. ebf1-deficient pre-pro-b cell lines (cultured on op9 feeders in the presence of il-7, flt3 ligand and stem cell factor (scf)) were transduced with bicistronic retroviruses expressing wild-type or mutant ebf1-gfp as described previously34. gfp-positive cells were sorted 24 h after transduction, and expression was confirmed by immunoblotting (data not shown). total rna was extracted using trizol reagent (invitrogen) according to the manufacturer's instructions. qrt-pcr analysis was performed in triplicate on a 7500 fast real-time pcr system (applied biosystems) using power sybr green master mix to measure the mrna levels of the ebf1 target genes igll1 and cd79b (refs. 33,34).  generation of clones overexpressing wild-type and ser102phe histone h1c in histone h1c–histone h1d–histone h1e triple-knockout escs. expression plasmids encoding flag-tagged wild-type or ser102phe human histone h1c were constructed by replacing the mouse histone h1d coding sequence with the sequence encoding flag-tagged wild-type or mutant human histone h1c in a 5-kb fragment encompassing the mouse hist1h1d upstream and downstream regulatory sequences, which were inserted into a vector containing a blasticidin-resistant gene. plasmid dna (20 μg) was transfected into 2 × 107 histone h1 triple-knockout escs, as previously described57. a total of 48 clones (24 clones for transfection with each vector) resistant to blasticidin were picked and screened by immunoblotting using an anti-flag antibody (sigma-aldrich, f1804; 1:2,000 dilution). cell clones with the highest levels of expression of flag-tagged wild-type and mutant histone h1c were selected for further analysis. (supplementary fig. 9)  preparation and analysis of histones. chromatin was prepared from escs, and histones were extracted with 0.2 n sulfuric acid, according to previously described protocols38, 39, 58. approximately 50 μg of total histones was injected into a c18 reverse-phase column (vydac) on an akta upc10 system (ge healthcare). effluent was monitored at 214 nm, and peak areas were analyzed with akta unicorn (v5.11) software. absorbance at 214 nm for the histone h1 and h2b peaks was adjusted according to the respective peptide bonds, and the histone h1/nucleosome ratio was calculated as previously described39, 58.\""
      ]
     },
     "execution_count": 584,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concatenate_data.loc[290,\"Text\"].lower()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 581,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Gene                                                  HIST1H1C\n",
       "Variation                                                S102F\n",
       "Class                                                        1\n",
       "Text          most of the histone h1 alterations were misse...\n",
       "Score                                                        1\n",
       "Name: 290, dtype: object"
      ]
     },
     "execution_count": 581,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data.loc[290,]"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "deep_projet",
   "language": "python",
   "name": "deep_projet"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
