{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import re\n",
    "import time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "file_train_text = \"../../data/training_text/training_text\"\n",
    "file_train_variant = \"../../data/training_variants/training_variants\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ragou/.local/lib/python3.8/site-packages/pandas/util/_decorators.py:311: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n",
      "  return func(*args, **kwargs)\n"
     ]
    }
   ],
   "source": [
    "train_text = pd.read_csv(file_train_text, sep = '\\|\\|')\n",
    "train_text.index.name = \"ID\"\n",
    "train_text.columns = [\"Text\"]\n",
    "\n",
    "train_variant = pd.read_csv(file_train_variant)\n",
    "train_variant.set_index(\"ID\",inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Y371H</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CBL</td>\n",
       "      <td>K382E</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>CBL</td>\n",
       "      <td>C381A</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>CBL</td>\n",
       "      <td>C396R</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Variation  Class\n",
       "ID                      \n",
       "8   CBL     Y371H      4\n",
       "11  CBL     K382E      4\n",
       "13  CBL     C381A      4\n",
       "21  CBL     C396R      4"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_variant[train_variant[\"Variation\"].isin([\"C396R\", \"H398Q\", \"Y371H\", \"K382E\", \"C381A\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Variation  Class\n",
       "ID                      \n",
       "2   CBL     Q249E      2"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_variant[train_variant[\"Variation\"] == \"Q249E\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 Text\n",
       "ID                                                   \n",
       "6   Oncogenic mutations in the monomeric Casitas B..."
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_text.loc[[6],]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_count = train_variant[\"Gene\"].value_counts().loc[lambda x : x > 100].to_dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_class = train_variant[train_variant[\"Gene\"].isin(gene_count.keys())]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA4EAAAFNCAYAAAC+H2oqAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/MnkTPAAAACXBIWXMAAAsTAAALEwEAmpwYAAAcvElEQVR4nO3dfbBlVXkn4N8rDVFQAaFDoCFCRtQQJyr2KEqijqgRv3AcZYxRiWOKpEqNikkkJhUdJ6ZMYuJn4hQBFSZGRfADE2J0xK9kIrFBogJmbAkKLR8tIirGCPrOH2e3XtvGvug9Z/e9+3mqTt29115nn/fs7qru311rr13dHQAAAKbhNmMXAAAAwOIIgQAAABMiBAIAAEyIEAgAADAhQiAAAMCECIEAAAATIgQCsGpU1Yur6i9H/PwPVtWvDNu/VFXvXcFzX1xVDx62V/R7VtULq+rUlTofAKubEAjALqWqnlxVm6rqa1V1VVX9bVX93Nh1ba+739TdD99Zv6p6Y1X9/jLO9zPd/cEfta6qenBVXbnduf+gu3/lRz03AGuDEAjALqOqTkryyiR/kOSAJD+Z5M+THDdiWXNVVevGrgGAaRECAdglVNXeSV6S5Jnd/fbuvrG7b+rud3f3b97Ce95WVVdX1Q1V9eGq+pklxx5ZVZdU1VeraktV/cbQvn9V/XVVfbmqvlRVH6mqHf57WFUPq6pPD+d/bZJacuyXq+rvh+2qqldU1bVV9ZWq+mRV3aOqTkzyS0l+axjZfPfQ//KqekFVfSLJjVW1bmh76JKPv21VvXWo/8KquueSz+6qusuS/TdW1e9X1V5J/jbJQcPnfa2qDtp+emlVPXaYfvrlYYrrTy85dnlV/UZVfWL43m+tqtsu448QgFVCCARgV3H/JLdN8o5b8Z6/TXJ4kh9PcmGSNy05dlqSX+3uOyS5R5LzhvbnJ7kyyfrMRhtfmKS3P3FV7Z/k7Ul+N8n+ST6b5OhbqOPhSR6Y5K5J9k5yfJLruvuUoaY/6u7bd/djlrznF5M8Ksk+3X3zDs55XJK3JblTkr9K8s6q2v0Wr0SS7r4xybFJvjB83u27+wvbfa+7JnlzkucO1+DcJO+uqj2WdDs+ySOSHJbkZ5P88g/6XABWFyEQgF3Ffkm+eAuBaIe6+/Xd/dXu/vckL05yz2FEMUluSnJEVd2xu6/v7guXtB+Y5M7DSONHuvv7QmCSRya5uLvP6u6bMpumevUtlHJTkjskuXuS6u5Lu/uqnZT/6u6+orv/7RaOX7Dks/80s4B81E7OuRz/LcnfdPf7hnO/PMntkjxgu9q+0N1fSvLuJPdagc8FYBchBAKwq7guyf7LvUeuqnarqpdV1Wer6itJLh8O7T/8/K+ZBbnPVdWHqur+Q/sfJ9mc5L1VdVlVnXwLH3FQkiu27QxB8Yoddezu85K8NsmfJbm2qk6pqjvu5Cvs8Fw7Ot7d385s9PKgnbxnOQ5K8rntzn1Fkg1L+iwNu19PcvsV+FwAdhFCIAC7in9M8u9JHrfM/k/ObMrkQzObgnno0F5J0t0f6+7jMpsq+s4kZw7tX+3u53f3TyV5bJKTquqYHZz/qiSHbNupqlq6v73ufnV33yfJEZlNC912H+OORhl/UPs2Sz/7NkkOTrJtaufXk+y5pO9P3IrzfiHJnZece9v32rKT9wGwRgiBAOwSuvuGJL+X5M+q6nFVtWdV7V5Vx1bVH+3gLXfILDRel1kg+oNtB6pqj+E5fnsPUx6/kuTbw7FHV9VdhvBzQ5JvbTu2nb9J8jNV9fhhdPLX871h6zuq6j9V1f2Ge/ZuTPKNJee8JslP3crLkST3WfLZzx2+60eHYxclefIwGvqIJA9a8r5rkuy3ZFrs9s5M8qiqOmao9/nDuf/vD1EjAKuQEAjALqO7/yTJSZktxrI1s2mKz8psJG97Z2Q2rXFLkkvy3YC0zVOTXD5MFf21zFbpTGYLyfyfJF/LbPTxz7v7Azuo5YtJnpjkZZkFzcOT/MMtlH7HJH+R5Pqhpusym3aazBaoOWJYiXNH3+OWvCuz+/euH77L44dAmyTPSfKYJF8evtd3ztvdn85s4ZfLhs/8nimk3f0vSZ6S5DVJvjic5zHd/c1bURsAq1jt+F54AAAA1iIjgQAAABMytxBYVa8fHpr7qSVtd6qq91XVZ4af+w7tVVWvrqrNw8Npj1zynhOG/p+pqhPmVS8AAMAUzHMk8I2ZPWh2qZOTvL+7D0/y/mE/mT3Y9vDhdWKS1yWz0JjkRUnul+S+SV60LTgCAABw680tBHb3h5N8abvm45KcPmyfnu8uA35ckjN65qNJ9qmqA5P8QpL3dfeXuvv6JO/L9wdLAAAAlmnR9wQe0N1XDdtXJzlg2N6Q731o7pVD2y21AwAA8ENYN9YHd3dX1YotTVpVJ2Y2lTR77bXXfe5+97uv1KkBAABWlQsuuOCL3b1+R8cWHQKvqaoDu/uqYbrntUP7liSHLOl38NC2JcmDt2v/4I5O3N2nJDklSTZu3NibNm1a2coBAABWiar63C0dW/R00HOSbFvh84TMHoS7rf1pwyqhRyW5YZg2+ndJHl5V+w4Lwjx8aAMAAOCHMLeRwKp6c2ajePtX1ZWZrfL5siRnVtUzknwuyfFD93OTPDLJ5iRfT/L0JOnuL1XV/0zysaHfS7p7+8VmAAAAWKbqXrHb8nYZpoMCAABTVlUXdPfGHR1b9HRQAAAARiQEAgAATIgQCAAAMCFCIAAAwIQIgQAAABMiBAIAAEyIEAgAADAhQiAAAMCErBu7AAAAdu7Sl543dgmrwk//zkPGLgF2eUYCAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJmSUEFhVz6uqi6vqU1X15qq6bVUdVlXnV9XmqnprVe0x9P2xYX/zcPzQMWoGAABYCxYeAqtqQ5JfT7Kxu++RZLckT0ryh0le0d13SXJ9kmcMb3lGkuuH9lcM/QAAAPghjDUddF2S21XVuiR7JrkqyUOSnDUcPz3J44bt44b9DMePqapaXKkAAABrx8JDYHdvSfLyJJ/PLPzdkOSCJF/u7puHblcm2TBsb0hyxfDem4f++y2yZgAAgLVijOmg+2Y2undYkoOS7JXkEStw3hOralNVbdq6deuPejoAAIA1aYzpoA9N8q/dvbW7b0ry9iRHJ9lnmB6aJAcn2TJsb0lySJIMx/dOct32J+3uU7p7Y3dvXL9+/by/AwAAwKo0Rgj8fJKjqmrP4d6+Y5JckuQDSZ4w9DkhybuG7XOG/QzHz+vuXmC9AAAAa8YY9wSen9kCLxcm+eRQwylJXpDkpKranNk9f6cNbzktyX5D+0lJTl50zQAAAGvFup13WXnd/aIkL9qu+bIk991B328keeIi6gIAAFjrxnpEBAAAACMQAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAlZN3YBAGvNhx74oLFLWBUe9OEPjV0CAEySkUAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmJBRQmBV7VNVZ1XVp6vq0qq6f1XdqareV1WfGX7uO/Stqnp1VW2uqk9U1ZFj1AwAALAWjDUS+Kok7+nuuye5Z5JLk5yc5P3dfXiS9w/7SXJsksOH14lJXrf4cgEAANaGhYfAqto7yQOTnJYk3f3N7v5ykuOSnD50Oz3J44bt45Kc0TMfTbJPVR240KIBAADWiDFGAg9LsjXJG6rq41V1alXtleSA7r5q6HN1kgOG7Q1Jrljy/iuHNgAAAG6lMULguiRHJnldd987yY357tTPJEl3d5K+NSetqhOralNVbdq6deuKFQsAALCWjBECr0xyZXefP+yflVkovGbbNM/h57XD8S1JDlny/oOHtu/R3ad098bu3rh+/fq5FQ8AALCaLTwEdvfVSa6oqrsNTcckuSTJOUlOGNpOSPKuYfucJE8bVgk9KskNS6aNAgAAcCusG+lzn53kTVW1R5LLkjw9s0B6ZlU9I8nnkhw/9D03ySOTbE7y9aEvAAAAP4RRQmB3X5Rk4w4OHbODvp3kmfOuCQAAYArGek4gAAAAIxACAQAAJkQIBAAAmBAhEAAAYELGWh10l3Cf3zxj7BJWhQv++GljlwAAAKwQI4EAAAATIgQCAABMiBAIAAAwIUIgAADAhAiBAAAAE7KsEFhV719OGwAAALu2H/iIiKq6bZI9k+xfVfsmqeHQHZNsmHNtAAAArLCdPSfwV5M8N8lBSS7Id0PgV5K8dn5lAQAAMA8/MAR296uSvKqqnt3dr1lQTQAAAMzJzkYCkyTd/ZqqekCSQ5e+p7vPmFNdAAAAzMGyQmBV/e8k/yHJRUm+NTR3EiEQAABgFVlWCEyyMckR3d3zLAYAAID5Wu5zAj+V5CfmWQgAAADzt9yRwP2TXFJV/5Tk37c1dvdj51IVAAAAc7HcEPjieRYBAADAYix3ddAPzbsQAAAA5m+5q4N+NbPVQJNkjyS7J7mxu+84r8IAAABYecsdCbzDtu2qqiTHJTlqXkUBAAAwH8tdHfQ7euadSX5h5csBAABgnpY7HfTxS3Zvk9lzA78xl4oAAACYm+WuDvqYJds3J7k8symhAAAArCLLvSfw6fMuBAAAgPlb1j2BVXVwVb2jqq4dXmdX1cHzLg4AAICVtdyFYd6Q5JwkBw2vdw9tAAAArCLLDYHru/sN3X3z8HpjkvVzrAsAAIA5WG4IvK6qnlJVuw2vpyS5bp6FAQAAsPKWGwL/e5Ljk1yd5KokT0jyy3OqCQAAgDlZ7iMiXpLkhO6+Pkmq6k5JXp5ZOAQAAGCVWO5I4M9uC4BJ0t1fSnLv+ZQEAADAvCw3BN6mqvbdtjOMBC53FBEAAIBdxHKD3J8k+ceqetuw/8QkL51PSaxln3/Jfxy7hFXhJ3/vk2OXAADAGrWsENjdZ1TVpiQPGZoe392XzK8sAAAA5mHZUzqH0Cf4AQAArGLLvScQAACANUAIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBCRguBVbVbVX28qv562D+sqs6vqs1V9daq2mNo/7Fhf/Nw/NCxagYAAFjtxhwJfE6SS5fs/2GSV3T3XZJcn+QZQ/szklw/tL9i6AcAAMAPYZQQWFUHJ3lUklOH/UrykCRnDV1OT/K4Yfu4YT/D8WOG/gAAANxKY40EvjLJbyX59rC/X5Ivd/fNw/6VSTYM2xuSXJEkw/Ebhv4AAADcSgsPgVX16CTXdvcFK3zeE6tqU1Vt2rp160qeGgAAYM0YYyTw6CSPrarLk7wls2mgr0qyT1WtG/ocnGTLsL0lySFJMhzfO8l125+0u0/p7o3dvXH9+vXz/QYAAACr1MJDYHf/dncf3N2HJnlSkvO6+5eSfCDJE4ZuJyR517B9zrCf4fh53d0LLBkAAGDN2JWeE/iCJCdV1ebM7vk7bWg/Lcl+Q/tJSU4eqT4AAIBVb93Ou8xPd38wyQeH7cuS3HcHfb6R5IkLLQwAAGCN2pVGAgEAAJgzIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYkHVjFwAArD4vfcoTxi5hVfidvzxr7BIAvo+RQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQjwsHoBV77XPf/fYJawKz/qTx4xdAgC7ACOBAAAAEyIEAgAATIgQCAAAMCFCIAAAwIQIgQAAABMiBAIAAEyIEAgAADAhCw+BVXVIVX2gqi6pqour6jlD+52q6n1V9Znh575De1XVq6tqc1V9oqqOXHTNAAAAa8UYI4E3J3l+dx+R5Kgkz6yqI5KcnOT93X14kvcP+0lybJLDh9eJSV63+JIBAADWhoWHwO6+qrsvHLa/muTSJBuSHJfk9KHb6UkeN2wfl+SMnvlokn2q6sDFVg0AALA2jHpPYFUdmuTeSc5PckB3XzUcujrJAcP2hiRXLHnblUPb9uc6sao2VdWmrVu3zq9oAACAVWy0EFhVt09ydpLndvdXlh7r7k7St+Z83X1Kd2/s7o3r169fwUoBAADWjlFCYFXtnlkAfFN3v31ovmbbNM/h57VD+5Ykhyx5+8FDGwAAALfSGKuDVpLTklza3X+65NA5SU4Ytk9I8q4l7U8bVgk9KskNS6aNAgAAcCusG+Ezj07y1CSfrKqLhrYXJnlZkjOr6hlJPpfk+OHYuUkemWRzkq8nefpCqwUAAFhDFh4Cu/vvk9QtHD5mB/07yTPnWhQAAMBEjLo6KAAAAIslBAIAAEyIEAgAADAhQiAAAMCECIEAAAATIgQCAABMiBAIAAAwIUIgAADAhAiBAAAAEyIEAgAATIgQCAAAMCFCIAAAwIQIgQAAABMiBAIAAEyIEAgAADAhQiAAAMCECIEAAAATIgQCAABMiBAIAAAwIUIgAADAhAiBAAAAEyIEAgAATIgQCAAAMCFCIAAAwIQIgQAAABOybuwCgPk6+jVHj13CqvAPz/6HsUsAAFgII4EAAAATIgQCAABMiBAIAAAwIUIgAADAhAiBAAAAEyIEAgAATIgQCAAAMCFCIAAAwIQIgQAAABMiBAIAAEyIEAgAADAhQiAAAMCECIEAAAATsm7sAgAAAJLkzLfdd+wSVoXjn/hPP9L7jQQCAABMiBAIAAAwIUIgAADAhAiBAAAAE2JhGAAA2IEXv/jFY5ewKrhOq8+qGQmsqkdU1b9U1eaqOnnsegAAAFajVRECq2q3JH+W5NgkRyT5xao6YtyqAAAAVp9VEQKT3DfJ5u6+rLu/meQtSY4buSYAAIBVZ7WEwA1Jrliyf+XQBgAAwK1Q3T12DTtVVU9I8oju/pVh/6lJ7tfdz1rS58QkJw67d0vyLwsvdGXsn+SLYxcxMa754rnmi+eaL55rvniu+eK55ovnmi/ear3md+7u9Ts6sFpWB92S5JAl+wcPbd/R3ackOWWRRc1DVW3q7o1j1zElrvniueaL55ovnmu+eK754rnmi+eaL95avOarZTrox5IcXlWHVdUeSZ6U5JyRawIAAFh1VsVIYHffXFXPSvJ3SXZL8vruvnjksgAAAFadVRECk6S7z01y7th1LMCqn9K6Crnmi+eaL55rvniu+eK55ovnmi+ea754a+6ar4qFYQAAAFgZq+WeQAAAAFaAELiLqKrXV9W1VfWpsWuZgqo6pKo+UFWXVNXFVfWcsWta66rqtlX1T1X1z8M1/x9j1zQVVbVbVX28qv567FqmoKour6pPVtVFVbVp7HqmoKr2qaqzqurTVXVpVd1/7JrWsqq62/D3e9vrK1X13LHrWuuq6nnDv5+fqqo3V9Vtx65prauq5wzX++K19nfcdNBdRFU9MMnXkpzR3fcYu561rqoOTHJgd19YVXdIckGSx3X3JSOXtmZVVSXZq7u/VlW7J/n7JM/p7o+OXNqaV1UnJdmY5I7d/eix61nrquryJBu7ezU+U2pVqqrTk3yku08dVhHfs7u/PHJZk1BVu2X22K77dffnxq5nraqqDZn9u3lEd/9bVZ2Z5NzufuO4la1dVXWPJG9Jct8k30zyniS/1t2bRy1shRgJ3EV094eTfGnsOqaiu6/q7guH7a8muTTJhnGrWtt65mvD7u7Dy2+h5qyqDk7yqCSnjl0LzENV7Z3kgUlOS5Lu/qYAuFDHJPmsALgQ65LcrqrWJdkzyRdGrmet++kk53f317v75iQfSvL4kWtaMUIgk1dVhya5d5LzRy5lzRumJV6U5Nok7+tu13z+Xpnkt5J8e+Q6pqSTvLeqLqiqE8cuZgIOS7I1yRuGac+nVtVeYxc1IU9K8uaxi1jruntLkpcn+XySq5Lc0N3vHbeqNe9TSX6+qvarqj2TPDLJISPXtGKEQCatqm6f5Owkz+3ur4xdz1rX3d/q7nslOTjJfYepFsxJVT06ybXdfcHYtUzMz3X3kUmOTfLMYbo/87MuyZFJXtfd905yY5KTxy1pGoapt49N8raxa1nrqmrfJMdl9kuPg5LsVVVPGbeqta27L03yh0nem9lU0IuSfGvMmlaSEMhkDfelnZ3kTd399rHrmZJhqtYHkjxi5FLWuqOTPHa4R+0tSR5SVX85bklr3/Ab+3T3tUnekdn9JMzPlUmuXDKz4KzMQiHzd2ySC7v7mrELmYCHJvnX7t7a3TcleXuSB4xc05rX3ad19326+4FJrk/y/8auaaUIgUzSsEjJaUku7e4/HbueKaiq9VW1z7B9uyQPS/LpUYta47r7t7v74O4+NLMpW+d1t98cz1FV7TUsNpVhSuLDM5tSxJx099VJrqiquw1NxySxyNdi/GJMBV2Uzyc5qqr2HP4Pc0xm6xkwR1X148PPn8zsfsC/GreilbNu7AKYqao3J3lwkv2r6sokL+ru08atak07OslTk3xyuEctSV7Y3eeOV9Kad2CS04eV5G6T5Mzu9sgC1poDkrxj9n+0rEvyV939nnFLmoRnJ3nTMD3xsiRPH7meNW/4JcfDkvzq2LVMQXefX1VnJbkwyc1JPp7klHGrmoSzq2q/JDcleeZaWnTKIyIAAAAmxHRQAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgFgGarqJ6rqLVX12aq6oKrOraq7VpXnAAKwqnhOIADsxPBw5nckOb27nzS03TOz5wICwKpiJBAAdu4/J7mpu//Xtobu/uckV2zbr6pDq+ojVXXh8HrA0H5gVX24qi6qqk9V1c9X1W5V9cZh/5NV9bzFfyUApspIIADs3D2SXLCTPtcmeVh3f6OqDk/y5iQbkzw5yd9190urarckeya5V5IN3X2PJKmqfeZVOABsTwgEgJWxe5LXVtW9knwryV2H9o8leX1V7Z7knd19UVVdluSnquo1Sf4myXvHKBiAaTIdFAB27uIk99lJn+cluSbJPTMbAdwjSbr7w0kemGRLkjdW1dO6+/qh3weT/FqSU+dTNgB8PyEQAHbuvCQ/VlUnbmuoqp9NcsiSPnsnuaq7v53kqUl2G/rdOck13f0XmYW9I6tq/yS36e6zk/xukiMX8zUAwHRQANip7u6q+i9JXllVL0jyjSSXJ3nukm5/nuTsqnpakvckuXFof3CS36yqm5J8LcnTkmxI8oaq2vbL2N+e93cAgG2qu8euAQAAgAUxHRQAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmJD/DwTZnzDiG3ukAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 1080x360 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure(figsize=(15,5))\n",
    "sns.countplot(x = train_variant[\"Class\"])\n",
    "plt.title(\"Class distribution\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA3sAAAHwCAYAAAAfJXbRAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/MnkTPAAAACXBIWXMAAAsTAAALEwEAmpwYAAAjSElEQVR4nO3de7ztdV3n8fcnwPsNhxMikMeULKnEPJqXMs2Z8TIVWkr6cJRxVGpCs8lJaXSUUiceo6mTlQ6NJjyGvOUlbRjvd8fUAyEK6EgqAnI5miJegsDP/LF+G5fbffbZ53DW3vt8eT4fj/3Ya/1u67sWi8V68bvs6u4AAAAwlh/a6AEAAACw94k9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9ANZdVW2tqq6q/af7/6eqjt1L2/75qvrs3P0vVtW/3BvbnrZ3TlU9YG9tDwAWRewBDKyqHl1VH6uqb1XV5dPt36qq2uixzevuh3b3KbtabgrEO+9iWx/q7rvsjXFV1aur6vnLtn9kd79/b2x/b6iqp1TV9qq6qqpevcL8B1XVZ6rq21X1vqq6w9y8G1fVq6rqG1V1aVX97roOfheq6t9V1Yc3ehwA+yqxBzCoqnp6kv+e5IVJbpfk4CS/meR+SW60gUNbmKU9hfuiqjp4D1f9cpLnJ3nVCts8KMmbkvyXJLdNsj3J6+YWOTHJEUnukOSBSZ5RVQ/Zw3EAsMmIPYABVdWtk/xhkt/q7r/u7it75u+7+7HdfdW03I2r6kVV9aWquqyqXlFVN53mPaCqLqqqp097BS+pqifMPcZO111hPPtNy36lqj6f5N8sm//+qnrSdPvOVfWBqrpiWv510/QPTot/sqq+WVW/PjfGZ1bVpUn+cmnasiHcs6rOraqvVdVfVtVNpm3+wJ6jpb2HVXVcksdmFkDfrKq3TfOvOyx0eg1eWlVfnn5eWlU3Xsvrt4ITpzH+XlXdbpXlvk93v6m735LkqyvM/tUk53T3G7r7nzKLu7tV1Y9P849N8rzu/lp3n5fkL5L8u509VlU9uarOq6orp7H+zDT9J6Z/hl+fDnP9lbl1rvtnO93/vtd8er1/s6o+N63/ZzXzE0lekeQ+0+v/9bW+JgDMiD2AMd0nyY2T/M0uljspyY8lOSrJnZMcmuQ5c/Nvl+TW0/QnJvmzqjpwjevOe3KSX0py9yTbkjxylTE9L8k7kxyY5LAkL0uS7r7/NP9u3X2L7l7aQ3W7zPZa3SHJcTvZ5mOTPDjJnaYxP3uVx8/0eCcnOS3Jf5se75dXWOxZSe6d2WtwtyT3Wrbt1V6/5Y5P8ttJfjrJZ6vqrVX1iKo6YFdjXcWRST4595y+leQfkhw5jeOQ+fnT7SNX2lBVPSqzWHx8klsl+ZUkX53G97bM/pn9cJKnJjmtqnbnUNpfSnLPzJ77MUkePMXnbyb56PT632Y3tgdAxB7AqA5K8pXuvmZpQlX932nPyXeq6v5VVZnF0X/s7n/s7iuT/Nckj57bzj8n+cPu/ufuPj3JN5PcZY3rzjsmyUu7+8Lu/sckf7TK2P85s3C7fXf/U3fv6pyt7yZ5bndf1d3f2ckyfzr32C9I8phdbHOtHpvZ63N5d+9I8gdJHjc3f8XXb6UNdfd3u/vd3f24zCL3LUn+Y5KLq+p5ezi+WyS5Ytm0K5LccpqXZfOX5q3kSZmF7yemvcTnd/cFmcXuLZKc1N1Xd/d7k/xtdu81Pqm7v97dX0ryvsziGYDrSewBjOmrSQ6aP4etu+877R35amaf/1uS3CzJGVMEfj3J26fp121nPhiTfDuzL/ZrWXfe7ZNcOHf/glXG/owkleTj0yGB/34Xz3XHdIjiapY/9u13sfxa3T7f/1yWb3tnr9+qpng+O8lZSQ7ITgJxDb6Z2V64ebdKcuU0L8vmL81byeGZ7RVc7vZJLuzu785NuyCzvZlrdenc7TW9RgDsmtgDGNNHk1yV5OhVlvlKku8kObK7bzP93Lq71/JFe3fXvSSzWFjyIzvbcHdf2t1P7u7bJ/mNJH9eq1+Bs9cw3uWP/eXp9rcyi9YkyQrnyu1q21/ObC/kStvebVV1WFWdUFXnJnltkh2ZHbZ6zB5u8pzMDi9d2v7NMzuU9Zzu/lpm/1zuNrf83aZ1VnLhtO5yX05yeFXNf6f4kSQXT7e/7zXO7NDWtVrLP1sAdkLsAQyou7+e2SGFf15Vj6yqW1bVD1XVUUluPi3z3cwuyPGSqvrhJKmqQ6vqwWvY/u6u+/okvz3FzIFJTtjZtqvqUVV12HT3a5l94V/aa3RZkh/d1fhWcPz02LfN7Dy7pfP9PpnZ+WtHTRdtOXHZert6vNckeXZVbanZlS+fk+R/7cH4UlUnZhZad8nsXLUjuvt506GNq623/zT2/ZLsV1U3mduj++YkP1lVvzYt85wkZ3f3Z6b5p07jP3C6aMuTk7x6Jw/1P5P8p6q6x3QBlTvX7M84fCyzvXHPqKoDavY3CH85s1hNZnsnf7WqbjZF+xN342W5LMlhVTXk1WMBFk3sAQyqu/9bkt/N7LDIy6af/5HkmUn+77TYM5Ocn+TvquobSd6dtR8yuDvr/kWSd2QWV2dm9ucAduaeST5WVd9M8tYkT+vuz0/zTkxyynTo6O7s7fqrzC4g8vnMDkV8fpJ09//L7Kql707yuSTLzw98ZZK7To/3lhW2+/zM/pzB2Uk+NT2356+w3Fq8JbPzFJ/Q3R/s7rXu1Xp2ZntZT0jyb6fbz06S6TzCX8vsPMWvJfnZfP95lc/N7PW4IMkHkrywu9++0oN09xum7fxVZod6viXJbbv76szi7qGZ7fH98ySPnwvKlyS5OrP33ymZXfRmrd6bWQBfWlVf2Y31AEhSa/9vCQAAAPsKe/YAAAAGJPYAAAAGJPYAAAAGJPYAAAAGJPYAAAAGtP+uF9m8DjrooN66detGDwMAAGBDnHHGGV/p7i0rzdunY2/r1q3Zvn37Rg8DAABgQ1TVBTub5zBOAACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAe2/0QPYaPf4vVM3eghsQme88PEbPQQAALhe7NkDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAY0MJir6oOr6r3VdW5VXVOVT1tmn5iVV1cVWdNPw+bW+f3q+r8qvpsVT14UWMDAAAY3f4L3PY1SZ7e3WdW1S2TnFFV75rmvaS7XzS/cFXdNcmjkxyZ5PZJ3l1VP9bd1y5wjAAAAENa2J697r6ku8+cbl+Z5Lwkh66yytFJXtvdV3X3F5Kcn+ReixofAADAyNblnL2q2prk7kk+Nk16SlWdXVWvqqoDp2mHJrlwbrWLsnocAgAAsBMLj72qukWSNyb5ne7+RpKXJ7lTkqOSXJLkj3dze8dV1faq2r5jx469PVwAAIAhLDT2quqAzELvtO5+U5J092XdfW13fzfJX+R7h2penOTwudUPm6Z9n+4+ubu3dfe2LVu2LHL4AAAA+6xFXo2zkrwyyXnd/eK56YfMLfaIJJ+ebr81yaOr6sZVdcckRyT5+KLGBwAAMLJFXo3zfkkel+RTVXXWNO0/J3lMVR2VpJN8MclvJEl3n1NVr09ybmZX8jzelTgBAAD2zMJir7s/nKRWmHX6Kuu8IMkLFjUmAACAG4p1uRonAAAA60vsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADGhhsVdVh1fV+6rq3Ko6p6qeNk2/bVW9q6o+N/0+cJpeVfUnVXV+VZ1dVT+zqLEBAACMbpF79q5J8vTuvmuSeyc5vqrumuSEJO/p7iOSvGe6nyQPTXLE9HNckpcvcGwAAABDW1jsdfcl3X3mdPvKJOclOTTJ0UlOmRY7JcnDp9tHJzm1Z/4uyW2q6pBFjQ8AAGBk63LOXlVtTXL3JB9LcnB3XzLNujTJwdPtQ5NcOLfaRdM0AAAAdtPCY6+qbpHkjUl+p7u/MT+vuztJ7+b2jquq7VW1fceOHXtxpAAAAONYaOxV1QGZhd5p3f2mafJlS4dnTr8vn6ZfnOTwudUPm6Z9n+4+ubu3dfe2LVu2LG7wAAAA+7BFXo2zkrwyyXnd/eK5WW9Ncux0+9gkfzM3/fHTVTnvneSKucM9AQAA2A37L3Db90vyuCSfqqqzpmn/OclJSV5fVU9MckGSY6Z5pyd5WJLzk3w7yRMWODYAAIChLSz2uvvDSWonsx+0wvKd5PhFjQcAAOCGZF2uxgkAAMD6EnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAAD2n+jBwCs7Et/+FMbPQQ2oR95zqc2eggAwD7Cnj0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABLSz2qupVVXV5VX16btqJVXVxVZ01/Txsbt7vV9X5VfXZqnrwosYFAABwQ7DIPXuvTvKQFaa/pLuPmn5OT5KqumuSRyc5clrnz6tqvwWODQAAYGgLi73u/mCSf1zj4kcneW13X9XdX0hyfpJ7LWpsAAAAo9uIc/aeUlVnT4d5HjhNOzTJhXPLXDRN+wFVdVxVba+q7Tt27Fj0WAEAAPZJ6x17L09ypyRHJbkkyR/v7ga6++Tu3tbd27Zs2bKXhwcAADCGdY297r6su6/t7u8m+Yt871DNi5McPrfoYdM0AAAA9sBux15VHVhVP70nD1ZVh8zdfUSSpSt1vjXJo6vqxlV1xyRHJPn4njwGAAAAyf5rWaiq3p/kV6blz0hyeVV9pLt/d5V1XpPkAUkOqqqLkjw3yQOq6qgkneSLSX4jSbr7nKp6fZJzk1yT5PjuvnbPnhIAAABrir0kt+7ub1TVk5Kc2t3PraqzV1uhux+zwuRXrrL8C5K8YI3jAQAAYBVrPYxz/+kQzGOS/O0CxwMAAMBesNbY+4Mk70hyfnd/oqp+NMnnFjcsAAAAro+1HsZ5SXdfd1GW7v58Vb14QWMCAADgelrrnr2XrXEaAAAAm8Cqe/aq6j5J7ptkS1XNX3nzVkn2W+TAAAAA2HO7OozzRkluMS13y7np30jyyEUNCgAAgOtn1djr7g8k+UBVvbq7L1inMQEAAHA9rfUCLTeuqpOTbJ1fp7t/cRGDAgAA4PpZa+y9IckrkvzPJNcubjgAAADsDWuNvWu6++ULHQkA+4z7vex+Gz0ENqGPPPUjGz0EAOas9U8vvK2qfquqDqmq2y79LHRkAAAA7LG17tk7dvr9e3PTOsmP7t3hAAAAsDesKfa6+46LHggAAAB7z5pir6oev9L07j517w4HAACAvWGth3Hec+72TZI8KMmZScQeAADAJrTWwzifOn+/qm6T5LWLGBAAAADX31qvxrnct5I4jw8AAGCTWus5e2/L7OqbSbJfkp9I8vpFDQoAAIDrZ63n7L1o7vY1SS7o7osWMB4AAAD2gjUdxtndH0jymSS3THJgkqsXOSgAAACunzXFXlUdk+TjSR6V5JgkH6uqRy5yYAAAAOy5tR7G+awk9+zuy5OkqrYkeXeSv17UwAAAANhza70a5w8thd7kq7uxLgAAAOtsrXv23l5V70jymun+ryc5fTFDAgAA4PpaNfaq6s5JDu7u36uqX03yc9OsjyY5bdGDAwAAYM/sas/eS5P8fpJ095uSvClJquqnpnm/vMCxAQAAsId2dd7dwd39qeUTp2lbFzIiAAAArrddxd5tVpl30704DgAAAPaiXcXe9qp68vKJVfWkJGcsZkgAAABcX7s6Z+93kry5qh6b78XdtiQ3SvKIBY4LAACA62HV2Ovuy5Lct6oemOQnp8n/u7vfu/CRAQAAsMfW9Hf2uvt9Sd634LEAAACwl6z1j6oDAGx6H7j/L2z0ENiEfuGDH9joIcCG2NUFWgAAANgH2bMHAADr4E+f/raNHgKbzFP++JcXun179gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAa0sNirqldV1eVV9em5abetqndV1eem3wdO06uq/qSqzq+qs6vqZxY1LgAAgBuCRe7Ze3WShyybdkKS93T3EUneM91PkocmOWL6OS7Jyxc4LgAAgOEtLPa6+4NJ/nHZ5KOTnDLdPiXJw+emn9ozf5fkNlV1yKLGBgAAMLr1Pmfv4O6+ZLp9aZKDp9uHJrlwbrmLpmkAAADsgQ27QEt3d5Le3fWq6riq2l5V23fs2LGAkQEAAOz71jv2Lls6PHP6ffk0/eIkh88td9g07Qd098ndva27t23ZsmWhgwUAANhXrXfsvTXJsdPtY5P8zdz0x09X5bx3kivmDvcEAABgN+2/qA1X1WuSPCDJQVV1UZLnJjkpyeur6olJLkhyzLT46UkeluT8JN9O8oRFjQsAAOCGYGGx192P2cmsB62wbCc5flFjAQAAuKHZsAu0AAAAsDhiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYED7b8SDVtUXk1yZ5Nok13T3tqq6bZLXJdma5ItJjunur23E+AAAAPZ1G7ln74HdfVR3b5vun5DkPd19RJL3TPcBAADYA5vpMM6jk5wy3T4lycM3bigAAAD7to2KvU7yzqo6o6qOm6Yd3N2XTLcvTXLwSitW1XFVtb2qtu/YsWM9xgoAALDP2ZBz9pL8XHdfXFU/nORdVfWZ+Znd3VXVK63Y3ScnOTlJtm3btuIyAAAAN3Qbsmevuy+efl+e5M1J7pXksqo6JEmm35dvxNgAAABGsO6xV1U3r6pbLt1O8q+TfDrJW5McOy12bJK/We+xAQAAjGIjDuM8OMmbq2rp8f+qu99eVZ9I8vqqemKSC5IcswFjAwAAGMK6x153fz7J3VaY/tUkD1rv8QAAAIxoM/3pBQAAAPYSsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADCgTRd7VfWQqvpsVZ1fVSds9HgAAAD2RZsq9qpqvyR/luShSe6a5DFVddeNHRUAAMC+Z1PFXpJ7JTm/uz/f3VcneW2Sozd4TAAAAPuczRZ7hya5cO7+RdM0AAAAdkN190aP4TpV9cgkD+nuJ033H5fkZ7v7KXPLHJfkuOnuXZJ8dt0HOq6DknxlowcBK/DeZDPz/mSz8t5kM/P+3Hvu0N1bVpqx/3qPZBcuTnL43P3DpmnX6e6Tk5y8noO6oaiq7d29baPHAct5b7KZeX+yWXlvspl5f66PzXYY5yeSHFFVd6yqGyV5dJK3bvCYAAAA9jmbas9ed19TVU9J8o4k+yV5VXefs8HDAgAA2OdsqthLku4+PcnpGz2OGyiHx7JZeW+ymXl/sll5b7KZeX+ug011gRYAAAD2js12zh4AAAB7gdgbSFVdW1VnVdUnq+rMqrrvNH1rVX1nmnduVZ1aVQdM8w6oqpOq6nPTOh+tqofObfOoquqqesiyx3pVVV1eVZ9e32fJCKrqX0zvx7Oq6tKqunjufk+/P11Vb6iqm03rfLGqPjXN2z63redV1dnT9HdW1e037pmxr5v7HF36OWGavn9V/dfps3Jp3rNWWW9rVT2gqq6Y7n+mql60cc+MEcy9z5Y+Hw9d5bP0Rqu8n9+/7HN0W1W9f8OeGPuk9freWVWHV9X7pm2dU1VPW/9nu+/adOfscb18p7uPSpKqenCSP0ryC9O8f+juo6pqvyTvSnJMktOSPC/JIUl+sruvqqqD59ZJksck+fD0++1z01+d5E+TnLqwZ8OwuvurSY5Kkqo6Mck3u/tF0/1vzr2PT0vym0lePK36wO5e/jd5Xtjd/2Va/reTPGdaB/bEdZ+jyzw/ye2S/FR3/1NV3TLJ01dbr6q2JvlQd/9SVd00yd9X1Zu7+yOLGTo3AN9Z9vn463P3T8zcZ+k0bWfv5yT54ap6aHf/n4WOmJGt1/fOa5I8vbvPnD57z6iqd3X3uYt9emMQe+O6VZKvLZ/Y3ddW1ceTHDrtMXlykjt291XT/MuSvD5JqqqSPCrJv0ryoaq6SXf/07TcB6cvMrBIH0ry06st0N3fmLt78yRORGavmvus3Dr3GXhlkhPXuo3u/k5VnZXk0EWMkRukXX4+7sILkzwridhjb1jY987uviTJJdPyV1bVeZl9loq9NRB7Y7np9GXiJpn9X5NfXL5AVd0kyc8meVqSOyf50rIvy/Pum+QL3f0P0+Ed/ybJGxcwbvgBVbV/kofme/9nr5O8s6o6yf/o7pPnln1BkscnuSLJA9d7rAxl6XN0yR8lOS+zz8or17jeF7r7EfMzq+rAJEck+eBeHCs3UCt8Pu7MD7yfu/t10+2PJnlEVT0wyWrvbdiZdf/eOe1ouHuSj+2dpzA+5+yN5TvdfVR3/3iShyQ5dfq/JElyp+lfyMuSXNLdZ69he49J8trp9mun+7BoS//x2J7kS0leOU3/ue7+mcy+4BxfVfdfWqG7n9Xdh2d2iMhT1nm8jGXpc3Tp53XLF6iqJ0znolxYVYevsN586P18VX0yycVJ3tHdl67Hk2BYO/t83JldvZ+fn+TZCxgnNwzr+r2zqm6RWfz9zirByDJib1Dd/dEkByXZMk36h+m46jsluUdV/UqS85P8SFXdavn60zHWv5bkOVX1xSQvS/KQ6VhpWKT5LydP7e6rk6S7L55+X57kzUnutcK6p2X2voW9aemz8pZJ0t1/OX2eXpFkv12s+6HuvluSI5M8saqOWuRAGd6Kn497qrvfm+SmSe69d4bHDdWiv3dOF3h5Y5LTuvtNC346QxF7g6qqH8/sS8hX56dPF7c4Icnvd/e3M/u/gv+9qm40rbelqh6V5EFJzu7uw7t7a3ffIbN/yb7v0CRYD1V187kP/Jsn+ddJPj3dP2Ju0aOTfGb9R8jI5j4r/3Q6JGnpi8mNdmMbX0hyUpJnLmSQsOeen+QZGz0I9m2L/N457S18ZZLzuvvFYbeIvbHcdDq06Kwkr0tybHdfu8Jyb0lys6r6+cwO39iR5Nya/RmFv03yjcx2nb952XpvnKanql6T2fH+d6mqi6rqiQt4PrDk4CQfng6H+3iS/93dS+eqnFSzy5CfnVkEuiQz18d1n6PTz0nT9GdldoGAT1fV32d2cYxTknx5N7b9iiT3d3Er1tHO3s/X6e7TM/seALtrvb533i/J45L84tx7+WGLeEIjqm4XrgMAABiNPXsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAMKeqDq6qv6qqz1fVGVX10aryN0YB2OeIPQCYTH+89y1JPtjdP9rd90jy6CSHbejAAGAPiD0A+J5fTHJ1d79iaUJ3X9DdL6uq/arqhVX1iao6u6p+I0mq6gFV9f6q+uuq+kxVnTZFY6rqHlX1gWkP4Tuq6pANel4A3ADtv9EDAIBN5MgkZ+5k3hOTXNHd96yqGyf5SFW9c5p392ndLyf5SJL7VdXHkrwsydHdvaOqfj3JC5L8+4U+AwCYiD0A2Imq+rMkP5fk6iQXJPnpqnrkNPvWSY6Y5n28uy+a1jkrydYkX0/yk0neNe3o2y/JJes3egBu6MQeAHzPOUl+belOdx9fVQcl2Z7kS0me2t3vmF+hqh6Q5Kq5Sddm9t/XSnJOd99nwWMGgBU5Zw8Avue9SW5SVf9hbtrNpt/vSPIfquqAJKmqH6uqm6+yrc8m2VJV95mWP6CqjlzEoAFgJfbsAcCku7uqHp7kJVX1jCQ7knwryTOTvCGzwzPPnC7AsiPJw1fZ1tXTIZ9/UlW3zuy/uS/NbO8hACxcdfdGjwEAAIC9zGGcAAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAA/r/+sCQoIjN2bkAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 1080x576 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure(figsize=(15,8))\n",
    "\n",
    "sns.barplot( x = list(gene_count.keys()), y = list(gene_count.values()))\n",
    "plt.title(\"Gene distribution > 100 count\")\n",
    "plt.ylabel(\"Counts\")\n",
    "plt.xlabel(\"Gene\")\n",
    "\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA3sAAAHwCAYAAAAfJXbRAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/MnkTPAAAACXBIWXMAAAsTAAALEwEAmpwYAAAwcUlEQVR4nO3de5xdZX0v/s+XJIgiApWLkJkk4I1LLBFQalVEPFJsvSGIIu3hppxetHj0HC8/j1WP9mhrFfByjoceRagW0FaPqIgiaEEFOeESi1wsRZJMRIhUULkIhOf3x+zomCaZmczs2ZOV9/v12q/Ze61nPeu796zMzCfPs9aq1loAAADolq0GXQAAAADTT9gDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhD4BZp6oWVVWrqrm911+pquOmqe9nV9VNY17fWlX/YTr67vX3/ao6ZLr664eqOr6qvrUJ232yqt7Tj5oAmH7CHsAWrKpeWVXfrap7quqO3vM/raoadG1jtdZe0Fo7a7x2vYD4hHH6uqy19uTpqGt94ae1tm9r7ZvT0T8ATIWwB7CFqqo3Jjk9yfuTPC7Jrkn+OMkzk2w9wNL6Zu1IIQBsCYQ9gC1QVW2f5L8n+dPW2j+01n7eRl3TWju2tfbLXrtHVNXfVNWKqrq9qj5WVY/srTukqkaq6o29UcHbquqEMfvY4LbrqWdOr+1PquqWJH+wzvpvVtWre8+fUFX/VFV399qf11t+aa/5sqr6RVW9YkyNb66qHyc5c+2ydUp4WlVdX1U/raozq2qbXp//brrj2tHDqjo5ybFJ3tTb3xd76381LbT3GZxWVT/qPU6rqkdM5PNbz2d0QlXdUFU/r6pbquo/jVk33vfisVV1flX9rKquTPL4De2n1/5ZVfWdqrqrqlZW1fHrabNjVX2pqlb3PrcvVdXQmPXH9+r8eVX9sKqO3dj3D4DpJ+wBbJmekeQRSb4wTrv3JXlSkiVJnpBkfpK/GLP+cUm27y0/KclHq2rHCW471muSvDDJU5McmOSojdT07iRfS7JjkqEkH06S1trBvfX7tdYe3VpbGyIel+S3kixMcvIG+jw2ye9lNAQ9Kcl/28j+09vfGUk+neSve/t70XqavS3J72T0M9gvydPX6Xtjn9+67sjoZ/SYJCckObWq9p9gXx9Ncn+S3ZKc2HusV1UtTPKVjH6uO/dqv3Y9TbdKcmZGP9cFSe5L8pFeH9sm+VCSF7TWtkvyu2P6WO/3D4DpJ+wBbJl2SvKT1tpDaxeMGcm5r6oO7p23d3KS/9xa+7fW2s+T/I8krxzTz4NJ/ntr7cHW2gVJfpHkyRPcdqyjk5zWWlvZWvu3JO/dSO0PZjRg7N5au7+1Nt6FRh5O8o7W2i9ba/dtoM1Hxuz7L5McM06fE3VsRj+fO1prq5O8K8kfjVm/3s9vfR211r7cWvvX3gjsP2U0MD17vL6qak6SI5P8RWvtntbadUk2dv7jq5J8vbV2Tq+vO1tr166nnjtba//YWru39/39yyTPGdPk4SSLq+qRrbXbWmvfH1PnZL5/AGwiYQ9gy3Rnkp3GnsPWWvvd1toOvXVbZXRU51FJruqFwLuSXNhb/qt+xgbGJPcmefQEtx1r9yQrx7xevpHa35SkklxZo1e+3OAoVc/q1tr947RZd9+7j9N+onbPb76Xdfve0Of371TVC6rqiqr6t97n+fsZDe3j9bVzkrmZ+Oc7nORfN7J+bT2Pqqr/XVXLq+pnSS5NskNVzWmt3ZPkFRk9B/S2qvpyVe3V23Sy3z8ANpGwB7BlujzJL5O8ZCNtfpLRqXn7ttZ26D22b62tN4xMcdvbMhoy1lqwoY5baz9urb2mtbZ7kv+U5H/Wxq/A2SZQ77r7/lHv+T0ZDa1Jkqp63CT7/lFGR7HW1/eE9c7z+8ckf5Nk114ovyCjoWk8q5M8lAl+vhkNhRs9p6/njRkdhTyotfaYJGun0VaStNa+2lp7fkanjt6Y5G97yyf7/QNgEwl7AFug1tpdGZ1S+D+r6qiq2q6qtqqqJUm27bV5OKN/oJ9aVbskSVXNr6rfm0D/k932M0n+vKqGeueZvWVDfVfVy8dcCOSnGQ1cD/de355kz/HqW48/6+37tzJ6nt3a8/2WJdm3qpb0LtryznW2G29/5yT5b1W1c1XtlNFzFj+1CfVtndFzLFcneaiqXpDksIls2Fpbk+RzSd7ZG43bJ8nG7ln46ST/oaqOrqq5vYu7LFlPu+0yGujv6n1u71i7oqp2raqX9M7d+2VGp5Q+3Fu3se8fANNI2APYQrXW/jrJGzI6re723uN/J3lzku/0mr05yc1JruhN1ft6NnBO2XpMZtu/TfLVjIarqzMaTjbkaUm+W1W/SHJ+klNaa7f01r0zyVm9qaNHT7DOJPn7jJ4Dd0tGpzC+J0laaz/I6FVLv57kX5Kse37Zx5Ps09vf/11Pv+9JsjTJ95L8c++9Tfqm5L1z4v48o6H4pxk9r+78SXTx2oxO6fxxkk9m9MIqG9rXioxOEX1jkn/L6IVV9ltP09OSPDKjo7hXZHSa7lpbZfTY+lGvj+ck+ZPeuo19/wCYRtXaRGa3AAAAsDkxsgcAANBBwh4AAEAHCXsAAAAdJOwBAAB0UN/CXlV9oqruqKrrxix7f1XdWFXfq6rPV9UOY9a9tapurqqbJnJZbwAAADasb1fjrKqDM3pfnbNba4t7yw5Lcklr7aGq+qskaa29uXfPn3OSPD3J7hm9xPWTevcG2qCddtqpLVq0qC/1AwAAzHZXXXXVT1prO69v3dx+7bS1dmlVLVpn2dfGvLwiyVG95y9Jcm5r7ZdJflhVN2c0+F2+sX0sWrQoS5cunb6iAQAANiNVtXxD6wZ5zt6JSb7Sez4/ycox60Z6ywAAANgEAwl7VfW2JA8l+fQmbHtyVS2tqqWrV6+e/uIAAAA6YMbDXlUdn+SFSY5tvz5hcFWS4THNhnrL/p3W2hmttQNbawfuvPN6p6YCAABs8fp2zt76VNXhSd6U5DmttXvHrDo/yd9X1QczeoGWJya5ciZrAwAAtkwPPvhgRkZGcv/99w+6lA3aZpttMjQ0lHnz5k14m76Fvao6J8khSXaqqpEk70jy1iSPSHJRVSXJFa21P26tfb+qPpPk+oxO7/yz8a7ECQAAMB1GRkay3XbbZdGiRenllFmltZY777wzIyMj2WOPPSa8XT+vxnnMehZ/fCPt/zLJX/arHgAAgPW5//77Z23QS5KqymMf+9hM9polg7waJwAAwKwwW4PeWptSn7AHAAAwjh//+Md55Stfmcc//vE54IAD8vu///v5wQ9+kMWLFw+6tA2a0Qu0AAAAbG5aazniiCNy3HHH5dxzz02SLFu2LLfffvuAK9s4I3sAAAAb8Y1vfCPz5s3LH//xH/9q2X777Zfh4V/fPe7WW2/Ns5/97Oy///7Zf//9853vfCdJctttt+Xggw/OkiVLsnjx4lx22WVZs2ZNjj/++CxevDhPecpTcuqpp/albiN7AAAAG3HdddflgAMO2GibXXbZJRdddFG22Wab/Mu//EuOOeaYLF26NH//93+f3/u938vb3va2rFmzJvfee2+uvfbarFq1Ktddd12S5K677upL3cIeAADAFD344IN57Wtfm2uvvTZz5szJD37wgyTJ0572tJx44ol58MEH89KXvjRLlizJnnvumVtuuSWve93r8gd/8Ac57LDD+lKTaZwAAAAbse++++aqq67aaJtTTz01u+66a5YtW5alS5fmgQceSJIcfPDBufTSSzN//vwcf/zxOfvss7Pjjjtm2bJlOeSQQ/Kxj30sr371q/tSt7AHAACwEYceemh++ctf5owzzvjVsu9973tZuXLlr17ffffd2W233bLVVlvl7/7u77JmzZokyfLly7PrrrvmNa95TV796lfn6quvzk9+8pM8/PDDOfLII/Oe97wnV199dV/qNo0TAABgI6oqn//85/P6178+f/VXf5VtttkmixYtymmnnfarNn/6p3+aI488MmeffXYOP/zwbLvttkmSb37zm3n/+9+fefPm5dGPfnTOPvvsrFq1KieccEIefvjhJMl73/ve/tTdWutLxzPhwAMPbEuXLh10GQAAwGbshhtuyN577z3oMsa1vjqr6qrW2oHra28aJwAAQAcJewAAAB0k7AEAAHSQsAfAFmfh8IJU1YQfC4cXDLpkAJg0V+MEYIuzYmRllp35xQm33++EF/WxGgDoDyN7AAAAHSTsAQAADNiJJ56YXXbZJYsXL562PoU9AACAMRYtXDipc7vHeyxauHDcfR5//PG58MILp/V9OGcPAABgjOUrVuTum2+ctv62f8Je47Y5+OCDc+utt07bPhMjewAAAJ0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAADdswxx+QZz3hGbrrppgwNDeXjH//4lPt0NU4AAIAxFi5YMKEraE6mv/Gcc84507a/tYQ9AACAMW5dvnzQJUwL0zgBAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAAGLCVK1fmuc99bvbZZ5/su+++Of3006fcp7AHAAAwxqKFC1NV0/ZYtHDhuPucO3duPvCBD+T666/PFVdckY9+9KO5/vrrp/Q+3GcPAABgjOUrVuSOy787bf3t8oyDxm2z2267ZbfddkuSbLfddtl7772zatWq7LPPPpu8XyN7AAAAs8itt96aa665JgcdNH5I3BhhDwAAYJb4xS9+kSOPPDKnnXZaHvOYx0ypL2EPAABgFnjwwQdz5JFH5thjj83LXvayKfcn7AEAAAxYay0nnXRS9t5777zhDW+Ylj6FPQAAgAH79re/nb/7u7/LJZdckiVLlmTJkiW54IILptSnq3ECAACMsXDBggldQXMy/Y3nWc96Vlpr07bPRNgDAAD4DbcuXz7oEqaFaZwAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAADNj999+fpz/96dlvv/2y77775h3veMeU+xT2AAAAxli4YEGqatoeE7nP3iMe8YhccsklWbZsWa699tpceOGFueKKK6b0PtxnDwAAYIwVK1fmlvMvnrb+9nzx88ZtU1V59KMfnSR58MEH8+CDD6aqprRfI3sAAACzwJo1a7JkyZLssssuef7zn5+DDjpoSv0JewAAALPAnDlzcu2112ZkZCRXXnllrrvuuin1J+wBAADMIjvssEOe+9zn5sILL5xSP8IeAADAgK1evTp33XVXkuS+++7LRRddlL322mtKfbpACwAAwIDddtttOe6447JmzZo8/PDDOfroo/PCF75wSn0KewAAAGMsGB6e0BU0J9PfeH77t38711xzzbTtMxH2AAAAfsPyFSsGXcK0cM4eAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAALPAmjVr8tSnPnXK99dbS9gDAAAYY+HwglTVtD0WDi+Y0H5PP/307L333tP2Pvp2n72q+kSSFya5o7W2uLfst5Kcl2RRkluTHN1a+2lVVZLTk/x+knuTHN9au7pftQEAAGzIipGVWXbmF6etv/1OeNG4bUZGRvLlL385b3vb2/LBD35wWvbbz5G9TyY5fJ1lb0lycWvtiUku7r1OkhckeWLvcXKS/9XHugAAAGaV17/+9fnrv/7rbLXV9EW0voW91tqlSf5tncUvSXJW7/lZSV46ZvnZbdQVSXaoqt36VRsAAMBs8aUvfSm77LJLDjjggGntd6bP2du1tXZb7/mPk+zaez4/ycox7UZ6ywAAADrt29/+ds4///wsWrQor3zlK3PJJZfkD//wD6fc78Au0NJaa0naZLerqpOramlVLV29enUfKgMAAJg5733vezMyMpJbb7015557bg499NB86lOfmnK/Mx32bl87PbP39Y7e8lVJhse0G+ot+3daa2e01g5srR24884797VYAACAzVXfrsa5AecnOS7J+3pfvzBm+Wur6twkByW5e8x0TwAAgBmzYGh4QlfQnEx/E3XIIYfkkEMOmZb99vPWC+ckOSTJTlU1kuQdGQ15n6mqk5IsT3J0r/kFGb3tws0ZvfXCCf2qCwAAYGOWr1wx6BKmRd/CXmvtmA2set562rYkf9avWgAAALY0A7tACwAAAP0j7AEAAHSQsAcAANBBwh4AAEAHzfStFwAAAFiPRYsWZbvttsucOXMyd+7cLF26dEr9CXsAAABjLBhekJUjK6etv+Gh4ayY4O0cvvGNb2SnnXaalv0KewAAAGOsHFmZSz7w2Wnr79A3vnza+poM5+wBAADMAlWVww47LAcccEDOOOOMKfdnZA8AAGAW+Na3vpX58+fnjjvuyPOf//zstddeOfjggze5PyN7AAAAs8D8+fOTJLvsskuOOOKIXHnllVPqT9gDAAAYsHvuuSc///nPf/X8a1/7WhYvXjylPk3jBAAAGLDbb789RxxxRJLkoYceyqte9aocfvjhU+pT2AMAABhjeGh4Wq+gOTw0PG6bPffcM8uWLZu2fSbCHgAAwG+Y6D3xZjvn7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAB0kLAHAAAwC9x111056qijstdee2XvvffO5ZdfPqX+hD0AAIAxFgwNp6qm7bFgAvfZS5JTTjklhx9+eG688cYsW7Yse++995Teh/vsAQAAjLFy1UjOe/tZ09bfK9593Lht7r777lx66aX55Cc/mSTZeuuts/XWW09pv0b2AAAABuyHP/xhdt5555xwwgl56lOfmle/+tW55557ptSnsAcAADBgDz30UK6++ur8yZ/8Sa655ppsu+22ed/73jelPoU9AACAARsaGsrQ0FAOOuigJMlRRx2Vq6++ekp9CnsAAAAD9rjHPS7Dw8O56aabkiQXX3xx9tlnnyn16QItAAAAs8CHP/zhHHvssXnggQey55575swzz5xSf8IeAADAGMPzhyZ0Bc3J9DcRS5YsydKlS6dtv8IeAADAGCtGVg66hGnhnD0AAIAOEvYAAAA6SNgDANgMLBhekKqa1GPB8IJBlw0MkHP2AAA2AytHVuaSD3x2Utsc+saX96kaYHNgZA8AAKCDhD0AAIABu+mmm7JkyZJfPR7zmMfktNNOm1KfpnECAACMMTw0nJFVI9PW39D8oawc53YOT37yk3PttdcmSdasWZP58+fniCOOmNJ+hT0AAIAxRlaN5PRTPjJt/Z1y+msn1f7iiy/O4x//+CxcuHBK+zWNEwAAYBY599xzc8wxx0y5H2EPAABglnjggQdy/vnn5+Uvn/rVdIU9AACAWeIrX/lK9t9//+y6665T7kvYAwAAmCXOOeecaZnCmQh7AAAAs8I999yTiy66KC972cumpT9X4wQAABhjaP7QpK+gOV5/E7HtttvmzjvvnLb9CnsAAABjjHdPvM2FaZwAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAADNipp56afffdN4sXL84xxxyT+++/f8p9CnsAAABjDM0fSlVN22O8++ytWrUqH/rQh7J06dJcd911WbNmTc4999wpvw/32QMAABhj1Y9W5a0nvnva+nvvJ94+bpuHHnoo9913X+bNm5d77703u++++5T3a2SPgRseGp7U/4wMDw0PumQAACZowST/1luwBf6tN3/+/PyX//JfsmDBguy2227Zfvvtc9hhh025XyN7DNzIqpGcfspHJtz+lNNf28dqAACYTitXjeS8t5814favePdxfaxmdvrpT3+aL3zhC/nhD3+YHXbYIS9/+cvzqU99Kn/4h384pX6N7AEAAAzQ17/+9eyxxx7ZeeedM2/evLzsZS/Ld77znSn3K+wBAAAM0IIFC3LFFVfk3nvvTWstF198cfbee+8p9yvsQUctWrhw0leKWrRw4aDLBgDY4hx00EE56qijsv/+++cpT3lKHn744Zx88slT7tc5e9BRy1esyN033zipbbZ/wl59qgYAYPMxf/f5E7qC5mT6G8+73vWuvOtd75q2fSbCHgAAwG8YWTUy6BKmhWmcAAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcNJOxV1X+uqu9X1XVVdU5VbVNVe1TVd6vq5qo6r6q2HkRtAAAAXTDjYa+q5if58yQHttYWJ5mT5JVJ/irJqa21JyT5aZKTZro2AACAQTj99NOzePHi7LvvvjnttNOmpc9BTeOcm+SRVTU3yaOS3Jbk0CT/0Ft/VpKXDqY0AABgSzZ//lCqatoe8+cPbXR/1113Xf72b/82V155ZZYtW5YvfelLufnmm6f8Pmb8PnuttVVV9TdJViS5L8nXklyV5K7W2kO9ZiNJxr/zIAAAwDT70Y9W5cSj/3za+vvEZz600fU33HBDDjrooDzqUY9KkjznOc/J5z73ubzpTW+a0n4HMY1zxyQvSbJHkt2TbJvk8Elsf3JVLa2qpatXr+5TlQAAADNj8eLFueyyy3LnnXfm3nvvzQUXXJCVK1dOud8ZH9lL8h+S/LC1tjpJqupzSZ6ZZIeqmtsb3RtKsmp9G7fWzkhyRpIceOCBbWZKBgAA6I+99947b37zm3PYYYdl2223zZIlSzJnzpwp9zuIc/ZWJPmdqnpUVVWS5yW5Psk3khzVa3Ncki8MoDYAAIAZd9JJJ+Wqq67KpZdemh133DFPetKTptznjIe91tp3M3ohlquT/HOvhjOSvDnJG6rq5iSPTfLxma4NAABgEO64444kyYoVK/K5z30ur3rVq6bc5yCmcaa19o4k71hn8S1Jnj6AcgAAAAbqyCOPzJ133pl58+blox/9aHbYYYcp9zmQsAcAADBb7b77/HGvoDnZ/sZz2WWXTdv+1hL2AAAAxli1amTQJUyLQd1UHQAAgD4S9gAAADpI2AMAALZ4rc3uW3hvSn3CHgAAsEXbZpttcuedd87awNday5133pltttlmUtu5QAsAALBFGxoaysjISFavXj3oUjZom222ydDQ0KS2EfYAAIAt2rx587LHHnsMuoxpZxonAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBM2rh8IJU1aQeC4cXDLpsAIDNztxBFwBsWVaMrMyyM784qW32O+FFfaoGAKC7jOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB00obBXVc+cyDIAAABmh4mO7H14gssAAACYBeZubGVVPSPJ7ybZuareMGbVY5LM6WdhAAAAbLqNhr0kWyd5dK/ddmOW/yzJUf0qCgAAgKnZaNhrrf1Tkn+qqk+21pbPUE0AAABM0Xgje2s9oqrOSLJo7DattUP7URQAAABTM9Gw99kkH0vyf5Ks6V85AAAATIeJhr2HWmv/q6+VAAAAMG0meuuFL1bVn1bVblX1W2sffa0MAACATTbRkb3jel//65hlLcme01sOAAAA02FCYa+1tke/CwEAAGD6TCjsVdV/XN/y1trZm7LTqtohoxd7WZzREcITk9yU5LyMXvHz1iRHt9Z+uin9AwAAbOkmes7e08Y8np3knUlePIX9np7kwtbaXkn2S3JDkrckubi19sQkF/deAwAAsAkmOo3zdWNf90bmzt2UHVbV9kkOTnJ8r+8HkjxQVS9Jckiv2VlJvpnkzZuyDwAAgC3dREf21nVPkk09j2+PJKuTnFlV11TV/6mqbZPs2lq7rdfmx0l23cT+AQAAtngTCntV9cWqOr/3+HJGz6/7/Cbuc26S/ZP8r9baUzMaHH9jymZrrWX0XL711XJyVS2tqqWrV6/exBIAgC3FooULU1WTeixauHDQZQNM2URvvfA3Y54/lGR5a21kE/c5kmSktfbd3ut/yGjYu72qdmut3VZVuyW5Y30bt9bOSHJGkhx44IHrDYQAAGstX7Eid99846S22f4Je/WpGoCZM6GRvdbaPyW5Mcl2SXZM8sCm7rC19uMkK6vqyb1Fz0tyfZLz8+v7+R2X5Aubug8AAIAt3URvvXB0kvdn9KIpleTDVfVfW2v/sIn7fV2ST1fV1kluSXJCRoPnZ6rqpCTLkxy9iX0DAABs8SY6jfNtSZ7WWrsjSapq5yRfz+gUzElrrV2b5MD1rHrepvQHAADAb5ro1Ti3Whv0eu6cxLYAAADMsImO7F1YVV9Nck7v9SuSXNCfkgAAAJiqjYa9qnpCRu9/91+r6mVJntVbdXmST/e7OAAAADbNeCN7pyV5a5K01j6X5HNJUlVP6a17UR9rAwAAYBONd97drq21f153YW/Zor5UBAAAwJSNF/Z22Mi6R05jHQAAAEyj8cLe0qp6zboLq+rVSa7qT0kAAABM1Xjn7L0+yeer6tj8OtwdmGTrJEf0sS4AAACmYKNhr7V2e5LfrarnJlncW/zl1tolfa8MAACATTah++y11r6R5Bt9rgUAAIBpMt45ewAAAGyGhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAOqyqJvxYMDQ86HKBaTR30AUAANA/5739rAm3fcW7j+tjJcBMM7IHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAzAwuEFqaoJP+ZuNWfQJQObmbmDLgAAYEu0YmRllp35xQm33++EF/WxGqCLjOwBAAB0kLAHAADQQcLeLLFo4cJJzdtftHDhoEsGAABmMefszRLLV6zIHZd/d8Ltd3nGQX2sBgAA2NwZ2QMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpoYGGvquZU1TVV9aXe6z2q6rtVdXNVnVdVWw+qNgAAgM3dIEf2Tklyw5jXf5Xk1NbaE5L8NMlJA6kKAACgAwYS9qpqKMkfJPk/vdeV5NAk/9BrclaSlw6iNgAAgC4Y1MjeaUnelOTh3uvHJrmrtfZQ7/VIkvkDqAsAAKATZjzsVdULk9zRWrtqE7c/uaqWVtXS1atXT3N1AAAA3TCIkb1nJnlxVd2a5NyMTt88PckOVTW312Yoyar1bdxaO6O1dmBr7cCdd955JuoFAADY7Mx42GutvbW1NtRaW5TklUkuaa0dm+QbSY7qNTsuyRdmujYAAICumE332XtzkjdU1c0ZPYfv4wOuBwAAYLM1d/wm/dNa+2aSb/ae35Lk6YOsBwAAoCtm08geAAAA00TYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoINmPOxV1XBVfaOqrq+q71fVKb3lv1VVF1XVv/S+7jjTtQEAAHTFIEb2HkryxtbaPkl+J8mfVdU+Sd6S5OLW2hOTXNx7DQAAwCaY8bDXWruttXZ17/nPk9yQZH6SlyQ5q9fsrCQvnenaAAAAumKg5+xV1aIkT03y3SS7ttZu6636cZJdB1UXAADA5m5gYa+qHp3kH5O8vrX2s7HrWmstSdvAdidX1dKqWrp69eoZqBQAAEiSBcMLUlWTeszZas6gy95izR3ETqtqXkaD3qdba5/rLb69qnZrrd1WVbsluWN927bWzkhyRpIceOCB6w2EAADA9Fs5sjKXfOCzk9rm0De+vE/VMJ5BXI2zknw8yQ2ttQ+OWXV+kuN6z49L8oWZrg0AAKArBjGy98wkf5Tkn6vq2t6y/y/J+5J8pqpOSrI8ydEDqA0AAKATZjzstda+laQ2sPp5M1kLAABAVw30apwAAAD0h7AHAADQQcIeAAD02aKFCyd1u4JFCxcOumQ6YCC3XgAAgC3J8hUrcvfNN064/fZP2KuP1bClMLIHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AwBQtXLAgVTWpx9w5cwZdNtBxcwddAADA5m7FypW55fyLJ7XNni9+Xp+qARhlZA8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9rYgC4cXpKom9Vg4vGDQZcOkLRganvSxvmBoeNBls4kWLpj8z7a5c+YMumwA6Lu5gy6AmbNiZGWWnfnFSW2z3wkv6lM10D8rV43kvLefNaltXvHu4/pUDf22YuXK3HL+xZPaZs8XP69P1QDA7GFkDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD3GNanL17tVw2ZvMt/vRQsXDrpcZrlFCxdO+rYIjiuAmTPZW3PN3cqtazYnbr3AuC75wGcn3PbQN768j5UwE+64/LsTbrvLMw7qYyV0wfIVK3L3zTdOapvtn7BXn6oBYF2TvTWX23JtXozsAQAAdJCwBwAA0EGmcQJTUlWTaj93zqbN9Z/MfuY4n4A+mOyxPjw0nBUrV/SpGgAYn7AHTMkt5188qfZ7vvh5m7Qf544yaJM5BhPHIQCDZxonAABAB826sFdVh1fVTVV1c1W9ZdD1bIpNudT4vLmTH2Sd7D42dfpcv+uaU5M/DCe7j6H5Q314pzA58+cPTeq4ne+43eIMDw1P6hgZHhoedMkAzGKzahpnVc1J8tEkz08ykuT/VdX5rbXrB1vZ5MzUpcZnavrcZJ339rMm1f4V7z5u0vt464nvnlT7937i7ZPeB0y3H/1oVU48+s8n3P4Tn/lQH6thNhpZNZLTT/nIhNufcvpr+1gNAJu72Tay9/QkN7fWbmmtPZDk3CQvGXBNAAAAm53ZFvbmJ1k55vVIbxkAAACTUK21QdfwK1V1VJLDW2uv7r3+oyQHtdZeO6bNyUlO7r18cpKbZrxQkmSnJD8ZdBFs8RyHzAaOQ2YDxyGD5hgcnIWttZ3Xt2JWnbOXZFWSsWebD/WW/Upr7YwkZ8xkUfx7VbW0tXbgoOtgy+Y4ZDZwHDIbOA4ZNMfg7DTbpnH+vyRPrKo9qmrrJK9Mcv6AawIAANjszKqRvdbaQ1X12iRfTTInySdaa98fcFkAAACbnVkV9pKktXZBkgsGXQfjMpWW2cBxyGzgOGQ2cBwyaI7BWWhWXaAFAACA6THbztkDAABgGgh7JEmqak1VXTvm8Zbe8rlV9T+q6l/GrHvbRrZbVFWHVNXdvdc3VtXfDO6dsbmoqseOOY5+XFWrxrxuva/XVdVnq+pRvW1urap/7q1bOqavd1fV93rLv1ZVuw/unbG5GfNzbe3xNn8jx+bWG/n5+c11jssDq+qbA3tjzGpjjqNlVXV1Vf1ub/miqrqvt+76qjq7qub11s2rqvf1fkdfXVWXV9ULxvS5pPfz8/B19vWJqrqjqq6b2XfJbDdTx2FVDVfVN3p9fb+qTpn5d7tlmHXn7DEw97XWlqxn+XuSPC7JU1pr91fVdkneuLHtqmpRkstaay+sqkcmuaaqPt9a+3Z/SqcLWmt3JlmSJFX1ziS/aK39Te/1L9YeZ1X16SR/nOSDvU2f21pb974+72+tvb3X/s+T/EVvG5iI+9Y53l4x5vU7M+bY7C3b0M/PJNmlql7QWvtKXyumC8Yed7+X5L1JntNb96+ttSVVNSfJRUmOTvLpJO9OsluSxa21X1bVrmO2SZJjknyr9/XCMcs/meQjSc7u27thczVTx+FDSd7YWru697flVVV1UWvt+v6+vS2PsMcG9UZPXpNkUWvt/iRprf08yTsn2kdr7b6qujbJ/H7UyBbpsiS/vbEGrbWfjXm5bRInJ7Opxj3exvH+JG9LIuwxGY9J8tN1F7bW1lTVlUnmj/kdvUdr7Ze99bcn+UySVFUleXmS5ye5rKq2GfO7/NLef8zCxvTtOGyt3Zbktl77n1fVDRn9W1HYm2bCHms9shfK1npvkhuSrOgFvIls98PW2hFjV1bVjkmemOTSaayVLVRVzU3ygvz6fwZbkq9VVUvyv1trZ4xp+5dJ/mOSu5M8d6ZrZfO3nuNtQ/7dz8/W2nm955cnOaKqnptkYz9LYe1xtE1GR0kOXbdBVW2T5KAkpyR5QkZ/R/9s3XY9v5vR38v/2ps+/AdJ/rEPddMtM34c9v7j4alJvjs9b4GxnLPHWve11paMeZy3boOqOqE3V3tlVQ2vZ7uxQe/ZVbUsyaokX22t/Xgm3gSdtfaXz9IkK5J8vLf8Wa21/TP6B/mfVdXBazdorb2ttTac0Skmr53hetm8beh425Dxfn6+J8l/60OddMva42ivJIcnObs3KpIkj+8dk7cnua219r0J9HdMknN7z8/tvYbxzOhxWFWPzmj4e/1GAiNTIOyxMTcnWdCbS53W2pm9edx3Z/Sm9xtzWWttvyT7Jjmpqpb0s1A6b+wf069rrT2QJK21Vb2vdyT5fJKnr2fbTyc5cuZKpQPWe7xtqtbaJUkemeR3pqc8uq61dnmSnZLs3Fv0r73fv49PckBVvTi//h39mHW3751TdWSSv6iqW5N8OMnha3+fw0T0+zjsXeDlH5N8urX2uT6/nS2WsMcGtdbuzej/aH+kN2S/9h/u1pPo44dJ3pfkzX0pki1WVW075hfGtkkOS3Jd7/UTxzR9SZIbZ75C+A3vSfKmQRfB5qGq9srof6reOXZ572JUb0ny1jG/o0+vqq172+1cVS9P8rwk32utDbfWFrXWFmb0j+rfONUCNqafx2FvtPDjSW5orX0w9I2wx1qPrN+8dPj7esvfltETaK+rqmsyerGCs5L8aBJ9fyzJwU4GZ5rtmuRbvenCVyb5cmtt7blV76vRy+Z/L6Mh0CWd6acN/fz8ldbaBUlWD6A2Nh+/Oo6SnJfkuNbamvW0+79JHlVVz87o9ODVSa6v0dsofCnJzzI6Ve7z62z3j73lqapzMno+6ZOraqSqTurD+2HzNFPH4TOT/FGSQ8f87Pz9fryhLV215iJ1AAAAXWNkDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AGKOqdq2qv6+qW6rqqqq6vKrcnwyAzY6wBwA9vRv9/t8kl7bW9mytHZDklUmGBloYAGwCYQ8Afu3QJA+01j62dkFrbXlr7cNVNaeq3l9V/6+qvldV/ylJquqQqvpmVf1DVd1YVZ/uhcZU1QFV9U+9EcKvVtVuA3pfAGyB5g66AACYRfZNcvUG1p2U5O7W2tOq6hFJvl1VX+ute2pv2x8l+XaSZ1bVd5N8OMlLWmurq+oVSf4yyYl9fQcA0CPsAcAGVNVHkzwryQNJlif57ao6qrd6+yRP7K27srU20tvm2iSLktyVZHGSi3oDfXOS3DZz1QOwpRP2AODXvp/kyLUvWmt/VlU7JVmaZEWS17XWvjp2g6o6JMkvxyxak9Hfr5Xk+621Z/S5ZgBYL+fsAcCvXZJkm6r6kzHLHtX7+tUkf1JV85Kkqp5UVdtupK+bkuxcVc/otZ9XVfv2o2gAWB8jewDQ01prVfXSJKdW1ZuSrE5yT5I3J/lsRqdnXt27AMvqJC/dSF8P9KZ8fqiqts/o79zTMjp6CAB9V621QdcAAADANDONEwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADooP8fbaQ8hx3YeEAAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 1080x576 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure(figsize=(15,8))\n",
    "sns.histplot(data=train_variant[train_variant[\"Gene\"].isin(gene_count.keys())],\n",
    "             x=\"Gene\", hue=\"Class\", multiple=\"dodge\", shrink=.7)\n",
    "plt.title(\"Gene distribution and class\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene             Variation  Class\n",
       "ID                                     \n",
       "0   FAM58A  Truncating Mutations      1\n",
       "1      CBL                 W802*      2\n",
       "2      CBL                 Q249E      2\n",
       "3      CBL                 N454D      3\n",
       "4      CBL                 L399V      4"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_variant.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Gene, Variation, Class]\n",
       "Index: []"
      ]
     },
     "execution_count": 200,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_variant[train_variant[\"Gene\"] == \"RAD51\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "amino_acid_dict = {'C' : 'CYS', 'D' : 'ASP', 'S' : 'SER', 'Q' : 'GLN', 'K' : 'LYS',\n",
    "    'I' : 'ILE', 'P' : 'PRO', 'T' : 'THR', 'F' : 'PHE', 'N' : 'ASN', \n",
    "    'G' : 'GLY', 'H' : 'HIS', 'L' : 'LEU', 'R' : 'ARG', 'W' : 'TRP', \n",
    "    'A' : 'ALA', 'V' : 'VAL', 'E' : 'GLU', 'Y' : 'TYR', 'M' : 'MET'}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Fonction pour la selection\n",
    "\n",
    "## clean_text\n",
    "Nettoie l'article, lower puis enlve les points qui pourrait nuire  la detection des phrases, (comme les \"...\", ou \"fig.A\")\n",
    "\n",
    "\n",
    "## find_match\n",
    "Va dtecter la phrase qui contient le mot, puis prend 1 phrase avant et aprs (au total 3 phrase) (voir diapo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def amino_three(amino):\n",
    "    return amino_acid_dict[amino]\n",
    "    \n",
    "def decompose_variation(variation):\n",
    "    decompose_aa_pos_aa = re.compile(\"([a-z]{1,})(\\d+)([a-z]{1,})\")\n",
    "    list_variation = decompose_aa_pos_aa.search(variation)\n",
    "    if list_variation:\n",
    "        aa1 = list_variation.group(1)\n",
    "        aa2 = list_variation.group(3)\n",
    "\n",
    "        if len(aa1) + len(aa2) == 2:\n",
    "            amino1 = amino_acid_dict[aa1.upper()].lower()\n",
    "            position = list_variation.group(2)\n",
    "            amino2 = amino_acid_dict[aa2.upper()].lower()\n",
    "            return [amino1,position,amino2]\n",
    "    return False\n",
    "\n",
    "def decompose_fusion(variation):\n",
    "    decompose_g1_g2_fusion = re.compile(\"(\\w+)\\s?(\\?|-)\\s?(\\w+)\\? fusion\")\n",
    "    list_variation = decompose_g1_g2_fusion.search(variation)\n",
    "    if list_variation:\n",
    "        gene1 = list_variation.group(1)\n",
    "        gene2 = list_variation.group(3)\n",
    "        print(gene1,gene2,variation, list_variation.group(0))\n",
    "        return \"(\" + gene1 + \"|\" + gene2 + \")\"\n",
    "    return False\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 212,
   "metadata": {},
   "outputs": [],
   "source": [
    "def clean_text(article):\n",
    "    dot3 = re.compile(\"[\\.]{2,}\")\n",
    "    fig = re.compile(\"fig[s]?\\.\")\n",
    "    decimal = re.compile(\"\\d+\\.\\d+\")    \n",
    "    etal = re.compile(\"et al\\.\")\n",
    "    ie = re.compile(\"i\\.e\\.\")\n",
    "    inc = re.compile(\"inc\\.\")\n",
    "    mutation_point = re.compile(\"[p|c]\\.\")\n",
    "    \n",
    "    clean_article = article.lower()\n",
    "    clean_article = dot3.sub(\".\", clean_article)\n",
    "    clean_article = fig.sub(\"\", clean_article)\n",
    "    clean_article = decimal.sub(\"\", clean_article)\n",
    "    clean_article = etal.sub(\"\", clean_article)\n",
    "    clean_article = ie.sub(\"\", clean_article)    \n",
    "    clean_article = inc.sub(\"\", clean_article)    \n",
    "    clean_article = mutation_point.sub(\"\", clean_article)    \n",
    "    \n",
    "    \n",
    "    return clean_article\n",
    "\n",
    "def join_tuple_string(strings_tuple):\n",
    "    return ' '.join(strings_tuple)\n",
    "\n",
    "def find_match(text, word):\n",
    "    clean = clean_text(text)\n",
    "    word = word.lower()\n",
    "    target_sentence = \"([^.]*{}[^.]*\\.)\".format(word)\n",
    "    before_after_target = \"([^.]*\\.){0,1}\"    \n",
    "    match_exp = re.compile(before_after_target + target_sentence + before_after_target)\n",
    "    match_text = match_exp.findall(clean)\n",
    "    final_match = \"\".join(list(map(join_tuple_string, match_text)))\n",
    "    return final_match\n",
    "\n",
    "def extract_match(line):\n",
    "    # Mettre celui qui consomme le moins en 1er --> ceux qui sont facile  faire sans trop de calcul !\n",
    "    # 1er : best match\n",
    "    # 2e ?\n",
    "    \n",
    "    \n",
    "    # Cleaning text     \n",
    "    text = clean_text(line[\"Text\"])\n",
    "    variation = line[\"Variation\"].lower()\n",
    "    gene = line[\"Gene\"].lower()\n",
    "    print(line.name)\n",
    "    # Try first match with inital variation value\n",
    "    # Quality score = 1 \n",
    "    initial_match = find_match(text, variation)    \n",
    "    if len(initial_match) != 0:\n",
    "        print(\"First match ! \", variation)\n",
    "        return initial_match, 1\n",
    "    \n",
    "\n",
    "    # Fusion of two genes\n",
    "    # Quality score = 2\n",
    "    if \"fusion\" in variation:\n",
    "        fusion_gene = decompose_fusion(variation)\n",
    "        if fusion_gene:\n",
    "            match_fusion_gene = find_match(text, fusion_gene)\n",
    "            if len(match_fusion_gene) != 0:\n",
    "                print(\"Fusion gene1 | gene2\", variation)\n",
    "                return match_fusion_gene, 2\n",
    "\n",
    "        # Try to match the word fusion at least..\n",
    "        # Quality score 4 (bad)\n",
    "        match_fusion = find_match(text,\"fusion\")\n",
    "        if len(match_fusion) != 0:\n",
    "            print(\"FUSION\", variation)\n",
    "            return match_fusion, 4\n",
    "        \n",
    "    aa_pos_aa = decompose_variation(variation)\n",
    "    if aa_pos_aa :\n",
    "        if len(aa_pos_aa) == 3:\n",
    "            # If we success to split variation in 3 group --> aa1 pos aa2\n",
    "            # Second try without the last amino acid --> y371\n",
    "            match_variation_aa_pos = find_match(text, variation[:-1])\n",
    "            if len(match_variation_aa_pos) != 0:\n",
    "                print(\"y371 aa_pos\", variation)\n",
    "                \n",
    "                return match_variation_aa_pos, 2\n",
    "\n",
    "            # Third try with 3 letter code of amino acid --> tyr371ser\n",
    "            match_aa_pos_aa = find_match(text, \"\".join(aa_pos_aa))\n",
    "            if len(match_aa_pos_aa) != 0:\n",
    "                print(\"aa_pos_aa\", variation)\n",
    "                return match_aa_pos_aa, 1.1\n",
    "            \n",
    "            # Try with 3 letter code without the last aa --> tyr371\n",
    "            match_aa_pos = find_match(text, aa_pos_aa[0] + aa_pos_aa[1])\n",
    "            if len(match_aa_pos) != 0:\n",
    "                print(\"aa_pos\", variation)\n",
    "                \n",
    "                return match_aa_pos,2\n",
    "            \n",
    "            # Match position only --> 371\n",
    "            match_pos = find_match(text, aa_pos_aa[1])\n",
    "            if len(match_pos) != 0:\n",
    "                print(\"pos\", variation)\n",
    "                \n",
    "                return match_pos,4\n",
    "            \n",
    "            # Match position around the real position --> 370 - 379\n",
    "            match_pos_weak = find_match(text, aa_pos_aa[1][:-1] + \"[0-9]\")\n",
    "            if len(match_pos_weak) != 0:\n",
    "                print(\"pos weak\", variation)\n",
    "                \n",
    "                return match_pos_weak,5\n",
    "            \n",
    "            \n",
    "            \n",
    "    return [],0\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exemple  la main juste pour tester"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "word_to_search = \"FAM58A\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "article = train_text.loc[0].values[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "34"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article.count(word_to_search)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "clean_match = find_match(article, word_to_search)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Before selection : total lenght 39672 and 34 occurences of the word FAM58A\n",
      "After selection :  total lenght 9214 and 34 occurences of the word FAM58A\n"
     ]
    }
   ],
   "source": [
    "print(\"Before selection : total lenght {} and {} occurences of the word {}\".format(\n",
    "    len(article),article.count(word_to_search) , word_to_search))\n",
    "\n",
    "print(\"After selection :  total lenght {} and {} occurences of the word {}\".format(\n",
    "    len(clean_match),clean_match.count(word_to_search.lower()), word_to_search))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>D171N</td>\n",
       "      <td>4</td>\n",
       "      <td>Introduction  Myelodysplastic syndromes (MDS) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>A122*</td>\n",
       "      <td>1</td>\n",
       "      <td>Introduction  Myelodysplastic syndromes (MDS) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>Fusions</td>\n",
       "      <td>1</td>\n",
       "      <td>The Runt-related transcription factor 1 gene (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>R80C</td>\n",
       "      <td>4</td>\n",
       "      <td>The RUNX1/AML1 gene is the most frequent targe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>K83E</td>\n",
       "      <td>4</td>\n",
       "      <td>The most frequent mutations associated with le...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3316 rows  4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene             Variation  Class  \\\n",
       "ID                                          \n",
       "0     FAM58A  Truncating Mutations      1   \n",
       "1        CBL                 W802*      2   \n",
       "2        CBL                 Q249E      2   \n",
       "3        CBL                 N454D      3   \n",
       "4        CBL                 L399V      4   \n",
       "...      ...                   ...    ...   \n",
       "3316   RUNX1                 D171N      4   \n",
       "3317   RUNX1                 A122*      1   \n",
       "3318   RUNX1               Fusions      1   \n",
       "3319   RUNX1                  R80C      4   \n",
       "3320   RUNX1                  K83E      4   \n",
       "\n",
       "                                                   Text  \n",
       "ID                                                       \n",
       "0     Cyclin-dependent kinases (CDKs) regulate a var...  \n",
       "1      Abstract Background  Non-small cell lung canc...  \n",
       "2      Abstract Background  Non-small cell lung canc...  \n",
       "3     Recent evidence has demonstrated that acquired...  \n",
       "4     Oncogenic mutations in the monomeric Casitas B...  \n",
       "...                                                 ...  \n",
       "3316  Introduction  Myelodysplastic syndromes (MDS) ...  \n",
       "3317  Introduction  Myelodysplastic syndromes (MDS) ...  \n",
       "3318  The Runt-related transcription factor 1 gene (...  \n",
       "3319  The RUNX1/AML1 gene is the most frequent targe...  \n",
       "3320  The most frequent mutations associated with le...  \n",
       "\n",
       "[3316 rows x 4 columns]"
      ]
     },
     "execution_count": 235,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concatenate_data = pd.merge(train_variant, train_text, on=\"ID\").dropna()\n",
    "concatenate_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {},
   "outputs": [],
   "source": [
    "concatenate_data[\"Text\"] = concatenate_data.apply(lambda line: clean_text(line[\"Text\"]), axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [],
   "source": [
    "concatenate_data[\"Variation\"] = train_variant[\"Variation\"].apply(lambda line: line.lower())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>cyclin-dependent kinases (cdks) regulate a var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>w802*</td>\n",
       "      <td>2</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>q249e</td>\n",
       "      <td>2</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>n454d</td>\n",
       "      <td>3</td>\n",
       "      <td>recent evidence has demonstrated that acquired...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>l399v</td>\n",
       "      <td>4</td>\n",
       "      <td>oncogenic mutations in the monomeric casitas b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>d171n</td>\n",
       "      <td>4</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>a122*</td>\n",
       "      <td>1</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>fusions</td>\n",
       "      <td>1</td>\n",
       "      <td>the runt-related transcription factor 1 gene (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>r80c</td>\n",
       "      <td>4</td>\n",
       "      <td>the runx1/aml1 gene is the most frequent targe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>k83e</td>\n",
       "      <td>4</td>\n",
       "      <td>the most frequent mutations associated with le...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3316 rows  4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene             Variation  Class  \\\n",
       "ID                                          \n",
       "0     FAM58A  truncating mutations      1   \n",
       "1        CBL                 w802*      2   \n",
       "2        CBL                 q249e      2   \n",
       "3        CBL                 n454d      3   \n",
       "4        CBL                 l399v      4   \n",
       "...      ...                   ...    ...   \n",
       "3316   RUNX1                 d171n      4   \n",
       "3317   RUNX1                 a122*      1   \n",
       "3318   RUNX1               fusions      1   \n",
       "3319   RUNX1                  r80c      4   \n",
       "3320   RUNX1                  k83e      4   \n",
       "\n",
       "                                                   Text  \n",
       "ID                                                       \n",
       "0     cyclin-dependent kinases (cdks) regulate a var...  \n",
       "1      abstract background  non-small cell lung canc...  \n",
       "2      abstract background  non-small cell lung canc...  \n",
       "3     recent evidence has demonstrated that acquired...  \n",
       "4     oncogenic mutations in the monomeric casitas b...  \n",
       "...                                                 ...  \n",
       "3316  introduction  myelodysplastic syndromes (mds) ...  \n",
       "3317  introduction  myelodysplastic syndromes (mds) ...  \n",
       "3318  the runt-related transcription factor 1 gene (...  \n",
       "3319  the runx1/aml1 gene is the most frequent targe...  \n",
       "3320  the most frequent mutations associated with le...  \n",
       "\n",
       "[3316 rows x 4 columns]"
      ]
     },
     "execution_count": 240,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concatenate_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Explications du problme\n",
    "Fonction poubelle juste pour voir ceux qui sont pas dtectable --> len(match) == 0\n",
    "\n",
    "Si ils sont pas dtectable, c'est que le nom donn pour leurs variation sont pas dans l'article, donc je vrfie  la main sur le site https://regex101.com/r/lvQZIf/1 qui sert  visualiser les expressions rgulires.\n",
    "En copiant collant tout l'article, je regarde comment je peux dtecter la variation avec un autre nom ducoup, comme le nom de base ne marche pas.\n",
    "\n",
    "Par exemple Id = 19, y371s n'est pas dans l'article mais il y est sous la forme tyr371ser.\n",
    "\n",
    "Cet regexpr ne marche pas (le texte est en minuscule) -> `([^.]*\\.){0,1}([^.]*y371s[^.]*\\.)([^.]*\\.){0,2}`\n",
    "\n",
    "Mais celle la marchera -> `([^.]*\\.){0,1}([^.]*tyr371ser[^.]*\\.)([^.]*\\.){0,2}` (a donne 3 match sur le site pour l'article id = 19)\n",
    "\n",
    "(pour les regex, il faut juste changer le nom au milieu aprs `*` et avant `[^.]`)\n",
    "\n",
    "Autre exemple, ID = 102, mutation r922* qui n'est mentionn sous aucunes formes, ni mme la position 922. Mais, il y a des mots qui peuvent faire penser  cette mutation du genre \"nonsense mutations\" car l'toile (\\*) veut dire que r922 est devenu un codon stop, donc c'est une mutation non sens. Faudrait voir si je peux convertir ces mutations en mot pour faciliter la dtection, ou utiliser un champ lexical.\n",
    "\n",
    "\n",
    "Donc l, j'essaie de faire plus ou moins au cas par cas (pas tous, y en a 880...) pour dterminer des groupes de variations indtectable pour revoir la dtection.\n",
    "Par exemple, pour les variations de type `y371s`, je peux prendre `y371`, ou tout simplement `371` qui peuvent dtecter la variation dans certain cas.\n",
    "\n",
    "\n",
    "Par contre, pour ceux o y a vraiment pas d'infos dans l'article avec la variation, du genre `t1365m` ID = 44; je sais pas quoi faire d'eux :)\n",
    "Car la position 1365 n'est mme pas mentionn dans l'article, donc je vois pas comment dtecter les infos lies  cette mutation...\n",
    "\n",
    "Soit je garde l'article entier, soit je prend le nom du gne pour la dtection, mais le pb avec a, c'est que ducoup tout ceux qui sont indtectables de ce gne l, auront les mmes lignes de dtect car je dtecterai uniquement en utilisant le nom du gne, donc y aura pas de diffrences entre eux mme si la variation diffre.\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "## Ne lance pas la cellule juste en bas, a prend un certain temps et de toute faon, tu as dj la sortie avec tout ceux qui ne marchent pas.\n",
    "\n",
    "La sortie est de la forme : \n",
    "`variation | id | len(article_original) | len(match)`\n",
    "\n",
    "Certaines variations sont des mots complets du genre `truncating mutations`, un peu relou pour la dtection, ce qu'on pourrait faire, c'est utiliser des mots qui ressemble  truncating grace  des mthodes de traitements de texte qui va prendre que le prfix par exemple et automatiser a pour les mots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def check_line(x):\n",
    "    extract = extract_match(x)\n",
    "    score = extract[1]\n",
    "    if score == 0:\n",
    "        print(x[\"Variation\"].lower(), x.name, len(x[\"Text\"]))\n",
    "\n",
    "        \n",
    "#concatenate_data.apply(lambda x : check_line(x) , axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Variable ID pour faciliter le print des deux cellules suivantes pour choper le text, la variation et le gene d'un ID et les copier coller sur le site regex\n",
    "\n",
    "Pour la dtection, je teste juste avec le nom de la variation, puis vraiment si y a rien, je regarde du cot du nom du gne"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "ID = 28    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\"sequencing studies have identified many recurrent coding mutations in human cancer genes; however, highly recurrent mutations involving regulatory regions have rarely been observed. here we describe two independent mutations within the tert core promoter that, when taken together, were observed in 71% (50 of 70) of melanomas and generate de novo consensus ets binding motifs. reporter assays showed that these mutations increase transcriptional activity from the tert promoter by 24-fold. examination of cancer cell lines derived from diverse tumor types revealed the same mutations in 16% (24 of 150) of cases, with preliminary evidence of elevated frequency in bladder and hepatocellular cancer cells. thus, somatic mutations in regulatory regions of the genome may represent an important tumorigenic mechanism.  systematic characterization of human cancer genomes has led to the discovery of a wide range of mutated genes that contribute to tumor development and progression. most of the somatic mutations in tumors reside within the protein-coding regions of genes or at splice junctions. to determine whether tumor genomes harbor recurrent mutations outside of protein-coding regions, we systematically queried noncoding somatic mutations using published whole-genome sequencing data.  analysis of whole-genome sequencing data from malignant melanomas (1, 2) revealed two somatic telomerase reverse transcriptase (tert) gene promoter mutations in 17 of 19 (89%) cases examined. the average sequence coverage at the tert promoter locus was 30-fold in normal samples and 60-fold in tumor samples ( s1a). each of these promoter mutations resulted in a cytidine-to-thymidine transition at a dipyrimidine motif indicative of ultraviolet (uv) lightinduced damage (chr5, 1,295,228 c>t and 1,295,250 c>t; hereafter termed c228t and c250t, respectively), and both mutations localized within 100 base pairs (bp) of the tert transcriptional start site (tss) (mean allelic fraction, ; range,  to ) (table s1). we validated these mutations by means of polymerase chain reaction and sanger sequencing tumor/normal sample pairs from both the discovery set ( 1a and  s1, b and c) and an extension set of 51 additional melanoma tumor/normal sample pairs. within this extension set, 33 tumors (65%) harbored one of the mutations. moreover, the mutations were mutually exclusive in both the discovery and extension sets (p =   107, fishers one-sided exact test). two tumors with a c228t transition also contained an adjacent c>t transition (at position chr5, 1,295,229), which is indicative of a dinucleotide cc>tt transition. together, these tert promoter mutations were observed in 50 of 70 (71%; 95% confidence interval: 59 to 82%, clopper-pearson method) melanomas examined ( 1b and table s1).   1  1 identification of tert promoter mutations in melanoma and cancer cell lines both c228t and c250t generated an identical 11-bp nucleotide stretch (5-ccccttccggg-3) containing a consensus binding site for e-twenty-six (ets) transcription factors (ggaa, reverse complement) within the tert promoter region. because ets transcription factors may become activated through dysregulation of mitogen-activated protein kinase (map kinase) signaling, we hypothesized that these promoter mutations might augment gene expression. to test this hypothesis, we used a reporter assay system in which the relevant portion of the mutant or wild-type tert core promoter was cloned upstream of the firefly luciferase gene (2). here, we tested both a core promoter fragment (132 to +5 relative to the tss) and the full core promoter (200 to +73). in comparison to the wild-type tert promoter, both mutations conferred approximately two- to fourfold increased transcriptional activity in five distinct cell line contexts ( 1c and  s1d). thus, each mutation was capable of augmenting transcriptional activity from the tert promoter.  to investigate whether similar tert promoter mutations occur in other cancer types, we examined sequencing data from this locus in 150 cell lines from the cancer cell line encyclopedia (ccle) (3). overall, 24 ccle lines (16%) contained either c228t or c250t (mean allelic fraction, ; range,  to ) (table s1). an increased frequency in melanoma was again noted (five of six lines tested), with additional evidence suggesting possible heightened prevalence (>25%; one-sided 95% confidence interval) in bladder (three of three lines) and hepatocellular cancer cell lines (four of six lines) ( 1d).  several lines of evidence support the hypothesis that these promoter mutations may function as driver events that contribute to oncogenesis through tert dysregulation and undergo positive selection, at least in human melanoma. first, the tert promoter mutations showed a combined frequency that exceeded those of braf and nras mutations, which activate known melanoma driver oncogenes (4, 5). in an analysis restricted to somatic mutations present at an allelic fraction of  or greater [to reduce artifacts of mutation calling (1)], the four most recurrent melanoma nucleotide substitutions included braf [chr7, 140,453,136 a>t (v600e)], nras [chr1, 115,256,529 t>c (q61r)], and the tert core promoter mutations c228t and c250t. second, although highly recurrent, c228t and c250t occurred in a wholly mutually exclusive fashion. this suggests the possibility that the mutations might be functionally redundant. third, the absence of other recurrent somatic mutations in the 3 kb upstream of the tert transcription start site in the queried melanomas (1) coupled with the absence of the described tert promoter mutations in 24 lung adenocarcinomas with comparably high somatic mutation rates (6) reduces the possibility that these recurrent tert promoter mutations are solely due to an increased background mutation rate at this locus. although the role of telomerase in tumorigenesis is well established, details regarding its dysregulation in cancer cells remain incompletely understood, particularly in melanoma (7). the tert promoter mutations identified here may link telomerase gene regulation and tumorigenic activation in this malignancy. the high prevalence of c228t and c250t suggests that these tert promoter mutations may comprise early genetic events in the genesis of melanoma and other cancer types. although tert expression alone is not sufficient to bypass oncogene-induced senescence, genomic tert activation may potentiate mechanisms by which melanocytes achieve immortalization in the setting of oncogenic mutations (8). these results therefore suggest that renewed efforts to develop clinically effective telomerase inhibitors may be warranted.  at the same time, promoter mutations likely represent only one potential mechanism of tert reactivation in a subset of human cancers. indeed, recurrent chromosomal copy gains spanning the tert locus have been described previously for several cancers, including melanoma (9, 10).  highly recurrent somatic mutations within a cancer gene promoter region have not previously been described. similarly, the de novo mutational generation of transcription factor binding motifs in tumor genomes was heretofore unknown, although an ets transcription factor binding motif was previously associated with a single-nucleotide polymorphism insertion at the mmp-1 locus (11). together, these findings raise the possibility that recurrent somatic mutations involving regulatory regions, in addition to coding sequences, may represent important driver events in cancer.  tert promoter mutations in familial and sporadic melanoma susanne horn,1,2 adina figl,1,2  sivaramakrishna rachakonda,1 christine fischer,3 antje sucker,2 andreas gast,1,2 stephanie kadel,1,2 iris moll,2 eduardo nagore,4 kari hemminki,1,5 dirk schadendorf,2 * rajiv kumar1 * cutaneous melanoma occurs in both familial and sporadic forms. we investigated a melanoma-prone family through linkage analysis and high-throughput sequencing and identified a disease-segregating germline mutation in the promoter of the telomerase reverse transcriptase (tert) gene, which encodes the catalytic subunit of telomerase. the mutation creates a new binding motif for ets transcription factors and ternary complex factors (tcfs) near the transcription start and, in reporter gene assays, caused up to twofold increase in transcription. we then screened the tert promoter in sporadic melanoma and observed recurrent ultraviolet signature somatic mutations in 125 of 168 (74%) of human cell lines derived from metastatic melanomas, 45 of 53 corresponding metastatic tumor tissues (85%), and 25 of 77 (33%) primary melanomas. the majority of those mutations occurred at two positions in the tert promoter and also generated binding motifs for ets/tcf transcription factors. the identification of germline mutations that cosegregate with disease in cancerprone families often provides genetic and mechanistic insights into the more common, sporadically arising cancers. in a study of cutaneous melanoma, the most malignant skin cancer, we investigated a large pedigree with 14 related melanoma patients who were not carriers of germline mutations in cdkn2a or cdk4, two known melanoma genes ( 1). multipoint linkage analysis showed a possible -mb linkage region on chromosome 5p with maximal logarithm of the odds ratio for linkage scores of  at rs1379917 and  at rs1968011. target-enriched high-throughput sequencing (hts) of the region was carried out on constitutional dna from the four affected and four unaffected members of the family with an average coverage between 55- and 108-fold (table s1) (1). the hts data revealed a single promoter variant, three intronic variants, and three nongene variants previously unknown and unique to the dna sequences of the affected individuals (table s2). the disease segregating variants, seven in total, were validated by sanger sequencing of dna from the individuals sequenced by hts and of dna from additional unaffected members of the family. the new variants were also detected in an unaffected member (754, table s3), who was 36 years old and carried multiple nevi. dna from affected individuals other than those sequenced by hts was not available for testing. of the seven unique variants identified, one variant (t>g), was located in the promoter at 57 base pairs (bp) from atg translation start site of the telomerase reverse transcriptase (tert ) gene. the tert gene encodes the catalytic reverse transcriptase subunit of telomerase, the ribonucleoprotein complex that maintains telomere length. the nucleotide change in the sequence cctgaa>ccggaa creates a new binding motif for ets transcription factors, with a general recognition motif gga(a/t). beyond the general motif for ets transcription factors, the familial mutation also generates a binding motif, ccggaa, for the ternary complex factors (tcfs) elk1 and elk4 (2, 3). to exclude the possibility that the detected promoter mutation in tert is a common germline variant, we screened germline dna from 140 sporadic melanoma cases and 165 healthy controls, and none carried the variant. screening of dna from index cases from 34 spanish melanoma families also did not show any mutations. no carriers were found in dbsnp and the 1000 genomes databases (data available for 18 individuals were obtained from ensembl). the familial mutation in the tert promoter was in complete allelic linkage with a common polymorphism rs2853669 (g>a) at 246 bp upstream from the atg start site (table s3). in previous work, this polymorphism was reported to disrupt an ets binding site, and it was associated with low telomerase activity in patients with non small cell lung cancer (4). in luciferase reporter gene assays, we found that the activity of constructs containing the mutation at 57 bp of the tert promoter was increased -fold and - fold over the wild-type construct in ma-mel-86a and human embryonic kidney (hek) 293t cells, respectively. a construct with both the tert mutation and the variant allele of the rs2853669 polymorphism showed a -fold increase in promoter activity in ma-mel-86a and and -fold increase in hek293 cells (mean from three measurements; details in supplementary text and  s1). the germline occurrence of the promoter mutation, creating an ets/tcf motif, can result in modification of tert expression in all tissues expressing ets/tcf. highest staining for the tcf elk1 protein has been reported in female-specific tissues, such as ovary and placenta. the increased expression of tcf elk1 protein in female-specific tissues may cause gender-related differences in 1 division of molecular genetic epidemiology, german cancer research center, im neuenheimer feld 580, 69120 heidelberg, germany. 2 department of dermatology, university hospital essen, 45122 essen, germany. 3 institute of human genetics, university of heidelberg, 69120 heidelberg, germany. 4 department of dermatology, instituto valenciano de oncologia, valencia, spain. 5 center for primary health care research, lund university, malm, sweden. *to whom correspondence should be addressed. e-mail: r.kumar@dkfz.de (r.k.); dirk.schadendorf@uk-essen.de (d.s.) these authors contributed equally to this work. www.sciencemag.org science vol 339 22 february 2013 959 reports  on june 2, 2017 http://science.sciencemag.org/ downloaded from  cancer susceptibility among carriers of the tert mutation (5) (supplementary text). two affected members of the family developed several different types of cancer (marked with # in  1). one affected individual presented with ovarian cancer at age 27 and melanoma at age 30. another individual was diagnosed with melanoma at age 20; later she developed ovarian cancer, renal cell carcinoma, bladder cancer, mammary carcinoma, and finally bronchial carcinoma, leading to her death at age 50. the mutation in the melanoma-prone family prompted us to screen melanoma cell lines derived from sporadic cases of metastatic melanoma. none of the cell lines carried the mutation detected in the family. however, we identified recurrent ultraviolet (uv)signature mutations in the tert core promoter in 74% (125 of 168) of the cell lines. the mutations were located within a 49-bp region starting from 100 bp upstream of the atg start site (table 1,  2,  s2, and table s4). there were two frequent mutations at 124 bp (g>a; c>t on opposite strand) and 146 bp (g>a); these mutations were mutually exclusive and occurred in 27 and 38% of cell lines, respectively. two tandem gg>aa (cc>tt) mutations at positions 124/125 bp and 138/139 bp were observed at a frequency of 9%. the tandem mutation at positions 138/139 bp could also be generated by a single-base mutation at 138 bp, because the base change at 139 bp has been reported as a rare polymorphism (rs35550267). the two most frequent single-base mutations as well as the two tandem mutations also result in the creation of ets/tcf binding motifs. mutations were confirmed in 45 of 53 (85%) available metastasized tumors corresponding to the cell lines. the somatic nature of the mutations was shown by the absence of mutations in corresponding dna from peripheral blood mononuclear cells available from 23 patients. somatic mutations in the tert promoter were more frequent than the braf mutations (53%, 90 of 169), cdkn2a alterations (50%, 84 of 169), and nras mutations (23%, 38 of 169;  s3). the occurrence of concomitant mutations in the tert promoter and braf was more frequent (47%) than by random chance (40%) with an odds ratio (or) of  [95% confidence interval (ci)  to ]. concomitant mutations in tert, braf, and cdkn2awere observed in 30% of cell lines compared with the expected frequency of such occurrence of 9% (or , 95% ci  to ). the high recurrence and specificity of the tert promoter mutations, together with the preliminary evidence from reporter assays that they have a functional effect on transcription, suggest that these mutations are driver rather than passenger events. extensive functional studies will be required to validate this hypothesis. the tert promoter mutations were also detected in 25 out of 77 (33%) paraffin embedded primary melanoma tumors (table 1 and table s5) at 124 bp (7/77; 9%) and 146 bp (5/77; 7%). four primary tumors carried the gg>aa tandem mutations at 124/125 bp, and eight primary tumors carried the gg>aa tandem mutations at 138/139 b reduced sensitivity to detect mutations in paraffin-embedded primary tumors because of contaminating normal cells cannot be ruled out. primary tumors harbored five additional mutations in the tert promoter, which were not present in metastases, and those did not generate ets/tcf binding motifs. we also screened dna extracted from 25 melanocytic nevi and only one carried a mutation at 101 bp, which did not create an ets/tcf motif. for both primary tumors and melanocytic nevi, matched normal control dna was not available for testing. the tert coding region has been reported to be somatically mutated in 1% of cancers (14 cancer types, 1271 unique samples) (6). mutations creating ets/tcf binding motifs in the tert promoter in melanoma have not been described in earlier sequencing projects. tcfs are a subfamily of ets transcription factors; two members of this subfamily, elk1 and elk4, are downstream targets of braf and regulate the expression of many genes (711). conceivably, tcf may represent a link between telomerase activity and the frequent braf activating mutations in melanoma ( s4) (12, 13). lastly, whether tert promoter mutations occur in other cancer types remains to be determined. we did not detect these mutations in a screen of 22 esophageal squamous cell c  reactivation of telomerase has been implicated in human tumorigenesis, but the underlying mechanisms remain poorly understood. here we report the presence of recurrent somatic mutations in the tert promoter in cancers of the central nervous system (43%), bladder (59%), thyroid (follicular cell-derived, 10%) and skin (melanoma, 29%). in thyroid cancers, the presence of tert promoter mutations (when occurring together with braf mutations) is significantly associated with higher tert mrna expression, and in glioblastoma we find a trend for increased telomerase expression in cases harbouring tert promoter mutations. both in thyroid cancers and glioblastoma, tert promoter mutations are significantly associated with older age of the patients. our results show that tert promoter mutations are relatively frequent in specific types of human cancers, where they lead to enhanced expression of telomerase. doi: /ncomms3185 1 institute of molecular pathology and immunology of the university of porto (ipatimup), 4200-465 porto, portugal. 2 institute of biomedical sciences of abel salazar, university of porto, 4050-313 porto, portugal. 3 medical faculty, university of porto, 4200-319 porto, portugal. 4 portuguese institute of oncologycoimbra centre (ipofg, epe), 3000-075 coimbra, portugal. 5 experimental pathology and therapeutics group, portuguese institute of oncology, 4200-072 porto, portugal. 6nucleo de investigacao em farmacia, centro de investigacaoem saude e ambiente (cisa), health school of the polytechnic institute of porto, 4400-330 vila nova de gaia, portugal. 7 research department, portuguese league against cancer (norte), 4200-177 porto, portugal. 8department of endocrinology, diabetes and metabolism, university hospital of coimbra, 3000-075 coimbra, portugal. 9 unit of endocrinology, medical faculty, university of coimbra, 3000-548 coimbra, portugal. 10 centre of molecular and environmental biology (cbma), department of biology, university of minho, campus de gualtar, 4710-057 braga, portugal. 11department of pathology, hospital s. joao, 4200-319 porto, portugal. 12department of pathology, hospital de braga, 4710-243 braga, portugal. 13 life and health sciences research institute (icvs), school of health sciences, university of minho, 4710-057 braga, portugal. 14 molecular oncology research center, barretos cancer hospital, cep 14784-400 sao paulo, brazil. 15department of pathology, clinical university hospital, sergas, medical faculty, university of santiago de compostela, idis, 15706 santiago de compostela, spain. * these authors contributed equally to this work. correspondence and requests for materials should be addressed to s. (email: psoares@ipatimupt). nature communications | 4:2185 | doi: /ncomms3185 | www.nature.com/naturecommunications 1 & 2013 macmillan publishers limited. all rights reserved. reactivation or re-expression of telomerase is referred to be a widespread feature in human cancers, although the genetic basis remains poorly understood in many cancer types. somatic mutations in the coding region of tert (human telomerase reverse transcriptase) are infrequent in human tumours, but germline and somatic mutations in tert promoter were recently found in a high percentage of human melanomas and human cancer cell lines1,2. such mutations occurred in two hotspot positions, located  124 and  146 bp upstream from the atg start site (  124 g4a and  146 g4a, c4t on opposite strand) and conferred enhanced tert promoter activity1,2, by putatively generating a consensus binding site (ggaa) for ets transcription factors within the tert promoter region1,2. our aim was to investigate whether the aforementioned tert promoter mutations were present in cancer types other than melanoma, having for that matter screened 741 primary tumours from the thyroid, kidney, bladder, gastrointestinal stromal tumour (gist), adrenal medulla (phaeochromocytomas) and central nervous system (cns). besides skin melanoma, we have also included ocular melanoma (not studied in the previous reports), benign lesions of the thyroid and skin and 58 human cancer-derived cell lines. our results highlight tert promoter mutations as frequent events in specific types of human cancers. results tert mutations in tumours and cell lines. overall, tert promoter mutations were found in 142 (19%) human tumour samples (table 1;  1a) and 14 (24%) human cell lines ( 1b; supplementary tables s1 and s2). the  124-bp mutation was the most frequent, being present in 99 cases, whereas the  146- bp mutation was present in 43 cases. the  124 g4a and  146 g4a mutations were found in a mutually exclusive fashion. two tandem gg4aa mutations at positions  124/  125 and  138/  139 bp were observed in one cell line each (supplementary  s1 and supplementary table s1). tert promoter mutations were not detected in normal thyroid tissue nor in benign lesions (nevi, thyroiditis, goitres and adenomas). no mutations were detected in 26 kidney cancers, 17 phaeochromocytomas and 36 gists. tert mutations in melanoma. we have analysed both skin and ocular melanomas, a type of melanoma that was not previously studied1,2. tert mutations were present in 16 out of 56 (29%) skin melanomas, but absent in the 25 ocular melanomas (table 1). all the four skin melanoma-derived cell lines but none of the six ocular melanoma-derived cell lines harboured tert mutations ( 1b and supplementary table s1). ten out of 16 (63%) skin melanomas with tert mutation also harboured the brafv600e mutation, whereas the braf mutation was present in 9 out of 38 (24%) melanomas without tert mutation (p  , fishers exact test, two-sided) (supplementary  s2 and supplementary table s3). tert mutations in cns tumours. in cns tumours, tert promoter mutations were found in 43% (51/118) of all cases with an equal prevalence of both mutations (table 1 and supplementary table s4). the frequency of tert mutations was different according to the tumour histology and grade ( 1c): although pilocytic astrocytomas (world health organization (who) grade 1) and diffuse astrocytoma (who grade 2) showed a lower frequency of mutations (8% and 15%, respectively), the most aggressive form, glioblastoma multiforme (gbm; who grade 4), present the highest frequency of tert mutations (62%). oligodendrogliomas (who grade 2) and anaplastic oligodendrogliomas (who grade 3) also harboured a high frequency of tert mutations (45% and 54%, respectively). the presence of tert mutations was significantly associated with a higher mean age at diagnosis (supplementary table s5) in cns patients as a whole (po; mannwhitney test, twotailed) and also in gbm patients only (p  ; mann whitney test, two-tailed); in oligodendroglioma patients, we found the same trend, although not statistically significant (p  ; mannwhitney test, two-tailed). in addition, we observed that, in a subset of 14 gbm (8 tert wild-type tumours and 6 tert-mutated tumours), the tumours with tert mutations showed a trend to increased immunohistochemistry (ihc) expression of telomerase, although this difference was not statistically significant (p  ; mannwhitney test, two-tailed) ( 2). tert mutations in thyroid cancer. we analysed a large series of follicular cell-derived thyroid samples including normal thyroid (n  27), benign (n  81) and malignant lesions (n  263) (table 1 and supplementary table s6). tert mutations were only detected in malignant tumours (10%) ( 1d), namely in table 1 | prevalence of tert promoter recurrent mutations in human cancers. organ/tissue number tert mutation, n (%) melanocytes nevi 9 0 skin melanoma 56 16 (29%) ocular melanoma 25 0 cns pilocytic astrocytoma 13 1 (8%) diffuse astrocytoma 20 3 (15%) oligodendroglioma 22 10 (45%) anaplastic oligodendroglioma 24 13 (54%) glioblastoma 39 24 (62%) thyroid benign 81 0 ptc 169 13 (8%) ftc 64 9 (14%) pdtc 14 3 (21%) atc 16 2 (13%) mtc 28 0 bladder low grade 21 14 (67%) high grade 61 34 (56%) kidney ccrcc 12 0 cromrcc 4 0 prcc 10 0 adrenal phaeochromocytoma 17 0 gi gist 36 0 total 741 142 (19%) atc, anaplastic thyroid carcinoma; ccrcc, clear cell renal cell carcinoma; cns, central nervous system; cromrcc, chromophobe renal cell carcinoma; ftc, follicular thyroid carcinoma; gi, gastrointestinal; gist, gastrointestinal stromal tumour; mtc, medullary thyroid carcinoma; pdtc, poorly differentiated thyroid carcinoma; prcc, papillary renal cell carcinoma; ptc, papillary thyroid carcinoma. article nature communications | doi: /ncomms3185 2 nature communications | 4:2185 | doi: /ncomms3185 | www.nature.com/naturecommunications & 2013 macmillan publishers limited. all rights reserved. 11% of papillary thyroid carcinomas (ptcs), 14% of follicular thyroid carcinomas, 21% of poorly differentiated thyroid carcinomas and 13% of the anaplastic thyroid carcinomas. within the group of ptc, tert mutations were detected to be associated to the so-called conventional ptc (cptc) (11%).the majority of tert-mutated thyroid cancers harboured the  124 g4a mutation (22 out of 27 cases) ( 1d and supplementary table s7). in thyroid cancer patients, tert mutations were significantly associated with guarded prognosis features such as larger tumours (p  ; mannwhitney test, two-tailed), older patients (po; mannwhitney test, two-tailed) and male gender (p  ; fishers exact test, two-sided) (supplementary table s5). after histotype stratification, such correlations were only kept in the group of cptc where we also found significant associations with lymph node metastasis (p  ; fishers exact test, two-sided) and brafv600e mutation (p  ; fishers exact test, two-sided) (supplementary  s2 and supplementary table s8). accordingly, we verified by quantitative rt-pcr (qpcr) that tert mrna is expressed at variable amounts in thyroid tumours, and a% of mutated cases cns (n=118) thyroid (n=263) bladder (n=82) skin melanoma (n=56) 146 g>a 124 g>a b 10 5 0 number of cell lines c 80 60 40 20 0 80 60 40 20 0 % of mutated cases % of mutated cases % of mutated cases pa (n=13) a (n=20) o (n=22) ao (n=24) gb (n=39) d 0 5 10 15 20 25 ptc (n=169) ftc (n=64) pdtc (n=14) atc (n=16) e 80 60 40 20 0 low grade (n=21) high grade (n=61) 146 g>a 124 g>a wt tandem 124/125 gg>aa tandem 138/139 gg>aa 146 g>a 124 g>a cns (n=1) lung (n=1) peripheral b lymphocytes (n=8) leukaemia (n=2) kidney (n=2) breast (n=8) gastric (n=8) colorectal (n=8) melanoma (uveal) (n=6) melanoma (skin) (n=4) thyroid (n=10) figure 1 | frequency and schematic illustration of tert promoter mutations in human cancers. graphics depict the overall frequency of tert mutations in the four tumour types where tert mutations were detected (a) and a bar plot showing the number of cell lines of different origin that harbour tert mutations (b). the frequency and type of tert mutations in different histological subtypes of cns (c), thyroid (d) and bladder (e) tumours is also shown. a, diffuse astrocytoma; ao, anaplastic oligodendroglioma; atc, anaplastic thyroid carcinoma; gb-glioblastoma; ftc, follicular thyroid carcinoma; o, oligodendroglioma; pa, pilocytic astrocytoma; pdtc, poorly differentiated thyroid carcinoma. a b c p=  10 8 6 4 2 0 tert wt (n=8) tert mutation (n=6) tert expression level (au) figure 2 | ihc for tert in two glioblastomas. (a)a tumour without tertexpression and without tert mutation. (b) a tumour with positive staining both in nucleus and cytoplasm that also harboured a tert mutation. the graph (c) displays quantification of tert nuclear expression level (ihc) in glioblastomas with and without tert mutation. we have measured the extent (o25%, 2550%, 5075% and 475%) and the intensity (absent, faint, moderate or strong) of tert nuclear staining in 14 glioblastomas (eight tert wild type and six tert mutant). the scoring was performed by two independent observers and is the product of extent and intensity of the staining (methods). scale bar, 200 mm. there is a trend to higher tert expression in tert-mutated gliomas, although the difference was not statistically significant (p  ; mannwhitney test, two-tailed). bars represent s.e.m. nature communications | doi: /ncomms3185 article nature communications | 4:2185 | doi: /ncomms3185 | www.nature.com/naturecommunications 3 & 2013 macmillan publishers limited. all rights reserved. the ones with coexistent tert and braf mutations showed the highest levels of tert mrna expression (p  , unpaired t-test;  3). tert mutations in bladder cancer. in bladder cancer, tert mutations were frequently detected (48/82; 59%) both in lowgrade tumours (14/21; 67%) and high-grade tumours (34/61; 56%) (table 1 and  1e; supplementary table s9). the  124 g4a mutation was present in 30 cases, whereas the  146 g4a was found in 18 cases. no association was found between the presence of the mutation and the age of the patients. no mutations were detected in 26 kidney cancers, 17 phaeochromocytoma and 36 gists (supplementary tables s10, s11 and s12, respectively). discussion recent findings have described that tert promoter mutations arise as a novel mechanism of telomerase reactivation/expression in human cancers1,2. in this work, we report for the first time the presence of these mutations in different histotypes of thyroid cancer, and our results validate the findings reported by others in melanoma, bladder and gliomas13. furthermore, we present for the first time evidence showing that tert promoter mutations are significantly associated with increased tert mrna expression in thyroid cancers. we also found a trend for increased telomerase expression in cases of gbm harbouring tert promoter mutations. in thyroid cancers, tert mrna was particularly high in cases harbouring both tert and braf mutations. the location of these mutations in the tert promoter, rather than the coding region of the gene, creating additional binding sites for transcription factors, represents also a novel mechanism of genetic activation in cancer. the frequency of tert mutations we detected in skin melanomas is similar to that reported by horn 1 in primary melanomas (33%), but lower than the frequency found by the same authors in metastatic melanomas (85%) and by huang 2 in a series of 70 melanomas and short-term cultures1,2. in our series, we have analysed few metastatic melanomas that did not show a significantly higher percentage of tert mutations. as our series is too small to allow a conclusion, it remains to be verified if there is an increased frequency of tert mutations in metastatic melanomas, as suggested by previous reports1,2. we have confirmed, in skin melanomas, the significant association between tert and braf mutations, as previously advanced by horn 1, which we now also demonstrate in thyroid cancers. interestingly, in benign nevi, which frequently harbour braf mutations, we did not detect tert mutations. this finding, together with the high frequency of tert mutations in metastatic melanomas1, suggests that, although braf mutations are thought to be an early event in melanoma genesis, tert mutations may occur at a later stage. no tert promoter mutations were found in ocular melanomas, which, at variance with skin melanomas, harbour gnaq mutations instead of braf mutations4,5. our results further support the assumption that ocular melanomas result from different etiopathogenic mechanism than skin melanomas5,6. our data on cns tumours indicate that tert promoter mutations are frequent events in gliomas, particularly in gbm where the highest mutation frequency was found. it has been shown that tert mutations conferred enhanced tert promoter activity in vitro1,2; our findings fit with this assumption because we observed that gbm with tert promoter mutations appear to display increased telomerase expression, despite the limited sensitivity of ihc and the low number of cases analysed. our results are also in concordance with those of lotsch 7 who reported that 60% of gbms were positive for tert mrna and telomerase activity7. it will be interesting to assess whether the gbm positive for telomerase activity described by lotsch 7 also harbour tert promoter mutations. our findings also indicate that tert mutations are associated with older patients, in accordance with a recent report3. the high frequency of tert mutations in cns tumours, particularly in intermediate and high-grade tumours, opens a window for new approaches in the therapy of cns tumours. in thyroid cancer, tert mutations were only found in follicular cell-derived cancers (ptc, follicular thyroid carcinoma, poorly differentiated thyroid carcinoma and anaplastic thyroid carcinoma). no mutations were detected in medullary thyroid carcinomas, as described by killela 3, nor in normal thyroid and benign lesions, such as goitre, adenomas or thyroiditis; this finding fits with previous studies that reported telomerase expression in malignant lesions and not in normal tissue or hyperplastic lesions8,9. tert mutations were associated with clinicopathological features (older age, increased tumour size and a b 5 4 3 2 1 0 1/()ct wt wt n wt wt b n b wt wt wt wt b b wt n wt pod n retr r t ret rt bt bt rt bt p< 4 3 2 1 0 1/()ct (mean) tert wt (n=18) tert mutation (n=6) figure 3 | quantification of tert mrna expression level in thyroid tumours. we have performed qpcr in normal thyroid tissue (n), a pool of mrna of nine normal thyroids (pool n) and thyroid tumours with different genetic backgrounds: without known genetic alterations (wt), with braf mutation (b), with ret rearrangement (ret), with ras mutation (r), with tert mutation (t), with tert and braf mutations (bt) or with ras and tert mutations (rt). (a) each bar represents tert mrna expression in an individual tumour measured in triplicate. tert mrna levels were normalized against tbp mrna levels. (b) quantification of tertmrna mean expression level in thyroid tumours with and without tert mutation. error bars represent s.e.m. the difference is statistically significant (po, unpaired t-test). article nature communications | doi: /ncomms3185 4 nature communications | 4:2185 | doi: /ncomms3185 | www.nature.com/naturecommunications & 2013 macmillan publishers limited. all rights reserved. male gender), but, after histotype stratification, these associations were only maintained in cpt studies in larger series will be necessary to clarify these associations. in cptc, tert mutations were associated with braf mutation, highlighting the coexistence of activation of braf and tert, previously reported in melanoma1 (supplementary  s2). our results reinforce the hypothesis of a link between braf activation and telomerase expression, which can be mediated by transcription factors binding to the newly created consensus binding sites, as previously advanced by horn 1. both in thyroid carcinoma and melanoma, it seems that a background status of activated braf enhances the effects of tert promoter mutation. our results in tert mrna expression strengthen this assumption, showing, for the first time, an increased tert expression in tumours harbouring braf and tert mutation ( 3). further observational and mechanistic studies are needed to clarify these points. tert alterations seem to constitute an early and frequent event in bladder cancer. these findings fit with the high prevalence of telomerase activity previously described in bladder tumours10. the recurrent hotspot mutations in tert were advanced to be, in melanomas, induced by ultraviolet radiation2. bladder is a target for several chemical carcinogens; it remains to be found whether tert mutations can also result from the action of such agents. the identification of tert mutations in urine may provide a biomarker for early diagnosis and monitoring of bladder cancer. no tert mutations were detected in kidney cancers, nor in 17 phaeochromocytoma and 36 gists. the underlying reasons for the tissue specificity of tert mutations remain to be clarified. we can hypothesize that tert mutations can be present in two settings. as described by killela 3, tert mutations can be relevant in tissues with relative low rates of self-renewal, which fits with our findings in follicular cell-derived thyroid cancer and gliomas where they are associated with the older age of the patients. on the other hand, these mutations can also result from environmental factors such as ultraviolet radiation and chemical carcinogens as suggested by their high frequency in melanoma, bladder and tongue13. in summary, our data identify tert mutations as common events in human cancers and support the assumption that tert promoter mutations may be one of the mechanisms that underlies telomerase reactivation in several types of human tumours.  introduction the telomere sequences at the chromosomal ends, composed of tandem repeats of ttaggg, are protected by a number of molecules that constitute the capping shelterin complex [1 ;  2]. the incomplete replication due to limitations of the process, called end-replication problem, results in shortening of telomeres in each successive mitotic cell division that eventually leads to replicative senescence referred to as the hayflick limit [3; 4 ;  5]. maintenance of telomere repeat length is dependent on sustained expression of telomerase holoenzyme that adds de novo repeat units at the end of each replication cycle [ 6 ;  7]. progressive attrition of telomeres is also defined as one of the hallmarks of aging of organisms [ 8 ;  9]. cancer cells, characteristically, acquire infinite capability to divide through maintenance of telomeres by sustained expression of telomerase or in its absence by an alternative lengthening of telomeres (alt) mechanism [ 10; 11; 12; 13; 14 ;  15].  human telomerase reverse transcriptase (tert) gene encodes the catalytic subunit of telomerase that together with a rna component, terc, maintains genomic integrity by telomere elongation [ 16]. though tert and terc are sufficient for in vitro telomerase activity, the in vivo telomerase functioning requires additional components that associate with tert and terc, to form the holoenzyme [ 17 ;  18]. those include dyskerin (dkc), nop10 ribonucleoprotein (nop10), gar1 ribonucleoprotein homolog (yeast) (gar1), nhp2 ribonucleoprotein (nhp2), reptin and pontin [ 11; 19; 20; 21 ;  22].  deregulation of telomerase has been shown to be a ubiquitous feature in human cancers with over 90% of cancers showing an upregulation of the enzyme [11; 23 ;  24]. the telomerase activity is dependent on a number of factors, regulated at various stages, which include gene transcription, mrna splicing, maturation and modifications of tert and terc, transport and localization of those components and assembly of active telomerase ribonucleoprotein to telomeres [ 4; 5; 16; 20; 21; 25; 26 ;  27]. the catalytic component tert acts as a determinant of telomerase activity and its transcription is repressed in most of the somatic cells with the exception of proliferative cells of self-renewing tissues [ 28; 29 ;  30]. an increased expression of tert has been consistently demonstrated to be a fundamental requirement for cellular transformation [ 4; 31; 32; 33; 34 ;  35].  the mechanism of tert upregulation in cancers had been attributed to several mechanisms including epigenetic deregulation as well as genetic amplification of the locus containing tert gene [ 36 ;  37]. in the absence of any evidence of a definite mechanism, the telomerase activity in tumor cells has been attributed to the assumption of stem cells being the progenitors in all cancers [38]. the normal stem cells in self-renewing tissues retain telomerase throughout lifetime replication thus abrogating a requirement for a positive selection [6]. the recently discovered tert promoter mutations add a new dimension to the acquisition of telomerase activity in human cancers. in this review we provide an overview and possible implications of the newly discovered mutations in the promoter of the tert gene in a wide range of cancers.  structure and regulation of the tert promoter the human tert gene is located on chromosome 5p and the promoter region of the gene is considered to be the most important regulatory element for telomerase expression. the tert promoter contains binding motifs for several factors that regulate the gene transcription and distinctly lacks a tata box or a similar sequence [ 39; 40; 41; 42 ;  43]. the core promoter region consists of 260 base pairs with several transcription-factor binding sites that include e-boxes where c-myc has been confirmed to bind and activate the transcription [ 44; 45; 46; 47 ;  48]. brca1 in conjunction with n-myc interacting protein (nmi) forms a complex with c-myc and inhibits tert promoter activity, that property is lost in some mutant forms of brca1 [ 49]. other sequence elements in tert promoter include gc-boxes, which are binding sites for zinc finger transcription factor, sp1 [ 4 ;  45]. transcription of the tert gene is also regulated by various hormones, cytokines and oncogenes [ 45]. several repressors of the tert transcription are also known. p53 has been shown to downregulate tert transcription in a sp1-dependent manner [ 50]. ets transcription factors that comprise over 30 members are prominently associated with telomerase activation [ 42 ;  51]. ets2 has been shown to form a complex with c-myc in a breast cancer cell line [ 44 ;  51]. ets transcription factors are also shown to be stimulated by oncogenes egf, her2/nez, ras and raf [ 52 ;  53]. the activation of oncogenes and inactivation of tumor suppressors are known to account for cellular immortalization through induction of tert transcription [ 54]. the high gc content around the transcription start site of the tert promoter confers epigenetic regulation through methylation and chromatin remodeling [ 37 ;  55].  tert promoter mutations in human cancers a discovery of a high-penetrant disease-segregating causal germline mutation in a melanoma family and highly specific and recurrent somatic mutations in tumors from unrelated patients in the tert promoter has likely provided a definite mechanism for cancer-specific tert activation [ 56 ;  57]. two independent studies using diverse approaches discovered non-coding mutations, mainly at two residues, within the core promoter region of the tert gene. one study was based on the identification of a causal gene mutation in a large melanoma pedigree where affected individuals presented a severe form of the disease with an early age of onset. the linkage analysis identified a  megabase telomeric region on chromosome 5p that included tert along with more than 80 other genes [ 56]. sequencing of the entire stretch of dna region in the family resulted in identification of a disease segregating a > c (t > g) single base change at 57 bp (chr 5: 1,295,161 hg19 coordinate) from atg start site. the germline mutation was present in affected and absent in unaffected individuals in the family with the exception of one. subsequent screening of cell lines derived from melanoma metastases from unrelated patients led to the detection of recurrent and mutually exclusive somatic mutations at two residues 124 and 146 from the atg start site in the tert promoter [ 56]. serendipitously, an independent study using a whole genome sequencing approach also reported the recurrent somatic tert promoter mutations at the same positions [ 57]. other mutations detected in tert promoter included the cc > tt tandem mutations at 124/125 and 138/139 bp from atg start site. the germline and somatic mutations in the non-coding part of the tert gene were defined by common salient features. one of the underlying features included a de novo creation of ccggaa/t general binding motifs for e-twenty six/ternary complex factors (ets/tcf) transcription factors, which differed from pre-existing ggaa/t ets binding sites within the tert promoter ( figure 1). the somatic mutations at both positions being c > t and the additional detection of cc > tt tandem mutations in a proportion of tumors augmented the evidence for the uv-origin of tumor specific nucleotide changes in melanoma as shown previously in studies based on whole genome sequencing [58]. interestingly, the mutations detected in the tert promoter in melanoma were more frequent than those in the braf gene.  schematic representation of a part of the tert promoter that contains residues, . figure 1.  schematic representation of a part of the tert promoter that contains residues, which are affected by a germline mutation in a melanoma family at the position 57 bp and recurrent somatic mutations at the positions 124 and 146 bp from the atg start site. the mutations create ccggaa/t binding motif for ets/tcf transcription factors that results in an increased tert expression. pre-existing binding sites for other transcriptions factors are shown above the sequence. figure options it was also observed that the tert promoter mutations tend to occur more often than expected by chance in tumors with either braf mutations (odds ratio [or] , 95% confidence interval [ci] ) or with concomitant alterations in both braf and cdkn2a (or , 95% ci ) [ 56]. braf mutations, due to occurrence and role in development of melanocytic nevi, are considered as the driver genetic lesions in melanoma [ 56; 59 ;  60]. the loss of cdkn2a has been suggested to play a role in the escape of melanocytes from braf induced senescence [61]. the acquisition of tert promoter mutations can be hypothesized to facilitate stabilization of the transformed genome through reversal of telomeric loss. most melanocytic nevi carry braf mutations, whereas tert promoter mutations and cdkn2a alterations are detected only in primary melanoma and beyond [ 56 ;  62].  bonafide of newly discovered non-coding mutations in the tert promoter was established by the detection in cancers other than melanoma [ 63]. the frequency of the mutations seems to vary between cancer types (table 1). the highest frequencies of the tert promoter mutations have so far been reported, besides melanoma, in pleomorphic dermal sarcoma, myxoid liposarcoma, glioma, urothelial cell carcinoma of bladder, basal and squamous cell carcinoma of skin, liver cancer and others [ 63; 64; 65; 66; 67; 68; 69; 70; 71; 72; 73; 74; 75 ;  76]. the mutations occur in other cancer types as well, albeit, at low frequencies [ 63 ;  67]. based on the prevalence in different cancer types it has been hypothesized that the tert promoter mutations mainly occur in tumors that are derived from tissues with low rates of self-renewal [ 63 ;  77]. unlike melanoma and other skin related malignancies, no tumor from the cancers affecting internal organs carried cc > tt tandem mutations in the tert promoter with the exception of that at the positions 138/139 bp from atg start site in bladder cancer. the tandem mutation reported so far in 4 of the 1231 bladder tumors could also be generated by a single-base mutation at 138 bp as the base change at 139 bp has been reported as a rare polymorphism represented by rs35550267 [ 56; 65 ;  66]. the differences in mutational pattern in cancer types are known to reflect etiological divergences and the c > t base change in tumors can also be attributed to apobec cytidine deaminase expression in cancer development [ 78 ;  79].  table 1. tert promoter mutations in cancers cancer type\\tmutation frequency (%)a\\tref. bladder cancer\\t887/1231 ()\\t[63; 64; 65; 66; 67 ;  80] glioma  ependymomas\\t1/36 ()\\t[63]  astrocytomas\\t574/1059 ()\\t[63; 64; 67; 69 ;  73]  mixed gliomas\\t102/188 ()\\t[63 ;  69]  oligodendrogliomas\\t46/72 ()\\t[63 ;  67] melanoma  cutaneous melanoma\\t136/256 (53)\\t[56; 57 ;  67]  ocular melanoma (not specified)\\t0/25\\t[67]   uveal melanoma\\t0/47\\t[68]   conjunctival melanoma\\t12/38 (32)\\t[68] squamous cell carcinoma (scc)  scc of head and neck\\t12/70 ()\\t[63]  scc of esophagus\\t5/313 ()\\t[77]  scc of the cervix\\t1/22 ()\\t[63]  scc of the skin\\t14/31 ()\\t[63 ;  76]  bowen's disease\\t1/11 ()\\t[76] basal cell carcinoma of skin\\t31/42 ()\\t[76] thyroidb  atc + pdtc\\t73/170 ()\\t[67; 70 ;  71]  dtc\\t41/336 ()\\t[70]  ftc\\t20/143 ()\\t[67 ;  70]  hcc\\t4/25 ()\\t[71]  ptc\\t61/506 ()\\t[67; 70 ;  71] atypical fibroxanthoma\\t25/27 ()\\t[75] myxoid liposarcoma\\t19/24 ()\\t[63] pleomorphic dermal sarcomas\\t26/34 ()\\t[75] liverc\\t218/531 ()\\t[63 ;  74] fibrosarcoma\\t1/3 ()\\t[63] dysembryoplastic neuroepithelial tumor\\t1/3 ()\\t[63] medulloblastoma\\t19/91 (%)\\t[63] solitary fibrous tumor (sft)\\t2/10 (%)\\t[63] ovarian, clear cell carcinoma\\t2/12 (%)\\t[63] ovarian, low grade serous\\t1/8 (%)\\t[63] malignant pleural mesothelioma\\t8/71 (%)\\t[72] endometrial cancer\\t2/19 (%)\\t[63] myxofibrosarcoma\\t1/10 (%)\\t[63] neuroblastoma\\t2/22 (%)\\t[63] osteosarcoma\\t1/23 (%)\\t[63] ref [63]: no mutations were found in acute myeloid leukemia (n = 48), alveolar rhabdomyosarcoma (n = 7), atypical lipomatous tumor (n = 10), breast carcinoma (n = 88), cholangiosarcoma (n = 28), central/conventional chondrosarcoma (n = 9), chronic lymphoid leukemia (n = 15), chronic myeloid leukemia (n = 6), colorectal adenocarcinoma (n = 22), embryonal rhabdomyosarcoma (n = 8), esthesioneuroblastoma (n = 11), extraskeletal myxoid chondrosarcoma (n = 3), fibrolammellar carcinoma of the liver (n = 12), gall bladder carcinoma (n = 10), hepatoblastoma (n = 3), leiomyosarcoma (n = 3), conventional lipoma (n = 8), low grade fibromyxoid sarcoma (n = 9), malignant peripheral nerve sheath tumor (n = 3), medullary thyroid carcinoma (n = 24), meningioma (n = 20), mesothelioma (n = 4), pancreatic acinar carcinoma (n = 25), pancreatic ductal adenocarcinoma (n = 24), pancreatic neuroendocrine tumor (n = 68), prostate carcinoma (n = 34), spinal ependymoma (n = 9), synovial sarcoma (n = 16), or undifferentiated pleomorphic soft tissue sarcoma (n = 10) samples.  ref [67]: no mutations were found in phaeochromocytoma (n = 17); ccrcc, cromrcc and prcc of the kidney (n = 26).  refs [63 ;  67]: no mutations were found in gastrointestinal stromal tumors (n = 45).  refs [56 ;  67]: no mutations were found in melanocytic nevi (n = 34).  refs [67 ;  70]: no mutations were found in benign thyroid tumors (n = 166) or medullary thyroid carcinoma (n = 44).  a includes all reported tert promoter mutations; most common mutations are 124c > t (chr 5:1,295,228 hg19 coordinate) and 146c > t (1,295,250). in melanoma 146c > t mutation is more frequent than the 124c > t; in cancers, especially in gliomas, thyroid cancers and bladder cancers the latter is the most common mutation. additionally, in melanoma two cc > tt tandem mutations affecting 124/125 and 138/139 residues were also detected with a combined frequency of 9%[56].  b atc, anaplastic thyroid carcinoma; ftc, follicular thyroid carcinoma; pdtc, poorly differentiated thyroid carcinoma; ptc, papillary thyroid carcinoma; hcc, hurthle cell cancers; dtc, differentiated thyroid cancer.  c includes hepatocellular carcinoma, cirrhotic tissue, cirrhotic macronodules, hepatocellular adenomas, hca with hcc foci.  table options functional aspects of tert promoter mutations the high recurrence, specificity and gain of function support that the non-coding tert promoter mutations are driver rather than passenger events in cancer development. the functional relevance of the mutations was indicated by the basic reporter assays that showed 24-fold increased promoter activity [ 56; 57 ;  80]. tumors from thyroid cancers, primary glioma, malignant pleural mesothelioma and liver cancers with tert promoter mutations were associated with higher gene expression than those without mutations [ 67; 69; 72 ;  74]. though limited at the moment, the available data do indicate a tendency of the tert promoter mutations being present in specific clinical and phenotypic subtypes and appear to be associated with adversarial forms of the disease. while in medulloblastomas the tert promoter mutations were inversely associated with increased otx2 expression; in primary adult glioma, the mutations occurred mainly in conjunction with egfr amplification [ 69]. glioma patients with tert promoter mutations showed an association with poorer survival than patients without mutations; in thyroid cancer, mutations are reportedly more frequent in advanced thyroid cancers than in papillary thyroid cancers [ 63; 70 ;  71].  the studies on bladder cancer consistently showed that tert promoter mutations are the most frequent lesions with even distribution across all stages and grades [ 65; 66 ;  80]. intriguingly, an observed interaction has raised a possibility of eventual use of the tert promoter mutations in conjunction with a common polymorphism within the sequence as biomarkers in bladder cancer. the data from bladder cancer showed that the variant allele of a common polymorphism at 245 bp from atg start site in the tert promoter acts as a modifier of the effect of tert promoter mutations on patient survival and disease recurrence [ 80]. bladder cancer patients with tert promoter mutations in tumors showed almost two-fold decreased survival and increased disease recurrence in the absence but not in the presence of the variant allele for the rs2853669 polymorphism [ 80]. mechanistic support for the observation was provided by the fact that mutations result in de novo creation of ets/tcf binding motifs; the variant allele of the rs2853669 polymorphism, on the contrary, disrupts a preexisting non-canonical ets2 binding site in the proximal region of the tert promoter, adjacent to an e-box [ 44].  the occurrence of highly specific tert promoter mutations indicates a strong selection pressure for the gene over-expression on path to cellular transformation. increased telomerase production has been demonstrated to promote cancer progression in an animal model [ 81]. the effect of the promoter mutations on tert expression can be tenable only in the presence of ets/tcf transcription factors that can specifically bind to the de novo sites created by the mutations. some of the ets/tcf transcription factors are downstream targets of mapk pathway, where braf is a prominent intermediate [ 82; 83 ;  84]. whether in melanoma activated braf is a driving force in selection of tert promoter mutations remains to be determined. nevertheless, expression of ets transcription factors is ubiquitous in melanoma and other cancers [ 85 ;  86].  many studies have stressed the function of the tert gene beyond its role in maintenance of the telomere; therefore the mutations in the tert promoter can affect non-canonical processes associated with tert [ 87; 88; 89; 90 ;  91]. tert acts as a modulator of wnt-b-catenin signaling pathway and induces stem cell characteristics in glioma; tert also regulates expression of nf-kb, a master regulator of inflammation [ 92; 93 ;  94]. tert was shown to be important for proliferation of p53-negative cells through atr mediated stabilization of etv1, which binds downstream of the transcriptional start site [95]. other non-canonical functions of tert include enhanced cell proliferation, decreased apoptosis, regulation of dna damage responses, chromatin state and increased cellular proliferation life span [ 96; 97 ;  98]. the effect of the mutations beyond transcription also remains a probability. the human tert promoter contains g-rich sequence and has potential for g-quadruplex formation that can potentially be targeted to regulate gene transcription [ 99]. g-quadruplexes have been also implicated in inhibition of telomerase and control of gene expression [100].  conceptual advancement and therapeutic possibilities the tert promoter mutations are thought to represent a conceptual advancement in the sense that those instead of altering an encoded protein modulate transcriptional regulation and represent first evidence of driver alterations in so called dark matter of the human genome [ 101 ;  102]. a host of germline variants discovered through genome wide association studies contribute to the susceptibility of various diseases through transcriptional deregulation [103]. while tert promoter mutations represent novel findings in human cancer, alterations in components associated with telomerase assembly, telomere protection or telomere recruitment are known to impact stem cell function and lifespan in mammals through various disorders [ 104 ;  105].  several strategies of therapeutic telomerase inhibition including small molecular inhibitors, immunotherapy, gene therapy, telomere and telomerase-proteins in different cancers have entered clinical trial [106]. it will be interesting to see if the tert promoter mutations, that increase gene expression, influence the current on-going research on targeted therapeutics or if the use of telomerase inhibitors in conjunction with kinase inhibitors like vemurafenib or similar small molecules in melanoma can alleviate recurrent resistance [ 107]. malignant cells, like all actively growing cells, must maintain their telomeres, but genetic mechanisms responsible for telomere maintenance in tumors have only recently been discovered. in particular, mutations of the telomere binding proteins alpha thalassemia/mental retardation syndrome x-linked (atrx) or death-domain associated protein (daxx) have been shown to underlie a telomere maintenance mechanism not involving telomerase (alternative lengthening of telomeres), and point mutations in the promoter of the telomerase reverse transcriptase (tert) gene increase telomerase expression and have been shown to occur in melanomas and a small number of other tumors. to further define the tumor types in which this latter mechanism plays a role, we surveyed 1,230 tumors of 60 different types. we found that tumors could be divided into types with low (<15%) and high (15%) frequencies of tert promoter mutations. the nine tert-high tumor types almost always originated in tissues with relatively low rates of self renewal, including melanomas, liposarcomas, hepatocellular carcinomas, urothelial carcinomas, squamous cell carcinomas of the tongue, medulloblastomas, and subtypes of gliomas (including 83% of primary glioblastoma, the most common brain tumor type). tert and atrx mutations were mutually exclusive, suggesting that these two genetic mechanisms confer equivalent selective growth advantages. in addition to their implications for understanding the relationship between telomeres and tumorigenesis, tert mutations provide a biomarker that may be useful for the early detection of urinary tract and liver tumors and aid in the classification and prognostication of brain tumors.  telomeres are nucleoprotein complexes at the ends of eukaryotic chromosomes that are required for chromosomal integrity. several hundred nucleotides of telomere repeats cap each chromosomal end, and in the absence of telomerase activity, telomeres shorten with each cell division (1). eventually, uncapped telomeres trigger cell death or senescence. cancer cells seem to divide ad infinitum and therefore, require some telomere maintenance mechanism to avoid this fate. because telomerase activity is generally higher in cancer cells than normal cells, it was originally believed that telomerase was somehow activated in cancer cells (26). however, it was subsequently realized that telomerase was only inactive in terminally differentiated cells and that normal stem cells in self-renewing tissues retained telomerase activity (1, 79). because normal stem cells must replicate throughout the long lifetimes of mammals (which can be more than a century in humans), it is clear that such cells must also retain telomerase activity. because normal stem cells are thought to be the progenitors of cancers, there would be no need to specifically activate telomerase in cancer cells; the enzyme was already active in the precursors, just as were the hundreds of other enzymes and proteins normally required for cell proliferation.  this view was challenged by the discovery of another mechanism for maintaining telomere length [, alternative lengthening of telomeres (alt)] (1012). alt occurs in the absence of telomerase activity and seems to be dependent on homologous recombination. it occurs in a particularly high fraction of certain tumor types, such as sarcomas, pancreatic neuroendocrine tumors, and brain tumors, but rarely in most common tumor types, such as those tumor types of the colon, breast, lung, prostate, or pancreas (13). why would cancer cells need alt if telomerase activity was already constitutively active in their precursors? this question was highlighted by the discovery that many alt cancers harbor mutations in alpha thalassemia/mental retardation syndrome x-linked (atrx) or death-domain associated protein (daxx), genes encoding proteins that interact with each other at telomeres (10, 11). presumably, the absence of functional atrx/daxx complexes permits the homologous recombination resulting in alt. at minimum, these data were compatible with the ideas that there could be a selective advantage for genetic alterations that results in telomere maintenance and that telomerase is not indefinitely activated in all normal stem cell precursors of cancers.  another challenge to the idea that genetic alterations were not required for telomerase activation in cancer was raised by the finding that mutations of the telomerase reverse transcriptase (tert) promoter occurred in 70% of melanomas and in a small number of tumor cell lines derived from various tissue types (14, 15). importantly, only 5 of 110 cell lines derived from lung, stomach, ovary, uterus, or prostate cancers harbored tert promoter mutations, whereas 19 mutations were found among 37 cell lines derived from various other tumor types. this situation is analogous to the situation for alt, which is infrequently observed in common epithelial cancers but is observed more regularly in tumors derived from nonepithelial cells, particularly sarcomas and brain tumors (13).  these findings prompted us to formulate a hypothesis about the mechanisms responsible for telomerase activity in cancers. we suggest that there are two ways to maintain telomere lengths as cells divide: (i) through epigenetic regulation of telomerase activity, which occurs in stem cells of tissues that are rapidly renewing, and (ii) through somatic mutations that maintain telomere lengths, such as mutations in the tert promoter or mutations in daxx or atrx. those cancers that originate in tissues that are constantly self-renewing, such as cancers of the epithelia of the gastrointestinal tract and skin or bone marrow, would be unlikely to harbor telomere-maintaining mutations, because telomerase is already epigenetically activated in their precursor cells. in contrast, tumors arising from cells that are not constantly self-renewing, such as neurons, glial cells, fibroblasts, hepatocytes, islet cells, and pancreatic ductal epithelial cells, might frequently harbor such mutations. a corollary of this hypothesis is that tumor types exhibiting high frequencies of alt would also exhibit high frequencies of tert mutations, and these mutations would be distributed in a mutually exclusive fashion. to test these hypotheses as well as answer other questions related to the role of tert promoter mutations in various cancer types, we determined the prevalence of tert promoter mutations in a large number of tumors.  go to: results we attempted to evaluate at least 20 individual specimens of common tumor types and fewer specimens of rare tumor types, depending on availability of specimens in our laboratories. in those tumor types in which our pilot studies showed a significant number of mutations, additional tumors were evaluated. melanomas and tumors of the lung, stomach, and esophagus were excluded, because they had already been adequately evaluated in the seminal papers cited (14, 15). when primary tumors rather than cell lines were used, we ensured that the fraction of neoplastic cells was >50% through histopathologic examination of frozen sections of the tissue blocks used for dna purification. in those cases in which the neoplastic content was <50%, we microdissected the lesions to enrich the neoplastic content to >50%. primers were designed to amplify the region containing the two tert mutations that were previously describedc228t and c250tcorresponding to the positions 124 and 146 bp, respectively, upstream of the tert atg start site (14, 15). the pcr fragments were then purified and analyzed by conventional sanger sequencing.  in all, we evaluated tert promoter mutations in 1,230 tumor specimens and identified 231 mutations (%) (table 1). c228t and c250t mutations accounted for % and % of the alterations, respectively (dataset s1). additionally, we detected four mutations that had not been observed previously: three c228a mutations and one c229a mutation (dataset s1). all four of these mutations as well as a representative subset of the c228t and c250t mutations (n = 59) were somatic, as evidenced by their absence in normal tissues of the patients containing the mutations in their tumors.  table 1. table 1. frequency of tert promoter mutations the 1,230 tumors represented 60 tumor types. in 26 of these tumor types, at least 15 individual tumors were evaluated (comprising a total of 1,043 individual tumors) ( 1). in the remaining tumor types, only a small number of samples (212) was available, in part because these tumor types are generally uncommon in western populations (table 1). among the tumor types in which at least 15 individual tumors were available for study, a clear distinction could be made. eighteen of these tumor types had only occasional tert promoter mutations (zero to three mutations, comprising 015% of the tumors of each type) ( 1). we classified these tumor types as tert-low (tert-l), because they had a low frequency of tert promoter mutations. eight other tumor types were classified as tert-high (tert-h) because of their relatively high prevalence of tert promoter mutations (1683% of the tumors of each type).   1.  1. frequency of tert promoter mutations; 15 or more tumors were analyzed in 26 tumor types. gliomas are divided into primary gbm, astrocytoma (including astrocytoma grades ii and iii, as well as secondary gbm), and oligodendroglioma. the tert-l tumor types included some of the most prevalent cancers, including epithelial tumors of the breast, prostate, thyroid, pancreas, gall bladder, uterus, and colon (as well as tumors of the lung, stomach, and esophagus based on prior studies) (14, 15) and leukemias. in fact, no tert mutations were identified in any specimen of 30 tumor types that we studied, comprising a total of 546 tumors (table 1). some nonepithelial cancers, such as synovial sarcomas, chordomas, neuroblastomas, osteosarcomas, and ependymomas, were also tert-l.  eight tert-h tumor types were identified (in addition to the previously described melanomas) (14, 15). these tumors included tumors of the cns, transitional cell carcinomas of the urinary tract, hepatocellular carcinomas, myxoid liposarcomas, and oral cavity carcinomas. although only a small number of tert-h tumors (other than melanomas) were examined in previous studies (15), mutations in gliomas, hepatocellular, and oral cavity carcinomas were detected, which would be expected on the basis of the high frequency of mutation in these tumors types (table 1).  clinical and molecular correlations in tert-h tumors. sarcomas. one of the highest frequencies of tert promoter mutation was found in myxoid liposarcoma (19 of 24 tumors, 79% with mutation). myxoid liposarcomas account for more than one-third of all liposarcomas and 10% of all adult soft tissue sarcomas (16). patients are relatively young, with a peak age range between 30 and 50 y. at the genetic level, the most characteristic change is a t(12;16) (q13;p11) chromosomal translocation that results in the fusion of the fus and ddit3 genes (16, 17). the cellular origin of these tumors is unknown, but preadipocytic progenitor cells and mesenchymal stem cells have been implicated (18); after embryogenesis, the mitotic activity of these cells is thought to be low. other sarcomas, also thought to originate from mesenchymal cells that do not self-renew in the absence of damage, were not tert-h (table 1). these sarcomas included synovial sarcomas (0% of 16 tumors) and osteosarcomas (% of 23 tumors). of note, myxoid liposarcomas have been previously shown to have a relatively high prevalence of alt (24% of 38 tumors) (13, 19). the data, in aggregate, are compatible with the idea that myxoid liposarcomas almost always genetically activate telomere maintenance genes through either tert promoter mutations or alt.  hepatocellular carcinomas. hepatocellular carcinomas (hccs) are the third leading cause of cancer mortality worldwide, and their incidence is increasing in the united states (20). most hccs in the united states are associated with hepatitis b or c virus infection, whereas others are associated with alcoholic cirrhosis; 44% of hcc samples that we evaluated harbored tert promoter mutations (27/61). this finding makes tert the most commonly mutated gene yet observed in this tumor type (21, 22). the mutations seemed to occur relatively early in tumorigenesis, because they were observed in 39% of stage i well-differentiated hccs (table s1). tert mutations were observed in virally associated tumors as well as cases without any underlying liver disease at similar frequencies (table s1). there was also no difference in the prevalence of tert promoter mutations with respect to sex, age, or ethnicity (table s1). alt has been observed in 7% of 121 hccs studied previously (13).  urinary tract cancers. urothelial carcinoma of the bladder is the fourth most common type of cancer in american males. in 2013, over 73,000 patients will be diagnosed with bladder cancer leading to approximately 15,000 deaths in the us alone (23). two-thirds of the 21 urothelial carcinomas of the bladder that we studied harbored tert promoter mutations. we were also able to evaluate 19 urothelial carcinomas of the upper urinary tract, a much less common anatomic site for this histopathologic subtype of tumor. nine of nineteen upper urinary tract urothelial carcinomas harbored tert mutations. tert mutations are, therefore, the most frequently mutated genes yet identified in urothelial carcinoma of either the bladder or upper urinary tract (24). the prevalence of alt in bladder cancers is very low (1% of 188 cancers) (13).  head and neck cancers. head and neck cancers are almost always squamous cell carcinomas and can occur throughout the oral cavity lining (mucous membranes of the cheek, hard and soft palate, tongue, supraglottis, et). it is the sixth most common cancer in the world, and 50,000 cases occurred in the united states in 2012. we identified tert promoter mutations in 17% of 70 oral cavity cancers that we evaluated. however, the anatomic distribution of the cases with tert promoter mutations was striking: 11 of 12 cancers with tert promoter mutations were in the oral tongue, although only 23 of 70 total cases originated in the oral tongue (p < , fisher exact probability test, two-tailed) (table s2). the basis for this extraordinary selectivity is curious given the shared characteristics of the squamous epithelium lining the tongue and other parts of the head and neck, including the oral cavity. moreover, we evaluated 22 squamous cell carcinomas of another site (the cervix) and found only one tert mutation (%) (table 1). most cervical squamous cell carcinomas and a subset of head and neck squamous cell carcinomas are caused by human papillomavirus, which can activate telomerase by expressing e6 and e7 viral oncogenes (25). these findings raise the possibility that human papillomavirus infection and tert mutation may be alternative mechanisms to activate telomerase among squamous cell carcinomas. we were unable to test correlations between tert promoter mutations and hpv status or other clinical parameters because of the small number of patients with available data (table s2). there have been no alt cases identified among 70 head and neck cancers, including 41 oral cavity cancers (13).  medulloblastomas. medulloblastoma is the most common malignant brain tumor of childhood (26). tert mutations occurred in 21% of 91 medulloblastomas that we evaluated. as with the oral cavity cancers, tert mutations were not distributed randomly among the medulloblastoma patients. although medulloblastomas are usually diagnosed at a young age, those medulloblastomas with tert mutations were diagnosed at a considerably older age (median = 6 vs. 16 y, p = , t test assuming unequal variances, two-tailed) ( s1a). this observation has important implications for understanding the basis for the selectivity of the tumor types harboring tert promoter mutations (discussion); 45 of 90 patients had been assessed previously for orthodenticle homeobox 2 (otx2) gene amplification and expression, and alterations in this transcription factor are known to correlate with clinically distinct molecular subtypes of medulloblastoma (27). otx2 expression was >100-fold higher in medulloblastoma patients without tert promoter mutations than in those patients with tert promoter mutations (note the log scale in  s1b). the high levels of otx2 expression were usually the result of otx2 gene amplification ( s1c). the association of tert promoter mutations with an older age at diagnosis and a lack of otx2 overexpression raises the possibility that tert mutations occur in a specific clinical and molecular subtype of medulloblastoma. the most likely molecular subtype of medulloblastoma that may be enriched for tert mutations is the noninfant sonic hedgehog subtype, which is characterized by an older age at diagnosis and lower expression of otx2 (28, 29). larger studies will be needed to make this association more definitive. alt has been observed in 7% of 55 medulloblastomas studied previously (13).  gliomas. gliomas are the most common cns tumor type and accounted for >14,000 deaths in the united states last year (30). histopathological and clinical criteria established by the world health organization are used to characterize these tumors into several subtypes (30). we considered the four main subtypes individually (table s3).  primary glioblastoma. these primary glioblastomas (gbms) are the most common malignant brain tumors in adults, accounting for 17% of all intracranial tumors, and they confer the worst survival (median of 15 mo) (31). these high-grade (grade iv) tumors have no detectable precursor lesions and have been referred to as de novo tumors. the prevalence of tert promoter mutations was remarkably high in gbms of adults (83% of 78 tumors) (table s3). this prevalence is higher than the prevalence of any other genetic mutation in this tumor type (32). these findings provide a molecular mechanism responsible for the high levels of tert mrna and telomerase activity observed in gbms (33).  for 51 of 78 primary gbm tumors, data on other common genetic alterations as well as clinical data were available ( 2a). interestingly, egfr amplification, a classic molecular feature of primary gbm, exclusively occurred in tumors with tert mutations (p = , fisher exact probability test, two-tailed). conversely, no association was identified between tert mutation and either tp53 mutation or cdkn2a deletion. importantly, the frequency of tert promoter mutations was considerably less in primary gbms of pediatric patients (11% of 19 tumors) than adult patients (discussion) (table s3). alt was observed in 11% of 105 adult gbm and 44% of pediatric gbm (, the reverse of the pattern observed for tert promoter mutations) (13). primary gbm patients without tert mutations survived considerably longer, on average, than patients with such mutations (median = 27 vs. 14 mo, p =  by the log rank test) ( s3).   2.  2. mutations of selected genes in glioma subtypes. (a) distribution of tert mutations and other genetic events in 51 primary gbms. (b) distribution of tert mutations and other genetic events among 40 astrocytomas, including grades iiiii astrocytomas . astrocytomas. infiltrative astrocytic tumors frequently progress, with recurrent lesions often of higher grade than the original lesions excised at surgery. they are most often grade ii or iii but can progress to grade iv (at which point they are often termed secondary gbms). astrocytomas of any stage rarely contained tert promoter mutations (10% of 40 total samples) (table s3). instead, they more frequently contained isocitrate dehydrogenase 1 (idh1) or isocitrate dehydrogenase 2 (idh2) mutations (75% of 40 tumors), atrx mutations (70% of 40 tumors), and tp53 mutations (73% of 40 tumors) ( 2b). alt has been observed in 63% of 57 astrocytomas, consistent with the high prevalence of atrx mutations (13). the lack of activating tert mutations in idh1 mutant tumors is also corroborated by the lack of tert mrna and telomerase activity observed in these lesions (33).  oligodendrogliomas. like astrocytomas, oligodendrogliomas often progress, and they frequently contain tert promoter mutations (78% of 45 tumor samples) (table s3). oligodendroglioma was the only tumor type studied (of all types, including non-cns tumors) (dataset s1) in which c250t mutations were nearly as frequent as c228t mutations. in oligodendrogliomas, 43% of tumors with tert mutations contained c250t substitutions, whereas in other gliomas, only 10% did (p < , fisher exact probability test, two-tailed). interestingly, 91% of 45 oligodendrogliomas that were evaluated for atrx and tert sequence alterations contained either an atrx coding or a tert promoter mutation, suggesting that genetic alterations resulting in telomere maintenance are required for tumorigenesis of this subtype.  oligodendrogliomas have long been known to contain characteristic losses of chromosome arms 1p and 19q, and these losses reflect inactivation of the cic gene on chromosome 19q and in some cases, inactivation of the fubp1 gene on chromosome 1p (3436). accordingly, 78% of 45 oligodendrogliomas contained chromosome arm 1p or 19q losses of heterozygosity ( 2c) (3436). moreover, nearly all of them contained idh1 or idh2 mutations (93%).  oligoastrocytomas. as their name implies, these tumors are mixed, with histologic features of both oligodendrogliomas and astrocytomas. this mixture, in part, reflects the difficulties in distinguishing the various glioma subtypes from one another on the basis of histopathologic or clinical criteria (37). the genetic features of this tumor subtype reflect this mixture: the prevalence of tert promoter mutations (25% of 24 tumors) was intermediate between oligodendrogliomas and astrocytomas, as were the frequencies of chromosome (chr) 1p/19q losses and idh1/2, tp53, and atrx mutations ( 2d).  alt vs. tert. alt has been observed in tumors of the cns (particularly gliomas) more frequently than tumors of any other tissue type. given that tert promoter mutations are also common in gliomas, the relationship between these two features could be determined with high confidence. the tumors depicted in  2 had previously been evaluated for alterations in atrx, which is a nearly perfect surrogate for the alt phenotype (11, 37). our data show that there were 50 gliomas with atrx mutations and 83 gliomas with tert mutations; 0 of 83 tumors with tert mutations contained atrx mutations (p < , fisher exact probability test, two-tailed).  go to: discussion the results described above, as well as the results published in refs. 14 and 15, provide evidence that supports one of the hypotheses raised in the introduction and refutes others. the first of these hypotheses was that tert mutations would only be observed in tumors derived from tissues that are not constantly self-renewing under normal circumstances. this hypothesis was supported in part: the vast majority of tert promoter mutations occurred in tumors derived from tissues that do not continually self-renew. the tert-h tumor types include only melanomas, certain subtypes of glioma, medulloblastomas, squamous cell cancers of the tongue, liposarcomas, hccs, and urinary tract cancers. the normal transitional cells of the urinary tract have very low proliferative indices (%  %), much lower than indices of gastrointestinal tract, bone marrow, or skin (38). normal hepatocytes also do not turnover often (39), and glial cells are thought to have limited capacity for self-renewal (40).  two other observations also support the hypothesis. pediatric primary gbms rarely contained tert mutations (11%), whereas adult primary gbms frequently did (83%). pediatric gbms are presumably derived from cells that are still dividing at the time of tumor initiation, and therefore, there is no selective advantage conferred by activating telomerase through a genetic mutation. adult gbms, in contrast, are presumably derived from postmitotic cells, and they should require telomerase activation. similarly, medulloblastomas are embryonal tumors that typically arise from precursor cells with high self-renewal rates that do not usually persist in adults. this finding is consistent with our observation that the mean age of medulloblastoma patients with tert mutations was considerably older than the mean age of medulloblastoma patients without tert mutations ( s1a).  there are, however, exceptions that belie the hypothesis that tert mutations occur only in nonself-renewing tissues. the epithelium that lines the tongue constantly self-renews, but many squamous carcinomas of the tongue harbored tert mutations (table s2). additionally, the squamous epithelia of the tongue certainly would not be expected to self-renew less than other squamous epithelia of the oral cavity, but the latter rarely harbored tert mutations (table s2). this finding may suggest that squamous carcinomas of the tongue originate from a different cell of origin than other oral cavity squamous carcinomas. conversely, only a subset of the tumor types derived from nonself-renewing tissues was tert-h. for example, the tert-h tumors included myxoid liposarcomas but not synovial sarcomas. moreover, cells of the pancreas (the islets of langerhans and the ductal epithelial cells) rarely renew, but pancreatic tumors of all types (pancreatic neuroendocrine tumors, acinar carcinomas, and pancreatic ductal adenocarcinomas) were all tert-l. the most that we can conclude at present is that nonself-renewing cell types are the major sources of tert-h tumors but that nonself-renewal is only one of the factors that determines whether tumor cells with tert promoter mutations will have a selective growth advantage over adjoining cells.  the first corollary to the hypothesis raised in the introduction was that tumor types that displayed alt would be those types that harbored tert promoter mutations. this corollary is soundly refuted by these data, at least in general terms. although tumor types of the cns and liposarcomas had high frequencies of alt as well as high frequencies of tert promoter mutations, these tumor types were the exceptions rather than the rule. for example, pancreatic neuroendocrine tumors have very high frequencies of alt but no evidence of tert mutations. conversely, bladder cancers frequently have tert mutations but never have alt (13). additionally, even among gliomas, pediatric gbms have high frequencies of alt and low frequencies of tert mutations, whereas adult gbms have the reverse pattern.  the second corollary was that the selective advantage afforded by tert mutation would be equivalent to the advantage afforded by atrx mutation (conferring alt). this hypothesis was most effectively tested in gliomas, in which both atrx coding and tert promoter mutations were common. there was a striking mutual exclusivity with respect to atrx and tert mutations (p < ), lending strong support to this idea.  these results also raise many unanswered questions. in some tumor types, such as gliomas, we can imagine that all tumors have genetically activated telomere maintenance programs through mutations in either tert or atrx. however, in other tumor types with frequent atrx mutations, such as pancreatic neuroendocrine tumors, what is responsible for activating telomerase in the fraction of cases not exhibiting alt if it is not a mutation in the tert promoter? similarly, what is responsible for activating telomerase in those tumors derived from nonself-renewing cell types in which neither alt nor tert mutations is frequently observed, such as synovial sarcomas or osteosarcomas? also, there are occasional individual tumors among the tert-l types that have tert promoter mutations (e.g., cervical cancers, ovarian cancers, and in ref. 15, lung cancers). what distinguishes these occasional cancers from others of the same histopathologic subtype? whole-genome sequencing studies, rather than those studies limited to the exome, might provide answers to these questions.  the results recorded here have practical as well as basic scientific implications. two-thirds of bladder cancers had tert promoter mutations, making it the most commonly mutated gene yet identified in invasive urothelial carcinoma of the bladder. given the persistently high mortality rate despite multimodality treatment in this group of patients, these mutations represent ideal urinary biomarkers to detect bladder cancers at an early stage and to follow patients for evidence of progression or recurrence once they have been diagnosed (41). similarly, the high prevalence of tert promoter mutations in hccs and glioma subtypes provides excellent candidate biomarkers for early detection (hcc) or monitoring (hcc in the plasma and gliomas in the cerebrospinal fluid) (42, 43).  another practical implication involves diagnostics. we conjecture that tumors with tert promoter or atrx mutations are derived from different precursor cells and that either type of precursor cell is different from those types that are the precursors of tumors without such mutations. this distinction could aid classification of the tumors in clinically meaningful ways. for example,  2 and  s2 outline the major genetic alterations occurring in the three most common types of gliomas. on the basis of the data in  2 ac, we speculate that oligodendrogliomas that lack tert mutations but contain atrx mutations may behave more like astrocytomas than oligodendrogliomas and vice versa. similarly, the primary gbms without tert mutations (15% of the total) may behave more like advanced progressive astrocytomas, which generally lack tert mutations. this possibility is supported by the observation that those primary gbm patients without tert mutations had a longer survival, on average, than other primary gbm patients ( s3).  go to: methods all clinical information and tissue were obtained with consent and institutional review board approval from the various institutions donating material to this study, and they were obtained in accordance with the health insurance portability and accountability act. tissue sections were reviewed by board-certified pathologists to ensure that 50% of the cells used for dna purification were neoplastic and confirm histopathological diagnosis. oligonucleotides with the sequences 5-m13-ggccgattcgacctctct-3 and 5-agcacctcgcggtagtgg-3, where m13 is a universal sequencing priming site with sequence 5-tgtaaaacgacggccagt-3, were used to pcr-amplify the proximal tert promoter containing c228 and c250 (chr5: 1,295,228; chr5: 1,295,250, respectively; hg19) for sanger sequencing using standard methods (44). primary gbm copy number data as well as alt status were derived from the data published in refs. 37, 45, and 46, and otx2 copy number expression was derived from the data published in ref. 27. brain tumor patients were treated at the tisch brain tumor center at duke. for the purposes of this study, secondary gbm designates a gbm that was resected >1 y after a prior diagnosis of a lower-grade glioma (grades iiii), and all other gbms were considered to be primary gbms. pediatric gbm samples were defined as those samples occurring before 21 y of age.\""
      ]
     },
     "execution_count": 242,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concatenate_data.loc[ID,\"Text\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 245,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "c228t tert\n"
     ]
    }
   ],
   "source": [
    "print(concatenate_data.loc[ID,\"Variation\"],\n",
    "concatenate_data.loc[ID,\"Gene\"].lower())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 246,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "1\n",
      "First match !  w802*\n",
      "2\n",
      "First match !  q249e\n",
      "3\n",
      "First match !  n454d\n",
      "4\n",
      "First match !  l399v\n",
      "5\n",
      "First match !  v391i\n",
      "6\n",
      "First match !  v430m\n",
      "7\n",
      "First match !  deletion\n",
      "8\n",
      "First match !  y371h\n",
      "9\n",
      "First match !  c384r\n",
      "10\n",
      "First match !  p395a\n",
      "11\n",
      "First match !  k382e\n",
      "12\n",
      "First match !  r420q\n",
      "13\n",
      "First match !  c381a\n",
      "14\n",
      "First match !  p428l\n",
      "15\n",
      "First match !  d390y\n",
      "16\n",
      "17\n",
      "First match !  q367p\n",
      "18\n",
      "First match !  m374v\n",
      "19\n",
      "aa_pos_aa y371s\n",
      "20\n",
      "First match !  h94y\n",
      "21\n",
      "First match !  c396r\n",
      "22\n",
      "First match !  g375p\n",
      "23\n",
      "First match !  s376f\n",
      "24\n",
      "First match !  p417a\n",
      "25\n",
      "First match !  h398y\n",
      "26\n",
      "First match !  s2g\n",
      "27\n",
      "First match !  y846c\n",
      "28\n",
      "First match !  c228t\n",
      "29\n",
      "First match !  h412y\n",
      "30\n",
      "First match !  h876q\n",
      "31\n",
      "First match !  promoter mutations\n",
      "32\n",
      "First match !  p704s\n",
      "33\n",
      "First match !  amplification\n",
      "34\n",
      "First match !  c250t\n",
      "35\n",
      "First match !  g1809r\n",
      "36\n",
      "First match !  g1809k\n",
      "37\n",
      "First match !  d1709e\n",
      "38\n",
      "First match !  d1709a\n",
      "39\n",
      "First match !  e1705a\n",
      "40\n",
      "First match !  d1810a\n",
      "41\n",
      "First match !  truncating mutations\n",
      "42\n",
      "First match !  e1705k\n",
      "43\n",
      "First match !  deletion\n",
      "44\n",
      "pos weak t1365m\n",
      "45\n",
      "First match !  v648g\n",
      "46\n",
      "47\n",
      "First match !  a707t\n",
      "48\n",
      "First match !  promoter hypermethylation\n",
      "49\n",
      "pos weak r1343l\n",
      "50\n",
      "First match !  a209t\n",
      "51\n",
      "First match !  y280h\n",
      "52\n",
      "First match !  d927g\n",
      "53\n",
      "First match !  n510k\n",
      "54\n",
      "First match !  f248s\n",
      "55\n",
      "First match !  l708p\n",
      "56\n",
      "First match !  v995m\n",
      "57\n",
      "First match !  y412f\n",
      "58\n",
      "First match !  f74s\n",
      "59\n",
      "First match !  r1040l\n",
      "60\n",
      "First match !  r453c\n",
      "61\n",
      "62\n",
      "First match !  a1022e\n",
      "63\n",
      "pos weak q984k\n",
      "64\n",
      "First match !  t605m\n",
      "65\n",
      "First match !  k218t\n",
      "66\n",
      "pos weak n1125i\n",
      "67\n",
      "y371 aa_pos y35c\n",
      "68\n",
      "First match !  y35n\n",
      "69\n",
      "y371 aa_pos y35h\n",
      "70\n",
      "First match !  deletion\n",
      "71\n",
      "72\n",
      "73\n",
      "74\n",
      "First match !  deletion\n",
      "75\n",
      "76\n",
      "First match !  p284l\n",
      "77\n",
      "First match !  q276*\n",
      "78\n",
      "First match !  t286a\n",
      "79\n",
      "First match !  t283a\n",
      "80\n",
      "81\n",
      "First match !  i290r\n",
      "82\n",
      "First match !  a77s\n",
      "83\n",
      "First match !  a77p\n",
      "84\n",
      "First match !  a77t\n",
      "85\n",
      "First match !  m90i\n",
      "86\n",
      "First match !  amplification\n",
      "87\n",
      "First match !  overexpression\n",
      "88\n",
      "89\n",
      "First match !  n45s\n",
      "90\n",
      "First match !  r487q\n",
      "91\n",
      "aa_pos_aa c41y\n",
      "92\n",
      "First match !  g52r\n",
      "93\n",
      "aa_pos_aa p83l\n",
      "94\n",
      "First match !  s241l\n",
      "95\n",
      "First match !  s387y\n",
      "96\n",
      "97\n",
      "First match !  r537p\n",
      "98\n",
      "First match !  d404g\n",
      "99\n",
      "First match !  p525l\n",
      "100\n",
      "101\n",
      "First match !  deletion\n",
      "102\n",
      "103\n",
      "First match !  r976h\n",
      "104\n",
      "105\n",
      "First match !  t1219i\n",
      "106\n",
      "pos g39e\n",
      "107\n",
      "First match !  deletion\n",
      "108\n",
      "109\n",
      "First match !  r1095h\n",
      "110\n",
      "111\n",
      "112\n",
      "113\n",
      "114\n",
      "115\n",
      "First match !  deletion\n",
      "116\n",
      "First match !  truncating mutations\n",
      "117\n",
      "118\n",
      "First match !  truncating mutations\n",
      "119\n",
      "First match !  deletion\n",
      "120\n",
      "121\n",
      "First match !  k700r\n",
      "122\n",
      "First match !  r625c\n",
      "123\n",
      "First match !  e622d\n",
      "124\n",
      "First match !  h662q\n",
      "125\n",
      "y371 aa_pos r625d\n",
      "126\n",
      "First match !  k666m\n",
      "127\n",
      "y371 aa_pos h662r\n",
      "128\n",
      "First match !  r625h\n",
      "129\n",
      "First match !  k666t\n",
      "130\n",
      "First match !  r625l\n",
      "131\n",
      "y371 aa_pos e622q\n",
      "132\n",
      "First match !  r625g\n",
      "133\n",
      "First match !  k666n\n",
      "134\n",
      "y371 aa_pos k666r\n",
      "135\n",
      "First match !  k700e\n",
      "136\n",
      "pos weak r957q\n",
      "137\n",
      "First match !  deletion\n",
      "138\n",
      "139\n",
      "140\n",
      "141\n",
      "142\n",
      "First match !  g719a\n",
      "143\n",
      "First match !  g735s\n",
      "144\n",
      "145\n",
      "146\n",
      "147\n",
      "148\n",
      "First match !  g724s\n",
      "149\n",
      "150\n",
      "First match !  egfrviii\n",
      "151\n",
      "pos weak v765a\n",
      "152\n",
      "First match !  egfrvii\n",
      "153\n",
      "First match !  l858m\n",
      "154\n",
      "First match !  h870r\n",
      "155\n",
      "156\n",
      "First match !  h773l\n",
      "157\n",
      "First match !  t710a\n",
      "158\n",
      "First match !  l747p\n",
      "159\n",
      "y371 aa_pos e709v\n",
      "160\n",
      "First match !  d761y\n",
      "161\n",
      "162\n",
      "y371 aa_pos h773y\n",
      "163\n",
      "First match !  v851a\n",
      "164\n",
      "FUSION egfr-rad51 fusion\n",
      "165\n",
      "First match !  d770_n771insnpg\n",
      "166\n",
      "First match !  e746_a750del\n",
      "167\n",
      "First match !  a864t\n",
      "168\n",
      "pos weak e746q\n",
      "169\n",
      "170\n",
      "First match !  egfrvv\n",
      "171\n",
      "172\n",
      "y371 aa_pos e709q\n",
      "173\n",
      "First match !  e804g\n",
      "174\n",
      "175\n",
      "176\n",
      "First match !  g796s\n",
      "177\n",
      "First match !  r836c\n",
      "178\n",
      "179\n",
      "First match !  g857e\n",
      "180\n",
      "First match !  g719s\n",
      "181\n",
      "First match !  l747v\n",
      "182\n",
      "pos weak t783a\n",
      "183\n",
      "First match !  l692f\n",
      "184\n",
      "185\n",
      "pos v726m\n",
      "186\n",
      "First match !  e746v\n",
      "187\n",
      "188\n",
      "First match !  exon 19 deletion\n",
      "189\n",
      "First match !  s768i\n",
      "190\n",
      "First match !  c620y\n",
      "191\n",
      "First match !  s921r\n",
      "192\n",
      "y371 aa_pos n826y\n",
      "193\n",
      "194\n",
      "First match !  h773_v774insh\n",
      "195\n",
      "196\n",
      "First match !  c628y\n",
      "197\n",
      "First match !  exon 20 insertion\n",
      "198\n",
      "First match !  exon 19 insertion\n",
      "199\n",
      "First match !  l704n\n",
      "200\n",
      "First match !  l861q\n",
      "201\n",
      "First match !  l747f\n",
      "202\n",
      "First match !  n826s\n",
      "203\n",
      "First match !  l747_e749del\n",
      "204\n",
      "First match !  amplification\n",
      "205\n",
      "First match !  l858q\n",
      "206\n",
      "207\n",
      "First match !  r108k\n",
      "208\n",
      "pos weak v774a\n",
      "209\n",
      "First match !  s645c\n",
      "210\n",
      "First match !  l858r\n",
      "211\n",
      "First match !  egfr-kdd\n",
      "212\n",
      "213\n",
      "pos weak v774m\n",
      "214\n",
      "215\n",
      "pos weak e746g\n",
      "216\n",
      "First match !  l861p\n",
      "217\n",
      "First match !  k745m\n",
      "218\n",
      "219\n",
      "First match !  l747s\n",
      "220\n",
      "First match !  s492r\n",
      "221\n",
      "First match !  s768_d770dup\n",
      "222\n",
      "223\n",
      "First match !  l861r\n",
      "224\n",
      "First match !  g719d\n",
      "225\n",
      "226\n",
      "First match !  a763_y764insfqea\n",
      "227\n",
      "First match !  s784f\n",
      "228\n",
      "First match !  y801h\n",
      "229\n",
      "230\n",
      "First match !  v742a\n",
      "231\n",
      "232\n",
      "233\n",
      "First match !  t725m\n",
      "234\n",
      "First match !  l703p\n",
      "235\n",
      "First match !  t790m\n",
      "236\n",
      "First match !  a767_v769dup\n",
      "237\n",
      "First match !  e866k\n",
      "238\n",
      "First match !  egfrviv\n",
      "239\n",
      "First match !  s768_v769insvas\n",
      "240\n",
      "241\n",
      "First match !  e734q\n",
      "242\n",
      "First match !  r776h\n",
      "243\n",
      "First match !  t263p\n",
      "244\n",
      "First match !  t785a\n",
      "245\n",
      "First match !  a750p\n",
      "246\n",
      "First match !  l838v\n",
      "247\n",
      "First match !  p753s\n",
      "248\n",
      "249\n",
      "250\n",
      "First match !  i853t\n",
      "251\n",
      "First match !  l833v\n",
      "252\n",
      "First match !  r324l\n",
      "253\n",
      "First match !  v769m\n",
      "254\n",
      "First match !  w731l\n",
      "255\n",
      "First match !  e884k\n",
      "256\n",
      "257\n",
      "First match !  g719c\n",
      "258\n",
      "First match !  e330k\n",
      "259\n",
      "260\n",
      "First match !  r841k\n",
      "261\n",
      "262\n",
      "263\n",
      "First match !  r776c\n",
      "264\n",
      "First match !  e868g\n",
      "265\n",
      "First match !  p848l\n",
      "266\n",
      "y371 aa_pos l861f\n",
      "267\n",
      "268\n",
      "FUSION egfr-purb fusion\n",
      "269\n",
      "First match !  e709g\n",
      "270\n",
      "First match !  e709k\n",
      "271\n",
      "First match !  p596l\n",
      "272\n",
      "First match !  e709a\n",
      "273\n",
      "First match !  d587h\n",
      "274\n",
      "y371 aa_pos g863d\n",
      "275\n",
      "pos weak l838p\n",
      "276\n",
      "277\n",
      "First match !  v843i\n",
      "278\n",
      "First match !  g465e\n",
      "279\n",
      "FUSION igh-nkx2 fusion\n",
      "280\n",
      "FUSION trb-nkx2-1 fusion\n",
      "281\n",
      "First match !  amplification\n",
      "282\n",
      "First match !  fusions\n",
      "283\n",
      "FUSION tra-nkx2-1 fusion\n",
      "284\n",
      "First match !  a339v\n",
      "285\n",
      "286\n",
      "First match !  deletion\n",
      "287\n",
      "288\n",
      "First match !  amplification\n",
      "289\n",
      "First match !  brd4-nut fusion\n",
      "290\n",
      "aa_pos_aa s102f\n",
      "291\n",
      "First match !  deletion\n",
      "292\n",
      "First match !  k373e\n",
      "293\n",
      "First match !  i157t\n",
      "294\n",
      "First match !  e161del\n",
      "295\n",
      "First match !  p85l\n",
      "296\n",
      "First match !  s428f\n",
      "297\n",
      "First match !  r117g\n",
      "298\n",
      "First match !  e321k\n",
      "299\n",
      "First match !  amplification\n",
      "300\n",
      "First match !  tmprss2-etv1 fusion\n",
      "301\n",
      "FUSION tmprss2-etv5 fusion\n",
      "302\n",
      "FUSION tmprss2-etv4 fusion\n",
      "303\n",
      "y371 aa_pos t75m\n",
      "304\n",
      "First match !  fusions\n",
      "305\n",
      "aa_pos_aa m160v\n",
      "306\n",
      "pos g35r\n",
      "307\n",
      "pos k28m\n",
      "308\n",
      "pos g35v\n",
      "309\n",
      "First match !  i279p\n",
      "310\n",
      "First match !  r331p\n",
      "311\n",
      "First match !  s330a\n",
      "312\n",
      "First match !  s308a\n",
      "313\n",
      "314\n",
      "First match !  k320e\n",
      "315\n",
      "First match !  s1986f\n",
      "316\n",
      "First match !  amplification\n",
      "317\n",
      "First match !  d2033n\n",
      "318\n",
      "First match !  s1986y\n",
      "319\n",
      "FUSION ezr-ros1 fusion\n",
      "320\n",
      "First match !  l1947r\n",
      "321\n",
      "FUSION ros1-cd74 fusion\n",
      "322\n",
      "FUSION lima1-ros1 fusion\n",
      "323\n",
      "First match !  v2098i\n",
      "324\n",
      "First match !  g2032r\n",
      "325\n",
      "First match !  g1971e\n",
      "326\n",
      "FUSION kdelr2-ros1 fusion\n",
      "327\n",
      "First match !  slc34a2-ros1 fusion\n",
      "328\n",
      "First match !  l1951r\n",
      "329\n",
      "FUSION msn-ros1 fusion\n",
      "330\n",
      "First match !  fusions\n",
      "331\n",
      "FUSION lrig3-ros1 fusion\n",
      "332\n",
      "FUSION sdc4-ros1 fusion\n",
      "333\n",
      "First match !  tfg-ros1 fusion\n",
      "334\n",
      "First match !  g2101a\n",
      "335\n",
      "FUSION cep85l-ros1 fusion\n",
      "336\n",
      "First match !  cd74-ros1 fusion\n",
      "337\n",
      "First match !  c2060g\n",
      "338\n",
      "FUSION ccdc6-ros1 fusion\n",
      "339\n",
      "First match !  e1935g\n",
      "340\n",
      "FUSION fig-ros1 fusion\n",
      "341\n",
      "342\n",
      "First match !  r732q\n",
      "343\n",
      "First match !  deletion\n",
      "344\n",
      "345\n",
      "First match !  d254n\n",
      "346\n",
      "347\n",
      "First match !  a298t\n",
      "348\n",
      "349\n",
      "First match !  d257n\n",
      "350\n",
      "351\n",
      "352\n",
      "First match !  fusions\n",
      "353\n",
      "aa_pos_aa y1414c\n",
      "354\n",
      "355\n",
      "356\n",
      "FUSION ep300-moz fusion\n",
      "357\n",
      "First match !  d1399y\n",
      "358\n",
      "FUSION ep300-mll fusion\n",
      "359\n",
      "First match !  deletion\n",
      "360\n",
      "aa_pos_aa p1502l\n",
      "361\n",
      "362\n",
      "aa_pos_aa p531l\n",
      "363\n",
      "First match !  a530v\n",
      "364\n",
      "aa_pos_aa p531s\n",
      "365\n",
      "First match !  a530t\n",
      "366\n",
      "First match !  y532c\n",
      "367\n",
      "First match !  p531a\n",
      "368\n",
      "369\n",
      "First match !  s215g\n",
      "370\n",
      "y371 aa_pos r273l\n",
      "371\n",
      "pos r202s\n",
      "372\n",
      "First match !  h179q\n",
      "373\n",
      "First match !  k120e\n",
      "374\n",
      "First match !  c277q\n",
      "375\n",
      "First match !  r290h\n",
      "376\n",
      "First match !  r337p\n",
      "377\n",
      "First match !  p278s\n",
      "378\n",
      "First match !  y234h\n",
      "379\n",
      "First match !  c277w\n",
      "380\n",
      "pos s241t\n",
      "381\n",
      "First match !  v173e\n",
      "382\n",
      "First match !  q331r\n",
      "383\n",
      "First match !  h115n\n",
      "384\n",
      "pos p219s\n",
      "385\n",
      "First match !  r249w\n",
      "386\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pos v274f\n",
      "387\n",
      "First match !  r175c\n",
      "388\n",
      "First match !  r248w\n",
      "389\n",
      "First match !  s121a\n",
      "390\n",
      "First match !  y220s\n",
      "391\n",
      "First match !  p153h\n",
      "392\n",
      "First match !  r181c\n",
      "393\n",
      "First match !  r337c\n",
      "394\n",
      "First match !  k120n\n",
      "395\n",
      "First match !  g245d\n",
      "396\n",
      "First match !  r175h\n",
      "397\n",
      "First match !  r181l\n",
      "398\n",
      "First match !  g244s\n",
      "399\n",
      "pos s240r\n",
      "400\n",
      "First match !  r273c\n",
      "401\n",
      "First match !  e221q\n",
      "402\n",
      "First match !  q331h\n",
      "403\n",
      "First match !  y220c\n",
      "404\n",
      "First match !  r342p\n",
      "405\n",
      "First match !  c242f\n",
      "406\n",
      "First match !  c242s\n",
      "407\n",
      "y371 aa_pos c135s\n",
      "408\n",
      "First match !  m237k\n",
      "409\n",
      "aa_pos v197l\n",
      "410\n",
      "First match !  s121c\n",
      "411\n",
      "First match !  s241f\n",
      "412\n",
      "413\n",
      "First match !  r248k\n",
      "414\n",
      "First match !  h214r\n",
      "415\n",
      "First match !  l330r\n",
      "416\n",
      "pos weak p151s\n",
      "417\n",
      "First match !  truncating mutations\n",
      "418\n",
      "First match !  r110l\n",
      "419\n",
      "First match !  c176f\n",
      "420\n",
      "First match !  h179y\n",
      "421\n",
      "First match !  c277r\n",
      "422\n",
      "First match !  l344r\n",
      "423\n",
      "First match !  r267q\n",
      "424\n",
      "First match !  r156h\n",
      "425\n",
      "First match !  d186a\n",
      "426\n",
      "First match !  c275s\n",
      "427\n",
      "First match !  e285k\n",
      "428\n",
      "429\n",
      "First match !  v157d\n",
      "430\n",
      "pos y236s\n",
      "431\n",
      "First match !  r337h\n",
      "432\n",
      "First match !  r213q\n",
      "433\n",
      "First match !  r280k\n",
      "434\n",
      "First match !  r110p\n",
      "435\n",
      "First match !  r249s\n",
      "436\n",
      "First match !  h168n\n",
      "437\n",
      "First match !  r290g\n",
      "438\n",
      "First match !  h193p\n",
      "439\n",
      "First match !  n239s\n",
      "440\n",
      "First match !  l348s\n",
      "441\n",
      "pos weak m133t\n",
      "442\n",
      "First match !  h179l\n",
      "443\n",
      "First match !  g244r\n",
      "444\n",
      "First match !  p316l\n",
      "445\n",
      "First match !  h297n\n",
      "446\n",
      "First match !  v272l\n",
      "447\n",
      "First match !  g244d\n",
      "448\n",
      "First match !  g245s\n",
      "449\n",
      "First match !  l348f\n",
      "450\n",
      "First match !  y163c\n",
      "451\n",
      "First match !  h179n\n",
      "452\n",
      "First match !  n235d\n",
      "453\n",
      "First match !  h168r\n",
      "454\n",
      "First match !  e285v\n",
      "455\n",
      "First match !  i255f\n",
      "456\n",
      "First match !  r158h\n",
      "457\n",
      "First match !  a161t\n",
      "458\n",
      "First match !  p278a\n",
      "459\n",
      "First match !  r249m\n",
      "460\n",
      "First match !  f341c\n",
      "461\n",
      "First match !  k292t\n",
      "462\n",
      "First match !  g266r\n",
      "463\n",
      "First match !  h179r\n",
      "464\n",
      "pos c141y\n",
      "465\n",
      "First match !  v143a\n",
      "466\n",
      "pos weak e326l\n",
      "467\n",
      "First match !  c229r\n",
      "468\n",
      "First match !  k291q\n",
      "469\n",
      "First match !  q144r\n",
      "470\n",
      "First match !  r248q\n",
      "471\n",
      "First match !  p250l\n",
      "472\n",
      "First match !  a347t\n",
      "473\n",
      "First match !  v157f\n",
      "474\n",
      "pos i251s\n",
      "475\n",
      "First match !  f134y\n",
      "476\n",
      "First match !  p152l\n",
      "477\n",
      "pos g199r\n",
      "478\n",
      "First match !  i162m\n",
      "479\n",
      "First match !  c135r\n",
      "480\n",
      "481\n",
      "First match !  a161s\n",
      "482\n",
      "First match !  g334r\n",
      "483\n",
      "First match !  v173l\n",
      "484\n",
      "First match !  r158l\n",
      "485\n",
      "First match !  h193n\n",
      "486\n",
      "First match !  r175l\n",
      "487\n",
      "pos p47s\n",
      "488\n",
      "First match !  r280a\n",
      "489\n",
      "First match !  r280t\n",
      "490\n",
      "First match !  k292i\n",
      "491\n",
      "First match !  y236c\n",
      "492\n",
      "y371 aa_pos r248l\n",
      "493\n",
      "First match !  r156c\n",
      "494\n",
      "pos y236d\n",
      "495\n",
      "pos t150i\n",
      "496\n",
      "First match !  l344p\n",
      "497\n",
      "First match !  h233n\n",
      "498\n",
      "First match !  p278l\n",
      "499\n",
      "First match !  k291e\n",
      "500\n",
      "First match !  e258v\n",
      "501\n",
      "pos weak s241y\n",
      "502\n",
      "First match !  c238f\n",
      "503\n",
      "First match !  f212y\n",
      "504\n",
      "First match !  r282w\n",
      "505\n",
      "First match !  r267p\n",
      "506\n",
      "First match !  h214q\n",
      "507\n",
      "First match !  t123a\n",
      "508\n",
      "First match !  s215c\n",
      "509\n",
      "First match !  r282q\n",
      "510\n",
      "First match !  r273h\n",
      "511\n",
      "First match !  g245a\n",
      "512\n",
      "First match !  c135y\n",
      "513\n",
      "First match !  r283q\n",
      "514\n",
      "First match !  r181h\n",
      "515\n",
      "First match !  p151h\n",
      "516\n",
      "First match !  f328v\n",
      "517\n",
      "First match !  y234c\n",
      "518\n",
      "First match !  p151a\n",
      "519\n",
      "First match !  r273g\n",
      "520\n",
      "First match !  v197e\n",
      "521\n",
      "pos e286k\n",
      "522\n",
      "First match !  k120m\n",
      "523\n",
      "First match !  i195t\n",
      "524\n",
      "First match !  g356a\n",
      "525\n",
      "First match !  r158c\n",
      "526\n",
      "First match !  h214n\n",
      "527\n",
      "First match !  c238s\n",
      "528\n",
      "pos weak l145r\n",
      "529\n",
      "First match !  p278r\n",
      "530\n",
      "First match !  r337l\n",
      "531\n",
      "First match !  g266e\n",
      "532\n",
      "533\n",
      "First match !  deletion\n",
      "534\n",
      "First match !  r321q\n",
      "535\n",
      "First match !  p305l\n",
      "536\n",
      "First match !  f346v\n",
      "537\n",
      "First match !  v319d\n",
      "538\n",
      "First match !  w368h\n",
      "539\n",
      "First match !  deletion\n",
      "540\n",
      "First match !  d450e\n",
      "541\n",
      "First match !  y406h\n",
      "542\n",
      "First match !  k420a\n",
      "543\n",
      "First match !  t413n\n",
      "544\n",
      "First match !  s276l\n",
      "545\n",
      "First match !  d300h\n",
      "546\n",
      "First match !  k375a\n",
      "547\n",
      "First match !  n387p\n",
      "548\n",
      "First match !  y426a\n",
      "549\n",
      "First match !  r428a\n",
      "550\n",
      "First match !  d300n\n",
      "551\n",
      "First match !  r310h\n",
      "552\n",
      "First match !  d450h\n",
      "553\n",
      "First match !  y298a\n",
      "554\n",
      "First match !  k341a\n",
      "555\n",
      "First match !  v356r\n",
      "556\n",
      "First match !  d258n\n",
      "557\n",
      "First match !  r373h\n",
      "558\n",
      "First match !  r287a\n",
      "559\n",
      "First match !  deletion\n",
      "560\n",
      "First match !  y297a\n",
      "561\n",
      "First match !  y238a\n",
      "562\n",
      "First match !  r243a\n",
      "563\n",
      "First match !  r373q\n",
      "564\n",
      "First match !  r292a\n",
      "565\n",
      "First match !  e239a\n",
      "566\n",
      "First match !  d408y\n",
      "567\n",
      "First match !  y384a\n",
      "568\n",
      "First match !  w406a\n",
      "569\n",
      "First match !  d408h\n",
      "570\n",
      "First match !  r268a\n",
      "571\n",
      "First match !  d408e\n",
      "572\n",
      "First match !  v277d\n",
      "573\n",
      "First match !  s425c\n",
      "574\n",
      "First match !  y237a\n",
      "575\n",
      "First match !  p336s\n",
      "576\n",
      "First match !  d537e\n",
      "577\n",
      "First match !  w509r\n",
      "578\n",
      "First match !  k507q\n",
      "579\n",
      "First match !  d537y\n",
      "580\n",
      "First match !  g508s\n",
      "581\n",
      "582\n",
      "First match !  r441p\n",
      "583\n",
      "First match !  r361c\n",
      "584\n",
      "First match !  l43v\n",
      "585\n",
      "First match !  truncating mutations\n",
      "586\n",
      "First match !  r497h\n",
      "587\n",
      "First match !  r515g\n",
      "588\n",
      "First match !  w398v\n",
      "589\n",
      "First match !  g419v\n",
      "590\n",
      "First match !  r420h\n",
      "591\n",
      "First match !  y513a\n",
      "592\n",
      "First match !  v465m\n",
      "593\n",
      "594\n",
      "First match !  v370d\n",
      "595\n",
      "First match !  r100t\n",
      "596\n",
      "First match !  p130s\n",
      "597\n",
      "First match !  deletion\n",
      "598\n",
      "First match !  k507a\n",
      "599\n",
      "First match !  e330g\n",
      "600\n",
      "First match !  k428a\n",
      "601\n",
      "First match !  r496h\n",
      "602\n",
      "First match !  e330k\n",
      "603\n",
      "First match !  c324y\n",
      "604\n",
      "605\n",
      "First match !  d351h\n",
      "606\n",
      "pos weak a532h\n",
      "607\n",
      "pos weak d357y\n",
      "608\n",
      "First match !  h132y\n",
      "609\n",
      "First match !  amplification\n",
      "610\n",
      "First match !  r24c\n",
      "611\n",
      "First match !  k22a\n",
      "612\n",
      "First match !  overexpression\n",
      "613\n",
      "First match !  a197t\n",
      "614\n",
      "First match !  amplification\n",
      "615\n",
      "First match !  r465h\n",
      "616\n",
      "617\n",
      "First match !  r479l\n",
      "618\n",
      "First match !  s462y\n",
      "619\n",
      "First match !  s227a\n",
      "620\n",
      "pos d520n\n",
      "621\n",
      "First match !  r479q\n",
      "622\n",
      "aa_pos r479h\n",
      "623\n",
      "First match !  r658q\n",
      "624\n",
      "First match !  r505c\n",
      "625\n",
      "First match !  r689q\n",
      "626\n",
      "First match !  k11r\n",
      "627\n",
      "First match !  t205a\n",
      "628\n",
      "pos r505l\n",
      "629\n",
      "First match !  deletion\n",
      "630\n",
      "First match !  i347m\n",
      "631\n",
      "First match !  copy number loss\n",
      "632\n",
      "pos weak g423r\n",
      "633\n",
      "First match !  r482q\n",
      "634\n",
      "First match !  amplification\n",
      "635\n",
      "First match !  deletion\n",
      "636\n",
      "637\n",
      "First match !  s31r\n",
      "638\n",
      "First match !  deletion\n",
      "639\n",
      "First match !  p133t\n",
      "640\n",
      "First match !  v109g\n",
      "641\n",
      "642\n",
      "First match !  q50p\n",
      "643\n",
      "First match !  p114l\n",
      "644\n",
      "First match !  e69g\n",
      "645\n",
      "y371 aa_pos g67r\n",
      "646\n",
      "First match !  l97r\n",
      "647\n",
      "First match !  d84g\n",
      "648\n",
      "649\n",
      "First match !  v126d\n",
      "650\n",
      "651\n",
      "First match !  d84h\n",
      "652\n",
      "First match !  m53i\n",
      "653\n",
      "y371 aa_pos t77p\n",
      "654\n",
      "First match !  n71i\n",
      "655\n",
      "First match !  g67s\n",
      "656\n",
      "First match !  n71s\n",
      "657\n",
      "y371 aa_pos r80p\n",
      "658\n",
      "First match !  d84v\n",
      "659\n",
      "First match !  d74n\n",
      "660\n",
      "y371 aa_pos d74y\n",
      "661\n",
      "First match !  p81l\n",
      "662\n",
      "First match !  l32p\n",
      "663\n",
      "y371 aa_pos d108h\n",
      "664\n",
      "First match !  deletion\n",
      "665\n",
      "y371 aa_pos g35r\n",
      "666\n",
      "First match !  d84y\n",
      "667\n",
      "First match !  n71k\n",
      "668\n",
      "First match !  r24p\n",
      "669\n",
      "First match !  r99p\n",
      "670\n",
      "First match !  a148t\n",
      "671\n",
      "First match !  g23d\n",
      "672\n",
      "First match !  r80l\n",
      "673\n",
      "First match !  g35a\n",
      "674\n",
      "First match !  h83y\n",
      "675\n",
      "First match !  a36p\n",
      "676\n",
      "First match !  p48l\n",
      "677\n",
      "aa_pos_aa d108n\n",
      "678\n",
      "First match !  p81t\n",
      "679\n",
      "First match !  g101w\n",
      "680\n",
      "First match !  a57v\n",
      "681\n",
      "First match !  r87l\n",
      "682\n",
      "First match !  h83n\n",
      "683\n",
      "First match !  d84n\n",
      "684\n",
      "First match !  p114s\n",
      "685\n",
      "First match !  i49s\n",
      "686\n",
      "First match !  r112g\n",
      "687\n",
      "First match !  s56i\n",
      "688\n",
      "689\n",
      "First match !  a60v\n",
      "690\n",
      "First match !  r87p\n",
      "691\n",
      "First match !  e88k\n",
      "692\n",
      "693\n",
      "694\n",
      "695\n",
      "First match !  deletion\n",
      "696\n",
      "First match !  hypermethylation\n",
      "697\n",
      "First match !  e35*\n",
      "698\n",
      "First match !  p40t\n",
      "699\n",
      "First match !  d86n\n",
      "700\n",
      "First match !  a23e\n",
      "701\n",
      "First match !  deletion\n",
      "702\n",
      "First match !  deletion\n",
      "703\n",
      "First match !  e812k\n",
      "704\n",
      "pos l49h\n",
      "705\n",
      "First match !  r678q\n",
      "706\n",
      "First match !  t798m\n",
      "707\n",
      "First match !  d808n\n",
      "708\n",
      "y371 aa_pos v659e\n",
      "709\n",
      "First match !  s760a\n",
      "710\n",
      "711\n",
      "First match !  l866m\n",
      "712\n",
      "First match !  c630y\n",
      "713\n",
      "First match !  i767m\n",
      "714\n",
      "First match !  s1002r\n",
      "715\n",
      "First match !  e719g\n",
      "716\n",
      "First match !  k753e\n",
      "717\n",
      "First match !  e719k\n",
      "718\n",
      "719\n",
      "First match !  g776s\n",
      "720\n",
      "First match !  s310f\n",
      "721\n",
      "First match !  s335c\n",
      "722\n",
      "First match !  l785f\n",
      "723\n",
      "pos weak l915m\n",
      "724\n",
      "First match !  c334s\n",
      "725\n",
      "726\n",
      "First match !  g660d\n",
      "727\n",
      "First match !  v777a\n",
      "728\n",
      "First match !  k753m\n",
      "729\n",
      "First match !  v842i\n",
      "730\n",
      "pos weak v750e\n",
      "731\n",
      "pos weak n319d\n",
      "732\n",
      "733\n",
      "First match !  v773a\n",
      "734\n",
      "First match !  e717k\n",
      "735\n",
      "First match !  k753a\n",
      "736\n",
      "First match !  t733i\n",
      "737\n",
      "First match !  y835f\n",
      "738\n",
      "First match !  s653c\n",
      "739\n",
      "740\n",
      "First match !  d769h\n",
      "741\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "First match !  l726i\n",
      "742\n",
      "First match !  h878y\n",
      "743\n",
      "First match !  exon 20 insertions\n",
      "744\n",
      "First match !  amplification\n",
      "745\n",
      "First match !  l755s\n",
      "746\n",
      "First match !  d821n\n",
      "747\n",
      "First match !  g309a\n",
      "748\n",
      "aa_pos_aa e265k\n",
      "749\n",
      "First match !  d769y\n",
      "750\n",
      "First match !  t798i\n",
      "751\n",
      "First match !  l755p\n",
      "752\n",
      "y371 aa_pos y1248f\n",
      "753\n",
      "First match !  p780l\n",
      "754\n",
      "First match !  v777l\n",
      "755\n",
      "First match !  s783p\n",
      "756\n",
      "First match !  d769a\n",
      "757\n",
      "First match !  c311r\n",
      "758\n",
      "759\n",
      "First match !  d277h\n",
      "760\n",
      "First match !  r143q\n",
      "761\n",
      "First match !  g309e\n",
      "762\n",
      "First match !  e321g\n",
      "763\n",
      "First match !  l726f\n",
      "764\n",
      "First match !  v777m\n",
      "765\n",
      "First match !  y803n\n",
      "766\n",
      "First match !  n857s\n",
      "767\n",
      "First match !  d845a\n",
      "768\n",
      "769\n",
      "First match !  s310y\n",
      "770\n",
      "First match !  v794m\n",
      "771\n",
      "772\n",
      "First match !  s846i\n",
      "773\n",
      "First match !  a1131t\n",
      "774\n",
      "First match !  g284r\n",
      "775\n",
      "First match !  p262h\n",
      "776\n",
      "First match !  v714m\n",
      "777\n",
      "First match !  t389k\n",
      "778\n",
      "First match !  amplification\n",
      "779\n",
      "First match !  q809r\n",
      "780\n",
      "First match !  r669c\n",
      "781\n",
      "First match !  a232v\n",
      "782\n",
      "First match !  v104m\n",
      "783\n",
      "First match !  e452k\n",
      "784\n",
      "First match !  ezr-erbb4 fusion\n",
      "785\n",
      "First match !  r306s\n",
      "786\n",
      "First match !  e563k\n",
      "787\n",
      "First match !  v348l\n",
      "788\n",
      "First match !  t244r\n",
      "789\n",
      "First match !  y285c\n",
      "790\n",
      "First match !  r544w\n",
      "791\n",
      "First match !  m887i\n",
      "792\n",
      "First match !  d595v\n",
      "793\n",
      "First match !  fusions\n",
      "794\n",
      "First match !  n181s\n",
      "795\n",
      "First match !  r838q\n",
      "796\n",
      "First match !  g785s\n",
      "797\n",
      "First match !  g936r\n",
      "798\n",
      "First match !  e317k\n",
      "799\n",
      "First match !  e542k\n",
      "800\n",
      "First match !  e872k\n",
      "801\n",
      "First match !  k935i\n",
      "802\n",
      "First match !  e836k\n",
      "803\n",
      "804\n",
      "pos weak e606g\n",
      "805\n",
      "806\n",
      "807\n",
      "808\n",
      "809\n",
      "aa_pos_aa r631c\n",
      "810\n",
      "y371 aa_pos d609e\n",
      "811\n",
      "First match !  d513y\n",
      "812\n",
      "aa_pos_aa v536m\n",
      "813\n",
      "aa_pos_aa r487w\n",
      "814\n",
      "815\n",
      "aa_pos_aa l461v\n",
      "816\n",
      "pos r592h\n",
      "817\n",
      "pos y24c\n",
      "818\n",
      "aa_pos_aa d423n\n",
      "819\n",
      "820\n",
      "First match !  v242f\n",
      "821\n",
      "aa_pos_aa r601q\n",
      "822\n",
      "First match !  a349p\n",
      "823\n",
      "First match !  p47a\n",
      "824\n",
      "First match !  m299i\n",
      "825\n",
      "First match !  r173c\n",
      "826\n",
      "First match !  truncating mutations\n",
      "827\n",
      "First match !  deletion\n",
      "828\n",
      "First match !  k52r\n",
      "829\n",
      "First match !  r109*\n",
      "830\n",
      "831\n",
      "aa_pos_aa r689s\n",
      "832\n",
      "First match !  deletion\n",
      "833\n",
      "834\n",
      "aa_pos_aa l230p\n",
      "835\n",
      "836\n",
      "aa_pos_aa s786f\n",
      "837\n",
      "FUSION bcr-abl1 fusion\n",
      "838\n",
      "First match !  l248v\n",
      "839\n",
      "First match !  g250e\n",
      "840\n",
      "First match !  m244v\n",
      "841\n",
      "First match !  fusions\n",
      "842\n",
      "First match !  e355a\n",
      "843\n",
      "First match !  l387m\n",
      "844\n",
      "First match !  e281k\n",
      "845\n",
      "First match !  e459k\n",
      "846\n",
      "First match !  f311l\n",
      "847\n",
      "First match !  y253f\n",
      "848\n",
      "First match !  t315i\n",
      "849\n",
      "First match !  e279k\n",
      "850\n",
      "First match !  e255v\n",
      "851\n",
      "First match !  e255k\n",
      "852\n",
      "First match !  h396p\n",
      "853\n",
      "First match !  y253h\n",
      "854\n",
      "First match !  v299l\n",
      "855\n",
      "First match !  l384m\n",
      "856\n",
      "First match !  v379i\n",
      "857\n",
      "First match !  m351t\n",
      "858\n",
      "First match !  v289f\n",
      "859\n",
      "First match !  q252h\n",
      "860\n",
      "First match !  f317l\n",
      "861\n",
      "First match !  h396r\n",
      "862\n",
      "First match !  f359c\n",
      "863\n",
      "864\n",
      "First match !  tmprss2-erg fusion\n",
      "865\n",
      "FUSION ewsr1-erg fusion\n",
      "866\n",
      "First match !  fus-erg fusion\n",
      "867\n",
      "First match !  fusions\n",
      "868\n",
      "869\n",
      "870\n",
      "871\n",
      "872\n",
      "First match !  a633t\n",
      "873\n",
      "First match !  r481g\n",
      "874\n",
      "First match !  h845y\n",
      "875\n",
      "876\n",
      "FUSION kdr-pdgfra fusion\n",
      "877\n",
      "First match !  n870s\n",
      "878\n",
      "First match !  q579r\n",
      "879\n",
      "First match !  r748g\n",
      "880\n",
      "First match !  r841k\n",
      "881\n",
      "882\n",
      "First match !  n659k\n",
      "883\n",
      "First match !  h570r\n",
      "884\n",
      "First match !  y555c\n",
      "885\n",
      "886\n",
      "First match !  e996k\n",
      "887\n",
      "First match !  v536e\n",
      "888\n",
      "First match !  s478p\n",
      "889\n",
      "890\n",
      "First match !  h650q\n",
      "891\n",
      "892\n",
      "y371 aa_pos a633v\n",
      "893\n",
      "894\n",
      "First match !  i562m\n",
      "895\n",
      "896\n",
      "First match !  v561a\n",
      "897\n",
      "898\n",
      "pos weak k830r\n",
      "899\n",
      "900\n",
      "901\n",
      "First match !  d842i\n",
      "902\n",
      "y371 aa_pos n659r\n",
      "903\n",
      "First match !  kif5b-pdgfra fusion\n",
      "904\n",
      "First match !  y849s\n",
      "905\n",
      "pos weak i834v\n",
      "906\n",
      "907\n",
      "First match !  g853d\n",
      "908\n",
      "First match !  d842v\n",
      "909\n",
      "First match !  t674i\n",
      "910\n",
      "First match !  p577s\n",
      "911\n",
      "912\n",
      "First match !  amplification\n",
      "913\n",
      "First match !  w349c\n",
      "914\n",
      "First match !  d846y\n",
      "915\n",
      "First match !  n848k\n",
      "916\n",
      "917\n",
      "First match !  l507p\n",
      "918\n",
      "First match !  n659s\n",
      "919\n",
      "First match !  v561d\n",
      "920\n",
      "921\n",
      "922\n",
      "First match !  strn-pdgfra fusion\n",
      "923\n",
      "First match !  d842y\n",
      "924\n",
      "First match !  y849c\n",
      "925\n",
      "First match !  bcr-pdgfra fusion\n",
      "926\n",
      "927\n",
      "First match !  etv6-pdgfra fusion\n",
      "928\n",
      "First match !  v658a\n",
      "929\n",
      "930\n",
      "First match !  f808l\n",
      "931\n",
      "FUSION kank1-pdgfrb fusion\n",
      "932\n",
      "First match !  r987w\n",
      "933\n",
      "FUSION nin-pdgfrb fusion\n",
      "934\n",
      "First match !  gpiap1-pdgfrb fusion\n",
      "935\n",
      "First match !  golga4-pdgfrb fusion\n",
      "936\n",
      "FUSION wdr48-pdgfrb fusion\n",
      "937\n",
      "First match !  tp53bp1-pdgfrb fusion\n",
      "938\n",
      "First match !  fusions\n",
      "939\n",
      "First match !  prkg2-pdgfrb fusion\n",
      "940\n",
      "First match !  ebf1-pdgfrb fusion\n",
      "941\n",
      "FUSION atf7ip-pdgfrb fusion\n",
      "942\n",
      "First match !  amplification\n",
      "943\n",
      "y371 aa_pos e1071w\n",
      "944\n",
      "First match !  pde4dip-pdgfrb fusion\n",
      "945\n",
      "First match !  bin2-pdgfrb fusion\n",
      "946\n",
      "aa_pos_aa r561c\n",
      "947\n",
      "FUSION cep85l-pdgfrb fusion\n",
      "948\n",
      "First match !  git2-pdgfrb fusion\n",
      "949\n",
      "FUSION sptbn1-pdgfrb fusion\n",
      "950\n",
      "FUSION hip1-pdgfrb fusion\n",
      "951\n",
      "FUSION kiaa1509-pdgfrb fusion\n",
      "952\n",
      "FUSION etv6-pdgfrb fusion\n",
      "953\n",
      "aa_pos_aa v665a\n",
      "954\n",
      "First match !  truncating mutations\n",
      "955\n",
      "First match !  deletion\n",
      "956\n",
      "First match !  v354e\n",
      "957\n",
      "First match !  d402y\n",
      "958\n",
      "First match !  p480l\n",
      "959\n",
      "960\n",
      "First match !  overexpression\n",
      "961\n",
      "First match !  m1_e165del\n",
      "962\n",
      "First match !  fusions\n",
      "963\n",
      "First match !  v534e\n",
      "964\n",
      "First match !  l536p\n",
      "965\n",
      "First match !  l469v\n",
      "966\n",
      "aa_pos_aa s463p\n",
      "967\n",
      "First match !  e380q\n",
      "968\n",
      "969\n",
      "FUSION esr1-yap1 fusion\n",
      "970\n",
      "aa_pos_aa l536q\n",
      "971\n",
      "First match !  l536h\n",
      "972\n",
      "First match !  v422del\n",
      "973\n",
      "First match !  esr1-ccdc170 fusion\n",
      "974\n",
      "pos weak l536r\n",
      "975\n",
      "First match !  s432l\n",
      "976\n",
      "First match !  y537c\n",
      "977\n",
      "First match !  y537s\n",
      "978\n",
      "First match !  tmprss2-etv1 fusion\n",
      "979\n",
      "First match !  fusions\n",
      "980\n",
      "FUSION ewsr1-etv1 fusion\n",
      "981\n",
      "First match !  r418g\n",
      "982\n",
      "First match !  r369q\n",
      "983\n",
      "First match !  fusions\n",
      "984\n",
      "First match !  p214l\n",
      "985\n",
      "First match !  r369w\n",
      "986\n",
      "First match !  385_418del\n",
      "987\n",
      "FUSION etv6-flt3 fusion\n",
      "988\n",
      "First match !  etv6-ntrk3 fusion\n",
      "989\n",
      "First match !  m224r\n",
      "990\n",
      "aa_pos_aa f158c\n",
      "991\n",
      "First match !  k82t\n",
      "992\n",
      "y371 aa_pos s35q\n",
      "993\n",
      "First match !  l117p\n",
      "994\n",
      "First match !  h68r\n",
      "995\n",
      "First match !  r246k\n",
      "996\n",
      "First match !  t417i\n",
      "997\n",
      "First match !  g305w\n",
      "998\n",
      "999\n",
      "1000\n",
      "aa_pos_aa h105r\n",
      "1001\n",
      "1002\n",
      "First match !  g305r\n",
      "1003\n",
      "1004\n",
      "aa_pos_aa h206d\n",
      "1005\n",
      "First match !  g1035s\n",
      "1006\n",
      "First match !  l191h\n",
      "1007\n",
      "First match !  l180p\n",
      "1008\n",
      "First match !  r1097h\n",
      "1009\n",
      "First match !  deletion\n",
      "1010\n",
      "First match !  s1036p\n",
      "1011\n",
      "First match !  g1567d\n",
      "1012\n",
      "First match !  l410r\n",
      "1013\n",
      "First match !  v299g\n",
      "1014\n",
      "First match !  h1620r\n",
      "1015\n",
      "1016\n",
      "First match !  l1584r\n",
      "1017\n",
      "First match !  q1554h\n",
      "1018\n",
      "First match !  t1623i\n",
      "1019\n",
      "First match !  h597y\n",
      "1020\n",
      "First match !  e1552del\n",
      "1021\n",
      "First match !  t1203k\n",
      "1022\n",
      "First match !  v1673f\n",
      "1023\n",
      "First match !  v705m\n",
      "1024\n",
      "First match !  p1675l\n",
      "1025\n",
      "First match !  g1596v\n",
      "1026\n",
      "First match !  r905w\n",
      "1027\n",
      "First match !  r905q\n",
      "1028\n",
      "First match !  v1673d\n",
      "1029\n",
      "First match !  r611w\n",
      "1030\n",
      "First match !  l830r\n",
      "1031\n",
      "First match !  s1653p\n",
      "1032\n",
      "First match !  r905g\n",
      "1033\n",
      "First match !  s1653f\n",
      "1034\n",
      "First match !  v769e\n",
      "1035\n",
      "First match !  q1503p\n",
      "1036\n",
      "First match !  deletion\n",
      "1037\n",
      "First match !  a889p\n",
      "1038\n",
      "First match !  w1610g\n",
      "1039\n",
      "First match !  r462c\n",
      "1040\n",
      "First match !  r1200w\n",
      "1041\n",
      "First match !  l792r\n",
      "1042\n",
      "First match !  l493v\n",
      "1043\n",
      "First match !  l493p\n",
      "1044\n",
      "First match !  c696y\n",
      "1045\n",
      "First match !  l826p\n",
      "1046\n",
      "First match !  a614d\n",
      "1047\n",
      "First match !  l146r\n",
      "1048\n",
      "First match !  r611q\n",
      "1049\n",
      "First match !  s1498n\n",
      "1050\n",
      "First match !  f615s\n",
      "1051\n",
      "First match !  y598c\n",
      "1052\n",
      "First match !  v705e\n",
      "1053\n",
      "First match !  l844r\n",
      "1054\n",
      "First match !  l448p\n",
      "1055\n",
      "First match !  p1709l\n",
      "1056\n",
      "First match !  e75g\n",
      "1057\n",
      "1058\n",
      "FUSION ewsr1-yy1 fusion\n",
      "1059\n",
      "First match !  ewsr1-ddit3 fusion\n",
      "1060\n",
      "1061\n",
      "First match !  fusions\n",
      "1062\n",
      "FUSION ewsr1-fli1 fusion\n",
      "1063\n",
      "FUSION ewsr1-nr4a3 fusion\n",
      "1064\n",
      "FUSION ewsr1-creb1 fusion\n",
      "1065\n",
      "FUSION ewsr1-etv4 fusion\n",
      "1066\n",
      "FUSION ewsr1-atf1 fusion\n",
      "1067\n",
      "FUSION ewsr1-wt1 fusion\n",
      "1068\n",
      "First match !  r206h\n",
      "1069\n",
      "First match !  g328e\n",
      "1070\n",
      "First match !  g328v\n",
      "1071\n",
      "pos weak g87r\n",
      "1072\n",
      "1073\n",
      "First match !  deletion\n",
      "1074\n",
      "First match !  s453fs*\n",
      "1075\n",
      "pos weak l455m\n",
      "1076\n",
      "First match !  f400i\n",
      "1077\n",
      "First match !  amplification\n",
      "1078\n",
      "First match !  m253k\n",
      "1079\n",
      "First match !  amplification\n",
      "1080\n",
      "1081\n",
      "pos weak y646n\n",
      "1082\n",
      "First match !  y646f\n",
      "1083\n",
      "pos weak y646h\n",
      "1084\n",
      "pos weak y646c\n",
      "1085\n",
      "pos weak y646s\n",
      "1086\n",
      "First match !  a677g\n",
      "1087\n",
      "First match !  overexpression\n",
      "1088\n",
      "First match !  amplification\n",
      "1089\n",
      "First match !  amplification\n",
      "1090\n",
      "First match !  k409q\n",
      "1091\n",
      "1092\n",
      "1093\n",
      "First match !  t1481fs\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1094\n",
      "1095\n",
      "First match !  e1286v\n",
      "1096\n",
      "First match !  t779fs\n",
      "1097\n",
      "First match !  truncating mutations\n",
      "1098\n",
      "First match !  deletion\n",
      "1099\n",
      "First match !  a95d\n",
      "1100\n",
      "First match !  e685v\n",
      "1101\n",
      "First match !  c91s\n",
      "1102\n",
      "First match !  truncating mutations\n",
      "1103\n",
      "First match !  f170i\n",
      "1104\n",
      "First match !  i47f\n",
      "1105\n",
      "First match !  c91a\n",
      "1106\n",
      "First match !  f81v\n",
      "1107\n",
      "First match !  deletion\n",
      "1108\n",
      "First match !  s858r\n",
      "1110\n",
      "First match !  truncating mutations\n",
      "1111\n",
      "First match !  h492r\n",
      "1112\n",
      "First match !  y510c\n",
      "1113\n",
      "First match !  deletion\n",
      "1114\n",
      "First match !  l274p\n",
      "1115\n",
      "First match !  w183a\n",
      "1116\n",
      "First match !  l210r\n",
      "1117\n",
      "1118\n",
      "1119\n",
      "First match !  r970c\n",
      "1120\n",
      "y371 aa_pos y1003c\n",
      "1121\n",
      "1122\n",
      "First match !  f1200i\n",
      "1123\n",
      "First match !  n375s\n",
      "1124\n",
      "aa_pos v1092i\n",
      "1125\n",
      "First match !  y1230c\n",
      "1126\n",
      "First match !  y1003f\n",
      "1127\n",
      "First match !  amplification\n",
      "1128\n",
      "1129\n",
      "1130\n",
      "First match !  d1010h\n",
      "1131\n",
      "1132\n",
      "1133\n",
      "First match !  overexpression\n",
      "1134\n",
      "First match !  y1003*\n",
      "1135\n",
      "First match !  y1235d\n",
      "1136\n",
      "1137\n",
      "First match !  d1010y\n",
      "1138\n",
      "1139\n",
      "1140\n",
      "1141\n",
      "1142\n",
      "pos weak m1250t\n",
      "1143\n",
      "First match !  ptprz1-met fusion\n",
      "1144\n",
      "1145\n",
      "First match !  y1230h\n",
      "1146\n",
      "First match !  e168d\n",
      "1147\n",
      "1148\n",
      "1149\n",
      "1150\n",
      "First match !  t992i\n",
      "1151\n",
      "First match !  s34f\n",
      "1152\n",
      "First match !  q157p\n",
      "1153\n",
      "First match !  s34y\n",
      "1154\n",
      "First match !  s3660l\n",
      "1155\n",
      "First match !  a1685s\n",
      "1156\n",
      "1157\n",
      "First match !  deletion\n",
      "1158\n",
      "First match !  deletion\n",
      "1159\n",
      "First match !  a4419s\n",
      "1160\n",
      "aa_pos_aa p4309a\n",
      "1161\n",
      "1162\n",
      "First match !  deletion\n",
      "1163\n",
      "1164\n",
      "First match !  r38h\n",
      "1165\n",
      "First match !  i31m\n",
      "1166\n",
      "First match !  g106d\n",
      "1167\n",
      "First match !  k111e\n",
      "1168\n",
      "First match !  r88q\n",
      "1169\n",
      "First match !  r108h\n",
      "1170\n",
      "First match !  m1043i\n",
      "1171\n",
      "First match !  e542g\n",
      "1172\n",
      "First match !  g106v\n",
      "1173\n",
      "First match !  d350g\n",
      "1174\n",
      "First match !  e545q\n",
      "1175\n",
      "First match !  v344a\n",
      "1176\n",
      "pos e78k\n",
      "1177\n",
      "1178\n",
      "First match !  r93q\n",
      "1179\n",
      "First match !  n1044k\n",
      "1180\n",
      "First match !  e116k\n",
      "1181\n",
      "First match !  v344g\n",
      "1182\n",
      "First match !  t1025s\n",
      "1183\n",
      "First match !  c420r\n",
      "1184\n",
      "First match !  k111n\n",
      "1185\n",
      "First match !  q60k\n",
      "1186\n",
      "First match !  n345i\n",
      "1187\n",
      "First match !  e545k\n",
      "1188\n",
      "First match !  e365k\n",
      "1189\n",
      "First match !  e81k\n",
      "1190\n",
      "First match !  q546k\n",
      "1191\n",
      "First match !  q546p\n",
      "1192\n",
      "First match !  e542k\n",
      "1193\n",
      "First match !  h701p\n",
      "1194\n",
      "First match !  r93w\n",
      "1195\n",
      "First match !  h1047r\n",
      "1196\n",
      "First match !  q546e\n",
      "1197\n",
      "First match !  a1066v\n",
      "1198\n",
      "First match !  amplification\n",
      "1199\n",
      "1200\n",
      "First match !  h1047l\n",
      "1201\n",
      "First match !  p449t\n",
      "1202\n",
      "First match !  p539r\n",
      "1203\n",
      "First match !  a1020v\n",
      "1204\n",
      "aa_pos_aa g914r\n",
      "1205\n",
      "y371 aa_pos e542q\n",
      "1206\n",
      "y371 aa_pos q546r\n",
      "1207\n",
      "First match !  e579k\n",
      "1208\n",
      "First match !  d1029y\n",
      "1209\n",
      "First match !  r115l\n",
      "1210\n",
      "1211\n",
      "First match !  g118d\n",
      "1212\n",
      "First match !  e545a\n",
      "1213\n",
      "First match !  h1047y\n",
      "1214\n",
      "First match !  e453a\n",
      "1215\n",
      "First match !  c378r\n",
      "1216\n",
      "First match !  e545g\n",
      "1217\n",
      "First match !  i391m\n",
      "1218\n",
      "First match !  e542v\n",
      "1219\n",
      "First match !  m1043v\n",
      "1220\n",
      "pos weak n553s\n",
      "1221\n",
      "First match !  e552k\n",
      "1222\n",
      "FUSION acpp-pik3cb fusion\n",
      "1223\n",
      "pos weak r48w\n",
      "1224\n",
      "First match !  fusions\n",
      "1225\n",
      "First match !  d1067v\n",
      "1226\n",
      "First match !  d1067y\n",
      "1227\n",
      "First match !  e633k\n",
      "1228\n",
      "1229\n",
      "First match !  d1067a\n",
      "1230\n",
      "First match !  l193f\n",
      "1231\n",
      "First match !  n82k\n",
      "1232\n",
      "First match !  h68y\n",
      "1233\n",
      "First match !  e124q\n",
      "1234\n",
      "First match !  e142d\n",
      "1235\n",
      "First match !  e135k\n",
      "1236\n",
      "First match !  y53h\n",
      "1237\n",
      "First match !  p123m\n",
      "1238\n",
      "First match !  e525k\n",
      "1239\n",
      "First match !  n334k\n",
      "1240\n",
      "First match !  e1021k\n",
      "1241\n",
      "First match !  fusions\n",
      "1242\n",
      "First match !  amplification\n",
      "1243\n",
      "First match !  yap1-mamld1 fusion\n",
      "1244\n",
      "First match !  yap1-fam118b fusion\n",
      "1245\n",
      "First match !  t576del\n",
      "1246\n",
      "First match !  g376r\n",
      "1247\n",
      "1248\n",
      "First match !  l370fs\n",
      "1249\n",
      "First match !  e218*\n",
      "1250\n",
      "First match !  deletion\n",
      "1251\n",
      "1252\n",
      "First match !  n564d\n",
      "1253\n",
      "First match !  w237_y242del\n",
      "1254\n",
      "First match !  e137k\n",
      "1255\n",
      "First match !  e160*\n",
      "1256\n",
      "1257\n",
      "1258\n",
      "First match !  d560y\n",
      "1259\n",
      "First match !  k379e\n",
      "1260\n",
      "First match !  r574fs\n",
      "1261\n",
      "1262\n",
      "1263\n",
      "First match !  k288q\n",
      "1264\n",
      "First match !  r162*\n",
      "1265\n",
      "First match !  r348*\n",
      "1266\n",
      "First match !  r262t\n",
      "1267\n",
      "First match !  n564k\n",
      "1268\n",
      "First match !  e439del\n",
      "1269\n",
      "aa_pos_aa g373r\n",
      "1270\n",
      "1271\n",
      "First match !  amplification\n",
      "1272\n",
      "First match !  n561d\n",
      "1273\n",
      "First match !  d557h\n",
      "1274\n",
      "First match !  a727v\n",
      "1275\n",
      "First match !  a171v\n",
      "1276\n",
      "1278\n",
      "First match !  g12v\n",
      "1279\n",
      "First match !  g13d\n",
      "1280\n",
      "First match !  amplification\n",
      "1281\n",
      "First match !  a146t\n",
      "1282\n",
      "First match !  k117r\n",
      "1283\n",
      "First match !  q61h\n",
      "1284\n",
      "y371 aa_pos g13r\n",
      "1285\n",
      "First match !  g13v\n",
      "1286\n",
      "First match !  a146v\n",
      "1287\n",
      "First match !  q61l\n",
      "1288\n",
      "First match !  q61k\n",
      "1289\n",
      "First match !  k117n\n",
      "1290\n",
      "First match !  t58i\n",
      "1291\n",
      "First match !  f28l\n",
      "1292\n",
      "First match !  g13c\n",
      "1293\n",
      "First match !  g12a\n",
      "1294\n",
      "First match !  q22k\n",
      "1295\n",
      "First match !  g12s\n",
      "1296\n",
      "First match !  y64a\n",
      "1297\n",
      "pos p34r\n",
      "1298\n",
      "First match !  g12c\n",
      "1299\n",
      "pos a59t\n",
      "1300\n",
      "1301\n",
      "First match !  r265s\n",
      "1302\n",
      "First match !  r100*\n",
      "1303\n",
      "aa_pos_aa m587v\n",
      "1304\n",
      "First match !  a111p\n",
      "1305\n",
      "1306\n",
      "First match !  r265c\n",
      "1307\n",
      "1308\n",
      "First match !  g101s\n",
      "1309\n",
      "1310\n",
      "pos weak e172k\n",
      "1311\n",
      "pos v506a\n",
      "1312\n",
      "1313\n",
      "1314\n",
      "pos weak r755s\n",
      "1315\n",
      "First match !  t117m\n",
      "1316\n",
      "pos g67r\n",
      "1317\n",
      "1318\n",
      "First match !  r217c\n",
      "1319\n",
      "First match !  i219v\n",
      "1320\n",
      "pos weak r725c\n",
      "1321\n",
      "pos weak p648s\n",
      "1322\n",
      "pos weak p648l\n",
      "1323\n",
      "1324\n",
      "pos weak a120s\n",
      "1325\n",
      "pos v384d\n",
      "1326\n",
      "First match !  r487*\n",
      "1327\n",
      "First match !  g67w\n",
      "1328\n",
      "pos weak l622h\n",
      "1329\n",
      "pos weak l582f\n",
      "1330\n",
      "First match !  k618t\n",
      "1331\n",
      "1332\n",
      "1333\n",
      "pos weak l749p\n",
      "1334\n",
      "First match !  t82a\n",
      "1335\n",
      "First match !  amplification\n",
      "1336\n",
      "First match !  amplification\n",
      "1337\n",
      "1338\n",
      "1339\n",
      "First match !  fusions\n",
      "1340\n",
      "First match !  e267g\n",
      "1341\n",
      "First match !  l321a\n",
      "1342\n",
      "First match !  q59e\n",
      "1343\n",
      "First match !  e40k\n",
      "1344\n",
      "First match !  k179m\n",
      "1345\n",
      "First match !  l362r\n",
      "1346\n",
      "First match !  v270a\n",
      "1347\n",
      "y371 aa_pos r370c\n",
      "1348\n",
      "First match !  l52r\n",
      "1349\n",
      "First match !  e49k\n",
      "1350\n",
      "First match !  e17k\n",
      "1351\n",
      "First match !  q79e\n",
      "1352\n",
      "First match !  g311d\n",
      "1353\n",
      "First match !  l78t\n",
      "1354\n",
      "First match !  q79k\n",
      "1355\n",
      "First match !  v271a\n",
      "1356\n",
      "First match !  p42t\n",
      "1357\n",
      "First match !  k39n\n",
      "1358\n",
      "First match !  w80a\n",
      "1359\n",
      "First match !  d32y\n",
      "1360\n",
      "First match !  t195i\n",
      "1361\n",
      "First match !  d325a\n",
      "1362\n",
      "First match !  k189n\n",
      "1363\n",
      "First match !  n53a\n",
      "1364\n",
      "First match !  d323h\n",
      "1365\n",
      "First match !  r23a\n",
      "1366\n",
      "First match !  l202f\n",
      "1367\n",
      "First match !  c77f\n",
      "1368\n",
      "First match !  d32h\n",
      "1369\n",
      "First match !  k181m\n",
      "1370\n",
      "First match !  i289m\n",
      "1371\n",
      "First match !  r170w\n",
      "1372\n",
      "First match !  r368c\n",
      "1373\n",
      "First match !  g161v\n",
      "1374\n",
      "First match !  fusions\n",
      "1375\n",
      "y371 aa_pos h355m\n",
      "1376\n",
      "First match !  amplification\n",
      "1377\n",
      "First match !  e17k\n",
      "1378\n",
      "First match !  d399n\n",
      "1379\n",
      "FUSION fgfr1op1-fgfr1 fusion\n",
      "1380\n",
      "First match !  a263v\n",
      "1381\n",
      "First match !  cep110-fgfr1 fusion\n",
      "1382\n",
      "First match !  znf198-fgfr1 fusion\n",
      "1383\n",
      "First match !  e664k\n",
      "1384\n",
      "First match !  amplification\n",
      "1385\n",
      "FUSION fgfr1-tacc1 fusion\n",
      "1386\n",
      "First match !  k656e\n",
      "1387\n",
      "First match !  fusions\n",
      "1388\n",
      "First match !  bcr-fgfr1 fusion\n",
      "1389\n",
      "FUSION erlin2?fgfr1 fusion\n",
      "1390\n",
      "First match !  n546k\n",
      "1391\n",
      "First match !  cux1-fgfr1 fusion\n",
      "1392\n",
      "First match !  g382d\n",
      "1393\n",
      "pos a500t\n",
      "1394\n",
      "First match !  k650e\n",
      "1395\n",
      "First match !  d617g\n",
      "1396\n",
      "First match !  k650t\n",
      "1397\n",
      "First match !  d641n\n",
      "1398\n",
      "FUSION igh-fgfr3 fusion\n",
      "1399\n",
      "First match !  n540s\n",
      "1400\n",
      "First match !  d641g\n",
      "1401\n",
      "pos weak y647c\n",
      "1402\n",
      "First match !  k650q\n",
      "1403\n",
      "First match !  s371c\n",
      "1404\n",
      "pos weak e627d\n",
      "1405\n",
      "First match !  k650r\n",
      "1406\n",
      "First match !  r248c\n",
      "1408\n",
      "pos weak v677i\n",
      "1409\n",
      "First match !  g637w\n",
      "1410\n",
      "FUSION fgfr3 - baiap2l1 fusion\n",
      "1411\n",
      "pos weak d646y\n",
      "1412\n",
      "1413\n",
      "pos weak h643d\n",
      "1414\n",
      "First match !  r669g\n",
      "1415\n",
      "First match !  i538v\n",
      "1416\n",
      "First match !  e466k\n",
      "1417\n",
      "First match !  f384l\n",
      "1418\n",
      "First match !  amplification\n",
      "1419\n",
      "First match !  g375c\n",
      "1420\n",
      "First match !  g697c\n",
      "1421\n",
      "First match !  v555m\n",
      "1422\n",
      "First match !  y373c\n",
      "1423\n",
      "First match !  s249c\n",
      "1424\n",
      "First match !  a391e\n",
      "1425\n",
      "First match !  k650m\n",
      "1426\n",
      "First match !  k650n\n",
      "1427\n",
      "First match !  g370c\n",
      "1428\n",
      "pos n653h\n",
      "1429\n",
      "First match !  fusions\n",
      "1430\n",
      "First match !  g380r\n",
      "1431\n",
      "First match !  f133v\n",
      "1432\n",
      "First match !  m117v\n",
      "1433\n",
      "First match !  r121q\n",
      "1434\n",
      "First match !  d140g\n",
      "1435\n",
      "First match !  y87c\n",
      "1436\n",
      "First match !  p94a\n",
      "1437\n",
      "First match !  w131a\n",
      "1438\n",
      "First match !  w131g\n",
      "1439\n",
      "First match !  s119n\n",
      "1440\n",
      "First match !  s80r\n",
      "1441\n",
      "First match !  g75r\n",
      "1442\n",
      "First match !  k129e\n",
      "1443\n",
      "First match !  e50k\n",
      "1444\n",
      "First match !  f133l\n",
      "1445\n",
      "First match !  y87n\n",
      "1446\n",
      "First match !  d130a\n",
      "1447\n",
      "First match !  n296i\n",
      "1448\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "First match !  f102c\n",
      "1449\n",
      "First match !  m117i\n",
      "1450\n",
      "1451\n",
      "First match !  r251q\n",
      "1452\n",
      "First match !  k517r\n",
      "1453\n",
      "First match !  e731k\n",
      "1454\n",
      "First match !  y375c\n",
      "1455\n",
      "First match !  k659n\n",
      "1456\n",
      "First match !  fusions\n",
      "1457\n",
      "First match !  t730s\n",
      "1458\n",
      "First match !  fgfr2-mgea5 fusion\n",
      "1459\n",
      "First match !  v755i\n",
      "1460\n",
      "First match !  g271e\n",
      "1461\n",
      "First match !  l770v\n",
      "1462\n",
      "First match !  r203c\n",
      "1463\n",
      "pos e565g\n",
      "1464\n",
      "First match !  fgfr2-ahcyl1 fusion\n",
      "1465\n",
      "First match !  a648t\n",
      "1466\n",
      "pos weak n549s\n",
      "1467\n",
      "First match !  p253r\n",
      "1468\n",
      "First match !  g701s\n",
      "1469\n",
      "First match !  t341p\n",
      "1470\n",
      "FUSION fgfr2?pphln1 fusion\n",
      "1471\n",
      "First match !  n549k\n",
      "1472\n",
      "First match !  s252w\n",
      "1473\n",
      "First match !  fgfr2-bicc1 fusion\n",
      "1474\n",
      "First match !  v248d\n",
      "1475\n",
      "First match !  k526e\n",
      "1476\n",
      "First match !  l617f\n",
      "1477\n",
      "First match !  i642v\n",
      "1478\n",
      "pos weak m535i\n",
      "1479\n",
      "pos weak k525e\n",
      "1480\n",
      "First match !  e636k\n",
      "1481\n",
      "FUSION fgfr2-tacc3 fusion\n",
      "1482\n",
      "FUSION fgfr2-kiaa1967 fusion\n",
      "1483\n",
      "1484\n",
      "First match !  c278f\n",
      "1485\n",
      "First match !  a389t\n",
      "1486\n",
      "pos weak l617m\n",
      "1487\n",
      "1488\n",
      "First match !  c382r\n",
      "1489\n",
      "pos weak k641n\n",
      "1490\n",
      "First match !  fgfr2-fam76a fusion\n",
      "1491\n",
      "First match !  fgfr2-ccdc6 fusion\n",
      "1492\n",
      "First match !  e219k\n",
      "1493\n",
      "First match !  k641r\n",
      "1494\n",
      "First match !  k310r\n",
      "1495\n",
      "First match !  d101y\n",
      "1496\n",
      "First match !  n549t\n",
      "1497\n",
      "First match !  m391r\n",
      "1498\n",
      "pos weak v564i\n",
      "1499\n",
      "pos weak k659e\n",
      "1500\n",
      "First match !  e475k\n",
      "1501\n",
      "pos n535k\n",
      "1502\n",
      "First match !  v550e\n",
      "1503\n",
      "First match !  amplification\n",
      "1504\n",
      "First match !  deletion\n",
      "1505\n",
      "First match !  truncating mutations\n",
      "1506\n",
      "First match !  g1128a\n",
      "1507\n",
      "First match !  g1128s\n",
      "1508\n",
      "First match !  l1122v\n",
      "1509\n",
      "First match !  t1151m\n",
      "1510\n",
      "First match !  npm-alk fusion\n",
      "1511\n",
      "First match !  g1123d\n",
      "1512\n",
      "First match !  d1203n\n",
      "1513\n",
      "First match !  ranbp2-alk fusion\n",
      "1514\n",
      "First match !  t1087i\n",
      "1515\n",
      "1516\n",
      "First match !  e1384k\n",
      "1517\n",
      "1518\n",
      "First match !  i1171n\n",
      "1519\n",
      "First match !  y1278s\n",
      "1520\n",
      "First match !  g1202r\n",
      "1521\n",
      "First match !  f1174i\n",
      "1522\n",
      "First match !  g881d\n",
      "1523\n",
      "First match !  s1206c\n",
      "1524\n",
      "First match !  i1171t\n",
      "1525\n",
      "First match !  cad-alk fusion\n",
      "1526\n",
      "First match !  l1152r\n",
      "1527\n",
      "First match !  c1156f\n",
      "1528\n",
      "First match !  g1123s\n",
      "1529\n",
      "First match !  s1206r\n",
      "1530\n",
      "First match !  f1174l\n",
      "1531\n",
      "First match !  s1206y\n",
      "1532\n",
      "First match !  strn-alk fusion\n",
      "1533\n",
      "FUSION ranbp1-alk fusion\n",
      "1534\n",
      "First match !  f1245c\n",
      "1535\n",
      "1536\n",
      "1537\n",
      "First match !  i1250t\n",
      "1538\n",
      "First match !  amplification\n",
      "1539\n",
      "First match !  t1343i\n",
      "1540\n",
      "1541\n",
      "First match !  l1198f\n",
      "1542\n",
      "First match !  f1245v\n",
      "1543\n",
      "First match !  g1201e\n",
      "1544\n",
      "First match !  v1180l\n",
      "1545\n",
      "First match !  d1091n\n",
      "1546\n",
      "First match !  kif5b-alk fusion\n",
      "1547\n",
      "First match !  l1196m\n",
      "1548\n",
      "First match !  p1139s\n",
      "1549\n",
      "1550\n",
      "First match !  g1269s\n",
      "1551\n",
      "First match !  g1269a\n",
      "1552\n",
      "First match !  l1196q\n",
      "1553\n",
      "First match !  a1099t\n",
      "1554\n",
      "y371 aa_pos g1286r\n",
      "1555\n",
      "pos weak r1060h\n",
      "1556\n",
      "First match !  d1270g\n",
      "1557\n",
      "y371 aa_pos e1210k\n",
      "1558\n",
      "First match !  n1178h\n",
      "1559\n",
      "First match !  k1062m\n",
      "1560\n",
      "First match !  fusions\n",
      "1561\n",
      "First match !  r1192p\n",
      "1562\n",
      "pos weak r1231q\n",
      "1563\n",
      "First match !  v597a\n",
      "1564\n",
      "First match !  d1349h\n",
      "1565\n",
      "First match !  r1464*\n",
      "1566\n",
      "First match !  i1183t\n",
      "1567\n",
      "First match !  l1198p\n",
      "1568\n",
      "First match !  c1156y\n",
      "1569\n",
      "First match !  l1152p\n",
      "1570\n",
      "First match !  h694r\n",
      "1571\n",
      "First match !  i1170n\n",
      "1572\n",
      "aa_pos r1275l\n",
      "1573\n",
      "First match !  i1170s\n",
      "1574\n",
      "First match !  a1234t\n",
      "1575\n",
      "First match !  s100f\n",
      "1576\n",
      "First match !  g325r\n",
      "1577\n",
      "1578\n",
      "First match !  amplification\n",
      "1579\n",
      "aa_pos_aa s505n\n",
      "1580\n",
      "First match !  w515k\n",
      "1581\n",
      "First match !  w515l\n",
      "1582\n",
      "First match !  p106l\n",
      "1583\n",
      "1584\n",
      "y371 aa_pos s217c\n",
      "1585\n",
      "y371 aa_pos s217t\n",
      "1586\n",
      "First match !  s217a\n",
      "1587\n",
      "First match !  t24a\n",
      "1588\n",
      "First match !  p86h\n",
      "1589\n",
      "First match !  d121g\n",
      "1590\n",
      "First match !  r64p\n",
      "1591\n",
      "y371 aa_pos y98h\n",
      "1592\n",
      "First match !  s68w\n",
      "1593\n",
      "pos v84l\n",
      "1594\n",
      "First match !  l128f\n",
      "1595\n",
      "First match !  truncating mutations\n",
      "1596\n",
      "First match !  v155a\n",
      "1597\n",
      "First match !  g114r\n",
      "1598\n",
      "y371 aa_pos h115r\n",
      "1599\n",
      "First match !  l118p\n",
      "1600\n",
      "First match !  r200w\n",
      "1601\n",
      "First match !  y98n\n",
      "1602\n",
      "First match !  q164r\n",
      "1603\n",
      "First match !  a149p\n",
      "1604\n",
      "First match !  l188q\n",
      "1605\n",
      "First match !  r69c\n",
      "1606\n",
      "First match !  y112h\n",
      "1607\n",
      "First match !  i151s\n",
      "1608\n",
      "First match !  l158p\n",
      "1609\n",
      "First match !  n78s\n",
      "1610\n",
      "First match !  f119s\n",
      "1611\n",
      "First match !  s65w\n",
      "1612\n",
      "First match !  q145h\n",
      "1613\n",
      "1614\n",
      "First match !  h191d\n",
      "1615\n",
      "pos weak l188v\n",
      "1616\n",
      "First match !  l118r\n",
      "1617\n",
      "First match !  w117c\n",
      "1618\n",
      "First match !  l63p\n",
      "1619\n",
      "First match !  e70k\n",
      "1620\n",
      "First match !  p25l\n",
      "1621\n",
      "First match !  r167w\n",
      "1622\n",
      "First match !  r82p\n",
      "1623\n",
      "First match !  s65l\n",
      "1624\n",
      "y371 aa_pos l158v\n",
      "1625\n",
      "First match !  s65a\n",
      "1626\n",
      "First match !  q96p\n",
      "1627\n",
      "First match !  deletion\n",
      "1628\n",
      "First match !  r167q\n",
      "1629\n",
      "First match !  v321m\n",
      "1630\n",
      "pos weak r154w\n",
      "1631\n",
      "pos weak r228k\n",
      "1632\n",
      "First match !  p326l\n",
      "1633\n",
      "1634\n",
      "y371 aa_pos n234i\n",
      "1635\n",
      "First match !  r134q\n",
      "1636\n",
      "pos g85r\n",
      "1637\n",
      "First match !  r304*\n",
      "1638\n",
      "aa_pos q72l\n",
      "1640\n",
      "1641\n",
      "First match !  d835n\n",
      "1642\n",
      "First match !  n676k\n",
      "1643\n",
      "1644\n",
      "First match !  d835a\n",
      "1645\n",
      "First match !  f590g\n",
      "1646\n",
      "First match !  t167a\n",
      "1647\n",
      "First match !  d324n\n",
      "1648\n",
      "First match !  y591d\n",
      "1649\n",
      "First match !  f594l\n",
      "1650\n",
      "First match !  v557i\n",
      "1651\n",
      "First match !  i867s\n",
      "1652\n",
      "First match !  d835del\n",
      "1653\n",
      "1654\n",
      "1655\n",
      "1656\n",
      "1657\n",
      "First match !  n676d\n",
      "1658\n",
      "1659\n",
      "pos a627t\n",
      "1660\n",
      "First match !  g831e\n",
      "1661\n",
      "First match !  r834q\n",
      "1662\n",
      "First match !  n676s\n",
      "1663\n",
      "First match !  m664i\n",
      "1664\n",
      "First match !  f691l\n",
      "1665\n",
      "First match !  m737i\n",
      "1666\n",
      "First match !  v194m\n",
      "1667\n",
      "First match !  v592a\n",
      "1668\n",
      "First match !  s451f\n",
      "1669\n",
      "y371 aa_pos s451e\n",
      "1670\n",
      "First match !  d835h\n",
      "1671\n",
      "First match !  y842c\n",
      "1672\n",
      "First match !  d835e\n",
      "1673\n",
      "First match !  y364h\n",
      "1674\n",
      "1675\n",
      "1676\n",
      "1677\n",
      "1678\n",
      "First match !  v579a\n",
      "1679\n",
      "1680\n",
      "First match !  n841i\n",
      "1681\n",
      "First match !  y572c\n",
      "1682\n",
      "1683\n",
      "First match !  d835y\n",
      "1684\n",
      "1685\n",
      "First match !  d839g\n",
      "1686\n",
      "First match !  amplification\n",
      "1687\n",
      "1688\n",
      "First match !  y599f\n",
      "1689\n",
      "First match !  g857a\n",
      "1690\n",
      "y371 aa_pos e5k\n",
      "1691\n",
      "First match !  e41a\n",
      "1692\n",
      "pos r20q\n",
      "1693\n",
      "First match !  g207e\n",
      "1694\n",
      "1695\n",
      "First match !  r802*\n",
      "1696\n",
      "1697\n",
      "First match !  i668v\n",
      "1698\n",
      "pos i18v\n",
      "1699\n",
      "First match !  s46i\n",
      "1700\n",
      "1701\n",
      "First match !  s46n\n",
      "1702\n",
      "First match !  t485k\n",
      "1703\n",
      "First match !  e541k\n",
      "1704\n",
      "1705\n",
      "First match !  p95h\n",
      "1706\n",
      "First match !  e1099k\n",
      "1707\n",
      "1708\n",
      "1709\n",
      "First match !  s297f\n",
      "1710\n",
      "First match !  p286r\n",
      "1711\n",
      "First match !  s459f\n",
      "1712\n",
      "pos weak s279y\n",
      "1713\n",
      "First match !  f367s\n",
      "1714\n",
      "First match !  v411l\n",
      "1715\n",
      "y371 aa_pos p286h\n",
      "1716\n",
      "pos weak l424i\n",
      "1717\n",
      "First match !  l424v\n",
      "1718\n",
      "aa_pos_aa s24f\n",
      "1719\n",
      "aa_pos_aa r11k\n",
      "1720\n",
      "y371 aa_pos a40e\n",
      "1721\n",
      "aa_pos_aa p26s\n",
      "1722\n",
      "1723\n",
      "First match !  i1307k\n",
      "1724\n",
      "First match !  n1026s\n",
      "1725\n",
      "First match !  deletion\n",
      "1726\n",
      "First match !  r640g\n",
      "1727\n",
      "First match !  a290t\n",
      "1728\n",
      "First match !  amplification\n",
      "1729\n",
      "1730\n",
      "1731\n",
      "First match !  deletion\n",
      "1732\n",
      "1733\n",
      "pos weak l390f\n",
      "1734\n",
      "First match !  d603n\n",
      "1735\n",
      "1736\n",
      "1737\n",
      "First match !  a636p\n",
      "1738\n",
      "First match !  a272v\n",
      "1739\n",
      "pos weak d603g\n",
      "1740\n",
      "aa_pos_aa g751r\n",
      "1741\n",
      "1742\n",
      "1743\n",
      "1744\n",
      "First match !  s860l\n",
      "1745\n",
      "First match !  truncating mutations\n",
      "1746\n",
      "First match !  s723f\n",
      "1747\n",
      "1748\n",
      "1749\n",
      "First match !  r389*\n",
      "1750\n",
      "First match !  y139d\n",
      "1751\n",
      "First match !  r132c\n",
      "1752\n",
      "First match !  r100a\n",
      "1753\n",
      "First match !  g123r\n",
      "1754\n",
      "First match !  r132h\n",
      "1755\n",
      "First match !  r132g\n",
      "1756\n",
      "First match !  r132q\n",
      "1757\n",
      "First match !  p33s\n",
      "1758\n",
      "First match !  a134d\n",
      "1759\n",
      "pos weak v294m\n",
      "1760\n",
      "pos weak h133q\n",
      "1761\n",
      "pos weak i130m\n",
      "1762\n",
      "pos i99m\n",
      "1763\n",
      "First match !  g70d\n",
      "1764\n",
      "First match !  r49c\n",
      "1765\n",
      "First match !  e190k\n",
      "1766\n",
      "First match !  r140q\n",
      "1767\n",
      "First match !  r172s\n",
      "1768\n",
      "First match !  r172g\n",
      "1769\n",
      "First match !  r172k\n",
      "1770\n",
      "First match !  r172m\n",
      "1771\n",
      "First match !  c528s\n",
      "1772\n",
      "First match !  e571k\n",
      "1773\n",
      "First match !  deletion\n",
      "1774\n",
      "First match !  truncating mutations\n",
      "1775\n",
      "1776\n",
      "First match !  deletion\n",
      "1777\n",
      "First match !  r1446h\n",
      "1778\n",
      "y371 aa_pos r1446c\n",
      "1779\n",
      "First match !  truncating mutations\n",
      "1780\n",
      "pos weak w1502a\n",
      "1781\n",
      "First match !  q1500p\n",
      "1782\n",
      "pos weak l702h\n",
      "1783\n",
      "1784\n",
      "First match !  p380r\n",
      "1785\n",
      "First match !  q58l\n",
      "1786\n",
      "First match !  arv567es\n",
      "1787\n",
      "First match !  a159t\n",
      "1788\n",
      "First match !  t878a\n",
      "1789\n",
      "pos weak h875y\n",
      "1790\n",
      "First match !  g216r\n",
      "1791\n",
      "pos weak w742l\n",
      "1792\n",
      "pos weak w742c\n",
      "1793\n",
      "pos weak t878s\n",
      "1794\n",
      "First match !  amplification\n",
      "1795\n",
      "1796\n",
      "First match !  f876l\n",
      "1797\n",
      "First match !  l272f\n",
      "1798\n",
      "First match !  q120e\n",
      "1799\n",
      "pos f877l\n",
      "1800\n",
      "First match !  g248v\n",
      "1801\n",
      "First match !  ar-v7\n",
      "1802\n",
      "First match !  n217i\n",
      "1803\n",
      "First match !  s214a\n",
      "1804\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "y371 aa_pos s214t\n",
      "1805\n",
      "First match !  q347_a348del\n",
      "1806\n",
      "First match !  f351l\n",
      "1807\n",
      "First match !  s214c\n",
      "1808\n",
      "First match !  g322s\n",
      "1809\n",
      "First match !  t37a\n",
      "1810\n",
      "First match !  g17v\n",
      "1811\n",
      "aa_pos_aa g17e\n",
      "1812\n",
      "aa_pos_aa g14v\n",
      "1813\n",
      "First match !  e40q\n",
      "1814\n",
      "aa_pos_aa l57v\n",
      "1815\n",
      "First match !  r5q\n",
      "1816\n",
      "1817\n",
      "First match !  t19n\n",
      "1818\n",
      "First match !  y42c\n",
      "1819\n",
      "First match !  e40t\n",
      "1820\n",
      "First match !  overexpression\n",
      "1821\n",
      "First match !  e40l\n",
      "1822\n",
      "First match !  g17a\n",
      "1823\n",
      "First match !  y34c\n",
      "1824\n",
      "First match !  e40w\n",
      "1825\n",
      "First match !  e40n\n",
      "1826\n",
      "1827\n",
      "First match !  w257g\n",
      "1828\n",
      "y371 aa_pos s256w\n",
      "1829\n",
      "pos p179r\n",
      "1830\n",
      "aa_pos r183p\n",
      "1831\n",
      "First match !  r182w\n",
      "1832\n",
      "pos r258h\n",
      "1833\n",
      "First match !  s256f\n",
      "1834\n",
      "pos r258c\n",
      "1835\n",
      "pos p179l\n",
      "1836\n",
      "aa_pos r183g\n",
      "1837\n",
      "y371 aa_pos w257c\n",
      "1838\n",
      "aa_pos r183q\n",
      "1839\n",
      "First match !  deletion\n",
      "1840\n",
      "First match !  truncating mutations\n",
      "1841\n",
      "First match !  r1625c\n",
      "1842\n",
      "First match !  m1255i\n",
      "1843\n",
      "First match !  a1374v\n",
      "1844\n",
      "First match !  g1125a\n",
      "1845\n",
      "First match !  h114y\n",
      "1846\n",
      "First match !  p186s\n",
      "1847\n",
      "First match !  s270l\n",
      "1848\n",
      "First match !  r264c\n",
      "1849\n",
      "First match !  h284n\n",
      "1850\n",
      "y371 aa_pos h284p\n",
      "1851\n",
      "1852\n",
      "First match !  r339w\n",
      "1853\n",
      "First match !  deletion\n",
      "1854\n",
      "First match !  amplification\n",
      "1855\n",
      "First match !  p2273s\n",
      "1856\n",
      "First match !  a1519t\n",
      "1857\n",
      "First match !  d2512g\n",
      "1858\n",
      "First match !  l1433s\n",
      "1859\n",
      "First match !  l2427r\n",
      "1860\n",
      "First match !  v2006l\n",
      "1861\n",
      "First match !  i2500f\n",
      "1862\n",
      "pos weak r2430m\n",
      "1863\n",
      "First match !  l2431p\n",
      "1864\n",
      "First match !  a8s\n",
      "1865\n",
      "First match !  r2505q\n",
      "1866\n",
      "First match !  i2500m\n",
      "1867\n",
      "First match !  e2419k\n",
      "1868\n",
      "First match !  t1977k\n",
      "1869\n",
      "First match !  f1888i\n",
      "1870\n",
      "First match !  c1483f\n",
      "1871\n",
      "First match !  w1456r\n",
      "1872\n",
      "First match !  v2006i\n",
      "1873\n",
      "First match !  a41p\n",
      "1874\n",
      "First match !  f1888v\n",
      "1875\n",
      "First match !  t1977r\n",
      "1876\n",
      "First match !  s2215t\n",
      "1877\n",
      "First match !  l2230v\n",
      "1878\n",
      "First match !  t1977i\n",
      "1879\n",
      "First match !  a1459p\n",
      "1880\n",
      "y371 aa_pos c1483w\n",
      "1881\n",
      "First match !  a2034v\n",
      "1882\n",
      "First match !  f1888l\n",
      "1883\n",
      "First match !  q2223k\n",
      "1884\n",
      "First match !  m2327i\n",
      "1885\n",
      "First match !  s2215f\n",
      "1886\n",
      "First match !  t1977s\n",
      "1887\n",
      "First match !  f2108l\n",
      "1888\n",
      "First match !  y1463s\n",
      "1889\n",
      "First match !  p2476l\n",
      "1890\n",
      "First match !  k1452n\n",
      "1891\n",
      "First match !  s2215y\n",
      "1892\n",
      "First match !  e2014k\n",
      "1893\n",
      "y371 aa_pos d2512y\n",
      "1894\n",
      "First match !  e1799k\n",
      "1895\n",
      "First match !  c1483y\n",
      "1896\n",
      "First match !  l1460p\n",
      "1897\n",
      "First match !  r2505*\n",
      "1898\n",
      "y371 aa_pos a41t\n",
      "1899\n",
      "First match !  c1483r\n",
      "1900\n",
      "First match !  deletion\n",
      "1901\n",
      "First match !  mll-tet1 fusion\n",
      "1902\n",
      "1903\n",
      "1904\n",
      "1905\n",
      "First match !  t910m\n",
      "1906\n",
      "1907\n",
      "First match !  g1232d\n",
      "1908\n",
      "First match !  truncating mutations\n",
      "1909\n",
      "First match !  deletion\n",
      "1910\n",
      "First match !  truncating mutations\n",
      "1911\n",
      "1912\n",
      "First match !  r514c\n",
      "1913\n",
      "First match !  e518k\n",
      "1914\n",
      "First match !  d384n\n",
      "1915\n",
      "First match !  v321m\n",
      "1916\n",
      "First match !  q477e\n",
      "1917\n",
      "y371 aa_pos d473g\n",
      "1918\n",
      "First match !  d473h\n",
      "1919\n",
      "First match !  h231r\n",
      "1920\n",
      "First match !  q635e\n",
      "1921\n",
      "First match !  n219d\n",
      "1922\n",
      "First match !  t241m\n",
      "1923\n",
      "pos weak l221r\n",
      "1924\n",
      "First match !  s387n\n",
      "1925\n",
      "First match !  i408v\n",
      "1926\n",
      "First match !  l412f\n",
      "1927\n",
      "First match !  w535l\n",
      "1928\n",
      "First match !  amplification\n",
      "1929\n",
      "First match !  f460l\n",
      "1930\n",
      "First match !  v414a\n",
      "1931\n",
      "First match !  a459v\n",
      "1932\n",
      "pos k575m\n",
      "1933\n",
      "First match !  e518a\n",
      "1934\n",
      "First match !  c49y\n",
      "1935\n",
      "First match !  f123i\n",
      "1936\n",
      "pos weak g116s\n",
      "1937\n",
      "First match !  g123s\n",
      "1938\n",
      "First match !  l232li\n",
      "1939\n",
      "First match !  l251p\n",
      "1940\n",
      "First match !  d387v\n",
      "1941\n",
      "First match !  l225li\n",
      "1942\n",
      "pos weak e127g\n",
      "1943\n",
      "pos s250p\n",
      "1944\n",
      "First match !  d401n\n",
      "1945\n",
      "First match !  e626k\n",
      "1946\n",
      "First match !  n81y\n",
      "1947\n",
      "First match !  d83v\n",
      "1948\n",
      "First match !  y69h\n",
      "1949\n",
      "First match !  e77k\n",
      "1950\n",
      "First match !  k4e\n",
      "1951\n",
      "1952\n",
      "First match !  t17a\n",
      "1953\n",
      "First match !  e1978*\n",
      "1954\n",
      "First match !  truncating mutations\n",
      "1955\n",
      "First match !  deletion\n",
      "1956\n",
      "First match !  n2875k\n",
      "1957\n",
      "First match !  d2870a\n",
      "1958\n",
      "First match !  d1853n\n",
      "1959\n",
      "aa_pos_aa g591v\n",
      "1960\n",
      "1961\n",
      "1962\n",
      "First match !  e322k\n",
      "1963\n",
      "First match !  amplification\n",
      "1964\n",
      "First match !  d32a\n",
      "1965\n",
      "First match !  s33f\n",
      "1966\n",
      "First match !  s37c\n",
      "1967\n",
      "First match !  g34e\n",
      "1968\n",
      "First match !  d32n\n",
      "1969\n",
      "First match !  k335i\n",
      "1970\n",
      "First match !  s33y\n",
      "1971\n",
      "First match !  s45a\n",
      "1972\n",
      "First match !  s45y\n",
      "1973\n",
      "First match !  s33c\n",
      "1974\n",
      "First match !  s37f\n",
      "1975\n",
      "First match !  s37y\n",
      "1976\n",
      "First match !  h36p\n",
      "1977\n",
      "First match !  s37a\n",
      "1978\n",
      "First match !  g34v\n",
      "1979\n",
      "1980\n",
      "First match !  t41a\n",
      "1981\n",
      "First match !  s33a\n",
      "1982\n",
      "First match !  s45f\n",
      "1983\n",
      "1984\n",
      "First match !  n387k\n",
      "1985\n",
      "First match !  i35s\n",
      "1986\n",
      "First match !  p44a\n",
      "1987\n",
      "First match !  s45p\n",
      "1988\n",
      "pos s23r\n",
      "1989\n",
      "pos t28i\n",
      "1990\n",
      "y371 aa_pos l115r\n",
      "1991\n",
      "First match !  c121s\n",
      "1992\n",
      "First match !  k57t\n",
      "1993\n",
      "First match !  i204t\n",
      "1994\n",
      "aa_pos i111a\n",
      "1995\n",
      "First match !  r49l\n",
      "1996\n",
      "First match !  p387s\n",
      "1997\n",
      "First match !  f129l\n",
      "1998\n",
      "First match !  q56_v60del\n",
      "1999\n",
      "pos weak s123t\n",
      "2000\n",
      "First match !  q56p\n",
      "2001\n",
      "First match !  f53c\n",
      "2002\n",
      "First match !  k97m\n",
      "2003\n",
      "First match !  e102_i103del\n",
      "2004\n",
      "y371 aa_pos v60m\n",
      "2005\n",
      "First match !  k59del\n",
      "2006\n",
      "pos d65n\n",
      "2007\n",
      "First match !  amplification\n",
      "2008\n",
      "First match !  q58_e62del\n",
      "2009\n",
      "First match !  y130c\n",
      "2010\n",
      "First match !  f53l\n",
      "2011\n",
      "aa_pos_aa i111p\n",
      "2012\n",
      "First match !  p124l\n",
      "2013\n",
      "First match !  i103n\n",
      "2014\n",
      "First match !  i111n\n",
      "2015\n",
      "First match !  p124q\n",
      "2016\n",
      "First match !  v60e\n",
      "2017\n",
      "First match !  v211d\n",
      "2018\n",
      "pos weak e120q\n",
      "2019\n",
      "First match !  f53s\n",
      "2020\n",
      "First match !  g128v\n",
      "2021\n",
      "First match !  p306h\n",
      "2022\n",
      "aa_pos i111r\n",
      "2023\n",
      "First match !  e203k\n",
      "2024\n",
      "2025\n",
      "First match !  k57e\n",
      "2026\n",
      "First match !  n382h\n",
      "2027\n",
      "First match !  p124s\n",
      "2028\n",
      "pos l37p\n",
      "2029\n",
      "First match !  r47q\n",
      "2030\n",
      "First match !  s222d\n",
      "2031\n",
      "First match !  d67n\n",
      "2032\n",
      "First match !  s226d\n",
      "2033\n",
      "First match !  f57v\n",
      "2034\n",
      "First match !  f57l\n",
      "2035\n",
      "First match !  k101m\n",
      "2036\n",
      "First match !  l46f\n",
      "2037\n",
      "First match !  n126d\n",
      "2038\n",
      "First match !  c125s\n",
      "2039\n",
      "First match !  f57c\n",
      "2040\n",
      "First match !  s222d\n",
      "2041\n",
      "First match !  e207k\n",
      "2042\n",
      "First match !  v35m\n",
      "2043\n",
      "First match !  truncating mutations\n",
      "2044\n",
      "2045\n",
      "First match !  m269r\n",
      "2046\n",
      "First match !  r552s\n",
      "2047\n",
      "First match !  r552g\n",
      "2048\n",
      "First match !  g434r\n",
      "2049\n",
      "First match !  e846k\n",
      "2050\n",
      "First match !  n233y\n",
      "2051\n",
      "pos t73i\n",
      "2052\n",
      "FUSION pvt1-myc fusion\n",
      "2053\n",
      "FUSION igl-myc fusion\n",
      "2054\n",
      "First match !  myc-nick\n",
      "2055\n",
      "First match !  fusions\n",
      "2056\n",
      "FUSION igh-myc fusion\n",
      "2057\n",
      "FUSION igk-myc fusion\n",
      "2058\n",
      "First match !  amplification\n",
      "2059\n",
      "First match !  amplification\n",
      "2060\n",
      "First match !  p44l\n",
      "2061\n",
      "First match !  f12l\n",
      "2062\n",
      "First match !  truncating mutations\n",
      "2063\n",
      "First match !  k398a\n",
      "2064\n",
      "First match !  l142p\n",
      "2065\n",
      "First match !  h65y\n",
      "2066\n",
      "First match !  a19v\n",
      "2067\n",
      "First match !  p70r\n",
      "2068\n",
      "First match !  d1384v\n",
      "2069\n",
      "First match !  deletion\n",
      "2070\n",
      "First match !  r1262a\n",
      "2071\n",
      "First match !  y1295a\n",
      "2072\n",
      "First match !  h1904r\n",
      "2073\n",
      "First match !  r1896m\n",
      "2074\n",
      "First match !  w1291r\n",
      "2075\n",
      "First match !  y1294a\n",
      "2076\n",
      "First match !  s1303n\n",
      "2077\n",
      "First match !  k1299e\n",
      "2078\n",
      "First match !  h1382y\n",
      "2079\n",
      "First match !  m1293a\n",
      "2080\n",
      "First match !  n1387a\n",
      "2081\n",
      "First match !  s1290a\n",
      "2082\n",
      "First match !  m232t\n",
      "2083\n",
      "First match !  s243n\n",
      "2084\n",
      "First match !  l265p\n",
      "2085\n",
      "aa_pos r812a\n",
      "2086\n",
      "2087\n",
      "2088\n",
      "First match !  amplification\n",
      "2089\n",
      "aa_pos q545a\n",
      "2090\n",
      "2091\n",
      "2092\n",
      "First match !  deletion\n",
      "2093\n",
      "First match !  truncating mutations\n",
      "2094\n",
      "First match !  y901c\n",
      "2095\n",
      "First match !  k975e\n",
      "2096\n",
      "First match !  deletion\n",
      "2097\n",
      "First match !  r882l\n",
      "2098\n",
      "First match !  w719*\n",
      "2099\n",
      "pos weak k765r\n",
      "2100\n",
      "First match !  l996f\n",
      "2101\n",
      "First match !  g909r\n",
      "2102\n",
      "2103\n",
      "pos weak d887n\n",
      "2104\n",
      "First match !  amplification\n",
      "2105\n",
      "aa_pos_aa l122r\n",
      "2106\n",
      "pos m1r\n",
      "2107\n",
      "pos m1k\n",
      "2108\n",
      "pos m1v\n",
      "2109\n",
      "pos m1t\n",
      "2110\n",
      "First match !  deletion\n",
      "2111\n",
      "First match !  truncating mutations\n",
      "2112\n",
      "2113\n",
      "First match !  amplification\n",
      "2114\n",
      "First match !  truncating mutations\n",
      "2115\n",
      "First match !  amplification\n",
      "2116\n",
      "2117\n",
      "First match !  deletion\n",
      "2118\n",
      "2119\n",
      "2120\n",
      "FUSION ccnd1-igh fusion\n",
      "2121\n",
      "2122\n",
      "First match !  amplification\n",
      "2123\n",
      "2124\n",
      "First match !  t286a\n",
      "2125\n",
      "y371 aa_pos t286i\n",
      "2126\n",
      "First match !  c47s\n",
      "2127\n",
      "First match !  p287a\n",
      "2128\n",
      "2129\n",
      "y371 aa_pos p287s\n",
      "2130\n",
      "y371 aa_pos p287t\n",
      "2131\n",
      "FUSION igh-bcl2 fusion\n",
      "2132\n",
      "First match !  d422n\n",
      "2133\n",
      "First match !  deletion\n",
      "2134\n",
      "First match !  r470c\n",
      "2135\n",
      "First match !  r272c\n",
      "2136\n",
      "First match !  g430c\n",
      "2137\n",
      "First match !  s243c\n",
      "2138\n",
      "First match !  r320q\n",
      "2139\n",
      "First match !  r415g\n",
      "2140\n",
      "First match !  v155f\n",
      "2141\n",
      "First match !  g423v\n",
      "2142\n",
      "2143\n",
      "First match !  g186r\n",
      "2144\n",
      "First match !  g480w\n",
      "2145\n",
      "First match !  g333s\n",
      "2146\n",
      "2147\n",
      "First match !  r571w\n",
      "2148\n",
      "First match !  d92g\n",
      "2149\n",
      "First match !  c124n\n",
      "2150\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "First match !  c124s\n",
      "2151\n",
      "First match !  l112r\n",
      "2152\n",
      "First match !  a34d\n",
      "2153\n",
      "First match !  e157g\n",
      "2154\n",
      "First match !  d107y\n",
      "2155\n",
      "First match !  a121p\n",
      "2156\n",
      "First match !  h118p\n",
      "2157\n",
      "First match !  a121v\n",
      "2158\n",
      "First match !  c136y\n",
      "2159\n",
      "First match !  r130*\n",
      "2160\n",
      "First match !  t401i\n",
      "2161\n",
      "First match !  l181p\n",
      "2162\n",
      "First match !  l112p\n",
      "2163\n",
      "First match !  g129r\n",
      "2164\n",
      "First match !  y68h\n",
      "2165\n",
      "pos p38l\n",
      "2166\n",
      "First match !  y174n\n",
      "2167\n",
      "First match !  t131l\n",
      "2168\n",
      "First match !  r15s\n",
      "2169\n",
      "First match !  y155c\n",
      "2170\n",
      "First match !  s227f\n",
      "2171\n",
      "First match !  c71y\n",
      "2172\n",
      "First match !  k125r\n",
      "2173\n",
      "First match !  s10n\n",
      "2174\n",
      "First match !  r173c\n",
      "2175\n",
      "First match !  d162h\n",
      "2176\n",
      "2177\n",
      "First match !  f347l\n",
      "2178\n",
      "First match !  p95s\n",
      "2179\n",
      "First match !  m199del\n",
      "2180\n",
      "First match !  n276s\n",
      "2181\n",
      "First match !  t131a\n",
      "2182\n",
      "First match !  s362l\n",
      "2183\n",
      "y371 aa_pos v217d\n",
      "2184\n",
      "First match !  g129a\n",
      "2185\n",
      "First match !  t167a\n",
      "2186\n",
      "First match !  a126g\n",
      "2187\n",
      "First match !  g165e\n",
      "2188\n",
      "First match !  p169h\n",
      "2189\n",
      "First match !  n48k\n",
      "2190\n",
      "First match !  g165r\n",
      "2191\n",
      "First match !  deletion\n",
      "2192\n",
      "First match !  r130k\n",
      "2193\n",
      "pos r15k\n",
      "2194\n",
      "First match !  h123d\n",
      "2195\n",
      "First match !  g127n\n",
      "2196\n",
      "First match !  l325f\n",
      "2197\n",
      "First match !  g20e\n",
      "2198\n",
      "First match !  s170r\n",
      "2199\n",
      "First match !  k125m\n",
      "2200\n",
      "First match !  c105f\n",
      "2201\n",
      "First match !  d92a\n",
      "2202\n",
      "First match !  a121e\n",
      "2203\n",
      "First match !  a126v\n",
      "2204\n",
      "First match !  y65c\n",
      "2205\n",
      "First match !  c136r\n",
      "2206\n",
      "First match !  t131s\n",
      "2207\n",
      "First match !  f341v\n",
      "2208\n",
      "First match !  a126s\n",
      "2209\n",
      "First match !  r130g\n",
      "2210\n",
      "First match !  h93q\n",
      "2211\n",
      "First match !  a126d\n",
      "2212\n",
      "First match !  g251c\n",
      "2213\n",
      "First match !  f241s\n",
      "2214\n",
      "First match !  i168f\n",
      "2215\n",
      "First match !  p95l\n",
      "2216\n",
      "First match !  d92e\n",
      "2217\n",
      "First match !  k125l\n",
      "2218\n",
      "First match !  c124r\n",
      "2219\n",
      "First match !  l23f\n",
      "2220\n",
      "First match !  d162g\n",
      "2221\n",
      "First match !  g129e\n",
      "2222\n",
      "First match !  k128n\n",
      "2223\n",
      "First match !  k289e\n",
      "2224\n",
      "First match !  i122v\n",
      "2225\n",
      "First match !  y68d\n",
      "2226\n",
      "First match !  v343e\n",
      "2227\n",
      "First match !  r159g\n",
      "2228\n",
      "First match !  t131i\n",
      "2229\n",
      "First match !  f21a\n",
      "2230\n",
      "First match !  r130l\n",
      "2231\n",
      "First match !  h123q\n",
      "2232\n",
      "First match !  v369g\n",
      "2233\n",
      "First match !  g127e\n",
      "2234\n",
      "First match !  k342n\n",
      "2235\n",
      "2236\n",
      "First match !  y16c\n",
      "2237\n",
      "First match !  d252g\n",
      "2238\n",
      "First match !  h61d\n",
      "2239\n",
      "First match !  l108p\n",
      "2240\n",
      "First match !  r173p\n",
      "2241\n",
      "First match !  d331g\n",
      "2242\n",
      "First match !  g165v\n",
      "2243\n",
      "First match !  k62r\n",
      "2244\n",
      "First match !  i122l\n",
      "2245\n",
      "First match !  r161g\n",
      "2246\n",
      "First match !  t160i\n",
      "2247\n",
      "First match !  r130a\n",
      "2248\n",
      "First match !  a39p\n",
      "2249\n",
      "First match !  d24y\n",
      "2250\n",
      "2251\n",
      "First match !  l345q\n",
      "2252\n",
      "First match !  s170n\n",
      "2253\n",
      "First match !  k125e\n",
      "2254\n",
      "First match !  i122s\n",
      "2255\n",
      "First match !  k128t\n",
      "2256\n",
      "y371 aa_pos v343l\n",
      "2257\n",
      "First match !  y27s\n",
      "2258\n",
      "First match !  m134l\n",
      "2259\n",
      "First match !  q110r\n",
      "2260\n",
      "First match !  m35r\n",
      "2261\n",
      "First match !  f154l\n",
      "2262\n",
      "First match !  p96q\n",
      "2263\n",
      "First match !  d326n\n",
      "2264\n",
      "First match !  h61r\n",
      "2265\n",
      "First match !  h93r\n",
      "2266\n",
      "First match !  d92v\n",
      "2267\n",
      "First match !  d92h\n",
      "2268\n",
      "First match !  r173h\n",
      "2269\n",
      "First match !  h93d\n",
      "2270\n",
      "First match !  h123y\n",
      "2271\n",
      "First match !  d92n\n",
      "2272\n",
      "First match !  k128q\n",
      "2273\n",
      "2274\n",
      "First match !  bcor-rara fusion\n",
      "2275\n",
      "First match !  zc3h7b-bcor fusion\n",
      "2276\n",
      "First match !  deletion\n",
      "2277\n",
      "FUSION fusions\n",
      "2278\n",
      "First match !  bcor-ccnb3 fusion\n",
      "2279\n",
      "First match !  truncating mutations\n",
      "2280\n",
      "aa_pos_aa r1515h\n",
      "2281\n",
      "First match !  truncating mutations\n",
      "2282\n",
      "aa_pos_aa r201w\n",
      "2283\n",
      "First match !  fusions\n",
      "2284\n",
      "First match !  cic-dux4 fusion\n",
      "2285\n",
      "2286\n",
      "First match !  h410r\n",
      "2287\n",
      "First match !  d661y\n",
      "2288\n",
      "First match !  y640f\n",
      "2289\n",
      "First match !  d661v\n",
      "2290\n",
      "First match !  s614r\n",
      "2291\n",
      "First match !  f384l\n",
      "2292\n",
      "y371 aa_pos f384y\n",
      "2293\n",
      "y371 aa_pos f384v\n",
      "2294\n",
      "First match !  amplification\n",
      "2295\n",
      "First match !  y149d\n",
      "2296\n",
      "First match !  p48r\n",
      "2297\n",
      "First match !  c569y\n",
      "2298\n",
      "First match !  deletion\n",
      "2299\n",
      "First match !  p48t\n",
      "2300\n",
      "First match !  s646f\n",
      "2301\n",
      "First match !  f958s\n",
      "2302\n",
      "2303\n",
      "First match !  k1026e\n",
      "2304\n",
      "First match !  y652h\n",
      "2305\n",
      "First match !  s703i\n",
      "2306\n",
      "First match !  l783f\n",
      "2307\n",
      "First match !  v658f\n",
      "2308\n",
      "First match !  a723d\n",
      "2309\n",
      "pos weak r879c\n",
      "2310\n",
      "First match !  a634d\n",
      "2311\n",
      "First match !  truncating mutations\n",
      "2312\n",
      "2313\n",
      "First match !  l910p\n",
      "2314\n",
      "First match !  y1035c\n",
      "2315\n",
      "First match !  f958v\n",
      "2316\n",
      "First match !  s729c\n",
      "2317\n",
      "First match !  k648n\n",
      "2318\n",
      "First match !  l611v\n",
      "2319\n",
      "First match !  k539l\n",
      "2320\n",
      "First match !  d544h\n",
      "2321\n",
      "First match !  y1045*\n",
      "2322\n",
      "First match !  spag9-jak2 fusion\n",
      "2323\n",
      "First match !  ssbp2-jak2 fusion\n",
      "2324\n",
      "First match !  w1038c\n",
      "2325\n",
      "First match !  y931c\n",
      "2326\n",
      "First match !  pcm1-jak2 fusion\n",
      "2327\n",
      "First match !  bcr-jak2 fusion\n",
      "2328\n",
      "First match !  r683t\n",
      "2329\n",
      "2330\n",
      "First match !  l983f\n",
      "2331\n",
      "2332\n",
      "First match !  pax5-jak2 fusion\n",
      "2333\n",
      "First match !  l1026f\n",
      "2334\n",
      "First match !  i1018f\n",
      "2335\n",
      "FUSION atf7ip-jak2 fusion\n",
      "2336\n",
      "First match !  s1039f\n",
      "2337\n",
      "First match !  s1025c\n",
      "2338\n",
      "First match !  v1075f\n",
      "2339\n",
      "First match !  t875n\n",
      "2340\n",
      "First match !  r683s\n",
      "2341\n",
      "First match !  y1045w\n",
      "2342\n",
      "First match !  c618r\n",
      "2343\n",
      "First match !  amplification\n",
      "2344\n",
      "First match !  f1061w\n",
      "2345\n",
      "First match !  h538q\n",
      "2346\n",
      "First match !  r683g\n",
      "2347\n",
      "First match !  i1018w\n",
      "2348\n",
      "First match !  tel-jak2 fusion\n",
      "2349\n",
      "First match !  r683k\n",
      "2350\n",
      "First match !  fusions\n",
      "2351\n",
      "First match !  amplification\n",
      "2352\n",
      "First match !  f31i\n",
      "2353\n",
      "First match !  k162d\n",
      "2354\n",
      "First match !  l67p\n",
      "2355\n",
      "First match !  m136r\n",
      "2356\n",
      "First match !  g163d\n",
      "2357\n",
      "First match !  w308c\n",
      "2358\n",
      "First match !  k78a\n",
      "2359\n",
      "First match !  d194y\n",
      "2360\n",
      "First match !  r304w\n",
      "2361\n",
      "First match !  deletion\n",
      "2362\n",
      "First match !  truncating mutations\n",
      "2363\n",
      "First match !  h174r\n",
      "2364\n",
      "First match !  a205t\n",
      "2365\n",
      "First match !  k78i\n",
      "2366\n",
      "First match !  f354l\n",
      "2367\n",
      "First match !  amplification\n",
      "2368\n",
      "First match !  deletion\n",
      "2369\n",
      "First match !  d61y\n",
      "2370\n",
      "First match !  t73i\n",
      "2371\n",
      "First match !  t507k\n",
      "2372\n",
      "First match !  e69k\n",
      "2373\n",
      "First match !  e76k\n",
      "2374\n",
      "First match !  g503v\n",
      "2375\n",
      "First match !  q79r\n",
      "2376\n",
      "First match !  r498l\n",
      "2377\n",
      "First match !  a72v\n",
      "2378\n",
      "First match !  d106a\n",
      "2379\n",
      "First match !  q510e\n",
      "2380\n",
      "First match !  m504v\n",
      "2381\n",
      "First match !  t468m\n",
      "2382\n",
      "First match !  v45l\n",
      "2383\n",
      "First match !  e76a\n",
      "2384\n",
      "First match !  p491s\n",
      "2385\n",
      "First match !  k590r\n",
      "2386\n",
      "First match !  f71i\n",
      "2387\n",
      "First match !  a72s\n",
      "2388\n",
      "First match !  g464a\n",
      "2389\n",
      "First match !  y63c\n",
      "2390\n",
      "First match !  s502t\n",
      "2391\n",
      "First match !  d61n\n",
      "2392\n",
      "First match !  y62c\n",
      "2393\n",
      "First match !  e139d\n",
      "2394\n",
      "First match !  r1391g\n",
      "2395\n",
      "First match !  y489c\n",
      "2396\n",
      "First match !  r1391s\n",
      "2397\n",
      "First match !  k1436q\n",
      "2398\n",
      "First match !  t1324n\n",
      "2399\n",
      "First match !  l1301r\n",
      "2400\n",
      "First match !  v1398d\n",
      "2401\n",
      "First match !  r1276p\n",
      "2402\n",
      "First match !  s1463f\n",
      "2403\n",
      "First match !  truncating mutations\n",
      "2404\n",
      "First match !  r2450*\n",
      "2405\n",
      "First match !  r1276q\n",
      "2406\n",
      "First match !  e1356g\n",
      "2407\n",
      "First match !  r1204w\n",
      "2408\n",
      "First match !  deletion\n",
      "2409\n",
      "First match !  r1204g\n",
      "2410\n",
      "First match !  c134w\n",
      "2411\n",
      "2412\n",
      "First match !  r1088c\n",
      "2413\n",
      "First match !  v1565i\n",
      "2414\n",
      "2415\n",
      "First match !  p1311t\n",
      "2416\n",
      "First match !  deletion\n",
      "2417\n",
      "First match !  e1735k\n",
      "2418\n",
      "First match !  r1753t\n",
      "2419\n",
      "First match !  c44y\n",
      "2420\n",
      "pos weak r1835p\n",
      "2421\n",
      "2422\n",
      "pos weak g1803a\n",
      "2423\n",
      "pos weak a1830t\n",
      "2424\n",
      "First match !  g1743r\n",
      "2425\n",
      "First match !  i68k\n",
      "2426\n",
      "First match !  c64g\n",
      "2427\n",
      "First match !  s4p\n",
      "2428\n",
      "First match !  k50l\n",
      "2429\n",
      "First match !  y105c\n",
      "2430\n",
      "First match !  l63f\n",
      "2431\n",
      "First match !  l1407p\n",
      "2432\n",
      "First match !  l1657p\n",
      "2433\n",
      "First match !  f79s\n",
      "2434\n",
      "aa_pos d1778h\n",
      "2435\n",
      "First match !  d1546n\n",
      "2436\n",
      "2437\n",
      "First match !  d96n\n",
      "2438\n",
      "y371 aa_pos w1837c\n",
      "2439\n",
      "First match !  deletion\n",
      "2440\n",
      "pos weak p1856s\n",
      "2441\n",
      "First match !  f1761i\n",
      "2442\n",
      "First match !  a1823t\n",
      "2443\n",
      "pos weak i1807s\n",
      "2444\n",
      "First match !  k38n\n",
      "2445\n",
      "First match !  s1651p\n",
      "2446\n",
      "First match !  l246v\n",
      "2447\n",
      "First match !  v191i\n",
      "2448\n",
      "First match !  t37r\n",
      "2449\n",
      "2450\n",
      "First match !  s72r\n",
      "2451\n",
      "First match !  y179c\n",
      "2452\n",
      "First match !  c39y\n",
      "2453\n",
      "y371 aa_pos k45n\n",
      "2454\n",
      "First match !  l52f\n",
      "2455\n",
      "First match !  h1746n\n",
      "2456\n",
      "First match !  p1749r\n",
      "2457\n",
      "First match !  h1746q\n",
      "2458\n",
      "First match !  i21v\n",
      "2459\n",
      "First match !  w1718c\n",
      "2460\n",
      "First match !  r170w\n",
      "2461\n",
      "First match !  t77m\n",
      "2462\n",
      "First match !  v1713a\n",
      "2463\n",
      "First match !  v1833m\n",
      "2464\n",
      "First match !  s1101n\n",
      "2465\n",
      "First match !  t1691i\n",
      "2466\n",
      "First match !  l147f\n",
      "2467\n",
      "First match !  h1421y\n",
      "2468\n",
      "First match !  e362h\n",
      "2469\n",
      "First match !  r170q\n",
      "2470\n",
      "2471\n",
      "First match !  e1794d\n",
      "2472\n",
      "First match !  l358r\n",
      "2473\n",
      "First match !  s308a\n",
      "2474\n",
      "First match !  v1741g\n",
      "2475\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "First match !  r1699l\n",
      "2476\n",
      "First match !  v1804a\n",
      "2477\n",
      "First match !  e23fs\n",
      "2478\n",
      "pos weak f1734s\n",
      "2479\n",
      "First match !  c61g\n",
      "2480\n",
      "First match !  d245v\n",
      "2481\n",
      "First match !  p142h\n",
      "2482\n",
      "pos l1705p\n",
      "2483\n",
      "First match !  h1686r\n",
      "2484\n",
      "pos weak p1776s\n",
      "2485\n",
      "First match !  m1783t\n",
      "2486\n",
      "pos weak s1841r\n",
      "2487\n",
      "First match !  v1534m\n",
      "2488\n",
      "First match !  w1782c\n",
      "2489\n",
      "First match !  g1706a\n",
      "2490\n",
      "First match !  g1788v\n",
      "2491\n",
      "First match !  w1837g\n",
      "2492\n",
      "First match !  a1708e\n",
      "2493\n",
      "First match !  s1512i\n",
      "2494\n",
      "y371 aa_pos m1783i\n",
      "2495\n",
      "2496\n",
      "y371 aa_pos m1783l\n",
      "2497\n",
      "First match !  g1706e\n",
      "2498\n",
      "First match !  r496c\n",
      "2499\n",
      "First match !  s153r\n",
      "2500\n",
      "First match !  n810y\n",
      "2501\n",
      "First match !  k1702e\n",
      "2502\n",
      "pos weak a1789s\n",
      "2503\n",
      "First match !  r866c\n",
      "2504\n",
      "pos weak v11a\n",
      "2505\n",
      "First match !  d1739v\n",
      "2506\n",
      "y371 aa_pos m1775e\n",
      "2507\n",
      "First match !  d67y\n",
      "2508\n",
      "First match !  d1344h\n",
      "2509\n",
      "First match !  v271l\n",
      "2510\n",
      "pos weak s1841n\n",
      "2511\n",
      "y371 aa_pos f1761s\n",
      "2512\n",
      "First match !  e1586g\n",
      "2513\n",
      "First match !  s1841a\n",
      "2514\n",
      "y371 aa_pos i89n\n",
      "2515\n",
      "First match !  l82p\n",
      "2516\n",
      "First match !  p798l\n",
      "2517\n",
      "First match !  e29v\n",
      "2518\n",
      "First match !  m1775k\n",
      "2519\n",
      "First match !  e1060a\n",
      "2520\n",
      "First match !  i90t\n",
      "2521\n",
      "First match !  i15t\n",
      "2522\n",
      "y371 aa_pos v1736g\n",
      "2523\n",
      "pos weak l1854p\n",
      "2524\n",
      "pos weak e1836k\n",
      "2525\n",
      "pos weak f1704s\n",
      "2526\n",
      "pos weak p1812s\n",
      "2527\n",
      "First match !  h1402y\n",
      "2528\n",
      "First match !  n550h\n",
      "2529\n",
      "2530\n",
      "First match !  g1738r\n",
      "2531\n",
      "First match !  s1497a\n",
      "2532\n",
      "2533\n",
      "First match !  q1785h\n",
      "2534\n",
      "First match !  h41r\n",
      "2535\n",
      "First match !  r1589p\n",
      "2536\n",
      "First match !  l165p\n",
      "2537\n",
      "First match !  c24r\n",
      "2538\n",
      "pos weak l1844r\n",
      "2539\n",
      "First match !  c1767s\n",
      "2540\n",
      "First match !  k65m\n",
      "2541\n",
      "y371 aa_pos v1809a\n",
      "2542\n",
      "pos y1703h\n",
      "2543\n",
      "First match !  c27a\n",
      "2544\n",
      "First match !  s1655f\n",
      "2545\n",
      "First match !  n1819y\n",
      "2546\n",
      "First match !  s1486c\n",
      "2547\n",
      "pos p1859r\n",
      "2548\n",
      "First match !  y1853*\n",
      "2549\n",
      "2550\n",
      "First match !  k56n\n",
      "2551\n",
      "First match !  s1651f\n",
      "2552\n",
      "2553\n",
      "pos weak d1778n\n",
      "2554\n",
      "First match !  y1703s\n",
      "2555\n",
      "First match !  a1669s\n",
      "2556\n",
      "First match !  w1837r\n",
      "2557\n",
      "First match !  v1833e\n",
      "2558\n",
      "First match !  p1806a\n",
      "2559\n",
      "First match !  m18t\n",
      "2560\n",
      "First match !  m1411t\n",
      "2561\n",
      "First match !  s1140g\n",
      "2562\n",
      "First match !  y1853c\n",
      "2563\n",
      "First match !  t1700a\n",
      "2564\n",
      "pos weak t1852s\n",
      "2565\n",
      "pos weak h1805p\n",
      "2566\n",
      "pos weak d1778y\n",
      "2567\n",
      "pos weak t1685a\n",
      "2568\n",
      "First match !  d1733g\n",
      "2569\n",
      "pos q1811r\n",
      "2570\n",
      "First match !  d1739g\n",
      "2571\n",
      "First match !  s1297del\n",
      "2572\n",
      "First match !  i89t\n",
      "2573\n",
      "First match !  c1697r\n",
      "2574\n",
      "First match !  l1764p\n",
      "2575\n",
      "First match !  p1637l\n",
      "2576\n",
      "First match !  g1770v\n",
      "2577\n",
      "y371 aa_pos a1752v\n",
      "2578\n",
      "First match !  g1656d\n",
      "2579\n",
      "pos weak g1763v\n",
      "2580\n",
      "First match !  i31m\n",
      "2581\n",
      "First match !  v1688del\n",
      "2582\n",
      "pos weak d1739y\n",
      "2583\n",
      "First match !  e1660g\n",
      "2584\n",
      "First match !  s1715r\n",
      "2585\n",
      "2586\n",
      "First match !  m1628v\n",
      "2587\n",
      "First match !  s186y\n",
      "2588\n",
      "First match !  s4f\n",
      "2589\n",
      "First match !  r1699w\n",
      "2590\n",
      "First match !  l28p\n",
      "2591\n",
      "First match !  n132k\n",
      "2592\n",
      "pos weak v1838e\n",
      "2593\n",
      "y371 aa_pos c24y\n",
      "2594\n",
      "pos e60l\n",
      "2595\n",
      "First match !  g1738e\n",
      "2596\n",
      "First match !  g1788d\n",
      "2597\n",
      "First match !  s59r\n",
      "2598\n",
      "First match !  v1809f\n",
      "2599\n",
      "First match !  m1689r\n",
      "2600\n",
      "First match !  d1739e\n",
      "2601\n",
      "First match !  m1400v\n",
      "2602\n",
      "First match !  m1008i\n",
      "2603\n",
      "First match !  e1282v\n",
      "2604\n",
      "First match !  e1214k\n",
      "2605\n",
      "First match !  v1665m\n",
      "2606\n",
      "First match !  c39r\n",
      "2607\n",
      "First match !  m1775v\n",
      "2608\n",
      "First match !  r1589h\n",
      "2609\n",
      "First match !  i26n\n",
      "2610\n",
      "pos n1819s\n",
      "2611\n",
      "First match !  v1808a\n",
      "2612\n",
      "First match !  h1862l\n",
      "2613\n",
      "First match !  t1720i\n",
      "2614\n",
      "First match !  m1r\n",
      "2615\n",
      "First match !  t47d\n",
      "2616\n",
      "pos weak p1771l\n",
      "2617\n",
      "First match !  v1736a\n",
      "2618\n",
      "pos weak c1787s\n",
      "2619\n",
      "First match !  v1810g\n",
      "2620\n",
      "First match !  n1236k\n",
      "2621\n",
      "First match !  r1699q\n",
      "2622\n",
      "2623\n",
      "First match !  a1789t\n",
      "2624\n",
      "2625\n",
      "First match !  k45q\n",
      "2626\n",
      "First match !  d1692n\n",
      "2627\n",
      "First match !  i1766s\n",
      "2628\n",
      "First match !  l1267s\n",
      "2629\n",
      "First match !  e143k\n",
      "2630\n",
      "First match !  t1720a\n",
      "2631\n",
      "y371 aa_pos l1780p\n",
      "2632\n",
      "First match !  i42v\n",
      "2633\n",
      "First match !  q12y\n",
      "2634\n",
      "First match !  w1718l\n",
      "2635\n",
      "pos weak p1856t\n",
      "2636\n",
      "pos weak r1751q\n",
      "2637\n",
      "First match !  a1752p\n",
      "2638\n",
      "2639\n",
      "First match !  v1653m\n",
      "2640\n",
      "pos weak n1730s\n",
      "2641\n",
      "First match !  e1644g\n",
      "2642\n",
      "First match !  h1686q\n",
      "2643\n",
      "First match !  q1826h\n",
      "2644\n",
      "First match !  r71g\n",
      "2645\n",
      "First match !  s1164i\n",
      "2646\n",
      "First match !  v1804d\n",
      "2647\n",
      "First match !  s1301r\n",
      "2648\n",
      "First match !  c44f\n",
      "2649\n",
      "First match !  v191d\n",
      "2650\n",
      "pos q1756c\n",
      "2651\n",
      "First match !  r841q\n",
      "2652\n",
      "First match !  m1652k\n",
      "2653\n",
      "pos weak r1751p\n",
      "2654\n",
      "First match !  c47g\n",
      "2655\n",
      "First match !  c360r\n",
      "2656\n",
      "2657\n",
      "First match !  s1715n\n",
      "2658\n",
      "y371 aa_pos k50e\n",
      "2659\n",
      "y371 aa_pos a1708v\n",
      "2660\n",
      "First match !  e1250k\n",
      "2661\n",
      "First match !  s1613c\n",
      "2662\n",
      "First match !  l668f\n",
      "2663\n",
      "First match !  p1614s\n",
      "2664\n",
      "First match !  a1843p\n",
      "2665\n",
      "First match !  c39s\n",
      "2666\n",
      "First match !  p1812r\n",
      "2667\n",
      "First match !  v1378i\n",
      "2668\n",
      "pos weak p1771r\n",
      "2669\n",
      "First match !  t1685i\n",
      "2670\n",
      "pos weak t1691k\n",
      "2671\n",
      "First match !  f1695l\n",
      "2672\n",
      "y371 aa_pos k45t\n",
      "2673\n",
      "First match !  s1473p\n",
      "2674\n",
      "First match !  m1652t\n",
      "2675\n",
      "First match !  e1346k\n",
      "2676\n",
      "pos weak r133h\n",
      "2677\n",
      "First match !  a1843t\n",
      "2678\n",
      "pos weak d1778g\n",
      "2679\n",
      "First match !  s1613g\n",
      "2680\n",
      "First match !  m18k\n",
      "2681\n",
      "First match !  d594n\n",
      "2682\n",
      "First match !  e501k\n",
      "2683\n",
      "2684\n",
      "First match !  a598t\n",
      "2685\n",
      "2686\n",
      "First match !  l485f\n",
      "2687\n",
      "2688\n",
      "2689\n",
      "First match !  a598v\n",
      "2690\n",
      "First match !  i326v\n",
      "2691\n",
      "First match !  q201h\n",
      "2692\n",
      "First match !  n581d\n",
      "2693\n",
      "First match !  a246p\n",
      "2694\n",
      "First match !  g596v\n",
      "2695\n",
      "First match !  v600r\n",
      "2696\n",
      "First match !  v471f\n",
      "2697\n",
      "First match !  t529m\n",
      "2698\n",
      "First match !  v600m\n",
      "2699\n",
      "First match !  g478c\n",
      "2700\n",
      "First match !  e501g\n",
      "2701\n",
      "First match !  r462e\n",
      "2702\n",
      "First match !  t529i\n",
      "2703\n",
      "First match !  trim24-braf fusion\n",
      "2704\n",
      "2705\n",
      "pos weak n581s\n",
      "2706\n",
      "First match !  k483e\n",
      "2707\n",
      "2708\n",
      "FUSION fam131b-braf fusion\n",
      "2709\n",
      "FUSION cul1-braf fusion\n",
      "2710\n",
      "First match !  l597q\n",
      "2711\n",
      "First match !  r671q\n",
      "2712\n",
      "2713\n",
      "First match !  kiaa1549-braf fusion\n",
      "2714\n",
      "2715\n",
      "First match !  k601n\n",
      "2716\n",
      "2717\n",
      "y371 aa_pos d594y\n",
      "2718\n",
      "First match !  p61braf\n",
      "2719\n",
      "First match !  v600e\n",
      "2720\n",
      "First match !  agk-braf fusion\n",
      "2721\n",
      "2722\n",
      "First match !  s151a\n",
      "2723\n",
      "pos weak g464r\n",
      "2724\n",
      "First match !  l597r\n",
      "2725\n",
      "First match !  papss1-braf fusion\n",
      "2726\n",
      "First match !  d594g\n",
      "2727\n",
      "pos r462i\n",
      "2728\n",
      "First match !  g469a\n",
      "2729\n",
      "aa_pos_aa k601q\n",
      "2730\n",
      "First match !  g469v\n",
      "2731\n",
      "FUSION mkrn1-braf fusion\n",
      "2732\n",
      "pos i463s\n",
      "2733\n",
      "pos f468c\n",
      "2734\n",
      "aa_pos_aa t241p\n",
      "2735\n",
      "First match !  d594a\n",
      "2736\n",
      "First match !  t529n\n",
      "2737\n",
      "First match !  g466a\n",
      "2738\n",
      "First match !  k483m\n",
      "2739\n",
      "First match !  l597s\n",
      "2740\n",
      "First match !  l584f\n",
      "2741\n",
      "aa_pos_aa w531c\n",
      "2742\n",
      "2743\n",
      "pos d594v\n",
      "2744\n",
      "First match !  snd1-braf fusion\n",
      "2745\n",
      "2746\n",
      "First match !  g466e\n",
      "2747\n",
      "y371 aa_pos g466r\n",
      "2748\n",
      "First match !  amplification\n",
      "2749\n",
      "First match !  g596r\n",
      "2750\n",
      "First match !  g469del\n",
      "2751\n",
      "2752\n",
      "First match !  akap9-braf fusion\n",
      "2753\n",
      "First match !  fusions\n",
      "2754\n",
      "First match !  v600k\n",
      "2756\n",
      "pos weak e586k\n",
      "2757\n",
      "First match !  v600d\n",
      "2758\n",
      "y371 aa_pos g469e\n",
      "2759\n",
      "First match !  v600g\n",
      "2760\n",
      "pos v600d_k601insfglat\n",
      "2761\n",
      "First match !  d287h\n",
      "2762\n",
      "pos weak g464v\n",
      "2763\n",
      "First match !  k601e\n",
      "2764\n",
      "aa_pos_aa e275k\n",
      "2765\n",
      "First match !  k499e\n",
      "2766\n",
      "First match !  l597v\n",
      "2767\n",
      "y371 aa_pos d594e\n",
      "2768\n",
      "pos t599i\n",
      "2769\n",
      "First match !  n581y\n",
      "2770\n",
      "First match !  g466v\n",
      "2771\n",
      "aa_pos_aa t599r\n",
      "2772\n",
      "First match !  y472c\n",
      "2773\n",
      "pos weak g464e\n",
      "2774\n",
      "First match !  p168t\n",
      "2775\n",
      "pos r2336p\n",
      "2776\n",
      "First match !  t2250a\n",
      "2777\n",
      "2778\n",
      "First match !  l2396f\n",
      "2779\n",
      "First match !  m784v\n",
      "2780\n",
      "First match !  v894i\n",
      "2781\n",
      "First match !  g2748d\n",
      "2782\n",
      "First match !  r2502c\n",
      "2783\n",
      "2784\n",
      "2785\n",
      "First match !  k2729n\n",
      "2786\n",
      "First match !  k1690n\n",
      "2787\n",
      "First match !  y3092c\n",
      "2788\n",
      "aa_pos_aa i2675v\n",
      "2789\n",
      "First match !  a2643g\n",
      "2790\n",
      "2791\n",
      "First match !  n1878k\n",
      "2792\n",
      "First match !  t3349a\n",
      "2793\n",
      "First match !  c1265s\n",
      "2794\n",
      "First match !  t582p\n",
      "2795\n",
      "First match !  t1354m\n",
      "2796\n",
      "First match !  l1904v\n",
      "2797\n",
      "2798\n",
      "First match !  s384f\n",
      "2799\n",
      "First match !  e2663v\n",
      "2800\n",
      "First match !  r3052w\n",
      "2801\n",
      "2802\n",
      "First match !  n2436i\n",
      "2803\n",
      "First match !  n319t\n",
      "2804\n",
      "First match !  r2659t\n",
      "2805\n",
      "First match !  e3002k\n",
      "2806\n",
      "First match !  g1194d\n",
      "2807\n",
      "First match !  q2858r\n",
      "2808\n",
      "First match !  k2411t\n",
      "2809\n",
      "2810\n",
      "First match !  v1306i\n",
      "2811\n",
      "First match !  n1102y\n",
      "2812\n",
      "First match !  f1524v\n",
      "2813\n",
      "2814\n",
      "2815\n",
      "2816\n",
      "2817\n",
      "First match !  y3098h\n",
      "2818\n",
      "2819\n",
      "First match !  y42c\n",
      "2820\n",
      "First match !  r1190w\n",
      "2821\n",
      "First match !  c1365y\n",
      "2822\n",
      "First match !  a2717s\n",
      "2823\n",
      "First match !  n372h\n",
      "2824\n",
      "First match !  r2520q\n",
      "2825\n",
      "First match !  s1172l\n",
      "2826\n",
      "First match !  q2384k\n",
      "2827\n",
      "First match !  p1819s\n",
      "2828\n",
      "First match !  r2842h\n",
      "2829\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "First match !  p655r\n",
      "2830\n",
      "First match !  l1019v\n",
      "2831\n",
      "First match !  r2888c\n",
      "2832\n",
      "2833\n",
      "2834\n",
      "First match !  k2950n\n",
      "2835\n",
      "First match !  r2784w\n",
      "2836\n",
      "First match !  i2627f\n",
      "2837\n",
      "2838\n",
      "First match !  d2312v\n",
      "2839\n",
      "First match !  d2723h\n",
      "2840\n",
      "First match !  deletion\n",
      "2841\n",
      "2842\n",
      "First match !  v2908g\n",
      "2843\n",
      "pos r18h\n",
      "2844\n",
      "aa_pos_aa s196n\n",
      "2845\n",
      "First match !  a1170v\n",
      "2846\n",
      "First match !  r2336h\n",
      "2847\n",
      "First match !  d806h\n",
      "2848\n",
      "pos weak v1643a\n",
      "2849\n",
      "First match !  h1966y\n",
      "2850\n",
      "First match !  t630i\n",
      "2851\n",
      "First match !  e2856a\n",
      "2852\n",
      "First match !  s1733f\n",
      "2853\n",
      "First match !  e462g\n",
      "2854\n",
      "First match !  i1349t\n",
      "2855\n",
      "First match !  d3095e\n",
      "2856\n",
      "First match !  r2973c\n",
      "2857\n",
      "First match !  d935n\n",
      "2858\n",
      "First match !  d2665g\n",
      "2859\n",
      "First match !  d3170g\n",
      "2860\n",
      "aa_pos_aa r2602t\n",
      "2861\n",
      "First match !  r2842c\n",
      "2862\n",
      "First match !  c554w\n",
      "2863\n",
      "First match !  n1228d\n",
      "2864\n",
      "First match !  v3079i\n",
      "2865\n",
      "First match !  h1918y\n",
      "2866\n",
      "aa_pos_aa l209f\n",
      "2867\n",
      "First match !  g602r\n",
      "2868\n",
      "First match !  i2285v\n",
      "2869\n",
      "pos weak s326r\n",
      "2870\n",
      "First match !  v2969m\n",
      "2871\n",
      "First match !  r2659k\n",
      "2872\n",
      "First match !  l2865v\n",
      "2873\n",
      "2874\n",
      "2875\n",
      "2876\n",
      "First match !  s869l\n",
      "2877\n",
      "First match !  k513r\n",
      "2878\n",
      "First match !  n56t\n",
      "2879\n",
      "First match !  w2626c\n",
      "2880\n",
      "aa_pos_aa s206c\n",
      "2881\n",
      "2882\n",
      "First match !  d2723g\n",
      "2883\n",
      "First match !  t2722r\n",
      "2884\n",
      "First match !  h2074n\n",
      "2885\n",
      "First match !  g2274v\n",
      "2886\n",
      "2887\n",
      "First match !  p375s\n",
      "2888\n",
      "aa_pos_aa r174c\n",
      "2889\n",
      "First match !  a75p\n",
      "2890\n",
      "First match !  n2113s\n",
      "2891\n",
      "First match !  d1280v\n",
      "2892\n",
      "2893\n",
      "First match !  r3052q\n",
      "2894\n",
      "First match !  s2670l\n",
      "2895\n",
      "2896\n",
      "First match !  q1396r\n",
      "2897\n",
      "First match !  k607t\n",
      "2898\n",
      "First match !  g1529r\n",
      "2899\n",
      "First match !  t352m\n",
      "2900\n",
      "2901\n",
      "First match !  k413e\n",
      "2902\n",
      "First match !  l46r\n",
      "2903\n",
      "First match !  a211d\n",
      "2904\n",
      "First match !  truncating mutations\n",
      "2905\n",
      "First match !  l535p\n",
      "2906\n",
      "2907\n",
      "First match !  k79e\n",
      "2908\n",
      "2909\n",
      "First match !  l64p\n",
      "2910\n",
      "pax8 ppar pax8-ppar? fusion pax8-ppar? fusion\n",
      "Fusion gene1 | gene2 pax8-ppar? fusion\n",
      "2911\n",
      "First match !  e79k\n",
      "2912\n",
      "First match !  e82g\n",
      "2913\n",
      "y371 aa_pos t80a\n",
      "2914\n",
      "First match !  e79q\n",
      "2915\n",
      "pos weak g81r\n",
      "2916\n",
      "y371 aa_pos w24r\n",
      "2917\n",
      "First match !  l30f\n",
      "2918\n",
      "First match !  w24c\n",
      "2919\n",
      "pos weak g81d\n",
      "2920\n",
      "First match !  t80k\n",
      "2921\n",
      "y371 aa_pos v32g\n",
      "2922\n",
      "y371 aa_pos d29y\n",
      "2923\n",
      "First match !  g31a\n",
      "2924\n",
      "y371 aa_pos d29h\n",
      "2925\n",
      "pos weak g81s\n",
      "2926\n",
      "y371 aa_pos i28t\n",
      "2927\n",
      "First match !  g31v\n",
      "2928\n",
      "First match !  g31r\n",
      "2929\n",
      "First match !  e82v\n",
      "2930\n",
      "First match !  e82d\n",
      "2931\n",
      "y371 aa_pos t80r\n",
      "2932\n",
      "y371 aa_pos w24s\n",
      "2933\n",
      "y371 aa_pos r380a\n",
      "2934\n",
      "First match !  truncating mutations\n",
      "2935\n",
      "y371 aa_pos r544s\n",
      "2936\n",
      "First match !  c481s\n",
      "2937\n",
      "First match !  y551f\n",
      "2938\n",
      "2939\n",
      "y371 aa_pos s36y\n",
      "2940\n",
      "First match !  m37k\n",
      "2941\n",
      "First match !  s32i\n",
      "2942\n",
      "First match !  q209l\n",
      "2943\n",
      "First match !  amplification\n",
      "2944\n",
      "First match !  c482r\n",
      "2945\n",
      "First match !  t771r\n",
      "2946\n",
      "First match !  r961w\n",
      "2947\n",
      "First match !  d717v\n",
      "2948\n",
      "First match !  a1065t\n",
      "2949\n",
      "First match !  q472h\n",
      "2950\n",
      "First match !  deletion\n",
      "2951\n",
      "First match !  truncating mutations\n",
      "2952\n",
      "First match !  q209l\n",
      "2953\n",
      "First match !  r844h\n",
      "2954\n",
      "First match !  q227l\n",
      "2955\n",
      "First match !  q227r\n",
      "2956\n",
      "First match !  r201h\n",
      "2957\n",
      "First match !  r201c\n",
      "2958\n",
      "First match !  deletion\n",
      "2959\n",
      "2960\n",
      "First match !  d816e\n",
      "2961\n",
      "First match !  k558n\n",
      "2962\n",
      "2963\n",
      "First match !  d816f\n",
      "2964\n",
      "First match !  a829p\n",
      "2965\n",
      "First match !  d816n\n",
      "2966\n",
      "2967\n",
      "First match !  t670i\n",
      "2968\n",
      "First match !  e839k\n",
      "2969\n",
      "First match !  k642e\n",
      "2970\n",
      "2971\n",
      "2972\n",
      "First match !  n822h\n",
      "2973\n",
      "First match !  y578c\n",
      "2974\n",
      "2975\n",
      "First match !  c809g\n",
      "2976\n",
      "2977\n",
      "2978\n",
      "2979\n",
      "First match !  h697y\n",
      "2980\n",
      "2981\n",
      "First match !  d816y\n",
      "2982\n",
      "aa_pos_aa f522c\n",
      "2983\n",
      "First match !  c443y\n",
      "2984\n",
      "2985\n",
      "First match !  w557r\n",
      "2986\n",
      "First match !  v559c\n",
      "2987\n",
      "2988\n",
      "First match !  m541l\n",
      "2989\n",
      "2990\n",
      "First match !  d820g\n",
      "2991\n",
      "2992\n",
      "2993\n",
      "2994\n",
      "First match !  d419del\n",
      "2995\n",
      "First match !  amplification\n",
      "2996\n",
      "2997\n",
      "2998\n",
      "2999\n",
      "First match !  k509i\n",
      "3000\n",
      "y371 aa_pos w557g\n",
      "3001\n",
      "First match !  l576p\n",
      "3002\n",
      "3003\n",
      "3004\n",
      "First match !  v559d\n",
      "3005\n",
      "First match !  d816v\n",
      "3006\n",
      "First match !  q556_k558del\n",
      "3007\n",
      "3008\n",
      "First match !  exon 11 mutations\n",
      "3009\n",
      "First match !  v560g\n",
      "3010\n",
      "y371 aa_pos d814v\n",
      "3011\n",
      "First match !  d816h\n",
      "3012\n",
      "First match !  n822y\n",
      "3013\n",
      "pos weak i653t\n",
      "3014\n",
      "pos weak n463s\n",
      "3015\n",
      "3016\n",
      "First match !  d579del\n",
      "3017\n",
      "First match !  v560d\n",
      "3018\n",
      "First match !  d737n\n",
      "3019\n",
      "3020\n",
      "First match !  v559g\n",
      "3021\n",
      "y371 aa_pos d816g\n",
      "3022\n",
      "First match !  a502_y503dup\n",
      "3023\n",
      "First match !  d820e\n",
      "3024\n",
      "3025\n",
      "First match !  exon 9 mutations\n",
      "3026\n",
      "First match !  s476i\n",
      "3027\n",
      "3028\n",
      "First match !  n505i\n",
      "3029\n",
      "First match !  y553n\n",
      "3030\n",
      "3031\n",
      "First match !  d572a\n",
      "3032\n",
      "First match !  n822i\n",
      "3033\n",
      "3034\n",
      "First match !  d60n\n",
      "3035\n",
      "3036\n",
      "3037\n",
      "3038\n",
      "3039\n",
      "3040\n",
      "3041\n",
      "First match !  d820a\n",
      "3042\n",
      "3043\n",
      "First match !  y823d\n",
      "3044\n",
      "3045\n",
      "3046\n",
      "pos weak y570h\n",
      "3047\n",
      "First match !  r796g\n",
      "3048\n",
      "3049\n",
      "First match !  v560e\n",
      "3050\n",
      "3051\n",
      "First match !  v654a\n",
      "3052\n",
      "First match !  e490k\n",
      "3053\n",
      "First match !  r634w\n",
      "3054\n",
      "3055\n",
      "3056\n",
      "y371 aa_pos d816a\n",
      "3057\n",
      "First match !  v559a\n",
      "3058\n",
      "3059\n",
      "First match !  amplification\n",
      "3060\n",
      "First match !  exon 1 mutations\n",
      "3061\n",
      "First match !  g44s\n",
      "3062\n",
      "First match !  exon 2 mutations\n",
      "3063\n",
      "First match !  l1224f\n",
      "3064\n",
      "3065\n",
      "pos weak i1616n\n",
      "3066\n",
      "3067\n",
      "pos weak r1598p\n",
      "3068\n",
      "pos l1574p\n",
      "3069\n",
      "3070\n",
      "pos weak l1596h\n",
      "3071\n",
      "pos weak l1600p\n",
      "3072\n",
      "3073\n",
      "3074\n",
      "pos weak i1680n\n",
      "3075\n",
      "pos l1593p\n",
      "3076\n",
      "pos weak a1701p\n",
      "3077\n",
      "3078\n",
      "3079\n",
      "pos weak l1678p\n",
      "3080\n",
      "pos weak r1608s\n",
      "3081\n",
      "First match !  fusions\n",
      "3082\n",
      "pos weak i1616t\n",
      "3083\n",
      "pos weak v1676d\n",
      "3084\n",
      "3085\n",
      "3086\n",
      "3087\n",
      "First match !  r1594q\n",
      "3088\n",
      "FUSION sec16a1-notch1 fusion\n",
      "3089\n",
      "pos weak g2430a\n",
      "3090\n",
      "pos weak f1592s\n",
      "3091\n",
      "3092\n",
      "pos weak v1576e\n",
      "3093\n",
      "FUSION mir143-notch1 fusion\n",
      "3094\n",
      "First match !  v1671i\n",
      "3095\n",
      "3096\n",
      "3097\n",
      "3098\n",
      "First match !  c157y\n",
      "3099\n",
      "First match !  p29s\n",
      "3100\n",
      "First match !  q61r\n",
      "3101\n",
      "First match !  c18y\n",
      "3102\n",
      "First match !  amplification\n",
      "3103\n",
      "First match !  g12v\n",
      "3104\n",
      "First match !  p29l\n",
      "3105\n",
      "First match !  r882c\n",
      "3106\n",
      "First match !  deletion\n",
      "3107\n",
      "First match !  truncating mutations\n",
      "3108\n",
      "First match !  amplification\n",
      "3109\n",
      "First match !  dnmt3b7\n",
      "3110\n",
      "First match !  deletion\n",
      "3111\n",
      "3112\n",
      "3113\n",
      "First match !  g264s\n",
      "3114\n",
      "FUSION hmga2-rad51b fusion\n",
      "3115\n",
      "First match !  truncating mutations\n",
      "3116\n",
      "First match !  g13e\n",
      "3117\n",
      "First match !  q22k\n",
      "3118\n",
      "First match !  q61h\n",
      "3119\n",
      "First match !  a11_g12insga\n",
      "3120\n",
      "First match !  g12f\n",
      "3121\n",
      "First match !  l19f\n",
      "3122\n",
      "First match !  g12d\n",
      "3123\n",
      "First match !  f156l\n",
      "3124\n",
      "First match !  y32s\n",
      "3125\n",
      "First match !  v14i\n",
      "3126\n",
      "First match !  y64a\n",
      "3127\n",
      "First match !  n116h\n",
      "3128\n",
      "First match !  e31k\n",
      "3129\n",
      "First match !  g60d\n",
      "3130\n",
      "First match !  t74p\n",
      "3131\n",
      "First match !  r164q\n",
      "3132\n",
      "First match !  g60r\n",
      "3133\n",
      "First match !  a146t\n",
      "3134\n",
      "First match !  g13c\n",
      "3135\n",
      "3136\n",
      "First match !  a146v\n",
      "3137\n",
      "First match !  p34l\n",
      "3138\n",
      "First match !  k147e\n",
      "3139\n",
      "First match !  g12r\n",
      "3140\n",
      "First match !  q61l\n",
      "3141\n",
      "First match !  t58i\n",
      "3142\n",
      "First match !  g12s\n",
      "3143\n",
      "First match !  k5n\n",
      "3144\n",
      "First match !  a18d\n",
      "3145\n",
      "First match !  s65n\n",
      "3146\n",
      "First match !  d153v\n",
      "3147\n",
      "First match !  g12a\n",
      "3148\n",
      "First match !  p34r\n",
      "3149\n",
      "First match !  k117n\n",
      "3150\n",
      "First match !  g12v\n",
      "3151\n",
      "First match !  amplification\n",
      "3152\n",
      "First match !  y40a\n",
      "3153\n",
      "First match !  q61r\n",
      "3154\n",
      "First match !  g12c\n",
      "3155\n",
      "First match !  d119n\n",
      "3156\n",
      "First match !  a59g\n",
      "3157\n",
      "First match !  f28l\n",
      "3158\n",
      "First match !  q22r\n",
      "3159\n",
      "First match !  q22e\n",
      "3160\n",
      "3161\n",
      "First match !  deletion\n",
      "3162\n",
      "First match !  fusions\n",
      "3163\n",
      "y371 aa_pos s259p\n",
      "3164\n",
      "First match !  esrp1-raf1 fusion\n",
      "3165\n",
      "First match !  i448v\n",
      "3166\n",
      "FUSION atg7-raf1 fusion\n",
      "3167\n",
      "First match !  s259f\n",
      "3168\n",
      "First match !  amplification\n",
      "3169\n",
      "First match !  s427g\n",
      "3170\n",
      "FUSION srgap3-raf1 fusion\n",
      "3171\n",
      "First match !  p261l\n",
      "3172\n",
      "y371 aa_pos s257w\n",
      "3173\n",
      "First match !  s259a\n",
      "3174\n",
      "First match !  f161l\n",
      "3175\n",
      "First match !  a151t\n",
      "3176\n",
      "First match !  t196a\n",
      "3177\n",
      "First match !  truncating mutations\n",
      "3178\n",
      "3179\n",
      "First match !  magi3-akt3 fusion\n",
      "3180\n",
      "First match !  e17k\n",
      "3181\n",
      "First match !  amplification\n",
      "3182\n",
      "First match !  fusions\n",
      "3183\n",
      "First match !  n551k\n",
      "3184\n",
      "First match !  truncating mutations\n",
      "3185\n",
      "First match !  r276w\n",
      "3186\n",
      "First match !  amplification\n",
      "3187\n",
      "First match !  t58i\n",
      "3188\n",
      "First match !  q61k\n",
      "3189\n",
      "First match !  g13v\n",
      "3190\n",
      "y371 aa_pos q61r\n",
      "3191\n",
      "First match !  g60e\n",
      "3192\n",
      "First match !  g13d\n",
      "3193\n",
      "First match !  g12v\n",
      "3194\n",
      "First match !  q61h\n",
      "3195\n",
      "First match !  t50i\n",
      "3196\n",
      "First match !  g12d\n",
      "3197\n",
      "First match !  q61l\n",
      "3198\n",
      "First match !  l481f\n",
      "3199\n",
      "First match !  y472h\n",
      "3200\n",
      "3201\n",
      "First match !  deletion\n",
      "3202\n",
      "3203\n",
      "First match !  deletion\n",
      "3204\n",
      "aa_pos_aa l607i\n",
      "3205\n",
      "aa_pos_aa r698w\n",
      "3206\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "First match !  r661w\n",
      "3207\n",
      "First match !  w563l\n",
      "3208\n",
      "First match !  c712r\n",
      "3209\n",
      "3210\n",
      "First match !  s567l\n",
      "3211\n",
      "First match !  c706f\n",
      "3212\n",
      "First match !  truncating mutations\n",
      "3213\n",
      "First match !  deletion\n",
      "3214\n",
      "First match !  amplification\n",
      "3215\n",
      "FUSION tpr-ntrk1 fusion\n",
      "3216\n",
      "First match !  v710a\n",
      "3217\n",
      "pos r342w\n",
      "3218\n",
      "First match !  fusions\n",
      "3219\n",
      "First match !  bcan-ntrk1 fusion\n",
      "3220\n",
      "First match !  nfasc-ntrk1 fusion\n",
      "3221\n",
      "First match !  trkaiii splice variant\n",
      "3222\n",
      "pos r342q\n",
      "3223\n",
      "FUSION delta-ntrk1 fusion\n",
      "3224\n",
      "First match !  cd74-ntrk1 fusion\n",
      "3225\n",
      "First match !  tpm3-ntrk1 fusion\n",
      "3226\n",
      "FUSION chtop-ntrk1 fusion\n",
      "3227\n",
      "First match !  amplification\n",
      "3228\n",
      "FUSION tfg-ntrk1 fusion\n",
      "3229\n",
      "First match !  lmna-ntrk1 fusion\n",
      "3230\n",
      "FUSION mprip-ntrk1 fusion\n",
      "3231\n",
      "First match !  m713i\n",
      "3232\n",
      "First match !  fusions\n",
      "3233\n",
      "First match !  r715g\n",
      "3234\n",
      "First match !  etv6-ntrk3 fusion\n",
      "3235\n",
      "First match !  btbd1-ntrk3 fusion\n",
      "3236\n",
      "First match !  fusions\n",
      "3237\n",
      "FUSION cpeb1-ntrk3 fusion\n",
      "3238\n",
      "First match !  g623r\n",
      "3239\n",
      "First match !  deletion\n",
      "3240\n",
      "First match !  l239r\n",
      "3241\n",
      "First match !  t654m\n",
      "3242\n",
      "3243\n",
      "First match !  l63v\n",
      "3244\n",
      "First match !  g505s\n",
      "3245\n",
      "y371 aa_pos g774v\n",
      "3246\n",
      "First match !  i638f\n",
      "3247\n",
      "First match !  amplification\n",
      "3248\n",
      "First match !  t654i\n",
      "3249\n",
      "First match !  fusions\n",
      "3250\n",
      "First match !  nsd1-nup98 fusion\n",
      "3251\n",
      "First match !  g325a\n",
      "3252\n",
      "3253\n",
      "First match !  deletion\n",
      "3254\n",
      "3255\n",
      "pos c248t\n",
      "3256\n",
      "3257\n",
      "FUSION ret-ccdc6 fusion\n",
      "3258\n",
      "First match !  a883t\n",
      "3259\n",
      "First match !  c609y\n",
      "3260\n",
      "First match !  t338i\n",
      "3261\n",
      "pos c634r\n",
      "3262\n",
      "First match !  s891a\n",
      "3263\n",
      "3264\n",
      "First match !  c618r\n",
      "3265\n",
      "pos c611y\n",
      "3266\n",
      "First match !  k603q\n",
      "3267\n",
      "First match !  d631g\n",
      "3268\n",
      "First match !  e921k\n",
      "3269\n",
      "First match !  fusions\n",
      "3270\n",
      "First match !  s904f\n",
      "3271\n",
      "First match !  l790f\n",
      "3272\n",
      "First match !  r873q\n",
      "3273\n",
      "First match !  y806c\n",
      "3274\n",
      "First match !  s765p\n",
      "3275\n",
      "First match !  v804l\n",
      "3276\n",
      "First match !  r897q\n",
      "3277\n",
      "First match !  d631a\n",
      "3278\n",
      "First match !  a883f\n",
      "3279\n",
      "First match !  c634s\n",
      "3280\n",
      "First match !  a919v\n",
      "3281\n",
      "First match !  c634w\n",
      "3282\n",
      "First match !  m980t\n",
      "3283\n",
      "First match !  e768d\n",
      "3284\n",
      "First match !  kif5b-ret fusion\n",
      "3285\n",
      "pos weak r833c\n",
      "3286\n",
      "First match !  v804m\n",
      "3287\n",
      "First match !  m918v\n",
      "3288\n",
      "pos c630r\n",
      "3289\n",
      "First match !  v648i\n",
      "3290\n",
      "First match !  i852m\n",
      "3291\n",
      "First match !  c620r\n",
      "3292\n",
      "pos c634y\n",
      "3293\n",
      "First match !  v804g\n",
      "3294\n",
      "aa_pos_aa r886w\n",
      "3295\n",
      "First match !  f893l\n",
      "3296\n",
      "First match !  y791f\n",
      "3297\n",
      "First match !  r177*\n",
      "3298\n",
      "3299\n",
      "First match !  r139g\n",
      "3300\n",
      "First match !  k83n\n",
      "3301\n",
      "First match !  r177q\n",
      "3302\n",
      "First match !  r166q\n",
      "3303\n",
      "pos weak p173s\n",
      "3304\n",
      "First match !  r201q\n",
      "3305\n",
      "3306\n",
      "First match !  w279*\n",
      "3307\n",
      "3308\n",
      "First match !  r174*\n",
      "3309\n",
      "First match !  d171g\n",
      "3310\n",
      "First match !  amplification\n",
      "3311\n",
      "First match !  runx1-evi1 fusion\n",
      "3312\n",
      "FUSION tel-runx1 fusion\n",
      "3313\n",
      "First match !  h78q\n",
      "3314\n",
      "aa_pos_aa g42r\n",
      "3315\n",
      "First match !  runx1-runx1t1 fusion\n",
      "3316\n",
      "First match !  d171n\n",
      "3317\n",
      "3318\n",
      "First match !  fusions\n",
      "3319\n",
      "First match !  r80c\n",
      "3320\n",
      "First match !  k83e\n",
      "done in 329.1251871585846 s\n"
     ]
    }
   ],
   "source": [
    "start = time.time()\n",
    "clean_match_data = concatenate_data.apply(lambda x: extract_match(x), axis = 1)\n",
    "clean_match = pd.DataFrame(list(clean_match_data), columns = [\"Text\",\"Score\"], index = clean_match_data.index)\n",
    "print(\"done in {} s\".format(time.time() - start))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ID\n",
       "0                                                 ([], 0)\n",
       "1       ( the c-cbl loh also positively correlated wit...\n",
       "2       ( the c-cbl loh also positively correlated wit...\n",
       "3       ( most of the changes were novel, although 4 c...\n",
       "4       ( all mutations from the second group were pre...\n",
       "                              ...                        \n",
       "3316    (2129 the vast majority of aml1 mutati...\n",
       "3317                                              ([], 0)\n",
       "3318    ( lpxn is preferentially expressed in hematopo...\n",
       "3319    () conversely, mutations mapping to the first ...\n",
       "3320    ( mutation analysis in fpd/aml pedigrees. elec...\n",
       "Length: 3316, dtype: object"
      ]
     },
     "execution_count": 247,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clean_match_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Text</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3263</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3298</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3305</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3307</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>402 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Text  Score\n",
       "ID              \n",
       "0      []    0.0\n",
       "16     []    0.0\n",
       "46     []    0.0\n",
       "61     []    0.0\n",
       "71     []    0.0\n",
       "...   ...    ...\n",
       "3263   []    0.0\n",
       "3298   []    0.0\n",
       "3305   []    0.0\n",
       "3307   []    0.0\n",
       "3317   []    0.0\n",
       "\n",
       "[402 rows x 2 columns]"
      ]
     },
     "execution_count": 248,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clean_match[clean_match[\"Score\"] == 0]"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "# import numpy as np\n",
    "CBL = train_variant[train_variant[\"Gene\"] == \"RUNX1\"]\n",
    "article_cbl = train_text.loc[CBL.index,]\n",
    "\n",
    "CBL.to_csv(\"cbl.txt\")\n",
    "dtf = pd.merge(pd.DataFrame(clean_article_CBL.index), clean_article_CBL, on =\"ID\")\n",
    "np.savetxt(\"cbl_clean_article.txt\",dtf, fmt = \"%d||%s\", header= \",\".join(dtf.columns), comments='')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "metadata": {},
   "outputs": [],
   "source": [
    "clean_match.index.name = \"ID\"\n",
    "new_data = pd.merge(concatenate_data,clean_match, on = \"ID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 250,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "      <td>most of the changes were novel, although 4 ca...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "      <td>all mutations from the second group were pred...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>CBL</td>\n",
       "      <td>V391I</td>\n",
       "      <td>4</td>\n",
       "      <td>all mutations from the second group were pred...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3315</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>RUNX1-RUNX1T1 Fusion</td>\n",
       "      <td>4</td>\n",
       "      <td>this truncated protein contained the rhd but ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>D171N</td>\n",
       "      <td>4</td>\n",
       "      <td>2129 the vast majority of aml1 mutatio...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>Fusions</td>\n",
       "      <td>1</td>\n",
       "      <td>lpxn is preferentially expressed in hematopoi...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>R80C</td>\n",
       "      <td>4</td>\n",
       "      <td>) conversely, mutations mapping to the first d...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>K83E</td>\n",
       "      <td>4</td>\n",
       "      <td>mutation analysis in fpd/aml pedigrees. elect...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2435 rows  5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene             Variation  Class  \\\n",
       "ID                                         \n",
       "1       CBL                 W802*      2   \n",
       "2       CBL                 Q249E      2   \n",
       "3       CBL                 N454D      3   \n",
       "4       CBL                 L399V      4   \n",
       "5       CBL                 V391I      4   \n",
       "...     ...                   ...    ...   \n",
       "3315  RUNX1  RUNX1-RUNX1T1 Fusion      4   \n",
       "3316  RUNX1                 D171N      4   \n",
       "3318  RUNX1               Fusions      1   \n",
       "3319  RUNX1                  R80C      4   \n",
       "3320  RUNX1                  K83E      4   \n",
       "\n",
       "                                                   Text  Score  \n",
       "ID                                                              \n",
       "1      the c-cbl loh also positively correlated with...    1.0  \n",
       "2      the c-cbl loh also positively correlated with...    1.0  \n",
       "3      most of the changes were novel, although 4 ca...    1.0  \n",
       "4      all mutations from the second group were pred...    1.0  \n",
       "5      all mutations from the second group were pred...    1.0  \n",
       "...                                                 ...    ...  \n",
       "3315   this truncated protein contained the rhd but ...    1.0  \n",
       "3316  2129 the vast majority of aml1 mutatio...    1.0  \n",
       "3318   lpxn is preferentially expressed in hematopoi...    1.0  \n",
       "3319  ) conversely, mutations mapping to the first d...    1.0  \n",
       "3320   mutation analysis in fpd/aml pedigrees. elect...    1.0  \n",
       "\n",
       "[2435 rows x 5 columns]"
      ]
     },
     "execution_count": 250,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data[new_data[\"Score\"] == 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3297</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>R177*</td>\n",
       "      <td>4</td>\n",
       "      <td>the mutation was present in all affected indi...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3298</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>Y113*</td>\n",
       "      <td>4</td>\n",
       "      <td>2129 the vast majority of aml1 mutatio...</td>\n",
       "      <td>5.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3299</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>R139G</td>\n",
       "      <td>4</td>\n",
       "      <td>the mutation was present in all affected indi...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3300</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>K83N</td>\n",
       "      <td>4</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3301</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>R177Q</td>\n",
       "      <td>4</td>\n",
       "      <td>because of his history of thrombocytopenia s...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3302</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>R166Q</td>\n",
       "      <td>4</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3303</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>P173S</td>\n",
       "      <td>4</td>\n",
       "      <td>familial platelet disorder with propensity to...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3304</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>R201Q</td>\n",
       "      <td>4</td>\n",
       "      <td>to explore the underlying mechanisms for bc, ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3305</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>S70fsX93</td>\n",
       "      <td>4</td>\n",
       "      <td>com/reviews/cancer  2015 macmillan publishers...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3306</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>W279*</td>\n",
       "      <td>1</td>\n",
       "      <td>(13) in the chromosomal joining region of t(3...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3307</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>balanced chromosomal translocations are freque...</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3308</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>R174*</td>\n",
       "      <td>7</td>\n",
       "      <td>to explore the underlying mechanisms for bc, ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3309</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>D171G</td>\n",
       "      <td>1</td>\n",
       "      <td>, patient number; sct, stem cell transplantati...</td>\n",
       "      <td>1.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3310</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>Amplification</td>\n",
       "      <td>7</td>\n",
       "      <td>this truncated protein conta the rhd but not ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3311</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>RUNX1-EVI1 Fusion</td>\n",
       "      <td>4</td>\n",
       "      <td>2129 the vast majority of aml1 mutatio...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3312</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>TEL-RUNX1 Fusion</td>\n",
       "      <td>4</td>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3313</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>H78Q</td>\n",
       "      <td>4</td>\n",
       "      <td>lpxn is preferentially expressed in hematopoi...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3314</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>G42R</td>\n",
       "      <td>6</td>\n",
       "      <td>) conversely, mutations mapping to the first d...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3315</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>RUNX1-RUNX1T1 Fusion</td>\n",
       "      <td>4</td>\n",
       "      <td>mutation analysis in fpd/aml pedigrees. elect...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene             Variation  Class  \\\n",
       "ID                                         \n",
       "3297  RUNX1                 R177*      4   \n",
       "3298  RUNX1                 Y113*      4   \n",
       "3299  RUNX1                 R139G      4   \n",
       "3300  RUNX1                  K83N      4   \n",
       "3301  RUNX1                 R177Q      4   \n",
       "3302  RUNX1                 R166Q      4   \n",
       "3303  RUNX1                 P173S      4   \n",
       "3304  RUNX1                 R201Q      4   \n",
       "3305  RUNX1              S70fsX93      4   \n",
       "3306  RUNX1                 W279*      1   \n",
       "3307  RUNX1  Truncating Mutations      1   \n",
       "3308  RUNX1                 R174*      7   \n",
       "3309  RUNX1                 D171G      1   \n",
       "3310  RUNX1         Amplification      7   \n",
       "3311  RUNX1     RUNX1-EVI1 Fusion      4   \n",
       "3312  RUNX1      TEL-RUNX1 Fusion      4   \n",
       "3313  RUNX1                  H78Q      4   \n",
       "3314  RUNX1                  G42R      6   \n",
       "3315  RUNX1  RUNX1-RUNX1T1 Fusion      4   \n",
       "\n",
       "                                                   Text  Score  \n",
       "ID                                                              \n",
       "3297   the mutation was present in all affected indi...    1.0  \n",
       "3298  2129 the vast majority of aml1 mutatio...    5.0  \n",
       "3299   the mutation was present in all affected indi...    1.0  \n",
       "3300                                                 []    0.0  \n",
       "3301    because of his history of thrombocytopenia s...    1.0  \n",
       "3302                                                 []    0.0  \n",
       "3303   familial platelet disorder with propensity to...    1.0  \n",
       "3304   to explore the underlying mechanisms for bc, ...    1.0  \n",
       "3305  com/reviews/cancer  2015 macmillan publishers...    1.0  \n",
       "3306  (13) in the chromosomal joining region of t(3...    1.0  \n",
       "3307  balanced chromosomal translocations are freque...    4.0  \n",
       "3308   to explore the underlying mechanisms for bc, ...    1.0  \n",
       "3309  , patient number; sct, stem cell transplantati...    1.1  \n",
       "3310   this truncated protein conta the rhd but not ...    1.0  \n",
       "3311  2129 the vast majority of aml1 mutatio...    1.0  \n",
       "3312                                                 []    0.0  \n",
       "3313   lpxn is preferentially expressed in hematopoi...    1.0  \n",
       "3314  ) conversely, mutations mapping to the first d...    1.0  \n",
       "3315   mutation analysis in fpd/aml pedigrees. elect...    1.0  "
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data[new_data[\"Gene\"] == \"RUNX1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "' mutation analysis in fpd/aml pedigrees. electropherograms from affected and control individuals showing the a>g substitution leading to the k83e missense mutation in pedigree 1 (a), the c>a substitution resulting in a nonsense mutation y260x in pedigree 3 (b), and the deletion of an a at the +3 position of the spldonor sof intron 4 in pedigree 2 (ivs4\\u2009+\\u20093dela, c).  arrows indicate the point of the mutations. extended haplotype analysis of each family indicated an approximate 3 megabase common nonrecombinant interval on human chromosome 21q containing runx1 in affected individuals. pedigree 1 showed an a>g substitution in exon 3 resulting in a missense mutation, k83e (figure 2a).  the mutation segregates with the disorder in the family and was absent in 188 unrelated control chromosomes. the mutation segregated with the dse in all family members tested. effects of runx1 fpd/aml mutations on dna binding and heterodization activities of the runt domainwe exam the function of the runt domain for 2 new (k83e and r135fsx177; swiss-prot q01196) and 5 descr (r139q, r174q, k90fsx101, r174x, r177x)10 fpd/aml point mutations.  on emsa analyses, r139q, r174x, and r177x alone showed barely detectable dna binding but produced a supershift band of eased intensity in the presence of pebp2/cbf, indicating that heterodization of these mutants with pebp2/cbf still occurs (figure3a).  k83e, r135fsx177, r174q, and k90fsx101 showed no detectable dna binding in the presence or absence of pebp/cbf.  an affinity assay of these mutants for heterodization with pebp2/cbf revealed that both missense mutants reta the ability to heterodwhereas both frameshift mutants did not (figure 3b). original magnification,  400. whthe k83e and r139q mutations are not within the known nls, both mutant proteins showed reduced nuclear localization.  k83e showed cytoplasmic localization in 30% of transfected cells, in contrast to the normal nuclear localization seen in k83n (a sporadic mutation; figure 4a). 11 this observation may suggest that the k83e mutation (k, posit hydrophilic, c6; e, negat hydrophilic, c5) results in greater loss of function than the k83n mutation (n, polar, hydrophobic, c4).  the r139q mutation is at a sinvolved in dna binding18 and was expected to affect dna binding rather than nuclear localization. however, cytoplasmic localization was observed in 40% of cells.  the observed defect in nuclear localization in k83e and r139q is consistent with earlier studies31 and may point to the existence of an additional untified n-terminal domain critical for nuclear localization. in contrast to runx1, pebp2/cbf is normally local in the cytoplasm before heterodization32 and can enter the nucleus only with runx however, the regulation of the heterodization remains unknown, and it has been shown that both proteins may have different cellular localizations when transfected in the same cells.11 35 with the exception of r177x, all of the mutants showed varying degrees of interference with the wild-type protein function in all cell l tested (u937, hl60, and jurkat).  most notably, k83e and r174q showed potent inhibition similar to that of pebp2/cbf-myh11 (figure 5b).  the results of cotransfection reporter assays using frameshift mutants are not reproducible within or between cell l.52 preferential binding of mutant runx1 proteins to pebp2/cbf may ease their half-l.  k83e may be the most potent nonchic mutant inhibitor of wild-type runx1 descr so far because k83 is a residue important for ubiquitinylation. mheterozygous for the runx1/aml1-eto or pebp2/cbf-myh11 knocked-ins showed defects in definithematopoiesis similar to those observed in runx1/ or pebp2/cbf/m53 54 thus, expression of strong dominant-negatinhibitors of runx1 seems to produce an embryonic lethal phenotype. this is consistent with the hypothesis that a second mutation has to occur in runx1 or other genes to cause leukemia among individuals harboring an inherrunx1 mutation, and these mutations are more ly to occur in the individuals with lower biologic activity of runx1.  for example, in 2 of the largest fpd/aml pedigrees, a markedly higher rate of leukemia is seen in the family with strong predicted dominant-negatk83e mutation (57%) compared with the pedigree with a complete deletion of runx1 (24%).  6. the ratio of leukemic fpd individuals to fpd-affected individuals in each pedigree is indicated on the right.  in the largest fpd/aml pedigrees, a markedly higher rate of leukemia is seen in the family with strong predicted dominant-negatk83e mutation (pedigree a1, dominant-negat 57%) compared with the pedigree with a complete deletion of runx1 (pedigree 1, haploinsufficiency, 24%). thus, we propose that the less functional pebp/cbf transcriptional regulation complex present in a hematopoietic cell due to variable inhibitory effects of heterozygous mutations, or mutations of both alleles, the higher the propensity to develop leukemia.) conversely, mutations mapping to the first dna interacting region, betap1-loop (betap1-betab), are seen in runx1 but not in runx2 (figure 2) (yoshida , 2003).  these runx1 mutants, r80c, k83n, and k83e showed strong affinity for the beta-subunit and hence a strong dominant negateffect, suggesting that haploinsufficiency is a dominant mechanism in runx2 mutants, whereas a dominant negateffect is important in runx1 mutants.  sequestration of the beta-subunit may be one important mechanism of the dominant negateffect caused by the runx1 mutants. adapted from nagata  (1999)  full figure and legend (307k)  top of page the lower the runx1 activity, the higher the leukemogenicity how does the dominant negateffect mentioned above relate to leukemogenesis? although the lim number of individuals and minimal clinical information evaluated prohibit a robust conclusion, families with mutations acting simply via haploinsufficiency have a lower e of leukemia than families with mutations acting in a dominant-negatfashion.  in two of the largest fpd/aml pedigrees, leukemia occurred at a markedly higher rate in the family with the strong dominant negatk83e mutation (57%) compared to the pedigree with a complete deletion of runx1 (24%) (table 2) (song , 1999; michaud , 2002).  thus, low runx1 activity appears to correlate with high leukemogenicity.'"
      ]
     },
     "execution_count": 74,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data.loc[3315,\"Text\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3315\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "([], 0)"
      ]
     },
     "execution_count": 76,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "extract_match(new_data.loc[3315,])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 251,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "27\n",
      "First match !  y846c\n",
      "28\n",
      "First match !  c228t\n",
      "29\n",
      "First match !  h412y\n",
      "30\n",
      "First match !  h876q\n",
      "31\n",
      "First match !  promoter mutations\n",
      "32\n",
      "First match !  p704s\n",
      "33\n",
      "First match !  amplification\n",
      "34\n",
      "First match !  c250t\n"
     ]
    }
   ],
   "source": [
    "RUNX1 = concatenate_data[concatenate_data[\"Gene\"] == \"TERT\"].apply(lambda x: extract_match(x), axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 252,
   "metadata": {},
   "outputs": [],
   "source": [
    "RUNX1_dtf = pd.DataFrame(list(RUNX1), columns = [\"Text\",\"Score\"], index = RUNX1.index)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ID\n",
       "27    ( in one case the proband is homozygous for a ...\n",
       "28    ( the average sequence coverage at the tert pr...\n",
       "29    ( both families contain compound heterozygotes...\n",
       "30    ( in one case the proband is homozygous for a ...\n",
       "31    ( to determine whether tumor genomes harbor re...\n",
       "32    ( both families contain compound heterozygotes...\n",
       "33    ( among them, acral lentiginous melanoma (alm)...\n",
       "34    ( the average sequence coverage at the tert pr...\n",
       "dtype: object"
      ]
     },
     "execution_count": 253,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "RUNX1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 254,
   "metadata": {},
   "outputs": [],
   "source": [
    "concat_RUNX1 = pd.merge(concatenate_data.loc[RUNX1_dtf.index,],RUNX1_dtf, on =\"ID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 255,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text_x</th>\n",
       "      <th>Text_y</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>TERT</td>\n",
       "      <td>y846c</td>\n",
       "      <td>4</td>\n",
       "      <td>heterozygous mutations in the telomerase compo...</td>\n",
       "      <td>in one case the proband is homozygous for a n...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>TERT</td>\n",
       "      <td>c228t</td>\n",
       "      <td>7</td>\n",
       "      <td>sequencing studies have identified many recurr...</td>\n",
       "      <td>the average sequence coverage at the tert pro...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>TERT</td>\n",
       "      <td>h412y</td>\n",
       "      <td>4</td>\n",
       "      <td>heterozygous mutations in the telomerase compo...</td>\n",
       "      <td>both families contain compound heterozygotes....</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>TERT</td>\n",
       "      <td>h876q</td>\n",
       "      <td>4</td>\n",
       "      <td>heterozygous mutations in the telomerase compo...</td>\n",
       "      <td>in one case the proband is homozygous for a n...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>TERT</td>\n",
       "      <td>promoter mutations</td>\n",
       "      <td>7</td>\n",
       "      <td>sequencing studies have identified many recurr...</td>\n",
       "      <td>to determine whether tumor genomes harbor rec...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>TERT</td>\n",
       "      <td>p704s</td>\n",
       "      <td>4</td>\n",
       "      <td>heterozygous mutations in the telomerase compo...</td>\n",
       "      <td>both families contain compound heterozygotes....</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>TERT</td>\n",
       "      <td>amplification</td>\n",
       "      <td>2</td>\n",
       "      <td>the current world health organization classifi...</td>\n",
       "      <td>among them, acral lentiginous melanoma (alm) ...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>TERT</td>\n",
       "      <td>c250t</td>\n",
       "      <td>7</td>\n",
       "      <td>sequencing studies have identified many recurr...</td>\n",
       "      <td>the average sequence coverage at the tert pro...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Gene           Variation  Class  \\\n",
       "ID                                    \n",
       "27  TERT               y846c      4   \n",
       "28  TERT               c228t      7   \n",
       "29  TERT               h412y      4   \n",
       "30  TERT               h876q      4   \n",
       "31  TERT  promoter mutations      7   \n",
       "32  TERT               p704s      4   \n",
       "33  TERT       amplification      2   \n",
       "34  TERT               c250t      7   \n",
       "\n",
       "                                               Text_x  \\\n",
       "ID                                                      \n",
       "27  heterozygous mutations in the telomerase compo...   \n",
       "28  sequencing studies have identified many recurr...   \n",
       "29  heterozygous mutations in the telomerase compo...   \n",
       "30  heterozygous mutations in the telomerase compo...   \n",
       "31  sequencing studies have identified many recurr...   \n",
       "32  heterozygous mutations in the telomerase compo...   \n",
       "33  the current world health organization classifi...   \n",
       "34  sequencing studies have identified many recurr...   \n",
       "\n",
       "                                               Text_y  Score  \n",
       "ID                                                            \n",
       "27   in one case the proband is homozygous for a n...      1  \n",
       "28   the average sequence coverage at the tert pro...      1  \n",
       "29   both families contain compound heterozygotes....      1  \n",
       "30   in one case the proband is homozygous for a n...      1  \n",
       "31   to determine whether tumor genomes harbor rec...      1  \n",
       "32   both families contain compound heterozygotes....      1  \n",
       "33   among them, acral lentiginous melanoma (alm) ...      1  \n",
       "34   the average sequence coverage at the tert pro...      1  "
      ]
     },
     "execution_count": 255,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concat_RUNX1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_data_best_score =  new_data[new_data[\"Score\"] == 1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 291,
   "metadata": {},
   "outputs": [],
   "source": [
    "count_new = new_data_best_score.apply(lambda line: line[\"Text_y\"].split(\".\")[0] in line[\"Text_x\"].lower(), axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text_x</th>\n",
       "      <th>Text_y</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>118</th>\n",
       "      <td>PBRM1</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>screening for tumor suppressor genes in breast...</td>\n",
       "      <td>screening for tumor suppressor genes in breas...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>330</th>\n",
       "      <td>ROS1</td>\n",
       "      <td>fusions</td>\n",
       "      <td>2</td>\n",
       "      <td>oncogenic gene fusions involving the 3 region...</td>\n",
       "      <td>oncogenic gene fusions involving the 3 regio...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>670</th>\n",
       "      <td>CDKN2A</td>\n",
       "      <td>a148t</td>\n",
       "      <td>6</td>\n",
       "      <td>background: a common missense variant of the c...</td>\n",
       "      <td>background: a common missense variant of the ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>793</th>\n",
       "      <td>ERBB4</td>\n",
       "      <td>fusions</td>\n",
       "      <td>2</td>\n",
       "      <td>purpose: to identify druggable oncogenic fusio...</td>\n",
       "      <td>purpose: to identify druggable oncogenic fusi...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>851</th>\n",
       "      <td>ABL1</td>\n",
       "      <td>e255k</td>\n",
       "      <td>2</td>\n",
       "      <td>the resistance to the tyrosine kinase inhibito...</td>\n",
       "      <td>the resistance to the tyrosine kinase inhibit...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>962</th>\n",
       "      <td>ESR1</td>\n",
       "      <td>fusions</td>\n",
       "      <td>2</td>\n",
       "      <td>the crucial role of recurrent gene fusions in ...</td>\n",
       "      <td>the crucial role of recurrent gene fusions in...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>979</th>\n",
       "      <td>ETV1</td>\n",
       "      <td>fusions</td>\n",
       "      <td>2</td>\n",
       "      <td>gene fusions involving the erythroblast transf...</td>\n",
       "      <td>gene fusions involving the erythroblast trans...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1419</th>\n",
       "      <td>FGFR3</td>\n",
       "      <td>g375c</td>\n",
       "      <td>7</td>\n",
       "      <td>two mutations in fgfr3, g380r and g375c are kn...</td>\n",
       "      <td>two mutations in fgfr3, g380r and g375c are k...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1424</th>\n",
       "      <td>FGFR3</td>\n",
       "      <td>a391e</td>\n",
       "      <td>7</td>\n",
       "      <td>the a391e mutation in fibroblast growth factor...</td>\n",
       "      <td>the a391e mutation in fibroblast growth facto...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1600</th>\n",
       "      <td>VHL</td>\n",
       "      <td>r200w</td>\n",
       "      <td>4</td>\n",
       "      <td>chuvash polycythemia (cp) is a rare congenital...</td>\n",
       "      <td>chuvash polycythemia (cp) is a rare congenita...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1614</th>\n",
       "      <td>VHL</td>\n",
       "      <td>h191d</td>\n",
       "      <td>4</td>\n",
       "      <td>chuvash polycythemia (cp) is a rare congenital...</td>\n",
       "      <td>chuvash polycythemia (cp) is a rare congenita...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1724</th>\n",
       "      <td>APC</td>\n",
       "      <td>n1026s</td>\n",
       "      <td>4</td>\n",
       "      <td>we identified the apc n1026s variant of unknow...</td>\n",
       "      <td>we identified the apc n1026s variant of unkno...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1934</th>\n",
       "      <td>CARD11</td>\n",
       "      <td>c49y</td>\n",
       "      <td>7</td>\n",
       "      <td>three new benta patients sharing the same nove...</td>\n",
       "      <td>three new benta patients sharing the same nov...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2084</th>\n",
       "      <td>MYD88</td>\n",
       "      <td>l265p</td>\n",
       "      <td>7</td>\n",
       "      <td>myd88 l265p is a somatic mutation that has bee...</td>\n",
       "      <td>myd88 l265p is a somatic mutation that has be...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2367</th>\n",
       "      <td>JUN</td>\n",
       "      <td>amplification</td>\n",
       "      <td>7</td>\n",
       "      <td>genomic amplification of c-jun and its upstrea...</td>\n",
       "      <td>genomic amplification of c-jun and its upstre...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2748</th>\n",
       "      <td>BRAF</td>\n",
       "      <td>amplification</td>\n",
       "      <td>7</td>\n",
       "      <td>comparative genomic hybridization (cgh) using ...</td>\n",
       "      <td>comparative genomic hybridization (cgh) using...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2969</th>\n",
       "      <td>KIT</td>\n",
       "      <td>k642e</td>\n",
       "      <td>7</td>\n",
       "      <td>we report the case of a man with an advanced n...</td>\n",
       "      <td>we report the case of a man with an advanced ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3155</th>\n",
       "      <td>KRAS</td>\n",
       "      <td>d119n</td>\n",
       "      <td>2</td>\n",
       "      <td>the d119n mutation of p21ras was prepared by s...</td>\n",
       "      <td>the d119n mutation of p21ras was prepared by ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3180</th>\n",
       "      <td>AKT3</td>\n",
       "      <td>e17k</td>\n",
       "      <td>7</td>\n",
       "      <td>recently, a rare activating mutation of akt1 (...</td>\n",
       "      <td>recently, a rare activating mutation of akt1 ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene             Variation  Class  \\\n",
       "ID                                          \n",
       "118    PBRM1  truncating mutations      1   \n",
       "330     ROS1               fusions      2   \n",
       "670   CDKN2A                 a148t      6   \n",
       "793    ERBB4               fusions      2   \n",
       "851     ABL1                 e255k      2   \n",
       "962     ESR1               fusions      2   \n",
       "979     ETV1               fusions      2   \n",
       "1419   FGFR3                 g375c      7   \n",
       "1424   FGFR3                 a391e      7   \n",
       "1600     VHL                 r200w      4   \n",
       "1614     VHL                 h191d      4   \n",
       "1724     APC                n1026s      4   \n",
       "1934  CARD11                  c49y      7   \n",
       "2084   MYD88                 l265p      7   \n",
       "2367     JUN         amplification      7   \n",
       "2748    BRAF         amplification      7   \n",
       "2969     KIT                 k642e      7   \n",
       "3155    KRAS                 d119n      2   \n",
       "3180    AKT3                  e17k      7   \n",
       "\n",
       "                                                 Text_x  \\\n",
       "ID                                                        \n",
       "118   screening for tumor suppressor genes in breast...   \n",
       "330   oncogenic gene fusions involving the 3 region...   \n",
       "670   background: a common missense variant of the c...   \n",
       "793   purpose: to identify druggable oncogenic fusio...   \n",
       "851   the resistance to the tyrosine kinase inhibito...   \n",
       "962   the crucial role of recurrent gene fusions in ...   \n",
       "979   gene fusions involving the erythroblast transf...   \n",
       "1419  two mutations in fgfr3, g380r and g375c are kn...   \n",
       "1424  the a391e mutation in fibroblast growth factor...   \n",
       "1600  chuvash polycythemia (cp) is a rare congenital...   \n",
       "1614  chuvash polycythemia (cp) is a rare congenital...   \n",
       "1724  we identified the apc n1026s variant of unknow...   \n",
       "1934  three new benta patients sharing the same nove...   \n",
       "2084  myd88 l265p is a somatic mutation that has bee...   \n",
       "2367  genomic amplification of c-jun and its upstrea...   \n",
       "2748  comparative genomic hybridization (cgh) using ...   \n",
       "2969  we report the case of a man with an advanced n...   \n",
       "3155  the d119n mutation of p21ras was prepared by s...   \n",
       "3180  recently, a rare activating mutation of akt1 (...   \n",
       "\n",
       "                                                 Text_y  Score  \n",
       "ID                                                              \n",
       "118    screening for tumor suppressor genes in breas...    1.0  \n",
       "330    oncogenic gene fusions involving the 3 regio...    1.0  \n",
       "670    background: a common missense variant of the ...    1.0  \n",
       "793    purpose: to identify druggable oncogenic fusi...    1.0  \n",
       "851    the resistance to the tyrosine kinase inhibit...    1.0  \n",
       "962    the crucial role of recurrent gene fusions in...    1.0  \n",
       "979    gene fusions involving the erythroblast trans...    1.0  \n",
       "1419   two mutations in fgfr3, g380r and g375c are k...    1.0  \n",
       "1424   the a391e mutation in fibroblast growth facto...    1.0  \n",
       "1600   chuvash polycythemia (cp) is a rare congenita...    1.0  \n",
       "1614   chuvash polycythemia (cp) is a rare congenita...    1.0  \n",
       "1724   we identified the apc n1026s variant of unkno...    1.0  \n",
       "1934   three new benta patients sharing the same nov...    1.0  \n",
       "2084   myd88 l265p is a somatic mutation that has be...    1.0  \n",
       "2367   genomic amplification of c-jun and its upstre...    1.0  \n",
       "2748   comparative genomic hybridization (cgh) using...    1.0  \n",
       "2969   we report the case of a man with an advanced ...    1.0  \n",
       "3155   the d119n mutation of p21ras was prepared by ...    1.0  \n",
       "3180   recently, a rare activating mutation of akt1 ...    1.0  "
      ]
     },
     "execution_count": 292,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data.loc[count_new[count_new == False].index,]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 300,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text_x</th>\n",
       "      <th>Text_y</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>697</th>\n",
       "      <td>CDKN2B</td>\n",
       "      <td>e35*</td>\n",
       "      <td>4</td>\n",
       "      <td>familial renal cell carcinoma (rcc) is genetic...</td>\n",
       "      <td>256g&gt;a asp86asn (d86n) and evolutionary conse...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1074</th>\n",
       "      <td>FOXA1</td>\n",
       "      <td>s453fs*</td>\n",
       "      <td>2</td>\n",
       "      <td>characterization of the prostate cancer transc...</td>\n",
       "      <td>together, our data suggest that aberrations i...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1302</th>\n",
       "      <td>MLH1</td>\n",
       "      <td>r100*</td>\n",
       "      <td>4</td>\n",
       "      <td>large-scale screening for germ-line mutations ...</td>\n",
       "      <td>in f73, among 34 randomly selected mlh1 cdna ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1326</th>\n",
       "      <td>MLH1</td>\n",
       "      <td>r487*</td>\n",
       "      <td>1</td>\n",
       "      <td>identification of a high-risk disease-causing ...</td>\n",
       "      <td>nc = mutation was not categorized as effect i...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1565</th>\n",
       "      <td>ALK</td>\n",
       "      <td>r1464*</td>\n",
       "      <td>3</td>\n",
       "      <td>in the era of personalized medicine, understan...</td>\n",
       "      <td>in order to treat patients accordingly requir...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1695</th>\n",
       "      <td>PMS2</td>\n",
       "      <td>r802*</td>\n",
       "      <td>1</td>\n",
       "      <td>identification of a high-risk disease-causing ...</td>\n",
       "      <td>this syndrome is characterized by caf-au-lai...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1897</th>\n",
       "      <td>MTOR</td>\n",
       "      <td>r2505*</td>\n",
       "      <td>3</td>\n",
       "      <td>genes encoding components of the pi3k-akt-mtor...</td>\n",
       "      <td>however, few cancer-associated mutations have...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1953</th>\n",
       "      <td>ATM</td>\n",
       "      <td>e1978*</td>\n",
       "      <td>1</td>\n",
       "      <td>over 1,000 reports have been published during ...</td>\n",
       "      <td>these meta-analyses included an average of  s...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2548</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>y1853*</td>\n",
       "      <td>4</td>\n",
       "      <td>abstract germline mutations that inactivate th...</td>\n",
       "      <td>among these mutations, there was an approxima...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3297</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>r177*</td>\n",
       "      <td>4</td>\n",
       "      <td>the aml1 gene is known as the most frequent ta...</td>\n",
       "      <td>they contained three major groups with distin...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3306</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>w279*</td>\n",
       "      <td>1</td>\n",
       "      <td>here we report two new runx1 mutations in one ...</td>\n",
       "      <td>because of his history of thrombocytopenia s...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3308</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>r174*</td>\n",
       "      <td>7</td>\n",
       "      <td>familial platelet disorder with propensity to...</td>\n",
       "      <td>familial platelet disorder with propensity to...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene Variation  Class  \\\n",
       "ID                              \n",
       "697   CDKN2B      e35*      4   \n",
       "1074   FOXA1   s453fs*      2   \n",
       "1302    MLH1     r100*      4   \n",
       "1326    MLH1     r487*      1   \n",
       "1565     ALK    r1464*      3   \n",
       "1695    PMS2     r802*      1   \n",
       "1897    MTOR    r2505*      3   \n",
       "1953     ATM    e1978*      1   \n",
       "2548   BRCA1    y1853*      4   \n",
       "3297   RUNX1     r177*      4   \n",
       "3306   RUNX1     w279*      1   \n",
       "3308   RUNX1     r174*      7   \n",
       "\n",
       "                                                 Text_x  \\\n",
       "ID                                                        \n",
       "697   familial renal cell carcinoma (rcc) is genetic...   \n",
       "1074  characterization of the prostate cancer transc...   \n",
       "1302  large-scale screening for germ-line mutations ...   \n",
       "1326  identification of a high-risk disease-causing ...   \n",
       "1565  in the era of personalized medicine, understan...   \n",
       "1695  identification of a high-risk disease-causing ...   \n",
       "1897  genes encoding components of the pi3k-akt-mtor...   \n",
       "1953  over 1,000 reports have been published during ...   \n",
       "2548  abstract germline mutations that inactivate th...   \n",
       "3297  the aml1 gene is known as the most frequent ta...   \n",
       "3306  here we report two new runx1 mutations in one ...   \n",
       "3308   familial platelet disorder with propensity to...   \n",
       "\n",
       "                                                 Text_y  Score  \n",
       "ID                                                              \n",
       "697    256g>a asp86asn (d86n) and evolutionary conse...    1.0  \n",
       "1074   together, our data suggest that aberrations i...    1.0  \n",
       "1302   in f73, among 34 randomly selected mlh1 cdna ...    1.0  \n",
       "1326   nc = mutation was not categorized as effect i...    1.0  \n",
       "1565   in order to treat patients accordingly requir...    1.0  \n",
       "1695   this syndrome is characterized by caf-au-lai...    1.0  \n",
       "1897   however, few cancer-associated mutations have...    1.0  \n",
       "1953   these meta-analyses included an average of  s...    1.0  \n",
       "2548   among these mutations, there was an approxima...    1.0  \n",
       "3297   they contained three major groups with distin...    1.0  \n",
       "3306    because of his history of thrombocytopenia s...    1.0  \n",
       "3308   familial platelet disorder with propensity to...    1.0  "
      ]
     },
     "execution_count": 300,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data_best_score[new_data_best_score.apply(lambda line: line[\"Variation\"] in line[\"Text_y\"], axis = 1) == False]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 340,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count      2435\n",
       "unique      229\n",
       "top       BRCA1\n",
       "freq        193\n",
       "Name: Gene, dtype: object"
      ]
     },
     "execution_count": 340,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data_best_score[\"Gene\"].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 338,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ID\n",
       "115     they act as transcriptional coactivators and ...\n",
       "116     in this study, we performed a mutational scre...\n",
       "Name: Text_y, dtype: object"
      ]
     },
     "execution_count": 338,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data_best_score[new_data_best_score[\"Gene\"] == \"KMT2D\"][\"Text_y\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "count = concat_RUNX1[concat_RUNX1[\"Score\"] == 1].apply(lambda line: line[\"Text_y\"].count(line[\"Variation\"].lower()), axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Int64Index([], dtype='int64', name='ID')"
      ]
     },
     "execution_count": 148,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index_hacker = count[count == 0].index\n",
    "index_hacker"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 341,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text_x</th>\n",
       "      <th>Text_y</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2417</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>e1735k</td>\n",
       "      <td>6</td>\n",
       "      <td>abstract  the brca1 gene from individuals at ...</td>\n",
       "      <td>published onlinefirst july 18, 2013; doi: /21...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2418</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>r1753t</td>\n",
       "      <td>4</td>\n",
       "      <td>mutations in brca1 and brca2 account for the m...</td>\n",
       "      <td>although hr is essentially error-free, nhej i...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2419</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>c44y</td>\n",
       "      <td>4</td>\n",
       "      <td>germline mutations of the breast cancer 1 (brc...</td>\n",
       "      <td>] table 1.  percentage of cells without spot i...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2424</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>g1743r</td>\n",
       "      <td>1</td>\n",
       "      <td>abstract  the brca1 gene from individuals at ...</td>\n",
       "      <td>among these six mutations, r1699w and r1699q ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2425</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>i68k</td>\n",
       "      <td>6</td>\n",
       "      <td>germline mutations of the breast cancer 1 (brc...</td>\n",
       "      <td>importantly, the inhibition of ubiquitin liga...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2674</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>m1652t</td>\n",
       "      <td>6</td>\n",
       "      <td>abstract germline mutations that inactivate th...</td>\n",
       "      <td>substitutions at position 1652 can result in ...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2675</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>e1346k</td>\n",
       "      <td>5</td>\n",
       "      <td>mutations in brca1 and brca2 account for the m...</td>\n",
       "      <td>published onlinefirst july 18, 2013; doi: /21...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2677</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>a1843t</td>\n",
       "      <td>1</td>\n",
       "      <td>abstract  the brca1 gene from individuals at ...</td>\n",
       "      <td>three of them are exposed on the face away fr...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2679</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>s1613g</td>\n",
       "      <td>5</td>\n",
       "      <td>germline-inactivating mutations of brca1 resul...</td>\n",
       "      <td>in contrast to the tumorigenic mutations exam...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2680</th>\n",
       "      <td>BRCA1</td>\n",
       "      <td>m18k</td>\n",
       "      <td>6</td>\n",
       "      <td>germline mutations of the breast cancer 1 (brc...</td>\n",
       "      <td>we therefore generated constructs of all vari...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>193 rows  6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Gene Variation  Class  \\\n",
       "ID                             \n",
       "2417  BRCA1    e1735k      6   \n",
       "2418  BRCA1    r1753t      4   \n",
       "2419  BRCA1      c44y      4   \n",
       "2424  BRCA1    g1743r      1   \n",
       "2425  BRCA1      i68k      6   \n",
       "...     ...       ...    ...   \n",
       "2674  BRCA1    m1652t      6   \n",
       "2675  BRCA1    e1346k      5   \n",
       "2677  BRCA1    a1843t      1   \n",
       "2679  BRCA1    s1613g      5   \n",
       "2680  BRCA1      m18k      6   \n",
       "\n",
       "                                                 Text_x  \\\n",
       "ID                                                        \n",
       "2417   abstract  the brca1 gene from individuals at ...   \n",
       "2418  mutations in brca1 and brca2 account for the m...   \n",
       "2419  germline mutations of the breast cancer 1 (brc...   \n",
       "2424   abstract  the brca1 gene from individuals at ...   \n",
       "2425  germline mutations of the breast cancer 1 (brc...   \n",
       "...                                                 ...   \n",
       "2674  abstract germline mutations that inactivate th...   \n",
       "2675  mutations in brca1 and brca2 account for the m...   \n",
       "2677   abstract  the brca1 gene from individuals at ...   \n",
       "2679  germline-inactivating mutations of brca1 resul...   \n",
       "2680  germline mutations of the breast cancer 1 (brc...   \n",
       "\n",
       "                                                 Text_y  Score  \n",
       "ID                                                              \n",
       "2417   published onlinefirst july 18, 2013; doi: /21...    1.0  \n",
       "2418   although hr is essentially error-free, nhej i...    1.0  \n",
       "2419  ] table 1.  percentage of cells without spot i...    1.0  \n",
       "2424   among these six mutations, r1699w and r1699q ...    1.0  \n",
       "2425   importantly, the inhibition of ubiquitin liga...    1.0  \n",
       "...                                                 ...    ...  \n",
       "2674   substitutions at position 1652 can result in ...    1.0  \n",
       "2675   published onlinefirst july 18, 2013; doi: /21...    1.0  \n",
       "2677   three of them are exposed on the face away fr...    1.0  \n",
       "2679   in contrast to the tumorigenic mutations exam...    1.0  \n",
       "2680   we therefore generated constructs of all vari...    1.0  \n",
       "\n",
       "[193 rows x 6 columns]"
      ]
     },
     "execution_count": 341,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data_best_score[new_data_best_score[\"Gene\"] == \"BRCA1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 342,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "#CBL = train_variant[train_variant[\"Gene\"] == \"RUNX1\"]\n",
    "#article_cbl = train_text.loc[CBL.index,]\n",
    "\n",
    "#CBL.to_csv(\"cbl.txt\")\n",
    "clean_TERT = new_data_best_score[new_data_best_score[\"Gene\"] == \"BRCA1\"][\"Text_y\"]\n",
    "dtf = pd.merge(pd.DataFrame(clean_TERT.index), clean_TERT, on =\"ID\")\n",
    "dtf.columns= [\"ID\",\"Text\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 343,
   "metadata": {},
   "outputs": [],
   "source": [
    "np.savetxt(\"../datas/brca1_clean_article.txt\",dtf, fmt = \"%d || %s\", header= \",\".join(dtf.columns), comments='')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "deep_projet",
   "language": "python",
   "name": "deep_projet"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
