{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import re\n",
    "import time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "file_train_text = \"../../data/training_text/training_text\"\n",
    "file_train_variant = \"../../data/training_variants/training_variants\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ragou/.local/lib/python3.8/site-packages/pandas/util/_decorators.py:311: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n",
      "  return func(*args, **kwargs)\n"
     ]
    }
   ],
   "source": [
    "train_text = pd.read_csv(file_train_text, sep = '\\|\\|')\n",
    "train_text.index.name = \"ID\"\n",
    "train_text.columns = [\"Text\"]\n",
    "\n",
    "train_variant = pd.read_csv(file_train_variant)\n",
    "train_variant.set_index(\"ID\",inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Y371H</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CBL</td>\n",
       "      <td>K382E</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>CBL</td>\n",
       "      <td>C381A</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>CBL</td>\n",
       "      <td>C396R</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Variation  Class\n",
       "ID                      \n",
       "8   CBL     Y371H      4\n",
       "11  CBL     K382E      4\n",
       "13  CBL     C381A      4\n",
       "21  CBL     C396R      4"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_variant[train_variant[\"Variation\"].isin([\"C396R\", \"H398Q\", \"Y371H\", \"K382E\", \"C381A\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Variation  Class\n",
       "ID                      \n",
       "2   CBL     Q249E      2"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_variant[train_variant[\"Variation\"] == \"Q249E\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 Text\n",
       "ID                                                   \n",
       "6   Oncogenic mutations in the monomeric Casitas B..."
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_text.loc[[6],]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_count = train_variant[\"Gene\"].value_counts().loc[lambda x : x > 100].to_dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_class = train_variant[train_variant[\"Gene\"].isin(gene_count.keys())]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA4EAAAFNCAYAAAC+H2oqAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/MnkTPAAAACXBIWXMAAAsTAAALEwEAmpwYAAAcvElEQVR4nO3dfbBlVXkn4N8rDVFQAaFDoCFCRtQQJyr2KEqijqgRv3AcZYxRiWOKpEqNikkkJhUdJ6ZMYuJn4hQBFSZGRfADE2J0xK9kIrFBogJmbAkKLR8tIirGCPrOH2e3XtvGvug9Z/e9+3mqTt29115nn/fs7qru311rr13dHQAAAKbhNmMXAAAAwOIIgQAAABMiBAIAAEyIEAgAADAhQiAAAMCECIEAAAATIgQCsGpU1Yur6i9H/PwPVtWvDNu/VFXvXcFzX1xVDx62V/R7VtULq+rUlTofAKubEAjALqWqnlxVm6rqa1V1VVX9bVX93Nh1ba+739TdD99Zv6p6Y1X9/jLO9zPd/cEfta6qenBVXbnduf+gu3/lRz03AGuDEAjALqOqTkryyiR/kOSAJD+Z5M+THDdiWXNVVevGrgGAaRECAdglVNXeSV6S5Jnd/fbuvrG7b+rud3f3b97Ce95WVVdX1Q1V9eGq+pklxx5ZVZdU1VeraktV/cbQvn9V/XVVfbmqvlRVH6mqHf57WFUPq6pPD+d/bZJacuyXq+rvh+2qqldU1bVV9ZWq+mRV3aOqTkzyS0l+axjZfPfQ//KqekFVfSLJjVW1bmh76JKPv21VvXWo/8KquueSz+6qusuS/TdW1e9X1V5J/jbJQcPnfa2qDtp+emlVPXaYfvrlYYrrTy85dnlV/UZVfWL43m+tqtsu448QgFVCCARgV3H/JLdN8o5b8Z6/TXJ4kh9PcmGSNy05dlqSX+3uOyS5R5LzhvbnJ7kyyfrMRhtfmKS3P3FV7Z/k7Ul+N8n+ST6b5OhbqOPhSR6Y5K5J9k5yfJLruvuUoaY/6u7bd/djlrznF5M8Ksk+3X3zDs55XJK3JblTkr9K8s6q2v0Wr0SS7r4xybFJvjB83u27+wvbfa+7JnlzkucO1+DcJO+uqj2WdDs+ySOSHJbkZ5P88g/6XABWFyEQgF3Ffkm+eAuBaIe6+/Xd/dXu/vckL05yz2FEMUluSnJEVd2xu6/v7guXtB+Y5M7DSONHuvv7QmCSRya5uLvP6u6bMpumevUtlHJTkjskuXuS6u5Lu/uqnZT/6u6+orv/7RaOX7Dks/80s4B81E7OuRz/LcnfdPf7hnO/PMntkjxgu9q+0N1fSvLuJPdagc8FYBchBAKwq7guyf7LvUeuqnarqpdV1Wer6itJLh8O7T/8/K+ZBbnPVdWHqur+Q/sfJ9mc5L1VdVlVnXwLH3FQkiu27QxB8Yoddezu85K8NsmfJbm2qk6pqjvu5Cvs8Fw7Ot7d385s9PKgnbxnOQ5K8rntzn1Fkg1L+iwNu19PcvsV+FwAdhFCIAC7in9M8u9JHrfM/k/ObMrkQzObgnno0F5J0t0f6+7jMpsq+s4kZw7tX+3u53f3TyV5bJKTquqYHZz/qiSHbNupqlq6v73ufnV33yfJEZlNC912H+OORhl/UPs2Sz/7NkkOTrJtaufXk+y5pO9P3IrzfiHJnZece9v32rKT9wGwRgiBAOwSuvuGJL+X5M+q6nFVtWdV7V5Vx1bVH+3gLXfILDRel1kg+oNtB6pqj+E5fnsPUx6/kuTbw7FHV9VdhvBzQ5JvbTu2nb9J8jNV9fhhdPLX871h6zuq6j9V1f2Ge/ZuTPKNJee8JslP3crLkST3WfLZzx2+60eHYxclefIwGvqIJA9a8r5rkuy3ZFrs9s5M8qiqOmao9/nDuf/vD1EjAKuQEAjALqO7/yTJSZktxrI1s2mKz8psJG97Z2Q2rXFLkkvy3YC0zVOTXD5MFf21zFbpTGYLyfyfJF/LbPTxz7v7Azuo5YtJnpjkZZkFzcOT/MMtlH7HJH+R5Pqhpusym3aazBaoOWJYiXNH3+OWvCuz+/euH77L44dAmyTPSfKYJF8evtd3ztvdn85s4ZfLhs/8nimk3f0vSZ6S5DVJvjic5zHd/c1bURsAq1jt+F54AAAA1iIjgQAAABMytxBYVa8fHpr7qSVtd6qq91XVZ4af+w7tVVWvrqrNw8Npj1zynhOG/p+pqhPmVS8AAMAUzHMk8I2ZPWh2qZOTvL+7D0/y/mE/mT3Y9vDhdWKS1yWz0JjkRUnul+S+SV60LTgCAABw680tBHb3h5N8abvm45KcPmyfnu8uA35ckjN65qNJ9qmqA5P8QpL3dfeXuvv6JO/L9wdLAAAAlmnR9wQe0N1XDdtXJzlg2N6Q731o7pVD2y21AwAA8ENYN9YHd3dX1YotTVpVJ2Y2lTR77bXXfe5+97uv1KkBAABWlQsuuOCL3b1+R8cWHQKvqaoDu/uqYbrntUP7liSHLOl38NC2JcmDt2v/4I5O3N2nJDklSTZu3NibNm1a2coBAABWiar63C0dW/R00HOSbFvh84TMHoS7rf1pwyqhRyW5YZg2+ndJHl5V+w4Lwjx8aAMAAOCHMLeRwKp6c2ajePtX1ZWZrfL5siRnVtUzknwuyfFD93OTPDLJ5iRfT/L0JOnuL1XV/0zysaHfS7p7+8VmAAAAWKbqXrHb8nYZpoMCAABTVlUXdPfGHR1b9HRQAAAARiQEAgAATIgQCAAAMCFCIAAAwIQIgQAAABMiBAIAAEyIEAgAADAhQiAAAMCErBu7AAAAdu7Sl543dgmrwk//zkPGLgF2eUYCAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJmSUEFhVz6uqi6vqU1X15qq6bVUdVlXnV9XmqnprVe0x9P2xYX/zcPzQMWoGAABYCxYeAqtqQ5JfT7Kxu++RZLckT0ryh0le0d13SXJ9kmcMb3lGkuuH9lcM/QAAAPghjDUddF2S21XVuiR7JrkqyUOSnDUcPz3J44bt44b9DMePqapaXKkAAABrx8JDYHdvSfLyJJ/PLPzdkOSCJF/u7puHblcm2TBsb0hyxfDem4f++y2yZgAAgLVijOmg+2Y2undYkoOS7JXkEStw3hOralNVbdq6deuPejoAAIA1aYzpoA9N8q/dvbW7b0ry9iRHJ9lnmB6aJAcn2TJsb0lySJIMx/dOct32J+3uU7p7Y3dvXL9+/by/AwAAwKo0Rgj8fJKjqmrP4d6+Y5JckuQDSZ4w9DkhybuG7XOG/QzHz+vuXmC9AAAAa8YY9wSen9kCLxcm+eRQwylJXpDkpKranNk9f6cNbzktyX5D+0lJTl50zQAAAGvFup13WXnd/aIkL9qu+bIk991B328keeIi6gIAAFjrxnpEBAAAACMQAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAlZN3YBAGvNhx74oLFLWBUe9OEPjV0CAEySkUAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmJBRQmBV7VNVZ1XVp6vq0qq6f1XdqareV1WfGX7uO/Stqnp1VW2uqk9U1ZFj1AwAALAWjDUS+Kok7+nuuye5Z5JLk5yc5P3dfXiS9w/7SXJsksOH14lJXrf4cgEAANaGhYfAqto7yQOTnJYk3f3N7v5ykuOSnD50Oz3J44bt45Kc0TMfTbJPVR240KIBAADWiDFGAg9LsjXJG6rq41V1alXtleSA7r5q6HN1kgOG7Q1Jrljy/iuHNgAAAG6lMULguiRHJnldd987yY357tTPJEl3d5K+NSetqhOralNVbdq6deuKFQsAALCWjBECr0xyZXefP+yflVkovGbbNM/h57XD8S1JDlny/oOHtu/R3ad098bu3rh+/fq5FQ8AALCaLTwEdvfVSa6oqrsNTcckuSTJOUlOGNpOSPKuYfucJE8bVgk9KskNS6aNAgAAcCusG+lzn53kTVW1R5LLkjw9s0B6ZlU9I8nnkhw/9D03ySOTbE7y9aEvAAAAP4RRQmB3X5Rk4w4OHbODvp3kmfOuCQAAYArGek4gAAAAIxACAQAAJkQIBAAAmBAhEAAAYELGWh10l3Cf3zxj7BJWhQv++GljlwAAAKwQI4EAAAATIgQCAABMiBAIAAAwIUIgAADAhAiBAAAAE7KsEFhV719OGwAAALu2H/iIiKq6bZI9k+xfVfsmqeHQHZNsmHNtAAAArLCdPSfwV5M8N8lBSS7Id0PgV5K8dn5lAQAAMA8/MAR296uSvKqqnt3dr1lQTQAAAMzJzkYCkyTd/ZqqekCSQ5e+p7vPmFNdAAAAzMGyQmBV/e8k/yHJRUm+NTR3EiEQAABgFVlWCEyyMckR3d3zLAYAAID5Wu5zAj+V5CfmWQgAAADzt9yRwP2TXFJV/5Tk37c1dvdj51IVAAAAc7HcEPjieRYBAADAYix3ddAPzbsQAAAA5m+5q4N+NbPVQJNkjyS7J7mxu+84r8IAAABYecsdCbzDtu2qqiTHJTlqXkUBAAAwH8tdHfQ7euadSX5h5csBAABgnpY7HfTxS3Zvk9lzA78xl4oAAACYm+WuDvqYJds3J7k8symhAAAArCLLvSfw6fMuBAAAgPlb1j2BVXVwVb2jqq4dXmdX1cHzLg4AAICVtdyFYd6Q5JwkBw2vdw9tAAAArCLLDYHru/sN3X3z8HpjkvVzrAsAAIA5WG4IvK6qnlJVuw2vpyS5bp6FAQAAsPKWGwL/e5Ljk1yd5KokT0jyy3OqCQAAgDlZ7iMiXpLkhO6+Pkmq6k5JXp5ZOAQAAGCVWO5I4M9uC4BJ0t1fSnLv+ZQEAADAvCw3BN6mqvbdtjOMBC53FBEAAIBdxHKD3J8k+ceqetuw/8QkL51PSaxln3/Jfxy7hFXhJ3/vk2OXAADAGrWsENjdZ1TVpiQPGZoe392XzK8sAAAA5mHZUzqH0Cf4AQAArGLLvScQAACANUAIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBCRguBVbVbVX28qv562D+sqs6vqs1V9daq2mNo/7Fhf/Nw/NCxagYAAFjtxhwJfE6SS5fs/2GSV3T3XZJcn+QZQ/szklw/tL9i6AcAAMAPYZQQWFUHJ3lUklOH/UrykCRnDV1OT/K4Yfu4YT/D8WOG/gAAANxKY40EvjLJbyX59rC/X5Ivd/fNw/6VSTYM2xuSXJEkw/Ebhv4AAADcSgsPgVX16CTXdvcFK3zeE6tqU1Vt2rp160qeGgAAYM0YYyTw6CSPrarLk7wls2mgr0qyT1WtG/ocnGTLsL0lySFJMhzfO8l125+0u0/p7o3dvXH9+vXz/QYAAACr1MJDYHf/dncf3N2HJnlSkvO6+5eSfCDJE4ZuJyR517B9zrCf4fh53d0LLBkAAGDN2JWeE/iCJCdV1ebM7vk7bWg/Lcl+Q/tJSU4eqT4AAIBVb93Ou8xPd38wyQeH7cuS3HcHfb6R5IkLLQwAAGCN2pVGAgEAAJgzIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYkHVjFwAArD4vfcoTxi5hVfidvzxr7BIAvo+RQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQjwsHoBV77XPf/fYJawKz/qTx4xdAgC7ACOBAAAAEyIEAgAATIgQCAAAMCFCIAAAwIQIgQAAABMiBAIAAEyIEAgAADAhCw+BVXVIVX2gqi6pqour6jlD+52q6n1V9Znh575De1XVq6tqc1V9oqqOXHTNAAAAa8UYI4E3J3l+dx+R5Kgkz6yqI5KcnOT93X14kvcP+0lybJLDh9eJSV63+JIBAADWhoWHwO6+qrsvHLa/muTSJBuSHJfk9KHb6UkeN2wfl+SMnvlokn2q6sDFVg0AALA2jHpPYFUdmuTeSc5PckB3XzUcujrJAcP2hiRXLHnblUPb9uc6sao2VdWmrVu3zq9oAACAVWy0EFhVt09ydpLndvdXlh7r7k7St+Z83X1Kd2/s7o3r169fwUoBAADWjlFCYFXtnlkAfFN3v31ovmbbNM/h57VD+5Ykhyx5+8FDGwAAALfSGKuDVpLTklza3X+65NA5SU4Ytk9I8q4l7U8bVgk9KskNS6aNAgAAcCusG+Ezj07y1CSfrKqLhrYXJnlZkjOr6hlJPpfk+OHYuUkemWRzkq8nefpCqwUAAFhDFh4Cu/vvk9QtHD5mB/07yTPnWhQAAMBEjLo6KAAAAIslBAIAAEyIEAgAADAhQiAAAMCECIEAAAATIgQCAABMiBAIAAAwIUIgAADAhAiBAAAAEyIEAgAATIgQCAAAMCFCIAAAwIQIgQAAABMiBAIAAEyIEAgAADAhQiAAAMCECIEAAAATIgQCAABMiBAIAAAwIUIgAADAhAiBAAAAEyIEAgAATIgQCAAAMCFCIAAAwIQIgQAAABOybuwCgPk6+jVHj13CqvAPz/6HsUsAAFgII4EAAAATIgQCAABMiBAIAAAwIUIgAADAhAiBAAAAEyIEAgAATIgQCAAAMCFCIAAAwIQIgQAAABMiBAIAAEyIEAgAADAhQiAAAMCECIEAAAATsm7sAgAAAJLkzLfdd+wSVoXjn/hPP9L7jQQCAABMiBAIAAAwIUIgAADAhAiBAAAAE2JhGAAA2IEXv/jFY5ewKrhOq8+qGQmsqkdU1b9U1eaqOnnsegAAAFajVRECq2q3JH+W5NgkRyT5xao6YtyqAAAAVp9VEQKT3DfJ5u6+rLu/meQtSY4buSYAAIBVZ7WEwA1Jrliyf+XQBgAAwK1Q3T12DTtVVU9I8oju/pVh/6lJ7tfdz1rS58QkJw67d0vyLwsvdGXsn+SLYxcxMa754rnmi+eaL55rvniu+eK55ovnmi/ear3md+7u9Ts6sFpWB92S5JAl+wcPbd/R3ackOWWRRc1DVW3q7o1j1zElrvniueaL55ovnmu+eK754rnmi+eaL95avOarZTrox5IcXlWHVdUeSZ6U5JyRawIAAFh1VsVIYHffXFXPSvJ3SXZL8vruvnjksgAAAFadVRECk6S7z01y7th1LMCqn9K6Crnmi+eaL55rvniu+eK55ovnmi+ea754a+6ar4qFYQAAAFgZq+WeQAAAAFaAELiLqKrXV9W1VfWpsWuZgqo6pKo+UFWXVNXFVfWcsWta66rqtlX1T1X1z8M1/x9j1zQVVbVbVX28qv567FqmoKour6pPVtVFVbVp7HqmoKr2qaqzqurTVXVpVd1/7JrWsqq62/D3e9vrK1X13LHrWuuq6nnDv5+fqqo3V9Vtx65prauq5wzX++K19nfcdNBdRFU9MMnXkpzR3fcYu561rqoOTHJgd19YVXdIckGSx3X3JSOXtmZVVSXZq7u/VlW7J/n7JM/p7o+OXNqaV1UnJdmY5I7d/eix61nrquryJBu7ezU+U2pVqqrTk3yku08dVhHfs7u/PHJZk1BVu2X22K77dffnxq5nraqqDZn9u3lEd/9bVZ2Z5NzufuO4la1dVXWPJG9Jct8k30zyniS/1t2bRy1shRgJ3EV094eTfGnsOqaiu6/q7guH7a8muTTJhnGrWtt65mvD7u7Dy2+h5qyqDk7yqCSnjl0LzENV7Z3kgUlOS5Lu/qYAuFDHJPmsALgQ65LcrqrWJdkzyRdGrmet++kk53f317v75iQfSvL4kWtaMUIgk1dVhya5d5LzRy5lzRumJV6U5Nok7+tu13z+Xpnkt5J8e+Q6pqSTvLeqLqiqE8cuZgIOS7I1yRuGac+nVtVeYxc1IU9K8uaxi1jruntLkpcn+XySq5Lc0N3vHbeqNe9TSX6+qvarqj2TPDLJISPXtGKEQCatqm6f5Owkz+3ur4xdz1rX3d/q7nslOTjJfYepFsxJVT06ybXdfcHYtUzMz3X3kUmOTfLMYbo/87MuyZFJXtfd905yY5KTxy1pGoapt49N8raxa1nrqmrfJMdl9kuPg5LsVVVPGbeqta27L03yh0nem9lU0IuSfGvMmlaSEMhkDfelnZ3kTd399rHrmZJhqtYHkjxi5FLWuqOTPHa4R+0tSR5SVX85bklr3/Ab+3T3tUnekdn9JMzPlUmuXDKz4KzMQiHzd2ySC7v7mrELmYCHJvnX7t7a3TcleXuSB4xc05rX3ad19326+4FJrk/y/8auaaUIgUzSsEjJaUku7e4/HbueKaiq9VW1z7B9uyQPS/LpUYta47r7t7v74O4+NLMpW+d1t98cz1FV7TUsNpVhSuLDM5tSxJx099VJrqiquw1NxySxyNdi/GJMBV2Uzyc5qqr2HP4Pc0xm6xkwR1X148PPn8zsfsC/GreilbNu7AKYqao3J3lwkv2r6sokL+ru08atak07OslTk3xyuEctSV7Y3eeOV9Kad2CS04eV5G6T5Mzu9sgC1poDkrxj9n+0rEvyV939nnFLmoRnJ3nTMD3xsiRPH7meNW/4JcfDkvzq2LVMQXefX1VnJbkwyc1JPp7klHGrmoSzq2q/JDcleeZaWnTKIyIAAAAmxHRQAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgFgGarqJ6rqLVX12aq6oKrOraq7VpXnAAKwqnhOIADsxPBw5nckOb27nzS03TOz5wICwKpiJBAAdu4/J7mpu//Xtobu/uckV2zbr6pDq+ojVXXh8HrA0H5gVX24qi6qqk9V1c9X1W5V9cZh/5NV9bzFfyUApspIIADs3D2SXLCTPtcmeVh3f6OqDk/y5iQbkzw5yd9190urarckeya5V5IN3X2PJKmqfeZVOABsTwgEgJWxe5LXVtW9knwryV2H9o8leX1V7Z7knd19UVVdluSnquo1Sf4myXvHKBiAaTIdFAB27uIk99lJn+cluSbJPTMbAdwjSbr7w0kemGRLkjdW1dO6+/qh3weT/FqSU+dTNgB8PyEQAHbuvCQ/VlUnbmuoqp9NcsiSPnsnuaq7v53kqUl2G/rdOck13f0XmYW9I6tq/yS36e6zk/xukiMX8zUAwHRQANip7u6q+i9JXllVL0jyjSSXJ3nukm5/nuTsqnpakvckuXFof3CS36yqm5J8LcnTkmxI8oaq2vbL2N+e93cAgG2qu8euAQAAgAUxHRQAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmJD/DwTZnzDiG3ukAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 1080x360 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure(figsize=(15,5))\n",
    "sns.countplot(x = train_variant[\"Class\"])\n",
    "plt.title(\"Class distribution\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA3sAAAHwCAYAAAAfJXbRAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/MnkTPAAAACXBIWXMAAAsTAAALEwEAmpwYAAAjSElEQVR4nO3de7ztdV3n8fcnwPsNhxMikMeULKnEPJqXMs2Z8TIVWkr6cJRxVGpCs8lJaXSUUiceo6mTlQ6NJjyGvOUlbRjvd8fUAyEK6EgqAnI5miJegsDP/LF+G5fbffbZ53DW3vt8eT4fj/3Ya/1u67sWi8V68bvs6u4AAAAwlh/a6AEAAACw94k9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9ANZdVW2tqq6q/af7/6eqjt1L2/75qvrs3P0vVtW/3BvbnrZ3TlU9YG9tDwAWRewBDKyqHl1VH6uqb1XV5dPt36qq2uixzevuh3b3KbtabgrEO+9iWx/q7rvsjXFV1aur6vnLtn9kd79/b2x/b6iqp1TV9qq6qqpevcL8B1XVZ6rq21X1vqq6w9y8G1fVq6rqG1V1aVX97roOfheq6t9V1Yc3ehwA+yqxBzCoqnp6kv+e5IVJbpfk4CS/meR+SW60gUNbmKU9hfuiqjp4D1f9cpLnJ3nVCts8KMmbkvyXJLdNsj3J6+YWOTHJEUnukOSBSZ5RVQ/Zw3EAsMmIPYABVdWtk/xhkt/q7r/u7it75u+7+7HdfdW03I2r6kVV9aWquqyqXlFVN53mPaCqLqqqp097BS+pqifMPcZO111hPPtNy36lqj6f5N8sm//+qnrSdPvOVfWBqrpiWv510/QPTot/sqq+WVW/PjfGZ1bVpUn+cmnasiHcs6rOraqvVdVfVtVNpm3+wJ6jpb2HVXVcksdmFkDfrKq3TfOvOyx0eg1eWlVfnn5eWlU3Xsvrt4ITpzH+XlXdbpXlvk93v6m735LkqyvM/tUk53T3G7r7nzKLu7tV1Y9P849N8rzu/lp3n5fkL5L8u509VlU9uarOq6orp7H+zDT9J6Z/hl+fDnP9lbl1rvtnO93/vtd8er1/s6o+N63/ZzXzE0lekeQ+0+v/9bW+JgDMiD2AMd0nyY2T/M0uljspyY8lOSrJnZMcmuQ5c/Nvl+TW0/QnJvmzqjpwjevOe3KSX0py9yTbkjxylTE9L8k7kxyY5LAkL0uS7r7/NP9u3X2L7l7aQ3W7zPZa3SHJcTvZ5mOTPDjJnaYxP3uVx8/0eCcnOS3Jf5se75dXWOxZSe6d2WtwtyT3Wrbt1V6/5Y5P8ttJfjrJZ6vqrVX1iKo6YFdjXcWRST4595y+leQfkhw5jeOQ+fnT7SNX2lBVPSqzWHx8klsl+ZUkX53G97bM/pn9cJKnJjmtqnbnUNpfSnLPzJ77MUkePMXnbyb56PT632Y3tgdAxB7AqA5K8pXuvmZpQlX932nPyXeq6v5VVZnF0X/s7n/s7iuT/Nckj57bzj8n+cPu/ufuPj3JN5PcZY3rzjsmyUu7+8Lu/sckf7TK2P85s3C7fXf/U3fv6pyt7yZ5bndf1d3f2ckyfzr32C9I8phdbHOtHpvZ63N5d+9I8gdJHjc3f8XXb6UNdfd3u/vd3f24zCL3LUn+Y5KLq+p5ezi+WyS5Ytm0K5LccpqXZfOX5q3kSZmF7yemvcTnd/cFmcXuLZKc1N1Xd/d7k/xtdu81Pqm7v97dX0ryvsziGYDrSewBjOmrSQ6aP4etu+877R35amaf/1uS3CzJGVMEfj3J26fp121nPhiTfDuzL/ZrWXfe7ZNcOHf/glXG/owkleTj0yGB/34Xz3XHdIjiapY/9u13sfxa3T7f/1yWb3tnr9+qpng+O8lZSQ7ITgJxDb6Z2V64ebdKcuU0L8vmL81byeGZ7RVc7vZJLuzu785NuyCzvZlrdenc7TW9RgDsmtgDGNNHk1yV5OhVlvlKku8kObK7bzP93Lq71/JFe3fXvSSzWFjyIzvbcHdf2t1P7u7bJ/mNJH9eq1+Bs9cw3uWP/eXp9rcyi9YkyQrnyu1q21/ObC/kStvebVV1WFWdUFXnJnltkh2ZHbZ6zB5u8pzMDi9d2v7NMzuU9Zzu/lpm/1zuNrf83aZ1VnLhtO5yX05yeFXNf6f4kSQXT7e/7zXO7NDWtVrLP1sAdkLsAQyou7+e2SGFf15Vj6yqW1bVD1XVUUluPi3z3cwuyPGSqvrhJKmqQ6vqwWvY/u6u+/okvz3FzIFJTtjZtqvqUVV12HT3a5l94V/aa3RZkh/d1fhWcPz02LfN7Dy7pfP9PpnZ+WtHTRdtOXHZert6vNckeXZVbanZlS+fk+R/7cH4UlUnZhZad8nsXLUjuvt506GNq623/zT2/ZLsV1U3mduj++YkP1lVvzYt85wkZ3f3Z6b5p07jP3C6aMuTk7x6Jw/1P5P8p6q6x3QBlTvX7M84fCyzvXHPqKoDavY3CH85s1hNZnsnf7WqbjZF+xN342W5LMlhVTXk1WMBFk3sAQyqu/9bkt/N7LDIy6af/5HkmUn+77TYM5Ocn+TvquobSd6dtR8yuDvr/kWSd2QWV2dm9ucAduaeST5WVd9M8tYkT+vuz0/zTkxyynTo6O7s7fqrzC4g8vnMDkV8fpJ09//L7Kql707yuSTLzw98ZZK7To/3lhW2+/zM/pzB2Uk+NT2356+w3Fq8JbPzFJ/Q3R/s7rXu1Xp2ZntZT0jyb6fbz06S6TzCX8vsPMWvJfnZfP95lc/N7PW4IMkHkrywu9++0oN09xum7fxVZod6viXJbbv76szi7qGZ7fH98ySPnwvKlyS5OrP33ymZXfRmrd6bWQBfWlVf2Y31AEhSa/9vCQAAAPsKe/YAAAAGJPYAAAAGJPYAAAAGJPYAAAAGJPYAAAAGtP+uF9m8DjrooN66detGDwMAAGBDnHHGGV/p7i0rzdunY2/r1q3Zvn37Rg8DAABgQ1TVBTub5zBOAACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAe2/0QPYaPf4vVM3eghsQme88PEbPQQAALhe7NkDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAY0MJir6oOr6r3VdW5VXVOVT1tmn5iVV1cVWdNPw+bW+f3q+r8qvpsVT14UWMDAAAY3f4L3PY1SZ7e3WdW1S2TnFFV75rmvaS7XzS/cFXdNcmjkxyZ5PZJ3l1VP9bd1y5wjAAAAENa2J697r6ku8+cbl+Z5Lwkh66yytFJXtvdV3X3F5Kcn+ReixofAADAyNblnL2q2prk7kk+Nk16SlWdXVWvqqoDp2mHJrlwbrWLsnocAgAAsBMLj72qukWSNyb5ne7+RpKXJ7lTkqOSXJLkj3dze8dV1faq2r5jx469PVwAAIAhLDT2quqAzELvtO5+U5J092XdfW13fzfJX+R7h2penOTwudUPm6Z9n+4+ubu3dfe2LVu2LHL4AAAA+6xFXo2zkrwyyXnd/eK56YfMLfaIJJ+ebr81yaOr6sZVdcckRyT5+KLGBwAAMLJFXo3zfkkel+RTVXXWNO0/J3lMVR2VpJN8MclvJEl3n1NVr09ybmZX8jzelTgBAAD2zMJir7s/nKRWmHX6Kuu8IMkLFjUmAACAG4p1uRonAAAA60vsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADGhhsVdVh1fV+6rq3Ko6p6qeNk2/bVW9q6o+N/0+cJpeVfUnVXV+VZ1dVT+zqLEBAACMbpF79q5J8vTuvmuSeyc5vqrumuSEJO/p7iOSvGe6nyQPTXLE9HNckpcvcGwAAABDW1jsdfcl3X3mdPvKJOclOTTJ0UlOmRY7JcnDp9tHJzm1Z/4uyW2q6pBFjQ8AAGBk63LOXlVtTXL3JB9LcnB3XzLNujTJwdPtQ5NcOLfaRdM0AAAAdtPCY6+qbpHkjUl+p7u/MT+vuztJ7+b2jquq7VW1fceOHXtxpAAAAONYaOxV1QGZhd5p3f2mafJlS4dnTr8vn6ZfnOTwudUPm6Z9n+4+ubu3dfe2LVu2LG7wAAAA+7BFXo2zkrwyyXnd/eK5WW9Ncux0+9gkfzM3/fHTVTnvneSKucM9AQAA2A37L3Db90vyuCSfqqqzpmn/OclJSV5fVU9MckGSY6Z5pyd5WJLzk3w7yRMWODYAAIChLSz2uvvDSWonsx+0wvKd5PhFjQcAAOCGZF2uxgkAAMD6EnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAAD2n+jBwCs7Et/+FMbPQQ2oR95zqc2eggAwD7Cnj0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABLSz2qupVVXV5VX16btqJVXVxVZ01/Txsbt7vV9X5VfXZqnrwosYFAABwQ7DIPXuvTvKQFaa/pLuPmn5OT5KqumuSRyc5clrnz6tqvwWODQAAYGgLi73u/mCSf1zj4kcneW13X9XdX0hyfpJ7LWpsAAAAo9uIc/aeUlVnT4d5HjhNOzTJhXPLXDRN+wFVdVxVba+q7Tt27Fj0WAEAAPZJ6x17L09ypyRHJbkkyR/v7ga6++Tu3tbd27Zs2bKXhwcAADCGdY297r6su6/t7u8m+Yt871DNi5McPrfoYdM0AAAA9sBux15VHVhVP70nD1ZVh8zdfUSSpSt1vjXJo6vqxlV1xyRHJPn4njwGAAAAyf5rWaiq3p/kV6blz0hyeVV9pLt/d5V1XpPkAUkOqqqLkjw3yQOq6qgkneSLSX4jSbr7nKp6fZJzk1yT5PjuvnbPnhIAAABrir0kt+7ub1TVk5Kc2t3PraqzV1uhux+zwuRXrrL8C5K8YI3jAQAAYBVrPYxz/+kQzGOS/O0CxwMAAMBesNbY+4Mk70hyfnd/oqp+NMnnFjcsAAAAro+1HsZ5SXdfd1GW7v58Vb14QWMCAADgelrrnr2XrXEaAAAAm8Cqe/aq6j5J7ptkS1XNX3nzVkn2W+TAAAAA2HO7OozzRkluMS13y7np30jyyEUNCgAAgOtn1djr7g8k+UBVvbq7L1inMQEAAHA9rfUCLTeuqpOTbJ1fp7t/cRGDAgAA4PpZa+y9IckrkvzPJNcubjgAAADsDWuNvWu6++ULHQkA+4z7vex+Gz0ENqGPPPUjGz0EAOas9U8vvK2qfquqDqmq2y79LHRkAAAA7LG17tk7dvr9e3PTOsmP7t3hAAAAsDesKfa6+46LHggAAAB7z5pir6oev9L07j517w4HAACAvWGth3Hec+72TZI8KMmZScQeAADAJrTWwzifOn+/qm6T5LWLGBAAAADX31qvxrnct5I4jw8AAGCTWus5e2/L7OqbSbJfkp9I8vpFDQoAAIDrZ63n7L1o7vY1SS7o7osWMB4AAAD2gjUdxtndH0jymSS3THJgkqsXOSgAAACunzXFXlUdk+TjSR6V5JgkH6uqRy5yYAAAAOy5tR7G+awk9+zuy5OkqrYkeXeSv17UwAAAANhza70a5w8thd7kq7uxLgAAAOtsrXv23l5V70jymun+ryc5fTFDAgAA4PpaNfaq6s5JDu7u36uqX03yc9OsjyY5bdGDAwAAYM/sas/eS5P8fpJ095uSvClJquqnpnm/vMCxAQAAsId2dd7dwd39qeUTp2lbFzIiAAAArrddxd5tVpl30704DgAAAPaiXcXe9qp68vKJVfWkJGcsZkgAAABcX7s6Z+93kry5qh6b78XdtiQ3SvKIBY4LAACA62HV2Ovuy5Lct6oemOQnp8n/u7vfu/CRAQAAsMfW9Hf2uvt9Sd634LEAAACwl6z1j6oDAGx6H7j/L2z0ENiEfuGDH9joIcCG2NUFWgAAANgH2bMHAADr4E+f/raNHgKbzFP++JcXun179gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAa0sNirqldV1eVV9em5abetqndV1eem3wdO06uq/qSqzq+qs6vqZxY1LgAAgBuCRe7Ze3WShyybdkKS93T3EUneM91PkocmOWL6OS7Jyxc4LgAAgOEtLPa6+4NJ/nHZ5KOTnDLdPiXJw+emn9ozf5fkNlV1yKLGBgAAMLr1Pmfv4O6+ZLp9aZKDp9uHJrlwbrmLpmkAAADsgQ27QEt3d5Le3fWq6riq2l5V23fs2LGAkQEAAOz71jv2Lls6PHP6ffk0/eIkh88td9g07Qd098ndva27t23ZsmWhgwUAANhXrXfsvTXJsdPtY5P8zdz0x09X5bx3kivmDvcEAABgN+2/qA1X1WuSPCDJQVV1UZLnJjkpyeur6olJLkhyzLT46UkeluT8JN9O8oRFjQsAAOCGYGGx192P2cmsB62wbCc5flFjAQAAuKHZsAu0AAAAsDhiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYED7b8SDVtUXk1yZ5Nok13T3tqq6bZLXJdma5ItJjunur23E+AAAAPZ1G7ln74HdfVR3b5vun5DkPd19RJL3TPcBAADYA5vpMM6jk5wy3T4lycM3bigAAAD7to2KvU7yzqo6o6qOm6Yd3N2XTLcvTXLwSitW1XFVtb2qtu/YsWM9xgoAALDP2ZBz9pL8XHdfXFU/nORdVfWZ+Znd3VXVK63Y3ScnOTlJtm3btuIyAAAAN3Qbsmevuy+efl+e5M1J7pXksqo6JEmm35dvxNgAAABGsO6xV1U3r6pbLt1O8q+TfDrJW5McOy12bJK/We+xAQAAjGIjDuM8OMmbq2rp8f+qu99eVZ9I8vqqemKSC5IcswFjAwAAGMK6x153fz7J3VaY/tUkD1rv8QAAAIxoM/3pBQAAAPYSsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADCgTRd7VfWQqvpsVZ1fVSds9HgAAAD2RZsq9qpqvyR/luShSe6a5DFVddeNHRUAAMC+Z1PFXpJ7JTm/uz/f3VcneW2Sozd4TAAAAPuczRZ7hya5cO7+RdM0AAAAdkN190aP4TpV9cgkD+nuJ033H5fkZ7v7KXPLHJfkuOnuXZJ8dt0HOq6DknxlowcBK/DeZDPz/mSz8t5kM/P+3Hvu0N1bVpqx/3qPZBcuTnL43P3DpmnX6e6Tk5y8noO6oaiq7d29baPHAct5b7KZeX+yWXlvspl5f66PzXYY5yeSHFFVd6yqGyV5dJK3bvCYAAAA9jmbas9ed19TVU9J8o4k+yV5VXefs8HDAgAA2OdsqthLku4+PcnpGz2OGyiHx7JZeW+ymXl/sll5b7KZeX+ug011gRYAAAD2js12zh4AAAB7gdgbSFVdW1VnVdUnq+rMqrrvNH1rVX1nmnduVZ1aVQdM8w6oqpOq6nPTOh+tqofObfOoquqqesiyx3pVVV1eVZ9e32fJCKrqX0zvx7Oq6tKqunjufk+/P11Vb6iqm03rfLGqPjXN2z63redV1dnT9HdW1e037pmxr5v7HF36OWGavn9V/dfps3Jp3rNWWW9rVT2gqq6Y7n+mql60cc+MEcy9z5Y+Hw9d5bP0Rqu8n9+/7HN0W1W9f8OeGPuk9freWVWHV9X7pm2dU1VPW/9nu+/adOfscb18p7uPSpKqenCSP0ryC9O8f+juo6pqvyTvSnJMktOSPC/JIUl+sruvqqqD59ZJksck+fD0++1z01+d5E+TnLqwZ8OwuvurSY5Kkqo6Mck3u/tF0/1vzr2PT0vym0lePK36wO5e/jd5Xtjd/2Va/reTPGdaB/bEdZ+jyzw/ye2S/FR3/1NV3TLJ01dbr6q2JvlQd/9SVd00yd9X1Zu7+yOLGTo3AN9Z9vn463P3T8zcZ+k0bWfv5yT54ap6aHf/n4WOmJGt1/fOa5I8vbvPnD57z6iqd3X3uYt9emMQe+O6VZKvLZ/Y3ddW1ceTHDrtMXlykjt291XT/MuSvD5JqqqSPCrJv0ryoaq6SXf/07TcB6cvMrBIH0ry06st0N3fmLt78yRORGavmvus3Dr3GXhlkhPXuo3u/k5VnZXk0EWMkRukXX4+7sILkzwridhjb1jY987uviTJJdPyV1bVeZl9loq9NRB7Y7np9GXiJpn9X5NfXL5AVd0kyc8meVqSOyf50rIvy/Pum+QL3f0P0+Ed/ybJGxcwbvgBVbV/kofme/9nr5O8s6o6yf/o7pPnln1BkscnuSLJA9d7rAxl6XN0yR8lOS+zz8or17jeF7r7EfMzq+rAJEck+eBeHCs3UCt8Pu7MD7yfu/t10+2PJnlEVT0wyWrvbdiZdf/eOe1ouHuSj+2dpzA+5+yN5TvdfVR3/3iShyQ5dfq/JElyp+lfyMuSXNLdZ69he49J8trp9mun+7BoS//x2J7kS0leOU3/ue7+mcy+4BxfVfdfWqG7n9Xdh2d2iMhT1nm8jGXpc3Tp53XLF6iqJ0znolxYVYevsN586P18VX0yycVJ3tHdl67Hk2BYO/t83JldvZ+fn+TZCxgnNwzr+r2zqm6RWfz9zirByDJib1Dd/dEkByXZMk36h+m46jsluUdV/UqS85P8SFXdavn60zHWv5bkOVX1xSQvS/KQ6VhpWKT5LydP7e6rk6S7L55+X57kzUnutcK6p2X2voW9aemz8pZJ0t1/OX2eXpFkv12s+6HuvluSI5M8saqOWuRAGd6Kn497qrvfm+SmSe69d4bHDdWiv3dOF3h5Y5LTuvtNC346QxF7g6qqH8/sS8hX56dPF7c4Icnvd/e3M/u/gv+9qm40rbelqh6V5EFJzu7uw7t7a3ffIbN/yb7v0CRYD1V187kP/Jsn+ddJPj3dP2Ju0aOTfGb9R8jI5j4r/3Q6JGnpi8mNdmMbX0hyUpJnLmSQsOeen+QZGz0I9m2L/N457S18ZZLzuvvFYbeIvbHcdDq06Kwkr0tybHdfu8Jyb0lys6r6+cwO39iR5Nya/RmFv03yjcx2nb952XpvnKanql6T2fH+d6mqi6rqiQt4PrDk4CQfng6H+3iS/93dS+eqnFSzy5CfnVkEuiQz18d1n6PTz0nT9GdldoGAT1fV32d2cYxTknx5N7b9iiT3d3Er1tHO3s/X6e7TM/seALtrvb533i/J45L84tx7+WGLeEIjqm4XrgMAABiNPXsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAMKeqDq6qv6qqz1fVGVX10aryN0YB2OeIPQCYTH+89y1JPtjdP9rd90jy6CSHbejAAGAPiD0A+J5fTHJ1d79iaUJ3X9DdL6uq/arqhVX1iao6u6p+I0mq6gFV9f6q+uuq+kxVnTZFY6rqHlX1gWkP4Tuq6pANel4A3ADtv9EDAIBN5MgkZ+5k3hOTXNHd96yqGyf5SFW9c5p392ndLyf5SJL7VdXHkrwsydHdvaOqfj3JC5L8+4U+AwCYiD0A2Imq+rMkP5fk6iQXJPnpqnrkNPvWSY6Y5n28uy+a1jkrydYkX0/yk0neNe3o2y/JJes3egBu6MQeAHzPOUl+belOdx9fVQcl2Z7kS0me2t3vmF+hqh6Q5Kq5Sddm9t/XSnJOd99nwWMGgBU5Zw8Avue9SW5SVf9hbtrNpt/vSPIfquqAJKmqH6uqm6+yrc8m2VJV95mWP6CqjlzEoAFgJfbsAcCku7uqHp7kJVX1jCQ7knwryTOTvCGzwzPPnC7AsiPJw1fZ1tXTIZ9/UlW3zuy/uS/NbO8hACxcdfdGjwEAAIC9zGGcAAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAA/r/+sCQoIjN2bkAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 1080x576 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure(figsize=(15,8))\n",
    "\n",
    "sns.barplot( x = list(gene_count.keys()), y = list(gene_count.values()))\n",
    "plt.title(\"Gene distribution > 100 count\")\n",
    "plt.ylabel(\"Counts\")\n",
    "plt.xlabel(\"Gene\")\n",
    "\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA3sAAAHwCAYAAAAfJXbRAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/MnkTPAAAACXBIWXMAAAsTAAALEwEAmpwYAAAwcUlEQVR4nO3de5xdZX0v/s+XJIgiApWLkJkk4I1LLBFQalVEPFJsvSGIIu3hppxetHj0HC8/j1WP9mhrFfByjoceRagW0FaPqIgiaEEFOeESi1wsRZJMRIhUULkIhOf3x+zomCaZmczs2ZOV9/v12q/Ze61nPeu796zMzCfPs9aq1loAAADolq0GXQAAAADTT9gDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhD4BZp6oWVVWrqrm911+pquOmqe9nV9VNY17fWlX/YTr67vX3/ao6ZLr664eqOr6qvrUJ232yqt7Tj5oAmH7CHsAWrKpeWVXfrap7quqO3vM/raoadG1jtdZe0Fo7a7x2vYD4hHH6uqy19uTpqGt94ae1tm9r7ZvT0T8ATIWwB7CFqqo3Jjk9yfuTPC7Jrkn+OMkzk2w9wNL6Zu1IIQBsCYQ9gC1QVW2f5L8n+dPW2j+01n7eRl3TWju2tfbLXrtHVNXfVNWKqrq9qj5WVY/srTukqkaq6o29UcHbquqEMfvY4LbrqWdOr+1PquqWJH+wzvpvVtWre8+fUFX/VFV399qf11t+aa/5sqr6RVW9YkyNb66qHyc5c+2ydUp4WlVdX1U/raozq2qbXp//brrj2tHDqjo5ybFJ3tTb3xd76381LbT3GZxWVT/qPU6rqkdM5PNbz2d0QlXdUFU/r6pbquo/jVk33vfisVV1flX9rKquTPL4De2n1/5ZVfWdqrqrqlZW1fHrabNjVX2pqlb3PrcvVdXQmPXH9+r8eVX9sKqO3dj3D4DpJ+wBbJmekeQRSb4wTrv3JXlSkiVJnpBkfpK/GLP+cUm27y0/KclHq2rHCW471muSvDDJU5McmOSojdT07iRfS7JjkqEkH06S1trBvfX7tdYe3VpbGyIel+S3kixMcvIG+jw2ye9lNAQ9Kcl/28j+09vfGUk+neSve/t70XqavS3J72T0M9gvydPX6Xtjn9+67sjoZ/SYJCckObWq9p9gXx9Ncn+S3ZKc2HusV1UtTPKVjH6uO/dqv3Y9TbdKcmZGP9cFSe5L8pFeH9sm+VCSF7TWtkvyu2P6WO/3D4DpJ+wBbJl2SvKT1tpDaxeMGcm5r6oO7p23d3KS/9xa+7fW2s+T/I8krxzTz4NJ/ntr7cHW2gVJfpHkyRPcdqyjk5zWWlvZWvu3JO/dSO0PZjRg7N5au7+1Nt6FRh5O8o7W2i9ba/dtoM1Hxuz7L5McM06fE3VsRj+fO1prq5O8K8kfjVm/3s9vfR211r7cWvvX3gjsP2U0MD17vL6qak6SI5P8RWvtntbadUk2dv7jq5J8vbV2Tq+vO1tr166nnjtba//YWru39/39yyTPGdPk4SSLq+qRrbXbWmvfH1PnZL5/AGwiYQ9gy3Rnkp3GnsPWWvvd1toOvXVbZXRU51FJruqFwLuSXNhb/qt+xgbGJPcmefQEtx1r9yQrx7xevpHa35SkklxZo1e+3OAoVc/q1tr947RZd9+7j9N+onbPb76Xdfve0Of371TVC6rqiqr6t97n+fsZDe3j9bVzkrmZ+Oc7nORfN7J+bT2Pqqr/XVXLq+pnSS5NskNVzWmt3ZPkFRk9B/S2qvpyVe3V23Sy3z8ANpGwB7BlujzJL5O8ZCNtfpLRqXn7ttZ26D22b62tN4xMcdvbMhoy1lqwoY5baz9urb2mtbZ7kv+U5H/Wxq/A2SZQ77r7/lHv+T0ZDa1Jkqp63CT7/lFGR7HW1/eE9c7z+8ckf5Nk114ovyCjoWk8q5M8lAl+vhkNhRs9p6/njRkdhTyotfaYJGun0VaStNa+2lp7fkanjt6Y5G97yyf7/QNgEwl7AFug1tpdGZ1S+D+r6qiq2q6qtqqqJUm27bV5OKN/oJ9aVbskSVXNr6rfm0D/k932M0n+vKqGeueZvWVDfVfVy8dcCOSnGQ1cD/de355kz/HqW48/6+37tzJ6nt3a8/2WJdm3qpb0LtryznW2G29/5yT5b1W1c1XtlNFzFj+1CfVtndFzLFcneaiqXpDksIls2Fpbk+RzSd7ZG43bJ8nG7ln46ST/oaqOrqq5vYu7LFlPu+0yGujv6n1u71i7oqp2raqX9M7d+2VGp5Q+3Fu3se8fANNI2APYQrXW/jrJGzI6re723uN/J3lzku/0mr05yc1JruhN1ft6NnBO2XpMZtu/TfLVjIarqzMaTjbkaUm+W1W/SHJ+klNaa7f01r0zyVm9qaNHT7DOJPn7jJ4Dd0tGpzC+J0laaz/I6FVLv57kX5Kse37Zx5Ps09vf/11Pv+9JsjTJ95L8c++9Tfqm5L1z4v48o6H4pxk9r+78SXTx2oxO6fxxkk9m9MIqG9rXioxOEX1jkn/L6IVV9ltP09OSPDKjo7hXZHSa7lpbZfTY+lGvj+ck+ZPeuo19/wCYRtXaRGa3AAAAsDkxsgcAANBBwh4AAEAHCXsAAAAdJOwBAAB0UN/CXlV9oqruqKrrxix7f1XdWFXfq6rPV9UOY9a9tapurqqbJnJZbwAAADasb1fjrKqDM3pfnbNba4t7yw5Lcklr7aGq+qskaa29uXfPn3OSPD3J7hm9xPWTevcG2qCddtqpLVq0qC/1AwAAzHZXXXXVT1prO69v3dx+7bS1dmlVLVpn2dfGvLwiyVG95y9Jcm5r7ZdJflhVN2c0+F2+sX0sWrQoS5cunb6iAQAANiNVtXxD6wZ5zt6JSb7Sez4/ycox60Z6ywAAANgEAwl7VfW2JA8l+fQmbHtyVS2tqqWrV6+e/uIAAAA6YMbDXlUdn+SFSY5tvz5hcFWS4THNhnrL/p3W2hmttQNbawfuvPN6p6YCAABs8fp2zt76VNXhSd6U5DmttXvHrDo/yd9X1QczeoGWJya5ciZrAwAAtkwPPvhgRkZGcv/99w+6lA3aZpttMjQ0lHnz5k14m76Fvao6J8khSXaqqpEk70jy1iSPSHJRVSXJFa21P26tfb+qPpPk+oxO7/yz8a7ECQAAMB1GRkay3XbbZdGiRenllFmltZY777wzIyMj2WOPPSa8XT+vxnnMehZ/fCPt/zLJX/arHgAAgPW5//77Z23QS5KqymMf+9hM9polg7waJwAAwKwwW4PeWptSn7AHAAAwjh//+Md55Stfmcc//vE54IAD8vu///v5wQ9+kMWLFw+6tA2a0Qu0AAAAbG5aazniiCNy3HHH5dxzz02SLFu2LLfffvuAK9s4I3sAAAAb8Y1vfCPz5s3LH//xH/9q2X777Zfh4V/fPe7WW2/Ns5/97Oy///7Zf//9853vfCdJctttt+Xggw/OkiVLsnjx4lx22WVZs2ZNjj/++CxevDhPecpTcuqpp/albiN7AAAAG3HdddflgAMO2GibXXbZJRdddFG22Wab/Mu//EuOOeaYLF26NH//93+f3/u938vb3va2rFmzJvfee2+uvfbarFq1Ktddd12S5K677upL3cIeAADAFD344IN57Wtfm2uvvTZz5szJD37wgyTJ0572tJx44ol58MEH89KXvjRLlizJnnvumVtuuSWve93r8gd/8Ac57LDD+lKTaZwAAAAbse++++aqq67aaJtTTz01u+66a5YtW5alS5fmgQceSJIcfPDBufTSSzN//vwcf/zxOfvss7Pjjjtm2bJlOeSQQ/Kxj30sr371q/tSt7AHAACwEYceemh++ctf5owzzvjVsu9973tZuXLlr17ffffd2W233bLVVlvl7/7u77JmzZokyfLly7PrrrvmNa95TV796lfn6quvzk9+8pM8/PDDOfLII/Oe97wnV199dV/qNo0TAABgI6oqn//85/P6178+f/VXf5VtttkmixYtymmnnfarNn/6p3+aI488MmeffXYOP/zwbLvttkmSb37zm3n/+9+fefPm5dGPfnTOPvvsrFq1KieccEIefvjhJMl73/ve/tTdWutLxzPhwAMPbEuXLh10GQAAwGbshhtuyN577z3oMsa1vjqr6qrW2oHra28aJwAAQAcJewAAAB0k7AEAAHSQsAfAFmfh8IJU1YQfC4cXDLpkAJg0V+MEYIuzYmRllp35xQm33++EF/WxGgDoDyN7AAAAHSTsAQAADNiJJ56YXXbZJYsXL562PoU9AACAMRYtXDipc7vHeyxauHDcfR5//PG58MILp/V9OGcPAABgjOUrVuTum2+ctv62f8Je47Y5+OCDc+utt07bPhMjewAAAJ0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAADdswxx+QZz3hGbrrppgwNDeXjH//4lPt0NU4AAIAxFi5YMKEraE6mv/Gcc84507a/tYQ9AACAMW5dvnzQJUwL0zgBAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAAGLCVK1fmuc99bvbZZ5/su+++Of3006fcp7AHAAAwxqKFC1NV0/ZYtHDhuPucO3duPvCBD+T666/PFVdckY9+9KO5/vrrp/Q+3GcPAABgjOUrVuSOy787bf3t8oyDxm2z2267ZbfddkuSbLfddtl7772zatWq7LPPPpu8XyN7AAAAs8itt96aa665JgcdNH5I3BhhDwAAYJb4xS9+kSOPPDKnnXZaHvOYx0ypL2EPAABgFnjwwQdz5JFH5thjj83LXvayKfcn7AEAAAxYay0nnXRS9t5777zhDW+Ylj6FPQAAgAH79re/nb/7u7/LJZdckiVLlmTJkiW54IILptSnq3ECAACMsXDBggldQXMy/Y3nWc96Vlpr07bPRNgDAAD4DbcuXz7oEqaFaZwAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAADNj999+fpz/96dlvv/2y77775h3veMeU+xT2AAAAxli4YEGqatoeE7nP3iMe8YhccsklWbZsWa699tpceOGFueKKK6b0PtxnDwAAYIwVK1fmlvMvnrb+9nzx88ZtU1V59KMfnSR58MEH8+CDD6aqprRfI3sAAACzwJo1a7JkyZLssssuef7zn5+DDjpoSv0JewAAALPAnDlzcu2112ZkZCRXXnllrrvuuin1J+wBAADMIjvssEOe+9zn5sILL5xSP8IeAADAgK1evTp33XVXkuS+++7LRRddlL322mtKfbpACwAAwIDddtttOe6447JmzZo8/PDDOfroo/PCF75wSn0KewAAAGMsGB6e0BU0J9PfeH77t38711xzzbTtMxH2AAAAfsPyFSsGXcK0cM4eAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAALPAmjVr8tSnPnXK99dbS9gDAAAYY+HwglTVtD0WDi+Y0H5PP/307L333tP2Pvp2n72q+kSSFya5o7W2uLfst5Kcl2RRkluTHN1a+2lVVZLTk/x+knuTHN9au7pftQEAAGzIipGVWXbmF6etv/1OeNG4bUZGRvLlL385b3vb2/LBD35wWvbbz5G9TyY5fJ1lb0lycWvtiUku7r1OkhckeWLvcXKS/9XHugAAAGaV17/+9fnrv/7rbLXV9EW0voW91tqlSf5tncUvSXJW7/lZSV46ZvnZbdQVSXaoqt36VRsAAMBs8aUvfSm77LJLDjjggGntd6bP2du1tXZb7/mPk+zaez4/ycox7UZ6ywAAADrt29/+ds4///wsWrQor3zlK3PJJZfkD//wD6fc78Au0NJaa0naZLerqpOramlVLV29enUfKgMAAJg5733vezMyMpJbb7015557bg499NB86lOfmnK/Mx32bl87PbP39Y7e8lVJhse0G+ot+3daa2e01g5srR24884797VYAACAzVXfrsa5AecnOS7J+3pfvzBm+Wur6twkByW5e8x0TwAAgBmzYGh4QlfQnEx/E3XIIYfkkEMOmZb99vPWC+ckOSTJTlU1kuQdGQ15n6mqk5IsT3J0r/kFGb3tws0ZvfXCCf2qCwAAYGOWr1wx6BKmRd/CXmvtmA2set562rYkf9avWgAAALY0A7tACwAAAP0j7AEAAHSQsAcAANBBwh4AAEAHzfStFwAAAFiPRYsWZbvttsucOXMyd+7cLF26dEr9CXsAAABjLBhekJUjK6etv+Gh4ayY4O0cvvGNb2SnnXaalv0KewAAAGOsHFmZSz7w2Wnr79A3vnza+poM5+wBAADMAlWVww47LAcccEDOOOOMKfdnZA8AAGAW+Na3vpX58+fnjjvuyPOf//zstddeOfjggze5PyN7AAAAs8D8+fOTJLvsskuOOOKIXHnllVPqT9gDAAAYsHvuuSc///nPf/X8a1/7WhYvXjylPk3jBAAAGLDbb789RxxxRJLkoYceyqte9aocfvjhU+pT2AMAABhjeGh4Wq+gOTw0PG6bPffcM8uWLZu2fSbCHgAAwG+Y6D3xZjvn7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAB0kLAHAAAwC9x111056qijstdee2XvvffO5ZdfPqX+hD0AAIAxFgwNp6qm7bFgAvfZS5JTTjklhx9+eG688cYsW7Yse++995Teh/vsAQAAjLFy1UjOe/tZ09bfK9593Lht7r777lx66aX55Cc/mSTZeuuts/XWW09pv0b2AAAABuyHP/xhdt5555xwwgl56lOfmle/+tW55557ptSnsAcAADBgDz30UK6++ur8yZ/8Sa655ppsu+22ed/73jelPoU9AACAARsaGsrQ0FAOOuigJMlRRx2Vq6++ekp9CnsAAAAD9rjHPS7Dw8O56aabkiQXX3xx9tlnnyn16QItAAAAs8CHP/zhHHvssXnggQey55575swzz5xSf8IeAADAGMPzhyZ0Bc3J9DcRS5YsydKlS6dtv8IeAADAGCtGVg66hGnhnD0AAIAOEvYAAAA6SNgDANgMLBhekKqa1GPB8IJBlw0MkHP2AAA2AytHVuaSD3x2Utsc+saX96kaYHNgZA8AAKCDhD0AAIABu+mmm7JkyZJfPR7zmMfktNNOm1KfpnECAACMMTw0nJFVI9PW39D8oawc53YOT37yk3PttdcmSdasWZP58+fniCOOmNJ+hT0AAIAxRlaN5PRTPjJt/Z1y+msn1f7iiy/O4x//+CxcuHBK+zWNEwAAYBY599xzc8wxx0y5H2EPAABglnjggQdy/vnn5+Uvn/rVdIU9AACAWeIrX/lK9t9//+y6665T7kvYAwAAmCXOOeecaZnCmQh7AAAAs8I999yTiy66KC972cumpT9X4wQAABhjaP7QpK+gOV5/E7HtttvmzjvvnLb9CnsAAABjjHdPvM2FaZwAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAADNipp56afffdN4sXL84xxxyT+++/f8p9CnsAAABjDM0fSlVN22O8++ytWrUqH/rQh7J06dJcd911WbNmTc4999wpvw/32QMAABhj1Y9W5a0nvnva+nvvJ94+bpuHHnoo9913X+bNm5d77703u++++5T3a2SPgRseGp7U/4wMDw0PumQAACZowST/1luwBf6tN3/+/PyX//JfsmDBguy2227Zfvvtc9hhh025XyN7DNzIqpGcfspHJtz+lNNf28dqAACYTitXjeS8t5814favePdxfaxmdvrpT3+aL3zhC/nhD3+YHXbYIS9/+cvzqU99Kn/4h384pX6N7AEAAAzQ17/+9eyxxx7ZeeedM2/evLzsZS/Ld77znSn3K+wBAAAM0IIFC3LFFVfk3nvvTWstF198cfbee+8p9yvsQUctWrhw0leKWrRw4aDLBgDY4hx00EE56qijsv/+++cpT3lKHn744Zx88slT7tc5e9BRy1esyN033zipbbZ/wl59qgYAYPMxf/f5E7qC5mT6G8+73vWuvOtd75q2fSbCHgAAwG8YWTUy6BKmhWmcAAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcNJOxV1X+uqu9X1XVVdU5VbVNVe1TVd6vq5qo6r6q2HkRtAAAAXTDjYa+q5if58yQHttYWJ5mT5JVJ/irJqa21JyT5aZKTZro2AACAQTj99NOzePHi7LvvvjnttNOmpc9BTeOcm+SRVTU3yaOS3Jbk0CT/0Ft/VpKXDqY0AABgSzZ//lCqatoe8+cPbXR/1113Xf72b/82V155ZZYtW5YvfelLufnmm6f8Pmb8PnuttVVV9TdJViS5L8nXklyV5K7W2kO9ZiNJxr/zIAAAwDT70Y9W5cSj/3za+vvEZz600fU33HBDDjrooDzqUY9KkjznOc/J5z73ubzpTW+a0n4HMY1zxyQvSbJHkt2TbJvk8Elsf3JVLa2qpatXr+5TlQAAADNj8eLFueyyy3LnnXfm3nvvzQUXXJCVK1dOud8ZH9lL8h+S/LC1tjpJqupzSZ6ZZIeqmtsb3RtKsmp9G7fWzkhyRpIceOCBbWZKBgAA6I+99947b37zm3PYYYdl2223zZIlSzJnzpwp9zuIc/ZWJPmdqnpUVVWS5yW5Psk3khzVa3Ncki8MoDYAAIAZd9JJJ+Wqq67KpZdemh133DFPetKTptznjIe91tp3M3ohlquT/HOvhjOSvDnJG6rq5iSPTfLxma4NAABgEO64444kyYoVK/K5z30ur3rVq6bc5yCmcaa19o4k71hn8S1Jnj6AcgAAAAbqyCOPzJ133pl58+blox/9aHbYYYcp9zmQsAcAADBb7b77/HGvoDnZ/sZz2WWXTdv+1hL2AAAAxli1amTQJUyLQd1UHQAAgD4S9gAAADpI2AMAALZ4rc3uW3hvSn3CHgAAsEXbZpttcuedd87awNday5133pltttlmUtu5QAsAALBFGxoaysjISFavXj3oUjZom222ydDQ0KS2EfYAAIAt2rx587LHHnsMuoxpZxonAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBM2rh8IJU1aQeC4cXDLpsAIDNztxBFwBsWVaMrMyyM784qW32O+FFfaoGAKC7jOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB00obBXVc+cyDIAAABmh4mO7H14gssAAACYBeZubGVVPSPJ7ybZuareMGbVY5LM6WdhAAAAbLqNhr0kWyd5dK/ddmOW/yzJUf0qCgAAgKnZaNhrrf1Tkn+qqk+21pbPUE0AAABM0Xgje2s9oqrOSLJo7DattUP7URQAAABTM9Gw99kkH0vyf5Ks6V85AAAATIeJhr2HWmv/q6+VAAAAMG0meuuFL1bVn1bVblX1W2sffa0MAACATTbRkb3jel//65hlLcme01sOAAAA02FCYa+1tke/CwEAAGD6TCjsVdV/XN/y1trZm7LTqtohoxd7WZzREcITk9yU5LyMXvHz1iRHt9Z+uin9AwAAbOkmes7e08Y8np3knUlePIX9np7kwtbaXkn2S3JDkrckubi19sQkF/deAwAAsAkmOo3zdWNf90bmzt2UHVbV9kkOTnJ8r+8HkjxQVS9Jckiv2VlJvpnkzZuyDwAAgC3dREf21nVPkk09j2+PJKuTnFlV11TV/6mqbZPs2lq7rdfmx0l23cT+AQAAtngTCntV9cWqOr/3+HJGz6/7/Cbuc26S/ZP8r9baUzMaHH9jymZrrWX0XL711XJyVS2tqqWrV6/exBIAgC3FooULU1WTeixauHDQZQNM2URvvfA3Y54/lGR5a21kE/c5kmSktfbd3ut/yGjYu72qdmut3VZVuyW5Y30bt9bOSHJGkhx44IHrDYQAAGstX7Eid99846S22f4Je/WpGoCZM6GRvdbaPyW5Mcl2SXZM8sCm7rC19uMkK6vqyb1Fz0tyfZLz8+v7+R2X5Aubug8AAIAt3URvvXB0kvdn9KIpleTDVfVfW2v/sIn7fV2ST1fV1kluSXJCRoPnZ6rqpCTLkxy9iX0DAABs8SY6jfNtSZ7WWrsjSapq5yRfz+gUzElrrV2b5MD1rHrepvQHAADAb5ro1Ti3Whv0eu6cxLYAAADMsImO7F1YVV9Nck7v9SuSXNCfkgAAAJiqjYa9qnpCRu9/91+r6mVJntVbdXmST/e7OAAAADbNeCN7pyV5a5K01j6X5HNJUlVP6a17UR9rAwAAYBONd97drq21f153YW/Zor5UBAAAwJSNF/Z22Mi6R05jHQAAAEyj8cLe0qp6zboLq+rVSa7qT0kAAABM1Xjn7L0+yeer6tj8OtwdmGTrJEf0sS4AAACmYKNhr7V2e5LfrarnJlncW/zl1tolfa8MAACATTah++y11r6R5Bt9rgUAAIBpMt45ewAAAGyGhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAOqyqJvxYMDQ86HKBaTR30AUAANA/5739rAm3fcW7j+tjJcBMM7IHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAzAwuEFqaoJP+ZuNWfQJQObmbmDLgAAYEu0YmRllp35xQm33++EF/WxGqCLjOwBAAB0kLAHAADQQcLeLLFo4cJJzdtftHDhoEsGAABmMefszRLLV6zIHZd/d8Ltd3nGQX2sBgAA2NwZ2QMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpoYGGvquZU1TVV9aXe6z2q6rtVdXNVnVdVWw+qNgAAgM3dIEf2Tklyw5jXf5Xk1NbaE5L8NMlJA6kKAACgAwYS9qpqKMkfJPk/vdeV5NAk/9BrclaSlw6iNgAAgC4Y1MjeaUnelOTh3uvHJrmrtfZQ7/VIkvkDqAsAAKATZjzsVdULk9zRWrtqE7c/uaqWVtXS1atXT3N1AAAA3TCIkb1nJnlxVd2a5NyMTt88PckOVTW312Yoyar1bdxaO6O1dmBr7cCdd955JuoFAADY7Mx42GutvbW1NtRaW5TklUkuaa0dm+QbSY7qNTsuyRdmujYAAICumE332XtzkjdU1c0ZPYfv4wOuBwAAYLM1d/wm/dNa+2aSb/ae35Lk6YOsBwAAoCtm08geAAAA00TYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoINmPOxV1XBVfaOqrq+q71fVKb3lv1VVF1XVv/S+7jjTtQEAAHTFIEb2HkryxtbaPkl+J8mfVdU+Sd6S5OLW2hOTXNx7DQAAwCaY8bDXWruttXZ17/nPk9yQZH6SlyQ5q9fsrCQvnenaAAAAumKg5+xV1aIkT03y3SS7ttZu6636cZJdB1UXAADA5m5gYa+qHp3kH5O8vrX2s7HrWmstSdvAdidX1dKqWrp69eoZqBQAAEiSBcMLUlWTeszZas6gy95izR3ETqtqXkaD3qdba5/rLb69qnZrrd1WVbsluWN927bWzkhyRpIceOCB6w2EAADA9Fs5sjKXfOCzk9rm0De+vE/VMJ5BXI2zknw8yQ2ttQ+OWXV+kuN6z49L8oWZrg0AAKArBjGy98wkf5Tkn6vq2t6y/y/J+5J8pqpOSrI8ydEDqA0AAKATZjzstda+laQ2sPp5M1kLAABAVw30apwAAAD0h7AHAADQQcIeAAD02aKFCyd1u4JFCxcOumQ6YCC3XgAAgC3J8hUrcvfNN064/fZP2KuP1bClMLIHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AwBQtXLAgVTWpx9w5cwZdNtBxcwddAADA5m7FypW55fyLJ7XNni9+Xp+qARhlZA8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9rYgC4cXpKom9Vg4vGDQZcOkLRganvSxvmBoeNBls4kWLpj8z7a5c+YMumwA6Lu5gy6AmbNiZGWWnfnFSW2z3wkv6lM10D8rV43kvLefNaltXvHu4/pUDf22YuXK3HL+xZPaZs8XP69P1QDA7GFkDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD3GNanL17tVw2ZvMt/vRQsXDrpcZrlFCxdO+rYIjiuAmTPZW3PN3cqtazYnbr3AuC75wGcn3PbQN768j5UwE+64/LsTbrvLMw7qYyV0wfIVK3L3zTdOapvtn7BXn6oBYF2TvTWX23JtXozsAQAAdJCwBwAA0EGmcQJTUlWTaj93zqbN9Z/MfuY4n4A+mOyxPjw0nBUrV/SpGgAYn7AHTMkt5188qfZ7vvh5m7Qf544yaJM5BhPHIQCDZxonAABAB826sFdVh1fVTVV1c1W9ZdD1bIpNudT4vLmTH2Sd7D42dfpcv+uaU5M/DCe7j6H5Q314pzA58+cPTeq4ne+43eIMDw1P6hgZHhoedMkAzGKzahpnVc1J8tEkz08ykuT/VdX5rbXrB1vZ5MzUpcZnavrcZJ339rMm1f4V7z5u0vt464nvnlT7937i7ZPeB0y3H/1oVU48+s8n3P4Tn/lQH6thNhpZNZLTT/nIhNufcvpr+1gNAJu72Tay9/QkN7fWbmmtPZDk3CQvGXBNAAAAm53ZFvbmJ1k55vVIbxkAAACTUK21QdfwK1V1VJLDW2uv7r3+oyQHtdZeO6bNyUlO7r18cpKbZrxQkmSnJD8ZdBFs8RyHzAaOQ2YDxyGD5hgcnIWttZ3Xt2JWnbOXZFWSsWebD/WW/Upr7YwkZ8xkUfx7VbW0tXbgoOtgy+Y4ZDZwHDIbOA4ZNMfg7DTbpnH+vyRPrKo9qmrrJK9Mcv6AawIAANjszKqRvdbaQ1X12iRfTTInySdaa98fcFkAAACbnVkV9pKktXZBkgsGXQfjMpWW2cBxyGzgOGQ2cBwyaI7BWWhWXaAFAACA6THbztkDAABgGgh7JEmqak1VXTvm8Zbe8rlV9T+q6l/GrHvbRrZbVFWHVNXdvdc3VtXfDO6dsbmoqseOOY5+XFWrxrxuva/XVdVnq+pRvW1urap/7q1bOqavd1fV93rLv1ZVuw/unbG5GfNzbe3xNn8jx+bWG/n5+c11jssDq+qbA3tjzGpjjqNlVXV1Vf1ub/miqrqvt+76qjq7qub11s2rqvf1fkdfXVWXV9ULxvS5pPfz8/B19vWJqrqjqq6b2XfJbDdTx2FVDVfVN3p9fb+qTpn5d7tlmHXn7DEw97XWlqxn+XuSPC7JU1pr91fVdkneuLHtqmpRkstaay+sqkcmuaaqPt9a+3Z/SqcLWmt3JlmSJFX1ziS/aK39Te/1L9YeZ1X16SR/nOSDvU2f21pb974+72+tvb3X/s+T/EVvG5iI+9Y53l4x5vU7M+bY7C3b0M/PJNmlql7QWvtKXyumC8Yed7+X5L1JntNb96+ttSVVNSfJRUmOTvLpJO9OsluSxa21X1bVrmO2SZJjknyr9/XCMcs/meQjSc7u27thczVTx+FDSd7YWru697flVVV1UWvt+v6+vS2PsMcG9UZPXpNkUWvt/iRprf08yTsn2kdr7b6qujbJ/H7UyBbpsiS/vbEGrbWfjXm5bRInJ7Opxj3exvH+JG9LIuwxGY9J8tN1F7bW1lTVlUnmj/kdvUdr7Ze99bcn+UySVFUleXmS5ye5rKq2GfO7/NLef8zCxvTtOGyt3Zbktl77n1fVDRn9W1HYm2bCHms9shfK1npvkhuSrOgFvIls98PW2hFjV1bVjkmemOTSaayVLVRVzU3ygvz6fwZbkq9VVUvyv1trZ4xp+5dJ/mOSu5M8d6ZrZfO3nuNtQ/7dz8/W2nm955cnOaKqnptkYz9LYe1xtE1GR0kOXbdBVW2T5KAkpyR5QkZ/R/9s3XY9v5vR38v/2ps+/AdJ/rEPddMtM34c9v7j4alJvjs9b4GxnLPHWve11paMeZy3boOqOqE3V3tlVQ2vZ7uxQe/ZVbUsyaokX22t/Xgm3gSdtfaXz9IkK5J8vLf8Wa21/TP6B/mfVdXBazdorb2ttTac0Skmr53hetm8beh425Dxfn6+J8l/60OddMva42ivJIcnObs3KpIkj+8dk7cnua219r0J9HdMknN7z8/tvYbxzOhxWFWPzmj4e/1GAiNTIOyxMTcnWdCbS53W2pm9edx3Z/Sm9xtzWWttvyT7Jjmpqpb0s1A6b+wf069rrT2QJK21Vb2vdyT5fJKnr2fbTyc5cuZKpQPWe7xtqtbaJUkemeR3pqc8uq61dnmSnZLs3Fv0r73fv49PckBVvTi//h39mHW3751TdWSSv6iqW5N8OMnha3+fw0T0+zjsXeDlH5N8urX2uT6/nS2WsMcGtdbuzej/aH+kN2S/9h/u1pPo44dJ3pfkzX0pki1WVW075hfGtkkOS3Jd7/UTxzR9SZIbZ75C+A3vSfKmQRfB5qGq9srof6reOXZ572JUb0ny1jG/o0+vqq172+1cVS9P8rwk32utDbfWFrXWFmb0j+rfONUCNqafx2FvtPDjSW5orX0w9I2wx1qPrN+8dPj7esvfltETaK+rqmsyerGCs5L8aBJ9fyzJwU4GZ5rtmuRbvenCVyb5cmtt7blV76vRy+Z/L6Mh0CWd6acN/fz8ldbaBUlWD6A2Nh+/Oo6SnJfkuNbamvW0+79JHlVVz87o9ODVSa6v0dsofCnJzzI6Ve7z62z3j73lqapzMno+6ZOraqSqTurD+2HzNFPH4TOT/FGSQ8f87Pz9fryhLV215iJ1AAAAXWNkDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AGKOqdq2qv6+qW6rqqqq6vKrcnwyAzY6wBwA9vRv9/t8kl7bW9mytHZDklUmGBloYAGwCYQ8Afu3QJA+01j62dkFrbXlr7cNVNaeq3l9V/6+qvldV/ylJquqQqvpmVf1DVd1YVZ/uhcZU1QFV9U+9EcKvVtVuA3pfAGyB5g66AACYRfZNcvUG1p2U5O7W2tOq6hFJvl1VX+ute2pv2x8l+XaSZ1bVd5N8OMlLWmurq+oVSf4yyYl9fQcA0CPsAcAGVNVHkzwryQNJlif57ao6qrd6+yRP7K27srU20tvm2iSLktyVZHGSi3oDfXOS3DZz1QOwpRP2AODXvp/kyLUvWmt/VlU7JVmaZEWS17XWvjp2g6o6JMkvxyxak9Hfr5Xk+621Z/S5ZgBYL+fsAcCvXZJkm6r6kzHLHtX7+tUkf1JV85Kkqp5UVdtupK+bkuxcVc/otZ9XVfv2o2gAWB8jewDQ01prVfXSJKdW1ZuSrE5yT5I3J/lsRqdnXt27AMvqJC/dSF8P9KZ8fqiqts/o79zTMjp6CAB9V621QdcAAADANDONEwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADooP8fbaQ8hx3YeEAAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 1080x576 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure(figsize=(15,8))\n",
    "sns.histplot(data=train_variant[train_variant[\"Gene\"].isin(gene_count.keys())],\n",
    "             x=\"Gene\", hue=\"Class\", multiple=\"dodge\", shrink=.7)\n",
    "plt.title(\"Gene distribution and class\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene             Variation  Class\n",
       "ID                                     \n",
       "0   FAM58A  Truncating Mutations      1\n",
       "1      CBL                 W802*      2\n",
       "2      CBL                 Q249E      2\n",
       "3      CBL                 N454D      3\n",
       "4      CBL                 L399V      4"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_variant.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Gene, Variation, Class]\n",
       "Index: []"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_variant[train_variant[\"Gene\"] == \"RAD51\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "amino_acid_dict = {'C' : 'CYS', 'D' : 'ASP', 'S' : 'SER', 'Q' : 'GLN', 'K' : 'LYS',\n",
    "    'I' : 'ILE', 'P' : 'PRO', 'T' : 'THR', 'F' : 'PHE', 'N' : 'ASN', \n",
    "    'G' : 'GLY', 'H' : 'HIS', 'L' : 'LEU', 'R' : 'ARG', 'W' : 'TRP', \n",
    "    'A' : 'ALA', 'V' : 'VAL', 'E' : 'GLU', 'Y' : 'TYR', 'M' : 'MET'}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Fonction pour la selection\n",
    "\n",
    "## clean_text\n",
    "Nettoie l'article, lower puis enlève les points qui pourrait nuire à la detection des phrases, (comme les \"...\", ou \"fig.A\")\n",
    "\n",
    "\n",
    "## find_match\n",
    "Va détecter la phrase qui contient le mot, puis prend 1 phrase avant et après (au total 3 phrase) (voir diapo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def amino_three(amino):\n",
    "    return amino_acid_dict[amino]\n",
    "    \n",
    "def decompose_variation(variation):\n",
    "    decompose_aa_pos_aa = re.compile(\"([a-z]{1,})(\\d+)([a-z]{1,})\")\n",
    "    list_variation = decompose_aa_pos_aa.search(variation)\n",
    "    if list_variation:\n",
    "        aa1 = list_variation.group(1)\n",
    "        aa2 = list_variation.group(3)\n",
    "\n",
    "        if len(aa1) + len(aa2) == 2:\n",
    "            amino1 = amino_acid_dict[aa1.upper()].lower()\n",
    "            position = list_variation.group(2)\n",
    "            amino2 = amino_acid_dict[aa2.upper()].lower()\n",
    "            return [amino1,position,amino2]\n",
    "    return False\n",
    "\n",
    "def decompose_fusion(variation):\n",
    "    decompose_g1_g2_fusion = re.compile(\"(\\w+)\\s?(\\?|-)\\s?(\\w+)\\? fusion\")\n",
    "    list_variation = decompose_g1_g2_fusion.search(variation)\n",
    "    if list_variation:\n",
    "        gene1 = list_variation.group(1)\n",
    "        gene2 = list_variation.group(3)\n",
    "        return \"(\" + gene1 + \"|\" + gene2 + \")\"\n",
    "    return False\n",
    "        \n",
    "    \n",
    "def decompose_dup(variation):\n",
    "    decompose_mut_pos = re.compile(\"([a-z]{1,})(\\d+)dup\")\n",
    "    list_mut = decompose_mut_pos.search(variation)\n",
    "    if list_mut:\n",
    "        mut = list_mut.group(1)\n",
    "        pos = list_mut.group(2)\n",
    "        return mut + \"\" + pos\n",
    "    return False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "def clean_text(article):\n",
    "    dot3 = re.compile(\"[\\.]{2,}\")\n",
    "    fig = re.compile(\"fig[s]?\\.\")\n",
    "    decimal = re.compile(\"\\d+\\.\\d+\")    \n",
    "    etal = re.compile(\"et al\\.\")\n",
    "    ie = re.compile(\"i\\.e\\.\")\n",
    "    inc = re.compile(\"inc\\.\")\n",
    "    mutation_point = re.compile(\"[p|c]\\.\")\n",
    "    \n",
    "    clean_article = article.lower()\n",
    "    clean_article = dot3.sub(\".\", clean_article)\n",
    "    clean_article = fig.sub(\"\", clean_article)\n",
    "    clean_article = decimal.sub(\"\", clean_article)\n",
    "    clean_article = etal.sub(\"\", clean_article)\n",
    "    clean_article = ie.sub(\"\", clean_article)    \n",
    "    clean_article = inc.sub(\"\", clean_article)    \n",
    "    clean_article = mutation_point.sub(\"\", clean_article)    \n",
    "    \n",
    "    \n",
    "    return clean_article\n",
    "\n",
    "def join_tuple_string(strings_tuple):\n",
    "    return ' '.join(strings_tuple)\n",
    "\n",
    "def find_match(text, word):\n",
    "    clean = clean_text(text)\n",
    "    word = word.lower()\n",
    "    target_sentence = \"([^.]*{}[^.]*\\.)\".format(word)\n",
    "    before_after_target = \"([^.]*\\.){0,1}\"    \n",
    "    match_exp = re.compile(before_after_target + target_sentence + before_after_target)\n",
    "    match_text = match_exp.findall(clean)\n",
    "    final_match = \"\".join(list(map(join_tuple_string, match_text)))\n",
    "    return final_match\n",
    "\n",
    "def extract_match(line):\n",
    "    # Mettre celui qui consomme le moins en 1er --> ceux qui sont facile à faire sans trop de calcul !\n",
    "    # 1er : best match\n",
    "    # 2e ?\n",
    "    \n",
    "    \n",
    "    # Cleaning text --> Already clean, no need    \n",
    "    text = line[\"Text\"]\n",
    "    variation = line[\"Variation\"]\n",
    "    #gene = line[\"Gene\"].lower()\n",
    "    \n",
    "    if len(text) < 10000:\n",
    "        return text,7\n",
    "    # Splice\n",
    "    if \"splice\" in variation:\n",
    "        match_splice = find_match(text, \"splice\")\n",
    "        if len(match_splice) != 0:\n",
    "            return match_splice, 2\n",
    "        \n",
    "    if \"fs\" in variation:\n",
    "        match_fs = find_match(text, \"frameshift\")\n",
    "        if len(match_fs) != 0:\n",
    "            return match_fs, 2\n",
    "        \n",
    "    # Amplification\n",
    "    if 'ampli' in variation:\n",
    "        match_ampli = find_match(text, \"(amplif|increse)\")\n",
    "        if len(match_ampli) != 0:\n",
    "            return match_ampli,3\n",
    "    \n",
    "    # Duplication\n",
    "    if \"dup\" in variation:\n",
    "        decomp_dup = decompose_dup(variation)\n",
    "        if decomp_dup:\n",
    "            match_mut_pos = find_match(text, decomp_dup)\n",
    "            if len(match_mut_pos) != 0:\n",
    "                return match_mut_pos,2\n",
    "        \n",
    "        match_dup = find_match(text, \"dup\")\n",
    "        if len(match_dup) != 0:\n",
    "            return match_dup, 3\n",
    "        \n",
    "    # Try with * --> w802*\n",
    "    if \"*\" in variation:\n",
    "        new_var = variation.replace(\"*\", \"\\\\*\")\n",
    "        match_star = find_match(text, new_var)\n",
    "        if len(match_star) != 0:\n",
    "            return match_star, 1\n",
    "        if \"fs\" in variation:\n",
    "            match_fs = find_match(text, \"fs\\\\*\")\n",
    "            if len(match_fs) != 0:\n",
    "                return match_fs, 3\n",
    "            \n",
    "        match_stop_nonsense = find_match(text,\"(stop|nonsense)\")\n",
    "        if len(match_stop_nonsense) != 0:\n",
    "            return match_stop_nonsense, 2\n",
    "\n",
    "    # Try first match with inital variation value\n",
    "    # Quality score = 1 \n",
    "    initial_match = find_match(text, variation)    \n",
    "    if len(initial_match) != 0:\n",
    "        #print(\"First match ! \", variation)\n",
    "        return initial_match, 1\n",
    "    \n",
    "    \n",
    "    # deletion and insertion\n",
    "    if \"del\" in variation or \"ins\" in variation:\n",
    "        match_delins = find_match(text, \"(deletion|insertion|delet|insert)\")\n",
    "        if len(match_delins) != 0:\n",
    "            return match_delins,2\n",
    "        \n",
    "        match_delins_sentence = find_match(text, \"(del|ins)(\\w|\\s){0,}(del|ins)\")\n",
    "        if len(match_delins_sentence) != 0:\n",
    "            return match_delins_sentence, 3\n",
    "    \n",
    "    # Trunc mutations\n",
    "    if \"trunc\" in variation:\n",
    "        match_trunc = find_match(text, \"trunc\")\n",
    "        if len(match_trunc) != 0:\n",
    "            #print(\"Trunc\", variation)\n",
    "            return match_trunc, 3\n",
    "        \n",
    "        match_shorte = find_match(text, \"shorte\")\n",
    "        if len(match_shorte) != 0:\n",
    "            return match_shorte,5\n",
    "\n",
    "    # Fusion of two genes\n",
    "    # Quality score = 2\n",
    "    if \"fusion\" in variation:\n",
    "        fusion_gene = decompose_fusion(variation)\n",
    "        if fusion_gene:\n",
    "            match_fusion_gene = find_match(text, fusion_gene)\n",
    "            if len(match_fusion_gene) != 0:\n",
    "                #print(\"Fusion gene1 | gene2\", variation)\n",
    "                return match_fusion_gene, 2\n",
    "\n",
    "        # Try to match the word fusion at least..\n",
    "        # Quality score 4 (bad)\n",
    "        match_fusion = find_match(text,\"fusion\")\n",
    "        if len(match_fusion) != 0:\n",
    "            #print(\"FUSION\", variation)\n",
    "            return match_fusion, 4\n",
    "        \n",
    "    aa_pos_aa = decompose_variation(variation)\n",
    "    if aa_pos_aa :\n",
    "        if len(aa_pos_aa) == 3:\n",
    "            # If we success to split variation in 3 group --> aa1 pos aa2\n",
    "            # Second try without the last amino acid --> y371\n",
    "            match_variation_aa_pos = find_match(text, variation[:-1])\n",
    "            if len(match_variation_aa_pos) != 0:\n",
    "                #print(\"y371 aa_pos\", variation)\n",
    "                \n",
    "                return match_variation_aa_pos, 2\n",
    "\n",
    "            # Third try with 3 letter code of amino acid --> tyr371ser\n",
    "            match_aa_pos_aa = find_match(text, \"\".join(aa_pos_aa))\n",
    "            if len(match_aa_pos_aa) != 0:\n",
    "                #print(\"aa_pos_aa\", variation)\n",
    "                return match_aa_pos_aa, 1\n",
    "            \n",
    "            # Try with 3 letter code without the last aa --> tyr371\n",
    "            match_aa_pos = find_match(text, aa_pos_aa[0] + aa_pos_aa[1])\n",
    "            if len(match_aa_pos) != 0:\n",
    "                #print(\"aa_pos\", variation)\n",
    "                \n",
    "                return match_aa_pos,2\n",
    "            \n",
    "            # Match position only --> 371\n",
    "            match_pos = find_match(text, aa_pos_aa[1])\n",
    "            if len(match_pos) != 0:\n",
    "                #print(\"pos\", variation)\n",
    "                \n",
    "                return match_pos,4\n",
    "            # Search word Substitution\n",
    "            match_substitution = find_match(text, \"substitu\")\n",
    "            if len(match_substitution) != 0:\n",
    "                return match_substitution, 3\n",
    "                \n",
    "            # Match position around the real position --> 370 - 379\n",
    "            match_pos_weak = find_match(text, aa_pos_aa[1][:-1] + \"[0-9]\")\n",
    "            if len(match_pos_weak) != 0:\n",
    "                #print(\"pos weak\", variation)\n",
    "                \n",
    "                return match_pos_weak,5\n",
    "            \n",
    "            \n",
    "    # score 6 ?\n",
    "    return [],0\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "r1209w 61 12951 0\n",
      "truncating mutations 88 15584 0\n",
      "r1076c 108 51812 0\n",
      "p691s 178 57832 0\n",
      "g863s 249 23329 0\n",
      "g810s 267 28321 0\n",
      "epigenetic silencing 350 25569 0\n",
      "r1627 354 21957 0\n",
      "c1385 355 21957 0\n",
      "dna binding domain missense mutations 480 32240 0\n",
      "amplification 807 38458 0\n",
      "a546d 968 36167 0\n",
      "ewsr1-etv1 fusion 1060 39717 0\n",
      "l388m 1072 13643 0\n",
      "truncating mutations 1080 77954 0\n",
      "s247f 1091 25303 0\n",
      "s247y 1092 25303 0\n",
      "h1094r 1141 17042 0\n",
      "e1051k 1228 65320 0\n",
      "p654l 1305 38073 0\n",
      "r659p 1307 38073 0\n",
      "r659l 1332 38073 0\n",
      "truncating mutations 1826 29176 0\n",
      "truncating mutations 1902 95852 0\n",
      "e144k 2024 33189 0\n",
      "r647a 2086 30805 0\n",
      "r583a 2087 30805 0\n",
      "truncating mutations 2285 167659 0\n",
      "r724h 2302 52705 0\n",
      "wildtype 3135 52295 0\n",
      "casp8l 3254 126757 0\n",
      "promoter hypermethylation 3256 79699 0\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "ID\n",
       "46      None\n",
       "61      None\n",
       "72      None\n",
       "88      None\n",
       "108     None\n",
       "178     None\n",
       "249     None\n",
       "267     None\n",
       "285     None\n",
       "350     None\n",
       "354     None\n",
       "355     None\n",
       "480     None\n",
       "581     None\n",
       "593     None\n",
       "807     None\n",
       "808     None\n",
       "819     None\n",
       "968     None\n",
       "1060    None\n",
       "1072    None\n",
       "1080    None\n",
       "1091    None\n",
       "1092    None\n",
       "1118    None\n",
       "1129    None\n",
       "1132    None\n",
       "1138    None\n",
       "1139    None\n",
       "1140    None\n",
       "1141    None\n",
       "1148    None\n",
       "1228    None\n",
       "1247    None\n",
       "1257    None\n",
       "1261    None\n",
       "1305    None\n",
       "1307    None\n",
       "1332    None\n",
       "1540    None\n",
       "1826    None\n",
       "1902    None\n",
       "1911    None\n",
       "2024    None\n",
       "2086    None\n",
       "2087    None\n",
       "2285    None\n",
       "2302    None\n",
       "2529    None\n",
       "3135    None\n",
       "3254    None\n",
       "3256    None\n",
       "3305    None\n",
       "dtype: object"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "to_fix.apply(lambda x : check_line(x) , axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "the mre11 complex, composed of rad50, nbs1 and mre11, has an essential role in the maintenance of genomic integrity and preventing cells from malignancy. here we report the association of three mre11 complex mutations with hereditary breast cancer susceptibility, studied by using a case–control design with 317 consecutive, newly diagnosed northern finnish breast cancer patients and 1000 geographically matched healthy controls (p = ). rad50 687delt displayed significantly elevated frequency in the studied patients (8 out of 317, or , 95% ci –, p= ), which indicates that it is a relatively common low-penetrance risk allele in this cohort. haplotype analysis and the screening of altogether 512 additional breast cancer cases from sweden, norway and iceland suggest that rad50 687delt is a finnish founder mutation, not present in the other nordic cohorts. the rad50 ivs3-1g>a splicing mutation leading to translational frameshift was observed in one patient, and the nbs1 leu150phe missense mutation affecting a conserved residue in the functionally important brca1 carboxy-terminal (brct) domain in two patients, both being absent from 1000 controls. microsatellite marker analysis showed that loss of the wild-type allele was not involved in the tumorigenesis in any of the studied mutation carriers, but they all showed increased genomic instability assessed by cytogenetic analysis of peripheral blood t-lymphocytes (p = ). in particular, the total number of chromosomal rearrangements was significantly increased (p = ). these findings suggest an effect for rad50 and nbs1 haploinsufficiency on genomic integrity and susceptibility to cancer.  go to: introduction rad50 is a highly conserved dna double-strand break (dsb) repair factor. together with nbs1 and mre11 it composes the mre11 complex with functions in sensing and early processing of dsb, cell cycle checkpoints, dna recombination and maintenance of telomeres (1–3). this protein complex integrates dna repair with checkpoint signaling through the atm kinase, acting both upstream and downstream of atm in the dna damage response pathway (4–7). even minor disturbances of mre11 complex activity have profound effects on the genomic integrity, and all three components have been implicated in recessive genetic instability disorders. hypomorphic mutations in mre11 and nbs1 result in ataxia-telangiectasia like disorder (atld) and nijmegen breakage syndrome (nbs), respectively (8,9), and recently mutations in rad50 have also been described in a patient with nbs phenotype (t. dörk, personal communication). while no cancer predisposition has been reported in atld (10,11), nbs patients are cancer-prone, particularly to lymphomas. in addition, heterozygous nbs carriers have been suggested to have an increased cancer risk, and germline alterations in nbs1 have been shown to associate with a variety of malignancies, including leukemia, malignant melanoma, prostate, breast and ovarian cancer (12–16).  on the basis of the central role of the mre11 complex in the maintenance of genomic integrity, 151 breast and/or ovarian cancer families were previously screened for germline mutations in the entire coding regions of the rad50, nbs1 and mre11 genes (17). two novel, potentially breast cancer-associated mutations were identified: rad50 687delt leading to frameshift with premature translation stop at codon 234 was observed in two families, and nbs1 leu150phe affecting an evolutionarily conserved residue in the middle of the functionally important brca1 carboxy-terminal (brct) domain was observed in one family. in addition, two other novel rad50 alterations, ile94leu and arg224his classified as unknown variants, were observed. this prompted us to evaluate the overall significance of these alterations in breast cancer susceptibility by studying a second breast cancer cohort. here we report the prevalence of these alleles, and a novel rad50 truncation allele, in an unselected cohort of 317 consecutive, newly diagnosed breast cancer cases and 1000 healthy population controls. for the further evaluation of the geographical occurrence of rad50 687delt, altogether 512 additional breast cancer patients from sweden, norway and iceland were screened for this mutation. we also provide evidence for a pathogenic effect of the three mre11 complex alterations on genomic instability assessed by cytogenetic analysis of peripheral blood lymphocytes.  go to: material and methods subjects  an unselected cohort of 317 breast cancer patients were screened for rad50 687delt, rad50 ile94leu, rad50 arg224his and nbs1 leu150phe alterations. dna samples were isolated from blood lymphocytes of patients operated at the oulu university hospital from april 2000 through july 2004. the diagnoses were pathologically confirmed. the mean age at cancer diagnosis was 58 years (ages varied between 29 and 95). a written informed consent was obtained from all patients, who also filled out a questionnaire concerning breast, ovarian and other cancer prevalence in their family. only patients reporting breast and/or ovarian cancer in first- or second-degree relatives were considered to be familial cases. the healthy control group (n = 1000) used as reference was collected from may through july 2002 and constituted of anonymous cancer-free finnish red cross blood donors originating from the same geographical region as the studied patients (17). the mean age of the controls was 41 years (ages varied between 18 and 66).  the blood samples of the patients identified in oulu selected for chromosomal analysis were collected at least 2 years after the initial breast cancer diagnosis. these patients filled out a questionnaire concerning cancer treatment received and other possible chronic diseases, to exclude those where recent or ongoing radio- and/or chemotherapy, or other medication might have influence on peripheral lymphocyte chromosomal stability. seven of the eight cases had received radiotherapy and the time from last treatment to sample collection for chromosomal analysis was for one patient 9 months, and for the rest 2–4 years. three patients had also received chemotherapy; the time from last treatment was for two cases 2 years and for one case 1 year. the controls used in chromosomal analysis were healthy, age-matched female individuals.  to evaluate the geographical prevalence of rad50 687delt, additional breast cancer cohorts from sweden (lund, 130 cases), norway (oslo, 136 cases) and iceland (reykjavik, 246 cases) were screened for this mutation.  approval for the study was obtained from the ethical boards of the involved university hospital health care districts.  mutation screening  conformation sensitive gel electrophoresis (18) was used for mutation detection. all findings were confirmed by sequencing with the li-cor ir2 4200-s dna analysis system (li-cor, lincoln, ne). the following primers were used: for rad50 687delt and arg224his in exon 5, forward 5′-gtgacagcataatatcccactg-3′, reverse 5′-ttgatttagccag-tccacga-3′; for rad50 ile94leu in exon 3, forward 5′-tgcctttttctcagaaccaac-3′, reverse 5′-gaaaacaaccat-caacttacagacc-3′; for nbs1 leu150phe in exon 4, forward 5′-agttttaaagtactaaaaattgccatc-3′, reverse 5′-acaaagg-gatggagtgggta-3′.  statistical analysis  fisher’s exact test was used to evaluate the differences in allele frequencies between cases and controls. independent samples t-test for normally distributed variables and mann–whitney u-test for non-normally distributed variables was used to determine the statistical difference in the number of different chromosomal aberrations between the two groups categorized according to their genetic status. comparisons were done with spss (version  for windows, spss, chicago, il), and all p values were two-sided.  evaluation of rad50 ivs3-1g>a effect on mrna splicing  lymphoblast cell lines were established from two carriers (br-0516, br-0490) of a novel rad50 ivs3-1g>a mutation identified during screening for the ile94leu alteration, and two non-carriers. cells were grown in rpmi 1640 medium containing 20% fetal bovine serum, 1% l-glutamine and gentamycin (10 μg/ml) at 37°c in a 5% co2 atmosphere. mrna was isolated with the fasttrack®  kit (invitrogen, carlsbad, ca) and cdna was synthesized using the revertaid™ first strand cdna synthesis kit (fermentas, hanover, md). the effect of rad50 ivs3-1g>a on splicing was estimated by cdna-specific amplification of a 488 bp fragment with following primers: forward 5′-aattttggttggacccaatg-3′ (exon 1), reverse 5′-cctgccgaagtgtttctaagg-3′ (exon 5). the presence of incorrectly spliced transcripts was evaluated by electrophoresis in 1% agarose gel and sequencing.  microsatellite marker analysis  dna markers d5s2057, d5s1984 and d5s2002 were used for haplotype analysis of rad50 687delt carriers, and to analyze allelic imbalance (ai) in the available tumors of rad50 687delt and rad50 ivs3-1g>a carriers. markers d8s1800, d8s88, d8s1811 and d8s1724 were used for the corresponding analyses of nbs1 leu150phe carriers. normal and tumor dna were extracted from paraffin-embedded tissues (19). the pcr products were analyzed with the li-cor using an ird800-labelled forward primer, and allele intensity ratios were quantified with the gene profiler  analysis program (scanalytics, fairfax, va). ai was calculated from the formula ai = (t2 × n1)/(t1 × n2), where t1/2 represents tumor and n1/2 the corresponding normal alleles. a value >  or <  was considered to indicate ai, meaning that the intensity of one allele had decreased >40% (20).  cytogenetic analyses  the cytogenetic analysis of peripheral blood t-lymphocytes of rad50 687delt, rad50 ivs3-1g>a, nbs1 leu150phe carriers, and mutation-negative, age-matched healthy controls was carried out in short-term cultures (21). a minimum of 50 giemsa-banded metaphases were evaluated for spontaneous chromosomal abnormalities by light microscopy and photographed with an automatic chromosome analyser (cytovision version , applied imaging, santa clara, ca).  go to: results rad50 687delt is a founder mutation with incomplete penetrance  a cohort of 317 consecutive, newly diagnosed breast cancer patients operated at the oulu university hospital from april 2000 through july 2004 were screened for the 4 mre11 complex alterations. the frequency of the observed alleles was compared with those of 1000 population controls. altogether eight rad50 687delt carriers (8 out of 317) were identified among the studied breast cancer cases, the frequency being significantly higher than in controls (6 out of 1000) (p = , or , 95% ci –). the mean age at diagnosis did not differ between patients with rad50 687delt mutation ( years) and non-carriers ( years). of the studied 317 cases 70 were considered familial, two of which (br-0492, br-0279) were found positive for rad50 687delt. the study of additional affected family members, however, indicated incomplete segregation of the mutation (figure 1a). two rad50 687delt carriers (br-0225, br-0234) had family history of other cancers (at least two cases in first- or second-degree relatives), whereas the rest (br-04126, br-018, br-0077 and br-01163) showed no evidence for clustering of cancer in the family.   1  1 pedigrees of (a) two rad50 687delt (b) one rad50 ivs3-1g>a, and (c) two nbs1 leu150phe mutation-positive families. filled/open symbols indicate cancer/non-cancer status. age at diagnosis, when known, is shown after cancer type (br, breast; csu, . owing to the relatively high frequency of rad50 687delt in the finnish cohort, altogether 512 breast cancer cases from sweden, norway and iceland were screened to obtain more information about the geographical occurrence of this allele. however, rad50 687delt was not observed, which suggests that it is a finnish founder mutation. the common origin of the mutation was confirmed by microsatellite marker analysis of the 14 carriers, which all shared the same haplotype not present in the 8 analyzed non-carriers (data not shown).  other alterations observed in the northern finnish unselected breast cancer cohort  of the other studied rad50 alterations arg224his was observed twice (2 out of 317, %), which did not differ from the controls (9 out of 1000, %), whereas ile94leu was not detected (0 out of 317). consequently, the current results do not support the association of these two changes with breast cancer susceptibility. however, when screening for the ile94leu alteration a novel rad50 splicing mutation, rad50 ivs3-1g>a, was observed in one patient (br-0490) with breast cancer at the age of 56. there were several breast cancer cases in the family (figure 1b), but only two additional ones were available for mutation testing: the sister of the index case was identified as carrier, whereas the maternal cousin tested negative. rad50 ivs3-1g>a was not present in 1000 population controls (table i).  table i table i pathogenica rad50 and nbs1 germline changes observed in unselected northern finnish breast cancer cases the nbs1 leu150phe substitution was observed in two patients (br-0483, br-02114), but not among controls (p = , table i). patient br-0483 had breast cancer at the age of 58, and was considered to be a familial case as her daughter had had ovarian cancer at age 17 (figure 1c). in addition, the daughter tested positive for the leu150phe allele. case br-02114 had breast cancer at the age of 54. she had family history of stomach and thyroid cancer (figure 1c), but no additional dna samples were available for mutation testing. the common origin of the two currently identified nbs1 leu150phe carriers and one previously observed carrier (17) was verified by microsatellite marker analysis (data not shown).  rad50 ivs3-1g>a leads to frameshift  the effect of rad50 ivs3-1g>a on mrna splicing was evaluated by cdna-specific amplification of rad50 exon 3. only one pcr product corresponding to the predicted size was observed in 1% agarose gel. however, sequencing revealed that ivs3-1g>a leads to one base pair deletion between exon 2 and 3 in the mrna sequence (214delg), thus resulting in a premature translation stop at codon 78. this observation was supported by the splice site prediction program (http://www.fruitfly.org/seq_tools/splice.html). even though ivs3-1g>a abolishes the existing acceptor site (score ), it creates a new one right next to the previous site (score ). the rad50 214delg transcript was present at lower levels than the wild-type, which indicates that it is unstable (data not shown).  loss of the wild-type allele is not implicated in the tumorigenesis of rad50 and nbs1 mutation carriers  the ai analysis was performed in the carriers of three potentially pathogenic alterations (table i). of the breast tumors of eight rad50 687delt, one rad50 ivs3-1g>a and two nbs1 leu150phe carriers, only two showed ai: one positive for rad50 687delt and one for nbs1 leu150phe. however, both tumors showed decrease in the intensity of the mutant allele (data not shown), which was confirmed by sequencing.  rad50 687delt, rad50 ivs3-1g>a and nbs1 leu150phe carriers display genomic instability  to investigate whether these three heterozygous mutations lead to genomic instability, cytogenetic analysis of peripheral blood t-lymphocytes from four affected rad50 687delt (br-018, br-0492, br-04126, br-01163), two rad50 ivs3-1g>a (br-0516, br-0490) and two nbs1 leu150phe (br-0483, br-0480) carriers was performed. six mutation-negative age-matched healthy controls were used as reference. the carriers showed a significant increase in the number of chromosomal aberrations (table ii and figure 2) (p = ). particularly, both simple and complex chromosomal rearrangements were found more frequently in carriers than in healthy controls (p =  and p = , respectively), whereas for chromosome/chromatid breaks and deletions the difference did not reach statistical significance (p = ). in contrast, telomeric associations were rare as they were seen only once in one rad50 687delt and in one nbs1 leu150phe carrier. all observed chromosomal aberrations were considered random, as no preference for specific break site or evidence for clonality was observed.   2  2 increase in spontaneous chromosomal aberrations in t-lymphocytes of heterozygous rad50 687delt, rad50 ivs3-1g>a and nbs1 leu150phe carriers. comparison of (a) the frequency of cells with chromosomal aberrations and (b) the frequency of aberration . table ii table ii chromosomal aberrations in heterozygous rad50 687delt, rad50 ivs3-1g>a and nbs1 leu150phe mutation carriers, and healthy controls go to: discussion given the importance of the mre11 complex in the maintenance of genomic integrity and in preventing cells from malignancy, we have evaluated the significance of four mre11 complex germline alterations in breast cancer susceptibility. of these rad50 687delt and nbs1 leu150phe alleles, in addition to a novel rad50 ivs3-1g>a splicing mutation, appeared to associate with predisposition to cancer.  the current results suggest that rad50 687delt is a finnish founder mutation, which confers increased risk of breast cancer. the risk is apparent in women unselected for family history: the frequency of rad50 687delt carriers among patients was 4-fold compared with healthy controls (95% ci –, p = ). no difference in allele frequency between cases with (2 out of 70) or without (6 out of 247) family history of breast and/or ovarian cancer was observed, which is at least partially due to the low number of cases considered as familial. nevertheless, in the two families displaying both rad50 687delt and breast cancer, clustering of the disease cannot be explained by this mutation alone. the incomplete segregation suggests that rad50 687delt is a low-penetrance susceptibility allele, which possibly multiplies the cancer risk in cooperation with susceptibility alleles in other genes and environmental factors, as suggested by the polygenic model for breast cancer susceptibility. according to this model the genetic susceptibility to breast cancer is due to several loci, each conferring a modest independent risk (22). consequently, these low-penetrance alleles might account for a substantial fraction of breast cancer incidence without necessarily causing evident familial clustering. this also suggests the possibility that a significant portion of the seemingly sporadic cancer cases could in fact be due to genetic predisposition (23).  it appears that rad50 687delt does not produce a stable protein product ( attwooll and j. petrini, unpublished data). this dosage variation may have direct influence on dna damage recognition and response pathway activities (24), which is supported by the increase of genetic instability in rad50 687delt carriers. the identification of another rad50 mutation, ivs3-1g>a, associated with genomic instability further suggests the existence of additional cancer susceptibility alleles in this evolutionarily conserved gene. these may, however, not only be very rare, affecting perhaps only one or two families, but also population specifi consequently, this suggests that screening of the rad50 gene for additional germline mutations in other populations might be beneficial.  the association of nbs1 leu150phe with breast cancer susceptibility is supported by the identification of two more carriers in the current cohort (%, 2 out of 317). yet, the families displaying nbs1 leu150phe do not show exclusively breast, but also other malignancies, including gynecological and stomach cancer. this feature was also evident in the previously reported mutation-positive family (17). although nbs1 leu150phe has been so far observed only in affected individuals, its frequency in the studied cohort is too low to reach statistical significance when compared with controls (p = ). however, its pathogenicity is supported by the increase in chromosomal instability in mutation carriers. it is possible that substitution of the evolutionarily conserved leu150 residue is critical for brct domain function. this domain seems essential for the interaction between nbs1 and histone γ-h2ax after irradiation (25), suggesting that it plays an important role in dna damage recognition. indeed, other changes in conserved residues of the brct domain, glygly136–137gluglu and tyr176ala, have been shown to disrupt the mre11 complex nuclear focus formation and to block nbs1 phosphorylation after irradiation (26). additionally, the nbs1 ile171val allele that is pathogenic for acute lymphoblastic leukemia and aplastic anemia (12,27) has been associated with an increase in numerical and structural chromosomal aberrations (27), suggesting a defect in the dna damage response.  the tested heterozygous carriers of rad50 687delt, rad50 ivs3-1g>a and nbs1 leu150phe all showed an increase in genomic instability compared with healthy controls. although based on a small number of individuals analysed, the total number of chromosome aberrations observed in mutation carrier t-lymphocytes (mean , sd ) was increased 7-fold compared with non-carrier healthy controls (mean , sd ) (p = ). in particular, the number of chromosomal rearrangements was significantly increased (p = ). as the mre11 complex has been implicated in multiple cellular responses to dna damage (3), the elevated frequency of spontaneous translocations in mutation carriers could reflect a deficiency in dna repair. in fact, experimental analyses have shown that dsbs are the principal lesions involved in the process of chromosomal aberration formation. defects in the dsb repair pathways may lead to broken chromosomes and chromosome rearrangements, and this genomic instability could be connected to increased mutation rates of cancer-related genes, ultimately contributing to tumorigenesis (28). interestingly, epidemiological studies have shown that people with elevated frequency of chromosomal aberrations in their peripheral blood lymphocytes have a significantly elevated risk of developing cancer. consequently, it has been suggested that the occurrence of chromosomal aberrations could be a relevant biomarker for cancer risk, reflecting either early biological effects of genotoxic carcinogens or individual cancer susceptibility (29).  the elevated spontaneous chromosomal instability of heterozygous mutation carriers suggests an effect for rad50 and nbs1 haploinsufficiency on genomic integrity. indeed, it has previously been indicated that in the case of tumor suppressor genes participating in the maintenance of the genomic stability the gene-dosage effect rather than biallelic inactivation is sufficient to exert a cellular phenotype that leads to tumorigenesis. however, these genes may show differences in tissue-specific protein-dosage thresholds, below which they fail to operate normally (24). for nbs1 heterozygosity and cancer predisposition there is supporting evidence also from mouse models. in addition to lymphomas, heterozygous nbs1 knockdown mice developed a wide array of tumors affecting the liver, mammary gland, prostate and lung. additionally, it was demonstrated that haploinsufficiency, not loss of heterozygosity, was the mechanism underlying tumor development. moreover, this study provided evidence that nbs1 heterozygosity predisposes cells to malignancy most likely because of chromosomal instability and defects in dna repair (30).  as the current results suggest that mre11 complex mutations act in a haploinsufficient manner, it raises the question about the importance of inactivation of these genes also in sporadic cancer cases. the first suggestion of reduced dosage of rad50 at the somatic level already comes from acute myeloid leukemia (aml), since a strong correlation between aml and chromosomal aberrations involving the rad50 containing 5q23-q31 chromosomal region has been observed (31). however, no direct implication of rad50 defects in aml has yet emerged. nevertheless, it is rather interesting that the currently identified rad50 ivs3-1g>a positive family also shows one case of leukemia.  in conclusion, current results support the contribution of rad50 687delt, rad50 ivs3-1g>a and nbs1 leu150phe in hereditary susceptibility to breast cancer, and probably also other cancers. in particular, rad50 687delt seems to be a relatively common low-penetrance breast cancer susceptibility allele in northern finland, and the risk associated with it is apparent in women unselected for family history. the elevated spontaneous chromosomal instability in peripheral blood lymphocytes of heterozygous mutation carriers suggests an effect for rad50 and nbs1 haploinsufficiency on genomic integrity and predisposition to cancer.  approximately 5–10% of all breast and ovarian cancers are due to genetic predisposition.1 germline mutations in two major susceptibility genes, brca1 and brca2, explain most of the hereditary ovarian cancer cases, but worldwide have been observed only in a fraction of the breast cancer families; in finland, about 20%.2–4 a small portion of the remaining cases, perhaps 1–2%, appears to be caused by germline mutations in other cancer susceptibility genes such as tp53, pten, ar, and atm. 5 recently, moderate or low penetrance germline mutations such as 1100delc of chk2 have also been implicated,6 suggesting that in many cases the phenotypic expression of the disease is accounted for by the combined effect of multiple but more subtle mutations in different target genes.5 7 despite all these discoveries, many breast cancer families, especially those with fewer than four cases and/or with later disease onset, have cancers due to mutations in still unknown genes. as the protein products of the genes so far indicated are central players in the pathways involved in cell cycle checkpoint functions and in the sensing, transduction, and repair of dna lesions,5 other similarly acting genes may represent new potential candidates for breast and/or ovarian cancer susceptibility. the mre11 complex, composed of the proteins mre11, nbs1, and rad50, is a vital component in the dna damage response and is at least partially responsible for connecting dna damage detection to dna repair and cell cycle checkpoint functions. the early association of the complex with damaged dna suggests its involvement in sensing dna double strand breaks (dsbs). in eukaryotes, dsb repair is handled by two different pathways, homologous recombination (hr) and non-homologous end joining (nhej), and the initial step in both is the recognition and signalling of the damage by the mre11 complex.8–10 in addition to dna damage detection and repair, the complex has an important role in co-ordinating checkpoint responses and dna replication,11–14 and is also required for the maintenance of telomeres.15 the sites important for the composition and various functional activities of the mre11 complex have been characterised (summarised in fig 1). because of its independent interaction with both nbs1 and rad50, mre11 has been viewed as the core of the complex, whereas the interaction between nbs1 and rad50 is indirect and mediated through mre 16 17 the sites in mre11 responsible for binding to nbs1 and rad50 have both been assigned to the aminoterminal half of the protein.16 18 in nbs1, the mre11 binding domain has been located to the extreme carboxyterminus of the protein,16 and the corresponding site in rad50 has been located to a 40 residue coiled coil region adjacent to the abc domain.19 20 additionally, several other important domains for the complex function have been identified. the dna nuclease activity is restricted to mre11 and specified by phosphodiesterase motifs in the aminoterminal half of the molecule.18 the aminoterminus of nbs1 exhibits a forkhead associated (fha) domain, followed by a brca1 carboxyterminal (brct) domain.21 both of these highly conserved domains are involved in protein–protein interactions and are therefore crucial for the functions of the mre11 complex, including its . . abbreviations: at, ataxia telangiectasia; atld, ataxia telangiectasialike disorde; basc, brca1 associated genome surveillance complex; brct, brca1 carboxyterminus; csge, conformation sensitive gel electrophoresis; dsb, double strand break; fha, forkhead associated; hr, homologous recombination; nbs, nijmegen breakage syndrome; nhej, non-homologous end joining key points n the mre11 complex has an essential role in maintaining genomic integrity and preventing cells from malignancy. the complex components interact with several proteins (for example brca1, atm and chk2) already connected to hereditary breast and ovarian cancer susceptibility, which makes mre11, nbs1, and rad50 suitable candidate genes for predisposition to these cancers. n we screened the affected index cases of 151 families with signs of hereditary susceptibility to breast and/or ovarian cancer for germline mutations in the mre11 complex genes. seven novel heterozygous alterations were observed, three of which were considered potentially disease-related: mre11 913crt (arg305trp), nbs1 448crt (leu150phe) and rad50 687delt (stop codon at 234). n rad50 687delt is expected to cause major truncation in both known protein isoforms and occurs in multiple individuals of two families. curiously, rad50 687delt was also present in some of the controls, but at a lower frequency, probably referring to incomplete penetrance. the mre11 and nbs1 alterations affected highly conserved amino acids and were seen in single cancer cases, but they were both absent in 1000 healthy controls. n our results suggest that these three mutations in the mre11 complex genes associate with hereditary susceptibility to breast and/or ovarian cancer, and may be important in other types of cancer too. this is the first time to our knowledge that germline mutations in rad50 have been linked to genetic disease in humans. 1 www.jmedgenet.com subcellular localisation after dna damage.17 22 rad50 in turn consists of bipartite n- and c-terminal atpase segments, which assemble into a single abc-type cassette at the end of the predicted antiparallel coiled coil.19 a functionally important zinc hook has been identified in the central portion of the coiled coil domain.23 this zinc hook is thought to mediate rad50 dimerisation, which allows the required molecular flexibility of the mre11 complex and keeps it functionally assembled during the dna recombination and repair process.14 the mre11 complex proteins interact with several important protein partners, many of which have been related to cancer predisposition. for instance, following ionising radiation, nbs1 is phosphorylated in an atm dependent manner to ensure proper response to dna damage.11 in turn, mre11 phosphorylation relies on nbs co-operation between a functional mre11 complex and atm is also required for activation of the chk2 kinase in response to dna damage.12 in addition, mre11, nbs1 and rad50 are linked through protein–protein interactions with brca1. after exposure to ionising radiation, brca1 forms nuclear foci that co-localise with the mre11 complex.25 along with brca1, atm, and other proteins, mre11, nbs1, and rad50 are also part of a multi-subunit protein complex referred to as the brca1 associated genome surveillance complex (basc).26 failure in the dna damage response and repair processes causes chromosomal instability that may result in cancer predisposition. for two of the mre11 complex genes, namely mre11 and nbs1, the connection to cancer has already been established: germline mutations in mre11 are associated with the ataxia telangiectasia-like disorder (atld),27 whereas defects in nbs1 cause nijmegen breakage syndrome (nbs).9 21 cells derived from atld and nbs patients display similar characteristics as those from ataxia telangiectasia (at) patients, and at due to germline mutations in the atm gene has been associated with an increased risk of developing various malignancies; breast cancer, lymphoma, and leukaemia in particular.28 as yet, germline mutations in rad50 have not been connected to genetic diseases in humans. however, in mice certain germline rad50 mutations result in growth defects and cancer predisposition.29 because mutations in atm, chk2, and brca1 have all been previously linked to hereditary predisposition to breast and/or ovarian cancer5 it is possible that germline mutations in the individual mre11 complex genes may also be directly involved in predisposition to these cancers. the suggested connection has been supported by a recent study, where the nbs1 657del5 founder mutation was found to be responsible for the occurrence of a small but significant portion of familial breast cancer cases.30 on these grounds, we tested 151 index patients from finnish families with signs of hereditary susceptibility to breast and/or ovarian cancer for possible germline mutations in the mre11, nbs1, and rad50 genes. the results of our study indicate novel disease associated alterations. materials and methods subjects and unaffected controls index cases diagnosed with breast or ovarian cancer of 151 families originating from northern finland were selected for the screening of germline mre11, nbs1, and rad50 mutations. of the studied families, 95 were associated with breast, 29 with breast–ovarian, and 4 with ovarian cancer. these families met the criteria for either high (76 families) or moderate (52 families) genetic susceptibility to breast and/or ovarian cancer. inclusion criteria for the families were one or more of the following: (a) two to three or more cases of breast and/or ovarian cancer in first or second degree relatives; (b) early disease onset ((35 years); (c) bilateral breast cancer; or (d) multiple primary tumours including breast or ovarian cancer in the same individual. the high risk families mainly included three or more cancer cases. all of the high risk families had been screened for germline mutations in brca1, brca2, chk2, and tp53, and eleven of them were found to be brca1 or brca2 mutation positive.4 31 32 besides high or moderate risk families, 23 families showing single cases of breast or ovarian cancer along with multiple cases (two or more) of other kinds of cancer in first or second degree relatives were included in the study. additionally, 1000 anonymous cancer free individuals were used as controls, and the occurrence of novel, potentially pathogenic alterations was also investigated in 192 breast cancer patients without family history of the disease. both groups derived from the same geographical area as the studied families. all patients had given their informed consent for obtaining pedigree data and blood specimens for a study on cancer susceptibility gene mutations. approval to perform the study was obtained from the ethics board of the northern ostrobotnia health care district and the finnish ministry of social affairs and health. screening for germline mutations dna from blood lymphocytes was extracted using the standard phenol–chloroform method or the puregene d- 50k purification kit (gentra, minneapolis, mn, usa). the coding regions and exon–intron boundaries of the mre11, nbs1, and rad50 genes were screened for germline mutations by conformation sensitive gel electrophoresis (csge), which has been shown to be a cost efficient way to scan genes with high sensitivity and specifity.4 33 samples with band shift were reamplified and the sequencing analysis was performed with the li-cor ir2 4200-s dna analysis system (li-cor , lincoln, ne, usa) using the sequitherm exeltm ii dna sequencing kit lc (epicentre technologies, madison, wi, usa). oligonucleotides for csge and sequencing were designed by using primer3 software based on sequence information obtained from publicly available databases. oligonucleotide sequences and pcr conditions are available upon request. loh analysis loss of heterozygosity (loh) analysis was performed to assess whether the tumours of the patients displaying the novel mre11 and rad50 germline mutations had lost the wildtype allele. the flanking microsatellite dna markers used for mre11 were d11s4176 and d11s1757; and for rad50 d5s2110, d5s2057, d5s1984, d5s2002, and d5s2117. the pcr products were analysed with the li-cor dna analysis system using the ird800 labelled forward primer. figure 1 components of the mre11 complex. the functional domains of each protein are marked above the corresponding diagram (see text for more details). in addition, the putative dna binding domain20 (ncbi; locuslink) of mre11 and the rad50 hook construct containing the zinc binding cys-x-x-cys-motif23 are shown. fha, forkhead associated; brct, brca1 carboxyterminal; mbs, mre11 binding sites. 2 electronic letter www.jmedgenet.com evaluation of rad50 687delt expression at the mrna level expression of the mutated allele at the mrna level was evaluated for rad50 687delt. using the fasttrackh  kit (invitrogen, carlsbad, ca, usa), mrna from lymphoblast cell lines was isolated, following which cdna synthesis was carried out with the generacertm kit (invitrogen). the use of the generacertm kit ensured that only transcripts containing full length cdna ends were used in pcr analysis. primers used for the cdna specific amplification of rad50 exon 5 were: forward 59-tttgaagcaaaagtttgatgaga-39 (located in exon 4) and reverse 59- tcattgtcaagtttcattattttagag-39 (located in exon 6). results novel germline alterations the mutation analysis of the mre11, nbs1, and rad50 genes revealed several germline alterations. eleven heterozygous changes in protein encoding regions were observed (summarised in tables 1 and 2). seven of the alterations were novel and three of these were considered potentially disease related. in addition, a large number of different apparently harmless alterations occurring in intron regions was seen (data not shown). mre11 and nbs1 in the mre11 gene, a single mutation in exon 10 was found in one index case with ovarian cancer (case 322, fig 2a). this novel single base change, a transition crt at position 913, resulted in substitution of arginine to tryptophan at codon 305. the position of the involved arginine is extremely conserved between several species, even those distantly related (fig 4) (yeast and flowering plant: data not shown). additionally, the flanking regions on both sides of this residue are well conserved, suggesting functional importance. in contrast to mre11, multiple alterations within the protein encoding region of nbs1 were detected, but only one was considered potentially pathogeni this novel change, a crt transition at nucleotide 448, results in substitution of leucine to phenylalanine at codon 150 and was displayed by one index case diagnosed with breast cancer (case 277, fig 2b). interestingly, this alteration occurs in the middle of the highly conserved brct domain, where the leucine residue at position 150 is conserved also in the mouse, rat, and chicken orthologues (fig 4). both the mre11 and nbs1 mutations were observed in single multi-cancer families including one ovarian or one breast cancer case, respectively, but co-segregation with the cancer phenotype could not be confirmed owing to lack of dna specimens from additional family members. however, both of the mutations were absent from a large cohort of population controls (n = 1000), indicating that these mutations may indeed relate to cancer predisposition (table 1). rad50 in rad50, four novel germline changes affecting exon regions were discovered (tables 1 and 2). of these changes, his68his appears to be a harmless polymorphism, but the roles of ile94leu and arg224his alterations remain more difficult to determine. both of these substitutions affect conserved residues, the former in the remarkably well conserved aminoterminal atpase domain and the latter in the coiled coil domain,19 20 but they also occurred in the healthy controls, although at lower frequency. interestingly, one of the breast cancer families displaying the rad50 ile94leu variant also showed partial co-segregation with the previously identified chk2 ile157thr unknown variant.31 until further clarifying studies have been carried out, the cancer predisposing role of these two rad50 variants remains unknown. however, at least one of the identified rad50 alterations, 687delt, is likely to be disease related. the rad50 687delt truncation mutation the rad50 687delt mutation in exon 5 leads to a premature stop at codon 234 and affects both isoforms of the protein. table 2 observed sequence variants occurring in the exon regions of the mre11 complex genes gene exon nt change effect on protein allele frequency* hereditary cases controls nbs1 2 102gra leu34leu  % (59/151) nd 5 553grc glu185gln  % (66/151) nd 10 1197trc asp399asp  % (76/151) nd 11 1720art leu574ile % (1/151) % (4/307)` 13 2016gra pro672pro  % (56/151) nd rad50 2 204crt his68his % (1/151) nd 3 280arc ile94leu % (2/151) % (3/1000) 5 671gra arg224his % (3/151) % (9/1000) *heterozygotes.  reported by varon et al (1998) or the snp database. `only 307 controls were tested for this alteration. nd, not done. table 1 novel mre11, nbs1 and rad50 mutations gene exon nt change effect on protein allele frequency* hereditary cases sporadic cases controls mre11 10 913crt arg305trp % (1/151) – (0/192) – (0/1000) nbs1 4 448crt leu150phe % (1/151) – (0/192) – (0/1000) rad50 5 687delt frameshift r stop codon at 234 % (2/151) – (0/192) % (6/1000) *heterozygotes. electronic letter 3 www.jmedgenet.com the deletion was found in two separate families (fig 3). in family 5, the index case and her sister (cases 828 and 474), both diagnosed with breast cancer, were carriers of the mutation. in family 6, the index (case 505) with ovarian cancer carried rad50 687delt, but had also tested positive for the brca1 3745delt mutation in one of our previous studies.4 in addition, the deceased mother with ovarian cancer was an obligatory carrier of at least the rad50 687delt mutation. in all the other cases tested, both the rad50 and the brca1 truncation mutation segregated incompletely with the disease. the observation of unaffected rad50 truncation mutation carriers in family 6, and also the surprisingly high prevalence at the population level (%) suggested not only that rad50 687delt, if truly pathogenic, exhibits low penetrance, but also that it might predispose carriers to cancer without clustering of the disease in the pedigrees. in order to test this hypothesis, we screened 192 breast cancer cases without family history of the disease for the rad50 truncation mutation. in parallel, we also screened the same cases for the mre11 and nbs1 mutations, as the disease segregation of these two alterations could not be confirmed from the pedigrees. as none of the tested alterations was observed in the studied cohort, it may suggest that these changes are mainly associated with patients with a family history of cancer. as rad50 687delt is expected to cause truncation of most of the protein product, we evaluated whether these mutant transcripts are eliminated by nonsense mediated decay. lymphoblast cell lines generated from blood specimens of three different mutation positive individuals (fig 3: family 5, case 828; family 6, cases 505 and 061) was used to produce mrna for the cdna synthesis. the results from sequencing analysis unambiguously demonstrated that in all studied cases both the wild type and the truncation allele were still present in the tested mrna pool (data not shown). mre11 and rad50 loh analysis in order to test the nature of the observed mutations, we investigated whether the tumorigenesis thought to be figure 2 families exhibiting the novel (a) mre11 arg305trp and (b) nbs1 leu150phe germline mutations. black circles (females) and boxes (males) indicate individuals with cancer. the grey filling indicates a pre-malignant lesion. the age at diagnosis, when known, is marked after the disease. the case number of individuals tested for a mutation is shown. figure 3 families with the novel rad50 687delt germline mutation. black circles (females) and boxes (males) indicate individuals with cancer. age at diagnosis, when known, is indicated after the malignancy. the case number of individuals analysed and with parallel genetic changes is shown. for a disease free mutation carrier, the age at monitoring is shown after the case number. 4 electronic letter www.jmedgenet.com associated with the novel germline mutations involved loss of the corresponding wild type allele. loh analysis was performed on the available tumour specimens from four different patients (cases 322, 828, 474, and 505; figs 2 and 3) displaying either the mre11 or rad50 mutation, but no loh was observed (data not shown). owing to the lack of suitable tumour material the corresponding loh analysis for the nbs1 mutation could not be performed. discussion the mre11 complex is an integral part of the proper dna damage response and repair of double strand breaks, essential for the wellbeing of any living organism. failure in these highly conserved processes leads to the loss of genomic integrity and consequently to an elevated risk of cancer.14 the cellular functions of the mre11 complex are indispensable, as seen in mice, where major deficiencies in any of the mre11, nbs1, or rad50 genes cause early embryonic lethality and only hypomorphic mutations appear to be viable.34–36 in humans, only a small number of hypomorphic germline mutations affecting the mre11 and nbs1 genes are currently known, mainly associating with the severe and cancer prone phenotypes of atld and nbs, respectively.21 27 37 disease related germline mutations in rad50 have not to our knowledge previously been reported. in the current study, we have examined the prevalence of nbs1, mre11, and rad50 germline mutations in index cases with breast or ovarian cancer from 151 families. of the novel heterozygous sequence alterations observed, arg305trp (mre11), leu150phe (nbs1), and 687delt (rad50) potentially relate to cancer susceptibility (table 1). as the loh analysis of tumours from mre11 or rad50 mutation carriers showed retention of the wild type alleles, it may indicate that these germline mutations are sufficient to cause the observed adverse phenotypic effect. however, silencing of the wild type allele by epigenetic mechanisms (for example by promoter region hypermethylation) cannot be excluded. the association of the novel mre11 and nbs1 missense mutations with hereditary cancer susceptibility is supported by their absence in both the controls and the unselected breast cancer cases. in addition, both amino acid substitutions occur in highly conserved protein domains. in mre11, the arg305trp substitution takes place at the very end of the first of two major protein domains,20 replacing the conserved positively charged hydrophilic arginine with the neutral and highly hydrophobic tryptophan. because of its position in the protein, it is possible that the arg305trp substitution may affect the structural organisation of the functionally active l shaped groove located between domains 1 and 2,38 and perhaps lower the affinity of mre11 for dna binding. the situation could be similar to that indicated for the asn117ser mutation in atld;27 despite the fact that the change occurs relatively remote from the active site, it has been suggested to disturb the macromolecular interaction of mre in nbs1, the leu150phe substitution involves the highly conserved brct domain (fig 4), where the conserved leucine is replaced with the biochemically non-similar phenylalanine. interestingly, another amino acid substitution affecting a conserved residue in the brct domain, ile171val, has previously been associated with cancer (acute lymphoblastic leukaemia).39 additionally, recent site specific mutagenesis experiments by cerosaletti & concannon40 indicate that changes at the conserved residues glygly136–137gluglu or tyr176ala are capable of disrupting mre11 complex nuclear focus formation and blocking nbs1 phosphorylation after irradiation, emphasising the functional importance of this region and especially the role of the conserved amino acids in the brct domain. however, in order to obtain proof of the proposed deleterious nature of the mre11 and nbs1 missense mutations discovered in the current study, the biochemical effects of these alterations has to be demonstrated by functional assays. furthermore, additional case–control series are needed to confirm their possible role of as cancer susceptibility alleles. of the novel germline alterations observed in this study, rad50 687delt was probably the most dramatic as it is predicted to cause about 80% shortening of both isoforms of the protein. the truncated protein would have retained only the n-terminal portion of the sequence necessary for binding of the rad50 dimer to mre11, but would have lost the corresponding and equally important c-terminal site.23 based on studies of lymphoblast cell lines we know that the deletion allele is expressed at the mrna level, but it is currently unknown if it actually gives rise to a truncated protein, how stable this product would be, and whether it has effects on the function of the wild type protein, perhaps by disturbing the functionally important dimerisation of rad50. if a shortened protein actually is produced, but is unstable, the effect is likely to be that of a null mutation. this dosage variation due to haploinsufficiency may have a direct figure 4 novel amino acid substitutions in mre11 and nbs1, and comparison of sequence homology between homo sapiens (hs), rattus norvegicus (rn), mus musculus (mm) and gallus gallus (gg). the sites showing changed residues are boxed. the amino acids conserved in all four species are drawn in blue, biochemically similar residues in green, and dissimilar ones in black. the sequence shown for mre11 extends to its last two phosphodiesterase motifs (pde iv and v)18 38 and for nbs1 corresponds to its brct domain.22 in nbs1 the two sites (136–137 and 176) marked with an asterisk (*) have been found to be critical for mre11 complex nuclear focus formation and phosphorylation of nbs1 after irradiation,40 and the one marked with an arrowhead (171.) has been associated with acute lymphoid leukaemia.39 electronic letter 5 www.jmedgenet.com influence on the activities in dna damage recognition and response pathways, and could modify the cancer risk by increasing genetic instability or by altering the effect of the normal dna damage response.41 however, the possibility that rad50 687delt is a hypomorphic mutation, analogous to the common slavic nbs founder mutation nbs1 657del5,42 also cannot be excluded. interestingly, according to the recent study by gorski et al, the frequency of the nbs1 657del5 mutation in the population controls they used is comparable with that of rad50 687delt in our study (both %). the rad50 687delt mutation was observed altogether in seven individuals, two of whom showed breast and two ovarian cancers, all at relatively young age (41–50 years old). however, in family 6 (fig 3 right) the situation is less straightforward, as at least one of the affected individuals (case 505) showed parallel involvement of the previously identified brca1 3745delt truncation mutation.4 in addition, there were several unaffected carriers of both mutations, but the three daughters of case 505 found positive for either the rad50 or the brca1 mutation, or both, may not yet contribute to the understanding of the disease penetrance owing to their relatively young age. although at present it is difficult to determine the exact role of these two germline mutations in this family, there are several scenarios worth exploring. it is possible that both rad50 and brca1 act independently on the resultant phenotype. on the other hand, co-occurrence of rad50 687delt could also strengthen the effect of the brca1 mutation. interestingly, a recent comparative genome hybridisation targeted linkage analysis has revealed a possible brca1 penetrance modifier locus on 5q in a location close to rad50. 43 44 besides penetrance modification, the rad50 mutation may also in some way influence the tissue specificity of a brca1 associated malignancy. however, it is also possible that in the genetic background provided by this particular family the mutation in rad50 does not significantly increase the risk of cancer in brca1 3745delt carriers. the surprisingly high frequency of rad50 687delt mutation carriers among the healthy population controls and its absence from the breast cancer cases without familial disease background suggests that rad50 687delt may be a new low penetrance susceptibility allele associated particularly with hereditary predisposition to cancer. at present it is unknown whether the rad50 687delt allele exists outside finland, as it is a well known fact that many disease related gene defects can be either unique or highly enriched in the finnish population, owing to founder effects, isolation, and genetic drift.45 however, the situation could also be similar to that observed for the cancer predisposing chk2 1100delc allele, which is enriched in southern finland,31 46 but also occurs at significant frequencies in many other european populations (for example, the uk6 ). in order to obtain more information about the geographical occurrence and the overall relevance of the rad50 687delt mutation, the intention is to study additional breast, ovarian, and other hereditary cancer materials along with suitable control specimens, initially from the nordic countries. in addition, functional studies are needed to reveal the exact details about the molecular consequences of the rad50 687delt mutation. in conclusion, this is the first time that germline mutations in rad50 have been linked to genetic diseases in humans. the rad50 deletion allele reported in this study provides evidence that deleterious germline mutations in the third member of the mre11 complex exist, are viable, and appear to associate with predisposition to cancer. additional investigations will be needed to confirm and follow up these preliminary leads on the role of mre11 complex gene mutations in hereditary susceptibility to cancer.  the genes predisposing to familial breast cancer are largely unknown, but 5 of the 6 known genes are involved in dna damage repair. rad50 is part of a highly conserved complex important in recognising, signalling and repairing dna double-strand breaks. recently, a truncating mutation in the rad50 gene, 687delt, was identified in 2 finnish breast cancer families. to evaluate the contribution of rad50 to familial breast cancer, we screened the whole coding region for mutations in 435 uk and 46 finnish familial breast cancer cases. we identified one truncating mutation, q350x, in one uk family. we screened a further 544 finnish familial breast cancer cases and 560 controls for the 687delt mutation, which was present in 3 cases (%) and 1 control (%). neither q350x nor 687delt segregated with cancer in the families in which they were identified. functional analyses suggested that rad50 687delt is a null allele as there was no detectable expression of the mutant protein. however, the wild-type allele was retained and expressed in breast tumors from mutation carriers. the abundance of the full-length rad50 protein was reduced in carrier lymphoblastoid cells, suggesting a possible haploinsufficiency mechanism. these data indicate that rad50 mutations are rare in familial breast cancer and either carry no, or a very small, increased risk of cancer. altogether, these results suggest rad50 can only be making a very minor contribution to familial breast cancer predisposition in uk and finland. © 2005 wiley-liss,  the genes predisposing to familial breast cancer are largely unknown. there are 2 major breast cancer susceptibility genes, brca1 and brca2, and 4 minor genes, tp53, chek2, atm and pten; but collectively mutations in these genes only account for approximately 20–30% of the familial risk of breast cancer.1, 2, 3 therefore, the susceptibility genes underlying most familial breast cancer remain to be identified.  five of the six known breast cancer susceptibility genes, brca1, brca2, chek2, atm and tp53, encode proteins implicated in genome integrity monitoring that have multiple interactions with each other, both directly and indirectly.4, 5 this suggests that mutations in some of the many other genes involved in the dna damage response pathways may also be involved in breast cancer susceptibility.  the highly conserved mre11/rad50/nbs1 (mrn) complex has a central role in many cellular responses to dna double-strand breaks, including homologous recombination, nonhomologous end-joining, telomere maintenance and dna damage checkpoint activation.6 the complex consists of the large coiled-coil atp-binding cassette atpase rad50, the nuclease mre11 and the checkpoint mediator nbs1.  hypomorphic biallelic nbs1 mutations cause the autosomal recessive condition, nijmegen breakage syndrome, which is characterised by microcephaly, immunodeficiency and tumor predisposition, particularly lymphomas.7 biallelic mre11 mutations cause ataxia-telangiectasia-like disorder (altd), which is characterised by slowly progressive ataxia and ocular apraxia.8 mutations in rad50 have not been associated with a defined human phenotype. however, a hypomorphic mutation of rad50 has been identified in a patient suffering from a syndrome broadly reminiscent of nbs.5 recently, a 687delt truncating rad50 mutation was reported in 2 breast cancer families from northern finland.9  rad50 is a 1,312-amino acid protein that contains 2 abc-atpase domains separated by 2 coiled-coil regions required for intramolecular interactions. a cys-x-x-cys motif located in the middle of the coiled-coil domain functions as a dimerization domain between 2 rad50 arms.10, 11 the 687delt mutation leads to a premature stop at codon 234 and the function of the mutant protein would be predicted to be severely compromised.  to evaluate the role of rad50 in breast cancer predisposition, we have screened the coding region of the gene in 481 uk and finnish breast cancer families and have specifically evaluated the finnish 687delt mutation in 590 finnish familial breast cancer cases and 560 controls. material and methods  patients and controls  familial breast cancer cases and controls were collected from uk and finland. the uk samples were collected through the breast cancer susceptibility collaboration (uk). consent was obtained from all cases and the research was approved by the london multiresearch ethics committee. samples from individuals with breast cancer from 702 familial breast cancer pedigrees were analyzed. all samples were screened for mutations in the full coding sequence of brca1 and brca2, as previously described.12 the breast cancer history in relatives of the index case was quantified using a family history score, with 1 point for an affected first-degree relative or a second primary in the index case,  point for an affected second-degree relative and  for an affected third-degree relative. the majority of cases had a family history score of at least 2 ( at least 2 first-degree relatives with breast cancer in addition to the affected index case, or equivalent) and no case had a family history score less than 1 ( at least one affected first-degree relative in addition to the affected index case, or equivalent). samples from 786 uk controls were obtained from human random control dna panels from the european collection of cell cultures (salisbury, uk).  the finnish series consisted of 590 familial breast cancer patients collected at the helsinki university central hospital, as previously described.13 of these, 338 breast cancer patients had a strong family history (3 or more first- or second-degree relatives with breast or ovarian cancer in the family, including the index case), as verified through the finnish cancer registry and hospital records, and 252 unrelated breast cancer cases had only a single affected first-degree relative. among the 590 patients, 502 had a family history of breast cancer only and 88 had a relative with ovarian cancer. for 435 cases, brca1 and brca2 mutations had been excluded as previously described,14, 15 and for 155 the brca1 and brca2 mutation status was unknown. controls were from 560 anonymous healthy blood donors from the finnish red cross blood transfusion service in helsinki. informed consent from the patients and permission for the study from the ethics committee of the department of obstetrics and gynecology, huch and the ministry of social affairs and health in finland were obtained. rad50 mutation analyses  the coding sequence and intron–exon boundaries of rad50 were analyzed using conformation sensitive gel electrophoresis, csge,16 in 435 uk and 46 finnish familial breast cancer cases. primers and conditions are available on request. genomic dna from cases showing mobility shifts on csge was bidirectionally sequenced using the bigdyeterminator cycle sequencing kit and an abi 3100 automated sequencer (applied biosystems, foster city, ca).  the exon 7, q350x mutation was analysed in 267 additional uk familial breast cancer patient samples and 786 population controls by bidirectional sequencing of exon 7. the finnish 687delt mutation and the q350x mutation were screened in finnish familial breast cancer patient samples and controls by minisequencing (primer extension).17 all positive minisequencing results were confirmed by reamplification from the original genomic dna sample and direct sequencing (abi bigdyeterminator cycle sequencing kit (v), abi prism 310 genetic analyzer (applied biosystems, foster city, ca). protein expression analysis of rad50 687delt  the rad50 protein expression was evaluated by immunoblotting on cell lysates from 3 lymphoblastoid cell lines established from heterozygous carriers of the 687delt mutation, compared with a control lymphoblastoid cell line homozygous for wild-type rad50. for immunoblotting analysis, proteins from total cell lysates were separated using denaturing polyacrylamide gel electrophoresis (10% sds-page) and blotted onto nitrocellulose membrane. the rad50 protein was visualized on parallel blots after incubation with 3 distinct primary antibodies (ab3622 from abcam, cambridge, uk, clone 13 from transduction laboratories, lexington, ky and 13b3 from genetex, san antonio, tx), using the ecl visualization reagents (amersham, piscataway, nj). immunohistochemistry of rad50 expression on 687delt carrier tumors  the rad50 protein expression was studied by immunohistochemical staining in 3 archival breast tumors from the same patients as above, compared to a series of normal breast tissues (n = 15), familial breast carcinomas with apparently wild-type rad50 (n = 25) and sporadic breast tumors (n = 27). for immunohistochemistry, the tumor sections were deparaffinized and processed for sensitive immunoperoxidase staining with the primary mouse monoclonal antibody against human rad50 (clone 2c6, abcam, 1:500 dilution), incubated overnight, followed by detection using the vectastain elite kit (vector laboratories, burlingame, ca), as previously described.18 for analyses of the mre11 and nbs1 proteins on parallel sections, rabbit antibodies no. 4895 against mre11 (cell signaling, diluted 1:1,500) and no. 3002 against nbs1 (cell signaling, diluted 1:100) were used, respectively. given that around 95% of epithelial cells were reproducibly and strongly positive on sections of normal breast, we regarded as an aberrant decrease when the rad50 protein was detectable in fewer than 70% of cancer cells in a particular lesion, and the staining intensity of the remaining positive cancer cells was reduced compared with normal cells present on the same section. bioinformatic analyses  sift-analysis19 (http://blocks.fhcrorg/sift/sift) and polyphen20 (http://genetics.bwh.harvard.edu/pph/) were used to evaluate functional significance of the rad50 missense variants found. sift program calculates tolerance scores for amino acid changes based on sequence alignment and conservation across protein family or across evolutionary history. polyphen (=polymorphism phenotyping) predicts possible impact of an amino acid substitution on the structure and function of a human protein, using straightforward physical and comparative considerations. results  rad50 mutation analysis  we screened the coding region and exon–intron boundaries of the rad50 gene in 435 uk and 46 finnish familial breast cancer cases and identified a single protein truncating mutation 1048c>t, q350x, in a woman who developed bilateral breast cancer at 43 years ( 1a, table i). the mutation was not present in her sibling, who developed breast cancer at 73 years. her mother developed ovarian cancer at 55 years and a paternal aunt developed breast cancer at 50 years. unfortunately, no further samples were available and therefore it is unknown whether the mutation arose de novo in the index case or was inherited from one of her parents, or whether the paternal aunt carried the mutation. to further evaluate the q350x mutation, 267 additional uk familial breast cancer cases, 786 uk controls, 235 finnish breast cancer cases and 319 finnish controls were screened for the mutation by direct sequencing or minisequencing. no further case or control with q350x was identified.  figure 1.  figure 1.  open figure pedigrees from 4 families with rad50 truncating mutations and 3 families with a possibly pathogenic missense mutation. (a) family b1321, a uk family with a q350x mutation. (b) uk families b682, b1339 and b1353 with missense mutations r327h, r224h and r193w, respectively. (c) three finnish families with the 687delt mutation. table i. rad50 variants found in uk and finnish samples nucleotide change\tamino acid change\texon\tuk\tfinnish cases\tcontrols\tcases\tcontrols truncating variants  1048 c>t\tq350x\t7\t1/702\t0/786\t0/281\t0/319  687delt\t \t5\t0/435\t \t3/590\t1/560 missense variants  280 a>c\ti94l\t3\t6/435\t \t0/46\t   373 g>a\tv127i\t4\t1/435\t \t0/46\t   577 c>t\tr193w\t5\t1/435\t \t0/46\t   671 g>a\tr224h\t5\t1/435\t \t0/46\t   695 c>a\ta232d\t5\t1/435\t \t0/46\t   943 g>t\tv315l\t7\t2/435\t \t0/46\t   980 g>a\tr327h\t7\t1/435\t \t0/46\t   2177 g>a\tr726h\t13\t1/435\t \t0/46\t   2525 t>c\tv842a\t16\t1/435\t \t0/46\t  synonymous variants  204 c>t\th68h\t2\t1/435\t \t0/46\t   3879 c>t\ti1293i\t5\t5/435\t \t0/46\t  intronic variants  ivs1+12g>a\t \tivs1\t1/435\t \t0/46\t   ivs9+73c>t\t \tivs9\t2/435\t \t0/46\t   ivs20+35c>t\t \tivs20\t2/435\t \t1/46\t   ivs20+49g>c\t \tivs20\t3/435\t \t0/46\t   ivs22-15delttc\t \tivs22\t1/435\t \t0/46\t   ivs22+24a>g\t \tivs22\t10/435\t \t0/46\t  we identified 11 coding rad50 sequence changes of which 9 were nonsynonymous and 2 were synonymous, 7 changes were intronic (table i). the exonic missense variants were evaluated for possible functional effect by sift and polyphen analysis, which suggested that r193w, which was found in one uk breast cancer case and alters a key residue in the mre11 binding segment, likely affects rad50 function. r224h and r327h may affect rad50 function but the other variants appear well tolerated by sift and polyphen analysis. pedigrees with tumor information of the families carrying the r193w, r224h and r327h missense mutations are shown in figure 1b.  the 687delt mutation found in the finnish population previously was evaluated among index cases from 590 finnish breast cancer families and 560 healthy population controls. the mutation was found in 3 familial patients (%) and one control sample (%). the characteristics of the breast cancer families with the 687delt mutation are shown in figure 1 complete mutation screening of the whole rad50 gene, among 46 finnish breast cancer families, did not reveal any other coding rad50 sequence variants present in breast cancer families from southern finland. the 687delt mutation was not found among the 435 uk families studied. protein expression analysis of rad50 687delt  immunoblotting analysis of rad50 protein in 687delt carrier lymphoblastoid cell lines showed a decreased overall abundance of the full-length protein product, compared with a normal control lymphoblastoid cell line. in addition, the truncated protein form was not detected in lysates from the cell lines with heterozygous rad50 687delt or with any of the 3 antibodies, including an antibody prepared specifically against the n-terminus of rad50, predicted to be preserved in the mutant protein ( 2a).  figure 2.  figure 2.  open figure (a) immunoblot showing a lower level of the full-length rad50 protein in the lymphoblastoid cell line heterozygous for 687delt (right), as compared to that in the noncarrier cell line (left). cdk7 was used as a loading control (the double band signal below). (b) immunohistochemical detection of rad50 in human breast tissues and breast carcinomas. (1) strong nuclear positivity for rad50 in epithelial and stromal cells of normal human breast tissue. insert: higher magnification shows that both luminal (inner layer) and myoepithelial (outer layer) cells are positive. (2) infiltrating ductal carcinoma showing the typical pattern of almost homogeneous rad50 expression. (3)–(5). examples of representative staining patterns for rad50 in breast carcinomas from carriers of the rad50 687delt mutation. (6) aberrant, grossly reduced or lost rad50 staining in a sporadic breast carcinoma contrasts with strong nuclear signal in the surrounding stromal cells. immunohistochemical analysis of 3 breast tumors from 687delt mutation carriers showed normal subcellular localisation of the rad50 protein. the overall staining pattern was nuclear and the signal detected by this semiquantitative immunohistochemistry approach seemed comparable in the 3 mutation carrier tumors, in a set of additional 25 familial breast carcinomas with apparently wild-type rad50, in the control normal breast tissues (n = 15, with both luminal epithelial and myoepithelial cells positive) and most of the 27 sporadic breast tumors examined ( 2b). on the other hand, a small subset of sporadic carcinomas (3 out of 27) showed grossly reduced abundance of the rad50 protein ( 2b). no obvious defects of either the mre11 or nbs1 protein were detected among the 28 familial breast tumors examined for rad50, in contrast to the 3 sporadic carcinomas that showed concomitant reduction in staining of all 3 proteins of the mrn complex. discussion  the possible role of rad50 in familial breast cancer predisposition is intriguing as rad50 participates in critical cellular functions of dna double-strand break repair, functionally interacting with other breast cancer predisposition genes.11, 21 in this study, we have investigated the role of rad50 in familial breast cancer by mutation analysis of the full gene in 481 breast cancer families from uk and finland, and have specifically evaluated a rad50 truncating mutation previously identified in 2 finnish breast cancer families.9  rad50 mutation analysis revealed only a single protein truncating mutation, q350x, in one of 481 breast cancer families studied. the mutation leads to truncation of 962 amino acids of the rad50 protein, including most of the coiled-coil domain, the zinc-hook structure and the carboxyterminal atp-binding domain.10, 22, 23 the mutation was identified in a young-onset bilateral breast cancer case, but was not present in her affected sister, although it is noteworthy that the sister did not develop breast cancer until 73 years. there was a history of breast cancer on the paternal side and ovarian cancer in the mother, but we were not able to evaluate whether or not either of these cases carried the mutation. no other case or control with the mutation was identified despite analysis of 1,488 uk samples and 717 finnish samples. these data indicate that rad50 q350x is rare. therefore, even if it is contributing to cancer predisposition in family b1321, it cannot be making a major contribution to familial breast cancer overall.  we identified 9 missense variants, 2 synonymous changes and 7 intronic rad50 variants (table i). two of the missense variants (i94l and r224h) and one of the synonymous changes (h68h) have been reported previously.9 there was no strong evidence to suggest that any of these variants are breast cancer susceptibility alleles. three variants, r913w, r224h and r327h, may affect rad50 function, but all are rare, each identified in only 1/481 cases, limiting any potential contribution to breast cancer susceptibility, even if they are pathogeni  the protein truncating mutation 687delt was detected in 3/590 (%) finnish familial breast cancer cases and 1/560 controls (%), but was not present in the 435 uk families studied. the mutation showed incomplete segregation with cancer in the families ( 1c). this is consistent with previous data showing that rad50 687delt did not segregate with cancer in one of the 2 families in which it was identified; the family also carried a pathogenic brca1 mutation.9 the combined 687delt frequency in families not tested positive for a brca1/2 mutation from finland does not differ significantly from the frequency in controls (4/730, % vs. 7/1,560, %, or , 95% ci –, p = ). the presence of the mutation at similar frequency in cases and controls and the lack of segregation with cancer suggest that it either carries no increased risk of breast cancer, or is a very low penetrance breast cancer susceptibility allele. the absence of the mutation in uk cases limited our ability to further evaluate the role of rad50 687delt in breast cancer predisposition.  to investigate the functional effect of rad50 687delt, we studied rad50 expression in lymphoblastoid cell lines from mutation carriers. this showed that the overall abundance of the full-length protein product was decreased and the truncated protein was not detected on immunoblotting. rad50 687delt thus appears to be a null allele with no detectable expression of the mutant protein, suggesting it is unlikely to have a dominant-negative effect on rad50 function. in addition, rad50 protein expression and nuclear localization appeared normal in breast tumors from 687delt mutation carriers, suggesting that the wild-type allele was retained and expressed. similarly to rad50, the abundance and localisation of the mre11 and nbs1 proteins appeared normal in all the 28 familial breast tumors examined by immunohistochemistry. the fact that we identified a small subset of sporadic breast tumors with a simultaneous gross reduction of all 3 proteins of the mrn complex is consistent with a previous study of sporadic carcinomas.24 furthermore, the latter cases also document that the antibodies and our assay conditions were appropriate to detect such aberrations at the protein level when they existed. the subset of mrn-deficient carcinomas (ref.24 and the present study) may reflect the recently reported constitutive activation of dna damage checkpoints in preinvasive human breast and other lesions, and the subsequent selection for defects of various components of the dna damage response network as a way to overcome the anticancer barrier of such activated checkpoints during tumor progression.25  our data on tumors and lymphoblastoid cell lines with the 687delt variant of rad50 are consistent with loss of heterozygosity analyses both in our study (data not shown) and in breast tumors from previously reported 687delt carriers, which demonstrated that the wild-type allele was not lost.9 however, the overall amount of rad50 in lymphoblastoid cells heterozygous for 687delt was reduced compared to that in noncarrier cells. these data suggest that rad50 is not acting as a classical tumor suppressor gene in familial breast cancer, although it is possible that rad50 haploinsufficiency is contributing to cancer. such a mechanism has been suggested by mouse models for important dna damage/repair proteins whose defects contribute to tumorigenicity via loss or decrease, including for example the tumor suppressor chk1 kinase.26  in conclusion, our data demonstrate that germ line variants in rad50 occur at very low frequency in breast cancer families from uk and finland. rare truncating variants may be associated with a small increased cancer risk, but do not appear to be acting as high penetrance mutations for breast cancer. rad50 can only be making a very small contribution to familial breast cancer in the uk and finnish populations.\n",
      "l234fs\n",
      "rad50\n"
     ]
    }
   ],
   "source": [
    "ID = 72       \n",
    "print(to_fix.loc[ID,\"Text\"])\n",
    "print(to_fix.loc[ID,\"Variation\"])\n",
    "print(to_fix.loc[ID,\"Gene\"].lower())               "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exemple à la main juste pour tester"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "word_to_search = \"FAM58A\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "article = train_text.loc[0].values[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "34"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article.count(word_to_search)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "clean_match = find_match(article, word_to_search)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Before selection : total lenght 39672 and 34 occurences of the word FAM58A\n",
      "After selection :  total lenght 9589 and 34 occurences of the word FAM58A\n"
     ]
    }
   ],
   "source": [
    "print(\"Before selection : total lenght {} and {} occurences of the word {}\".format(\n",
    "    len(article),article.count(word_to_search) , word_to_search))\n",
    "\n",
    "print(\"After selection :  total lenght {} and {} occurences of the word {}\".format(\n",
    "    len(clean_match),clean_match.count(word_to_search.lower()), word_to_search))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>D171N</td>\n",
       "      <td>4</td>\n",
       "      <td>Introduction  Myelodysplastic syndromes (MDS) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>A122*</td>\n",
       "      <td>1</td>\n",
       "      <td>Introduction  Myelodysplastic syndromes (MDS) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>Fusions</td>\n",
       "      <td>1</td>\n",
       "      <td>The Runt-related transcription factor 1 gene (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>R80C</td>\n",
       "      <td>4</td>\n",
       "      <td>The RUNX1/AML1 gene is the most frequent targe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>K83E</td>\n",
       "      <td>4</td>\n",
       "      <td>The most frequent mutations associated with le...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3316 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene             Variation  Class  \\\n",
       "ID                                          \n",
       "0     FAM58A  Truncating Mutations      1   \n",
       "1        CBL                 W802*      2   \n",
       "2        CBL                 Q249E      2   \n",
       "3        CBL                 N454D      3   \n",
       "4        CBL                 L399V      4   \n",
       "...      ...                   ...    ...   \n",
       "3316   RUNX1                 D171N      4   \n",
       "3317   RUNX1                 A122*      1   \n",
       "3318   RUNX1               Fusions      1   \n",
       "3319   RUNX1                  R80C      4   \n",
       "3320   RUNX1                  K83E      4   \n",
       "\n",
       "                                                   Text  \n",
       "ID                                                       \n",
       "0     Cyclin-dependent kinases (CDKs) regulate a var...  \n",
       "1      Abstract Background  Non-small cell lung canc...  \n",
       "2      Abstract Background  Non-small cell lung canc...  \n",
       "3     Recent evidence has demonstrated that acquired...  \n",
       "4     Oncogenic mutations in the monomeric Casitas B...  \n",
       "...                                                 ...  \n",
       "3316  Introduction  Myelodysplastic syndromes (MDS) ...  \n",
       "3317  Introduction  Myelodysplastic syndromes (MDS) ...  \n",
       "3318  The Runt-related transcription factor 1 gene (...  \n",
       "3319  The RUNX1/AML1 gene is the most frequent targe...  \n",
       "3320  The most frequent mutations associated with le...  \n",
       "\n",
       "[3316 rows x 4 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concatenate_data = pd.merge(train_variant, train_text, on=\"ID\").dropna()\n",
    "concatenate_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "concatenate_data[\"Text\"] = concatenate_data.apply(lambda line: clean_text(line[\"Text\"]), axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "concatenate_data[\"Variation\"] = train_variant[\"Variation\"].apply(lambda line: line.lower())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>cyclin-dependent kinases (cdks) regulate a var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>w802*</td>\n",
       "      <td>2</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>q249e</td>\n",
       "      <td>2</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>n454d</td>\n",
       "      <td>3</td>\n",
       "      <td>recent evidence has demonstrated that acquired...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>l399v</td>\n",
       "      <td>4</td>\n",
       "      <td>oncogenic mutations in the monomeric casitas b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>d171n</td>\n",
       "      <td>4</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>a122*</td>\n",
       "      <td>1</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>fusions</td>\n",
       "      <td>1</td>\n",
       "      <td>the runt-related transcription factor 1 gene (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>r80c</td>\n",
       "      <td>4</td>\n",
       "      <td>the runx1/aml1 gene is the most frequent targe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>k83e</td>\n",
       "      <td>4</td>\n",
       "      <td>the most frequent mutations associated with le...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3316 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene             Variation  Class  \\\n",
       "ID                                          \n",
       "0     FAM58A  truncating mutations      1   \n",
       "1        CBL                 w802*      2   \n",
       "2        CBL                 q249e      2   \n",
       "3        CBL                 n454d      3   \n",
       "4        CBL                 l399v      4   \n",
       "...      ...                   ...    ...   \n",
       "3316   RUNX1                 d171n      4   \n",
       "3317   RUNX1                 a122*      1   \n",
       "3318   RUNX1               fusions      1   \n",
       "3319   RUNX1                  r80c      4   \n",
       "3320   RUNX1                  k83e      4   \n",
       "\n",
       "                                                   Text  \n",
       "ID                                                       \n",
       "0     cyclin-dependent kinases (cdks) regulate a var...  \n",
       "1      abstract background  non-small cell lung canc...  \n",
       "2      abstract background  non-small cell lung canc...  \n",
       "3     recent evidence has demonstrated that acquired...  \n",
       "4     oncogenic mutations in the monomeric casitas b...  \n",
       "...                                                 ...  \n",
       "3316  introduction  myelodysplastic syndromes (mds) ...  \n",
       "3317  introduction  myelodysplastic syndromes (mds) ...  \n",
       "3318  the runt-related transcription factor 1 gene (...  \n",
       "3319  the runx1/aml1 gene is the most frequent targe...  \n",
       "3320  the most frequent mutations associated with le...  \n",
       "\n",
       "[3316 rows x 4 columns]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concatenate_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Explications du problème\n",
    "Fonction poubelle juste pour voir ceux qui sont pas détectable --> len(match) == 0\n",
    "\n",
    "Si ils sont pas détectable, c'est que le nom donné pour leurs variation sont pas dans l'article, donc je vérfie à la main sur le site https://regex101.com/r/lvQZIf/1 qui sert à visualiser les expressions régulières.\n",
    "En copiant collant tout l'article, je regarde comment je peux détecter la variation avec un autre nom ducoup, comme le nom de base ne marche pas.\n",
    "\n",
    "Par exemple Id = 19, y371s n'est pas dans l'article mais il y est sous la forme tyr371ser.\n",
    "\n",
    "Cet regexpr ne marche pas (le texte est en minuscule) -> `([^.]*\\.){0,1}([^.]*y371s[^.]*\\.)([^.]*\\.){0,2}`\n",
    "\n",
    "Mais celle la marchera -> `([^.]*\\.){0,1}([^.]*tyr371ser[^.]*\\.)([^.]*\\.){0,2}` (ça donne 3 match sur le site pour l'article id = 19)\n",
    "\n",
    "(pour les regex, il faut juste changer le nom au milieu après `*` et avant `[^.]`)\n",
    "\n",
    "Autre exemple, ID = 102, mutation r922* qui n'est mentionné sous aucunes formes, ni même la position 922. Mais, il y a des mots qui peuvent faire penser à cette mutation du genre \"nonsense mutations\" car l'étoile (\\*) veut dire que r922 est devenu un codon stop, donc c'est une mutation non sens. Faudrait voir si je peux convertir ces mutations en mot pour faciliter la détection, ou utiliser un champ lexical.\n",
    "\n",
    "\n",
    "Donc là, j'essaie de faire plus ou moins au cas par cas (pas tous, y en a 880...) pour déterminer des groupes de variations indétectable pour revoir la détection.\n",
    "Par exemple, pour les variations de type `y371s`, je peux prendre `y371`, ou tout simplement `371` qui peuvent détecter la variation dans certain cas.\n",
    "\n",
    "\n",
    "Par contre, pour ceux où y a vraiment pas d'infos dans l'article avec la variation, du genre `t1365m` ID = 44; je sais pas quoi faire d'eux :)\n",
    "Car la position 1365 n'est même pas mentionné dans l'article, donc je vois pas comment détecter les infos liées à cette mutation...\n",
    "\n",
    "Soit je garde l'article entier, soit je prend le nom du gène pour la détection, mais le pb avec ça, c'est que ducoup tout ceux qui sont indétectables de ce gène là, auront les mêmes lignes de détecté car je détecterai uniquement en utilisant le nom du gène, donc y aura pas de différences entre eux même si la variation diffère.\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "## Ne lance pas la cellule juste en bas, ça prend un certain temps et de toute façon, tu as déjà la sortie avec tout ceux qui ne marchent pas.\n",
    "\n",
    "La sortie est de la forme : \n",
    "`variation | id | len(article_original) | len(match)`\n",
    "\n",
    "Certaines variations sont des mots complets du genre `truncating mutations`, un peu relou pour la détection, ce qu'on pourrait faire, c'est utiliser des mots qui ressemble à truncating grace à des méthodes de traitements de texte qui va prendre que le préfix par exemple et automatiser ça pour les mots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def check_line(x):\n",
    "    extract = extract_match(x)\n",
    "    score = extract[1]\n",
    "    \n",
    "    if score == 0:\n",
    "        print(x[\"Variation\"].lower(), x.name, len(x[\"Text\"]), score)\n",
    "\n",
    "        \n",
    "#concatenate_data.apply(lambda x : check_line(x) , axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## \"trunc\" : 10 indectectable\n",
    "## \"shorte\" : 7 --> pour shorter shorten\n",
    "le reste, aucunes infos... même \"mutation\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>cyclin-dependent kinases (cdks) regulate a var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>CBL</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>to determine if residual cylindrical refractiv...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>DICER1</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>mesenchymal cell populations contribute to mic...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>RYBP</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>the mouse double minute 2 (mdm2)–p53 interacti...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>TGFBR2</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>signaling by transforming growth factor-β (tgf...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3200</th>\n",
       "      <td>RASA1</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>phospholipase c gamma 1 (plc gamma 1) and p21r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3202</th>\n",
       "      <td>BCL2L11</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>lung cancer is the leading cause of major canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3212</th>\n",
       "      <td>RB1</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>retinoblastoma is a pediatric tumor of the dev...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3252</th>\n",
       "      <td>CASP8</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>evidence exists that alterations of the genes ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3307</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>the runx genes have come to prominence recentl...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>92 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene             Variation  Class  \\\n",
       "ID                                           \n",
       "0      FAM58A  truncating mutations      1   \n",
       "16        CBL  truncating mutations      1   \n",
       "41     DICER1  truncating mutations      1   \n",
       "88       RYBP  truncating mutations      1   \n",
       "100    TGFBR2  truncating mutations      1   \n",
       "...       ...                   ...    ...   \n",
       "3200    RASA1  truncating mutations      1   \n",
       "3202  BCL2L11  truncating mutations      1   \n",
       "3212      RB1  truncating mutations      1   \n",
       "3252    CASP8  truncating mutations      1   \n",
       "3307    RUNX1  truncating mutations      1   \n",
       "\n",
       "                                                   Text  \n",
       "ID                                                       \n",
       "0     cyclin-dependent kinases (cdks) regulate a var...  \n",
       "16    to determine if residual cylindrical refractiv...  \n",
       "41    mesenchymal cell populations contribute to mic...  \n",
       "88    the mouse double minute 2 (mdm2)–p53 interacti...  \n",
       "100   signaling by transforming growth factor-β (tgf...  \n",
       "...                                                 ...  \n",
       "3200  phospholipase c gamma 1 (plc gamma 1) and p21r...  \n",
       "3202  lung cancer is the leading cause of major canc...  \n",
       "3212  retinoblastoma is a pediatric tumor of the dev...  \n",
       "3252  evidence exists that alterations of the genes ...  \n",
       "3307  the runx genes have come to prominence recentl...  \n",
       "\n",
       "[92 rows x 4 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trunc = concatenate_data[concatenate_data[\"Variation\"] == \"truncating mutations\"]\n",
    "trunc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "truncating mutations 88 15584\n",
      "truncating mutations 1080 77954\n",
      "truncating mutations 1826 29176\n",
      "truncating mutations 1902 95852\n",
      "truncating mutations 1911 6154\n",
      "truncating mutations 2146 1407\n",
      "truncating mutations 2285 167659\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "ID\n",
       "0       None\n",
       "16      None\n",
       "41      None\n",
       "88      None\n",
       "100     None\n",
       "        ... \n",
       "3200    None\n",
       "3202    None\n",
       "3212    None\n",
       "3252    None\n",
       "3307    None\n",
       "Length: 92, dtype: object"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trunc.apply(lambda x : check_line(x) , axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fix les classes 4  --> 60 à fix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'bad' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-29-deaeed22b8fe>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mbad_class4\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbad\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"Class\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m4\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mclass4\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mconcatenate_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mbad_class4\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mclass4_tofix\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mclass4\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mclass4\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"Class\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m4\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'bad' is not defined"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'class4_tofix' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-30-3dde65d4650d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mclass4_tofix\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0mcheck_line\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'class4_tofix' is not defined"
     ]
    }
   ],
   "source": [
    "class4_tofix.apply(lambda x : check_line(x) , axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'tgfbr1\\\\*6a'"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = \"tgfbr1*6a\"\n",
    "a = a.replace(\"*\",\"\\\\*\")\n",
    "a\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'tgfbr1*6a'"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "re.findall(\"tgfbr1\\\\*6a\",\"tgfbr1*6a\")[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fix les codons stop --> 60~ -----------> plus que 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>w802*</td>\n",
       "      <td>2</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>CCND3</td>\n",
       "      <td>q276*</td>\n",
       "      <td>7</td>\n",
       "      <td>of the generalizations to emerge from studies ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>TGFBR1</td>\n",
       "      <td>tgfbr1*6a</td>\n",
       "      <td>4</td>\n",
       "      <td>breast cancer is the most common female malign...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>MSH6</td>\n",
       "      <td>r922*</td>\n",
       "      <td>4</td>\n",
       "      <td>identification of a high-risk disease-causing ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>MSH6</td>\n",
       "      <td>e1322*</td>\n",
       "      <td>4</td>\n",
       "      <td>human colorectal cancer cell lines are used wi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3297</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>r177*</td>\n",
       "      <td>4</td>\n",
       "      <td>the aml1 gene is known as the most frequent ta...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3298</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>y113*</td>\n",
       "      <td>4</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3306</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>w279*</td>\n",
       "      <td>1</td>\n",
       "      <td>here we report two new runx1 mutations in one ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3308</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>r174*</td>\n",
       "      <td>7</td>\n",
       "      <td>familial platelet disorder with propensity to...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>a122*</td>\n",
       "      <td>1</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>64 rows × 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene  Variation  Class  \\\n",
       "ID                               \n",
       "1        CBL      w802*      2   \n",
       "77     CCND3      q276*      7   \n",
       "96    TGFBR1  tgfbr1*6a      4   \n",
       "102     MSH6      r922*      4   \n",
       "104     MSH6     e1322*      4   \n",
       "...      ...        ...    ...   \n",
       "3297   RUNX1      r177*      4   \n",
       "3298   RUNX1      y113*      4   \n",
       "3306   RUNX1      w279*      1   \n",
       "3308   RUNX1      r174*      7   \n",
       "3317   RUNX1      a122*      1   \n",
       "\n",
       "                                                   Text  \n",
       "ID                                                       \n",
       "1      abstract background  non-small cell lung canc...  \n",
       "77    of the generalizations to emerge from studies ...  \n",
       "96    breast cancer is the most common female malign...  \n",
       "102   identification of a high-risk disease-causing ...  \n",
       "104   human colorectal cancer cell lines are used wi...  \n",
       "...                                                 ...  \n",
       "3297  the aml1 gene is known as the most frequent ta...  \n",
       "3298  introduction  myelodysplastic syndromes (mds) ...  \n",
       "3306  here we report two new runx1 mutations in one ...  \n",
       "3308   familial platelet disorder with propensity to...  \n",
       "3317  introduction  myelodysplastic syndromes (mds) ...  \n",
       "\n",
       "[64 rows x 4 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "codon_stop = concatenate_data[concatenate_data[\"Variation\"].apply(lambda var: \"*\" in var)]\n",
    "codon_stop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "q233* 1583 738\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "ID\n",
       "1       None\n",
       "77      None\n",
       "96      None\n",
       "102     None\n",
       "104     None\n",
       "        ... \n",
       "3297    None\n",
       "3298    None\n",
       "3306    None\n",
       "3308    None\n",
       "3317    None\n",
       "Length: 64, dtype: object"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "codon_stop.apply(lambda x : check_line(x) , axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fix KIT gene (del, delins...) -> 41 à fix --->  plus que 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "kit_gene = concatenate_data[concatenate_data[\"Gene\"] == \"KIT\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "kit_gene.apply(lambda x : check_line(x) , axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fix EGFR : 10 à fix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "egfr_gene = concatenate_data[concatenate_data[\"Gene\"] == \"EGFR\"]\n",
    "egfr_gene\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "egfr_gene.apply(lambda x : check_line(x) , axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Variable ID pour faciliter le print des deux cellules suivantes pour choper le text, la variation et le gene d'un ID et les copier coller sur le site regex\n",
    "\n",
    "Pour la détection, je teste juste avec le nom de la variation, puis vraiment si y a rien, je regarde du coté du nom du gène"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "ID = 171 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'somatic mutations introduced into the epidermal growth factor receptor (egfr) gene in non-small-cell lung cancer (nsclc) are important factors to determine therapeutic responses to gefitinib. the current diagnostic test measures the overall egfr mutation status of the cancer tissue, and may ignore the presence of nonmutated, gefitinib-unresponsive cancer cells. twenty-one nsclc patients with egfr mutations were recruited for the study. all patients were treated with gefitinib after surgical treatment. fifty to sixty areas of nsclc tumors were sampled from each tissue, and their egfr mutation states were determined by a primer extension assay. this assay discriminates between egfr mutation-positive and -negative cancer cells within a single tumor tissue. fifteen tissues consisted only of cells with egfr mutations, but the remaining six tissues contained both mutated and non-mutated cells. time to disease progression and overall survival after gefitinib treatment were significantly shorter in those patients with egfr heterogeneity (p =  and p = , respectively). a considerable proportion of nsclc contains a heterogeneous population of both egfr mutated and non-mutated cancer cells, resulting in a reduced response to gefitinib. the intratumor genetic heterogeneity of a target molecule such as egfr would be an important factor to consider when treating patients with molecular target agents. (cancer sci 2008; 99: 929–935) lung cancer is a major cause of death. in addition, because many patients are diagnosed at an advanced cancer stage, effective anticancer drugs are urgently needed. tki of egfr ( gefitinib and erlotinib) are a new entity of anticancer drugs and have the potential to improve the therapy of advanced-stage lung cancer.(1) egfr is a transmembrane protein that induces a signal transduction cascade upon ligand binding. it is activated through autophosphorylation of a tyrosine residue in the cytoplasmic domain, causing the receptor to dimerize and internalize, and triggering a cascade leading to cellular proliferation, inhibition of apoptosis, and other cancer-related activities.(2) egfr tki inhibit this cascade by blocking autophosphorylation. there have been extensive studies on the clinical effects of gefitinib. the objective response rate to gefitinib is from 9 to 19%.(3,4) patients’ characteristics, such as adenocarcinoma histology, female sex, non-smoking history, and asian race, are associated with an increased response to gefitinib, and may guide treatment decisions.(3–5) in addition, many studies have shown a correlation between somatic egfr mutations and increased responsiveness to gefitinib.(6–10) such egfr mutations are found frequently in nsclc, and are predominantly located in exons 19 and 21, corresponding to the tyrosine kinase domain. a similar correlation is also reported with erlotinib.(11,12) japanese patients with nsclc have a high proportion of egfr mutations, and these mutations have been correlated with an increased responsiveness to gefitinib.(7,10) however, there are nsclc patients with egfr mutations that have poor responses to gefitinib, suggesting that there are other factors determining responsiveness to gefitinib.(10) one of the biological factors possibly related to the responsiveness of gefitinib is genetic heterogeneity within a single tumor tissue.(13) a tumor tissue usually consists of cancer cells with different genetic alterations, and such intratumor heterogeneity could contribute to resistance to anticancer drugs. in the present report, we examined genetic intratumor heterogeneity of egfr mutations in nsclc tissues using a refined sampling method and mutation detection by primer extension. methods patients and clinical characteristics. we recruited 21 nsclc patients who were treated with gefitinib as monotherapy for postoperative recurrence. surgical resection was complete in all patients except no. 4, whose resection was incomplete due to pleural dissemination. patients had surgery between december 1996 and april 2003 and initiated gefitinib treatment from august 2002 to october 2005 at osaka medical center for cancer and cardiovascular diseases. the ethical committee of osaka medical center for cancer and cardiovascular diseases (chaired by dr hiroshi nakajima) approved the current research, and informed consent was obtained for all cases. all patients were japanese, and clinical information was obtained by reviewing patient medical records. patients were diagnosed with nsclc by histological examination of hematoxylin–eosinstained sections of the formalin-fixed, paraffin-embedded primary tumor. surviving patients were censored at the date of last contact with the institution. tumor recurrences were not necessarily confirmed pathologically but were diagnosed with imaging, clinical manifestation, and laboratory examination. in patients with measurable disease, tumor responses were assessed by the response evaluation criteria in solid tumors(14) and categorized as a complete response, partial response, or no objective response. tumor response was not evaluated in two patients (patients 16 and 17) because these patients had no measurable disease. however, their diseases, which were malignant effusion in the thoracic cavity (patient 16) and pericardiac space (patient 17), were well controlled after drainage with gefitinib treatment for 30 and 450 days, respectively. time to disease progression and overall survival were calculated from the start date of gefitinib therapy to the date of disease progression or death. determination of the egfr mutation status. tumor samples collected during surgery were frozen rapidly in liquid nitrogen 3 to whom correspondence should be addressed. e-mail: katou-ki@mpref.osaka.jp abbreviations: egfr, epidermal growth factor receptor; nsclc, non-small-cell lung cancer; pcr, polymerase chain reaction; snp, single nucleotide polymorphism; tki, tyrosine kinase inhibitor. 930 doi: /j.1349-.00782.x © 2008 japanese cancer association and stored at –80° the tumor was divided into two pieces: one was used for rna extraction and cdna sequencing, and the other was used for tissue sections. total rna was purified from tumor tissues with trizol reagent (invitrogen, carlsbad ca, usa) and purified rna was quantitated and assessed for purity by ultraviolet spectrophotometry. cdna was generated from 5 μg of rna with superscript iii (invitrogen) according to the manufacturer’s protocol. an egfr gene fragment covering exons 18–24 was amplified using primers 5′-ggatcggcctc ttcatgc-3′ and 5′-aactttgggcgactatctgc-3′. the amplified products were sequenced directly using the bigdye terminator cycle sequencing kit (version ; applied biosystems, foster city, ca, usa) with the abi prism 3730 (applied biosystems). after the results of the direct sequencing of pcr products for snap shot (applied biosystems), oligonucleotide primers were designed next to the mutated nucleotides. multipoint microsampling. tumors were cut into 35 μm-thick sections in a cryostat, and stained with % toluidine blue. about 50–60 areas, 104 μm2  in size including 30–50 cells, were selected randomly avoiding normal cells, and microdissected using the as lmd system (leica wetzler, germany). genomic dna was extracted from each sampled area with the qiaamp dna micro kit (qiagen, hilden, germany), and suspended with 20 μl of distilled water. a dna fragment containing the mutated or snp locus was amplified by pcr. polymerase chain reactions consisted of 1 × pcr buffer (toyobo, osaka, japan),  mm mgso4,  mm dntp each  u/μl kod plus dna polymerase and template dna in a 10-μl reaction volume. the base of interest was determined by the snap shot assay, according to the manufacturer’s guidelines with minor modifications. the snap shot assay is based on a primer-extension assay: each extension primer binds to a complementary template in the presence of fluorescently labeled ddntp and dna polymerase, and the polymerase extends the primer by one nucleotide, adding a single dideoxynucleotide to its 3′ end. fluorescent dyes used for dideoxynucleotides were as follows: a, dr6g; c, dtamra; g, dr110; and t, drox. before the snap shot assay,  μl of pcr products were treated with 1 u shrimp alkaline phosphatase (takara, kyoto, japan) and 1 u exonuclease i (takara) for 1 h at 37°c in a final volume of 10 μl, followed by inactivation of the enzymes for 30 min at 80° snap shot thermal cycling was carried out in 5 μl, containing  μl of treated pcr product,  μl snapshot ready reaction premix (applied biosystems), 1 mm extension primer, and  μl distilled water. cycling conditions were as follows: denaturation with 25 cycles of 10 s at 96°c, annealing with 5 s at 50°c, and extension with 30 s at 60° to remove unincorporated ddntp 5 μl of snapshot products were incubated at 37°c for 1 h with 1 u shrimp alkaline phosphatase in a final volume of 6 μl, and the enzyme was deactivated as described above. a total of  μl of treated snap shot reaction was denatured in  μl of distilled water for 5 min at 95°c, and was analyzed using an abi prism 3100 genetic analyzer (applied biosystems). the fragment analysis was done with genescan software (applied biosystems). the mutation status or snp base on each allele was determined from the incorporated nucleotides. because the intrinsic fluorescense intensity of each dye is different, we measured the relative fluorescense intensity ratio, and adjusted the outputs of the dna analyzer. the values used to adjust differences in the intrinsic fluorescence intensities of dyes were as follows: dr110/drox, ; dr6g/dr110, ; and dtamara/drox, . the primer sequences for genomic dna amplification and the snap shot assay are listed in tables s1 and s2, respectively. the snp loci used for control studies were selected from jsnp (a repository of japanese single nucleotide polymorphism)(15) using the following criteria: (1) they were on chromosome 16, which rarely changes in lung cancer; and (2) high frequency of the minor allele. measurement of egfr amplification. the experiment was carried out as described previously,(10) except for the primers used in egfr amplification. the sequences of the primers and taqman probe were 5′-ggaggaccgtcgcttggt-3′, 5′-tcttga catgctgcggtgtt-3′, and 5′-caccgcgacctggcagc ca-3′. control dna was purified from leukocytes of five normal individuals. the threshold of the amplification was set at three as described previously. statistical analysis. statistical analysis (kaplan–meier analysis) was carried out using spss windows version (spss , chicago, il, us). two-sided p-values were calculated using the log-rank test. results principle of the assay. theoretical and experimental studies on colorectal cancer using microsatellite instability have shown that cancer cells of the same genotype locate contiguously.(16) therefore, if a sample excised from a tumor tissue is small enough, it will contain a genetically identical population of cancer cells. if a tumor tissue consists of cancer cells with different egfr mutation states, we can differentiate them by analyzing mutation states of multiple areas. this method is quantitative, and the ratio between gefitinib-sensitive cells (egfr mutation positive) and nonsensitive cells (egfr mutation negative, wild type) can be determined simply by counting the number of areas within a tumor of each genetic type. we analyzed 50–60 areas from each tissue. it should be noted that the sampling must not be contaminated with non-cancerous cells: a pure population was achieved using laser-capture microdissection to observe tissue sections. because the amount of genomic dna is extremely limited in our study, we chose snap shot, a primer extension assay. in snap shot, oligonucleotide primers were designed next to the mutated egfr nucleotide. using fluorescent dye-labeled dideoxynucleotides, a single nucleotide was added by dna polymerase, and the mutation status was determined. first, we determined the optimal sample size to obtain unbiased pcr amplification by changing the areas excised by lasercapture microdissection. we defined the size as approximately 100 × 100 × 35 μm, which enabled four assays and contained 30–60 cells. then, we checked the quantitative ability of the pcr assay. using genomic dna from a small area and serially diluted cloned egfr cdna as a standard, we determined the amount of genomic dna recovered from one small area. we then checked the linearity of assay with various amounts of the cloned dna. the calibration curve is shown in  s1. there was good linearity between inoculated cdna and the fluorescensce peak ratio, confirming the validity of our assay. the outline of the analysis is shown in figure 1. intratumor heterogeneity of egfr mutations. as a preliminary experiment, we examined lung cancer tissues from several patients confirmed to carry egfr mutations. in one cancer tissue, we found that some areas had wild-type egfr (no egfr mutation). a typical example is shown in figure 2. using genomic dna from two areas of this tumor, we observed two peaks representing the mutated and non-mutated alleles. in contrast, sections from two other areas showed peaks derived only from the non-mutated alleles. this tumor is therefore a heterogenous mixture of cancer cells with and without egfr mutations. it should be noted that all excised areas contained only cancer cells and were morphologically indistinguishable. to exclude the possibility that the heterogeneity was an experimental artifact, we carried out the assay with two snp loci, jst003692 and jst004556, which were heterozygous, c and t, in patient no. 17. because the two heterozygous alleles were quantitatively equivalent in all cells, deviation from the equal allele frequency should be due to experimental artifacts. we sampled 60 areas taniguchi  cancer sci | may 2008 | vol. 99 | no. 5 | 931 © 2008 japanese cancer association and analyzed them using the snap shot assay. there were no areas with 0 or 100% t allele. the mean and sd of the t allele was  and , respectively, for jst003692, and  and , respectively, for jst004556. calculating from these values, the chance for 0% t allele is  × 10–5 and  × 10–7, respectively. therefore, we can exclude the possibility that the above heterogeneity ( identi- fication of wild-type egfr) was due to experimental artifacts. additionally, to exclude the possible contamination of normal cells, we determined the mutation status of p53. the analysis of bulk tissue indicated that tissue from patient no. 17 contained only the mutated p53 allele. the spots with only wild-type egfr allele also contained only the mutation p53 allele (data not shown). these spots did not contain normal cells. correlation with response to gefitinib. we then investigated tissues from 21 patients. table 1 is a summary of the clinical information for these patients. cancer tissues were classified into two groups: those consisting only of egfr-mutated cells (15 cases, group i) and those with egfr-mutated and nonmutated cells (six cases, group ii). eight tissues in group i had an equal proportion of mutated and non-mutated egfr (the median was <61%), suggesting that these cells were heterozygous for the mutated egfr gene. examples are shown in figure 3a– divergence of the ratio of egfr-mutated and nonmutated alleles among tumor sections was partly due to that introduced during the assay. we excluded the existence of other reported mutations by sequencing the amplified products. in addition, with samples from patients 17, 19, and 20, we confirmed that the lack of mutated alleles was not due to chromosome copy number variation, using snp markers to confirm that chromosome 7p was a heterozygote. the other seven tissues in group i also consisted of the mutated and non-mutated genes, but the frequency of the mutated egfr gene was higher than the non-mutated gene. examples are shown in figure 3d–f. this increase in the mutant egfr was due either to specific amplification of the mutated egfr gene or to polyploidy ( dominance of the chromosome carrying the mutated gene). the remaining six tissues consisted not only of egfr-mutated cells but also of non-mutated cells ( 3g–l). the proportion of the non-mutated cells varied from  to 100%. patient no. 21 did not contain any mutated cells: additionally we confirmed the snap shot results by dna sequencing, that is, direct and sampled sequencing of 68 cdna clones from the block used for the snap shot analysis. the discrepancy from the initial analysis was due to heterogeneity among the sample, because we used a different part of the tumor for the initial sequencing analysis. we evaluated the effect of intratumor heterogeneity on the responsiveness to gefitinib therapy by measuring time to disease progression and overall survival. as shown in figure 4, both time to progression (median of group i, 18 months; median of group ii,  months; p = ) and overall survival (median of group i, 27 months; median of group ii,  months; p = ) were significantly less in group ii. there was no significant difference both in time to progression and overall survival after gefitinib treatment between patients heterozygous for the egfr mutation (patients 1–8) and those with the dominant mutated egfr allele (patient 9–15) (time to progression, p = ; overall survival, p = ) ( s2). the number of cases with egfr amplification was four in group i and two in group ii (table 1). thus, the differences were not due to egfr amplification. discussion epidermal growth factor receptor tki are potentially effective anticancer agents, but only a fraction of patients respond to egfr tki therapy. it is important to distinguish patients who are most likely to respond. in japan, somatic mutations found in the egfr tyrosine kinase domain are the best predictive factor of responsiveness to egfr tki, although there are contradictory studies in other nations.(17,18) so far, egfr mutation status has been viewed as homogeneous within a single tumor tissue. however, we found that 6 out of 21 mutation-positive nsclc consisted not only of mutated cells but also of nonmutated cells. in addition, these patients did not respond well to gefitinib: both time to progression and overall survival after gefinitib treatment were significantly reduced compared to patients with only egfr-mutant tumor cells. non-mutated cells are likely to be unresponsive to gefitinib therapy and contribute to disease progression. unfortunately, this hypothesis is not proven due to the unavailability of tissue after treatment. in particular, it is important to examine the fraction of non-mutated cells in recurrent tumors. another possibility is that mutation-positive cells of this tissue type would not respond well to gefitinib. through the course of gefitinib therapy, patients that initially respond to gefitinib lose responsiveness. a second egfr mutation at amino acid position 790 (t790m) was found in tumors that acquired resistance to gefitinib.(19,20) it is interesting to compare the presence of t790m and egfr heterogeneity with disease progression during gefitinib therapy. however, this direct  1. outline of the analysis method. egfr, epidermal growth factor receptor; lcm, laser capture microdissection. 932 doi: /j.1349-.00782.x © 2008 japanese cancer association table 1. summary of clinical information of patients  2. an example of the intratumor heterogeneity of the epidermal growth factor receptor mutation. the tissue is patient no. 17 in table 1. (left) a microscopic view of a lung tumor section. red circles indicate regions excised by laser-capture dissection. (right) electropherograms of the snap shot assay of the corresponding regions. blue peak, amplified fragment with ddg; green peak, amplified fragment with dda. patient no. sex stage histology mutation wt area (%) mutated allele (%) heterogeneity group egfr amplification response period between surgery and first gefitinib (months) outcome 1 male iii a ad l858r   i  pr 51 dead 2 female ii b ad l858r   i  pr 82 alive 3 male ii a ad l858r   i  cr 88 alive 4 female iii b ad l858r   i  cr 7 dead 5 female iii a ad l858r   i  cr 52 alive 6 female iii a ad dele746-a750   i  cr 88 dead 7 female ii b ad dele746-a750   i  pr 48 alive 8 female i a ad dele746-a750   i  pr 2 alive 9 male ii b ad l858r   i  pr 38 dead 10 female i a ad dele746-a750   i  pr 18 alive 11 female i a ad dele746-a750   i  pr 17 alive (lost from follow up) 12 female i b ad dele746-a750   i  pr 37 alive 13 female i b ad l861q, g719s   i  pr 35 alive 14 female iii a ad dell747-t751   i  pr 11 alive (lost from follow up) 15 female i b ad dele750-k758   i  pr 31 alive 16 female ii b ad l858r   ii  sd 16 dead 17 male iii a ad dele746-a750   ii  pr 13 dead 18 female iii a ad dele746-a750   ii  cr 39 alive 19 female iii b ad e746-t751 insip   ii  pr 42 dead 20 female ii b ad dele746-a750   ii  pr 12 dead 21 male iii b adsq dela859-l883 insv  na ii  pd 73 dead mutated allele (%), ratios of the mutated allele of all areas was calculated, and their median value is shown. wt area (%), fraction of areas consisting only of the non-mutant allele. ad, adenocarcinoma; adsq, adenosquamous carcinoma; cr, complete response; egfr, epidermal growth factor receptor; na, not applicable; pd, progressive disease; pr, partial response; sd, stable disease. taniguchi  cancer sci | may 2008 | vol. 99 | no. 5 | 933 © 2008 japanese cancer association comparison was difficult for the following reasons. first, we observed egfr tumor heterogeneity before gefitinib therapy, whereas most of the t790m mutations were found in tissues after gefitinib therapy.(19–21) the presence of the t790m mutation in nsclc before therapy is very rare.(22) therefore, the t790m mutation is not informative for predicting gefitinib responsiveness. second, the reported cases of nsclc with the t790m mutation initially responded to gefitinib and then acquired resistance. data for gefitinib non-responders are limited.(22) the patients with egfr heterogeneity did not respond as well as other patients, and two out of six cases were categorized as non-responders. egfr heterogeneity is likely to correlate with the initial response to gefitinib rather than the frequency of acquired resistance. additional studies with biopsy samples during and after gefitinib therapy will provide a clearer picture of the effects of both the t790m mutation and egfr heterogeneity on gefitinib responsiveness. the overall response rate of this study was 81% (17 out of 21), and similar to that in egfr-mutated patients reported in japan. in general, time to progression is a better indicator of gefitinib efficacy than tumor response. in our case, due to the small sample size, the difference in the gefitinib effect was apparent with time to progression, and not with tumor response.  3. intratumor heterogeneity of the epidermal growth factor receptor (egfr) gene. vertical axis indicates the percentage of the mutated egfr gene. horizontal axis areas ordered by the percentage of the mutated gene. blue regions correspond to the mutated genes, and purple regions correspond to the non-mutated genes. columns only consisting of purple regions represent areas consisting of non-mutated cells. each panel corresponds to the following patients: (a) 1; (b) 4; (c) 5; (d) 13; (e) 14; (f) 15; (g) 16; (h) 17; (i) 18; (j) 19; (k) 20; and (l) 21. 934 doi: /j.1349-.00782.x © 2008 japanese cancer association in the present study, the egfr mutation status was obtained from primary tumors, but gefitinib therapy was used for patients who had relapsed or developed metastatic tumors. one may wonder whether the egfr mutation status is the same both in primary and metastatic tumors. although there is no data on egfr to address this possibility, there are studies on her2. trastuzumab is an effective therapy for breast cancer patients that overexpress her2, and patients are selected by a diagnostic test to detect her2 expression levels. her2 expression in primary and metastatic lesions of the same patients was examined, and the degree of 90–95% of cases was concordant in both lesions.(23,24) intratumor genetic heterogeneity has been considered as a potential cause of anticancer drug resistance. in addition, there have been preliminary studies to elucidate the heterogeneity.(16,25–28) however, there have been no systematic studies to link tumor heterogeneity with drug resistance. we introduced a new technical framework, which is simple in principle but practically demanding. the main obstacles are the sampling step and numerous pcr reactions. these technical problems will be solved with the development of a specific sampling machine. although there are other technical possibilities, our method of sectioning tumor tissue into small parts allows a variety of analytical techniques, including gene expression and epigenetic analysis. we are now expecting a second generation of molecular target therapies.(29) selecting patients who are most likely to benefit is important, and intratumor heterogeneity will be an important factor for evaluating their efficacy.'"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concatenate_data.loc[ID,\"Text\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "a859_l883delinsv egfr\n"
     ]
    }
   ],
   "source": [
    "print(concatenate_data.loc[ID,\"Variation\"],\n",
    "concatenate_data.loc[ID,\"Gene\"].lower())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CLEAN DATA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "done in 411.4533226490021 s\n"
     ]
    }
   ],
   "source": [
    "start = time.time()\n",
    "clean_match_data = concatenate_data.apply(lambda x: extract_match(x), axis = 1)\n",
    "clean_match = pd.DataFrame(list(clean_match_data), columns = [\"Text\",\"Score\"], index = clean_match_data.index)\n",
    "print(\"done in {} s\".format(time.time() - start))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "ID\n",
       "0       (here, we deorphanize cdk10 by identifying cyc...\n",
       "1       ( the c-cbl loh also positively correlated wit...\n",
       "2       ( the c-cbl loh also positively correlated wit...\n",
       "3       ( most of the changes were novel, although 4 c...\n",
       "4       ( all mutations from the second group were pre...\n",
       "                              ...                        \n",
       "3316    (21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutati...\n",
       "3317    ( (a) spleen from morbid mice/d171n (left) and...\n",
       "3318    ( lpxn is preferentially expressed in hematopo...\n",
       "3319    () conversely, mutations mapping to the first ...\n",
       "3320    ( mutation analysis in fpd/aml pedigrees. elec...\n",
       "Length: 3316, dtype: object"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clean_match_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "full_x_data = pd.DataFrame(clean_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'new_data' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-22-5ac3197fac06>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mnew_data\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mnew_data\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"Score\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mline\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mline\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"Text_x\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'new_data' is not defined"
     ]
    }
   ],
   "source": [
    "new_data[new_data[\"Score\"] == 1].apply(lambda line: print(len(line[\"Text_x\"])), axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Text</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>here, we deorphanize cdk10 by identifying cycl...</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>most of the changes were novel, although 4 ca...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>all mutations from the second group were pred...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutatio...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>(a) spleen from morbid mice/d171n (left) and ...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>lpxn is preferentially expressed in hematopoi...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>) conversely, mutations mapping to the first d...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>mutation analysis in fpd/aml pedigrees. elect...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3316 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Text  Score\n",
       "ID                                                            \n",
       "0     here, we deorphanize cdk10 by identifying cycl...    3.0\n",
       "1      the c-cbl loh also positively correlated with...    1.0\n",
       "2      the c-cbl loh also positively correlated with...    1.0\n",
       "3      most of the changes were novel, although 4 ca...    1.0\n",
       "4      all mutations from the second group were pred...    1.0\n",
       "...                                                 ...    ...\n",
       "3316  21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutatio...    1.0\n",
       "3317   (a) spleen from morbid mice/d171n (left) and ...    2.0\n",
       "3318   lpxn is preferentially expressed in hematopoi...    1.0\n",
       "3319  ) conversely, mutations mapping to the first d...    1.0\n",
       "3320   mutation analysis in fpd/aml pedigrees. elect...    1.0\n",
       "\n",
       "[3316 rows x 2 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_x_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "full_y_data = concatenate_data.loc[:,[\"Gene\",\"Variation\",\"Class\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Text</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>350</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>480</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>807</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>968</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1060</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1072</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1080</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1091</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1092</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1141</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1228</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1305</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1307</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1332</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1826</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1902</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2024</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2086</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2285</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2302</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3135</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3254</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3256</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "     Text  Score\n",
       "ID              \n",
       "61     []      0\n",
       "88     []      0\n",
       "108    []      0\n",
       "178    []      0\n",
       "249    []      0\n",
       "267    []      0\n",
       "350    []      0\n",
       "354    []      0\n",
       "355    []      0\n",
       "480    []      0\n",
       "807    []      0\n",
       "968    []      0\n",
       "1060   []      0\n",
       "1072   []      0\n",
       "1080   []      0\n",
       "1091   []      0\n",
       "1092   []      0\n",
       "1141   []      0\n",
       "1228   []      0\n",
       "1305   []      0\n",
       "1307   []      0\n",
       "1332   []      0\n",
       "1826   []      0\n",
       "1902   []      0\n",
       "2024   []      0\n",
       "2086   []      0\n",
       "2087   []      0\n",
       "2285   []      0\n",
       "2302   []      0\n",
       "3135   []      0\n",
       "3254   []      0\n",
       "3256   []      0"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index_zero = clean_match[clean_match[\"Score\"] == 0]\n",
    "index_zero"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TEXT_X et TEXT_Y pour check si les articles correspondent --> Pas de pb de décalages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "clean_match.index.name = \"ID\"\n",
    "new_data = pd.merge(concatenate_data,clean_match, on = \"ID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text_x</th>\n",
       "      <th>Text_y</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>cyclin-dependent kinases (cdks) regulate a var...</td>\n",
       "      <td>here, we deorphanize cdk10 by identifying cycl...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>w802*</td>\n",
       "      <td>2</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>q249e</td>\n",
       "      <td>2</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>n454d</td>\n",
       "      <td>3</td>\n",
       "      <td>recent evidence has demonstrated that acquired...</td>\n",
       "      <td>most of the changes were novel, although 4 ca...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>l399v</td>\n",
       "      <td>4</td>\n",
       "      <td>oncogenic mutations in the monomeric casitas b...</td>\n",
       "      <td>all mutations from the second group were pred...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>d171n</td>\n",
       "      <td>4</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "      <td>21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutatio...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>a122*</td>\n",
       "      <td>1</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "      <td>(a) spleen from morbid mice/d171n (left) and ...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>fusions</td>\n",
       "      <td>1</td>\n",
       "      <td>the runt-related transcription factor 1 gene (...</td>\n",
       "      <td>lpxn is preferentially expressed in hematopoi...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>r80c</td>\n",
       "      <td>4</td>\n",
       "      <td>the runx1/aml1 gene is the most frequent targe...</td>\n",
       "      <td>) conversely, mutations mapping to the first d...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>k83e</td>\n",
       "      <td>4</td>\n",
       "      <td>the most frequent mutations associated with le...</td>\n",
       "      <td>mutation analysis in fpd/aml pedigrees. elect...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3316 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene             Variation  Class  \\\n",
       "ID                                          \n",
       "0     FAM58A  truncating mutations      1   \n",
       "1        CBL                 w802*      2   \n",
       "2        CBL                 q249e      2   \n",
       "3        CBL                 n454d      3   \n",
       "4        CBL                 l399v      4   \n",
       "...      ...                   ...    ...   \n",
       "3316   RUNX1                 d171n      4   \n",
       "3317   RUNX1                 a122*      1   \n",
       "3318   RUNX1               fusions      1   \n",
       "3319   RUNX1                  r80c      4   \n",
       "3320   RUNX1                  k83e      4   \n",
       "\n",
       "                                                 Text_x  \\\n",
       "ID                                                        \n",
       "0     cyclin-dependent kinases (cdks) regulate a var...   \n",
       "1      abstract background  non-small cell lung canc...   \n",
       "2      abstract background  non-small cell lung canc...   \n",
       "3     recent evidence has demonstrated that acquired...   \n",
       "4     oncogenic mutations in the monomeric casitas b...   \n",
       "...                                                 ...   \n",
       "3316  introduction  myelodysplastic syndromes (mds) ...   \n",
       "3317  introduction  myelodysplastic syndromes (mds) ...   \n",
       "3318  the runt-related transcription factor 1 gene (...   \n",
       "3319  the runx1/aml1 gene is the most frequent targe...   \n",
       "3320  the most frequent mutations associated with le...   \n",
       "\n",
       "                                                 Text_y  Score  \n",
       "ID                                                              \n",
       "0     here, we deorphanize cdk10 by identifying cycl...      3  \n",
       "1      the c-cbl loh also positively correlated with...      1  \n",
       "2      the c-cbl loh also positively correlated with...      1  \n",
       "3      most of the changes were novel, although 4 ca...      1  \n",
       "4      all mutations from the second group were pred...      1  \n",
       "...                                                 ...    ...  \n",
       "3316  21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutatio...      1  \n",
       "3317   (a) spleen from morbid mice/d171n (left) and ...      2  \n",
       "3318   lpxn is preferentially expressed in hematopoi...      1  \n",
       "3319  ) conversely, mutations mapping to the first d...      1  \n",
       "3320   mutation analysis in fpd/aml pedigrees. elect...      1  \n",
       "\n",
       "[3316 rows x 6 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text_x</th>\n",
       "      <th>Text_y</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>cyclin-dependent kinases (cdks) regulate a var...</td>\n",
       "      <td>here, we deorphanize cdk10 by identifying cycl...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>w802*</td>\n",
       "      <td>2</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>q249e</td>\n",
       "      <td>2</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>n454d</td>\n",
       "      <td>3</td>\n",
       "      <td>recent evidence has demonstrated that acquired...</td>\n",
       "      <td>most of the changes were novel, although 4 ca...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>l399v</td>\n",
       "      <td>4</td>\n",
       "      <td>oncogenic mutations in the monomeric casitas b...</td>\n",
       "      <td>all mutations from the second group were pred...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>d171n</td>\n",
       "      <td>4</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "      <td>21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutatio...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>a122*</td>\n",
       "      <td>1</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "      <td>(a) spleen from morbid mice/d171n (left) and ...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>fusions</td>\n",
       "      <td>1</td>\n",
       "      <td>the runt-related transcription factor 1 gene (...</td>\n",
       "      <td>lpxn is preferentially expressed in hematopoi...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>r80c</td>\n",
       "      <td>4</td>\n",
       "      <td>the runx1/aml1 gene is the most frequent targe...</td>\n",
       "      <td>) conversely, mutations mapping to the first d...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>k83e</td>\n",
       "      <td>4</td>\n",
       "      <td>the most frequent mutations associated with le...</td>\n",
       "      <td>mutation analysis in fpd/aml pedigrees. elect...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3284 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene             Variation  Class  \\\n",
       "ID                                          \n",
       "0     FAM58A  truncating mutations      1   \n",
       "1        CBL                 w802*      2   \n",
       "2        CBL                 q249e      2   \n",
       "3        CBL                 n454d      3   \n",
       "4        CBL                 l399v      4   \n",
       "...      ...                   ...    ...   \n",
       "3316   RUNX1                 d171n      4   \n",
       "3317   RUNX1                 a122*      1   \n",
       "3318   RUNX1               fusions      1   \n",
       "3319   RUNX1                  r80c      4   \n",
       "3320   RUNX1                  k83e      4   \n",
       "\n",
       "                                                 Text_x  \\\n",
       "ID                                                        \n",
       "0     cyclin-dependent kinases (cdks) regulate a var...   \n",
       "1      abstract background  non-small cell lung canc...   \n",
       "2      abstract background  non-small cell lung canc...   \n",
       "3     recent evidence has demonstrated that acquired...   \n",
       "4     oncogenic mutations in the monomeric casitas b...   \n",
       "...                                                 ...   \n",
       "3316  introduction  myelodysplastic syndromes (mds) ...   \n",
       "3317  introduction  myelodysplastic syndromes (mds) ...   \n",
       "3318  the runt-related transcription factor 1 gene (...   \n",
       "3319  the runx1/aml1 gene is the most frequent targe...   \n",
       "3320  the most frequent mutations associated with le...   \n",
       "\n",
       "                                                 Text_y  Score  \n",
       "ID                                                              \n",
       "0     here, we deorphanize cdk10 by identifying cycl...      3  \n",
       "1      the c-cbl loh also positively correlated with...      1  \n",
       "2      the c-cbl loh also positively correlated with...      1  \n",
       "3      most of the changes were novel, although 4 ca...      1  \n",
       "4      all mutations from the second group were pred...      1  \n",
       "...                                                 ...    ...  \n",
       "3316  21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutatio...      1  \n",
       "3317   (a) spleen from morbid mice/d171n (left) and ...      2  \n",
       "3318   lpxn is preferentially expressed in hematopoi...      1  \n",
       "3319  ) conversely, mutations mapping to the first d...      1  \n",
       "3320   mutation analysis in fpd/aml pedigrees. elect...      1  \n",
       "\n",
       "[3284 rows x 6 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data_best_score =  new_data[new_data[\"Score\"] != 0]\n",
    "new_data_best_score"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Ceux qui ont un score de 0 --> PB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bad = new_data[new_data[\"Score\"] == 0]\n",
    "bad[\"Gene\"].tolist()\n",
    "bad_class4 = bad[\"Class\"] == 4\n",
    "class4 = concatenate_data.loc[bad_class4.index,]\n",
    "class4_tofix = class4[class4[\"Class\"] == 4]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check si la 1ere phrase y est bien dans l'article x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "count_new = new_data_best_score.apply(lambda line: line[\"Text_y\"].split(\".\")[0] in line[\"Text_x\"].lower(), axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Si 1ere phrase y == 1ere phrase x, apparemment ça fait pas le comptage mais c'est good !"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_data.loc[count_new[count_new == False].index,]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check si la variation est bien dans les phrases car c'est du score 1 --> perfect match\n",
    "Ceux qui y sont pas, c'est les mutations codons stop avec `*` qui a une signification dans les expression réguliere, donc à retravailler !"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text_x</th>\n",
       "      <th>Text_y</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>cyclin-dependent kinases (cdks) regulate a var...</td>\n",
       "      <td>here, we deorphanize cdk10 by identifying cycl...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>CBL</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>to determine if residual cylindrical refractiv...</td>\n",
       "      <td>full size image (58 kb) previous figures/tab...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>CBL</td>\n",
       "      <td>y371s</td>\n",
       "      <td>4</td>\n",
       "      <td>acquired uniparental disomy (aupd) is a common...</td>\n",
       "      <td>transformed nih3t3 cells showed pi3 kinase-de...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>PTPRT</td>\n",
       "      <td>t1365m</td>\n",
       "      <td>4</td>\n",
       "      <td>tyrosine phosphorylation, regulated by protein...</td>\n",
       "      <td>mutations in both intracellular ptp domains (...</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>PTPRT</td>\n",
       "      <td>t844m</td>\n",
       "      <td>4</td>\n",
       "      <td>introduction  preceding hematologic disorders ...</td>\n",
       "      <td>introduction  preceding hematologic disorders ...</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3307</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>the runx genes have come to prominence recentl...</td>\n",
       "      <td>the most divergent feature is the extended gl...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3308</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>r174*</td>\n",
       "      <td>7</td>\n",
       "      <td>familial platelet disorder with propensity to...</td>\n",
       "      <td>familial platelet disorder with propensity to...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3312</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>tel-runx1 fusion</td>\n",
       "      <td>4</td>\n",
       "      <td>balanced chromosomal translocations are freque...</td>\n",
       "      <td>balanced chromosomal translocations are freque...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3314</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>g42r</td>\n",
       "      <td>6</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "      <td>, patient number; sct, stem cell transplantati...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>a122*</td>\n",
       "      <td>1</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "      <td>(a) spleen from morbid mice/d171n (left) and ...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>860 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene             Variation  Class  \\\n",
       "ID                                          \n",
       "0     FAM58A  truncating mutations      1   \n",
       "16       CBL  truncating mutations      1   \n",
       "19       CBL                 y371s      4   \n",
       "44     PTPRT                t1365m      4   \n",
       "46     PTPRT                 t844m      4   \n",
       "...      ...                   ...    ...   \n",
       "3307   RUNX1  truncating mutations      1   \n",
       "3308   RUNX1                 r174*      7   \n",
       "3312   RUNX1      tel-runx1 fusion      4   \n",
       "3314   RUNX1                  g42r      6   \n",
       "3317   RUNX1                 a122*      1   \n",
       "\n",
       "                                                 Text_x  \\\n",
       "ID                                                        \n",
       "0     cyclin-dependent kinases (cdks) regulate a var...   \n",
       "16    to determine if residual cylindrical refractiv...   \n",
       "19    acquired uniparental disomy (aupd) is a common...   \n",
       "44    tyrosine phosphorylation, regulated by protein...   \n",
       "46    introduction  preceding hematologic disorders ...   \n",
       "...                                                 ...   \n",
       "3307  the runx genes have come to prominence recentl...   \n",
       "3308   familial platelet disorder with propensity to...   \n",
       "3312  balanced chromosomal translocations are freque...   \n",
       "3314  introduction  myelodysplastic syndromes (mds) ...   \n",
       "3317  introduction  myelodysplastic syndromes (mds) ...   \n",
       "\n",
       "                                                 Text_y  Score  \n",
       "ID                                                              \n",
       "0     here, we deorphanize cdk10 by identifying cycl...      3  \n",
       "16      full size image (58 kb) previous figures/tab...      3  \n",
       "19     transformed nih3t3 cells showed pi3 kinase-de...      1  \n",
       "44     mutations in both intracellular ptp domains (...      5  \n",
       "46    introduction  preceding hematologic disorders ...      7  \n",
       "...                                                 ...    ...  \n",
       "3307   the most divergent feature is the extended gl...      3  \n",
       "3308   familial platelet disorder with propensity to...      2  \n",
       "3312  balanced chromosomal translocations are freque...      4  \n",
       "3314  , patient number; sct, stem cell transplantati...      1  \n",
       "3317   (a) spleen from morbid mice/d171n (left) and ...      2  \n",
       "\n",
       "[860 rows x 6 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data_best_score[new_data_best_score.apply(lambda line: line[\"Variation\"] in line[\"Text_y\"], axis = 1) == False]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_data_best_score = new_data_best_score.drop([1074,1897])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count      3284\n",
       "unique      257\n",
       "top       BRCA1\n",
       "freq        264\n",
       "Name: Gene, dtype: object"
      ]
     },
     "execution_count": 35,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data_best_score[\"Gene\"].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_data_best_score[new_data_best_score[\"Gene\"] == \"KMT2D\"][\"Text_y\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_data_best_score[new_data_best_score[\"Gene\"] == \"BRCA1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_data_best_score"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Fix les score == 0 164\n",
    "145"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Text</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>here, we deorphanize cdk10 by identifying cycl...</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>most of the changes were novel, although 4 ca...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>all mutations from the second group were pred...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutatio...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>(a) spleen from morbid mice/d171n (left) and ...</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>lpxn is preferentially expressed in hematopoi...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>) conversely, mutations mapping to the first d...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>mutation analysis in fpd/aml pedigrees. elect...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3316 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Text  Score\n",
       "ID                                                            \n",
       "0     here, we deorphanize cdk10 by identifying cycl...      3\n",
       "1      the c-cbl loh also positively correlated with...      1\n",
       "2      the c-cbl loh also positively correlated with...      1\n",
       "3      most of the changes were novel, although 4 ca...      1\n",
       "4      all mutations from the second group were pred...      1\n",
       "...                                                 ...    ...\n",
       "3316  21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutatio...      1\n",
       "3317   (a) spleen from morbid mice/d171n (left) and ...      2\n",
       "3318   lpxn is preferentially expressed in hematopoi...      1\n",
       "3319  ) conversely, mutations mapping to the first d...      1\n",
       "3320   mutation analysis in fpd/aml pedigrees. elect...      1\n",
       "\n",
       "[3316 rows x 2 columns]"
      ]
     },
     "execution_count": 127,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clean_match"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "index_zero = clean_match[clean_match[\"Score\"] == 0].index"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>PTPRT</td>\n",
       "      <td>r1209w</td>\n",
       "      <td>4</td>\n",
       "      <td>tyrosine phosphorylation, regulated by protein...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>RYBP</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>the mouse double minute 2 (mdm2)–p53 interacti...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>MSH6</td>\n",
       "      <td>r1076c</td>\n",
       "      <td>1</td>\n",
       "      <td>in the united states, approximately % of all i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>178</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>p691s</td>\n",
       "      <td>2</td>\n",
       "      <td>in selected patients with advanced non-small c...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>249</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>g863s</td>\n",
       "      <td>2</td>\n",
       "      <td>‘classical' mutations in the egfr tyrosine kin...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>267</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>g810s</td>\n",
       "      <td>7</td>\n",
       "      <td>purpose: epidermal growth factor receptor (egf...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>350</th>\n",
       "      <td>CDH1</td>\n",
       "      <td>epigenetic silencing</td>\n",
       "      <td>1</td>\n",
       "      <td>in diffuse gastric carcinoma, despite common e...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>354</th>\n",
       "      <td>EP300</td>\n",
       "      <td>r1627</td>\n",
       "      <td>4</td>\n",
       "      <td>the transcriptional coactivator p300/cbp (creb...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>355</th>\n",
       "      <td>EP300</td>\n",
       "      <td>c1385</td>\n",
       "      <td>4</td>\n",
       "      <td>the transcriptional coactivator p300/cbp (creb...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>480</th>\n",
       "      <td>TP53</td>\n",
       "      <td>dna binding domain missense mutations</td>\n",
       "      <td>4</td>\n",
       "      <td>mutations in the p53 tumor suppressor are the ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>807</th>\n",
       "      <td>ERCC2</td>\n",
       "      <td>amplification</td>\n",
       "      <td>2</td>\n",
       "      <td>introduction  platinum-based chemotherapy has ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>968</th>\n",
       "      <td>ESR1</td>\n",
       "      <td>a546d</td>\n",
       "      <td>7</td>\n",
       "      <td>recent studies have identified somatic esr1 mu...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1060</th>\n",
       "      <td>EWSR1</td>\n",
       "      <td>ewsr1-etv1 fusion</td>\n",
       "      <td>7</td>\n",
       "      <td>fibroblast growth factors (fgfs) expressed in...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1072</th>\n",
       "      <td>FOXA1</td>\n",
       "      <td>l388m</td>\n",
       "      <td>1</td>\n",
       "      <td>characterization of the prostate cancer transc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1080</th>\n",
       "      <td>ARID1A</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>arid1a (baf250a) promotes the formation of swi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1091</th>\n",
       "      <td>RXRA</td>\n",
       "      <td>s247f</td>\n",
       "      <td>6</td>\n",
       "      <td>urothelial carcinoma of the bladder is a commo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1092</th>\n",
       "      <td>RXRA</td>\n",
       "      <td>s247y</td>\n",
       "      <td>6</td>\n",
       "      <td>urothelial carcinoma of the bladder is a commo...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1141</th>\n",
       "      <td>MET</td>\n",
       "      <td>h1094r</td>\n",
       "      <td>7</td>\n",
       "      <td>hereditary papifiary renal carcinoma (hprc) is...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1228</th>\n",
       "      <td>PIK3CB</td>\n",
       "      <td>e1051k</td>\n",
       "      <td>2</td>\n",
       "      <td>deregulation of the pi3k signaling pathway is ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1305</th>\n",
       "      <td>MLH1</td>\n",
       "      <td>p654l</td>\n",
       "      <td>6</td>\n",
       "      <td>mismatch-repair factors have a prominent role ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1307</th>\n",
       "      <td>MLH1</td>\n",
       "      <td>r659p</td>\n",
       "      <td>6</td>\n",
       "      <td>mismatch-repair factors have a prominent role ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1332</th>\n",
       "      <td>MLH1</td>\n",
       "      <td>r659l</td>\n",
       "      <td>1</td>\n",
       "      <td>mismatch-repair factors have a prominent role ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1826</th>\n",
       "      <td>NCOR1</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>estrogen receptors (ers) are normally expresse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1902</th>\n",
       "      <td>TET1</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>tet2 is a close relative of tet1, an enzyme th...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2024</th>\n",
       "      <td>MAP2K1</td>\n",
       "      <td>e144k</td>\n",
       "      <td>7</td>\n",
       "      <td>histiocytic neoplasms are clonal, hematopoieti...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2086</th>\n",
       "      <td>AGO2</td>\n",
       "      <td>r647a</td>\n",
       "      <td>1</td>\n",
       "      <td>argonaute (ago) proteins mediate silencing of ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2087</th>\n",
       "      <td>AGO2</td>\n",
       "      <td>r583a</td>\n",
       "      <td>1</td>\n",
       "      <td>argonaute (ago) proteins mediate silencing of ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2285</th>\n",
       "      <td>INPP4B</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>although the gastrointestinal (gi) tract posse...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2302</th>\n",
       "      <td>JAK1</td>\n",
       "      <td>r724h</td>\n",
       "      <td>7</td>\n",
       "      <td>regulatory t (t reg) cells have a major role i...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3135</th>\n",
       "      <td>KRAS</td>\n",
       "      <td>wildtype</td>\n",
       "      <td>3</td>\n",
       "      <td>to characterize somatic alterations in colorec...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3254</th>\n",
       "      <td>CASP8</td>\n",
       "      <td>casp8l</td>\n",
       "      <td>4</td>\n",
       "      <td>caspase-8 (fas-associating protein with death ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3256</th>\n",
       "      <td>CASP8</td>\n",
       "      <td>promoter hypermethylation</td>\n",
       "      <td>4</td>\n",
       "      <td>neuroblastoma, the most common paediatric soli...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene                              Variation  Class  \\\n",
       "ID                                                           \n",
       "61     PTPRT                                 r1209w      4   \n",
       "88      RYBP                   truncating mutations      1   \n",
       "108     MSH6                                 r1076c      1   \n",
       "178     EGFR                                  p691s      2   \n",
       "249     EGFR                                  g863s      2   \n",
       "267     EGFR                                  g810s      7   \n",
       "350     CDH1                   epigenetic silencing      1   \n",
       "354    EP300                                  r1627      4   \n",
       "355    EP300                                  c1385      4   \n",
       "480     TP53  dna binding domain missense mutations      4   \n",
       "807    ERCC2                          amplification      2   \n",
       "968     ESR1                                  a546d      7   \n",
       "1060   EWSR1                      ewsr1-etv1 fusion      7   \n",
       "1072   FOXA1                                  l388m      1   \n",
       "1080  ARID1A                   truncating mutations      1   \n",
       "1091    RXRA                                  s247f      6   \n",
       "1092    RXRA                                  s247y      6   \n",
       "1141     MET                                 h1094r      7   \n",
       "1228  PIK3CB                                 e1051k      2   \n",
       "1305    MLH1                                  p654l      6   \n",
       "1307    MLH1                                  r659p      6   \n",
       "1332    MLH1                                  r659l      1   \n",
       "1826   NCOR1                   truncating mutations      1   \n",
       "1902    TET1                   truncating mutations      1   \n",
       "2024  MAP2K1                                  e144k      7   \n",
       "2086    AGO2                                  r647a      1   \n",
       "2087    AGO2                                  r583a      1   \n",
       "2285  INPP4B                   truncating mutations      1   \n",
       "2302    JAK1                                  r724h      7   \n",
       "3135    KRAS                               wildtype      3   \n",
       "3254   CASP8                                 casp8l      4   \n",
       "3256   CASP8              promoter hypermethylation      4   \n",
       "\n",
       "                                                   Text  \n",
       "ID                                                       \n",
       "61    tyrosine phosphorylation, regulated by protein...  \n",
       "88    the mouse double minute 2 (mdm2)–p53 interacti...  \n",
       "108   in the united states, approximately % of all i...  \n",
       "178   in selected patients with advanced non-small c...  \n",
       "249   ‘classical' mutations in the egfr tyrosine kin...  \n",
       "267   purpose: epidermal growth factor receptor (egf...  \n",
       "350   in diffuse gastric carcinoma, despite common e...  \n",
       "354   the transcriptional coactivator p300/cbp (creb...  \n",
       "355   the transcriptional coactivator p300/cbp (creb...  \n",
       "480   mutations in the p53 tumor suppressor are the ...  \n",
       "807   introduction  platinum-based chemotherapy has ...  \n",
       "968   recent studies have identified somatic esr1 mu...  \n",
       "1060   fibroblast growth factors (fgfs) expressed in...  \n",
       "1072  characterization of the prostate cancer transc...  \n",
       "1080  arid1a (baf250a) promotes the formation of swi...  \n",
       "1091  urothelial carcinoma of the bladder is a commo...  \n",
       "1092  urothelial carcinoma of the bladder is a commo...  \n",
       "1141  hereditary papifiary renal carcinoma (hprc) is...  \n",
       "1228  deregulation of the pi3k signaling pathway is ...  \n",
       "1305  mismatch-repair factors have a prominent role ...  \n",
       "1307  mismatch-repair factors have a prominent role ...  \n",
       "1332  mismatch-repair factors have a prominent role ...  \n",
       "1826  estrogen receptors (ers) are normally expresse...  \n",
       "1902  tet2 is a close relative of tet1, an enzyme th...  \n",
       "2024  histiocytic neoplasms are clonal, hematopoieti...  \n",
       "2086  argonaute (ago) proteins mediate silencing of ...  \n",
       "2087  argonaute (ago) proteins mediate silencing of ...  \n",
       "2285  although the gastrointestinal (gi) tract posse...  \n",
       "2302  regulatory t (t reg) cells have a major role i...  \n",
       "3135  to characterize somatic alterations in colorec...  \n",
       "3254  caspase-8 (fas-associating protein with death ...  \n",
       "3256  neuroblastoma, the most common paediatric soli...  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "to_fix = concatenate_data.loc[index_zero,:]\n",
    "to_fix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "SCOOOOORE 0\n",
      "t844m 46 9068 0\n",
      "SCOOOOORE 0\n",
      "r1209w 61 12951 0\n",
      "SCOOOOORE g\n",
      "SCOOOOORE 0\n",
      "l234fs 72 76499 0\n",
      "SCOOOOORE 0\n",
      "l1273f 73 26872 0\n",
      "SCOOOOORE g\n",
      "SCOOOOORE 0\n",
      "i290a 80 42197 0\n",
      "SCOOOOORE 0\n",
      "truncating mutations 88 15584 0\n",
      "SCOOOOORE 0\n",
      "r1076c 108 51812 0\n",
      "SCOOOOORE 0\n",
      "v509a 111 40001 0\n",
      "SCOOOOORE 5\n",
      "SCOOOOORE 5\n",
      "SCOOOOORE 0\n",
      "e758g 155 23513 0\n",
      "SCOOOOORE 0\n",
      "k806a 161 23513 0\n",
      "SCOOOOORE 0\n",
      "p691s 178 57832 0\n",
      "SCOOOOORE 0\n",
      "d837n 222 22044 0\n",
      "SCOOOOORE 0\n",
      "g863s 249 23329 0\n",
      "SCOOOOORE 5\n",
      "SCOOOOORE 0\n",
      "g810s 267 28321 0\n",
      "SCOOOOORE 0\n",
      "a113_splice 285 35251 0\n",
      "SCOOOOORE 5\n",
      "SCOOOOORE 5\n",
      "SCOOOOORE 5\n",
      "SCOOOOORE 0\n",
      "epigenetic silencing 350 25569 0\n",
      "SCOOOOORE 0\n",
      "r1627 354 21957 0\n",
      "SCOOOOORE 0\n",
      "c1385 355 21957 0\n",
      "SCOOOOORE 0\n",
      "dna binding domain missense mutations 480 32240 0\n",
      "SCOOOOORE 0\n",
      "r378a 581 9691 0\n",
      "SCOOOOORE 0\n",
      "d493a 593 9691 0\n",
      "SCOOOOORE 0\n",
      "y353l 604 47177 0\n",
      "SCOOOOORE 0\n",
      "s562l 616 25082 0\n",
      "SCOOOOORE 5\n",
      "SCOOOOORE 5\n",
      "SCOOOOORE 5\n",
      "SCOOOOORE 0\n",
      "v839g 710 46529 0\n",
      "SCOOOOORE 2\n",
      "SCOOOOORE 2\n",
      "SCOOOOORE 0\n",
      "g665a 803 38458 0\n",
      "SCOOOOORE 0\n",
      "a717g 805 29780 0\n",
      "SCOOOOORE 0\n",
      "n238s 806 38458 0\n",
      "SCOOOOORE 0\n",
      "amplification 807 38458 0\n",
      "SCOOOOORE 0\n",
      "s746fs 808 29780 0\n",
      "SCOOOOORE 0\n",
      "p463l 814 38458 0\n",
      "SCOOOOORE 0\n",
      "f568fs 819 29780 0\n",
      "SCOOOOORE 0\n",
      "e875g 833 20126 0\n",
      "SCOOOOORE 5\n",
      "SCOOOOORE 0\n",
      "p130s 875 36845 0\n",
      "SCOOOOORE 0\n",
      "s584l 885 26982 0\n",
      "SCOOOOORE 0\n",
      "g829r 891 36845 0\n",
      "SCOOOOORE 0\n",
      "d1071n 906 36845 0\n",
      "SCOOOOORE 0\n",
      "a546d 968 36167 0\n",
      "SCOOOOORE 5\n",
      "SCOOOOORE 5\n",
      "SCOOOOORE 0\n",
      "ewsr1-etv1 fusion 1060 39717 0\n",
      "SCOOOOORE 0\n",
      "l388m 1072 13643 0\n",
      "SCOOOOORE 0\n",
      "truncating mutations 1080 77954 0\n",
      "SCOOOOORE 0\n",
      "s247f 1091 25303 0\n",
      "SCOOOOORE 0\n",
      "s247y 1092 25303 0\n",
      "SCOOOOORE 0\n",
      "x1007_splice 1118 100366 0\n",
      "SCOOOOORE 0\n",
      "v1188l 1121 54324 0\n",
      "SCOOOOORE 0\n",
      "l1195v 1128 54324 0\n",
      "SCOOOOORE 0\n",
      "x963_splice 1129 100366 0\n",
      "SCOOOOORE 0\n",
      "h1094y 1131 54324 0\n",
      "SCOOOOORE 0\n",
      "x1009_splice 1132 100366 0\n",
      "SCOOOOORE 0\n",
      "h1094l 1136 63850 0\n",
      "SCOOOOORE 0\n",
      "981_1028splice 1138 100366 0\n",
      "SCOOOOORE 0\n",
      "d1010n 1139 5670 0\n",
      "SCOOOOORE 0\n",
      "x1008_splice 1140 100366 0\n",
      "SCOOOOORE 0\n",
      "h1094r 1141 17042 0\n",
      "SCOOOOORE 0\n",
      "h1106d 1144 31046 0\n",
      "SCOOOOORE 0\n",
      "v1070e 1147 31501 0\n",
      "SCOOOOORE 0\n",
      "963_d1010splice 1148 100366 0\n",
      "SCOOOOORE 0\n",
      "n1100y 1149 29531 0\n",
      "SCOOOOORE 0\n",
      "e1051k 1228 65320 0\n",
      "SCOOOOORE 0\n",
      "x475_splice 1247 57420 0\n",
      "SCOOOOORE 0\n",
      "x582_splice 1257 106218 0\n",
      "SCOOOOORE 0\n",
      "x434_splice 1261 57420 0\n",
      "SCOOOOORE 0\n",
      "k376n 1270 38803 0\n",
      "SCOOOOORE 0\n",
      "s273c 1276 38803 0\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-170-a11f1689bb15>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mto_fix\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0mcheck_line\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36mapply\u001b[0;34m(self, func, axis, raw, result_type, args, **kwargs)\u001b[0m\n\u001b[1;32m   8739\u001b[0m             \u001b[0mkwargs\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   8740\u001b[0m         )\n\u001b[0;32m-> 8741\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   8742\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   8743\u001b[0m     def applymap(\n",
      "\u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/apply.py\u001b[0m in \u001b[0;36mapply\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    686\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_raw\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    687\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 688\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_standard\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    689\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    690\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0magg\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/apply.py\u001b[0m in \u001b[0;36mapply_standard\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    810\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    811\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mapply_standard\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 812\u001b[0;31m         \u001b[0mresults\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mres_index\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_series_generator\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    813\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    814\u001b[0m         \u001b[0;31m# wrap results\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.8/site-packages/pandas/core/apply.py\u001b[0m in \u001b[0;36mapply_series_generator\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    826\u001b[0m             \u001b[0;32mfor\u001b[0m \u001b[0mi\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mv\u001b[0m \u001b[0;32min\u001b[0m \u001b[0menumerate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mseries_gen\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    827\u001b[0m                 \u001b[0;31m# ignore SettingWithCopy here in case the user mutates\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 828\u001b[0;31m                 \u001b[0mresults\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mv\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    829\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0misinstance\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresults\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mi\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mABCSeries\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    830\u001b[0m                     \u001b[0;31m# If we have a view on v, we need to make a copy because\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-170-a11f1689bb15>\u001b[0m in \u001b[0;36m<lambda>\u001b[0;34m(x)\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mto_fix\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0mcheck_line\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-169-e42ce4669824>\u001b[0m in \u001b[0;36mcheck_line\u001b[0;34m(x)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mcheck_line\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m     \u001b[0mextract\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mextract_match\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m     \u001b[0mscore\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mextract\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"SCOOOOORE\"\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mscore\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-141-34071c3e04be>\u001b[0m in \u001b[0;36mextract_match\u001b[0;34m(line)\u001b[0m\n\u001b[1;32m     77\u001b[0m     \u001b[0;31m# Try first match with inital variation value\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     78\u001b[0m     \u001b[0;31m# Quality score = 1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 79\u001b[0;31m     \u001b[0minitial_match\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfind_match\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtext\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mvariation\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     80\u001b[0m     \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0minitial_match\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m!=\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     81\u001b[0m         \u001b[0;31m#print(\"First match ! \", variation)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-141-34071c3e04be>\u001b[0m in \u001b[0;36mfind_match\u001b[0;34m(text, word)\u001b[0m\n\u001b[1;32m     29\u001b[0m     \u001b[0mbefore_after_target\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"([^.]*\\.){0,1}\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     30\u001b[0m     \u001b[0mmatch_exp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mre\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcompile\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbefore_after_target\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mtarget_sentence\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mbefore_after_target\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 31\u001b[0;31m     \u001b[0mmatch_text\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmatch_exp\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfindall\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mclean\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     32\u001b[0m     \u001b[0mfinal_match\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m\"\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmap\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mjoin_tuple_string\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmatch_text\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     33\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mfinal_match\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "to_fix.apply(lambda x : check_line(x) , axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "identification of a high-risk disease-causing constitutional mutation in a cancer patient guides the clinical management of their whole family, with implications for counseling, cancer treatment options and presymptomatic surveillance, and considerations of risk-reducing surgery and/or medication regimens1. carriers of mutations in the mismatch repair (mmr) genes mlh1, msh2, msh6 and pms2 causing lynch syndrome1 have a substantially increased risk of colorectal and endometrial cancers, along with increased risk of ovarian, gastric, small bowel, urothelial, brain, hepatobiliary, pancreatic, bladder, kidney, prostate and breast cancers1, 2, 3, 4, 5, 6, 7, 8. however, intensive management reduces mortality9.  sequence variants of uncertain functional and clinical relevance are common in genetic test reports. although several lines of evidence can be evaluated to assess the clinical implications of these variants, usually none of these approaches can be used on its own to obtain clinically useful interpretations and, for many variants, comprehensive data are lacking. laboratories are generally conservative in designating pathogenic variants, defining variants as being of 'uncertain significance' unless overwhelming evidence of pathogenicity exists. several schemes for the classification of variants in genes associated with mendelian conditions have been proposed for use in the clinical setting. because clinically useful actions are currently only considered for high-penetrance mutations, all of these systems are aimed at differentiating high-penetrance from low-penetrance and neutral variants and do not consider variants of intermediate risk. these schemes differ in the range and format of data used for classification and in the number of variant classes10, 11, 12. the international agency for research on cancer (iarc) classification system, endorsed by the human variome project (hvp), facilitates standardized categorization by defining classes that can be linked to validated quantitative measures of causality and/or pathogenicity from statistical models13, 14, 15, 16 or to validated interpretation of qualitative data17. importantly, only the five-class iarc system has been linked to clinical recommendations for all classes, including clinical testing and full high-risk surveillance guidelines for class 5 (pathogenic) and class 4 (likely pathogenic) variants; advice to treat as if “no mutation associated with disease has been detected” for class 1 (not pathogenic) and class 2 (likely not pathogenic) variants; and acquisition of additional data to provide more robust classification for class 2 (likely not pathogenic), class 3 (uncertain) and class 4 (likely pathogenic) variants.  locus-specific databases (lsdbs) are an important source of information for clinicians and researchers in assessing data and forming opinions on the clinical relevance of disease-associated gene sequence variants, and these databases have a fundamental role in variant classification owing to their added value from having aggregated data. consistent and normalized data curation is critical to the value derived from databases in categorizing the relationship between genetic variation and disease—especially for clinical applications. it has previously been recommended by the iarc working group that a panel covering a range of expertise in variant classification provide consensus opinion on variant pathogenicity before publicly accessible display of such information18. another important component of the classifications provided by lsdbs is transparency regarding the criteria and supporting information used for classification, which allows lsdb users to consider the information for their own applications in research or clinical settings18.   insight has merged multiple gene mutation and variant repositories to create the insight colon cancer gene variant database for mmr and other colon cancer susceptibility genes19, 20, 21, 22, 23, hosted by the leiden open variation database (lovd). following recommendations for lsdb curation18, insight formed an international panel of researchers and clinicians to review mmr gene variants submitted to the database. to encourage the submission of unpublished clinical and research data to further facilitate variant classification, the microattribution approach24 was implemented using open researcher and contributor identification (orcid). here we present the results of the insight variant interpretation committee (vic) effort to develop, test and apply a 5-tiered scheme to classify 2,360 unique constitutional mmr gene variants.  results abstract• introduction• results• discussion• methods• accession codes• references• acknowledgments• author information• supplementary information curation of mmr gene variants as of the end of december 2012, after 3,458 alterations to standardize nomenclature had been made, there were 12,635 submissions of 2,730 unique mmr gene variants lodged in the insight database. furthermore, 370 unique variants (%) were not identified in constitutional (germline) dna (see supplementary  1 and supplementary table 1 for details) and were excluded from further analyses because (i) no evidence existed that these occurred as constitutional variants and (ii) no clinical information was available to assess their potential roles in hereditary disease. the 2,360 constitutional variants included 932 mlh1 (39%), 842 msh2 (36%), 449 msh6 (19%) and 137 pms2 (6%) variants. most variants (800; 34%) were nonsense or frameshift changes predicted to cause protein truncation, with the next largest group (746; 32%) consisting of nonsynonymous variants that were not obviously truncating, including missense substitutions, small in-frame insertion-deletion mutations (indels) and read-through alterations of the translation termination codon.  variants had originally been assigned a classification by submitters according to the following classes: pathogenic, probably pathogenic, no known pathogenicity, probably no pathogenicity or effect unknown. no information was recorded to document the rationale for classification or the standards used to classify variants. when considering the 1,382 constitutional variants with multiple entries in the insight database, discordance in classification between submitters was found for 869 variants. some of this discordance arose because of classification based on single data points or references, such as results from a single functional assay22 or inferences from individual publications originally lodged in the mismatch repair genes variant database23 (see the example in supplementary table 2).  development of a five-tiered system for classification the insight vic (online methods) was established in 2007 to address discrepancies in the classification of mmr gene variants lodged in the insight database. since march 2011, the vic has made a concerted effort to develop standardized criteria for variant classification, employing a modified delphi consensus process25 to evaluate current scientific evidence and reach consensus. in line with hvp26, the iarc classification system10 for variant categorization (table 1) was adopted by insight for the classification of mmr gene variants. briefly, multiple lines of evidence were required for classification, and evidence for each variant had to include data associating the variant with both clinical and functional consequences (online methods).  table 1: insight variant classification scheme with accompanying recommendations for family management, adapted from the iarc five-tiered classification system full table figures/tables index the scheme was first tested on a subset of 117 mmr gene variants, and the criteria evolved and were refined by consensus to accommodate new data and inconsistencies over multiple classification teleconferences and face-to-face meetings. final criteria were then applied retrospectively and to all remaining unique variants listed in the database (supplementary table 3). an overview of the insight classification criteria is shown in figure 1 (see supplementary table 4 and the supplementary note for detailed criteria and justifications). at the close of each vic teleconference or meeting, consensus classifications were noted. where necessary, action items to improve or clarify classification included (i) calls for missing clinical and functional information for specific variants to committee members and the general insight membership; (ii) requests for more detailed data or data clarification from the authors of original publications; and (iii) reassessment of classification after additional data were obtained. at the end of the process, the insight database was updated with the final consensus classifications and the supporting data to ensure transparency.  figure 1: overview of the five-tiered insight classification guidelines. overview of the five-tiered insight classification guidelines. (a) simplified guidelines describing the levels and types of evidence required for each tier of the classification. full guidelines are described in the supplementary note, and the detailed rationale behind each criterion is provided in supplementary table 3. the lynch syndrome molecular phenotype described in classes 4 and 5 includes information on msi and/or loss of expression of relevant protein(s), as determined by immunohistochemistry. in this study, variants resulting in the introduction of a premature termination codon or large genomic deletions affecting functionally important domains, generally considered pathogenic on the basis of dna sequence alone, are referred to as class 5a (assumed pathogenic) variants. all other variants categorized in class 5 are termed class 5b variants. (b) flowchart used to assist in the interpretation of available data from functional assays. assays reviewed for classification are shown in supplementary table 4, and the values used to define abrogated or normal function are shown in supplementary table 5. cutoffs of <25% and >75% protein expression, as used in previous publications47, 48, are very conservative given that abrogated function has been reported to be associated with a decrease in mlh1 expression of ~50% or more49. for variants that had normal, inconclusive or intermediate mmr activity in two independent assays but were deficient in protein function in two independent assays, abrogated function was assigned. af, allele frequency; pp, posterior probability of pathogenicity derived by multifactorial likelihood analysis; cmmrd, constitutional mmr deficiency (mim 276300); lr, likelihood ratio; ls, lynch syndrome; mss, microsatellite stable; crc, colorectal cancer; ihc, immunohistochemistry; nmd, nonsense-mediated decay.  full size image (421 kb) figures/tables index next the major issues faced by the committee in the review process included redundancy across multiple sources (resolved through discussion with the authors of original publications), paucity of information, incomplete or inaccurate data and difficulties in the interpretation of the results of functional assays. to facilitate the interpretation of findings from functional assays, supporting information and flowcharts were developed ( 1b and supplementary tables 5 and 6), and multiple meetings were coordinated that were dedicated to the review of variants with apparently discordant results from functional assays (supplementary table 3).  validation of the insight qualitative classification criteria nonsense or frameshift alterations or large genomic deletions interrupting functionally important domains are generally considered pathogenic on the basis of dna sequence alone; these variants were designated here as class 5a (assumed pathogenic). there were 990 assumed pathogenic variants in the database, 640 of which were private mutations. to demonstrate the robustness of the qualitative classification criteria, 170 assumed pathogenic variants (68 mlh1, 75 msh2, 13 msh6 and 14 pms2) were reviewed as a validation set against the class 5 (pathogenic) qualitative criteria required for the assignment of variants to class 5b (online methods and supplementary table 7). designation of a variant as class 5b required evidence of abrogated protein function, at least two tumors with microsatellite instability (msi) or appropriate loss of mmr protein expression and a segregation likelihood ratio of >10:1 (incorporating gene-specific cumulative risks27) or variant cosegregation with disease reported in at least two amsterdam criteria–positive families. class 5b was attained by all 60 validation set variants that had sufficient clinical data to assess these required criteria. the other 110 validation set variants could not be assigned to class 5b, largely because family cosegregation and tumor data were scarce or unobtainable—presumably because these variants are accepted as disease causing and are routinely used for clinical presymptomatic testing in families (online methods). of these variants, 72 were assigned to class 4 owing to lack of only 1 point of evidence, and 38 variants were assigned to class 3 owing to insufficient data. however, only 2 of 13 msh6 and 2 of 14 pms2 variants fulfilled class 5b criteria, reflecting the lower penetrance and later age of onset associated with deleterious variants in pms2 (ref. 28) and msh6 (ref. 29). together, these results indicate that the criteria for classification using qualitative data were sufficiently stringent to ensure conservative classification.  classification of 2,360 constitutional mmr gene variants of the 12,006 eligible variant entries in the insight database, submitter and final classifications differed for 7,935 (66%), including changes from class 1 (not pathogenic) to class 5 (pathogenic) and vice versa ( 2a). the overall breakdown of final classifications is shown in figure 2b. in addition to the 990 assumed pathogenic truncating or large-deletion variants (class 5a), consistent medical management is now also possible for the remaining 1,370 not obviously truncating variants; these include 167 class 5 (pathogenic) variants (class 5b) (12%), 183 class 4 (likely pathogenic) variants (14%), 86 class 2 (likely not pathogenic) variants (6%) and 169 class 1 (not pathogenic) variants (12%).  figure 2: outcome of standardized five-tiered insight classification of constitutional mmr gene variants. outcome of standardized five-tiered insight classification of constitutional mmr gene variants. (a) the graph plots the proportion of five-tiered classifications for all documented constitutional variants in the database against the original lovd database classifications assigned by submitters. class 5a is a subset of class 5 containing assumed pathogenic nonsense mutations, small frameshift indels and large deletions. class 5b includes variants that are not obviously truncating but are considered to be pathogenic on the basis of combined evidence (supplementary note). results show that standardized classification led to altered classifications for a considerable proportion of variant entries, including the downgrading of variants submitted as pathogenic (24%) and the upgrading of variants with unknown pathogenicity to likely pathogenic (%) or pathogenic (48%). in addition, clinically important misclassifications were identified for unique variants initially submitted as not pathogenic (54 unique variants reclassified as class 5b variants and 25 reclassified as class 4 variants) and unique variants submitted as pathogenic (28 unique variants reclassified as class 1 variants, 16 reclassified as class 2 variants and 218 reclassified as class 3 variants). (b) pie chart showing the distribution of final insight vic consensus classifications.  full size image (214 kb) download excel source data (40 kb) previous figures/tables index next nonsynonymous variants made up the majority of class 3 variants (524/765; 68%) and newly assigned class 5b variants (91/167; 54%) ( 3 and supplementary  2; see supplementary table 8 for detailed information supporting classifications). substitutions at canonical dinucleotide splice sites fell predominantly in class 4 owing to lack of functional rna analyses; however, if experimentally tested, these variants would likely be moved to class 5b. intronic variants outside of conserved splice sites were the most prevalent variant type in class 1.  figure 3: classifications of all documented unique variants by variant type. classifications of all documented unique variants by variant type. the plot represents the proportion of the different variant types within the five classes. class 5a is a subset of class 5 containing assumed pathogenic mutations (nonsense mutations, small frameshift indels and large deletions). all other variants placed in class 5 are termed class 5b variants (supplementary note). the different variant types include the following: atg/utr, variants in the initiation codon and the 5′ or 3′ utr; i, intronic variants outside of the canonical splice-site dinucleotides; s, synonymous variants; ns, not obviously truncating nonsynonymous variants outside of the kozak consensus sequence, , missense, small in-frame indel and read-through translation termination codon alterations; ss, variants in the canonical splice-site dinucleotides; lgdup, large genomic duplications; lgdel, large genomic deletions or disrupting inversions; ptc, variants that introduce premature termination codons, , nonsense mutations and small frameshift indels. see supplementary figure 2 for further details of variant types by mmr gene.  full size image (144 kb) download excel source data (34 kb) previous figures/tables index next final categorization (online methods) of not obviously truncating variants as class 1, 2, 4 or 5 was achieved by applying qualitative criteria for 391 variants, quantitative multifactorial likelihood analysis methodology (based on bioinformatic prior probabilities and evidence from segregation and/or tumor data; see thompson 16) for 192 variants and either quantitative or qualitative criteria for 26 variants. where classifications derived using quantitative criteria differed from those generated with qualitative criteria, this difference reflected the amount of data available rather than deficiencies in the classification criteria, with no variants considered to fall into class 1 or 2 using one approach and class 4 or 5 using the other. six synonymous variants were categorized in class 5b owing to their effects on splicing. of the substitutions occurring in initiation codons (often assumed to be pathogenic30, 31, 32), only one of nine had sufficient evidence to determine pathogenicity.  implementing microattribution microattribution is a means to incentivize the placement of unpublished data in the public domain by assigning scholarly contribution to authors similar to the citation conventions afforded to journal articles33. retrospective and prospective microattribution was implemented to acknowledge and encourage the submission of unpublished data to the insight database, including the submission of additional detailed clinical information from authors of published reports. microattribution was assigned for initial variant submission, segregation and family history data, pathology diagnostics (msi analysis and immunohistochemistry) information, data from in vitro functional assays (mainly rna splicing) and the determination of variant frequencies in normal individuals. as of july 2013, a total of 6,015 microattributions were conferred, including 3,763 for variant submission, 2,111 for family and tumor pathology data, 97 for data from in vitro assays and 25 for frequency data. notably, 19% of the microattributions for clinical and functional data contributed additional information critical for the classification of variants in the class 5a (assumed pathogenic) validation subset. these data also highlighted the fact that clinical testing for these assumed pathogenic variants is mostly undertaken in the presymptomatic setting. the contribution of microattribution to the final classification of not obviously truncating variants is shown in figure 4. notably, classification was altered for 57 of 169 (34%) variants for which unpublished data were obtained. moreover, implementation of microattribution stimulated the submission of 128 new mmr gene variants yet to be classified.  figure 4: contribution of microattribution to the classification of variants that are not obviously truncating. contribution of microattribution to the classification of variants that are not obviously truncating. dark shading (yes) indicates the proportion of total variants for each class where the additional data obtained through microattribution contributed to final classification.  full size image (98 kb) download excel source data (32 kb) previous figures/tables index next class 3 variants of uncertain relevance missense variants in mmr genes are abundant among class 3 (uncertain) variants and present a considerable clinical problem. quantitative multifactorial likelihood analysis is an effective approach for the classification of missense variants, as validated bioinformatic predictions34 based on amino acid conservation and physicochemical properties can be used as a surrogate for the in vitro effects of variants on protein function. in silico analyses previously shown to be highly accurate (area under receiving operator characteristic (roc) curve of )34 were used to estimate the prior probability of pathogenicity for all 481 class 3 (uncertain) missense variants ( 5) to prioritize requests for data to facilitate future multifactorial analysis. the distribution of prior probabilities for mlh1 and msh2 class 3 variants was clearly bimodal, suggesting that ~50% of mlh1 and msh2 missense variants may be classified as pathogenic after further investigation. in total, 401 missense variants had extreme prior probabilities of ≤20% or ≥80%, with 270 having probabilities of <10% or >90%, indicating that classification of a variant as class 1 or class 5 could be easily performed by incorporating segregation or tumor information. it is also possible that some class 3 variants with low prior probability of pathogenicity based on the predicted missense alteration would cause splicing aberrations, as already observed for 42 of the 746 not obviously truncating nonsynonymous variants. incorporation of validated bioinformatic splicing prediction tools into the mmr gene multifactorial model, as is under development for brca1 and brca2 (ref. 35), will assist in the prioritization of variants with the potential to disrupt splicing.  figure 5: probabilities of pathogenicity for 481 class 3 missense variants of uncertain effect derived by in silico analysis. probabilities of pathogenicity for 481 class 3 missense variants of uncertain effect derived by in silico analysis. (a–e) distribution of probabilities of pathogenicity as estimated from a calibrated algorithm based on customized mapp and polyphen-2 scores34 for mlh1, n = 186 (a), msh2, n = 169 (b), msh6, n = 145 (c), pms2, n = 24 (d) and all four mmr genes (e), showing stratification of variants with prior probabilities of ≤20% or ≥80% to prioritize variants for further investigation and classification.  full size image (69 kb) download excel source data (36 kb) previous figures/tables index in investigation of the potential effects of class 3 regulatory variants (online methods), all 15 5′ utr variants fell within multiple transcription factor binding sites, but no evidence for interruption of microrna binding was found for 6 variants in the 3′ utr (data not shown). multifactorial analyses and transcription assays would help elucidate whether these variants affect gene function.  discussion abstract• introduction• results• discussion• methods• accession codes• references• acknowledgments• author information• supplementary information the insight vic has successfully undertaken a collaborative effort to establish standardized variant interpretation guidelines, encourage data submission and provide objective assessment of mmr gene variants involved in lynch syndrome. the criteria developed provide a basis for the standardized clinical classification of variants to inform patient and family management through genetic counseling10. this initiative has achieved the systematic evaluation of 2,360 constitutional variants, which will benefit thousands of families internationally. notably, 605 variants not resulting in the introduction of a premature termination codon, including 217 nonsynonymous substitutions, have now been assigned to class 5 (pathogenic) and class 4 (likely pathogenic) or to class 1 (not pathogenic) and class 2 (likely not pathogenic). the assigned classes of these variants can now also be used as standards for the calibration of functional assays36, 37.  the clinical relevance of 32% of the variants investigated remains uncertain. a large proportion of these (71%) were private variants occurring in only one family, and these variants are difficult to classify owing to the paucity of available clinical information. clinicians have a fundamental role in promoting the collection of segregation information and other data relevant for classification. we anticipate that the development of this interpretation scheme, along with the implementation of microattribution, will incentivize assistance in the accumulation of clinical data. the value of microattribution for data accrual has previously been demonstrated for hemoglobinopathies24, and the insight initiative now demonstrates the clinical usefulness of data collection. the promotion of standardized data formats will assist in the transition to fully quantitative, unbiased classification, eventually incorporating other components of the qualitative guidelines. in addition, the difficulties experienced in interpreting apparently discordant data from functional assays emphasize the importance of assay validation and standardization38, 39. such experience will be directly applicable to the functional analysis of deep intronic and regulatory variants, which are increasingly detected with the advancement of dna sequencing technologies.  to accommodate the lower penetrance and reported lower degrees of tumor msi associated with msh6 and pms2 mutations28, 29, 40, 41, 42, 43, 44, gene-specific criteria should also be considered for future iterations of the classification guidelines, for example, stipulating the inclusion of segregation odds for msh6 and pms2 variants for classification or using modified panels to detect msi status.  another challenging issue to contemplate will be incorporating alleles of intermediate risk45 into classification schemes, including the clinical recommendations that might be linked to such variants. the identification of a subset of mmr gene alleles with apparently discordant clinical and functional features that renders them resistant to classification will provide the basis for future studies to define the most appropriate methodology and criteria to identify such variants. further studies will also be required to assess whether variants resulting in abrogated dna damage response but normal mmr46 are associated with the same clinical features as classical pathogenic mutations in mmr genes.  the insight database is a well-recognized resource for the clinical and research communities, receiving over 20,000 hits per month. the development and adoption of standard templates allows transparency in the review process. database users can view relevant information and sources in relation to information on guideline interpretation when considering the classification provided by the committee. the guidelines must evolve to accommodate additional kinds of evidence, but we anticipate no clinical issues as long as variant classifications are dated and linked to a dated set of guidelines with the supporting information used to derive classifications. final classifications have also been submitted to the ncbi clinvar database for higher exposure, but expert classifications and underlying data rest with insight.  to our knowledge, this is the first large-scale comprehensive classification effort demonstrating the value of evaluation by expert panel in the curation of an lsdb and providing summary information used to assign variant pathogenicity. this initiative also shows how classification may be assisted by promoting standardized data submission from stakeholders in the clinical and research settings, thereby allowing access to unpublished clinical and functional information used to facilitate variant classification. thus, the insight initiative provides a key model of lsdb-centric multidisciplinary collaboration for the transparent interpretation of dna variants.  methods abstract• introduction• results• discussion• methods• accession codes• references• acknowledgments• author information• supplementary information insight variant interpretation committee expertise. the insight vic (current chair, m.g.) includes 40 multidisciplinary experts from 5 continents (see supplementary table 9 for disciplines covered by vic members). the committee is responsible to its governance committee, which in turn is responsible to the insight council. insight has recently joined hvp and is a founding member of its gene and disease specific council. the insight council specifically considered the need and responsibility associated with classification assignment on its database and took all reasonable steps to both invite the highest possible expertise to contribute to the classification process and to ensure that its processes and legal standing are robust.  insight database curation. mutalyzer50 was used to standardize the nomenclature of all variants present in the database as of december 2012. variants with multiple submissions that were originally assigned a classification of pathogenic or probably pathogenic as well as no known pathogenicity or probably no pathogenicity were included in the group of discordant variants used to test the classification criteria. all unique variants identified in the database were assigned to one of the following sources: constitutional, somatic, artificial or unknown.  development of five-tiered insight classification criteria. the insight classification criteria were developed using the delphi method25. a five-tiered classification system originally developed for consistent classification of mmr gene variants identified in participants of the colon cancer family registry16, 34 was selected as a baseline for the insight classification criteria. this system included the option of classification on the basis of the posterior probabilities arising from multifactorial likelihood analysis15, 16, 51, 52 as well as multiple combinations of qualitative data not yet calibrated for inclusion in quantitative analyses but that are often reported in the literature or available from clinical sources. these baseline classification criteria were critically reviewed by insight vic members attending the insight san antonio meeting in april 2011 and by vic members via e-mail. in response to comment, the rules were amended for clarity, to apply a more stringent interpretation of functional assay data and to consider additional points of evidence. these insight rules were used for variant classification over a series of 11 meetings (10 teleconferences and 1 face-to-face meeting), with further changes incorporated after each meeting to include additional points of evidence identified to be relevant during the review process as the committee encountered different combinations of useful data from published and unpublished sources. for example, after discussion, co-occurrence of a variant with a pathogenic mutation in the same gene with clinical information regarding a constitutional mmr deficiency phenotype53 was included as an in vivo test of mmr function, and 1000 genomes project data54 were accepted as a test for population frequency. consistency of the accumulative evidence required for a given class was reviewed by presentation of the rules at a face-to-face meeting of committee members. supporting documentation was developed to assist in the interpretation of splicing and functional assay results by b.a.t. in consultation with a subset of committee members with specific expertise in this field ( 1b and supplementary tables 4 and 5). where necessary, rule alterations were applied retrospectively to variants evaluated in previous meetings. the finalized rules (shown in simplified format in  1 and detailed in the supplementary note) were then used to assess all remaining variants lodged in the insight database.  classification of mmr gene variants by literature review and data collation. variants occurring in the 1000 genomes project54 with allele frequency greater than 1% were automatically classified as class 1 variants. committee members were invited to participate in at least one classification meeting. a core group participated in each meeting, with attendance invited from vic membership to make up the balance. before each meeting, participants were assigned, through randomization, a subset of variants to be assessed. each attendee was provided literature pertaining to the list of variants to be discussed and, where relevant, additional unpublished clinical or research information submitted by committee members to insight curator j.- meeting attendees were requested to thoroughly review and summarize all information pertaining to the subset of variants in a spreadsheet template and to provide a class assignment based on their interpretation of the information accessed. all reviewer summaries, submitted clinical information and results from causality analysis were compiled into a single file to allow the comparison of data and class assignments for each variant and were circulated to the teleconference participants. during committee meetings, variants were discussed one at a time, assessing the following: class assigned by each reviewer; rationale for classification according to the classification guidelines; difficulties in interpreting specific data sources; assessment of possible redundancy of information due to multiple publications including all or some of the same information pertaining to a variant; differences in interpretation of the guidelines as provided and adjustments required to improve their clarity; the consensus view on variant class considering the preceding discussion; and action required to obtain additional information for refining the classification of variants that remained in class 2, 3 or 4 at the close of discussion. where classifications differed using qualitative and quantitative criteria, these differences were due to differences in the availability of specific data types for the two approaches, and the most extreme classification was assigned for relevant variants. b.a.t. prioritized variants for examination by identifying and classifying any variants for which rules-based classification could be applied, such as variants that were truncating or comprised a large deletion from nomenclature, canonical splice site with no splicing data or frequency of >1% in a control reference grou b.a.t. then collated all information for all remaining unique variants (including those reviewed previously in teleconferences) and determined which variants had sufficient information to allow classification outside of class 3. summary information for these variants was circulated for independent class assignment by at least three reviewers from the vic, and classification was finalized at teleconferences or by e-mail.  validation of qualitative criteria. a subset of truncating variants and large genomic deletions was selected to validate the qualitative classification criteria. variants were selected on the basis of the availability of data from the first point of evidence in the qualitative class 5 criterion, , in vitro functional assay results (for example, protein truncation test or genomic or mrna confirmation of large deletions); constitutional mmr deficiency syndrome phenotype; or different haplotypes across multiple families. published and unpublished data for these variants were then used to validate the other points of evidence required for classification as a class 5 (pathogenic) variant.  preliminary analysis of class 3 (uncertain) variants. in silico probabilities of pathogenicity were estimated for all class 3 missense variants, as described elsewhere34. preliminary bioinformatic analysis of class 3 regulatory variants was undertaken using encyclopedia of dna elements (encode) data55 on the ucsc genome browser.  implementation of the microattribution process. the variant interpretation process uses both published and unpublished data. for published literature, the pubmed id (pmid) was used to reference the original work. some unpublished data were recorded in the insight database at study initiation, and insight members were also requested by e-mail to contribute information important for variant classification using a standardized submission template. data submitters were requested to provide a permanent, publicly searchable unique id, preferably from the orcid system, to facilitate the adoption of the microattribution approach. microattribution was assigned for the different types of information corresponding to the points of evidence required for classification—namely, submitters were allocated one credit of microattribution for each type of information received, including (i) a variant (mandatory), (ii) family history or pedigree, (iii) msi information, (iv) immunohistochemistry data, (v) in vitro functional data, (vi) data from rna splicing assays and (vii) population frequency data. all unpublished data received by the vic were recorded in microattribution tables for each element type, with each microattribution table listing a unique researcher id along with submitted information. microattribution counts for submitters are publically available on the insight website. additionally, the data will be made available in nanopublication format.  statistical analysis. multifactorial likelihood analysis was performed for variants with appropriate tumor and segregation data available, using previously reported methods16, 34, 51 that are described briefly as follows. bayes factor analysis was conducted by b.a.t. to assess mlh1, msh2, msh6 and pms2 variant causality from segregation data16, 51 for both published and unpublished pedigrees with sufficient relevant information on cancer and variant carrier status. penetrance estimates from senter 28 were used in the bayes segregation analysis27 of pms2 variants. where family relationship status was unknown, a conservative segregation likelihood ratio was derived—, setting affected carriers as first-degree relatives—which is less informative than segregation between second-degree relatives. colorectal tumor msi and somatic braf mutation status were used to assign likelihood ratios according to tumor phenotype16. for each variant, the individual likelihood ratios (cosegregation, tumor) were multiplied to calculate the odds for causality. then, a posterior probability was calculated by combining the prior probability (in silico for missense variants34 or on the basis of sequence location for all other variants13) and the odds for causality using bayes rule where posterior = (prior × odds × (1/(1−prior)))/(prior × odds × (1/(1 − prior)) + 1). stata 11 was used to calculate the sample size for the truncating variant validation set, h0: p = , assuming α =  (one-sided) and power = .  all other analyses were completed using the statistical package r and graphpad prism 6. for meta-analysis of population frequency data, the proportions were combined using an inverse variance random-effects model to account for heterogeneity between studies.\n",
      "v509a\n",
      "msh6\n"
     ]
    }
   ],
   "source": [
    "ID = 111  \n",
    "print(to_fix.loc[ID,\"Text\"])\n",
    "print(to_fix.loc[ID,\"Variation\"])\n",
    "print(to_fix.loc[ID,\"Gene\"].lower())                 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Save les datas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Text_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>here, we deorphanize cdk10 by identifying cycl...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>most of the changes were novel, although 4 ca...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>all mutations from the second group were pred...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3279</th>\n",
       "      <td>3316</td>\n",
       "      <td>21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutatio...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3280</th>\n",
       "      <td>3317</td>\n",
       "      <td>(a) spleen from morbid mice/d171n (left) and ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3281</th>\n",
       "      <td>3318</td>\n",
       "      <td>lpxn is preferentially expressed in hematopoi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3282</th>\n",
       "      <td>3319</td>\n",
       "      <td>) conversely, mutations mapping to the first d...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3283</th>\n",
       "      <td>3320</td>\n",
       "      <td>mutation analysis in fpd/aml pedigrees. elect...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3284 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        ID                                             Text_y\n",
       "0        0  here, we deorphanize cdk10 by identifying cycl...\n",
       "1        1   the c-cbl loh also positively correlated with...\n",
       "2        2   the c-cbl loh also positively correlated with...\n",
       "3        3   most of the changes were novel, although 4 ca...\n",
       "4        4   all mutations from the second group were pred...\n",
       "...    ...                                                ...\n",
       "3279  3316  21⇓⇓⇓⇓⇓⇓⇓–29 the vast majority of aml1 mutatio...\n",
       "3280  3317   (a) spleen from morbid mice/d171n (left) and ...\n",
       "3281  3318   lpxn is preferentially expressed in hematopoi...\n",
       "3282  3319  ) conversely, mutations mapping to the first d...\n",
       "3283  3320   mutation analysis in fpd/aml pedigrees. elect...\n",
       "\n",
       "[3284 rows x 2 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "#CBL = train_variant[train_variant[\"Gene\"] == \"RUNX1\"]\n",
    "#article_cbl = train_text.loc[CBL.index,]\n",
    "\n",
    "#CBL.to_csv(\"cbl.txt\")\n",
    "#clean_TERT = new_data_best_score[new_data_best_score[\"Gene\"] == \"BRCA1\"][\"Text_y\"]\n",
    "dtf = pd.merge(pd.DataFrame(new_data_best_score.index), new_data_best_score[\"Text_y\"], on =\"ID\")\n",
    "dtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "np.savetxt(\"../datas/all_data_clean.txt\",dtf, fmt = \"%d || %s\", header= \",\".join(dtf.columns), comments='')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: sklearn in /home/ragou/.local/lib/python3.8/site-packages (0.0)\n",
      "Collecting scikit-learn\n",
      "  Downloading scikit_learn-1.0-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (25.8 MB)\n",
      "\u001b[K     |████████████████████████████████| 25.8 MB 122 kB/s  eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: threadpoolctl>=2.0.0 in /home/ragou/.local/lib/python3.8/site-packages (from scikit-learn->sklearn) (2.1.0)\n",
      "Requirement already satisfied: numpy>=1.14.6 in /usr/lib/python3/dist-packages (from scikit-learn->sklearn) (1.17.4)\n",
      "Requirement already satisfied: joblib>=0.11 in /home/ragou/.local/lib/python3.8/site-packages (from scikit-learn->sklearn) (0.17.0)\n",
      "Requirement already satisfied: scipy>=1.1.0 in /home/ragou/.local/lib/python3.8/site-packages (from scikit-learn->sklearn) (1.7.1)\n",
      "Installing collected packages: scikit-learn\n",
      "Successfully installed scikit-learn-1.0\n"
     ]
    }
   ],
   "source": [
    "!pip3 install sklearn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on package sklearn:\n",
      "\n",
      "NAME\n",
      "    sklearn\n",
      "\n",
      "DESCRIPTION\n",
      "    Machine learning module for Python\n",
      "    ==================================\n",
      "    \n",
      "    sklearn is a Python module integrating classical machine\n",
      "    learning algorithms in the tightly-knit world of scientific Python\n",
      "    packages (numpy, scipy, matplotlib).\n",
      "    \n",
      "    It aims to provide simple and efficient solutions to learning problems\n",
      "    that are accessible to everybody and reusable in various contexts:\n",
      "    machine-learning as a versatile tool for science and engineering.\n",
      "    \n",
      "    See http://scikit-learn.org for complete documentation.\n",
      "\n",
      "PACKAGE CONTENTS\n",
      "    __check_build (package)\n",
      "    _build_utils (package)\n",
      "    _config\n",
      "    _distributor_init\n",
      "    _isotonic\n",
      "    _loss (package)\n",
      "    _min_dependencies\n",
      "    base\n",
      "    calibration\n",
      "    cluster (package)\n",
      "    compose (package)\n",
      "    conftest\n",
      "    covariance (package)\n",
      "    cross_decomposition (package)\n",
      "    datasets (package)\n",
      "    decomposition (package)\n",
      "    discriminant_analysis\n",
      "    dummy\n",
      "    ensemble (package)\n",
      "    exceptions\n",
      "    experimental (package)\n",
      "    externals (package)\n",
      "    feature_extraction (package)\n",
      "    feature_selection (package)\n",
      "    gaussian_process (package)\n",
      "    impute (package)\n",
      "    inspection (package)\n",
      "    isotonic\n",
      "    kernel_approximation\n",
      "    kernel_ridge\n",
      "    linear_model (package)\n",
      "    manifold (package)\n",
      "    metrics (package)\n",
      "    mixture (package)\n",
      "    model_selection (package)\n",
      "    multiclass\n",
      "    multioutput\n",
      "    naive_bayes\n",
      "    neighbors (package)\n",
      "    neural_network (package)\n",
      "    pipeline\n",
      "    preprocessing (package)\n",
      "    random_projection\n",
      "    semi_supervised (package)\n",
      "    setup\n",
      "    svm (package)\n",
      "    tests (package)\n",
      "    tree (package)\n",
      "    utils (package)\n",
      "\n",
      "FUNCTIONS\n",
      "    clone(estimator, *, safe=True)\n",
      "        Constructs a new unfitted estimator with the same parameters.\n",
      "        \n",
      "        Clone does a deep copy of the model in an estimator\n",
      "        without actually copying attached data. It yields a new estimator\n",
      "        with the same parameters that has not been fitted on any data.\n",
      "        \n",
      "        If the estimator's `random_state` parameter is an integer (or if the\n",
      "        estimator doesn't have a `random_state` parameter), an *exact clone* is\n",
      "        returned: the clone and the original estimator will give the exact same\n",
      "        results. Otherwise, *statistical clone* is returned: the clone might\n",
      "        yield different results from the original estimator. More details can be\n",
      "        found in :ref:`randomness`.\n",
      "        \n",
      "        Parameters\n",
      "        ----------\n",
      "        estimator : {list, tuple, set} of estimator instance or a single             estimator instance\n",
      "            The estimator or group of estimators to be cloned.\n",
      "        \n",
      "        safe : bool, default=True\n",
      "            If safe is False, clone will fall back to a deep copy on objects\n",
      "            that are not estimators.\n",
      "    \n",
      "    config_context(**new_config)\n",
      "        Context manager for global scikit-learn configuration\n",
      "        \n",
      "        Parameters\n",
      "        ----------\n",
      "        assume_finite : bool, default=False\n",
      "            If True, validation for finiteness will be skipped,\n",
      "            saving time, but leading to potential crashes. If\n",
      "            False, validation for finiteness will be performed,\n",
      "            avoiding error.  Global default: False.\n",
      "        \n",
      "        working_memory : int, default=1024\n",
      "            If set, scikit-learn will attempt to limit the size of temporary arrays\n",
      "            to this number of MiB (per job when parallelised), often saving both\n",
      "            computation time and memory on expensive operations that can be\n",
      "            performed in chunks. Global default: 1024.\n",
      "        \n",
      "        print_changed_only : bool, default=True\n",
      "            If True, only the parameters that were set to non-default\n",
      "            values will be printed when printing an estimator. For example,\n",
      "            ``print(SVC())`` while True will only print 'SVC()', but would print\n",
      "            'SVC(C=1.0, cache_size=200, ...)' with all the non-changed parameters\n",
      "            when False. Default is True.\n",
      "        \n",
      "            .. versionchanged:: 0.23\n",
      "               Default changed from False to True.\n",
      "        \n",
      "        display : {'text', 'diagram'}, default='text'\n",
      "            If 'diagram', estimators will be displayed as a diagram in a Jupyter\n",
      "            lab or notebook context. If 'text', estimators will be displayed as\n",
      "            text. Default is 'text'.\n",
      "        \n",
      "            .. versionadded:: 0.23\n",
      "        \n",
      "        Notes\n",
      "        -----\n",
      "        All settings, not just those presently modified, will be returned to\n",
      "        their previous values when the context manager is exited.\n",
      "        \n",
      "        Examples\n",
      "        --------\n",
      "        >>> import sklearn\n",
      "        >>> from sklearn.utils.validation import assert_all_finite\n",
      "        >>> with sklearn.config_context(assume_finite=True):\n",
      "        ...     assert_all_finite([float('nan')])\n",
      "        >>> with sklearn.config_context(assume_finite=True):\n",
      "        ...     with sklearn.config_context(assume_finite=False):\n",
      "        ...         assert_all_finite([float('nan')])\n",
      "        Traceback (most recent call last):\n",
      "        ...\n",
      "        ValueError: Input contains NaN, ...\n",
      "        \n",
      "        See Also\n",
      "        --------\n",
      "        set_config : Set global scikit-learn configuration.\n",
      "        get_config : Retrieve current values of the global configuration.\n",
      "    \n",
      "    get_config()\n",
      "        Retrieve current values for configuration set by :func:`set_config`\n",
      "        \n",
      "        Returns\n",
      "        -------\n",
      "        config : dict\n",
      "            Keys are parameter names that can be passed to :func:`set_config`.\n",
      "        \n",
      "        See Also\n",
      "        --------\n",
      "        config_context : Context manager for global scikit-learn configuration.\n",
      "        set_config : Set global scikit-learn configuration.\n",
      "    \n",
      "    set_config(assume_finite=None, working_memory=None, print_changed_only=None, display=None)\n",
      "        Set global scikit-learn configuration\n",
      "        \n",
      "        .. versionadded:: 0.19\n",
      "        \n",
      "        Parameters\n",
      "        ----------\n",
      "        assume_finite : bool, default=None\n",
      "            If True, validation for finiteness will be skipped,\n",
      "            saving time, but leading to potential crashes. If\n",
      "            False, validation for finiteness will be performed,\n",
      "            avoiding error.  Global default: False.\n",
      "        \n",
      "            .. versionadded:: 0.19\n",
      "        \n",
      "        working_memory : int, default=None\n",
      "            If set, scikit-learn will attempt to limit the size of temporary arrays\n",
      "            to this number of MiB (per job when parallelised), often saving both\n",
      "            computation time and memory on expensive operations that can be\n",
      "            performed in chunks. Global default: 1024.\n",
      "        \n",
      "            .. versionadded:: 0.20\n",
      "        \n",
      "        print_changed_only : bool, default=None\n",
      "            If True, only the parameters that were set to non-default\n",
      "            values will be printed when printing an estimator. For example,\n",
      "            ``print(SVC())`` while True will only print 'SVC()' while the default\n",
      "            behaviour would be to print 'SVC(C=1.0, cache_size=200, ...)' with\n",
      "            all the non-changed parameters.\n",
      "        \n",
      "            .. versionadded:: 0.21\n",
      "        \n",
      "        display : {'text', 'diagram'}, default=None\n",
      "            If 'diagram', estimators will be displayed as a diagram in a Jupyter\n",
      "            lab or notebook context. If 'text', estimators will be displayed as\n",
      "            text. Default is 'text'.\n",
      "        \n",
      "            .. versionadded:: 0.23\n",
      "        \n",
      "        See Also\n",
      "        --------\n",
      "        config_context : Context manager for global scikit-learn configuration.\n",
      "        get_config : Retrieve current values of the global configuration.\n",
      "    \n",
      "    show_versions()\n",
      "        Print useful debugging information\"\n",
      "        \n",
      "        .. versionadded:: 0.20\n",
      "\n",
      "DATA\n",
      "    __SKLEARN_SETUP__ = False\n",
      "    __all__ = ['calibration', 'cluster', 'covariance', 'cross_decompositio...\n",
      "\n",
      "VERSION\n",
      "    1.0\n",
      "\n",
      "FILE\n",
      "    /home/ragou/.local/lib/python3.8/site-packages/sklearn/__init__.py\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(sklearn)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Train et Test data 70/30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sklearn.model_selection\n",
    "train_x,test_x, train_y,test_y = sklearn.model_selection.train_test_split(full_x_data,full_y_data, train_size = 0.7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "x_train :  (2321, 2)\n",
      "x_test :  (995, 2)\n",
      "y_train :  (2321, 3)\n",
      "y_test :  (995, 3)\n"
     ]
    }
   ],
   "source": [
    "print(\"x_train : \",train_x.shape)\n",
    "print(\"x_test : \",test_x.shape)\n",
    "print(\"y_train : \",train_y.shape)\n",
    "print(\"y_test : \",test_y.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Text</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3114</th>\n",
       "      <td>the pathogenetically critical sequences seem ...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181</th>\n",
       "      <td>the pip3 catalytic activity of the resulting ...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>914</th>\n",
       "      <td>however, most isoforms with a substitution in...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2156</th>\n",
       "      <td>this prompted us to examine additional inheri...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1902</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2804</th>\n",
       "      <td>the presence of the mutations was confirmed b...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>711</th>\n",
       "      <td>the effect of trastuzumab and neratinib on so...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>the smad4 point mutations were amplified with...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>in the br21 clinical trial, there were no g71...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2321 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Text  Score\n",
       "ID                                                            \n",
       "3114   the pathogenetically critical sequences seem ...      4\n",
       "2181   the pip3 catalytic activity of the resulting ...      1\n",
       "914    however, most isoforms with a substitution in...      1\n",
       "2156   this prompted us to examine additional inheri...      1\n",
       "1902                                                 []      0\n",
       "...                                                 ...    ...\n",
       "2804   the presence of the mutations was confirmed b...      1\n",
       "145                                                  []      0\n",
       "711    the effect of trastuzumab and neratinib on so...      1\n",
       "602    the smad4 point mutations were amplified with...      1\n",
       "216    in the br21 clinical trial, there were no g71...      1\n",
       "\n",
       "[2321 rows x 2 columns]"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3114</th>\n",
       "      <td>RAD51B</td>\n",
       "      <td>hmga2-rad51b fusion</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>t131a</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>914</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>d846y</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2156</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>h118p</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1902</th>\n",
       "      <td>TET1</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2804</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>r2659t</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>k467t</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>711</th>\n",
       "      <td>ERBB2</td>\n",
       "      <td>l866m</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>SMAD4</td>\n",
       "      <td>e330k</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>l861p</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2321 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene             Variation  Class\n",
       "ID                                       \n",
       "3114  RAD51B   hmga2-rad51b fusion      8\n",
       "2181    PTEN                 t131a      4\n",
       "914   PDGFRA                 d846y      7\n",
       "2156    PTEN                 h118p      4\n",
       "1902    TET1  truncating mutations      1\n",
       "...      ...                   ...    ...\n",
       "2804   BRCA2                r2659t      1\n",
       "145     EGFR                 k467t      2\n",
       "711    ERBB2                 l866m      7\n",
       "602    SMAD4                 e330k      4\n",
       "216     EGFR                 l861p      2\n",
       "\n",
       "[2321 rows x 3 columns]"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Text</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3114</td>\n",
       "      <td>the pathogenetically critical sequences seem ...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2181</td>\n",
       "      <td>the pip3 catalytic activity of the resulting ...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>914</td>\n",
       "      <td>however, most isoforms with a substitution in...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2156</td>\n",
       "      <td>this prompted us to examine additional inheri...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1902</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2316</th>\n",
       "      <td>2804</td>\n",
       "      <td>the presence of the mutations was confirmed b...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2317</th>\n",
       "      <td>145</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2318</th>\n",
       "      <td>711</td>\n",
       "      <td>the effect of trastuzumab and neratinib on so...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2319</th>\n",
       "      <td>602</td>\n",
       "      <td>the smad4 point mutations were amplified with...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2320</th>\n",
       "      <td>216</td>\n",
       "      <td>in the br21 clinical trial, there were no g71...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2321 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        ID                                               Text  Score\n",
       "0     3114   the pathogenetically critical sequences seem ...      4\n",
       "1     2181   the pip3 catalytic activity of the resulting ...      1\n",
       "2      914   however, most isoforms with a substitution in...      1\n",
       "3     2156   this prompted us to examine additional inheri...      1\n",
       "4     1902                                                 []      0\n",
       "...    ...                                                ...    ...\n",
       "2316  2804   the presence of the mutations was confirmed b...      1\n",
       "2317   145                                                 []      0\n",
       "2318   711   the effect of trastuzumab and neratinib on so...      1\n",
       "2319   602   the smad4 point mutations were amplified with...      1\n",
       "2320   216   in the br21 clinical trial, there were no g71...      1\n",
       "\n",
       "[2321 rows x 3 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dtf = pd.merge(pd.DataFrame(train_x.index), train_x, on =\"ID\")\n",
    "dtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "dtf = pd.merge(pd.DataFrame(train_x.index), train_x, on =\"ID\")\n",
    "dtf\n",
    "np.savetxt(\"../datas/x_train.txt\",dtf, fmt = \"%d || %s || %d\", header= \",\".join(dtf.columns), comments='')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "dtf = pd.merge(pd.DataFrame(test_x.index), test_x, on =\"ID\")\n",
    "dtf\n",
    "np.savetxt(\"../datas/x_test.txt\",dtf, fmt = \"%d || %s || %d\", header= \",\".join(dtf.columns), comments='')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "train_y.to_csv(\"y_train.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_y.to_csv(\"../datas/y_test.txt\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "deep_projet",
   "language": "python",
   "name": "deep_projet"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
