{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import seaborn as sns\n",
    "import matplotlib.pyplot as plt\n",
    "import re\n",
    "import time"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "file_train_text = \"../../data/training_text/training_text\"\n",
    "file_train_variant = \"../../data/training_variants/training_variants\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ragou/.local/lib/python3.8/site-packages/pandas/util/_decorators.py:311: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support regex separators (separators > 1 char and different from '\\s+' are interpreted as regex); you can avoid this warning by specifying engine='python'.\n",
      "  return func(*args, **kwargs)\n"
     ]
    }
   ],
   "source": [
    "train_text = pd.read_csv(file_train_text, sep = '\\|\\|')\n",
    "train_text.index.name = \"ID\"\n",
    "train_text.columns = [\"Text\"]\n",
    "\n",
    "train_variant = pd.read_csv(file_train_variant)\n",
    "train_variant.set_index(\"ID\",inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Y371H</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>CBL</td>\n",
       "      <td>K382E</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>CBL</td>\n",
       "      <td>C381A</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>CBL</td>\n",
       "      <td>C396R</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Variation  Class\n",
       "ID                      \n",
       "8   CBL     Y371H      4\n",
       "11  CBL     K382E      4\n",
       "13  CBL     C381A      4\n",
       "21  CBL     C396R      4"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_variant[train_variant[\"Variation\"].isin([\"C396R\", \"H398Q\", \"Y371H\", \"K382E\", \"C381A\"])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   Gene Variation  Class\n",
       "ID                      \n",
       "2   CBL     Q249E      2"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_variant[train_variant[\"Variation\"] == \"Q249E\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                 Text\n",
       "ID                                                   \n",
       "6   Oncogenic mutations in the monomeric Casitas B..."
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_text.loc[[6],]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_count = train_variant[\"Gene\"].value_counts().loc[lambda x : x > 100].to_dict()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "gene_class = train_variant[train_variant[\"Gene\"].isin(gene_count.keys())]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA4EAAAFNCAYAAAC+H2oqAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/MnkTPAAAACXBIWXMAAAsTAAALEwEAmpwYAAAcvElEQVR4nO3dfbBlVXkn4N8rDVFQAaFDoCFCRtQQJyr2KEqijqgRv3AcZYxRiWOKpEqNikkkJhUdJ6ZMYuJn4hQBFSZGRfADE2J0xK9kIrFBogJmbAkKLR8tIirGCPrOH2e3XtvGvug9Z/e9+3mqTt29115nn/fs7qru311rr13dHQAAAKbhNmMXAAAAwOIIgQAAABMiBAIAAEyIEAgAADAhQiAAAMCECIEAAAATIgQCsGpU1Yur6i9H/PwPVtWvDNu/VFXvXcFzX1xVDx62V/R7VtULq+rUlTofAKubEAjALqWqnlxVm6rqa1V1VVX9bVX93Nh1ba+739TdD99Zv6p6Y1X9/jLO9zPd/cEfta6qenBVXbnduf+gu3/lRz03AGuDEAjALqOqTkryyiR/kOSAJD+Z5M+THDdiWXNVVevGrgGAaRECAdglVNXeSV6S5Jnd/fbuvrG7b+rud3f3b97Ce95WVVdX1Q1V9eGq+pklxx5ZVZdU1VeraktV/cbQvn9V/XVVfbmqvlRVH6mqHf57WFUPq6pPD+d/bZJacuyXq+rvh+2qqldU1bVV9ZWq+mRV3aOqTkzyS0l+axjZfPfQ//KqekFVfSLJjVW1bmh76JKPv21VvXWo/8KquueSz+6qusuS/TdW1e9X1V5J/jbJQcPnfa2qDtp+emlVPXaYfvrlYYrrTy85dnlV/UZVfWL43m+tqtsu448QgFVCCARgV3H/JLdN8o5b8Z6/TXJ4kh9PcmGSNy05dlqSX+3uOyS5R5LzhvbnJ7kyyfrMRhtfmKS3P3FV7Z/k7Ul+N8n+ST6b5OhbqOPhSR6Y5K5J9k5yfJLruvuUoaY/6u7bd/djlrznF5M8Ksk+3X3zDs55XJK3JblTkr9K8s6q2v0Wr0SS7r4xybFJvjB83u27+wvbfa+7JnlzkucO1+DcJO+uqj2WdDs+ySOSHJbkZ5P88g/6XABWFyEQgF3Ffkm+eAuBaIe6+/Xd/dXu/vckL05yz2FEMUluSnJEVd2xu6/v7guXtB+Y5M7DSONHuvv7QmCSRya5uLvP6u6bMpumevUtlHJTkjskuXuS6u5Lu/uqnZT/6u6+orv/7RaOX7Dks/80s4B81E7OuRz/LcnfdPf7hnO/PMntkjxgu9q+0N1fSvLuJPdagc8FYBchBAKwq7guyf7LvUeuqnarqpdV1Wer6itJLh8O7T/8/K+ZBbnPVdWHqur+Q/sfJ9mc5L1VdVlVnXwLH3FQkiu27QxB8Yoddezu85K8NsmfJbm2qk6pqjvu5Cvs8Fw7Ot7d385s9PKgnbxnOQ5K8rntzn1Fkg1L+iwNu19PcvsV+FwAdhFCIAC7in9M8u9JHrfM/k/ObMrkQzObgnno0F5J0t0f6+7jMpsq+s4kZw7tX+3u53f3TyV5bJKTquqYHZz/qiSHbNupqlq6v73ufnV33yfJEZlNC912H+OORhl/UPs2Sz/7NkkOTrJtaufXk+y5pO9P3IrzfiHJnZece9v32rKT9wGwRgiBAOwSuvuGJL+X5M+q6nFVtWdV7V5Vx1bVH+3gLXfILDRel1kg+oNtB6pqj+E5fnsPUx6/kuTbw7FHV9VdhvBzQ5JvbTu2nb9J8jNV9fhhdPLX871h6zuq6j9V1f2Ge/ZuTPKNJee8JslP3crLkST3WfLZzx2+60eHYxclefIwGvqIJA9a8r5rkuy3ZFrs9s5M8qiqOmao9/nDuf/vD1EjAKuQEAjALqO7/yTJSZktxrI1s2mKz8psJG97Z2Q2rXFLkkvy3YC0zVOTXD5MFf21zFbpTGYLyfyfJF/LbPTxz7v7Azuo5YtJnpjkZZkFzcOT/MMtlH7HJH+R5Pqhpusym3aazBaoOWJYiXNH3+OWvCuz+/euH77L44dAmyTPSfKYJF8evtd3ztvdn85s4ZfLhs/8nimk3f0vSZ6S5DVJvjic5zHd/c1bURsAq1jt+F54AAAA1iIjgQAAABMytxBYVa8fHpr7qSVtd6qq91XVZ4af+w7tVVWvrqrNw8Npj1zynhOG/p+pqhPmVS8AAMAUzHMk8I2ZPWh2qZOTvL+7D0/y/mE/mT3Y9vDhdWKS1yWz0JjkRUnul+S+SV60LTgCAABw680tBHb3h5N8abvm45KcPmyfnu8uA35ckjN65qNJ9qmqA5P8QpL3dfeXuvv6JO/L9wdLAAAAlmnR9wQe0N1XDdtXJzlg2N6Q731o7pVD2y21AwAA8ENYN9YHd3dX1YotTVpVJ2Y2lTR77bXXfe5+97uv1KkBAABWlQsuuOCL3b1+R8cWHQKvqaoDu/uqYbrntUP7liSHLOl38NC2JcmDt2v/4I5O3N2nJDklSTZu3NibNm1a2coBAABWiar63C0dW/R00HOSbFvh84TMHoS7rf1pwyqhRyW5YZg2+ndJHl5V+w4Lwjx8aAMAAOCHMLeRwKp6c2ajePtX1ZWZrfL5siRnVtUzknwuyfFD93OTPDLJ5iRfT/L0JOnuL1XV/0zysaHfS7p7+8VmAAAAWKbqXrHb8nYZpoMCAABTVlUXdPfGHR1b9HRQAAAARiQEAgAATIgQCAAAMCFCIAAAwIQIgQAAABMiBAIAAEyIEAgAADAhQiAAAMCErBu7AAAAdu7Sl543dgmrwk//zkPGLgF2eUYCAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJmSUEFhVz6uqi6vqU1X15qq6bVUdVlXnV9XmqnprVe0x9P2xYX/zcPzQMWoGAABYCxYeAqtqQ5JfT7Kxu++RZLckT0ryh0le0d13SXJ9kmcMb3lGkuuH9lcM/QAAAPghjDUddF2S21XVuiR7JrkqyUOSnDUcPz3J44bt44b9DMePqapaXKkAAABrx8JDYHdvSfLyJJ/PLPzdkOSCJF/u7puHblcm2TBsb0hyxfDem4f++y2yZgAAgLVijOmg+2Y2undYkoOS7JXkEStw3hOralNVbdq6deuPejoAAIA1aYzpoA9N8q/dvbW7b0ry9iRHJ9lnmB6aJAcn2TJsb0lySJIMx/dOct32J+3uU7p7Y3dvXL9+/by/AwAAwKo0Rgj8fJKjqmrP4d6+Y5JckuQDSZ4w9DkhybuG7XOG/QzHz+vuXmC9AAAAa8YY9wSen9kCLxcm+eRQwylJXpDkpKranNk9f6cNbzktyX5D+0lJTl50zQAAAGvFup13WXnd/aIkL9qu+bIk991B328keeIi6gIAAFjrxnpEBAAAACMQAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAlZN3YBAGvNhx74oLFLWBUe9OEPjV0CAEySkUAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmJBRQmBV7VNVZ1XVp6vq0qq6f1XdqareV1WfGX7uO/Stqnp1VW2uqk9U1ZFj1AwAALAWjDUS+Kok7+nuuye5Z5JLk5yc5P3dfXiS9w/7SXJsksOH14lJXrf4cgEAANaGhYfAqto7yQOTnJYk3f3N7v5ykuOSnD50Oz3J44bt45Kc0TMfTbJPVR240KIBAADWiDFGAg9LsjXJG6rq41V1alXtleSA7r5q6HN1kgOG7Q1Jrljy/iuHNgAAAG6lMULguiRHJnldd987yY357tTPJEl3d5K+NSetqhOralNVbdq6deuKFQsAALCWjBECr0xyZXefP+yflVkovGbbNM/h57XD8S1JDlny/oOHtu/R3ad098bu3rh+/fq5FQ8AALCaLTwEdvfVSa6oqrsNTcckuSTJOUlOGNpOSPKuYfucJE8bVgk9KskNS6aNAgAAcCusG+lzn53kTVW1R5LLkjw9s0B6ZlU9I8nnkhw/9D03ySOTbE7y9aEvAAAAP4RRQmB3X5Rk4w4OHbODvp3kmfOuCQAAYArGek4gAAAAIxACAQAAJkQIBAAAmBAhEAAAYELGWh10l3Cf3zxj7BJWhQv++GljlwAAAKwQI4EAAAATIgQCAABMiBAIAAAwIUIgAADAhAiBAAAAE7KsEFhV719OGwAAALu2H/iIiKq6bZI9k+xfVfsmqeHQHZNsmHNtAAAArLCdPSfwV5M8N8lBSS7Id0PgV5K8dn5lAQAAMA8/MAR296uSvKqqnt3dr1lQTQAAAMzJzkYCkyTd/ZqqekCSQ5e+p7vPmFNdAAAAzMGyQmBV/e8k/yHJRUm+NTR3EiEQAABgFVlWCEyyMckR3d3zLAYAAID5Wu5zAj+V5CfmWQgAAADzt9yRwP2TXFJV/5Tk37c1dvdj51IVAAAAc7HcEPjieRYBAADAYix3ddAPzbsQAAAA5m+5q4N+NbPVQJNkjyS7J7mxu+84r8IAAABYecsdCbzDtu2qqiTHJTlqXkUBAAAwH8tdHfQ7euadSX5h5csBAABgnpY7HfTxS3Zvk9lzA78xl4oAAACYm+WuDvqYJds3J7k8symhAAAArCLLvSfw6fMuBAAAgPlb1j2BVXVwVb2jqq4dXmdX1cHzLg4AAICVtdyFYd6Q5JwkBw2vdw9tAAAArCLLDYHru/sN3X3z8HpjkvVzrAsAAIA5WG4IvK6qnlJVuw2vpyS5bp6FAQAAsPKWGwL/e5Ljk1yd5KokT0jyy3OqCQAAgDlZ7iMiXpLkhO6+Pkmq6k5JXp5ZOAQAAGCVWO5I4M9uC4BJ0t1fSnLv+ZQEAADAvCw3BN6mqvbdtjOMBC53FBEAAIBdxHKD3J8k+ceqetuw/8QkL51PSaxln3/Jfxy7hFXhJ3/vk2OXAADAGrWsENjdZ1TVpiQPGZoe392XzK8sAAAA5mHZUzqH0Cf4AQAArGLLvScQAACANUAIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBCRguBVbVbVX28qv562D+sqs6vqs1V9daq2mNo/7Fhf/Nw/NCxagYAAFjtxhwJfE6SS5fs/2GSV3T3XZJcn+QZQ/szklw/tL9i6AcAAMAPYZQQWFUHJ3lUklOH/UrykCRnDV1OT/K4Yfu4YT/D8WOG/gAAANxKY40EvjLJbyX59rC/X5Ivd/fNw/6VSTYM2xuSXJEkw/Ebhv4AAADcSgsPgVX16CTXdvcFK3zeE6tqU1Vt2rp160qeGgAAYM0YYyTw6CSPrarLk7wls2mgr0qyT1WtG/ocnGTLsL0lySFJMhzfO8l125+0u0/p7o3dvXH9+vXz/QYAAACr1MJDYHf/dncf3N2HJnlSkvO6+5eSfCDJE4ZuJyR517B9zrCf4fh53d0LLBkAAGDN2JWeE/iCJCdV1ebM7vk7bWg/Lcl+Q/tJSU4eqT4AAIBVb93Ou8xPd38wyQeH7cuS3HcHfb6R5IkLLQwAAGCN2pVGAgEAAJgzIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYkHVjFwAArD4vfcoTxi5hVfidvzxr7BIAvo+RQAAAgAkRAgEAACZECAQAAJgQIRAAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQjwsHoBV77XPf/fYJawKz/qTx4xdAgC7ACOBAAAAEyIEAgAATIgQCAAAMCFCIAAAwIQIgQAAABMiBAIAAEyIEAgAADAhCw+BVXVIVX2gqi6pqour6jlD+52q6n1V9Znh575De1XVq6tqc1V9oqqOXHTNAAAAa8UYI4E3J3l+dx+R5Kgkz6yqI5KcnOT93X14kvcP+0lybJLDh9eJSV63+JIBAADWhoWHwO6+qrsvHLa/muTSJBuSHJfk9KHb6UkeN2wfl+SMnvlokn2q6sDFVg0AALA2jHpPYFUdmuTeSc5PckB3XzUcujrJAcP2hiRXLHnblUPb9uc6sao2VdWmrVu3zq9oAACAVWy0EFhVt09ydpLndvdXlh7r7k7St+Z83X1Kd2/s7o3r169fwUoBAADWjlFCYFXtnlkAfFN3v31ovmbbNM/h57VD+5Ykhyx5+8FDGwAAALfSGKuDVpLTklza3X+65NA5SU4Ytk9I8q4l7U8bVgk9KskNS6aNAgAAcCusG+Ezj07y1CSfrKqLhrYXJnlZkjOr6hlJPpfk+OHYuUkemWRzkq8nefpCqwUAAFhDFh4Cu/vvk9QtHD5mB/07yTPnWhQAAMBEjLo6KAAAAIslBAIAAEyIEAgAADAhQiAAAMCECIEAAAATIgQCAABMiBAIAAAwIUIgAADAhAiBAAAAEyIEAgAATIgQCAAAMCFCIAAAwIQIgQAAABMiBAIAAEyIEAgAADAhQiAAAMCECIEAAAATIgQCAABMiBAIAAAwIUIgAADAhAiBAAAAEyIEAgAATIgQCAAAMCFCIAAAwIQIgQAAABOybuwCgPk6+jVHj13CqvAPz/6HsUsAAFgII4EAAAATIgQCAABMiBAIAAAwIUIgAADAhAiBAAAAEyIEAgAATIgQCAAAMCFCIAAAwIQIgQAAABMiBAIAAEyIEAgAADAhQiAAAMCECIEAAAATsm7sAgAAAJLkzLfdd+wSVoXjn/hPP9L7jQQCAABMiBAIAAAwIUIgAADAhAiBAAAAE2JhGAAA2IEXv/jFY5ewKrhOq8+qGQmsqkdU1b9U1eaqOnnsegAAAFajVRECq2q3JH+W5NgkRyT5xao6YtyqAAAAVp9VEQKT3DfJ5u6+rLu/meQtSY4buSYAAIBVZ7WEwA1Jrliyf+XQBgAAwK1Q3T12DTtVVU9I8oju/pVh/6lJ7tfdz1rS58QkJw67d0vyLwsvdGXsn+SLYxcxMa754rnmi+eaL55rvniu+eK55ovnmi/ear3md+7u9Ts6sFpWB92S5JAl+wcPbd/R3ackOWWRRc1DVW3q7o1j1zElrvniueaL55ovnmu+eK754rnmi+eaL95avOarZTrox5IcXlWHVdUeSZ6U5JyRawIAAFh1VsVIYHffXFXPSvJ3SXZL8vruvnjksgAAAFadVRECk6S7z01y7th1LMCqn9K6Crnmi+eaL55rvniu+eK55ovnmi+ea754a+6ar4qFYQAAAFgZq+WeQAAAAFaAELiLqKrXV9W1VfWpsWuZgqo6pKo+UFWXVNXFVfWcsWta66rqtlX1T1X1z8M1/x9j1zQVVbVbVX28qv567FqmoKour6pPVtVFVbVp7HqmoKr2qaqzqurTVXVpVd1/7JrWsqq62/D3e9vrK1X13LHrWuuq6nnDv5+fqqo3V9Vtx65prauq5wzX++K19nfcdNBdRFU9MMnXkpzR3fcYu561rqoOTHJgd19YVXdIckGSx3X3JSOXtmZVVSXZq7u/VlW7J/n7JM/p7o+OXNqaV1UnJdmY5I7d/eix61nrquryJBu7ezU+U2pVqqrTk3yku08dVhHfs7u/PHJZk1BVu2X22K77dffnxq5nraqqDZn9u3lEd/9bVZ2Z5NzufuO4la1dVXWPJG9Jct8k30zyniS/1t2bRy1shRgJ3EV094eTfGnsOqaiu6/q7guH7a8muTTJhnGrWtt65mvD7u7Dy2+h5qyqDk7yqCSnjl0LzENV7Z3kgUlOS5Lu/qYAuFDHJPmsALgQ65LcrqrWJdkzyRdGrmet++kk53f317v75iQfSvL4kWtaMUIgk1dVhya5d5LzRy5lzRumJV6U5Nok7+tu13z+Xpnkt5J8e+Q6pqSTvLeqLqiqE8cuZgIOS7I1yRuGac+nVtVeYxc1IU9K8uaxi1jruntLkpcn+XySq5Lc0N3vHbeqNe9TSX6+qvarqj2TPDLJISPXtGKEQCatqm6f5Owkz+3ur4xdz1rX3d/q7nslOTjJfYepFsxJVT06ybXdfcHYtUzMz3X3kUmOTfLMYbo/87MuyZFJXtfd905yY5KTxy1pGoapt49N8raxa1nrqmrfJMdl9kuPg5LsVVVPGbeqta27L03yh0nem9lU0IuSfGvMmlaSEMhkDfelnZ3kTd399rHrmZJhqtYHkjxi5FLWuqOTPHa4R+0tSR5SVX85bklr3/Ab+3T3tUnekdn9JMzPlUmuXDKz4KzMQiHzd2ySC7v7mrELmYCHJvnX7t7a3TcleXuSB4xc05rX3ad19326+4FJrk/y/8auaaUIgUzSsEjJaUku7e4/HbueKaiq9VW1z7B9uyQPS/LpUYta47r7t7v74O4+NLMpW+d1t98cz1FV7TUsNpVhSuLDM5tSxJx099VJrqiquw1NxySxyNdi/GJMBV2Uzyc5qqr2HP4Pc0xm6xkwR1X148PPn8zsfsC/GreilbNu7AKYqao3J3lwkv2r6sokL+ru08atak07OslTk3xyuEctSV7Y3eeOV9Kad2CS04eV5G6T5Mzu9sgC1poDkrxj9n+0rEvyV939nnFLmoRnJ3nTMD3xsiRPH7meNW/4JcfDkvzq2LVMQXefX1VnJbkwyc1JPp7klHGrmoSzq2q/JDcleeZaWnTKIyIAAAAmxHRQAACACRECAQAAJkQIBAAAmBAhEAAAYEKEQAAAgAkRAgFgGarqJ6rqLVX12aq6oKrOraq7VpXnAAKwqnhOIADsxPBw5nckOb27nzS03TOz5wICwKpiJBAAdu4/J7mpu//Xtobu/uckV2zbr6pDq+ojVXXh8HrA0H5gVX24qi6qqk9V1c9X1W5V9cZh/5NV9bzFfyUApspIIADs3D2SXLCTPtcmeVh3f6OqDk/y5iQbkzw5yd9190urarckeya5V5IN3X2PJKmqfeZVOABsTwgEgJWxe5LXVtW9knwryV2H9o8leX1V7Z7knd19UVVdluSnquo1Sf4myXvHKBiAaTIdFAB27uIk99lJn+cluSbJPTMbAdwjSbr7w0kemGRLkjdW1dO6+/qh3weT/FqSU+dTNgB8PyEQAHbuvCQ/VlUnbmuoqp9NcsiSPnsnuaq7v53kqUl2G/rdOck13f0XmYW9I6tq/yS36e6zk/xukiMX8zUAwHRQANip7u6q+i9JXllVL0jyjSSXJ3nukm5/nuTsqnpakvckuXFof3CS36yqm5J8LcnTkmxI8oaq2vbL2N+e93cAgG2qu8euAQAAgAUxHRQAAGBChEAAAIAJEQIBAAAmRAgEAACYECEQAABgQoRAAACACRECAQAAJkQIBAAAmJD/DwTZnzDiG3ukAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 1080x360 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure(figsize=(15,5))\n",
    "sns.countplot(x = train_variant[\"Class\"])\n",
    "plt.title(\"Class distribution\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA3sAAAHwCAYAAAAfJXbRAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/MnkTPAAAACXBIWXMAAAsTAAALEwEAmpwYAAAjSElEQVR4nO3de7ztdV3n8fcnwPsNhxMikMeULKnEPJqXMs2Z8TIVWkr6cJRxVGpCs8lJaXSUUiceo6mTlQ6NJjyGvOUlbRjvd8fUAyEK6EgqAnI5miJegsDP/LF+G5fbffbZ53DW3vt8eT4fj/3Ya/1u67sWi8V68bvs6u4AAAAwlh/a6AEAAACw94k9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9ANZdVW2tqq6q/af7/6eqjt1L2/75qvrs3P0vVtW/3BvbnrZ3TlU9YG9tDwAWRewBDKyqHl1VH6uqb1XV5dPt36qq2uixzevuh3b3KbtabgrEO+9iWx/q7rvsjXFV1aur6vnLtn9kd79/b2x/b6iqp1TV9qq6qqpevcL8B1XVZ6rq21X1vqq6w9y8G1fVq6rqG1V1aVX97roOfheq6t9V1Yc3ehwA+yqxBzCoqnp6kv+e5IVJbpfk4CS/meR+SW60gUNbmKU9hfuiqjp4D1f9cpLnJ3nVCts8KMmbkvyXJLdNsj3J6+YWOTHJEUnukOSBSZ5RVQ/Zw3EAsMmIPYABVdWtk/xhkt/q7r/u7it75u+7+7HdfdW03I2r6kVV9aWquqyqXlFVN53mPaCqLqqqp097BS+pqifMPcZO111hPPtNy36lqj6f5N8sm//+qnrSdPvOVfWBqrpiWv510/QPTot/sqq+WVW/PjfGZ1bVpUn+cmnasiHcs6rOraqvVdVfVtVNpm3+wJ6jpb2HVXVcksdmFkDfrKq3TfOvOyx0eg1eWlVfnn5eWlU3Xsvrt4ITpzH+XlXdbpXlvk93v6m735LkqyvM/tUk53T3G7r7nzKLu7tV1Y9P849N8rzu/lp3n5fkL5L8u509VlU9uarOq6orp7H+zDT9J6Z/hl+fDnP9lbl1rvtnO93/vtd8er1/s6o+N63/ZzXzE0lekeQ+0+v/9bW+JgDMiD2AMd0nyY2T/M0uljspyY8lOSrJnZMcmuQ5c/Nvl+TW0/QnJvmzqjpwjevOe3KSX0py9yTbkjxylTE9L8k7kxyY5LAkL0uS7r7/NP9u3X2L7l7aQ3W7zPZa3SHJcTvZ5mOTPDjJnaYxP3uVx8/0eCcnOS3Jf5se75dXWOxZSe6d2WtwtyT3Wrbt1V6/5Y5P8ttJfjrJZ6vqrVX1iKo6YFdjXcWRST4595y+leQfkhw5jeOQ+fnT7SNX2lBVPSqzWHx8klsl+ZUkX53G97bM/pn9cJKnJjmtqnbnUNpfSnLPzJ77MUkePMXnbyb56PT632Y3tgdAxB7AqA5K8pXuvmZpQlX932nPyXeq6v5VVZnF0X/s7n/s7iuT/Nckj57bzj8n+cPu/ufuPj3JN5PcZY3rzjsmyUu7+8Lu/sckf7TK2P85s3C7fXf/U3fv6pyt7yZ5bndf1d3f2ckyfzr32C9I8phdbHOtHpvZ63N5d+9I8gdJHjc3f8XXb6UNdfd3u/vd3f24zCL3LUn+Y5KLq+p5ezi+WyS5Ytm0K5LccpqXZfOX5q3kSZmF7yemvcTnd/cFmcXuLZKc1N1Xd/d7k/xtdu81Pqm7v97dX0ryvsziGYDrSewBjOmrSQ6aP4etu+877R35amaf/1uS3CzJGVMEfj3J26fp121nPhiTfDuzL/ZrWXfe7ZNcOHf/glXG/owkleTj0yGB/34Xz3XHdIjiapY/9u13sfxa3T7f/1yWb3tnr9+qpng+O8lZSQ7ITgJxDb6Z2V64ebdKcuU0L8vmL81byeGZ7RVc7vZJLuzu785NuyCzvZlrdenc7TW9RgDsmtgDGNNHk1yV5OhVlvlKku8kObK7bzP93Lq71/JFe3fXvSSzWFjyIzvbcHdf2t1P7u7bJ/mNJH9eq1+Bs9cw3uWP/eXp9rcyi9YkyQrnyu1q21/ObC/kStvebVV1WFWdUFXnJnltkh2ZHbZ6zB5u8pzMDi9d2v7NMzuU9Zzu/lpm/1zuNrf83aZ1VnLhtO5yX05yeFXNf6f4kSQXT7e/7zXO7NDWtVrLP1sAdkLsAQyou7+e2SGFf15Vj6yqW1bVD1XVUUluPi3z3cwuyPGSqvrhJKmqQ6vqwWvY/u6u+/okvz3FzIFJTtjZtqvqUVV12HT3a5l94V/aa3RZkh/d1fhWcPz02LfN7Dy7pfP9PpnZ+WtHTRdtOXHZert6vNckeXZVbanZlS+fk+R/7cH4UlUnZhZad8nsXLUjuvt506GNq623/zT2/ZLsV1U3mduj++YkP1lVvzYt85wkZ3f3Z6b5p07jP3C6aMuTk7x6Jw/1P5P8p6q6x3QBlTvX7M84fCyzvXHPqKoDavY3CH85s1hNZnsnf7WqbjZF+xN342W5LMlhVTXk1WMBFk3sAQyqu/9bkt/N7LDIy6af/5HkmUn+77TYM5Ocn+TvquobSd6dtR8yuDvr/kWSd2QWV2dm9ucAduaeST5WVd9M8tYkT+vuz0/zTkxyynTo6O7s7fqrzC4g8vnMDkV8fpJ09//L7Kql707yuSTLzw98ZZK7To/3lhW2+/zM/pzB2Uk+NT2356+w3Fq8JbPzFJ/Q3R/s7rXu1Xp2ZntZT0jyb6fbz06S6TzCX8vsPMWvJfnZfP95lc/N7PW4IMkHkrywu9++0oN09xum7fxVZod6viXJbbv76szi7qGZ7fH98ySPnwvKlyS5OrP33ymZXfRmrd6bWQBfWlVf2Y31AEhSa/9vCQAAAPsKe/YAAAAGJPYAAAAGJPYAAAAGJPYAAAAGJPYAAAAGtP+uF9m8DjrooN66detGDwMAAGBDnHHGGV/p7i0rzdunY2/r1q3Zvn37Rg8DAABgQ1TVBTub5zBOAACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAYk9AACAAe2/0QPYaPf4vVM3eghsQme88PEbPQQAALhe7NkDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAYkNgDAAAY0MJir6oOr6r3VdW5VXVOVT1tmn5iVV1cVWdNPw+bW+f3q+r8qvpsVT14UWMDAAAY3f4L3PY1SZ7e3WdW1S2TnFFV75rmvaS7XzS/cFXdNcmjkxyZ5PZJ3l1VP9bd1y5wjAAAAENa2J697r6ku8+cbl+Z5Lwkh66yytFJXtvdV3X3F5Kcn+ReixofAADAyNblnL2q2prk7kk+Nk16SlWdXVWvqqoDp2mHJrlwbrWLsnocAgAAsBMLj72qukWSNyb5ne7+RpKXJ7lTkqOSXJLkj3dze8dV1faq2r5jx469PVwAAIAhLDT2quqAzELvtO5+U5J092XdfW13fzfJX+R7h2penOTwudUPm6Z9n+4+ubu3dfe2LVu2LHL4AAAA+6xFXo2zkrwyyXnd/eK56YfMLfaIJJ+ebr81yaOr6sZVdcckRyT5+KLGBwAAMLJFXo3zfkkel+RTVXXWNO0/J3lMVR2VpJN8MclvJEl3n1NVr09ybmZX8jzelTgBAAD2zMJir7s/nKRWmHX6Kuu8IMkLFjUmAACAG4p1uRonAAAA60vsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADEjsAQAADGhhsVdVh1fV+6rq3Ko6p6qeNk2/bVW9q6o+N/0+cJpeVfUnVXV+VZ1dVT+zqLEBAACMbpF79q5J8vTuvmuSeyc5vqrumuSEJO/p7iOSvGe6nyQPTXLE9HNckpcvcGwAAABDW1jsdfcl3X3mdPvKJOclOTTJ0UlOmRY7JcnDp9tHJzm1Z/4uyW2q6pBFjQ8AAGBk63LOXlVtTXL3JB9LcnB3XzLNujTJwdPtQ5NcOLfaRdM0AAAAdtPCY6+qbpHkjUl+p7u/MT+vuztJ7+b2jquq7VW1fceOHXtxpAAAAONYaOxV1QGZhd5p3f2mafJlS4dnTr8vn6ZfnOTwudUPm6Z9n+4+ubu3dfe2LVu2LG7wAAAA+7BFXo2zkrwyyXnd/eK5WW9Ncux0+9gkfzM3/fHTVTnvneSKucM9AQAA2A37L3Db90vyuCSfqqqzpmn/OclJSV5fVU9MckGSY6Z5pyd5WJLzk3w7yRMWODYAAIChLSz2uvvDSWonsx+0wvKd5PhFjQcAAOCGZF2uxgkAAMD6EnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAAD2n+jBwCs7Et/+FMbPQQ2oR95zqc2eggAwD7Cnj0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABiT0AAIABLSz2qupVVXV5VX16btqJVXVxVZ01/Txsbt7vV9X5VfXZqnrwosYFAABwQ7DIPXuvTvKQFaa/pLuPmn5OT5KqumuSRyc5clrnz6tqvwWODQAAYGgLi73u/mCSf1zj4kcneW13X9XdX0hyfpJ7LWpsAAAAo9uIc/aeUlVnT4d5HjhNOzTJhXPLXDRN+wFVdVxVba+q7Tt27Fj0WAEAAPZJ6x17L09ypyRHJbkkyR/v7ga6++Tu3tbd27Zs2bKXhwcAADCGdY297r6su6/t7u8m+Yt871DNi5McPrfoYdM0AAAA9sBux15VHVhVP70nD1ZVh8zdfUSSpSt1vjXJo6vqxlV1xyRHJPn4njwGAAAAyf5rWaiq3p/kV6blz0hyeVV9pLt/d5V1XpPkAUkOqqqLkjw3yQOq6qgkneSLSX4jSbr7nKp6fZJzk1yT5PjuvnbPnhIAAABrir0kt+7ub1TVk5Kc2t3PraqzV1uhux+zwuRXrrL8C5K8YI3jAQAAYBVrPYxz/+kQzGOS/O0CxwMAAMBesNbY+4Mk70hyfnd/oqp+NMnnFjcsAAAAro+1HsZ5SXdfd1GW7v58Vb14QWMCAADgelrrnr2XrXEaAAAAm8Cqe/aq6j5J7ptkS1XNX3nzVkn2W+TAAAAA2HO7OozzRkluMS13y7np30jyyEUNCgAAgOtn1djr7g8k+UBVvbq7L1inMQEAAHA9rfUCLTeuqpOTbJ1fp7t/cRGDAgAA4PpZa+y9IckrkvzPJNcubjgAAADsDWuNvWu6++ULHQkA+4z7vex+Gz0ENqGPPPUjGz0EAOas9U8vvK2qfquqDqmq2y79LHRkAAAA7LG17tk7dvr9e3PTOsmP7t3hAAAAsDesKfa6+46LHggAAAB7z5pir6oev9L07j517w4HAACAvWGth3Hec+72TZI8KMmZScQeAADAJrTWwzifOn+/qm6T5LWLGBAAAADX31qvxrnct5I4jw8AAGCTWus5e2/L7OqbSbJfkp9I8vpFDQoAAIDrZ63n7L1o7vY1SS7o7osWMB4AAAD2gjUdxtndH0jymSS3THJgkqsXOSgAAACunzXFXlUdk+TjSR6V5JgkH6uqRy5yYAAAAOy5tR7G+awk9+zuy5OkqrYkeXeSv17UwAAAANhza70a5w8thd7kq7uxLgAAAOtsrXv23l5V70jymun+ryc5fTFDAgAA4PpaNfaq6s5JDu7u36uqX03yc9OsjyY5bdGDAwAAYM/sas/eS5P8fpJ095uSvClJquqnpnm/vMCxAQAAsId2dd7dwd39qeUTp2lbFzIiAAAArrddxd5tVpl30704DgAAAPaiXcXe9qp68vKJVfWkJGcsZkgAAABcX7s6Z+93kry5qh6b78XdtiQ3SvKIBY4LAACA62HV2Ovuy5Lct6oemOQnp8n/u7vfu/CRAQAAsMfW9Hf2uvt9Sd634LEAAACwl6z1j6oDAGx6H7j/L2z0ENiEfuGDH9joIcCG2NUFWgAAANgH2bMHAADr4E+f/raNHgKbzFP++JcXun179gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAYk9gAAAAa0sNirqldV1eVV9em5abetqndV1eem3wdO06uq/qSqzq+qs6vqZxY1LgAAgBuCRe7Ze3WShyybdkKS93T3EUneM91PkocmOWL6OS7Jyxc4LgAAgOEtLPa6+4NJ/nHZ5KOTnDLdPiXJw+emn9ozf5fkNlV1yKLGBgAAMLr1Pmfv4O6+ZLp9aZKDp9uHJrlwbrmLpmkAAADsgQ27QEt3d5Le3fWq6riq2l5V23fs2LGAkQEAAOz71jv2Lls6PHP6ffk0/eIkh88td9g07Qd098ndva27t23ZsmWhgwUAANhXrXfsvTXJsdPtY5P8zdz0x09X5bx3kivmDvcEAABgN+2/qA1X1WuSPCDJQVV1UZLnJjkpyeur6olJLkhyzLT46UkeluT8JN9O8oRFjQsAAOCGYGGx192P2cmsB62wbCc5flFjAQAAuKHZsAu0AAAAsDhiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYEBiDwAAYED7b8SDVtUXk1yZ5Nok13T3tqq6bZLXJdma5ItJjunur23E+AAAAPZ1G7ln74HdfVR3b5vun5DkPd19RJL3TPcBAADYA5vpMM6jk5wy3T4lycM3bigAAAD7to2KvU7yzqo6o6qOm6Yd3N2XTLcvTXLwSitW1XFVtb2qtu/YsWM9xgoAALDP2ZBz9pL8XHdfXFU/nORdVfWZ+Znd3VXVK63Y3ScnOTlJtm3btuIyAAAAN3Qbsmevuy+efl+e5M1J7pXksqo6JEmm35dvxNgAAABGsO6xV1U3r6pbLt1O8q+TfDrJW5McOy12bJK/We+xAQAAjGIjDuM8OMmbq2rp8f+qu99eVZ9I8vqqemKSC5IcswFjAwAAGMK6x153fz7J3VaY/tUkD1rv8QAAAIxoM/3pBQAAAPYSsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADAgsQcAADCgTRd7VfWQqvpsVZ1fVSds9HgAAAD2RZsq9qpqvyR/luShSe6a5DFVddeNHRUAAMC+Z1PFXpJ7JTm/uz/f3VcneW2Sozd4TAAAAPuczRZ7hya5cO7+RdM0AAAAdkN190aP4TpV9cgkD+nuJ033H5fkZ7v7KXPLHJfkuOnuXZJ8dt0HOq6DknxlowcBK/DeZDPz/mSz8t5kM/P+3Hvu0N1bVpqx/3qPZBcuTnL43P3DpmnX6e6Tk5y8noO6oaiq7d29baPHAct5b7KZeX+yWXlvspl5f66PzXYY5yeSHFFVd6yqGyV5dJK3bvCYAAAA9jmbas9ed19TVU9J8o4k+yV5VXefs8HDAgAA2OdsqthLku4+PcnpGz2OGyiHx7JZeW+ymXl/sll5b7KZeX+ug011gRYAAAD2js12zh4AAAB7gdgbSFVdW1VnVdUnq+rMqrrvNH1rVX1nmnduVZ1aVQdM8w6oqpOq6nPTOh+tqofObfOoquqqesiyx3pVVV1eVZ9e32fJCKrqX0zvx7Oq6tKqunjufk+/P11Vb6iqm03rfLGqPjXN2z63redV1dnT9HdW1e037pmxr5v7HF36OWGavn9V/dfps3Jp3rNWWW9rVT2gqq6Y7n+mql60cc+MEcy9z5Y+Hw9d5bP0Rqu8n9+/7HN0W1W9f8OeGPuk9freWVWHV9X7pm2dU1VPW/9nu+/adOfscb18p7uPSpKqenCSP0ryC9O8f+juo6pqvyTvSnJMktOSPC/JIUl+sruvqqqD59ZJksck+fD0++1z01+d5E+TnLqwZ8OwuvurSY5Kkqo6Mck3u/tF0/1vzr2PT0vym0lePK36wO5e/jd5Xtjd/2Va/reTPGdaB/bEdZ+jyzw/ye2S/FR3/1NV3TLJ01dbr6q2JvlQd/9SVd00yd9X1Zu7+yOLGTo3AN9Z9vn463P3T8zcZ+k0bWfv5yT54ap6aHf/n4WOmJGt1/fOa5I8vbvPnD57z6iqd3X3uYt9emMQe+O6VZKvLZ/Y3ddW1ceTHDrtMXlykjt291XT/MuSvD5JqqqSPCrJv0ryoaq6SXf/07TcB6cvMrBIH0ry06st0N3fmLt78yRORGavmvus3Dr3GXhlkhPXuo3u/k5VnZXk0EWMkRukXX4+7sILkzwridhjb1jY987uviTJJdPyV1bVeZl9loq9NRB7Y7np9GXiJpn9X5NfXL5AVd0kyc8meVqSOyf50rIvy/Pum+QL3f0P0+Ed/ybJGxcwbvgBVbV/kofme/9nr5O8s6o6yf/o7pPnln1BkscnuSLJA9d7rAxl6XN0yR8lOS+zz8or17jeF7r7EfMzq+rAJEck+eBeHCs3UCt8Pu7MD7yfu/t10+2PJnlEVT0wyWrvbdiZdf/eOe1ouHuSj+2dpzA+5+yN5TvdfVR3/3iShyQ5dfq/JElyp+lfyMuSXNLdZ69he49J8trp9mun+7BoS//x2J7kS0leOU3/ue7+mcy+4BxfVfdfWqG7n9Xdh2d2iMhT1nm8jGXpc3Tp53XLF6iqJ0znolxYVYevsN586P18VX0yycVJ3tHdl67Hk2BYO/t83JldvZ+fn+TZCxgnNwzr+r2zqm6RWfz9zirByDJib1Dd/dEkByXZMk36h+m46jsluUdV/UqS85P8SFXdavn60zHWv5bkOVX1xSQvS/KQ6VhpWKT5LydP7e6rk6S7L55+X57kzUnutcK6p2X2voW9aemz8pZJ0t1/OX2eXpFkv12s+6HuvluSI5M8saqOWuRAGd6Kn497qrvfm+SmSe69d4bHDdWiv3dOF3h5Y5LTuvtNC346QxF7g6qqH8/sS8hX56dPF7c4Icnvd/e3M/u/gv+9qm40rbelqh6V5EFJzu7uw7t7a3ffIbN/yb7v0CRYD1V187kP/Jsn+ddJPj3dP2Ju0aOTfGb9R8jI5j4r/3Q6JGnpi8mNdmMbX0hyUpJnLmSQsOeen+QZGz0I9m2L/N457S18ZZLzuvvFYbeIvbHcdDq06Kwkr0tybHdfu8Jyb0lys6r6+cwO39iR5Nya/RmFv03yjcx2nb952XpvnKanql6T2fH+d6mqi6rqiQt4PrDk4CQfng6H+3iS/93dS+eqnFSzy5CfnVkEuiQz18d1n6PTz0nT9GdldoGAT1fV32d2cYxTknx5N7b9iiT3d3Er1tHO3s/X6e7TM/seALtrvb533i/J45L84tx7+WGLeEIjqm4XrgMAABiNPXsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAAAADEnsAMKeqDq6qv6qqz1fVGVX10aryN0YB2OeIPQCYTH+89y1JPtjdP9rd90jy6CSHbejAAGAPiD0A+J5fTHJ1d79iaUJ3X9DdL6uq/arqhVX1iao6u6p+I0mq6gFV9f6q+uuq+kxVnTZFY6rqHlX1gWkP4Tuq6pANel4A3ADtv9EDAIBN5MgkZ+5k3hOTXNHd96yqGyf5SFW9c5p392ndLyf5SJL7VdXHkrwsydHdvaOqfj3JC5L8+4U+AwCYiD0A2Imq+rMkP5fk6iQXJPnpqnrkNPvWSY6Y5n28uy+a1jkrydYkX0/yk0neNe3o2y/JJes3egBu6MQeAHzPOUl+belOdx9fVQcl2Z7kS0me2t3vmF+hqh6Q5Kq5Sddm9t/XSnJOd99nwWMGgBU5Zw8Avue9SW5SVf9hbtrNpt/vSPIfquqAJKmqH6uqm6+yrc8m2VJV95mWP6CqjlzEoAFgJfbsAcCku7uqHp7kJVX1jCQ7knwryTOTvCGzwzPPnC7AsiPJw1fZ1tXTIZ9/UlW3zuy/uS/NbO8hACxcdfdGjwEAAIC9zGGcAAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAAxJ7AAAAA/r/+sCQoIjN2bkAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 1080x576 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure(figsize=(15,8))\n",
    "\n",
    "sns.barplot( x = list(gene_count.keys()), y = list(gene_count.values()))\n",
    "plt.title(\"Gene distribution > 100 count\")\n",
    "plt.ylabel(\"Counts\")\n",
    "plt.xlabel(\"Gene\")\n",
    "\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAA3sAAAHwCAYAAAAfJXbRAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjQuMywgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/MnkTPAAAACXBIWXMAAAsTAAALEwEAmpwYAAAwcUlEQVR4nO3de5xdZX0v/s+XJIgiApWLkJkk4I1LLBFQalVEPFJsvSGIIu3hppxetHj0HC8/j1WP9mhrFfByjoceRagW0FaPqIgiaEEFOeESi1wsRZJMRIhUULkIhOf3x+zomCaZmczs2ZOV9/v12q/Ze61nPeu796zMzCfPs9aq1loAAADolq0GXQAAAADTT9gDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhD4BZp6oWVVWrqrm911+pquOmqe9nV9VNY17fWlX/YTr67vX3/ao6ZLr664eqOr6qvrUJ232yqt7Tj5oAmH7CHsAWrKpeWVXfrap7quqO3vM/raoadG1jtdZe0Fo7a7x2vYD4hHH6uqy19uTpqGt94ae1tm9r7ZvT0T8ATIWwB7CFqqo3Jjk9yfuTPC7Jrkn+OMkzk2w9wNL6Zu1IIQBsCYQ9gC1QVW2f5L8n+dPW2j+01n7eRl3TWju2tfbLXrtHVNXfVNWKqrq9qj5WVY/srTukqkaq6o29UcHbquqEMfvY4LbrqWdOr+1PquqWJH+wzvpvVtWre8+fUFX/VFV399qf11t+aa/5sqr6RVW9YkyNb66qHyc5c+2ydUp4WlVdX1U/raozq2qbXp//brrj2tHDqjo5ybFJ3tTb3xd76381LbT3GZxWVT/qPU6rqkdM5PNbz2d0QlXdUFU/r6pbquo/jVk33vfisVV1flX9rKquTPL4De2n1/5ZVfWdqrqrqlZW1fHrabNjVX2pqlb3PrcvVdXQmPXH9+r8eVX9sKqO3dj3D4DpJ+wBbJmekeQRSb4wTrv3JXlSkiVJnpBkfpK/GLP+cUm27y0/KclHq2rHCW471muSvDDJU5McmOSojdT07iRfS7JjkqEkH06S1trBvfX7tdYe3VpbGyIel+S3kixMcvIG+jw2ye9lNAQ9Kcl/28j+09vfGUk+neSve/t70XqavS3J72T0M9gvydPX6Xtjn9+67sjoZ/SYJCckObWq9p9gXx9Ncn+S3ZKc2HusV1UtTPKVjH6uO/dqv3Y9TbdKcmZGP9cFSe5L8pFeH9sm+VCSF7TWtkvyu2P6WO/3D4DpJ+wBbJl2SvKT1tpDaxeMGcm5r6oO7p23d3KS/9xa+7fW2s+T/I8krxzTz4NJ/ntr7cHW2gVJfpHkyRPcdqyjk5zWWlvZWvu3JO/dSO0PZjRg7N5au7+1Nt6FRh5O8o7W2i9ba/dtoM1Hxuz7L5McM06fE3VsRj+fO1prq5O8K8kfjVm/3s9vfR211r7cWvvX3gjsP2U0MD17vL6qak6SI5P8RWvtntbadUk2dv7jq5J8vbV2Tq+vO1tr166nnjtba//YWru39/39yyTPGdPk4SSLq+qRrbXbWmvfH1PnZL5/AGwiYQ9gy3Rnkp3GnsPWWvvd1toOvXVbZXRU51FJruqFwLuSXNhb/qt+xgbGJPcmefQEtx1r9yQrx7xevpHa35SkklxZo1e+3OAoVc/q1tr947RZd9+7j9N+onbPb76Xdfve0Of371TVC6rqiqr6t97n+fsZDe3j9bVzkrmZ+Oc7nORfN7J+bT2Pqqr/XVXLq+pnSS5NskNVzWmt3ZPkFRk9B/S2qvpyVe3V23Sy3z8ANpGwB7BlujzJL5O8ZCNtfpLRqXn7ttZ26D22b62tN4xMcdvbMhoy1lqwoY5baz9urb2mtbZ7kv+U5H/Wxq/A2SZQ77r7/lHv+T0ZDa1Jkqp63CT7/lFGR7HW1/eE9c7z+8ckf5Nk114ovyCjoWk8q5M8lAl+vhkNhRs9p6/njRkdhTyotfaYJGun0VaStNa+2lp7fkanjt6Y5G97yyf7/QNgEwl7AFug1tpdGZ1S+D+r6qiq2q6qtqqqJUm27bV5OKN/oJ9aVbskSVXNr6rfm0D/k932M0n+vKqGeueZvWVDfVfVy8dcCOSnGQ1cD/de355kz/HqW48/6+37tzJ6nt3a8/2WJdm3qpb0LtryznW2G29/5yT5b1W1c1XtlNFzFj+1CfVtndFzLFcneaiqXpDksIls2Fpbk+RzSd7ZG43bJ8nG7ln46ST/oaqOrqq5vYu7LFlPu+0yGujv6n1u71i7oqp2raqX9M7d+2VGp5Q+3Fu3se8fANNI2APYQrXW/jrJGzI6re723uN/J3lzku/0mr05yc1JruhN1ft6NnBO2XpMZtu/TfLVjIarqzMaTjbkaUm+W1W/SHJ+klNaa7f01r0zyVm9qaNHT7DOJPn7jJ4Dd0tGpzC+J0laaz/I6FVLv57kX5Kse37Zx5Ps09vf/11Pv+9JsjTJ95L8c++9Tfqm5L1z4v48o6H4pxk9r+78SXTx2oxO6fxxkk9m9MIqG9rXioxOEX1jkn/L6IVV9ltP09OSPDKjo7hXZHSa7lpbZfTY+lGvj+ck+ZPeuo19/wCYRtXaRGa3AAAAsDkxsgcAANBBwh4AAEAHCXsAAAAdJOwBAAB0UN/CXlV9oqruqKrrxix7f1XdWFXfq6rPV9UOY9a9tapurqqbJnJZbwAAADasb1fjrKqDM3pfnbNba4t7yw5Lcklr7aGq+qskaa29uXfPn3OSPD3J7hm9xPWTevcG2qCddtqpLVq0qC/1AwAAzHZXXXXVT1prO69v3dx+7bS1dmlVLVpn2dfGvLwiyVG95y9Jcm5r7ZdJflhVN2c0+F2+sX0sWrQoS5cunb6iAQAANiNVtXxD6wZ5zt6JSb7Sez4/ycox60Z6ywAAANgEAwl7VfW2JA8l+fQmbHtyVS2tqqWrV6+e/uIAAAA6YMbDXlUdn+SFSY5tvz5hcFWS4THNhnrL/p3W2hmttQNbawfuvPN6p6YCAABs8fp2zt76VNXhSd6U5DmttXvHrDo/yd9X1QczeoGWJya5ciZrAwAAtkwPPvhgRkZGcv/99w+6lA3aZpttMjQ0lHnz5k14m76Fvao6J8khSXaqqpEk70jy1iSPSHJRVSXJFa21P26tfb+qPpPk+oxO7/yz8a7ECQAAMB1GRkay3XbbZdGiRenllFmltZY777wzIyMj2WOPPSa8XT+vxnnMehZ/fCPt/zLJX/arHgAAgPW5//77Z23QS5KqymMf+9hM9polg7waJwAAwKwwW4PeWptSn7AHAAAwjh//+Md55Stfmcc//vE54IAD8vu///v5wQ9+kMWLFw+6tA2a0Qu0AAAAbG5aazniiCNy3HHH5dxzz02SLFu2LLfffvuAK9s4I3sAAAAb8Y1vfCPz5s3LH//xH/9q2X777Zfh4V/fPe7WW2/Ns5/97Oy///7Zf//9853vfCdJctttt+Xggw/OkiVLsnjx4lx22WVZs2ZNjj/++CxevDhPecpTcuqpp/albiN7AAAAG3HdddflgAMO2GibXXbZJRdddFG22Wab/Mu//EuOOeaYLF26NH//93+f3/u938vb3va2rFmzJvfee2+uvfbarFq1Ktddd12S5K677upL3cIeAADAFD344IN57Wtfm2uvvTZz5szJD37wgyTJ0572tJx44ol58MEH89KXvjRLlizJnnvumVtuuSWve93r8gd/8Ac57LDD+lKTaZwAAAAbse++++aqq67aaJtTTz01u+66a5YtW5alS5fmgQceSJIcfPDBufTSSzN//vwcf/zxOfvss7Pjjjtm2bJlOeSQQ/Kxj30sr371q/tSt7AHAACwEYceemh++ctf5owzzvjVsu9973tZuXLlr17ffffd2W233bLVVlvl7/7u77JmzZokyfLly7PrrrvmNa95TV796lfn6quvzk9+8pM8/PDDOfLII/Oe97wnV199dV/qNo0TAABgI6oqn//85/P6178+f/VXf5VtttkmixYtymmnnfarNn/6p3+aI488MmeffXYOP/zwbLvttkmSb37zm3n/+9+fefPm5dGPfnTOPvvsrFq1KieccEIefvjhJMl73/ve/tTdWutLxzPhwAMPbEuXLh10GQAAwGbshhtuyN577z3oMsa1vjqr6qrW2oHra28aJwAAQAcJewAAAB0k7AEAAHSQsAfAFmfh8IJU1YQfC4cXDLpkAJg0V+MEYIuzYmRllp35xQm33++EF/WxGgDoDyN7AAAAHSTsAQAADNiJJ56YXXbZJYsXL562PoU9AACAMRYtXDipc7vHeyxauHDcfR5//PG58MILp/V9OGcPAABgjOUrVuTum2+ctv62f8Je47Y5+OCDc+utt07bPhMjewAAAJ0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAADdswxx+QZz3hGbrrppgwNDeXjH//4lPt0NU4AAIAxFi5YMKEraE6mv/Gcc84507a/tYQ9AACAMW5dvnzQJUwL0zgBAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAAGLCVK1fmuc99bvbZZ5/su+++Of3006fcp7AHAAAwxqKFC1NV0/ZYtHDhuPucO3duPvCBD+T666/PFVdckY9+9KO5/vrrp/Q+3GcPAABgjOUrVuSOy787bf3t8oyDxm2z2267ZbfddkuSbLfddtl7772zatWq7LPPPpu8XyN7AAAAs8itt96aa665JgcdNH5I3BhhDwAAYJb4xS9+kSOPPDKnnXZaHvOYx0ypL2EPAABgFnjwwQdz5JFH5thjj83LXvayKfcn7AEAAAxYay0nnXRS9t5777zhDW+Ylj6FPQAAgAH79re/nb/7u7/LJZdckiVLlmTJkiW54IILptSnq3ECAACMsXDBggldQXMy/Y3nWc96Vlpr07bPRNgDAAD4DbcuXz7oEqaFaZwAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAADNj999+fpz/96dlvv/2y77775h3veMeU+xT2AAAAxli4YEGqatoeE7nP3iMe8YhccsklWbZsWa699tpceOGFueKKK6b0PtxnDwAAYIwVK1fmlvMvnrb+9nzx88ZtU1V59KMfnSR58MEH8+CDD6aqprRfI3sAAACzwJo1a7JkyZLssssuef7zn5+DDjpoSv0JewAAALPAnDlzcu2112ZkZCRXXnllrrvuuin1J+wBAADMIjvssEOe+9zn5sILL5xSP8IeAADAgK1evTp33XVXkuS+++7LRRddlL322mtKfbpACwAAwIDddtttOe6447JmzZo8/PDDOfroo/PCF75wSn0KewAAAGMsGB6e0BU0J9PfeH77t38711xzzbTtMxH2AAAAfsPyFSsGXcK0cM4eAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAALPAmjVr8tSnPnXK99dbS9gDAAAYY+HwglTVtD0WDi+Y0H5PP/307L333tP2Pvp2n72q+kSSFya5o7W2uLfst5Kcl2RRkluTHN1a+2lVVZLTk/x+knuTHN9au7pftQEAAGzIipGVWXbmF6etv/1OeNG4bUZGRvLlL385b3vb2/LBD35wWvbbz5G9TyY5fJ1lb0lycWvtiUku7r1OkhckeWLvcXKS/9XHugAAAGaV17/+9fnrv/7rbLXV9EW0voW91tqlSf5tncUvSXJW7/lZSV46ZvnZbdQVSXaoqt36VRsAAMBs8aUvfSm77LJLDjjggGntd6bP2du1tXZb7/mPk+zaez4/ycox7UZ6ywAAADrt29/+ds4///wsWrQor3zlK3PJJZfkD//wD6fc78Au0NJaa0naZLerqpOramlVLV29enUfKgMAAJg5733vezMyMpJbb7015557bg499NB86lOfmnK/Mx32bl87PbP39Y7e8lVJhse0G+ot+3daa2e01g5srR24884797VYAACAzVXfrsa5AecnOS7J+3pfvzBm+Wur6twkByW5e8x0TwAAgBmzYGh4QlfQnEx/E3XIIYfkkEMOmZb99vPWC+ckOSTJTlU1kuQdGQ15n6mqk5IsT3J0r/kFGb3tws0ZvfXCCf2qCwAAYGOWr1wx6BKmRd/CXmvtmA2set562rYkf9avWgAAALY0A7tACwAAAP0j7AEAAHSQsAcAANBBwh4AAEAHzfStFwAAAFiPRYsWZbvttsucOXMyd+7cLF26dEr9CXsAAABjLBhekJUjK6etv+Gh4ayY4O0cvvGNb2SnnXaalv0KewAAAGOsHFmZSz7w2Wnr79A3vnza+poM5+wBAADMAlWVww47LAcccEDOOOOMKfdnZA8AAGAW+Na3vpX58+fnjjvuyPOf//zstddeOfjggze5PyN7AAAAs8D8+fOTJLvsskuOOOKIXHnllVPqT9gDAAAYsHvuuSc///nPf/X8a1/7WhYvXjylPk3jBAAAGLDbb789RxxxRJLkoYceyqte9aocfvjhU+pT2AMAABhjeGh4Wq+gOTw0PG6bPffcM8uWLZu2fSbCHgAAwG+Y6D3xZjvn7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAB0kLAHAAAwC9x111056qijstdee2XvvffO5ZdfPqX+hD0AAIAxFgwNp6qm7bFgAvfZS5JTTjklhx9+eG688cYsW7Yse++995Teh/vsAQAAjLFy1UjOe/tZ09bfK9593Lht7r777lx66aX55Cc/mSTZeuuts/XWW09pv0b2AAAABuyHP/xhdt5555xwwgl56lOfmle/+tW55557ptSnsAcAADBgDz30UK6++ur8yZ/8Sa655ppsu+22ed/73jelPoU9AACAARsaGsrQ0FAOOuigJMlRRx2Vq6++ekp9CnsAAAAD9rjHPS7Dw8O56aabkiQXX3xx9tlnnyn16QItAAAAs8CHP/zhHHvssXnggQey55575swzz5xSf8IeAADAGMPzhyZ0Bc3J9DcRS5YsydKlS6dtv8IeAADAGCtGVg66hGnhnD0AAIAOEvYAAAA6SNgDANgMLBhekKqa1GPB8IJBlw0MkHP2AAA2AytHVuaSD3x2Utsc+saX96kaYHNgZA8AAKCDhD0AAIABu+mmm7JkyZJfPR7zmMfktNNOm1KfpnECAACMMTw0nJFVI9PW39D8oawc53YOT37yk3PttdcmSdasWZP58+fniCOOmNJ+hT0AAIAxRlaN5PRTPjJt/Z1y+msn1f7iiy/O4x//+CxcuHBK+zWNEwAAYBY599xzc8wxx0y5H2EPAABglnjggQdy/vnn5+Uvn/rVdIU9AACAWeIrX/lK9t9//+y6665T7kvYAwAAmCXOOeecaZnCmQh7AAAAs8I999yTiy66KC972cumpT9X4wQAABhjaP7QpK+gOV5/E7HtttvmzjvvnLb9CnsAAABjjHdPvM2FaZwAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAADNipp56afffdN4sXL84xxxyT+++/f8p9CnsAAABjDM0fSlVN22O8++ytWrUqH/rQh7J06dJcd911WbNmTc4999wpvw/32QMAABhj1Y9W5a0nvnva+nvvJ94+bpuHHnoo9913X+bNm5d77703u++++5T3a2SPgRseGp7U/4wMDw0PumQAACZowST/1luwBf6tN3/+/PyX//JfsmDBguy2227Zfvvtc9hhh025XyN7DNzIqpGcfspHJtz+lNNf28dqAACYTitXjeS8t5814favePdxfaxmdvrpT3+aL3zhC/nhD3+YHXbYIS9/+cvzqU99Kn/4h384pX6N7AEAAAzQ17/+9eyxxx7ZeeedM2/evLzsZS/Ld77znSn3K+wBAAAM0IIFC3LFFVfk3nvvTWstF198cfbee+8p9yvsQUctWrhw0leKWrRw4aDLBgDY4hx00EE56qijsv/+++cpT3lKHn744Zx88slT7tc5e9BRy1esyN033zipbbZ/wl59qgYAYPMxf/f5E7qC5mT6G8+73vWuvOtd75q2fSbCHgAAwG8YWTUy6BKmhWmcAAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcNJOxV1X+uqu9X1XVVdU5VbVNVe1TVd6vq5qo6r6q2HkRtAAAAXTDjYa+q5if58yQHttYWJ5mT5JVJ/irJqa21JyT5aZKTZro2AACAQTj99NOzePHi7LvvvjnttNOmpc9BTeOcm+SRVTU3yaOS3Jbk0CT/0Ft/VpKXDqY0AABgSzZ//lCqatoe8+cPbXR/1113Xf72b/82V155ZZYtW5YvfelLufnmm6f8Pmb8PnuttVVV9TdJViS5L8nXklyV5K7W2kO9ZiNJxr/zIAAAwDT70Y9W5cSj/3za+vvEZz600fU33HBDDjrooDzqUY9KkjznOc/J5z73ubzpTW+a0n4HMY1zxyQvSbJHkt2TbJvk8Elsf3JVLa2qpatXr+5TlQAAADNj8eLFueyyy3LnnXfm3nvvzQUXXJCVK1dOud8ZH9lL8h+S/LC1tjpJqupzSZ6ZZIeqmtsb3RtKsmp9G7fWzkhyRpIceOCBbWZKBgAA6I+99947b37zm3PYYYdl2223zZIlSzJnzpwp9zuIc/ZWJPmdqnpUVVWS5yW5Psk3khzVa3Ncki8MoDYAAIAZd9JJJ+Wqq67KpZdemh133DFPetKTptznjIe91tp3M3ohlquT/HOvhjOSvDnJG6rq5iSPTfLxma4NAABgEO64444kyYoVK/K5z30ur3rVq6bc5yCmcaa19o4k71hn8S1Jnj6AcgAAAAbqyCOPzJ133pl58+blox/9aHbYYYcp9zmQsAcAADBb7b77/HGvoDnZ/sZz2WWXTdv+1hL2AAAAxli1amTQJUyLQd1UHQAAgD4S9gAAADpI2AMAALZ4rc3uW3hvSn3CHgAAsEXbZpttcuedd87awNday5133pltttlmUtu5QAsAALBFGxoaysjISFavXj3oUjZom222ydDQ0KS2EfYAAIAt2rx587LHHnsMuoxpZxonAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBM2rh8IJU1aQeC4cXDLpsAIDNztxBFwBsWVaMrMyyM784qW32O+FFfaoGAKC7jOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB00obBXVc+cyDIAAABmh4mO7H14gssAAACYBeZubGVVPSPJ7ybZuareMGbVY5LM6WdhAAAAbLqNhr0kWyd5dK/ddmOW/yzJUf0qCgAAgKnZaNhrrf1Tkn+qqk+21pbPUE0AAABM0Xgje2s9oqrOSLJo7DattUP7URQAAABTM9Gw99kkH0vyf5Ks6V85AAAATIeJhr2HWmv/q6+VAAAAMG0meuuFL1bVn1bVblX1W2sffa0MAACATTbRkb3jel//65hlLcme01sOAAAA02FCYa+1tke/CwEAAGD6TCjsVdV/XN/y1trZm7LTqtohoxd7WZzREcITk9yU5LyMXvHz1iRHt9Z+uin9AwAAbOkmes7e08Y8np3knUlePIX9np7kwtbaXkn2S3JDkrckubi19sQkF/deAwAAsAkmOo3zdWNf90bmzt2UHVbV9kkOTnJ8r+8HkjxQVS9Jckiv2VlJvpnkzZuyDwAAgC3dREf21nVPkk09j2+PJKuTnFlV11TV/6mqbZPs2lq7rdfmx0l23cT+AQAAtngTCntV9cWqOr/3+HJGz6/7/Cbuc26S/ZP8r9baUzMaHH9jymZrrWX0XL711XJyVS2tqqWrV6/exBIAgC3FooULU1WTeixauHDQZQNM2URvvfA3Y54/lGR5a21kE/c5kmSktfbd3ut/yGjYu72qdmut3VZVuyW5Y30bt9bOSHJGkhx44IHrDYQAAGstX7Eid99846S22f4Je/WpGoCZM6GRvdbaPyW5Mcl2SXZM8sCm7rC19uMkK6vqyb1Fz0tyfZLz8+v7+R2X5Aubug8AAIAt3URvvXB0kvdn9KIpleTDVfVfW2v/sIn7fV2ST1fV1kluSXJCRoPnZ6rqpCTLkxy9iX0DAABs8SY6jfNtSZ7WWrsjSapq5yRfz+gUzElrrV2b5MD1rHrepvQHAADAb5ro1Ti3Whv0eu6cxLYAAADMsImO7F1YVV9Nck7v9SuSXNCfkgAAAJiqjYa9qnpCRu9/91+r6mVJntVbdXmST/e7OAAAADbNeCN7pyV5a5K01j6X5HNJUlVP6a17UR9rAwAAYBONd97drq21f153YW/Zor5UBAAAwJSNF/Z22Mi6R05jHQAAAEyj8cLe0qp6zboLq+rVSa7qT0kAAABM1Xjn7L0+yeer6tj8OtwdmGTrJEf0sS4AAACmYKNhr7V2e5LfrarnJlncW/zl1tolfa8MAACATTah++y11r6R5Bt9rgUAAIBpMt45ewAAAGyGhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAOqyqJvxYMDQ86HKBaTR30AUAANA/5739rAm3fcW7j+tjJcBMM7IHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAB0kLAHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AAEAHCXsAAAAdJOwBAAzAwuEFqaoJP+ZuNWfQJQObmbmDLgAAYEu0YmRllp35xQm33++EF/WxGqCLjOwBAAB0kLAHAADQQcLeLLFo4cJJzdtftHDhoEsGAABmMefszRLLV6zIHZd/d8Ltd3nGQX2sBgAA2NwZ2QMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpoYGGvquZU1TVV9aXe6z2q6rtVdXNVnVdVWw+qNgAAgM3dIEf2Tklyw5jXf5Xk1NbaE5L8NMlJA6kKAACgAwYS9qpqKMkfJPk/vdeV5NAk/9BrclaSlw6iNgAAgC4Y1MjeaUnelOTh3uvHJrmrtfZQ7/VIkvkDqAsAAKATZjzsVdULk9zRWrtqE7c/uaqWVtXS1atXT3N1AAAA3TCIkb1nJnlxVd2a5NyMTt88PckOVTW312Yoyar1bdxaO6O1dmBr7cCdd955JuoFAADY7Mx42GutvbW1NtRaW5TklUkuaa0dm+QbSY7qNTsuyRdmujYAAICumE332XtzkjdU1c0ZPYfv4wOuBwAAYLM1d/wm/dNa+2aSb/ae35Lk6YOsBwAAoCtm08geAAAA00TYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoINmPOxV1XBVfaOqrq+q71fVKb3lv1VVF1XVv/S+7jjTtQEAAHTFIEb2HkryxtbaPkl+J8mfVdU+Sd6S5OLW2hOTXNx7DQAAwCaY8bDXWruttXZ17/nPk9yQZH6SlyQ5q9fsrCQvnenaAAAAumKg5+xV1aIkT03y3SS7ttZu6636cZJdB1UXAADA5m5gYa+qHp3kH5O8vrX2s7HrWmstSdvAdidX1dKqWrp69eoZqBQAAEiSBcMLUlWTeszZas6gy95izR3ETqtqXkaD3qdba5/rLb69qnZrrd1WVbsluWN927bWzkhyRpIceOCB6w2EAADA9Fs5sjKXfOCzk9rm0De+vE/VMJ5BXI2zknw8yQ2ttQ+OWXV+kuN6z49L8oWZrg0AAKArBjGy98wkf5Tkn6vq2t6y/y/J+5J8pqpOSrI8ydEDqA0AAKATZjzstda+laQ2sPp5M1kLAABAVw30apwAAAD0h7AHAADQQcIeAAD02aKFCyd1u4JFCxcOumQ6YCC3XgAAgC3J8hUrcvfNN064/fZP2KuP1bClMLIHAADQQcIeAABABwl7AAAAHSTsAQAAdJCwBwAA0EHCHgAAQAcJewAAAB0k7AEAAHSQsAcAANBBwh4AwBQtXLAgVTWpx9w5cwZdNtBxcwddAADA5m7FypW55fyLJ7XNni9+Xp+qARhlZA8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADoIGEPAACgg4Q9AACADhL2AAAAOkjYAwAA6CBhDwAAoIOEPQAAgA4S9rYgC4cXpKom9Vg4vGDQZcOkLRganvSxvmBoeNBls4kWLpj8z7a5c+YMumwA6Lu5gy6AmbNiZGWWnfnFSW2z3wkv6lM10D8rV43kvLefNaltXvHu4/pUDf22YuXK3HL+xZPaZs8XP69P1QDA7GFkDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD3GNanL17tVw2ZvMt/vRQsXDrpcZrlFCxdO+rYIjiuAmTPZW3PN3cqtazYnbr3AuC75wGcn3PbQN768j5UwE+64/LsTbrvLMw7qYyV0wfIVK3L3zTdOapvtn7BXn6oBYF2TvTWX23JtXozsAQAAdJCwBwAA0EGmcQJTUlWTaj93zqbN9Z/MfuY4n4A+mOyxPjw0nBUrV/SpGgAYn7AHTMkt5188qfZ7vvh5m7Qf544yaJM5BhPHIQCDZxonAABAB826sFdVh1fVTVV1c1W9ZdD1bIpNudT4vLmTH2Sd7D42dfpcv+uaU5M/DCe7j6H5Q314pzA58+cPTeq4ne+43eIMDw1P6hgZHhoedMkAzGKzahpnVc1J8tEkz08ykuT/VdX5rbXrB1vZ5MzUpcZnavrcZJ339rMm1f4V7z5u0vt464nvnlT7937i7ZPeB0y3H/1oVU48+s8n3P4Tn/lQH6thNhpZNZLTT/nIhNufcvpr+1gNAJu72Tay9/QkN7fWbmmtPZDk3CQvGXBNAAAAm53ZFvbmJ1k55vVIbxkAAACTUK21QdfwK1V1VJLDW2uv7r3+oyQHtdZeO6bNyUlO7r18cpKbZrxQkmSnJD8ZdBFs8RyHzAaOQ2YDxyGD5hgcnIWttZ3Xt2JWnbOXZFWSsWebD/WW/Upr7YwkZ8xkUfx7VbW0tXbgoOtgy+Y4ZDZwHDIbOA4ZNMfg7DTbpnH+vyRPrKo9qmrrJK9Mcv6AawIAANjszKqRvdbaQ1X12iRfTTInySdaa98fcFkAAACbnVkV9pKktXZBkgsGXQfjMpWW2cBxyGzgOGQ2cBwyaI7BWWhWXaAFAACA6THbztkDAABgGgh7JEmqak1VXTvm8Zbe8rlV9T+q6l/GrHvbRrZbVFWHVNXdvdc3VtXfDO6dsbmoqseOOY5+XFWrxrxuva/XVdVnq+pRvW1urap/7q1bOqavd1fV93rLv1ZVuw/unbG5GfNzbe3xNn8jx+bWG/n5+c11jssDq+qbA3tjzGpjjqNlVXV1Vf1ub/miqrqvt+76qjq7qub11s2rqvf1fkdfXVWXV9ULxvS5pPfz8/B19vWJqrqjqq6b2XfJbDdTx2FVDVfVN3p9fb+qTpn5d7tlmHXn7DEw97XWlqxn+XuSPC7JU1pr91fVdkneuLHtqmpRkstaay+sqkcmuaaqPt9a+3Z/SqcLWmt3JlmSJFX1ziS/aK39Te/1L9YeZ1X16SR/nOSDvU2f21pb974+72+tvb3X/s+T/EVvG5iI+9Y53l4x5vU7M+bY7C3b0M/PJNmlql7QWvtKXyumC8Yed7+X5L1JntNb96+ttSVVNSfJRUmOTvLpJO9OsluSxa21X1bVrmO2SZJjknyr9/XCMcs/meQjSc7u27thczVTx+FDSd7YWru697flVVV1UWvt+v6+vS2PsMcG9UZPXpNkUWvt/iRprf08yTsn2kdr7b6qujbJ/H7UyBbpsiS/vbEGrbWfjXm5bRInJ7Opxj3exvH+JG9LIuwxGY9J8tN1F7bW1lTVlUnmj/kdvUdr7Ze99bcn+UySVFUleXmS5ye5rKq2GfO7/NLef8zCxvTtOGyt3Zbktl77n1fVDRn9W1HYm2bCHms9shfK1npvkhuSrOgFvIls98PW2hFjV1bVjkmemOTSaayVLVRVzU3ygvz6fwZbkq9VVUvyv1trZ4xp+5dJ/mOSu5M8d6ZrZfO3nuNtQ/7dz8/W2nm955cnOaKqnptkYz9LYe1xtE1GR0kOXbdBVW2T5KAkpyR5QkZ/R/9s3XY9v5vR38v/2ps+/AdJ/rEPddMtM34c9v7j4alJvjs9b4GxnLPHWve11paMeZy3boOqOqE3V3tlVQ2vZ7uxQe/ZVbUsyaokX22t/Xgm3gSdtfaXz9IkK5J8vLf8Wa21/TP6B/mfVdXBazdorb2ttTac0Skmr53hetm8beh425Dxfn6+J8l/60OddMva42ivJIcnObs3KpIkj+8dk7cnua219r0J9HdMknN7z8/tvYbxzOhxWFWPzmj4e/1GAiNTIOyxMTcnWdCbS53W2pm9edx3Z/Sm9xtzWWttvyT7Jjmpqpb0s1A6b+wf069rrT2QJK21Vb2vdyT5fJKnr2fbTyc5cuZKpQPWe7xtqtbaJUkemeR3pqc8uq61dnmSnZLs3Fv0r73fv49PckBVvTi//h39mHW3751TdWSSv6iqW5N8OMnha3+fw0T0+zjsXeDlH5N8urX2uT6/nS2WsMcGtdbuzej/aH+kN2S/9h/u1pPo44dJ3pfkzX0pki1WVW075hfGtkkOS3Jd7/UTxzR9SZIbZ75C+A3vSfKmQRfB5qGq9srof6reOXZ572JUb0ny1jG/o0+vqq172+1cVS9P8rwk32utDbfWFrXWFmb0j+rfONUCNqafx2FvtPDjSW5orX0w9I2wx1qPrN+8dPj7esvfltETaK+rqmsyerGCs5L8aBJ9fyzJwU4GZ5rtmuRbvenCVyb5cmtt7blV76vRy+Z/L6Mh0CWd6acN/fz8ldbaBUlWD6A2Nh+/Oo6SnJfkuNbamvW0+79JHlVVz87o9ODVSa6v0dsofCnJzzI6Ve7z62z3j73lqapzMno+6ZOraqSqTurD+2HzNFPH4TOT/FGSQ8f87Pz9fryhLV215iJ1AAAAXWNkDwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AGKOqdq2qv6+qW6rqqqq6vKrcnwyAzY6wBwA9vRv9/t8kl7bW9mytHZDklUmGBloYAGwCYQ8Afu3QJA+01j62dkFrbXlr7cNVNaeq3l9V/6+qvldV/ylJquqQqvpmVf1DVd1YVZ/uhcZU1QFV9U+9EcKvVtVuA3pfAGyB5g66AACYRfZNcvUG1p2U5O7W2tOq6hFJvl1VX+ute2pv2x8l+XaSZ1bVd5N8OMlLWmurq+oVSf4yyYl9fQcA0CPsAcAGVNVHkzwryQNJlif57ao6qrd6+yRP7K27srU20tvm2iSLktyVZHGSi3oDfXOS3DZz1QOwpRP2AODXvp/kyLUvWmt/VlU7JVmaZEWS17XWvjp2g6o6JMkvxyxak9Hfr5Xk+621Z/S5ZgBYL+fsAcCvXZJkm6r6kzHLHtX7+tUkf1JV85Kkqp5UVdtupK+bkuxcVc/otZ9XVfv2o2gAWB8jewDQ01prVfXSJKdW1ZuSrE5yT5I3J/lsRqdnXt27AMvqJC/dSF8P9KZ8fqiqts/o79zTMjp6CAB9V621QdcAAADANDONEwAAoIOEPQAAgA4S9gAAADpI2AMAAOggYQ8AAKCDhD0AAIAOEvYAAAA6SNgDAADooP8fbaQ8hx3YeEAAAAAASUVORK5CYII=\n",
      "text/plain": [
       "<Figure size 1080x576 with 1 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "plt.figure(figsize=(15,8))\n",
    "sns.histplot(data=train_variant[train_variant[\"Gene\"].isin(gene_count.keys())],\n",
    "             x=\"Gene\", hue=\"Class\", multiple=\"dodge\", shrink=.7)\n",
    "plt.title(\"Gene distribution and class\")\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      Gene             Variation  Class\n",
       "ID                                     \n",
       "0   FAM58A  Truncating Mutations      1\n",
       "1      CBL                 W802*      2\n",
       "2      CBL                 Q249E      2\n",
       "3      CBL                 N454D      3\n",
       "4      CBL                 L399V      4"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_variant.head(5)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [Gene, Variation, Class]\n",
       "Index: []"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_variant[train_variant[\"Gene\"] == \"RAD51\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "amino_acid_dict = {'C' : 'CYS', 'D' : 'ASP', 'S' : 'SER', 'Q' : 'GLN', 'K' : 'LYS',\n",
    "    'I' : 'ILE', 'P' : 'PRO', 'T' : 'THR', 'F' : 'PHE', 'N' : 'ASN', \n",
    "    'G' : 'GLY', 'H' : 'HIS', 'L' : 'LEU', 'R' : 'ARG', 'W' : 'TRP', \n",
    "    'A' : 'ALA', 'V' : 'VAL', 'E' : 'GLU', 'Y' : 'TYR', 'M' : 'MET'}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Fonction pour la selection\n",
    "\n",
    "## clean_text\n",
    "Nettoie l'article, lower puis enlve les points qui pourrait nuire  la detection des phrases, (comme les \"...\", ou \"fig.A\")\n",
    "\n",
    "\n",
    "## find_match\n",
    "Va dtecter la phrase qui contient le mot, puis prend 1 phrase avant et aprs (au total 3 phrase) (voir diapo)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def amino_three(amino):\n",
    "    return amino_acid_dict[amino]\n",
    "    \n",
    "def decompose_variation(variation):\n",
    "    decompose_aa_pos_aa = re.compile(\"([a-z]{1,})(\\d+)([a-z]{1,})\")\n",
    "    list_variation = decompose_aa_pos_aa.search(variation)\n",
    "    if list_variation:\n",
    "        aa1 = list_variation.group(1)\n",
    "        aa2 = list_variation.group(3)\n",
    "\n",
    "        if len(aa1) + len(aa2) == 2:\n",
    "            amino1 = amino_acid_dict[aa1.upper()].lower()\n",
    "            position = list_variation.group(2)\n",
    "            amino2 = amino_acid_dict[aa2.upper()].lower()\n",
    "            return [amino1,position,amino2]\n",
    "    return False\n",
    "\n",
    "def decompose_fusion(variation):\n",
    "    decompose_g1_g2_fusion = re.compile(\"(\\w+)\\s?(\\?|-)\\s?(\\w+)\\? fusion\")\n",
    "    list_variation = decompose_g1_g2_fusion.search(variation)\n",
    "    if list_variation:\n",
    "        gene1 = list_variation.group(1)\n",
    "        gene2 = list_variation.group(3)\n",
    "        print(gene1,gene2,variation, list_variation.group(0))\n",
    "        return \"(\" + gene1 + \"|\" + gene2 + \")\"\n",
    "    return False\n",
    "        \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "def clean_text(article):\n",
    "    dot3 = re.compile(\"[\\.]{2,}\")\n",
    "    fig = re.compile(\"fig[s]?\\.\")\n",
    "    decimal = re.compile(\"\\d+\\.\\d+\")    \n",
    "    etal = re.compile(\"et al\\.\")\n",
    "    ie = re.compile(\"i\\.e\\.\")\n",
    "    inc = re.compile(\"inc\\.\")\n",
    "    mutation_point = re.compile(\"[p|c]\\.\")\n",
    "    \n",
    "    clean_article = article.lower()\n",
    "    clean_article = dot3.sub(\".\", clean_article)\n",
    "    clean_article = fig.sub(\"\", clean_article)\n",
    "    clean_article = decimal.sub(\"\", clean_article)\n",
    "    clean_article = etal.sub(\"\", clean_article)\n",
    "    clean_article = ie.sub(\"\", clean_article)    \n",
    "    clean_article = inc.sub(\"\", clean_article)    \n",
    "    clean_article = mutation_point.sub(\"\", clean_article)    \n",
    "    \n",
    "    \n",
    "    return clean_article\n",
    "\n",
    "def join_tuple_string(strings_tuple):\n",
    "    return ' '.join(strings_tuple)\n",
    "\n",
    "def find_match(text, word):\n",
    "    clean = clean_text(text)\n",
    "    word = word.lower()\n",
    "    target_sentence = \"([^.]*{}[^.]*\\.)\".format(word)\n",
    "    before_after_target = \"([^.]*\\.){0,1}\"    \n",
    "    match_exp = re.compile(before_after_target + target_sentence + before_after_target)\n",
    "    match_text = match_exp.findall(clean)\n",
    "    final_match = \"\".join(list(map(join_tuple_string, match_text)))\n",
    "    return final_match\n",
    "\n",
    "def extract_match(line):\n",
    "    # Mettre celui qui consomme le moins en 1er --> ceux qui sont facile  faire sans trop de calcul !\n",
    "    # 1er : best match\n",
    "    # 2e ?\n",
    "    \n",
    "    \n",
    "    # Cleaning text --> Already clean, no need    \n",
    "    text = line[\"Text\"]\n",
    "    variation = line[\"Variation\"]\n",
    "    #gene = line[\"Gene\"].lower()\n",
    "    #print(line.name)\n",
    "    if \"dup\" in variation:\n",
    "        match_dup = find_match(text, \"dup\")\n",
    "        if len(match_dup) != 0:\n",
    "            return match_dup, 2\n",
    "        \n",
    "    # Try with * --> w802*\n",
    "    if \"*\" in variation:\n",
    "        new_var = variation.replace(\"*\", \"\\\\*\")\n",
    "        match_star = find_match(text, new_var)\n",
    "        if len(match_star) != 0:\n",
    "            return match_star, 1\n",
    "        if \"fs\" in variation:\n",
    "            match_fs = find_match(text, \"fs\\\\*\")\n",
    "            if len(match_fs) != 0:\n",
    "                return match_fs, 3\n",
    "            \n",
    "        match_stop_nonsense = find_match(text,\"(stop|nonsense)\")\n",
    "        if len(match_stop_nonsense) != 0:\n",
    "            return match_stop_nonsense, 2\n",
    "\n",
    "    # Try first match with inital variation value\n",
    "    # Quality score = 1 \n",
    "    initial_match = find_match(text, variation)    \n",
    "    if len(initial_match) != 0:\n",
    "        #print(\"First match ! \", variation)\n",
    "        return initial_match, 1\n",
    "    \n",
    "    \n",
    "    # deletion and insertion\n",
    "    if \"del\" in variation or \"ins\" in variation:\n",
    "        match_delins = find_match(text, \"(deletion|insertion)\")\n",
    "        if len(match_delins) != 0:\n",
    "            return match_delins,2\n",
    "        \n",
    "        match_delins_sentence = find_match(text, \"(del|ins)(\\w|\\s){0,}(del|ins)\")\n",
    "        if len(match_delins_sentence) != 0:\n",
    "            return match_delins_sentence, 3\n",
    "    \n",
    "    # Trunc mutations\n",
    "    if \"trunc\" in variation:\n",
    "        match_trunc = find_match(text, \"trunc\")\n",
    "        if len(match_trunc) != 0:\n",
    "            #print(\"Trunc\", variation)\n",
    "            return match_trunc, 3\n",
    "        \n",
    "        match_shorte = find_match(text, \"shorte\")\n",
    "        if len(match_shorte) != 0:\n",
    "            return match_shorte,5\n",
    "\n",
    "    # Fusion of two genes\n",
    "    # Quality score = 2\n",
    "    if \"fusion\" in variation:\n",
    "        fusion_gene = decompose_fusion(variation)\n",
    "        if fusion_gene:\n",
    "            match_fusion_gene = find_match(text, fusion_gene)\n",
    "            if len(match_fusion_gene) != 0:\n",
    "                #print(\"Fusion gene1 | gene2\", variation)\n",
    "                return match_fusion_gene, 2\n",
    "\n",
    "        # Try to match the word fusion at least..\n",
    "        # Quality score 4 (bad)\n",
    "        match_fusion = find_match(text,\"fusion\")\n",
    "        if len(match_fusion) != 0:\n",
    "            #print(\"FUSION\", variation)\n",
    "            return match_fusion, 4\n",
    "        \n",
    "    aa_pos_aa = decompose_variation(variation)\n",
    "    if aa_pos_aa :\n",
    "        if len(aa_pos_aa) == 3:\n",
    "            # If we success to split variation in 3 group --> aa1 pos aa2\n",
    "            # Second try without the last amino acid --> y371\n",
    "            match_variation_aa_pos = find_match(text, variation[:-1])\n",
    "            if len(match_variation_aa_pos) != 0:\n",
    "                #print(\"y371 aa_pos\", variation)\n",
    "                \n",
    "                return match_variation_aa_pos, 2\n",
    "\n",
    "            # Third try with 3 letter code of amino acid --> tyr371ser\n",
    "            match_aa_pos_aa = find_match(text, \"\".join(aa_pos_aa))\n",
    "            if len(match_aa_pos_aa) != 0:\n",
    "                #print(\"aa_pos_aa\", variation)\n",
    "                return match_aa_pos_aa, 1\n",
    "            \n",
    "            # Try with 3 letter code without the last aa --> tyr371\n",
    "            match_aa_pos = find_match(text, aa_pos_aa[0] + aa_pos_aa[1])\n",
    "            if len(match_aa_pos) != 0:\n",
    "                #print(\"aa_pos\", variation)\n",
    "                \n",
    "                return match_aa_pos,2\n",
    "            \n",
    "            # Match position only --> 371\n",
    "            match_pos = find_match(text, aa_pos_aa[1])\n",
    "            if len(match_pos) != 0:\n",
    "                #print(\"pos\", variation)\n",
    "                \n",
    "                return match_pos,4\n",
    "            \n",
    "            # Match position around the real position --> 370 - 379\n",
    "            match_pos_weak = find_match(text, aa_pos_aa[1][:-1] + \"[0-9]\")\n",
    "            if len(match_pos_weak) != 0:\n",
    "                #print(\"pos weak\", variation)\n",
    "                \n",
    "                return match_pos_weak,5\n",
    "            \n",
    "            \n",
    "    # score 6 ?\n",
    "    return [],0\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Exemple  la main juste pour tester"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "word_to_search = \"FAM58A\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "article = train_text.loc[0].values[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "34"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "article.count(word_to_search)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "clean_match = find_match(article, word_to_search)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Before selection : total lenght 39672 and 34 occurences of the word FAM58A\n",
      "After selection :  total lenght 9589 and 34 occurences of the word FAM58A\n"
     ]
    }
   ],
   "source": [
    "print(\"Before selection : total lenght {} and {} occurences of the word {}\".format(\n",
    "    len(article),article.count(word_to_search) , word_to_search))\n",
    "\n",
    "print(\"After selection :  total lenght {} and {} occurences of the word {}\".format(\n",
    "    len(clean_match),clean_match.count(word_to_search.lower()), word_to_search))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>Truncating Mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>Cyclin-dependent kinases (CDKs) regulate a var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>W802*</td>\n",
       "      <td>2</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>Q249E</td>\n",
       "      <td>2</td>\n",
       "      <td>Abstract Background  Non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>N454D</td>\n",
       "      <td>3</td>\n",
       "      <td>Recent evidence has demonstrated that acquired...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>L399V</td>\n",
       "      <td>4</td>\n",
       "      <td>Oncogenic mutations in the monomeric Casitas B...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>D171N</td>\n",
       "      <td>4</td>\n",
       "      <td>Introduction  Myelodysplastic syndromes (MDS) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>A122*</td>\n",
       "      <td>1</td>\n",
       "      <td>Introduction  Myelodysplastic syndromes (MDS) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>Fusions</td>\n",
       "      <td>1</td>\n",
       "      <td>The Runt-related transcription factor 1 gene (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>R80C</td>\n",
       "      <td>4</td>\n",
       "      <td>The RUNX1/AML1 gene is the most frequent targe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>K83E</td>\n",
       "      <td>4</td>\n",
       "      <td>The most frequent mutations associated with le...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3316 rows  4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene             Variation  Class  \\\n",
       "ID                                          \n",
       "0     FAM58A  Truncating Mutations      1   \n",
       "1        CBL                 W802*      2   \n",
       "2        CBL                 Q249E      2   \n",
       "3        CBL                 N454D      3   \n",
       "4        CBL                 L399V      4   \n",
       "...      ...                   ...    ...   \n",
       "3316   RUNX1                 D171N      4   \n",
       "3317   RUNX1                 A122*      1   \n",
       "3318   RUNX1               Fusions      1   \n",
       "3319   RUNX1                  R80C      4   \n",
       "3320   RUNX1                  K83E      4   \n",
       "\n",
       "                                                   Text  \n",
       "ID                                                       \n",
       "0     Cyclin-dependent kinases (CDKs) regulate a var...  \n",
       "1      Abstract Background  Non-small cell lung canc...  \n",
       "2      Abstract Background  Non-small cell lung canc...  \n",
       "3     Recent evidence has demonstrated that acquired...  \n",
       "4     Oncogenic mutations in the monomeric Casitas B...  \n",
       "...                                                 ...  \n",
       "3316  Introduction  Myelodysplastic syndromes (MDS) ...  \n",
       "3317  Introduction  Myelodysplastic syndromes (MDS) ...  \n",
       "3318  The Runt-related transcription factor 1 gene (...  \n",
       "3319  The RUNX1/AML1 gene is the most frequent targe...  \n",
       "3320  The most frequent mutations associated with le...  \n",
       "\n",
       "[3316 rows x 4 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concatenate_data = pd.merge(train_variant, train_text, on=\"ID\").dropna()\n",
    "concatenate_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "concatenate_data[\"Text\"] = concatenate_data.apply(lambda line: clean_text(line[\"Text\"]), axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "concatenate_data[\"Variation\"] = train_variant[\"Variation\"].apply(lambda line: line.lower())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>cyclin-dependent kinases (cdks) regulate a var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>w802*</td>\n",
       "      <td>2</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>q249e</td>\n",
       "      <td>2</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>n454d</td>\n",
       "      <td>3</td>\n",
       "      <td>recent evidence has demonstrated that acquired...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>l399v</td>\n",
       "      <td>4</td>\n",
       "      <td>oncogenic mutations in the monomeric casitas b...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>d171n</td>\n",
       "      <td>4</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>a122*</td>\n",
       "      <td>1</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>fusions</td>\n",
       "      <td>1</td>\n",
       "      <td>the runt-related transcription factor 1 gene (...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>r80c</td>\n",
       "      <td>4</td>\n",
       "      <td>the runx1/aml1 gene is the most frequent targe...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>k83e</td>\n",
       "      <td>4</td>\n",
       "      <td>the most frequent mutations associated with le...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3316 rows  4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene             Variation  Class  \\\n",
       "ID                                          \n",
       "0     FAM58A  truncating mutations      1   \n",
       "1        CBL                 w802*      2   \n",
       "2        CBL                 q249e      2   \n",
       "3        CBL                 n454d      3   \n",
       "4        CBL                 l399v      4   \n",
       "...      ...                   ...    ...   \n",
       "3316   RUNX1                 d171n      4   \n",
       "3317   RUNX1                 a122*      1   \n",
       "3318   RUNX1               fusions      1   \n",
       "3319   RUNX1                  r80c      4   \n",
       "3320   RUNX1                  k83e      4   \n",
       "\n",
       "                                                   Text  \n",
       "ID                                                       \n",
       "0     cyclin-dependent kinases (cdks) regulate a var...  \n",
       "1      abstract background  non-small cell lung canc...  \n",
       "2      abstract background  non-small cell lung canc...  \n",
       "3     recent evidence has demonstrated that acquired...  \n",
       "4     oncogenic mutations in the monomeric casitas b...  \n",
       "...                                                 ...  \n",
       "3316  introduction  myelodysplastic syndromes (mds) ...  \n",
       "3317  introduction  myelodysplastic syndromes (mds) ...  \n",
       "3318  the runt-related transcription factor 1 gene (...  \n",
       "3319  the runx1/aml1 gene is the most frequent targe...  \n",
       "3320  the most frequent mutations associated with le...  \n",
       "\n",
       "[3316 rows x 4 columns]"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concatenate_data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Explications du problme\n",
    "Fonction poubelle juste pour voir ceux qui sont pas dtectable --> len(match) == 0\n",
    "\n",
    "Si ils sont pas dtectable, c'est que le nom donn pour leurs variation sont pas dans l'article, donc je vrfie  la main sur le site https://regex101.com/r/lvQZIf/1 qui sert  visualiser les expressions rgulires.\n",
    "En copiant collant tout l'article, je regarde comment je peux dtecter la variation avec un autre nom ducoup, comme le nom de base ne marche pas.\n",
    "\n",
    "Par exemple Id = 19, y371s n'est pas dans l'article mais il y est sous la forme tyr371ser.\n",
    "\n",
    "Cet regexpr ne marche pas (le texte est en minuscule) -> `([^.]*\\.){0,1}([^.]*y371s[^.]*\\.)([^.]*\\.){0,2}`\n",
    "\n",
    "Mais celle la marchera -> `([^.]*\\.){0,1}([^.]*tyr371ser[^.]*\\.)([^.]*\\.){0,2}` (a donne 3 match sur le site pour l'article id = 19)\n",
    "\n",
    "(pour les regex, il faut juste changer le nom au milieu aprs `*` et avant `[^.]`)\n",
    "\n",
    "Autre exemple, ID = 102, mutation r922* qui n'est mentionn sous aucunes formes, ni mme la position 922. Mais, il y a des mots qui peuvent faire penser  cette mutation du genre \"nonsense mutations\" car l'toile (\\*) veut dire que r922 est devenu un codon stop, donc c'est une mutation non sens. Faudrait voir si je peux convertir ces mutations en mot pour faciliter la dtection, ou utiliser un champ lexical.\n",
    "\n",
    "\n",
    "Donc l, j'essaie de faire plus ou moins au cas par cas (pas tous, y en a 880...) pour dterminer des groupes de variations indtectable pour revoir la dtection.\n",
    "Par exemple, pour les variations de type `y371s`, je peux prendre `y371`, ou tout simplement `371` qui peuvent dtecter la variation dans certain cas.\n",
    "\n",
    "\n",
    "Par contre, pour ceux o y a vraiment pas d'infos dans l'article avec la variation, du genre `t1365m` ID = 44; je sais pas quoi faire d'eux :)\n",
    "Car la position 1365 n'est mme pas mentionn dans l'article, donc je vois pas comment dtecter les infos lies  cette mutation...\n",
    "\n",
    "Soit je garde l'article entier, soit je prend le nom du gne pour la dtection, mais le pb avec a, c'est que ducoup tout ceux qui sont indtectables de ce gne l, auront les mmes lignes de dtect car je dtecterai uniquement en utilisant le nom du gne, donc y aura pas de diffrences entre eux mme si la variation diffre.\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "## Ne lance pas la cellule juste en bas, a prend un certain temps et de toute faon, tu as dj la sortie avec tout ceux qui ne marchent pas.\n",
    "\n",
    "La sortie est de la forme : \n",
    "`variation | id | len(article_original) | len(match)`\n",
    "\n",
    "Certaines variations sont des mots complets du genre `truncating mutations`, un peu relou pour la dtection, ce qu'on pourrait faire, c'est utiliser des mots qui ressemble  truncating grace  des mthodes de traitements de texte qui va prendre que le prfix par exemple et automatiser a pour les mots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def check_line(x):\n",
    "    extract = extract_match(x)\n",
    "    score = extract[1]\n",
    "    if score == 0:\n",
    "        print(x[\"Variation\"].lower(), x.name, len(x[\"Text\"]))\n",
    "\n",
    "        \n",
    "#concatenate_data.apply(lambda x : check_line(x) , axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## \"trunc\" : 10 indectectable\n",
    "## \"shorte\" : 7 --> pour shorter shorten\n",
    "le reste, aucunes infos... mme \"mutation\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>cyclin-dependent kinases (cdks) regulate a var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>CBL</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>to determine if residual cylindrical refractiv...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>DICER1</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>mesenchymal cell populations contribute to mic...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>RYBP</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>the mouse double minute 2 (mdm2)p53 interacti...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>100</th>\n",
       "      <td>TGFBR2</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>signaling by transforming growth factor- (tgf...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3200</th>\n",
       "      <td>RASA1</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>phospholipase c gamma 1 (plc gamma 1) and p21r...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3202</th>\n",
       "      <td>BCL2L11</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>lung cancer is the leading cause of major canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3212</th>\n",
       "      <td>RB1</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>retinoblastoma is a pediatric tumor of the dev...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3252</th>\n",
       "      <td>CASP8</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>evidence exists that alterations of the genes ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3307</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>the runx genes have come to prominence recentl...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>92 rows  4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Gene             Variation  Class  \\\n",
       "ID                                           \n",
       "0      FAM58A  truncating mutations      1   \n",
       "16        CBL  truncating mutations      1   \n",
       "41     DICER1  truncating mutations      1   \n",
       "88       RYBP  truncating mutations      1   \n",
       "100    TGFBR2  truncating mutations      1   \n",
       "...       ...                   ...    ...   \n",
       "3200    RASA1  truncating mutations      1   \n",
       "3202  BCL2L11  truncating mutations      1   \n",
       "3212      RB1  truncating mutations      1   \n",
       "3252    CASP8  truncating mutations      1   \n",
       "3307    RUNX1  truncating mutations      1   \n",
       "\n",
       "                                                   Text  \n",
       "ID                                                       \n",
       "0     cyclin-dependent kinases (cdks) regulate a var...  \n",
       "16    to determine if residual cylindrical refractiv...  \n",
       "41    mesenchymal cell populations contribute to mic...  \n",
       "88    the mouse double minute 2 (mdm2)p53 interacti...  \n",
       "100   signaling by transforming growth factor- (tgf...  \n",
       "...                                                 ...  \n",
       "3200  phospholipase c gamma 1 (plc gamma 1) and p21r...  \n",
       "3202  lung cancer is the leading cause of major canc...  \n",
       "3212  retinoblastoma is a pediatric tumor of the dev...  \n",
       "3252  evidence exists that alterations of the genes ...  \n",
       "3307  the runx genes have come to prominence recentl...  \n",
       "\n",
       "[92 rows x 4 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trunc = concatenate_data[concatenate_data[\"Variation\"] == \"truncating mutations\"]\n",
    "trunc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "truncating mutations 88 15584\n",
      "truncating mutations 1080 77954\n",
      "truncating mutations 1826 29176\n",
      "truncating mutations 1902 95852\n",
      "truncating mutations 1911 6154\n",
      "truncating mutations 2146 1407\n",
      "truncating mutations 2285 167659\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "ID\n",
       "0       None\n",
       "16      None\n",
       "41      None\n",
       "88      None\n",
       "100     None\n",
       "        ... \n",
       "3200    None\n",
       "3202    None\n",
       "3212    None\n",
       "3252    None\n",
       "3307    None\n",
       "Length: 92, dtype: object"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "trunc.apply(lambda x : check_line(x) , axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fix les classes 4  --> 60  fix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'bad' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-29-deaeed22b8fe>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mbad_class4\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbad\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"Class\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m4\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0mclass4\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mconcatenate_data\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mbad_class4\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mclass4_tofix\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mclass4\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mclass4\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"Class\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;36m4\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'bad' is not defined"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'class4_tofix' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-30-3dde65d4650d>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mclass4_tofix\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;32mlambda\u001b[0m \u001b[0mx\u001b[0m \u001b[0;34m:\u001b[0m \u001b[0mcheck_line\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'class4_tofix' is not defined"
     ]
    }
   ],
   "source": [
    "class4_tofix.apply(lambda x : check_line(x) , axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'tgfbr1\\\\*6a'"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = \"tgfbr1*6a\"\n",
    "a = a.replace(\"*\",\"\\\\*\")\n",
    "a\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'tgfbr1*6a'"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "re.findall(\"tgfbr1\\\\*6a\",\"tgfbr1*6a\")[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fix les codons stop --> 60~ -----------> plus que 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>w802*</td>\n",
       "      <td>2</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>CCND3</td>\n",
       "      <td>q276*</td>\n",
       "      <td>7</td>\n",
       "      <td>of the generalizations to emerge from studies ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>TGFBR1</td>\n",
       "      <td>tgfbr1*6a</td>\n",
       "      <td>4</td>\n",
       "      <td>breast cancer is the most common female malign...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>102</th>\n",
       "      <td>MSH6</td>\n",
       "      <td>r922*</td>\n",
       "      <td>4</td>\n",
       "      <td>identification of a high-risk disease-causing ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>MSH6</td>\n",
       "      <td>e1322*</td>\n",
       "      <td>4</td>\n",
       "      <td>human colorectal cancer cell lines are used wi...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3297</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>r177*</td>\n",
       "      <td>4</td>\n",
       "      <td>the aml1 gene is known as the most frequent ta...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3298</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>y113*</td>\n",
       "      <td>4</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3306</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>w279*</td>\n",
       "      <td>1</td>\n",
       "      <td>here we report two new runx1 mutations in one ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3308</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>r174*</td>\n",
       "      <td>7</td>\n",
       "      <td>familial platelet disorder with propensity to...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>a122*</td>\n",
       "      <td>1</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>64 rows  4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene  Variation  Class  \\\n",
       "ID                               \n",
       "1        CBL      w802*      2   \n",
       "77     CCND3      q276*      7   \n",
       "96    TGFBR1  tgfbr1*6a      4   \n",
       "102     MSH6      r922*      4   \n",
       "104     MSH6     e1322*      4   \n",
       "...      ...        ...    ...   \n",
       "3297   RUNX1      r177*      4   \n",
       "3298   RUNX1      y113*      4   \n",
       "3306   RUNX1      w279*      1   \n",
       "3308   RUNX1      r174*      7   \n",
       "3317   RUNX1      a122*      1   \n",
       "\n",
       "                                                   Text  \n",
       "ID                                                       \n",
       "1      abstract background  non-small cell lung canc...  \n",
       "77    of the generalizations to emerge from studies ...  \n",
       "96    breast cancer is the most common female malign...  \n",
       "102   identification of a high-risk disease-causing ...  \n",
       "104   human colorectal cancer cell lines are used wi...  \n",
       "...                                                 ...  \n",
       "3297  the aml1 gene is known as the most frequent ta...  \n",
       "3298  introduction  myelodysplastic syndromes (mds) ...  \n",
       "3306  here we report two new runx1 mutations in one ...  \n",
       "3308   familial platelet disorder with propensity to...  \n",
       "3317  introduction  myelodysplastic syndromes (mds) ...  \n",
       "\n",
       "[64 rows x 4 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "codon_stop = concatenate_data[concatenate_data[\"Variation\"].apply(lambda var: \"*\" in var)]\n",
    "codon_stop"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "q233* 1583 738\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "ID\n",
       "1       None\n",
       "77      None\n",
       "96      None\n",
       "102     None\n",
       "104     None\n",
       "        ... \n",
       "3297    None\n",
       "3298    None\n",
       "3306    None\n",
       "3308    None\n",
       "3317    None\n",
       "Length: 64, dtype: object"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "codon_stop.apply(lambda x : check_line(x) , axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fix KIT gene (del, delins...) -> 41  fix --->  plus que 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "kit_gene = concatenate_data[concatenate_data[\"Gene\"] == \"KIT\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "kit_gene.apply(lambda x : check_line(x) , axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Fix EGFR : 10  fix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "egfr_gene = concatenate_data[concatenate_data[\"Gene\"] == \"EGFR\"]\n",
    "egfr_gene\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "egfr_gene.apply(lambda x : check_line(x) , axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Variable ID pour faciliter le print des deux cellules suivantes pour choper le text, la variation et le gene d'un ID et les copier coller sur le site regex\n",
    "\n",
    "Pour la dtection, je teste juste avec le nom de la variation, puis vraiment si y a rien, je regarde du cot du nom du gne"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "ID = 171 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'somatic mutations introduced into the epidermal growth factor receptor (egfr) gene in non-small-cell lung cancer (nsclc) are important factors to determine therapeutic responses to gefitinib. the current diagnostic test measures the overall egfr mutation status of the cancer tissue, and may ignore the presence of nonmutated, gefitinib-unresponsive cancer cells. twenty-one nsclc patients with egfr mutations were recruited for the study. all patients were treated with gefitinib after surgical treatment. fifty to sixty areas of nsclc tumors were sampled from each tissue, and their egfr mutation states were determined by a primer extension assay. this assay discriminates between egfr mutation-positive and -negative cancer cells within a single tumor tissue. fifteen tissues consisted only of cells with egfr mutations, but the remaining six tissues contained both mutated and non-mutated cells. time to disease progression and overall survival after gefitinib treatment were significantly shorter in those patients with egfr heterogeneity (p =  and p = , respectively). a considerable proportion of nsclc contains a heterogeneous population of both egfr mutated and non-mutated cancer cells, resulting in a reduced response to gefitinib. the intratumor genetic heterogeneity of a target molecule such as egfr would be an important factor to consider when treating patients with molecular target agents. (cancer sci 2008; 99: 929935) lung cancer is a major cause of death. in addition, because many patients are diagnosed at an advanced cancer stage, effective anticancer drugs are urgently needed. tki of egfr ( gefitinib and erlotinib) are a new entity of anticancer drugs and have the potential to improve the therapy of advanced-stage lung cancer.(1) egfr is a transmembrane protein that induces a signal transduction cascade upon ligand binding. it is activated through autophosphorylation of a tyrosine residue in the cytoplasmic domain, causing the receptor to dimerize and internalize, and triggering a cascade leading to cellular proliferation, inhibition of apoptosis, and other cancer-related activities.(2) egfr tki inhibit this cascade by blocking autophosphorylation. there have been extensive studies on the clinical effects of gefitinib. the objective response rate to gefitinib is from 9 to 19%.(3,4) patients characteristics, such as adenocarcinoma histology, female sex, non-smoking history, and asian race, are associated with an increased response to gefitinib, and may guide treatment decisions.(35) in addition, many studies have shown a correlation between somatic egfr mutations and increased responsiveness to gefitinib.(610) such egfr mutations are found frequently in nsclc, and are predominantly located in exons 19 and 21, corresponding to the tyrosine kinase domain. a similar correlation is also reported with erlotinib.(11,12) japanese patients with nsclc have a high proportion of egfr mutations, and these mutations have been correlated with an increased responsiveness to gefitinib.(7,10) however, there are nsclc patients with egfr mutations that have poor responses to gefitinib, suggesting that there are other factors determining responsiveness to gefitinib.(10) one of the biological factors possibly related to the responsiveness of gefitinib is genetic heterogeneity within a single tumor tissue.(13) a tumor tissue usually consists of cancer cells with different genetic alterations, and such intratumor heterogeneity could contribute to resistance to anticancer drugs. in the present report, we examined genetic intratumor heterogeneity of egfr mutations in nsclc tissues using a refined sampling method and mutation detection by primer extension. methods patients and clinical characteristics. we recruited 21 nsclc patients who were treated with gefitinib as monotherapy for postoperative recurrence. surgical resection was complete in all patients except no. 4, whose resection was incomplete due to pleural dissemination. patients had surgery between december 1996 and april 2003 and initiated gefitinib treatment from august 2002 to october 2005 at osaka medical center for cancer and cardiovascular diseases. the ethical committee of osaka medical center for cancer and cardiovascular diseases (chaired by dr hiroshi nakajima) approved the current research, and informed consent was obtained for all cases. all patients were japanese, and clinical information was obtained by reviewing patient medical records. patients were diagnosed with nsclc by histological examination of hematoxylineosinstained sections of the formalin-fixed, paraffin-embedded primary tumor. surviving patients were censored at the date of last contact with the institution. tumor recurrences were not necessarily confirmed pathologically but were diagnosed with imaging, clinical manifestation, and laboratory examination. in patients with measurable disease, tumor responses were assessed by the response evaluation criteria in solid tumors(14) and categorized as a complete response, partial response, or no objective response. tumor response was not evaluated in two patients (patients 16 and 17) because these patients had no measurable disease. however, their diseases, which were malignant effusion in the thoracic cavity (patient 16) and pericardiac space (patient 17), were well controlled after drainage with gefitinib treatment for 30 and 450 days, respectively. time to disease progression and overall survival were calculated from the start date of gefitinib therapy to the date of disease progression or death. determination of the egfr mutation status. tumor samples collected during surgery were frozen rapidly in liquid nitrogen 3 to whom correspondence should be addressed. e-mail: katou-ki@mpref.osaka.jp abbreviations: egfr, epidermal growth factor receptor; nsclc, non-small-cell lung cancer; pcr, polymerase chain reaction; snp, single nucleotide polymorphism; tki, tyrosine kinase inhibitor. 930 doi: /j.1349-.00782.x  2008 japanese cancer association and stored at 80 the tumor was divided into two pieces: one was used for rna extraction and cdna sequencing, and the other was used for tissue sections. total rna was purified from tumor tissues with trizol reagent (invitrogen, carlsbad ca, usa) and purified rna was quantitated and assessed for purity by ultraviolet spectrophotometry. cdna was generated from 5 g of rna with superscript iii (invitrogen) according to the manufacturers protocol. an egfr gene fragment covering exons 1824 was amplified using primers 5-ggatcggcctc ttcatgc-3 and 5-aactttgggcgactatctgc-3. the amplified products were sequenced directly using the bigdye terminator cycle sequencing kit (version ; applied biosystems, foster city, ca, usa) with the abi prism 3730 (applied biosystems). after the results of the direct sequencing of pcr products for snap shot (applied biosystems), oligonucleotide primers were designed next to the mutated nucleotides. multipoint microsampling. tumors were cut into 35 m-thick sections in a cryostat, and stained with % toluidine blue. about 5060 areas, 104 m2  in size including 3050 cells, were selected randomly avoiding normal cells, and microdissected using the as lmd system (leica wetzler, germany). genomic dna was extracted from each sampled area with the qiaamp dna micro kit (qiagen, hilden, germany), and suspended with 20 l of distilled water. a dna fragment containing the mutated or snp locus was amplified by pcr. polymerase chain reactions consisted of 1  pcr buffer (toyobo, osaka, japan),  mm mgso4,  mm dntp each  u/l kod plus dna polymerase and template dna in a 10-l reaction volume. the base of interest was determined by the snap shot assay, according to the manufacturers guidelines with minor modifications. the snap shot assay is based on a primer-extension assay: each extension primer binds to a complementary template in the presence of fluorescently labeled ddntp and dna polymerase, and the polymerase extends the primer by one nucleotide, adding a single dideoxynucleotide to its 3 end. fluorescent dyes used for dideoxynucleotides were as follows: a, dr6g; c, dtamra; g, dr110; and t, drox. before the snap shot assay,  l of pcr products were treated with 1 u shrimp alkaline phosphatase (takara, kyoto, japan) and 1 u exonuclease i (takara) for 1 h at 37c in a final volume of 10 l, followed by inactivation of the enzymes for 30 min at 80 snap shot thermal cycling was carried out in 5 l, containing  l of treated pcr product,  l snapshot ready reaction premix (applied biosystems), 1 mm extension primer, and  l distilled water. cycling conditions were as follows: denaturation with 25 cycles of 10 s at 96c, annealing with 5 s at 50c, and extension with 30 s at 60 to remove unincorporated ddntp 5 l of snapshot products were incubated at 37c for 1 h with 1 u shrimp alkaline phosphatase in a final volume of 6 l, and the enzyme was deactivated as described above. a total of  l of treated snap shot reaction was denatured in  l of distilled water for 5 min at 95c, and was analyzed using an abi prism 3100 genetic analyzer (applied biosystems). the fragment analysis was done with genescan software (applied biosystems). the mutation status or snp base on each allele was determined from the incorporated nucleotides. because the intrinsic fluorescense intensity of each dye is different, we measured the relative fluorescense intensity ratio, and adjusted the outputs of the dna analyzer. the values used to adjust differences in the intrinsic fluorescence intensities of dyes were as follows: dr110/drox, ; dr6g/dr110, ; and dtamara/drox, . the primer sequences for genomic dna amplification and the snap shot assay are listed in tables s1 and s2, respectively. the snp loci used for control studies were selected from jsnp (a repository of japanese single nucleotide polymorphism)(15) using the following criteria: (1) they were on chromosome 16, which rarely changes in lung cancer; and (2) high frequency of the minor allele. measurement of egfr amplification. the experiment was carried out as described previously,(10) except for the primers used in egfr amplification. the sequences of the primers and taqman probe were 5-ggaggaccgtcgcttggt-3, 5-tcttga catgctgcggtgtt-3, and 5-caccgcgacctggcagc ca-3. control dna was purified from leukocytes of five normal individuals. the threshold of the amplification was set at three as described previously. statistical analysis. statistical analysis (kaplanmeier analysis) was carried out using spss windows version (spss , chicago, il, us). two-sided p-values were calculated using the log-rank test. results principle of the assay. theoretical and experimental studies on colorectal cancer using microsatellite instability have shown that cancer cells of the same genotype locate contiguously.(16) therefore, if a sample excised from a tumor tissue is small enough, it will contain a genetically identical population of cancer cells. if a tumor tissue consists of cancer cells with different egfr mutation states, we can differentiate them by analyzing mutation states of multiple areas. this method is quantitative, and the ratio between gefitinib-sensitive cells (egfr mutation positive) and nonsensitive cells (egfr mutation negative, wild type) can be determined simply by counting the number of areas within a tumor of each genetic type. we analyzed 5060 areas from each tissue. it should be noted that the sampling must not be contaminated with non-cancerous cells: a pure population was achieved using laser-capture microdissection to observe tissue sections. because the amount of genomic dna is extremely limited in our study, we chose snap shot, a primer extension assay. in snap shot, oligonucleotide primers were designed next to the mutated egfr nucleotide. using fluorescent dye-labeled dideoxynucleotides, a single nucleotide was added by dna polymerase, and the mutation status was determined. first, we determined the optimal sample size to obtain unbiased pcr amplification by changing the areas excised by lasercapture microdissection. we defined the size as approximately 100  100  35 m, which enabled four assays and contained 3060 cells. then, we checked the quantitative ability of the pcr assay. using genomic dna from a small area and serially diluted cloned egfr cdna as a standard, we determined the amount of genomic dna recovered from one small area. we then checked the linearity of assay with various amounts of the cloned dna. the calibration curve is shown in  s1. there was good linearity between inoculated cdna and the fluorescensce peak ratio, confirming the validity of our assay. the outline of the analysis is shown in figure 1. intratumor heterogeneity of egfr mutations. as a preliminary experiment, we examined lung cancer tissues from several patients confirmed to carry egfr mutations. in one cancer tissue, we found that some areas had wild-type egfr (no egfr mutation). a typical example is shown in figure 2. using genomic dna from two areas of this tumor, we observed two peaks representing the mutated and non-mutated alleles. in contrast, sections from two other areas showed peaks derived only from the non-mutated alleles. this tumor is therefore a heterogenous mixture of cancer cells with and without egfr mutations. it should be noted that all excised areas contained only cancer cells and were morphologically indistinguishable. to exclude the possibility that the heterogeneity was an experimental artifact, we carried out the assay with two snp loci, jst003692 and jst004556, which were heterozygous, c and t, in patient no. 17. because the two heterozygous alleles were quantitatively equivalent in all cells, deviation from the equal allele frequency should be due to experimental artifacts. we sampled 60 areas taniguchi  cancer sci | may 2008 | vol. 99 | no. 5 | 931  2008 japanese cancer association and analyzed them using the snap shot assay. there were no areas with 0 or 100% t allele. the mean and sd of the t allele was  and , respectively, for jst003692, and  and , respectively, for jst004556. calculating from these values, the chance for 0% t allele is   105 and   107, respectively. therefore, we can exclude the possibility that the above heterogeneity ( identi- fication of wild-type egfr) was due to experimental artifacts. additionally, to exclude the possible contamination of normal cells, we determined the mutation status of p53. the analysis of bulk tissue indicated that tissue from patient no. 17 contained only the mutated p53 allele. the spots with only wild-type egfr allele also contained only the mutation p53 allele (data not shown). these spots did not contain normal cells. correlation with response to gefitinib. we then investigated tissues from 21 patients. table 1 is a summary of the clinical information for these patients. cancer tissues were classified into two groups: those consisting only of egfr-mutated cells (15 cases, group i) and those with egfr-mutated and nonmutated cells (six cases, group ii). eight tissues in group i had an equal proportion of mutated and non-mutated egfr (the median was <61%), suggesting that these cells were heterozygous for the mutated egfr gene. examples are shown in figure 3a divergence of the ratio of egfr-mutated and nonmutated alleles among tumor sections was partly due to that introduced during the assay. we excluded the existence of other reported mutations by sequencing the amplified products. in addition, with samples from patients 17, 19, and 20, we confirmed that the lack of mutated alleles was not due to chromosome copy number variation, using snp markers to confirm that chromosome 7p was a heterozygote. the other seven tissues in group i also consisted of the mutated and non-mutated genes, but the frequency of the mutated egfr gene was higher than the non-mutated gene. examples are shown in figure 3df. this increase in the mutant egfr was due either to specific amplification of the mutated egfr gene or to polyploidy ( dominance of the chromosome carrying the mutated gene). the remaining six tissues consisted not only of egfr-mutated cells but also of non-mutated cells ( 3gl). the proportion of the non-mutated cells varied from  to 100%. patient no. 21 did not contain any mutated cells: additionally we confirmed the snap shot results by dna sequencing, that is, direct and sampled sequencing of 68 cdna clones from the block used for the snap shot analysis. the discrepancy from the initial analysis was due to heterogeneity among the sample, because we used a different part of the tumor for the initial sequencing analysis. we evaluated the effect of intratumor heterogeneity on the responsiveness to gefitinib therapy by measuring time to disease progression and overall survival. as shown in figure 4, both time to progression (median of group i, 18 months; median of group ii,  months; p = ) and overall survival (median of group i, 27 months; median of group ii,  months; p = ) were significantly less in group ii. there was no significant difference both in time to progression and overall survival after gefitinib treatment between patients heterozygous for the egfr mutation (patients 18) and those with the dominant mutated egfr allele (patient 915) (time to progression, p = ; overall survival, p = ) ( s2). the number of cases with egfr amplification was four in group i and two in group ii (table 1). thus, the differences were not due to egfr amplification. discussion epidermal growth factor receptor tki are potentially effective anticancer agents, but only a fraction of patients respond to egfr tki therapy. it is important to distinguish patients who are most likely to respond. in japan, somatic mutations found in the egfr tyrosine kinase domain are the best predictive factor of responsiveness to egfr tki, although there are contradictory studies in other nations.(17,18) so far, egfr mutation status has been viewed as homogeneous within a single tumor tissue. however, we found that 6 out of 21 mutation-positive nsclc consisted not only of mutated cells but also of nonmutated cells. in addition, these patients did not respond well to gefitinib: both time to progression and overall survival after gefinitib treatment were significantly reduced compared to patients with only egfr-mutant tumor cells. non-mutated cells are likely to be unresponsive to gefitinib therapy and contribute to disease progression. unfortunately, this hypothesis is not proven due to the unavailability of tissue after treatment. in particular, it is important to examine the fraction of non-mutated cells in recurrent tumors. another possibility is that mutation-positive cells of this tissue type would not respond well to gefitinib. through the course of gefitinib therapy, patients that initially respond to gefitinib lose responsiveness. a second egfr mutation at amino acid position 790 (t790m) was found in tumors that acquired resistance to gefitinib.(19,20) it is interesting to compare the presence of t790m and egfr heterogeneity with disease progression during gefitinib therapy. however, this direct  1. outline of the analysis method. egfr, epidermal growth factor receptor; lcm, laser capture microdissection. 932 doi: /j.1349-.00782.x  2008 japanese cancer association table 1. summary of clinical information of patients  2. an example of the intratumor heterogeneity of the epidermal growth factor receptor mutation. the tissue is patient no. 17 in table 1. (left) a microscopic view of a lung tumor section. red circles indicate regions excised by laser-capture dissection. (right) electropherograms of the snap shot assay of the corresponding regions. blue peak, amplified fragment with ddg; green peak, amplified fragment with dda. patient no. sex stage histology mutation wt area (%) mutated allele (%) heterogeneity group egfr amplification response period between surgery and first gefitinib (months) outcome 1 male iii a ad l858r   i  pr 51 dead 2 female ii b ad l858r   i  pr 82 alive 3 male ii a ad l858r   i  cr 88 alive 4 female iii b ad l858r   i  cr 7 dead 5 female iii a ad l858r   i  cr 52 alive 6 female iii a ad dele746-a750   i  cr 88 dead 7 female ii b ad dele746-a750   i  pr 48 alive 8 female i a ad dele746-a750   i  pr 2 alive 9 male ii b ad l858r   i  pr 38 dead 10 female i a ad dele746-a750   i  pr 18 alive 11 female i a ad dele746-a750   i  pr 17 alive (lost from follow up) 12 female i b ad dele746-a750   i  pr 37 alive 13 female i b ad l861q, g719s   i  pr 35 alive 14 female iii a ad dell747-t751   i  pr 11 alive (lost from follow up) 15 female i b ad dele750-k758   i  pr 31 alive 16 female ii b ad l858r   ii  sd 16 dead 17 male iii a ad dele746-a750   ii  pr 13 dead 18 female iii a ad dele746-a750   ii  cr 39 alive 19 female iii b ad e746-t751 insip   ii  pr 42 dead 20 female ii b ad dele746-a750   ii  pr 12 dead 21 male iii b adsq dela859-l883 insv  na ii  pd 73 dead mutated allele (%), ratios of the mutated allele of all areas was calculated, and their median value is shown. wt area (%), fraction of areas consisting only of the non-mutant allele. ad, adenocarcinoma; adsq, adenosquamous carcinoma; cr, complete response; egfr, epidermal growth factor receptor; na, not applicable; pd, progressive disease; pr, partial response; sd, stable disease. taniguchi  cancer sci | may 2008 | vol. 99 | no. 5 | 933  2008 japanese cancer association comparison was difficult for the following reasons. first, we observed egfr tumor heterogeneity before gefitinib therapy, whereas most of the t790m mutations were found in tissues after gefitinib therapy.(1921) the presence of the t790m mutation in nsclc before therapy is very rare.(22) therefore, the t790m mutation is not informative for predicting gefitinib responsiveness. second, the reported cases of nsclc with the t790m mutation initially responded to gefitinib and then acquired resistance. data for gefitinib non-responders are limited.(22) the patients with egfr heterogeneity did not respond as well as other patients, and two out of six cases were categorized as non-responders. egfr heterogeneity is likely to correlate with the initial response to gefitinib rather than the frequency of acquired resistance. additional studies with biopsy samples during and after gefitinib therapy will provide a clearer picture of the effects of both the t790m mutation and egfr heterogeneity on gefitinib responsiveness. the overall response rate of this study was 81% (17 out of 21), and similar to that in egfr-mutated patients reported in japan. in general, time to progression is a better indicator of gefitinib efficacy than tumor response. in our case, due to the small sample size, the difference in the gefitinib effect was apparent with time to progression, and not with tumor response.  3. intratumor heterogeneity of the epidermal growth factor receptor (egfr) gene. vertical axis indicates the percentage of the mutated egfr gene. horizontal axis areas ordered by the percentage of the mutated gene. blue regions correspond to the mutated genes, and purple regions correspond to the non-mutated genes. columns only consisting of purple regions represent areas consisting of non-mutated cells. each panel corresponds to the following patients: (a) 1; (b) 4; (c) 5; (d) 13; (e) 14; (f) 15; (g) 16; (h) 17; (i) 18; (j) 19; (k) 20; and (l) 21. 934 doi: /j.1349-.00782.x  2008 japanese cancer association in the present study, the egfr mutation status was obtained from primary tumors, but gefitinib therapy was used for patients who had relapsed or developed metastatic tumors. one may wonder whether the egfr mutation status is the same both in primary and metastatic tumors. although there is no data on egfr to address this possibility, there are studies on her2. trastuzumab is an effective therapy for breast cancer patients that overexpress her2, and patients are selected by a diagnostic test to detect her2 expression levels. her2 expression in primary and metastatic lesions of the same patients was examined, and the degree of 9095% of cases was concordant in both lesions.(23,24) intratumor genetic heterogeneity has been considered as a potential cause of anticancer drug resistance. in addition, there have been preliminary studies to elucidate the heterogeneity.(16,2528) however, there have been no systematic studies to link tumor heterogeneity with drug resistance. we introduced a new technical framework, which is simple in principle but practically demanding. the main obstacles are the sampling step and numerous pcr reactions. these technical problems will be solved with the development of a specific sampling machine. although there are other technical possibilities, our method of sectioning tumor tissue into small parts allows a variety of analytical techniques, including gene expression and epigenetic analysis. we are now expecting a second generation of molecular target therapies.(29) selecting patients who are most likely to benefit is important, and intratumor heterogeneity will be an important factor for evaluating their efficacy.'"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "concatenate_data.loc[ID,\"Text\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "a859_l883delinsv egfr\n"
     ]
    }
   ],
   "source": [
    "print(concatenate_data.loc[ID,\"Variation\"],\n",
    "concatenate_data.loc[ID,\"Gene\"].lower())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# CLEAN DATA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pax8 ppar pax8-ppar? fusion pax8-ppar? fusion\n",
      "done in 381.8074963092804 s\n"
     ]
    }
   ],
   "source": [
    "start = time.time()\n",
    "clean_match_data = concatenate_data.apply(lambda x: extract_match(x), axis = 1)\n",
    "clean_match = pd.DataFrame(list(clean_match_data), columns = [\"Text\",\"Score\"], index = clean_match_data.index)\n",
    "print(\"done in {} s\".format(time.time() - start))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Text</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>here, we deorphanize cdk10 by identifying cycl...</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>most of the changes were novel, although 4 ca...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>all mutations from the second group were pred...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>2129 the vast majority of aml1 mutatio...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>(a) spleen from morbid mice/d171n (left) and ...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>lpxn is preferentially expressed in hematopoi...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>) conversely, mutations mapping to the first d...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>mutation analysis in fpd/aml pedigrees. elect...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3316 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Text  Score\n",
       "ID                                                            \n",
       "0     here, we deorphanize cdk10 by identifying cycl...    3.0\n",
       "1      the c-cbl loh also positively correlated with...    1.0\n",
       "2      the c-cbl loh also positively correlated with...    1.0\n",
       "3      most of the changes were novel, although 4 ca...    1.0\n",
       "4      all mutations from the second group were pred...    1.0\n",
       "...                                                 ...    ...\n",
       "3316  2129 the vast majority of aml1 mutatio...    1.0\n",
       "3317   (a) spleen from morbid mice/d171n (left) and ...    2.0\n",
       "3318   lpxn is preferentially expressed in hematopoi...    1.0\n",
       "3319  ) conversely, mutations mapping to the first d...    1.0\n",
       "3320   mutation analysis in fpd/aml pedigrees. elect...    1.0\n",
       "\n",
       "[3316 rows x 2 columns]"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clean_match"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "full_x_data = pd.DataFrame(clean_match)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text_x</th>\n",
       "      <th>Text_y</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>cyclin-dependent kinases (cdks) regulate a var...</td>\n",
       "      <td>here, we deorphanize cdk10 by identifying cycl...</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>w802*</td>\n",
       "      <td>2</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>q249e</td>\n",
       "      <td>2</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>n454d</td>\n",
       "      <td>3</td>\n",
       "      <td>recent evidence has demonstrated that acquired...</td>\n",
       "      <td>most of the changes were novel, although 4 ca...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>l399v</td>\n",
       "      <td>4</td>\n",
       "      <td>oncogenic mutations in the monomeric casitas b...</td>\n",
       "      <td>all mutations from the second group were pred...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>d171n</td>\n",
       "      <td>4</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "      <td>2129 the vast majority of aml1 mutatio...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>a122*</td>\n",
       "      <td>1</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "      <td>(a) spleen from morbid mice/d171n (left) and ...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>fusions</td>\n",
       "      <td>1</td>\n",
       "      <td>the runt-related transcription factor 1 gene (...</td>\n",
       "      <td>lpxn is preferentially expressed in hematopoi...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>r80c</td>\n",
       "      <td>4</td>\n",
       "      <td>the runx1/aml1 gene is the most frequent targe...</td>\n",
       "      <td>) conversely, mutations mapping to the first d...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>k83e</td>\n",
       "      <td>4</td>\n",
       "      <td>the most frequent mutations associated with le...</td>\n",
       "      <td>mutation analysis in fpd/aml pedigrees. elect...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3316 rows  6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene             Variation  Class  \\\n",
       "ID                                          \n",
       "0     FAM58A  truncating mutations      1   \n",
       "1        CBL                 w802*      2   \n",
       "2        CBL                 q249e      2   \n",
       "3        CBL                 n454d      3   \n",
       "4        CBL                 l399v      4   \n",
       "...      ...                   ...    ...   \n",
       "3316   RUNX1                 d171n      4   \n",
       "3317   RUNX1                 a122*      1   \n",
       "3318   RUNX1               fusions      1   \n",
       "3319   RUNX1                  r80c      4   \n",
       "3320   RUNX1                  k83e      4   \n",
       "\n",
       "                                                 Text_x  \\\n",
       "ID                                                        \n",
       "0     cyclin-dependent kinases (cdks) regulate a var...   \n",
       "1      abstract background  non-small cell lung canc...   \n",
       "2      abstract background  non-small cell lung canc...   \n",
       "3     recent evidence has demonstrated that acquired...   \n",
       "4     oncogenic mutations in the monomeric casitas b...   \n",
       "...                                                 ...   \n",
       "3316  introduction  myelodysplastic syndromes (mds) ...   \n",
       "3317  introduction  myelodysplastic syndromes (mds) ...   \n",
       "3318  the runt-related transcription factor 1 gene (...   \n",
       "3319  the runx1/aml1 gene is the most frequent targe...   \n",
       "3320  the most frequent mutations associated with le...   \n",
       "\n",
       "                                                 Text_y  Score  \n",
       "ID                                                              \n",
       "0     here, we deorphanize cdk10 by identifying cycl...    3.0  \n",
       "1      the c-cbl loh also positively correlated with...    1.0  \n",
       "2      the c-cbl loh also positively correlated with...    1.0  \n",
       "3      most of the changes were novel, although 4 ca...    1.0  \n",
       "4      all mutations from the second group were pred...    1.0  \n",
       "...                                                 ...    ...  \n",
       "3316  2129 the vast majority of aml1 mutatio...    1.0  \n",
       "3317   (a) spleen from morbid mice/d171n (left) and ...    2.0  \n",
       "3318   lpxn is preferentially expressed in hematopoi...    1.0  \n",
       "3319  ) conversely, mutations mapping to the first d...    1.0  \n",
       "3320   mutation analysis in fpd/aml pedigrees. elect...    1.0  \n",
       "\n",
       "[3316 rows x 6 columns]"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Text</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>here, we deorphanize cdk10 by identifying cycl...</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>most of the changes were novel, although 4 ca...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>all mutations from the second group were pred...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>2129 the vast majority of aml1 mutatio...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>(a) spleen from morbid mice/d171n (left) and ...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>lpxn is preferentially expressed in hematopoi...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>) conversely, mutations mapping to the first d...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>mutation analysis in fpd/aml pedigrees. elect...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3316 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Text  Score\n",
       "ID                                                            \n",
       "0     here, we deorphanize cdk10 by identifying cycl...    3.0\n",
       "1      the c-cbl loh also positively correlated with...    1.0\n",
       "2      the c-cbl loh also positively correlated with...    1.0\n",
       "3      most of the changes were novel, although 4 ca...    1.0\n",
       "4      all mutations from the second group were pred...    1.0\n",
       "...                                                 ...    ...\n",
       "3316  2129 the vast majority of aml1 mutatio...    1.0\n",
       "3317   (a) spleen from morbid mice/d171n (left) and ...    2.0\n",
       "3318   lpxn is preferentially expressed in hematopoi...    1.0\n",
       "3319  ) conversely, mutations mapping to the first d...    1.0\n",
       "3320   mutation analysis in fpd/aml pedigrees. elect...    1.0\n",
       "\n",
       "[3316 rows x 2 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_x_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {},
   "outputs": [],
   "source": [
    "full_y_data = concatenate_data.loc[:,[\"Gene\",\"Variation\",\"Class\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Text</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3135</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3242</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3254</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3256</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3305</th>\n",
       "      <td>[]</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>164 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Text  Score\n",
       "ID              \n",
       "46     []    0.0\n",
       "61     []    0.0\n",
       "71     []    0.0\n",
       "72     []    0.0\n",
       "73     []    0.0\n",
       "...   ...    ...\n",
       "3135   []    0.0\n",
       "3242   []    0.0\n",
       "3254   []    0.0\n",
       "3256   []    0.0\n",
       "3305   []    0.0\n",
       "\n",
       "[164 rows x 2 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "clean_match[clean_match[\"Score\"] == 0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TEXT_X et TEXT_Y pour check si les articles correspondent --> Pas de pb de dcalages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "clean_match.index.name = \"ID\"\n",
    "new_data = pd.merge(concatenate_data,clean_match, on = \"ID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "      <th>Text_x</th>\n",
       "      <th>Text_y</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>FAM58A</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "      <td>cyclin-dependent kinases (cdks) regulate a var...</td>\n",
       "      <td>here, we deorphanize cdk10 by identifying cycl...</td>\n",
       "      <td>3.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CBL</td>\n",
       "      <td>w802*</td>\n",
       "      <td>2</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>CBL</td>\n",
       "      <td>q249e</td>\n",
       "      <td>2</td>\n",
       "      <td>abstract background  non-small cell lung canc...</td>\n",
       "      <td>the c-cbl loh also positively correlated with...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>CBL</td>\n",
       "      <td>n454d</td>\n",
       "      <td>3</td>\n",
       "      <td>recent evidence has demonstrated that acquired...</td>\n",
       "      <td>most of the changes were novel, although 4 ca...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>CBL</td>\n",
       "      <td>l399v</td>\n",
       "      <td>4</td>\n",
       "      <td>oncogenic mutations in the monomeric casitas b...</td>\n",
       "      <td>all mutations from the second group were pred...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3316</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>d171n</td>\n",
       "      <td>4</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "      <td>2129 the vast majority of aml1 mutatio...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3317</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>a122*</td>\n",
       "      <td>1</td>\n",
       "      <td>introduction  myelodysplastic syndromes (mds) ...</td>\n",
       "      <td>(a) spleen from morbid mice/d171n (left) and ...</td>\n",
       "      <td>2.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3318</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>fusions</td>\n",
       "      <td>1</td>\n",
       "      <td>the runt-related transcription factor 1 gene (...</td>\n",
       "      <td>lpxn is preferentially expressed in hematopoi...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3319</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>r80c</td>\n",
       "      <td>4</td>\n",
       "      <td>the runx1/aml1 gene is the most frequent targe...</td>\n",
       "      <td>) conversely, mutations mapping to the first d...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3320</th>\n",
       "      <td>RUNX1</td>\n",
       "      <td>k83e</td>\n",
       "      <td>4</td>\n",
       "      <td>the most frequent mutations associated with le...</td>\n",
       "      <td>mutation analysis in fpd/aml pedigrees. elect...</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3316 rows  6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene             Variation  Class  \\\n",
       "ID                                          \n",
       "0     FAM58A  truncating mutations      1   \n",
       "1        CBL                 w802*      2   \n",
       "2        CBL                 q249e      2   \n",
       "3        CBL                 n454d      3   \n",
       "4        CBL                 l399v      4   \n",
       "...      ...                   ...    ...   \n",
       "3316   RUNX1                 d171n      4   \n",
       "3317   RUNX1                 a122*      1   \n",
       "3318   RUNX1               fusions      1   \n",
       "3319   RUNX1                  r80c      4   \n",
       "3320   RUNX1                  k83e      4   \n",
       "\n",
       "                                                 Text_x  \\\n",
       "ID                                                        \n",
       "0     cyclin-dependent kinases (cdks) regulate a var...   \n",
       "1      abstract background  non-small cell lung canc...   \n",
       "2      abstract background  non-small cell lung canc...   \n",
       "3     recent evidence has demonstrated that acquired...   \n",
       "4     oncogenic mutations in the monomeric casitas b...   \n",
       "...                                                 ...   \n",
       "3316  introduction  myelodysplastic syndromes (mds) ...   \n",
       "3317  introduction  myelodysplastic syndromes (mds) ...   \n",
       "3318  the runt-related transcription factor 1 gene (...   \n",
       "3319  the runx1/aml1 gene is the most frequent targe...   \n",
       "3320  the most frequent mutations associated with le...   \n",
       "\n",
       "                                                 Text_y  Score  \n",
       "ID                                                              \n",
       "0     here, we deorphanize cdk10 by identifying cycl...    3.0  \n",
       "1      the c-cbl loh also positively correlated with...    1.0  \n",
       "2      the c-cbl loh also positively correlated with...    1.0  \n",
       "3      most of the changes were novel, although 4 ca...    1.0  \n",
       "4      all mutations from the second group were pred...    1.0  \n",
       "...                                                 ...    ...  \n",
       "3316  2129 the vast majority of aml1 mutatio...    1.0  \n",
       "3317   (a) spleen from morbid mice/d171n (left) and ...    2.0  \n",
       "3318   lpxn is preferentially expressed in hematopoi...    1.0  \n",
       "3319  ) conversely, mutations mapping to the first d...    1.0  \n",
       "3320   mutation analysis in fpd/aml pedigrees. elect...    1.0  \n",
       "\n",
       "[3316 rows x 6 columns]"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_data_best_score =  new_data[new_data[\"Score\"] == 1]\n",
    "new_data_best_score"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Ceux qui ont un score de 0 --> PB"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bad = new_data[new_data[\"Score\"] == 0]\n",
    "bad[\"Gene\"].tolist()\n",
    "bad_class4 = bad[\"Class\"] == 4\n",
    "class4 = concatenate_data.loc[bad_class4.index,]\n",
    "class4_tofix = class4[class4[\"Class\"] == 4]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check si la 1ere phrase y est bien dans l'article x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "count_new = new_data_best_score.apply(lambda line: line[\"Text_y\"].split(\".\")[0] in line[\"Text_x\"].lower(), axis = 1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Si 1ere phrase y == 1ere phrase x, apparemment a fait pas le comptage mais c'est good !"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_data.loc[count_new[count_new == False].index,]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Check si la variation est bien dans les phrases car c'est du score 1 --> perfect match\n",
    "Ceux qui y sont pas, c'est les mutations codons stop avec `*` qui a une signification dans les expression rguliere, donc  retravailler !"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_data_best_score[new_data_best_score.apply(lambda line: line[\"Variation\"] in line[\"Text_y\"], axis = 1) == False]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_data_best_score = new_data_best_score.drop([1074,1897])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_data_best_score[\"Gene\"].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_data_best_score[new_data_best_score[\"Gene\"] == \"KMT2D\"][\"Text_y\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_data_best_score[new_data_best_score[\"Gene\"] == \"BRCA1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_data_best_score"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Save les datas"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "#CBL = train_variant[train_variant[\"Gene\"] == \"RUNX1\"]\n",
    "#article_cbl = train_text.loc[CBL.index,]\n",
    "\n",
    "#CBL.to_csv(\"cbl.txt\")\n",
    "#clean_TERT = new_data_best_score[new_data_best_score[\"Gene\"] == \"BRCA1\"][\"Text_y\"]\n",
    "dtf = pd.merge(pd.DataFrame(new_data_best_score.index), new_data_best_score[\"Text_y\"], on =\"ID\")\n",
    "dtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#np.savetxt(\"../datas/new_data_clean.txt\",dtf, fmt = \"%d || %s\", header= \",\".join(dtf.columns), comments='')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Requirement already satisfied: sklearn in /home/ragou/.local/lib/python3.8/site-packages (0.0)\n",
      "Collecting scikit-learn\n",
      "  Downloading scikit_learn-1.0-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (25.8 MB)\n",
      "\u001b[K     || 25.8 MB 122 kB/s  eta 0:00:01\n",
      "\u001b[?25hRequirement already satisfied: threadpoolctl>=2.0.0 in /home/ragou/.local/lib/python3.8/site-packages (from scikit-learn->sklearn) (2.1.0)\n",
      "Requirement already satisfied: numpy>=1.14.6 in /usr/lib/python3/dist-packages (from scikit-learn->sklearn) (1.17.4)\n",
      "Requirement already satisfied: joblib>=0.11 in /home/ragou/.local/lib/python3.8/site-packages (from scikit-learn->sklearn) (0.17.0)\n",
      "Requirement already satisfied: scipy>=1.1.0 in /home/ragou/.local/lib/python3.8/site-packages (from scikit-learn->sklearn) (1.7.1)\n",
      "Installing collected packages: scikit-learn\n",
      "Successfully installed scikit-learn-1.0\n"
     ]
    }
   ],
   "source": [
    "!pip3 install sklearn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Help on package sklearn:\n",
      "\n",
      "NAME\n",
      "    sklearn\n",
      "\n",
      "DESCRIPTION\n",
      "    Machine learning module for Python\n",
      "    ==================================\n",
      "    \n",
      "    sklearn is a Python module integrating classical machine\n",
      "    learning algorithms in the tightly-knit world of scientific Python\n",
      "    packages (numpy, scipy, matplotlib).\n",
      "    \n",
      "    It aims to provide simple and efficient solutions to learning problems\n",
      "    that are accessible to everybody and reusable in various contexts:\n",
      "    machine-learning as a versatile tool for science and engineering.\n",
      "    \n",
      "    See http://scikit-learn.org for complete documentation.\n",
      "\n",
      "PACKAGE CONTENTS\n",
      "    __check_build (package)\n",
      "    _build_utils (package)\n",
      "    _config\n",
      "    _distributor_init\n",
      "    _isotonic\n",
      "    _loss (package)\n",
      "    _min_dependencies\n",
      "    base\n",
      "    calibration\n",
      "    cluster (package)\n",
      "    compose (package)\n",
      "    conftest\n",
      "    covariance (package)\n",
      "    cross_decomposition (package)\n",
      "    datasets (package)\n",
      "    decomposition (package)\n",
      "    discriminant_analysis\n",
      "    dummy\n",
      "    ensemble (package)\n",
      "    exceptions\n",
      "    experimental (package)\n",
      "    externals (package)\n",
      "    feature_extraction (package)\n",
      "    feature_selection (package)\n",
      "    gaussian_process (package)\n",
      "    impute (package)\n",
      "    inspection (package)\n",
      "    isotonic\n",
      "    kernel_approximation\n",
      "    kernel_ridge\n",
      "    linear_model (package)\n",
      "    manifold (package)\n",
      "    metrics (package)\n",
      "    mixture (package)\n",
      "    model_selection (package)\n",
      "    multiclass\n",
      "    multioutput\n",
      "    naive_bayes\n",
      "    neighbors (package)\n",
      "    neural_network (package)\n",
      "    pipeline\n",
      "    preprocessing (package)\n",
      "    random_projection\n",
      "    semi_supervised (package)\n",
      "    setup\n",
      "    svm (package)\n",
      "    tests (package)\n",
      "    tree (package)\n",
      "    utils (package)\n",
      "\n",
      "FUNCTIONS\n",
      "    clone(estimator, *, safe=True)\n",
      "        Constructs a new unfitted estimator with the same parameters.\n",
      "        \n",
      "        Clone does a deep copy of the model in an estimator\n",
      "        without actually copying attached data. It yields a new estimator\n",
      "        with the same parameters that has not been fitted on any data.\n",
      "        \n",
      "        If the estimator's `random_state` parameter is an integer (or if the\n",
      "        estimator doesn't have a `random_state` parameter), an *exact clone* is\n",
      "        returned: the clone and the original estimator will give the exact same\n",
      "        results. Otherwise, *statistical clone* is returned: the clone might\n",
      "        yield different results from the original estimator. More details can be\n",
      "        found in :ref:`randomness`.\n",
      "        \n",
      "        Parameters\n",
      "        ----------\n",
      "        estimator : {list, tuple, set} of estimator instance or a single             estimator instance\n",
      "            The estimator or group of estimators to be cloned.\n",
      "        \n",
      "        safe : bool, default=True\n",
      "            If safe is False, clone will fall back to a deep copy on objects\n",
      "            that are not estimators.\n",
      "    \n",
      "    config_context(**new_config)\n",
      "        Context manager for global scikit-learn configuration\n",
      "        \n",
      "        Parameters\n",
      "        ----------\n",
      "        assume_finite : bool, default=False\n",
      "            If True, validation for finiteness will be skipped,\n",
      "            saving time, but leading to potential crashes. If\n",
      "            False, validation for finiteness will be performed,\n",
      "            avoiding error.  Global default: False.\n",
      "        \n",
      "        working_memory : int, default=1024\n",
      "            If set, scikit-learn will attempt to limit the size of temporary arrays\n",
      "            to this number of MiB (per job when parallelised), often saving both\n",
      "            computation time and memory on expensive operations that can be\n",
      "            performed in chunks. Global default: 1024.\n",
      "        \n",
      "        print_changed_only : bool, default=True\n",
      "            If True, only the parameters that were set to non-default\n",
      "            values will be printed when printing an estimator. For example,\n",
      "            ``print(SVC())`` while True will only print 'SVC()', but would print\n",
      "            'SVC(C=1.0, cache_size=200, ...)' with all the non-changed parameters\n",
      "            when False. Default is True.\n",
      "        \n",
      "            .. versionchanged:: 0.23\n",
      "               Default changed from False to True.\n",
      "        \n",
      "        display : {'text', 'diagram'}, default='text'\n",
      "            If 'diagram', estimators will be displayed as a diagram in a Jupyter\n",
      "            lab or notebook context. If 'text', estimators will be displayed as\n",
      "            text. Default is 'text'.\n",
      "        \n",
      "            .. versionadded:: 0.23\n",
      "        \n",
      "        Notes\n",
      "        -----\n",
      "        All settings, not just those presently modified, will be returned to\n",
      "        their previous values when the context manager is exited.\n",
      "        \n",
      "        Examples\n",
      "        --------\n",
      "        >>> import sklearn\n",
      "        >>> from sklearn.utils.validation import assert_all_finite\n",
      "        >>> with sklearn.config_context(assume_finite=True):\n",
      "        ...     assert_all_finite([float('nan')])\n",
      "        >>> with sklearn.config_context(assume_finite=True):\n",
      "        ...     with sklearn.config_context(assume_finite=False):\n",
      "        ...         assert_all_finite([float('nan')])\n",
      "        Traceback (most recent call last):\n",
      "        ...\n",
      "        ValueError: Input contains NaN, ...\n",
      "        \n",
      "        See Also\n",
      "        --------\n",
      "        set_config : Set global scikit-learn configuration.\n",
      "        get_config : Retrieve current values of the global configuration.\n",
      "    \n",
      "    get_config()\n",
      "        Retrieve current values for configuration set by :func:`set_config`\n",
      "        \n",
      "        Returns\n",
      "        -------\n",
      "        config : dict\n",
      "            Keys are parameter names that can be passed to :func:`set_config`.\n",
      "        \n",
      "        See Also\n",
      "        --------\n",
      "        config_context : Context manager for global scikit-learn configuration.\n",
      "        set_config : Set global scikit-learn configuration.\n",
      "    \n",
      "    set_config(assume_finite=None, working_memory=None, print_changed_only=None, display=None)\n",
      "        Set global scikit-learn configuration\n",
      "        \n",
      "        .. versionadded:: 0.19\n",
      "        \n",
      "        Parameters\n",
      "        ----------\n",
      "        assume_finite : bool, default=None\n",
      "            If True, validation for finiteness will be skipped,\n",
      "            saving time, but leading to potential crashes. If\n",
      "            False, validation for finiteness will be performed,\n",
      "            avoiding error.  Global default: False.\n",
      "        \n",
      "            .. versionadded:: 0.19\n",
      "        \n",
      "        working_memory : int, default=None\n",
      "            If set, scikit-learn will attempt to limit the size of temporary arrays\n",
      "            to this number of MiB (per job when parallelised), often saving both\n",
      "            computation time and memory on expensive operations that can be\n",
      "            performed in chunks. Global default: 1024.\n",
      "        \n",
      "            .. versionadded:: 0.20\n",
      "        \n",
      "        print_changed_only : bool, default=None\n",
      "            If True, only the parameters that were set to non-default\n",
      "            values will be printed when printing an estimator. For example,\n",
      "            ``print(SVC())`` while True will only print 'SVC()' while the default\n",
      "            behaviour would be to print 'SVC(C=1.0, cache_size=200, ...)' with\n",
      "            all the non-changed parameters.\n",
      "        \n",
      "            .. versionadded:: 0.21\n",
      "        \n",
      "        display : {'text', 'diagram'}, default=None\n",
      "            If 'diagram', estimators will be displayed as a diagram in a Jupyter\n",
      "            lab or notebook context. If 'text', estimators will be displayed as\n",
      "            text. Default is 'text'.\n",
      "        \n",
      "            .. versionadded:: 0.23\n",
      "        \n",
      "        See Also\n",
      "        --------\n",
      "        config_context : Context manager for global scikit-learn configuration.\n",
      "        get_config : Retrieve current values of the global configuration.\n",
      "    \n",
      "    show_versions()\n",
      "        Print useful debugging information\"\n",
      "        \n",
      "        .. versionadded:: 0.20\n",
      "\n",
      "DATA\n",
      "    __SKLEARN_SETUP__ = False\n",
      "    __all__ = ['calibration', 'cluster', 'covariance', 'cross_decompositio...\n",
      "\n",
      "VERSION\n",
      "    1.0\n",
      "\n",
      "FILE\n",
      "    /home/ragou/.local/lib/python3.8/site-packages/sklearn/__init__.py\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "help(sklearn)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Test sur les training data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "import sklearn.model_selection\n",
    "train_x,test_x, train_y,test_y = sklearn.model_selection.train_test_split(full_x_data,full_y_data, train_size = 0.7)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "x_train :  (2321, 2)\n",
      "x_test :  (995, 2)\n",
      "y_train :  (2321, 3)\n",
      "y_test :  (995, 3)\n"
     ]
    }
   ],
   "source": [
    "print(\"x_train : \",train_x.shape)\n",
    "print(\"x_test : \",test_x.shape)\n",
    "print(\"y_train : \",train_y.shape)\n",
    "print(\"y_test : \",test_y.shape)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Text</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3114</th>\n",
       "      <td>the pathogenetically critical sequences seem ...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181</th>\n",
       "      <td>the pip3 catalytic activity of the resulting ...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>914</th>\n",
       "      <td>however, most isoforms with a substitution in...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2156</th>\n",
       "      <td>this prompted us to examine additional inheri...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1902</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2804</th>\n",
       "      <td>the presence of the mutations was confirmed b...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>711</th>\n",
       "      <td>the effect of trastuzumab and neratinib on so...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>the smad4 point mutations were amplified with...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>in the br21 clinical trial, there were no g71...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2321 rows  2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   Text  Score\n",
       "ID                                                            \n",
       "3114   the pathogenetically critical sequences seem ...      4\n",
       "2181   the pip3 catalytic activity of the resulting ...      1\n",
       "914    however, most isoforms with a substitution in...      1\n",
       "2156   this prompted us to examine additional inheri...      1\n",
       "1902                                                 []      0\n",
       "...                                                 ...    ...\n",
       "2804   the presence of the mutations was confirmed b...      1\n",
       "145                                                  []      0\n",
       "711    the effect of trastuzumab and neratinib on so...      1\n",
       "602    the smad4 point mutations were amplified with...      1\n",
       "216    in the br21 clinical trial, there were no g71...      1\n",
       "\n",
       "[2321 rows x 2 columns]"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Gene</th>\n",
       "      <th>Variation</th>\n",
       "      <th>Class</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>ID</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3114</th>\n",
       "      <td>RAD51B</td>\n",
       "      <td>hmga2-rad51b fusion</td>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2181</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>t131a</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>914</th>\n",
       "      <td>PDGFRA</td>\n",
       "      <td>d846y</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2156</th>\n",
       "      <td>PTEN</td>\n",
       "      <td>h118p</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1902</th>\n",
       "      <td>TET1</td>\n",
       "      <td>truncating mutations</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2804</th>\n",
       "      <td>BRCA2</td>\n",
       "      <td>r2659t</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>k467t</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>711</th>\n",
       "      <td>ERBB2</td>\n",
       "      <td>l866m</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>SMAD4</td>\n",
       "      <td>e330k</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>216</th>\n",
       "      <td>EGFR</td>\n",
       "      <td>l861p</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2321 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Gene             Variation  Class\n",
       "ID                                       \n",
       "3114  RAD51B   hmga2-rad51b fusion      8\n",
       "2181    PTEN                 t131a      4\n",
       "914   PDGFRA                 d846y      7\n",
       "2156    PTEN                 h118p      4\n",
       "1902    TET1  truncating mutations      1\n",
       "...      ...                   ...    ...\n",
       "2804   BRCA2                r2659t      1\n",
       "145     EGFR                 k467t      2\n",
       "711    ERBB2                 l866m      7\n",
       "602    SMAD4                 e330k      4\n",
       "216     EGFR                 l861p      2\n",
       "\n",
       "[2321 rows x 3 columns]"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "train_y"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>Text</th>\n",
       "      <th>Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3114</td>\n",
       "      <td>the pathogenetically critical sequences seem ...</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2181</td>\n",
       "      <td>the pip3 catalytic activity of the resulting ...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>914</td>\n",
       "      <td>however, most isoforms with a substitution in...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2156</td>\n",
       "      <td>this prompted us to examine additional inheri...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1902</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2316</th>\n",
       "      <td>2804</td>\n",
       "      <td>the presence of the mutations was confirmed b...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2317</th>\n",
       "      <td>145</td>\n",
       "      <td>[]</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2318</th>\n",
       "      <td>711</td>\n",
       "      <td>the effect of trastuzumab and neratinib on so...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2319</th>\n",
       "      <td>602</td>\n",
       "      <td>the smad4 point mutations were amplified with...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2320</th>\n",
       "      <td>216</td>\n",
       "      <td>in the br21 clinical trial, there were no g71...</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>2321 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        ID                                               Text  Score\n",
       "0     3114   the pathogenetically critical sequences seem ...      4\n",
       "1     2181   the pip3 catalytic activity of the resulting ...      1\n",
       "2      914   however, most isoforms with a substitution in...      1\n",
       "3     2156   this prompted us to examine additional inheri...      1\n",
       "4     1902                                                 []      0\n",
       "...    ...                                                ...    ...\n",
       "2316  2804   the presence of the mutations was confirmed b...      1\n",
       "2317   145                                                 []      0\n",
       "2318   711   the effect of trastuzumab and neratinib on so...      1\n",
       "2319   602   the smad4 point mutations were amplified with...      1\n",
       "2320   216   in the br21 clinical trial, there were no g71...      1\n",
       "\n",
       "[2321 rows x 3 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dtf = pd.merge(pd.DataFrame(train_x.index), train_x, on =\"ID\")\n",
    "dtf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "dtf = pd.merge(pd.DataFrame(train_x.index), train_x, on =\"ID\")\n",
    "dtf\n",
    "np.savetxt(\"../datas/x_train.txt\",dtf, fmt = \"%d || %s || %d\", header= \",\".join(dtf.columns), comments='')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "metadata": {},
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "dtf = pd.merge(pd.DataFrame(test_x.index), test_x, on =\"ID\")\n",
    "dtf\n",
    "np.savetxt(\"../datas/x_test.txt\",dtf, fmt = \"%d || %s || %d\", header= \",\".join(dtf.columns), comments='')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": [
    "train_y.to_csv(\"y_train.txt\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "test_y.to_csv(\"../datas/y_test.txt\")"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "deep_projet",
   "language": "python",
   "name": "deep_projet"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
